--- a/ospf6d/ospf6_abr.c	2018-10-08 20:32:57.000000000 +0800
+++ b/ospf6d/ospf6_abr.c	2018-10-20 01:00:36.479411021 +0800
@@ -1209,8 +1209,8 @@
 		/* For Inter-Prefix route: Update RIB/FIB,
 		 * For Inter-Router trigger summary update
 		 */
-		if (table->hook_add)
-			(*table->hook_add)(old_route);
+		//if (table->hook_add)
+		//	(*table->hook_add)(old_route);
 
 		/* Delete new route */
 		ospf6_route_delete(route);
