diff --git a/drivers/media/i2c/ov01a1s.c b/drivers/media/i2c/ov01a1s.c
index 9b537ab63..1377fe26d 100644
--- a/drivers/media/i2c/ov01a1s.c
+++ b/drivers/media/i2c/ov01a1s.c
@@ -961,15 +961,18 @@ static int ov01a1s_probe(struct i2c_client *client)
 {
 	struct ov01a1s *ov01a1s;
 	int ret = 0;
+#if IS_ENABLED(CONFIG_INTEL_VSC)
 	struct vsc_mipi_config conf;
 	struct vsc_camera_status status;
 	s64 link_freq;
+#endif
 
 	ov01a1s = devm_kzalloc(&client->dev, sizeof(*ov01a1s), GFP_KERNEL);
 	if (!ov01a1s)
 		return -ENOMEM;
 	ov01a1s->client = client;
 
+#if IS_ENABLED(CONFIG_INTEL_VSC)
 	conf.lane_num = OV01A1S_DATA_LANES;
 	/* frequency unit 100k */
 	conf.freq = OV01A1S_LINK_FREQ_400MHZ / 100000;
@@ -985,6 +988,7 @@ static int ov01a1s_probe(struct i2c_client *client)
 		return -EPROBE_DEFER;
 	else if (ret)
 		return ret;
+#endif
 
 	ov01a1s_set_power(ov01a1s, 1);
 
