diff --git a/42_bieaz b/42_bieaz
index 98f303c..967c719 100755
--- a/42_bieaz
+++ b/42_bieaz
@@ -199,7 +199,7 @@ while [ "x$list" != "x" ] ; do
   gettext_printf "Found linux image: %s\n" "$linux" >&2
   basename=`basename $linux`
   dirname=`dirname $linux`
-  version=`echo $basename | sed -e "s,^[^0-9]*-,,g"`
+  version=`echo $basename | sed -e "s,vmlinuz-,,g"`
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
 
