diff -pNaru5 src.old/le-1.16.4/src/color.cc src/le-1.16.4/src/color.cc
--- src.old/le-1.16.4/src/color.cc	2017-07-20 11:35:20.000000000 -0400
+++ src/le-1.16.4/src/color.cc	2017-07-29 14:32:39.757342181 -0400
@@ -31,11 +31,11 @@ struct attr attr_table[MAX_COLOR_NO];
 int   attr_num;
 
 int   can_use_default_colors;
 
 int   next_pair;
-static int find_pair(int fg,int bg)
+int find_pair(int fg,int bg)
 {
    if(!can_use_default_colors)
    {
       if(fg==NO_COLOR)
 	 fg=COLOR_WHITE;
diff -pNaru5 src.old/le-1.16.4/src/keynames.cc src/le-1.16.4/src/keynames.cc
--- src.old/le-1.16.4/src/keynames.cc	2017-07-20 11:35:20.000000000 -0400
+++ src/le-1.16.4/src/keynames.cc	2017-07-29 14:32:39.757342181 -0400
@@ -351,11 +351,11 @@ static void MakeCodeNameTableExt()
 #define EXT_NAMES(tp,i,limit,index,table) ((i >= limit) ? tp->ext_Names[index] : table[i])
 #define ExtStrname(tp,i,names)  EXT_NAMES(tp, i, STRCOUNT, (i - (tp->num_Strings - tp->ext_Strings)) + (tp->ext_Numbers + tp->ext_Booleans), names)
 
    if(!cur_term)
       return;
-   TERMTYPE *tp = &cur_term->type;
+   TERMTYPE *tp = (TERMTYPE *)(&cur_term);
    if(!tp)
       return;
    if(NUM_STRINGS(tp)<=STRCOUNT)
       return;
    CodeNameTableExt=(CodeName*)malloc(sizeof(CodeName)*(NUM_STRINGS(tp)-STRCOUNT));
