From 7796485f7fd696f9e6b78b77eee65fc142be16f0 Mon Sep 17 00:00:00 2001
From: Mateusz Gozdek <mgozdekof@gmail.com>
Date: Thu, 28 May 2020 09:32:03 +0200
Subject: [PATCH] Use shared version of libjpeg-turbo

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>
---
 linux/Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git linux/Makefile linux/Makefile
index 0072361..9ccf019 100644
--- linux/Makefile
+++ linux/Makefile
@@ -6,15 +6,11 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 # Use at your own risk. See README file for more details.
 
-JPEG_DIR ?= /opt/libjpeg-turbo
-JPEG_INCLUDE ?= $(JPEG_DIR)/include
-JPEG_LIB ?= $(JPEG_DIR)/lib`getconf LONG_BIT`
-
 CC    = -no-pie
 GTK   = `pkg-config --libs --cflags gtk+-2.0` `pkg-config --libs x11`
 LIBAV = `pkg-config --libs --cflags libswscale libavutil`
 LIBS  =  -lspeex -lasound -lpthread -lm
-JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.a
+JPEG  = `pkg-config --libs --cflags libturbojpeg`
 SRC      = src/connection.c src/settings.c src/decoder*.c src/av.c src/adb.c
 NO_WARN  = -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
 
-- 
2.26.2

