diff --git a/lucidglyph.sh b/lucidglyph.sh
index 53cd571..62ef974 100755
--- a/lucidglyph.sh
+++ b/lucidglyph.sh
@@ -58,7 +58,7 @@ M_DEST_UNINSTALL_FILE="uninstaller.sh"
 DISABLE_ENVIRONMENT=${DISABLE_ENVIRONMENT:-} # TODO: Remove in 1.0.0
 
 ENVIRONMENT_DIR="$MODULES_DIR/environment"
-DEST_ENVIRONMENT="$DEST_CONF/environment"
+DEST_ENVIRONMENT="${DEST_ENVIRONMENT:-$DEST_CONF/environment}"
 
 # Fontconfig group
 DISABLE_FONTCONFIG=${DISABLE_FONTCONFIG:-} # TODO: Remove in 1.0.0
