From c70f3a7973eef99c6709f11be75d7fe1bdc29efd Mon Sep 17 00:00:00 2001
From: Jan Cholasta <jan@cholasta.net>
Date: Mon, 5 Mar 2018 16:17:24 +0100
Subject: [PATCH] Fix soundfont search path

---
 src/gameconfigfile.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gameconfigfile.cpp b/src/gameconfigfile.cpp
index 2979420f9..c8f22895c 100644
--- a/src/gameconfigfile.cpp
+++ b/src/gameconfigfile.cpp
@@ -149,6 +149,8 @@ FGameConfigFile::FGameConfigFile ()
 #else
 		SetValueForKey("Path", "$HOME/" GAME_DIR "/soundfonts", true);
 		SetValueForKey("Path", "$HOME/" GAME_DIR "/fm_banks", true);
+		SetValueForKey("Path", SHARE_DIR "/soundfonts", true);
+		SetValueForKey("Path", SHARE_DIR "/fm_banks", true);
 		SetValueForKey("Path", "/usr/local/share/doom/soundfonts", true);
 		SetValueForKey("Path", "/usr/local/share/doom/fm_banks", true);
 		SetValueForKey("Path", "/usr/local/share/games/doom/soundfonts", true);
@@ -157,6 +159,7 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey("Path", "/usr/share/doom/fm_banks", true);
 		SetValueForKey("Path", "/usr/share/games/doom/soundfonts", true);
 		SetValueForKey("Path", "/usr/share/games/doom/fm_banks", true);
+		SetValueForKey("Path", "/usr/share/soundfonts", true);
 #endif
 	}
 
-- 
2.19.0

