diff --git a/metadata.json b/metadata.json
index e4c9691..c5d8a70 100644
--- a/metadata.json
+++ b/metadata.json
@@ -3,8 +3,8 @@
   "name": "TopIcons Plus",
   "gettext-domain": "TopIcons-Plus",
   "shell-version": [
-    "3.34",
-    "3.36"
+    "40",
+    "41"
   ],
   "settings-schema": "org.gnome.shell.extensions.topicons",
   "url": "https://github.com/phocean/TopIcons-plus",
diff --git a/prefs.js b/prefs.js
index bd372d2..f6b1277 100644
--- a/prefs.js
+++ b/prefs.js
@@ -12,7 +12,7 @@ const Gettext = imports.gettext.domain(Me.metadata['gettext-domain']);
 const _ = Gettext.gettext;
 
 function init() {
-    Convenience.initTranslations();
+    ExtensionUtils.initTranslations();
 }
 
 const TopIconsPlusSettings = new GObject.Class({
@@ -180,8 +180,6 @@ const TopIconsPlusSettings = new GObject.Class({
 });
 
 function buildPrefsWidget() {
-     let widget = new TopIconsPlusSettings();
-     widget.show_all();
+     return new TopIconsPlusSettings();
 
-     return widget;
 }
