Index: source/d_findiwads.cpp
===================================================================
--- source/d_findiwads.cpp	(revision 2592)
+++ source/d_findiwads.cpp	(revision 2593)
@@ -425,6 +425,7 @@
    // Default Linux locations
    paths.addNew() = "/usr/local/share/games/doom";
    paths.addNew() = "/usr/share/games/doom";
+   paths.addNew() = "/usr/share/doom";
 #endif
 
    // add base/game paths
