From b9f79a3af9b21e72eb0064119e5b63adbf6a6e74 Mon Sep 17 00:00:00 2001
From: Mike Swanson <mikeonthecomputer@gmail.com>
Date: Thu, 6 Apr 2023 09:58:03 -0700
Subject: [PATCH] manual: build with asciidoc

---
 manual/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/manual/Makefile b/manual/Makefile
index 0b285d90..db67e568 100644
--- a/manual/Makefile
+++ b/manual/Makefile
@@ -1,7 +1,8 @@
 all: freedoom-manual.pdf
 
 freedoom-manual.pdf: manual.adoc
-	-asciidoctor-pdf -a toc manual.adoc -o $@
+	a2x -f pdf -a toc manual.adoc
+	mv manual.pdf freedoom-manual.pdf
 
 clean:
 	$(RM) freedoom-manual.pdf
-- 
2.40.0

