--- a/qtbase/mkspecs/android-clang/qmake.conf	2019-02-07 11:03:08.250521061 -0300
+++ b/qtbase/mkspecs/android-clang/qmake.conf	2019-02-07 11:10:50.389466359 -0300
@@ -34,11 +34,12 @@
 QMAKE_LINK    = $$QMAKE_CXX $$QMAKE_CFLAGS -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libatomic.a
 equals(ANDROID_TARGET_ARCH, armeabi-v7a): QMAKE_LINK += -Wl,--exclude-libs,libunwind.a

-QMAKE_CFLAGS += -DANDROID_HAS_WSTRING --sysroot=$$NDK_ROOT/sysroot \
+QMAKE_CFLAGS += -DANDROID_HAS_WSTRING --sysroot=$$ANDROID_PLATFORM_ROOT_PATH \
                 -isystem $$NDK_ROOT/sysroot/usr/include/$$NDK_TOOLS_PREFIX \
                 -isystem $$NDK_ROOT/sources/cxx-stl/llvm-libc++/include \
                 -isystem $$NDK_ROOT/sources/android/support/include \
-                -isystem $$NDK_ROOT/sources/cxx-stl/llvm-libc++abi/include
+                -isystem $$NDK_ROOT/sources/cxx-stl/llvm-libc++abi/include \
+                -isystem $$NDK_ROOT/sysroot/usr/include

 ANDROID_SOURCES_CXX_STL_LIBDIR = $$NDK_ROOT/sources/cxx-stl/llvm-libc++/libs/$$ANDROID_TARGET_ARCH

@@ -54,3 +54,6 @@
 QMAKE_CFLAGS_OPTIMIZE_SIZE = -Oz

 include(../common/android-base-tail.conf)
+
+QMAKE_PKG_CONFIG = $$[QT_INSTALL_PREFIX]
+QMAKE_PKG_CONFIG = android-$$section(QMAKE_PKG_CONFIG, /, 3, 3)-pkg-config
