--- a/src/hu_lib.h
+++ b/src/hu_lib.h
@@ -118,7 +118,7 @@ void HUlib_DrawSText(hu_stext_t *s, dboolean external);
 // erases all stext lines
 void HUlib_EraseSText(hu_stext_t *s);
 
-void (*althudtextfunc)(int, int, byte *, patch_t *, int);
+extern void (*althudtextfunc)(int, int, byte *, patch_t *, int);
 
 extern dboolean STCFN034;
 extern dboolean idbehold;

--- a/src/i_video.h
+++ b/src/i_video.h
@@ -86,8 +86,8 @@ void I_SetGamma(float value);
 void I_WindowResizeBlit(void);
 #endif
 
-void (*blitfunc)(void);
-void (*mapblitfunc)(void);
+extern void (*blitfunc)(void);
+extern void (*mapblitfunc)(void);
 
 extern dboolean     sendpause;
 extern dboolean     quitting;

--- a/src/r_main.h
+++ b/src/r_main.h
@@ -101,36 +101,36 @@ extern fixed_t      fractionaltic;
 // Function pointers to switch refresh/drawing functions.
 // Used to select shadow mode etc.
 //
-void (*colfunc)(void);
-void (*wallcolfunc)(void);
-void (*bmapwallcolfunc)(void);
-void (*segcolfunc)(void);
-void (*transcolfunc)(void);
-void (*basecolfunc)(void);
-void (*fuzzcolfunc)(void);
-void (*tlcolfunc)(void);
-void (*tl50colfunc)(void);
-void (*tl50segcolfunc)(void);
-void (*tl33colfunc)(void);
-void (*tlgreencolfunc)(void);
-void (*tlredcolfunc)(void);
-void (*tlredwhitecolfunc1)(void);
-void (*tlredwhitecolfunc2)(void);
-void (*tlredwhite50colfunc)(void);
-void (*tlbluecolfunc)(void);
-void (*tlgreen33colfunc)(void);
-void (*tlred33colfunc)(void);
-void (*tlblue25colfunc)(void);
-void (*redtobluecolfunc)(void);
-void (*tlredtoblue33colfunc)(void);
-void (*skycolfunc)(void);
-void (*redtogreencolfunc)(void);
-void (*tlredtogreen33colfunc)(void);
-void (*psprcolfunc)(void);
-void (*spanfunc)(void);
-void (*bloodsplatcolfunc)(void);
-void (*megaspherecolfunc)(void);
-void (*supershotguncolfunc)(void);
+extern void (*colfunc)(void);
+extern void (*wallcolfunc)(void);
+extern void (*bmapwallcolfunc)(void);
+extern void (*segcolfunc)(void);
+extern void (*transcolfunc)(void);
+extern void (*basecolfunc)(void);
+extern void (*fuzzcolfunc)(void);
+extern void (*tlcolfunc)(void);
+extern void (*tl50colfunc)(void);
+extern void (*tl50segcolfunc)(void);
+extern void (*tl33colfunc)(void);
+extern void (*tlgreencolfunc)(void);
+extern void (*tlredcolfunc)(void);
+extern void (*tlredwhitecolfunc1)(void);
+extern void (*tlredwhitecolfunc2)(void);
+extern void (*tlredwhite50colfunc)(void);
+extern void (*tlbluecolfunc)(void);
+extern void (*tlgreen33colfunc)(void);
+extern void (*tlred33colfunc)(void);
+extern void (*tlblue25colfunc)(void);
+extern void (*redtobluecolfunc)(void);
+extern void (*tlredtoblue33colfunc)(void);
+extern void (*skycolfunc)(void);
+extern void (*redtogreencolfunc)(void);
+extern void (*tlredtogreen33colfunc)(void);
+extern void (*psprcolfunc)(void);
+extern void (*spanfunc)(void);
+extern void (*bloodsplatcolfunc)(void);
+extern void (*megaspherecolfunc)(void);
+extern void (*supershotguncolfunc)(void);
 
 //
 // Utility functions.
