From ad64507654c9e7861247a9811f8e3fa737a0333f Mon Sep 17 00:00:00 2001
From: KoffeinFlummi <koffeinflummi@gmail.com>
Date: Fri, 10 Jul 2015 09:35:36 +0200
Subject: [PATCH] Add vim-style navigation bindings

---
 src/bindings.cpp | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/bindings.cpp b/src/bindings.cpp
index 4f3a88e..ce515a6 100644
--- a/src/bindings.cpp
+++ b/src/bindings.cpp
@@ -467,6 +467,22 @@ bool BindingsConfiguration::read(const std::string &file)
 void BindingsConfiguration::generateDefaults()
 {
 	NC::Key::Type k = NC::Key::None;
+	if (notBound(k = stringToKey("h")))
+		bind(k, Actions::Type::PreviousColumn);
+	if (notBound(k = stringToKey("k")))
+		bind(k, Actions::Type::ScrollUp);
+	if (notBound(k = stringToKey("j")))
+		bind(k, Actions::Type::ScrollDown);
+	if (notBound(k = stringToKey("l")))
+		bind(k, Actions::Type::NextColumn);
+	if (notBound(k = stringToKey("ctrl-b")))
+		bind(k, Actions::Type::PageUp);
+	if (notBound(k = stringToKey("ctrl-f")))
+		bind(k, Actions::Type::PageDown);
+	if (notBound(k = stringToKey("ctrl-u")))
+		bind(k, Actions::Type::PageUp);
+	if (notBound(k = stringToKey("ctrl-d")))
+		bind(k, Actions::Type::PageDown);
 	if (notBound(k = stringToKey("mouse")))
 		bind(k, Actions::Type::MouseEvent);
 	if (notBound(k = stringToKey("up")))
@@ -649,7 +665,7 @@ void BindingsConfiguration::generateDefaults()
 		bind(k, Actions::Type::ShowArtistInfo);
 	if (notBound(k = stringToKey("g")))
 		bind(k, Actions::Type::JumpToPositionInSong);
-	if (notBound(k = stringToKey("l")))
+	if (notBound(k = stringToKey("L")))
 		bind(k, Actions::Type::ShowLyrics);
 	if (notBound(k = stringToKey("ctrl-v")))
 		bind(k, Actions::Type::SelectRange);
-- 
2.5.0

