diff --unified --recursive --text a/thirdparty/discord-rpc/rapidjson/include/rapidjson/document.h b/thirdparty/discord-rpc/rapidjson/include/rapidjson/document.h
--- a/thirdparty/discord-rpc/rapidjson/include/rapidjson/document.h	2024-05-05 23:48:30.000000000 -0400
+++ b/thirdparty/discord-rpc/rapidjson/include/rapidjson/document.h	2024-05-12 21:43:50.237508798 -0400
@@ -316,8 +316,6 @@
 
     GenericStringRef(const GenericStringRef& rhs) : s(rhs.s), length(rhs.length) {}
 
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
     //! implicit conversion to plain CharType pointer
     operator const Ch *() const { return s; }
 
@@ -328,6 +326,8 @@
     //! Disallow construction from non-const array
     template<SizeType N>
     GenericStringRef(CharType (&str)[N]) /* = delete */;
+    //! Copy assignment operator not permitted - immutable type
+    GenericStringRef& operator=(const GenericStringRef& rhs) /* = delete */;
 };
 
 //! Mark a character pointer as constant string
