diff -Naur envoy-1.26.2.a/.bazelrc envoy-1.26.2.b/.bazelrc
--- envoy-1.26.2.a/.bazelrc	2023-06-05 04:42:11.000000000 -0400
+++ envoy-1.26.2.b/.bazelrc	2023-06-26 18:38:28.905340126 -0400
@@ -58,7 +58,7 @@
 build:linux --fission=dbg,opt
 build:linux --features=per_object_debug_info
 build:linux --action_env=BAZEL_LINKLIBS=-l%:libstdc++.a
-build:linux --action_env=BAZEL_LINKOPTS=-lm
+build:linux --action_env=BAZEL_LINKOPTS=-lm:-lpsl
 
 # TODO(keith): remove once https://github.com/DataDog/dd-opentracing-cpp/pull/252 is integrated
 # this avoids warnings/errors on arm64 Linux builds
@@ -170,7 +170,7 @@
 build:libc++ --action_env=LDFLAGS=-stdlib=libc++
 build:libc++ --action_env=BAZEL_CXXOPTS=-stdlib=libc++
 build:libc++ --action_env=BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a
-build:libc++ --action_env=BAZEL_LINKOPTS=-lm:-pthread
+build:libc++ --action_env=BAZEL_LINKOPTS=-lm:-pthread:-lpsl
 build:libc++ --define force_libcpp=enabled
 
 # Optimize build for binary size reduction.
