From 82f9e9e520536c8c73583e04bbb608e44185a0fa Mon Sep 17 00:00:00 2001
From: "j.r" <j.r@jugendhacker.de>
Date: Thu, 22 Oct 2020 18:13:54 +0200
Subject: [PATCH] Add ldflags to Makefile

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 71c574a..ae01955 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 .POSIX:
 all:
-	$(CC) -o among-sus $(CFLAGS) main.c
+	$(CC) -o among-sus $(CFLAGS) main.c $(LDFLAGS)
 
 clean:
 	rm -f among-sus
-- 
2.29.0

