From 9f0da4fe1696e82ec74d6ea8e730dbac52be9a7b Mon Sep 17 00:00:00 2001
From: Solomon Choina <shlomochoina@gmail.com>
Date: Sun, 4 Aug 2019 19:24:18 -0400
Subject: [PATCH] Wrap 3.3, delete fail and nats

---
 scotty.cabal | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/scotty.cabal b/scotty.cabal
index 57e9879..7d025f7 100644
--- a/scotty.cabal
+++ b/scotty.cabal
@@ -82,11 +82,9 @@ Library
                        case-insensitive    >= 1.0.0.1  && < 1.3,
                        data-default-class  >= 0.0.1    && < 0.2,
                        exceptions          >= 0.7      && < 0.11,
-                       fail,
                        http-types          >= 0.8.2    && < 0.13,
                        monad-control       >= 1.0.0.3  && < 1.1,
                        mtl                 >= 2.1.2    && < 2.3,
-                       nats                >= 0.1      && < 2,
                        network             >= 2.6.0.2  && < 3.2,
                        regex-compat        >= 0.95.1   && < 0.96,
                        text                >= 0.11.3.1 && < 1.3,
@@ -95,7 +93,7 @@ Library
                        transformers-compat >= 0.4      && < 0.7,
                        wai                 >= 3.0.0    && < 3.3,
                        wai-extra           >= 3.0.0    && < 3.1,
-                       warp                >= 3.0.13   && < 3.3
+                       warp                >= 3.0.13   && < 3.4
 
   GHC-options: -Wall -fno-warn-orphans
 
-- 
2.22.0

