--- a/absl/algorithm/CMakeLists.txt
+++ b/absl/algorithm/CMakeLists.txt
@@ -26,19 +26,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    algorithm_test
-  SRCS
-    "algorithm_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::algorithm
-    absl::config
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     algorithm_container
@@ -53,19 +40,3 @@
     absl::nullability
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    container_test
-  SRCS
-    "container_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::algorithm_container
-    absl::base
-    absl::core_headers
-    absl::memory
-    absl::span
-    GTest::gmock_main
-)
--- a/absl/base/CMakeLists.txt
+++ b/absl/base/CMakeLists.txt
@@ -80,19 +80,6 @@
     ${ABSL_DEFAULT_COPTS}
 )
 
-absl_cc_test(
-  NAME
-    nullability_test
-  SRCS
-    "nullability_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::core_headers
-    absl::nullability
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -273,20 +260,6 @@
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
-    exception_testing
-  HDRS
-    "internal/exception_testing.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  DEPS
-    absl::config
-    GTest::gtest
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
     pretty_function
   HDRS
     "internal/pretty_function.h"
@@ -297,40 +270,6 @@
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
-    exception_safety_testing
-  HDRS
-    "internal/exception_safety_testing.h"
-  SRCS
-    "internal/exception_safety_testing.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::pretty_function
-    absl::memory
-    absl::meta
-    absl::strings
-    absl::utility
-    GTest::gtest
-  TESTONLY
-)
-
-absl_cc_test(
-  NAME
-    absl_exception_safety_testing_test
-  SRCS
-    "exception_safety_testing_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::exception_safety_testing
-    absl::memory
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
     atomic_hook_test_helper
   SRCS
     "internal/atomic_hook_test_helper.cc"
@@ -342,127 +281,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    atomic_hook_test
-  SRCS
-    "internal/atomic_hook_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::atomic_hook_test_helper
-    absl::atomic_hook
-    absl::core_headers
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    bit_cast_test
-  SRCS
-    "bit_cast_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::core_headers
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    errno_saver_test
-  SRCS
-    "internal/errno_saver_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::errno_saver
-    absl::strerror
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    throw_delegate_test
-  SRCS
-    "throw_delegate_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::throw_delegate
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    inline_variable_test
-  SRCS
-    "internal/inline_variable_testing.h"
-    "inline_variable_test.cc"
-    "inline_variable_test_a.cc"
-    "inline_variable_test_b.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base_internal
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    invoke_test
-  SRCS
-    "invoke_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base_internal
-    absl::memory
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    spinlock_test_common
-  SRCS
-    "spinlock_test_common.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::base_internal
-    absl::core_headers
-    absl::synchronization
-    GTest::gtest
-  TESTONLY
-)
-
-# On bazel BUILD this target use "alwayslink = 1" which is not implemented here
-absl_cc_test(
-  NAME
-    spinlock_test
-  SRCS
-    "spinlock_test_common.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::base_internal
-    absl::config
-    absl::core_headers
-    absl::synchronization
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -482,87 +300,6 @@
 
 absl_cc_test(
   NAME
-    endian_test
-  SRCS
-    "internal/endian_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::endian
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    config_test
-  SRCS
-    "config_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::synchronization
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    call_once_test
-  SRCS
-    "call_once_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::core_headers
-    absl::synchronization
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    no_destructor_test
-  SRCS
-    "no_destructor_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::no_destructor
-    absl::config
-    absl::raw_logging_internal
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    raw_logging_test
-  SRCS
-    "raw_logging_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    sysinfo_test
-  SRCS
-    "internal/sysinfo_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::synchronization
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
     low_level_alloc_test
   SRCS
     "internal/low_level_alloc_test.cc"
@@ -574,21 +311,6 @@
     Threads::Threads
 )
 
-absl_cc_test(
-  NAME
-    thread_identity_test
-  SRCS
-    "internal/thread_identity_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::core_headers
-    absl::synchronization
-    Threads::Threads
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -606,18 +328,6 @@
 
 absl_cc_test(
   NAME
-    scoped_set_env_test
-  SRCS
-    "internal/scoped_set_env_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::scoped_set_env
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
     cmake_thread_test
   SRCS
     "internal/cmake_thread_test.cc"
@@ -627,20 +337,6 @@
     absl::base
 )
 
-absl_cc_test(
-  NAME
-    log_severity_test
-  SRCS
-    "log_severity_test.cc"
-  DEPS
-    absl::flags_internal
-    absl::flags_marshalling
-    absl::log_severity
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -659,20 +355,6 @@
     absl::errno_saver
 )
 
-absl_cc_test(
-  NAME
-    strerror_test
-  SRCS
-    "internal/strerror_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strerror
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -687,18 +369,6 @@
     absl::config
 )
 
-absl_cc_test(
-  NAME
-    fast_type_id_test
-  SRCS
-    "internal/fast_type_id_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::fast_type_id
-    GTest::gtest_main
-)
-
 absl_cc_library(
   NAME
     prefetch
@@ -712,28 +382,3 @@
     absl::config
     absl::core_headers
 )
-
-absl_cc_test(
-  NAME
-    prefetch_test
-  SRCS
-    "prefetch_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::prefetch
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    optimization_test
-  SRCS
-    "optimization_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::core_headers
-    absl::optional
-    GTest::gtest_main
-)
--- a/absl/cleanup/CMakeLists.txt
+++ b/absl/cleanup/CMakeLists.txt
@@ -40,17 +40,3 @@
     absl::core_headers
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    cleanup_test
-  SRCS
-    "cleanup_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::cleanup
-    absl::config
-    absl::utility
-    GTest::gmock_main
-)
--- a/absl/container/CMakeLists.txt
+++ b/absl/container/CMakeLists.txt
@@ -62,32 +62,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    btree_test
-  SRCS
-    "btree_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::algorithm_container
-    absl::btree
-    absl::btree_test_common
-    absl::compare
-    absl::core_headers
-    absl::flags
-    absl::hash_testing
-    absl::optional
-    absl::random_random
-    absl::raw_logging_internal
-    absl::strings
-    absl::test_allocator
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -101,23 +75,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    compressed_tuple_test
-  SRCS
-    "internal/compressed_tuple_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any
-    absl::compressed_tuple
-    absl::memory
-    absl::optional
-    absl::test_instance_tracker
-    absl::utility
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     fixed_array
@@ -136,37 +93,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    fixed_array_test
-  SRCS
-    "fixed_array_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::fixed_array
-    absl::config
-    absl::exception_testing
-    absl::hash_testing
-    absl::memory
-    absl::test_allocator
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    fixed_array_exception_safety_test
-  SRCS
-    "fixed_array_exception_safety_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::fixed_array
-    absl::config
-    absl::exception_safety_testing
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -205,54 +131,6 @@
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
-    test_allocator
-  HDRS
-    "internal/test_allocator.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  DEPS
-    absl::config
-    GTest::gmock
-)
-
-absl_cc_test(
-  NAME
-    inlined_vector_test
-  SRCS
-    "inlined_vector_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::check
-    absl::config
-    absl::core_headers
-    absl::exception_testing
-    absl::hash_testing
-    absl::inlined_vector
-    absl::memory
-    absl::strings
-    absl::test_allocator
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    inlined_vector_exception_safety_test
-  SRCS
-    "inlined_vector_exception_safety_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::inlined_vector
-    absl::config
-    absl::exception_safety_testing
-    GTest::gmock_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
     test_instance_tracker
   HDRS
     "internal/test_instance_tracker.h"
@@ -265,18 +143,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    test_instance_tracker_test
-  SRCS
-    "internal/test_instance_tracker_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     flat_hash_map
@@ -294,26 +160,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    flat_hash_map_test
-  SRCS
-    "flat_hash_map_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any
-    absl::check
-    absl::flat_hash_map
-    absl::hash_generator_testing
-    absl::type_traits
-    absl::unordered_map_constructor_test
-    absl::unordered_map_lookup_test
-    absl::unordered_map_members_test
-    absl::unordered_map_modifiers_test
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     flat_hash_set
@@ -331,29 +177,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    flat_hash_set_test
-  SRCS
-    "flat_hash_set_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-    "-DUNORDERED_SET_CXX17"
-  DEPS
-    absl::check
-    absl::config
-    absl::container_memory
-    absl::flat_hash_set
-    absl::hash_generator_testing
-    absl::memory
-    absl::strings
-    absl::unordered_set_constructor_test
-    absl::unordered_set_lookup_test
-    absl::unordered_set_members_test
-    absl::unordered_set_modifiers_test
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     node_hash_map
@@ -372,24 +195,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    node_hash_map_test
-  SRCS
-    "node_hash_map_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::node_hash_map
-    absl::tracked
-    absl::unordered_map_constructor_test
-    absl::unordered_map_lookup_test
-    absl::unordered_map_members_test
-    absl::unordered_map_modifiers_test
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     node_hash_set
@@ -407,24 +212,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    node_hash_set_test
-  SRCS
-    "node_hash_set_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-    "-DUNORDERED_SET_CXX17"
-  DEPS
-    absl::hash_generator_testing
-    absl::node_hash_set
-    absl::unordered_set_constructor_test
-    absl::unordered_set_lookup_test
-    absl::unordered_set_members_test
-    absl::unordered_set_modifiers_test
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -441,22 +228,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    container_memory_test
-  SRCS
-    "internal/container_memory_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::container_memory
-    absl::no_destructor
-    absl::strings
-    absl::test_instance_tracker
-    absl::type_traits
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -473,23 +244,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    hash_function_defaults_test
-  SRCS
-    "internal/hash_function_defaults_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::cord
-    absl::cord_test_helpers
-    absl::hash_function_defaults
-    absl::hash
-    absl::random_random
-    absl::strings
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -523,18 +277,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    hash_policy_testing_test
-  SRCS
-    "internal/hash_policy_testing_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_policy_testing
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -549,18 +291,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    hash_policy_traits_test
-  SRCS
-    "internal/hash_policy_traits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_policy_traits
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -574,19 +304,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    common_policy_traits_test
-  SRCS
-    "internal/common_policy_traits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::common_policy_traits
-    absl::config
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -608,19 +325,6 @@
     absl::time
 )
 
-absl_cc_test(
-  NAME
-    hashtablez_sampler_test
-  SRCS
-    "internal/hashtablez_sampler_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::hashtablez_sampler
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -659,20 +363,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    node_slot_policy_test
-  SRCS
-    "internal/node_slot_policy_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::hash_policy_traits
-    absl::node_slot_policy
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -734,49 +424,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    raw_hash_set_test
-  SRCS
-    "internal/raw_hash_set_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::container_memory
-    absl::core_headers
-    absl::flat_hash_map
-    absl::flat_hash_set
-    absl::hash
-    absl::hash_function_defaults
-    absl::hash_policy_testing
-    absl::hashtable_debug
-    absl::hashtablez_sampler
-    absl::log
-    absl::memory
-    absl::prefetch
-    absl::raw_hash_set
-    absl::strings
-    absl::test_allocator
-    absl::type_traits
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    raw_hash_set_allocator_test
-  SRCS
-    "internal/raw_hash_set_allocator_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::raw_hash_set
-    absl::tracked
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -796,22 +443,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    layout_test
-  SRCS
-    "internal/layout_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::layout
-    absl::check
-    absl::config
-    absl::span
-    absl::utility
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -824,166 +455,3 @@
     absl::config
   TESTONLY
 )
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_map_constructor_test
-  HDRS
-    "internal/unordered_map_constructor_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_map_lookup_test
-  HDRS
-    "internal/unordered_map_lookup_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_map_members_test
-  HDRS
-    "internal/unordered_map_members_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::type_traits
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_map_modifiers_test
-  HDRS
-    "internal/unordered_map_modifiers_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_set_constructor_test
-  HDRS
-    "internal/unordered_set_constructor_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_set_lookup_test
-  HDRS
-    "internal/unordered_set_lookup_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_set_members_test
-  HDRS
-    "internal/unordered_set_members_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::type_traits
-    GTest::gmock
-  TESTONLY
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    unordered_set_modifiers_test
-  HDRS
-    "internal/unordered_set_modifiers_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash_generator_testing
-    absl::hash_policy_testing
-    GTest::gmock
-  TESTONLY
-)
-
-absl_cc_test(
-  NAME
-    unordered_set_test
-  SRCS
-    "internal/unordered_set_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::unordered_set_constructor_test
-    absl::unordered_set_lookup_test
-    absl::unordered_set_members_test
-    absl::unordered_set_modifiers_test
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    unordered_map_test
-  SRCS
-    "internal/unordered_map_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::unordered_map_constructor_test
-    absl::unordered_map_lookup_test
-    absl::unordered_map_members_test
-    absl::unordered_map_modifiers_test
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    sample_element_size_test
-  SRCS
-    "sample_element_size_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flat_hash_map
-    absl::flat_hash_set
-    absl::node_hash_map
-    absl::node_hash_set
-    GTest::gmock_main
-)
--- a/absl/debugging/CMakeLists.txt
+++ b/absl/debugging/CMakeLists.txt
@@ -46,19 +46,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    stacktrace_test
-  SRCS
-    "stacktrace_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::stacktrace
-    absl::core_headers
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     symbolize
@@ -90,29 +77,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    symbolize_test
-  SRCS
-    "symbolize_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-    $<$<BOOL:${MSVC}>:-Z7>
-  LINKOPTS
-    $<$<BOOL:${MSVC}>:-DEBUG>
-  DEPS
-    absl::base
-    absl::check
-    absl::config
-    absl::core_headers
-    absl::log
-    absl::memory
-    absl::stack_consumption
-    absl::strings
-    absl::symbolize
-    GTest::gmock
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -150,24 +114,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    failure_signal_handler_test
-  SRCS
-    "failure_signal_handler_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::check
-    absl::failure_signal_handler
-    absl::stacktrace
-    absl::symbolize
-    absl::strings
-    absl::raw_logging_internal
-    Threads::Threads
-    GTest::gmock
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -206,23 +152,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    demangle_test
-  SRCS
-    "internal/demangle_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::demangle_internal
-    absl::stack_consumption
-    absl::config
-    absl::core_headers
-    absl::log
-    absl::memory
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     leak_check
@@ -238,22 +167,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    leak_check_test
-  SRCS
-    "leak_check_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::leak_check
-    absl::base
-    absl::log
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -271,20 +184,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    stack_consumption_test
-  SRCS
-    "internal/stack_consumption_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::stack_consumption
-    absl::core_headers
-    absl::log
-    GTest::gmock_main
-)
-
 # component target
 absl_cc_library(
   NAME
--- a/absl/flags/CMakeLists.txt
+++ b/absl/flags/CMakeLists.txt
@@ -293,179 +293,3 @@
     absl::strings
     absl::synchronization
 )
-
-############################################################################
-# Unit tests in alphabetical order.
-
-absl_cc_test(
-  NAME
-    flags_commandlineflag_test
-  SRCS
-    "commandlineflag_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags
-    absl::flags_commandlineflag
-    absl::flags_commandlineflag_internal
-    absl::flags_config
-    absl::flags_private_handle_accessor
-    absl::flags_reflection
-    absl::memory
-    absl::strings
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_config_test
-  SRCS
-    "config_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_config
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_flag_test
-  SRCS
-    "flag_test.cc"
-    "flag_test_defs.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::core_headers
-    absl::flags
-    absl::flags_config
-    absl::flags_internal
-    absl::flags_marshalling
-    absl::flags_reflection
-    absl::int128
-    absl::strings
-    absl::time
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_marshalling_test
-  SRCS
-    "marshalling_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_marshalling
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_parse_test
-  SRCS
-    "parse_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags
-    absl::flags_parse
-    absl::flags_reflection
-    absl::flags_usage_internal
-    absl::log
-    absl::scoped_set_env
-    absl::span
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    flags_path_util_test
-  SRCS
-    "internal/path_util_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_path_util
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_program_name_test
-  SRCS
-    "internal/program_name_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_program_name
-    absl::strings
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_reflection_test
-  SRCS
-    "reflection_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_commandlineflag_internal
-    absl::flags
-    absl::flags_reflection
-    absl::flags_usage
-    absl::memory
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    flags_sequence_lock_test
-  SRCS
-    "internal/sequence_lock_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::flags_internal
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    flags_usage_config_test
-  SRCS
-    "usage_config_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_config
-    absl::flags_path_util
-    absl::flags_program_name
-    absl::strings
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    flags_usage_test
-  SRCS
-    "internal/usage_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags_config
-    absl::flags
-    absl::flags_path_util
-    absl::flags_program_name
-    absl::flags_parse
-    absl::flags_reflection
-    absl::flags_usage
-    absl::strings
-    GTest::gmock
-)
--- a/absl/functional/CMakeLists.txt
+++ b/absl/functional/CMakeLists.txt
@@ -32,24 +32,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    any_invocable_test
-  SRCS
-    "any_invocable_test.cc"
-    "internal/any_invocable.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any_invocable
-    absl::base_internal
-    absl::config
-    absl::core_headers
-    absl::type_traits
-    absl::utility
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     bind_front
@@ -65,19 +47,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    bind_front_test
-  SRCS
-    "bind_front_test.cc"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  DEPS
-    absl::bind_front
-    absl::memory
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     function_ref
@@ -95,20 +64,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    function_ref_test
-  SRCS
-    "function_ref_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::function_ref
-    absl::memory
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     overload
@@ -120,15 +75,3 @@
     absl::meta
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    overload_test
-  SRCS
-    "overload_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    GTest::gmock_main
-)
--- a/absl/hash/CMakeLists.txt
+++ b/absl/hash/CMakeLists.txt
@@ -42,71 +42,6 @@
   PUBLIC
 )
 
-absl_cc_library(
-  NAME
-    hash_testing
-  HDRS
-    "hash_testing.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::spy_hash_state
-    absl::meta
-    absl::strings
-    absl::variant
-    GTest::gmock
-  TESTONLY
-  PUBLIC
-)
-
-absl_cc_test(
-  NAME
-    hash_test
-  SRCS
-    "hash_test.cc"
-    "internal/hash_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::btree
-    absl::cord_test_helpers
-    absl::core_headers
-    absl::flat_hash_map
-    absl::flat_hash_set
-    absl::hash
-    absl::hash_testing
-    absl::int128
-    absl::memory
-    absl::meta
-    absl::node_hash_map
-    absl::node_hash_set
-    absl::optional
-    absl::spy_hash_state
-    absl::string_view
-    absl::variant
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    hash_instantiated_test
-  SRCS
-    "hash_instantiated_test.cc"
-    "internal/hash_test.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::hash
-    absl::hash_testing
-    absl::config
-    absl::btree
-    absl::flat_hash_map
-    absl::flat_hash_set
-    absl::node_hash_map
-    absl::node_hash_set
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 #
 # Note: Even though external code should not depend on this target
@@ -143,18 +78,6 @@
     absl::endian
 )
 
-absl_cc_test(
-  NAME
-    city_test
-  SRCS
-    "internal/city_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::city
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -171,16 +94,3 @@
     absl::int128
     absl::prefetch
 )
-
-absl_cc_test(
-  NAME
-    low_level_hash_test
-  SRCS
-    "internal/low_level_hash_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::low_level_hash
-    absl::strings
-    GTest::gmock_main
-)
--- a/absl/log/CMakeLists.txt
+++ b/absl/log/CMakeLists.txt
@@ -318,51 +318,6 @@
 
 absl_cc_library(
   NAME
-    log_internal_test_helpers
-  SRCS
-    "internal/test_helpers.cc"
-  HDRS
-    "internal/test_helpers.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::config
-    absl::log_globals
-    absl::log_initialize
-    absl::log_internal_globals
-    absl::log_severity
-    GTest::gtest
-  TESTONLY
-)
-
-absl_cc_library(
-  NAME
-    log_internal_test_matchers
-  SRCS
-    "internal/test_matchers.cc"
-  HDRS
-    "internal/test_matchers.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::config
-    absl::core_headers
-    absl::log_entry
-    absl::log_internal_test_helpers
-    absl::log_severity
-    absl::strings
-    absl::time
-    GTest::gtest
-    GTest::gmock
-  TESTONLY
-)
-
-absl_cc_library(
-  NAME
     log_internal_voidify
   SRCS
   HDRS
@@ -627,31 +582,6 @@
 
 absl_cc_library(
   NAME
-    scoped_mock_log
-  SRCS
-    "scoped_mock_log.cc"
-  HDRS
-    "scoped_mock_log.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::config
-    absl::log_entry
-    absl::log_severity
-    absl::log_sink
-    absl::log_sink_registry
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gmock
-    GTest::gtest
-  PUBLIC
-  TESTONLY
-)
-
-absl_cc_library(
-  NAME
     log_internal_structured
   HDRS
     "internal/structured.h"
@@ -733,27 +663,6 @@
     absl::absl_vlog_is_on
 )
 
-absl_cc_test(
-  NAME
-    vlog_is_on_test
-  SRCS
-    "vlog_is_on_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::log_flags
-    absl::log_globals
-    absl::scoped_mock_log
-    absl::vlog_is_on
-    absl::log_severity
-    absl::flags
-    absl::optional
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     log_internal_fnmatch
@@ -769,376 +678,3 @@
     absl::config
     absl::strings
 )
-
-# Test targets
-
-absl_cc_test(
-  NAME
-    absl_check_test
-  SRCS
-    "absl_check_test.cc"
-    "check_test_impl.inc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::absl_check
-    absl::config
-    absl::core_headers
-    absl::log_internal_test_helpers
-    absl::status
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    absl_log_basic_test
-  SRCS
-    "log_basic_test.cc"
-    "log_basic_test_impl.inc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::base
-    absl::absl_log
-    absl::log_entry
-    absl::log_globals
-    absl::log_severity
-    absl::log_internal_test_actions
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::scoped_mock_log
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    check_test
-  SRCS
-    "check_test.cc"
-    "check_test_impl.inc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::check
-    absl::config
-    absl::core_headers
-    absl::log_internal_test_helpers
-    absl::status
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    die_if_null_test
-  SRCS
-    "die_if_null_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::die_if_null
-    absl::log_internal_test_helpers
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    log_basic_test
-  SRCS
-    "log_basic_test.cc"
-    "log_basic_test_impl.inc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::base
-    absl::log
-    absl::log_entry
-    absl::log_globals
-    absl::log_severity
-    absl::log_internal_test_actions
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::scoped_mock_log
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_entry_test
-  SRCS
-    "log_entry_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::config
-    absl::core_headers
-    absl::log_entry
-    absl::log_internal_append_truncated
-    absl::log_internal_format
-    absl::log_internal_globals
-    absl::log_internal_test_helpers
-    absl::log_severity
-    absl::span
-    absl::strings
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_flags_test
-  SRCS
-    "flags_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_flags
-    absl::log_globals
-    absl::log_internal_flags
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_severity
-    absl::flags
-    absl::flags_reflection
-    absl::scoped_mock_log
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_globals_test
-  SRCS
-    "globals_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_globals
-    absl::log_internal_globals
-    absl::log_internal_test_helpers
-    absl::log_severity
-    absl::scoped_mock_log
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_format_test
-  SRCS
-    "log_format_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::check
-    absl::log
-    absl::log_internal_test_matchers
-    absl::optional
-    absl::scoped_mock_log
-    absl::str_format
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_macro_hygiene_test
-  SRCS
-    "log_macro_hygiene_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_severity
-    absl::scoped_mock_log
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_sink_test
-  SRCS
-    "log_sink_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_internal_test_actions
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_sink
-    absl::log_sink_registry
-    absl::log_severity
-    absl::scoped_mock_log
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_streamer_test
-  SRCS
-    "log_streamer_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::base
-    absl::core_headers
-    absl::log
-    absl::log_internal_test_actions
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_streamer
-    absl::log_severity
-    absl::scoped_mock_log
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_modifier_methods_test
-  SRCS
-    "log_modifier_methods_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::log_internal_test_actions
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_sink
-    absl::scoped_mock_log
-    absl::strings
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    scoped_mock_log_test
-  SRCS
-    "scoped_mock_log_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_globals
-    absl::log_internal_globals
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_severity
-    absl::memory
-    absl::scoped_mock_log
-    absl::strings
-    absl::synchronization
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    log_internal_stderr_log_sink_test
-  SRCS
-    "internal/stderr_log_sink_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_globals
-    absl::log_internal_test_helpers
-    absl::log_severity
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_stripping_test
-  SRCS
-    "stripping_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::check
-    absl::flags_program_name
-    absl::log
-    absl::log_internal_test_helpers
-    absl::log_severity
-    absl::status
-    absl::strerror
-    absl::strings
-    absl::str_format
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    log_structured_test
-  SRCS
-    "structured_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::log_internal_test_helpers
-    absl::log_internal_test_matchers
-    absl::log_structured
-    absl::scoped_mock_log
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    internal_fnmatch_test
-  SRCS
-    "internal/fnmatch_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log_internal_fnmatch
-    GTest::gmock_main
-)
--- a/absl/memory/CMakeLists.txt
+++ b/absl/memory/CMakeLists.txt
@@ -26,16 +26,3 @@
     absl::meta
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    memory_test
-  SRCS
-    "memory_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::memory
-    absl::core_headers
-    GTest::gmock_main
-)
--- a/absl/meta/CMakeLists.txt
+++ b/absl/meta/CMakeLists.txt
@@ -27,21 +27,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    type_traits_test
-  SRCS
-    "type_traits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::time
-    absl::core_headers
-    absl::type_traits
-    GTest::gmock_main
-)
-
 # component target
 absl_cc_library(
   NAME
--- a/absl/numeric/CMakeLists.txt
+++ b/absl/numeric/CMakeLists.txt
@@ -27,20 +27,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    bits_test
-  SRCS
-    "bits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::bits
-    absl::core_headers
-    absl::random_random
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     int128
@@ -59,23 +45,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    int128_test
-  SRCS
-    "int128_stream_test.cc"
-    "int128_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::int128
-    absl::base
-    absl::hash_testing
-    absl::type_traits
-    absl::strings
-    GTest::gmock_main
-)
-
 # component target
 absl_cc_library(
   NAME
--- a/absl/profiling/CMakeLists.txt
+++ b/absl/profiling/CMakeLists.txt
@@ -24,19 +24,6 @@
     absl::synchronization
 )
 
-absl_cc_test(
-  NAME
-    sample_recorder_test
-  SRCS
-    "internal/sample_recorder_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::sample_recorder
-    absl::time
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     exponential_biased
@@ -51,19 +38,6 @@
     absl::core_headers
 )
 
-absl_cc_test(
-  NAME
-    exponential_biased_test
-  SRCS
-    "internal/exponential_biased_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::exponential_biased
-    absl::strings
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     periodic_sampler
@@ -78,16 +52,3 @@
     absl::exponential_biased
 )
 
-absl_cc_test(
-  NAME
-    periodic_sampler_test
-  SRCS
-    "internal/periodic_sampler_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::core_headers
-    absl::periodic_sampler
-    GTest::gmock_main
-)
-
--- a/absl/random/CMakeLists.txt
+++ b/absl/random/CMakeLists.txt
@@ -48,24 +48,6 @@
     absl::type_traits
 )
 
-absl_cc_test(
-  NAME
-    random_bit_gen_ref_test
-  SRCS
-    "bit_gen_ref_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_bit_gen_ref
-    absl::random_random
-    absl::random_internal_sequence_urbg
-    absl::fast_type_id
-    GTest::gmock
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -99,67 +81,6 @@
 
 absl_cc_library(
   NAME
-    random_mocking_bit_gen
-  HDRS
-    "mock_distributions.h"
-    "mocking_bit_gen.h"
-  COPTS
-    ${ABSL_DEFAULT_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::flat_hash_map
-    absl::raw_logging_internal
-    absl::random_distributions
-    absl::random_internal_distribution_caller
-    absl::random_internal_mock_overload_set
-    absl::random_random
-    absl::strings
-    absl::span
-    absl::type_traits
-    absl::utility
-    absl::variant
-    GTest::gmock
-    GTest::gtest
-  PUBLIC
-  TESTONLY
-)
-
-absl_cc_test(
-  NAME
-    random_mock_distributions_test
-  SRCS
-    "mock_distributions_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_mocking_bit_gen
-    absl::random_random
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_mocking_bit_gen_test
-  SRCS
-    "mocking_bit_gen_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_bit_gen_ref
-    absl::random_mocking_bit_gen
-    absl::random_random
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_library(
-  NAME
     random_distributions
   SRCS
     "discrete_distribution.cc"
@@ -232,290 +153,6 @@
     absl::span
 )
 
-absl_cc_test(
-  NAME
-    random_bernoulli_distribution_test
-  SRCS
-    "bernoulli_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_distributions
-    absl::random_random
-    absl::random_internal_sequence_urbg
-    absl::random_internal_pcg_engine
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_beta_distribution_test
-  SRCS
-    "beta_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::numeric_representation
-    absl::random_distributions
-    absl::random_random
-    absl::random_internal_distribution_test_util
-    absl::random_internal_sequence_urbg
-    absl::random_internal_pcg_engine
-    absl::strings
-    absl::str_format
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_distributions_test
-  SRCS
-    "distributions_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_distributions
-    absl::random_random
-    absl::random_internal_distribution_test_util
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_generators_test
-  SRCS
-    "generators_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_distributions
-    absl::random_random
-    absl::raw_logging_internal
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_log_uniform_int_distribution_test
-  SRCS
-    "log_uniform_int_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    absl::str_format
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_discrete_distribution_test
-  SRCS
-    "discrete_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_poisson_distribution_test
-  SRCS
-    "poisson_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_distributions
-    absl::random_random
-    absl::core_headers
-    absl::flat_hash_map
-    absl::log
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::strings
-    absl::str_format
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_exponential_distribution_test
-  SRCS
-    "exponential_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::numeric_representation
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    absl::str_format
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_gaussian_distribution_test
-  SRCS
-    "gaussian_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::log
-    absl::numeric_representation
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    absl::str_format
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_uniform_int_distribution_test
-  SRCS
-    "uniform_int_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_uniform_real_distribution_test
-  SRCS
-    "uniform_real_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::numeric_representation
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_zipf_distribution_test
-  SRCS
-    "zipf_distribution_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_distributions
-    absl::random_internal_distribution_test_util
-    absl::random_internal_pcg_engine
-    absl::random_internal_sequence_urbg
-    absl::random_random
-    absl::strings
-    GTest::gmock
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_examples_test
-  SRCS
-    "examples_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_random
-    GTest::gtest_main
-)
-
-absl_cc_test(
-  NAME
-    random_seed_sequences_test
-  SRCS
-    "seed_sequences_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_seed_sequences
-    absl::random_internal_nonsecure_base
-    absl::random_random
-    GTest::gmock
-    GTest::gtest_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -887,274 +524,6 @@
 )
 
 # Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_traits_test
-  SRCS
-    "internal/traits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_traits
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_generate_real_test
-  SRCS
-    "internal/generate_real_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::bits
-    absl::flags
-    absl::random_internal_generate_real
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_distribution_test_util_test
-  SRCS
-    "internal/distribution_test_util_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_distribution_test_util
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_fastmath_test
-  SRCS
-    "internal/fastmath_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_fastmath
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_explicit_seed_seq_test
-  SRCS
-    "internal/explicit_seed_seq_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_explicit_seed_seq
-    absl::random_seed_sequences
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_salted_seed_seq_test
-  SRCS
-    "internal/salted_seed_seq_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_salted_seed_seq
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_chi_square_test
-  SRCS
-    "internal/chi_square_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::core_headers
-    absl::random_internal_distribution_test_util
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_fast_uniform_bits_test
-  SRCS
-    "internal/fast_uniform_bits_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_fast_uniform_bits
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_nonsecure_base_test
-  SRCS
-    "internal/nonsecure_base_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_nonsecure_base
-    absl::random_random
-    absl::random_distributions
-    absl::random_seed_sequences
-    absl::strings
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_seed_material_test
-  SRCS
-    "internal/seed_material_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_seed_material
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_pool_urbg_test
-  SRCS
-    "internal/pool_urbg_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_pool_urbg
-    absl::span
-    absl::type_traits
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_pcg_engine_test
-  SRCS
-    "internal/pcg_engine_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_explicit_seed_seq
-    absl::random_internal_pcg_engine
-    absl::time
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_randen_engine_test
-  SRCS
-    "internal/randen_engine_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_internal_explicit_seed_seq
-    absl::random_internal_randen_engine
-    absl::strings
-    absl::time
-    GTest::gmock
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_randen_test
-  SRCS
-    "internal/randen_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_randen
-    absl::type_traits
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_randen_slow_test
-  SRCS
-    "internal/randen_slow_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::endian
-    absl::random_internal_randen_slow
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_randen_hwaes_test
-  SRCS
-    "internal/randen_hwaes_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::log
-    absl::random_internal_platform
-    absl::random_internal_randen_hwaes
-    absl::random_internal_randen_hwaes_impl
-    absl::str_format
-    GTest::gmock
-    GTest::gtest
-)
-
-# Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
     random_internal_uniform_helper
@@ -1169,51 +538,3 @@
     absl::random_internal_traits
     absl::type_traits
 )
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_uniform_helper_test
-  SRCS
-    "internal/uniform_helper_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_uniform_helper
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_iostream_state_saver_test
-  SRCS
-    "internal/iostream_state_saver_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_iostream_state_saver
-    GTest::gtest_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_test(
-  NAME
-    random_internal_wide_multiply_test
-  SRCS
-      internal/wide_multiply_test.cc
-  COPTS
-    ${ABSL_TEST_COPTS}
-  LINKOPTS
-    ${ABSL_DEFAULT_LINKOPTS}
-  DEPS
-    absl::random_internal_wide_multiply
-    absl::bits
-    absl::int128
-    GTest::gmock
-    GTest::gtest_main
-)
--- a/absl/status/CMakeLists.txt
+++ b/absl/status/CMakeLists.txt
@@ -49,20 +49,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    status_test
-  SRCS
-   "status_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::status
-    absl::str_format
-    absl::strings
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     statusor
@@ -88,17 +74,3 @@
     absl::variant
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    statusor_test
-  SRCS
-   "statusor_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::status
-    absl::statusor
-    absl::strings
-    GTest::gmock_main
-)
--- a/absl/strings/CMakeLists.txt
+++ b/absl/strings/CMakeLists.txt
@@ -141,330 +141,6 @@
     absl::type_traits
 )
 
-absl_cc_test(
-  NAME
-    match_test
-  SRCS
-    "match_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::base
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    escaping_test
-  SRCS
-    "escaping_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::core_headers
-    absl::fixed_array
-    GTest::gmock_main
-    absl::check
-)
-
-absl_cc_test(
-  NAME
-    has_absl_stringify_test
-  SRCS
-    "has_absl_stringify_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::optional
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    has_ostream_operator_test
-  SRCS
-    "has_ostream_operator_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::has_ostream_operator
-    absl::optional
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    ascii_test
-  SRCS
-    "ascii_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    damerau_levenshtein_distance_test
-  SRCS
-    "internal/damerau_levenshtein_distance_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::base
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    memutil_test
-  SRCS
-    "internal/memutil.h"
-    "internal/memutil_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    utf8_test
-  SRCS
-    "internal/utf8_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings_internal
-    absl::base
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    string_constant_test
-  SRCS
-    "internal/string_constant_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::type_traits
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    string_view_test
-  SRCS
-    "string_view_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::config
-    absl::core_headers
-    absl::dynamic_annotations
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    substitute_test
-  SRCS
-    "substitute_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_replace_test
-  SRCS
-    "str_replace_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_split_test
-  SRCS
-    "str_split_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::core_headers
-    absl::dynamic_annotations
-    absl::btree
-    absl::flat_hash_map
-    absl::node_hash_map
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    ostringstream_test
-  SRCS
-    "internal/ostringstream_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings_internal
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    resize_uninitialized_test
-  SRCS
-    "internal/resize_uninitialized.h"
-    "internal/resize_uninitialized_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::core_headers
-    absl::type_traits
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_join_test
-  SRCS
-    "str_join_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::base
-    absl::core_headers
-    absl::memory
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_cat_test
-  SRCS
-    "str_cat_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::str_format
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    numbers_test
-  SRCS
-    "internal/numbers_test_common.h"
-    "numbers_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::core_headers
-    absl::int128
-    absl::log
-    absl::pow10_helper
-    absl::random_distributions
-    absl::random_random
-    absl::strings
-    absl::strings_internal
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    strip_test
-  SRCS
-    "strip_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::base
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    charset_test
-  SRCS
-    "charset_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    charconv_test
-  SRCS
-    "charconv_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::str_format
-    absl::pow10_helper
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    charconv_parse_test
-  SRCS
-    "internal/charconv_parse.h"
-    "internal/charconv_parse_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::check
-    absl::config
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    charconv_bigint_test
-  SRCS
-    "internal/charconv_bigint.h"
-    "internal/charconv_bigint_test.cc"
-    "internal/charconv_parse.h"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::strings
-    absl::config
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     str_format
@@ -518,137 +194,6 @@
     absl::span
 )
 
-absl_cc_test(
-  NAME
-    str_format_test
-  SRCS
-    "str_format_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord
-    absl::core_headers
-    absl::span
-    absl::str_format
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_extension_test
-  SRCS
-    "internal/str_format/extension_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::str_format
-    absl::str_format_internal
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_arg_test
-  SRCS
-    "internal/str_format/arg_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::str_format
-    absl::str_format_internal
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_bind_test
-  SRCS
-    "internal/str_format/bind_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::str_format_internal
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_checker_test
-  SRCS
-    "internal/str_format/checker_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::str_format
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_convert_test
-  SRCS
-    "internal/str_format/convert_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::core_headers
-    absl::int128
-    absl::log
-    absl::raw_logging_internal
-    absl::span
-    absl::str_format
-    absl::str_format_internal
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_output_test
-  SRCS
-    "internal/str_format/output_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::str_format_internal
-    absl::cord
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    str_format_parser_test
-  SRCS
-    "internal/str_format/parser_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::str_format_internal
-    absl::string_view
-    absl::config
-    absl::core_headers
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    char_formatting_test
-  SRCS
-    "char_formatting_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::str_format
-    absl::strings
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -664,19 +209,6 @@
   TESTONLY
 )
 
-absl_cc_test(
-  NAME
-    pow10_helper_test
-  SRCS
-    "internal/pow10_helper_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::pow10_helper
-    absl::str_format
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -727,21 +259,6 @@
     absl::config
 )
 
-absl_cc_test(
-  NAME
-    cordz_update_tracker_test
-  SRCS
-    "internal/cordz_update_tracker_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cordz_update_tracker
-    absl::core_headers
-    absl::synchronization
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -759,18 +276,6 @@
     absl::raw_logging_internal
 )
 
-absl_cc_test(
-  NAME
-    cordz_functions_test
-  SRCS
-    "internal/cordz_functions_test.cc"
-  DEPS
-    absl::config
-    absl::cordz_functions
-    absl::cordz_test_helpers
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -803,23 +308,6 @@
     absl::synchronization
 )
 
-absl_cc_test(
-  NAME
-    cordz_handle_test
-  SRCS
-    "internal/cordz_handle_test.cc"
-  DEPS
-    absl::config
-    absl::cordz_handle
-    absl::cordz_test_helpers
-    absl::memory
-    absl::random_random
-    absl::random_distributions
-    absl::synchronization
-    absl::time
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -847,49 +335,6 @@
     absl::time
 )
 
-absl_cc_test(
-  NAME
-    cordz_info_test
-  SRCS
-    "internal/cordz_info_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord_internal
-    absl::cordz_test_helpers
-    absl::cordz_handle
-    absl::cordz_info
-    absl::cordz_statistics
-    absl::cordz_test_helpers
-    absl::cordz_update_tracker
-    absl::span
-    absl::stacktrace
-    absl::symbolize
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cordz_info_statistics_test
-  SRCS
-    "internal/cordz_info_statistics_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord
-    absl::cord_internal
-    absl::cordz_info
-    absl::cordz_sample_token
-    absl::cordz_statistics
-    absl::cordz_update_scope
-    absl::cordz_update_tracker
-    absl::crc_cord_state
-    absl::thread_pool
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -906,29 +351,6 @@
     absl::cordz_info
 )
 
-absl_cc_test(
-  NAME
-    cordz_sample_token_test
-  SRCS
-    "internal/cordz_sample_token_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord_internal
-    absl::cordz_handle
-    absl::cordz_info
-    absl::cordz_info
-    absl::cordz_sample_token
-    absl::cordz_test_helpers
-    absl::memory
-    absl::random_random
-    absl::synchronization
-    absl::thread_pool
-    absl::time
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -945,24 +367,6 @@
     absl::core_headers
 )
 
-absl_cc_test(
-  NAME
-    cordz_update_scope_test
-  SRCS
-    "internal/cordz_update_scope_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord_internal
-    absl::cordz_info
-    absl::cordz_test_helpers
-    absl::cordz_update_scope
-    absl::cordz_update_tracker
-    absl::core_headers
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     cord
@@ -1052,142 +456,3 @@
     absl::strings
   TESTONLY
 )
-
-absl_cc_test(
-  NAME
-    cord_test
-  SRCS
-    "cord_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::check
-    absl::config
-    absl::cord
-    absl::cord_test_helpers
-    absl::cordz_test_helpers
-    absl::core_headers
-    absl::endian
-    absl::fixed_array
-    absl::function_ref
-    absl::hash
-    absl::log
-    absl::optional
-    absl::random_random
-    absl::str_format
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cord_data_edge_test
-  SRCS
-    "internal/cord_data_edge_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::cord_internal
-    absl::cord_rep_test_util
-    absl::core_headers
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cord_rep_btree_test
-  SRCS
-    "internal/cord_rep_btree_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::cleanup
-    absl::config
-    absl::cord_internal
-    absl::cord_rep_test_util
-    absl::core_headers
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cord_rep_btree_navigator_test
-  SRCS
-    "internal/cord_rep_btree_navigator_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::cord_internal
-    absl::cord_rep_test_util
-    absl::core_headers
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cord_rep_btree_reader_test
-  SRCS
-    "internal/cord_rep_btree_reader_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::cord_internal
-    absl::cord_rep_test_util
-    absl::core_headers
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cord_rep_crc_test
-  SRCS
-    "internal/cord_rep_crc_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::cord_internal
-    absl::cord_rep_test_util
-    absl::crc_cord_state
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    cordz_test
-  SRCS
-    "cordz_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::cord
-    absl::cord_internal
-    absl::cord_test_helpers
-    absl::cordz_test_helpers
-    absl::cordz_functions
-    absl::cordz_info
-    absl::cordz_sample_token
-    absl::cordz_statistics
-    absl::cordz_update_tracker
-    absl::base
-    absl::config
-    absl::core_headers
-    absl::raw_logging_internal
-    absl::strings
-    GTest::gmock_main
-)
--- a/absl/synchronization/CMakeLists.txt
+++ b/absl/synchronization/CMakeLists.txt
@@ -51,21 +51,6 @@
     absl::time
 )
 
-absl_cc_test(
-  NAME
-    kernel_timeout_internal_test
-  SRCS
-    "internal/kernel_timeout_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::kernel_timeout_internal
-    absl::config
-    absl::random_random
-    absl::time
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     synchronization
@@ -117,47 +102,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    barrier_test
-  SRCS
-    "barrier_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::synchronization
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    blocking_counter_test
-  SRCS
-    "blocking_counter_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::synchronization
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    graphcycles_test
-  SRCS
-    "internal/graphcycles_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::check
-    absl::core_headers
-    absl::graphcycles_internal
-    absl::log
-    GTest::gmock_main
-)
-
 # Internal-only target, do not depend on directly.
 absl_cc_library(
   NAME
@@ -174,103 +118,6 @@
 )
 
 absl_cc_test(
-  NAME
-    mutex_test
-  SRCS
-    "mutex_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::synchronization
-    absl::thread_pool
-    absl::base
-    absl::check
-    absl::config
-    absl::core_headers
-    absl::log
-    absl::memory
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    mutex_method_pointer_test
-  SRCS
-    "mutex_method_pointer_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::config
-    absl::synchronization
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    notification_test
-  SRCS
-    "notification_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::synchronization
-    absl::time
-    GTest::gmock_main
-)
-
-# Internal-only target, do not depend on directly.
-absl_cc_library(
-  NAME
-    per_thread_sem_test_common
-  SRCS
-    "internal/per_thread_sem_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::synchronization
-    absl::base
-    absl::config
-    absl::strings
-    absl::time
-    GTest::gmock
-  TESTONLY
-)
-
-absl_cc_test(
-  NAME
-    per_thread_sem_test
-  SRCS
-    "internal/per_thread_sem_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::per_thread_sem_test_common
-    absl::synchronization
-    absl::strings
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    waiter_test
-  SRCS
-    "internal/waiter_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::config
-    absl::kernel_timeout_internal
-    absl::random_random
-    absl::synchronization
-    absl::thread_pool
-    absl::time
-    GTest::gmock_main
-)
-
-absl_cc_test(
   NAME
     lifetime_test
   SRCS
--- a/absl/time/CMakeLists.txt
+++ b/absl/time/CMakeLists.txt
@@ -103,39 +103,3 @@
     absl::time_zone
   TESTONLY
 )
-
-absl_cc_test(
-  NAME
-    time_test
-  SRCS
-    "civil_time_test.cc"
-    "clock_test.cc"
-    "duration_test.cc"
-    "format_test.cc"
-    "time_test.cc"
-    "time_zone_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::time_internal_test_util
-    absl::time
-    absl::config
-    absl::core_headers
-    absl::strings
-    absl::str_format
-    absl::time_zone
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    flag_test
-  SRCS
-    "flag_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::flags
-    absl::flags_reflection
-    GTest::gmock_main
-)
--- a/absl/types/CMakeLists.txt
+++ b/absl/types/CMakeLists.txt
@@ -57,52 +57,6 @@
     absl::raw_logging_internal
 )
 
-absl_cc_test(
-  NAME
-    any_test
-  SRCS
-    "any_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any
-    absl::config
-    absl::exception_testing
-    absl::log
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    any_test_noexceptions
-  SRCS
-    "any_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any
-    absl::config
-    absl::exception_testing
-    absl::raw_logging_internal
-    absl::test_instance_tracker
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    any_exception_safety_test
-  SRCS
-    "any_exception_safety_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::any
-    absl::config
-    absl::exception_safety_testing
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     span
@@ -121,46 +75,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    span_test
-  SRCS
-    "span_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::span
-    absl::base
-    absl::config
-    absl::core_headers
-    absl::exception_testing
-    absl::fixed_array
-    absl::inlined_vector
-    absl::hash_testing
-    absl::strings
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    span_test_noexceptions
-  SRCS
-    "span_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::span
-    absl::base
-    absl::config
-    absl::core_headers
-    absl::exception_testing
-    absl::fixed_array
-    absl::inlined_vector
-    absl::hash_testing
-    absl::strings
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     optional
@@ -212,36 +126,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    optional_test
-  SRCS
-    "optional_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::optional
-    absl::config
-    absl::log
-    absl::strings
-    absl::type_traits
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    optional_exception_safety_test
-  SRCS
-    "optional_exception_safety_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::optional
-    absl::config
-    absl::exception_safety_testing
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     variant
@@ -261,23 +145,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    variant_test
-  SRCS
-    "variant_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::variant
-    absl::config
-    absl::core_headers
-    absl::memory
-    absl::type_traits
-    absl::strings
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     compare
@@ -291,31 +158,3 @@
     absl::type_traits
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    compare_test
-  SRCS
-    "compare_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::base
-    absl::compare
-    GTest::gmock_main
-)
-
-absl_cc_test(
-  NAME
-    variant_exception_safety_test
-  SRCS
-    "variant_exception_safety_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::variant
-    absl::config
-    absl::exception_safety_testing
-    absl::memory
-    GTest::gmock_main
-)
--- a/absl/utility/CMakeLists.txt
+++ b/absl/utility/CMakeLists.txt
@@ -28,21 +28,6 @@
   PUBLIC
 )
 
-absl_cc_test(
-  NAME
-    utility_test
-  SRCS
-    "utility_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::utility
-    absl::core_headers
-    absl::memory
-    absl::strings
-    GTest::gmock_main
-)
-
 absl_cc_library(
   NAME
     if_constexpr
@@ -54,15 +39,3 @@
     absl::config
   PUBLIC
 )
-
-absl_cc_test(
-  NAME
-    if_constexpr_test
-  SRCS
-    "internal/if_constexpr_test.cc"
-  COPTS
-    ${ABSL_TEST_COPTS}
-  DEPS
-    absl::if_constexpr
-    GTest::gmock_main
-)
--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -736,15 +736,6 @@
   ${ARGN}
   )
 
-  if (ABSL_INTERNAL_MAKE_DLL_TEST)
-    set(_dll "abseil_test_dll")
-    set(_dll_files ${ABSL_INTERNAL_TEST_DLL_FILES})
-    set(_dll_libs "abseil_dll" "GTest::gtest" "GTest::gmock")
-    set(_dll_compile_definitions "GTEST_LINKED_AS_SHARED_LIBRARY=1")
-    set(_dll_includes ${absl_gtest_src_dir}/googletest/include ${absl_gtest_src_dir}/googlemock/include)
-    set(_dll_consume "ABSL_CONSUME_TEST_DLL")
-    set(_dll_build "ABSL_BUILD_TEST_DLL")
-  else()
     set(_dll "abseil_dll")
     set(_dll_files ${ABSL_INTERNAL_DLL_FILES})
     set(_dll_libs "")
@@ -752,7 +743,6 @@
     set(_dll_includes "")
     set(_dll_consume "ABSL_CONSUME_DLL")
     set(_dll_build "ABSL_BUILD_DLL")
-  endif()
 
   add_library(
     ${_dll}
