diff --git a/dashboard/CMakeLists.txt b/dashboard/CMakeLists.txt
index 081816a..41a06fb 100644
--- a/dashboard/CMakeLists.txt
+++ b/dashboard/CMakeLists.txt
@@ -60,7 +60,12 @@ if (WIVRN_CHECK_CAPSYSNICE)
 endif()
 
 file(GLOB TRANSLATION_FILES ${CMAKE_CURRENT_SOURCE_DIR}/wivrn_*.ts)
-qt_add_translations(wivrn-dashboard TS_FILES ${TRANSLATION_FILES})
+
+qt_add_translations(wivrn-dashboard SOURCE_TARGETS wivrn-dashboard TS_FILES ${TRANSLATION_FILES})
+
+# https://16bpp.net/blog/post/localizing-a-qt-app-or-anything-else-for-that-matter/, CMake Setup
+# https://gist.github.com/define-private-public/f01ff2f9bbb74eded144558a2e02966c#file-cmakelists-2-txt
+add_dependencies(wivrn-dashboard release_translations)
 
 install(TARGETS wivrn-dashboard)
 
-- 
2.47.0

