From 3e8294b50a34d7060ec0636afb1a7ad9e448db5b Mon Sep 17 00:00:00 2001
From: Thom Wiggers <thom@thomwiggers.nl>
Date: Thu, 24 Jan 2019 16:09:37 +0100
Subject: [PATCH] Marie Kondo

---
 src/remove.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/remove.c b/src/remove.c
index 8102d4e..74ff93e 100644
--- a/src/remove.c
+++ b/src/remove.c
@@ -288,12 +288,12 @@ prompt (FTS const *fts, FTSENT const *ent, bool is_dir,
                        problems, it may be more convenient to translate these
                        strings instead as: "%1$s: %3$s is write-protected and
                        is of type '%2$s' -- remove it? ".  */
-                    ? _("%s: remove write-protected %s %s? ")
-                    : _("%s: remove %s %s? ")),
+                    ? _("%s: does write-protected %s %s spark joy? ")
+                    : _("%s: does %s %s spark joy? ")),
                    program_name, file_type (sbuf), quoted_name);
         }
 
-      if (!yesno ())
+      if (yesno ())
         return RM_USER_DECLINED;
     }
   return RM_OK;
-- 
2.20.1

