diff --git a/configs/mx53loco_defconfig b/configs/mx53loco_defconfig
index 3fba84c..813b517 100644
--- a/configs/mx53loco_defconfig
+++ b/configs/mx53loco_defconfig
@@ -29,3 +29,6 @@ CONFIG_USB_ETHER_SMSC95XX=y
 CONFIG_VIDEO=y
 # CONFIG_VIDEO_SW_CURSOR is not set
 CONFIG_OF_LIBFDT=y
+CONFIG_CMD_PART=y
+CONFIG_PARTITION_UUIDS=y
+CONFIG_IDENT_STRING="\nMX53loco Arch Linux ARM"
diff --git a/include/configs/mx53loco.h b/include/configs/mx53loco.h
index e973b35..341e5ef 100644
--- a/include/configs/mx53loco.h
+++ b/include/configs/mx53loco.h
@@ -84,14 +84,15 @@
 	"ip_dyn=yes\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
-	"mmcroot=/dev/mmcblk0p2 rw rootwait\0" \
+	"bootdir=/\0" \
+	"mmcroot=/dev/mmcblk0p1 rw rootwait\0" \
 	"mmcargs=setenv bootargs console=ttymxc0,${baudrate} root=${mmcroot}\0" \
 	"loadbootscript=" \
-		"load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
+		"load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}${image}\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${bootdir}dtbs/${fdt_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -140,7 +141,16 @@
 		"else " \
 			"if run loadimage; then " \
 				"run mmcboot; " \
-			"else run netboot; " \
+			"else " \
+				"setenv bootdir /boot/; " \
+				"if run loadbootscript; then " \
+					"run bootscript; " \
+				"else " \
+					"if run loadimage; then " \
+						"run mmcboot; " \
+					"else run netboot; " \
+					"fi; " \
+				"fi; " \
 			"fi; " \
 		"fi; " \
 	"else run netboot; fi"
