diff --git a/src/spiceqxl_display.c b/src/spiceqxl_display.c
index a3a8978..748c2db 100644
--- a/src/spiceqxl_display.c
+++ b/src/spiceqxl_display.c
@@ -350,6 +350,6 @@ void qxl_add_spice_display_interface(qxl_screen_t *qxl)
 
 void spiceqxl_display_monitors_config(qxl_screen_t *qxl)
 {
-    spice_qxl_monitors_config_async(&qxl->display_sin, (QXLPHYSICAL)qxl->monitors_config,
+    spice_qxl_monitors_config_async(&qxl->display_sin, qxl->monitors_config,
                                     MEMSLOT_GROUP, 0);
 }
