From 3e7d71e1986af1dbecbc8b2772982f0382bda17b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <bevan@bi-co.net>
Date: Sat, 15 Feb 2025 20:10:54 +0100
Subject: [PATCH] Allow building against QWT 6.3.0

---
 cmake/Modules/FindQwt.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/Modules/FindQwt.cmake b/cmake/Modules/FindQwt.cmake
index 4e778add..3309d0e6 100644
--- a/cmake/Modules/FindQwt.cmake
+++ b/cmake/Modules/FindQwt.cmake
@@ -20,7 +20,7 @@ find_path(QWT_INCLUDE_DIRS
 	/sw/include/qwt
 	/usr/local/lib/qwt.framework/Headers
 	/usr/local/lib/qwt-qt5/lib/framework/Headers
-	/usr/local/qwt-6.2.0/include
+	/usr/local/qwt-6.3.0/include
 )
 if (APPLE)
 	set(CMAKE_FIND_LIBRARY_SUFFIXES " " " .dylib" ".so" ".a ")
@@ -38,7 +38,7 @@ find_library (QWT_LIBRARIES
 	/sw/lib
 	/usr/local/lib/qwt.framework
 	/usr/local/lib/qwt-qt6/lib/framework
-	/usr/local/qwt-6.2.0/lib
+	/usr/local/qwt-6.3.0/lib
 )
 
 set(QWT_FOUND FALSE)
@@ -49,13 +49,13 @@ if(QWT_INCLUDE_DIRS)
   set(QWT_VERSION "No Version")
   string(REGEX MATCH "[0-9]+.[0-9]+.[0-9]+" QWT_VERSION ${QWT_STRING_VERSION})
   string(COMPARE LESS ${QWT_VERSION} "5.2.0" QWT_WRONG_VERSION)
-  string(COMPARE GREATER ${QWT_VERSION} "6.2.0" QWT_WRONG_VERSION)
+  string(COMPARE GREATER ${QWT_VERSION} "6.3.0" QWT_WRONG_VERSION)
 
   message(STATUS "QWT Version: ${QWT_VERSION}")
   if(NOT QWT_WRONG_VERSION)
     set(QWT_FOUND TRUE)
   else(NOT QWT_WRONG_VERSION)
-    message(STATUS "QWT Version must be >= 5.2 and <= 6.2.0, Found ${QWT_VERSION}")
+    message(STATUS "QWT Version must be >= 5.2 and <= 6.3.0, Found ${QWT_VERSION}")
   endif(NOT QWT_WRONG_VERSION)
 
 endif(QWT_INCLUDE_DIRS)
-- 
2.48.1

