From 834b4f0160c1a8943b9be12eb0a0b9494c9f640f Mon Sep 17 00:00:00 2001
From: Julianne H <julianne.hervier@qonfucius.team>
Date: Thu, 16 Feb 2023 11:48:25 +0100
Subject: [PATCH] makepkg: Fix Cargo.toml

---
 bkmr/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bkmr/Cargo.toml b/bkmr/Cargo.toml
index c18d1b2..de3ee47 100644
--- a/bkmr/Cargo.toml
+++ b/bkmr/Cargo.toml
@@ -34,7 +34,7 @@ assert_cmd = "2.0.7"
 ctor = "0.1.26"
 mockall = "0.11.3"
 rstest = "0.16.0"
-test-log = "0.2.21"
+test-log = "0.2.11"
 tuikit = "0.5.0"
 select = "0.6.0"
 reqwest = {version = "0.11.13", features = ["blocking", "json"] }
-- 
2.39.2

