From 546c7fdd794206e504faa9374d5f9e53f17b853f Mon Sep 17 00:00:00 2001
From: Olivier Mehani <olivier.mehani@nicta.com.au>
Date: Tue, 17 Jun 2014 16:23:02 +1000
Subject: [PATCH] Use str() as unicode() if missing

Signed-off-by: Olivier Mehani <olivier.mehani@nicta.com.au>
---
 GTG/plugins/hamster/hamster.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/GTG/plugins/hamster/hamster.py b/GTG/plugins/hamster/hamster.py
index 2badf8b..ada250e 100644
--- a/GTG/plugins/hamster/hamster.py
+++ b/GTG/plugins/hamster/hamster.py
@@ -28,7 +28,11 @@ import datetime
 from GTG import _
 from GTG.core.task import Task
 
+try:
+    unicode("")
+except:
+    unicode=str
 
 class hamsterPlugin:
     PLUGIN_NAMESPACE = 'hamster-plugin'
     DEFAULT_PREFERENCES = {
-- 
2.0.0

