From 6bb6d0b8338b59cb3120516a797547f5f8fa26b7 Mon Sep 17 00:00:00 2001
From: soloturn <soloturn@gmail.com>
Date: Fri, 31 Dec 2021 18:10:49 +0100
Subject: [PATCH] arch aur pachtes

---
 utils/build-presets.ini | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 7f5a7cdc6a6..d85bb468b50 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -818,11 +818,9 @@ llbuild
 swiftpm
 swift-driver
 xctest
-libicu
 libcxx
 swiftdocc
 
-build-ninja
 install-llvm
 install-swift
 install-lldb
@@ -830,8 +828,7 @@ install-llbuild
 install-swiftpm
 install-swift-driver
 install-xctest
-install-libicu
-install-prefix=/usr
+install-prefix=/usr/lib/swift-git
 install-libcxx
 install-sourcekit-lsp
 install-swiftdocc
@@ -844,14 +841,13 @@ build-swift-stdlib-unittest-extra
 
 test-installable-package
 
-# Build the benchmarks against the toolchain.
-toolchain-benchmarks
+# does not work outside /usr, Build the benchmarks against the toolchain.
+# toolchain-benchmarks
 
 # Path to the root of the installation filesystem.
 install-destdir=%(install_destdir)s
 
 # Path to the .tar.gz package we would create.
-installable-package=%(installable_package)s
 
 # This ensures the default module cache
 # location is local to this run, allowing
-- 
2.34.1

