From ea37bb9a26424309a50aaf289b40c6ac68baf9b1 Mon Sep 17 00:00:00 2001
From: FabioLolix <fabio.loli@disroot.org>
Date: Fri, 24 Apr 2026 19:32:08 +0200
Subject: [PATCH] remove lock

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index eff74fb..c76e6d4 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -57,7 +57,7 @@ doc = [
 ]
 
 [build-system]
-requires = ["uv_build>=0.8.14,<0.9.0"]
+requires = ["uv_build>=0.9.0"]
 build-backend = "uv_build"
 
 [tool.uv.build-backend]
-- 
2.53.0

