From 42d4d7c7c00389532bf57e91d9c2df15457d0269 Mon Sep 17 00:00:00 2001
From: y2keeth <y2keeth@yahoo.com>
Date: Sat, 7 Nov 2015 12:02:59 -0500
Subject: [PATCH] Patch from Debian #803857 to support ffmpeg 2.9

---
 neo/renderer/Cinematic.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/neo/renderer/Cinematic.cpp b/neo/renderer/Cinematic.cpp
index a9a35a6..3865d0d 100644
--- a/neo/renderer/Cinematic.cpp
+++ b/neo/renderer/Cinematic.cpp
@@ -439,8 +439,8 @@ idCinematicLocal::~idCinematicLocal()
 	
 	// RB: TODO double check this. It seems we have different versions of ffmpeg on Kubuntu 13.10 and the win32 development files
 #if defined(_WIN32) || defined(_WIN64)
-	avcodec_free_frame( &frame );
-	avcodec_free_frame( &frame2 );
+	av_frame_free( &frame );
+	av_frame_free( &frame2 );
 #else
 	av_freep( &frame );
 	av_freep( &frame2 );
@@ -557,12 +557,12 @@ bool idCinematicLocal::InitFromFFMPEGFile( const char* qpath, bool amilooping )
 	framePos = -1;
 	common->Printf( "Loaded FFMPEG file: '%s', looping=%d%dx%d, %f FPS, %f sec\n", qpath, looping, CIN_WIDTH, CIN_HEIGHT, frameRate, durationSec );
 	image = ( byte* )Mem_Alloc( CIN_WIDTH * CIN_HEIGHT * 4 * 2, TAG_CINEMATIC );
-	avpicture_fill( ( AVPicture* )frame2, image, PIX_FMT_BGR32, CIN_WIDTH, CIN_HEIGHT );
+	avpicture_fill( ( AVPicture* )frame2, image, AV_PIX_FMT_BGR32, CIN_WIDTH, CIN_HEIGHT );
 	if( img_convert_ctx )
 	{
 		sws_freeContext( img_convert_ctx );
 	}
-	img_convert_ctx = sws_getContext( dec_ctx->width, dec_ctx->height, dec_ctx->pix_fmt, CIN_WIDTH, CIN_HEIGHT, PIX_FMT_BGR32, SWS_BICUBIC, NULL, NULL, NULL );
+	img_convert_ctx = sws_getContext( dec_ctx->width, dec_ctx->height, dec_ctx->pix_fmt, CIN_WIDTH, CIN_HEIGHT, AV_PIX_FMT_BGR32, SWS_BICUBIC, NULL, NULL, NULL );
 	status = FMV_PLAY;
 	
 	startTime = 0;
-- 
2.7.1

