--- /dev/null
+++ b/setup.py
@@ -0,0 +1,24 @@
+from setuptools import setup
+
+with open('README.md') as fh:
+    long_description = fh.read()
+
+setup(
+    name='infinityctl',
+    author='ring',
+    author_email='ring@8chan.co',
+    description='Command line interface for 8chan',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    license='GPL3',
+    classifiers=[
+        'Programming Language :: Python :: 3',
+        'Development Status :: 5 - Production/Stable',
+        'Environment :: Console',
+        'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
+        'Intended Audience :: End Users/Desktop'],
+    py_modules=['infinitychan', 'infinityctl', 'botmod', 'threadviewer', 'urls'],
+    scripts=['infinityctl', 'onionctl', 'finalchanctl', 'vichanctl'],
+    install_requires=['urllib3'],
+    python_requires='>=3.0'
+)
