diff --git a/ebpf_prog/Makefile b/ebpf_prog/Makefile
index 4a472a85..d8b65b68 100644
--- a/ebpf_prog/Makefile
+++ b/ebpf_prog/Makefile
@@ -69,6 +69,7 @@ $(info EXTRA_FLAGS    = $(EXTRA_FLAGS))
 SRC := $(wildcard *.c)
 BIN := $(SRC:.c=.o)
 CFLAGS = -I. \
+	-I$(KERNEL_HEADERS)/arch/$(ARCH)/include/ \
 	-I$(KERNEL_HEADERS)/arch/$(ARCH)/include/generated/ \
 	-I$(KERNEL_HEADERS)/include \
 	-include $(KERNEL_DIR)/include/linux/kconfig.h \
@@ -88,7 +89,9 @@ CFLAGS = -I. \
 	-Wno-gnu-variable-sized-type-not-at-end \
 	-Wno-address-of-packed-member \
 	-Wno-tautological-compare \
-	-Wno-unknown-warning-option  \
+	-Wno-unknown-warning-option \
+	-fcf-protection \
+	-DCC_USING_FENTRY \
 	-fno-stack-protector \
 	-g -O2 -emit-llvm
 
