--- a/cpuset/commands/shield.py	2018-02-22 12:56:37.000000000 -0500
+++ b/cpuset/commands/shield.py	2018-02-22 12:58:00.748829750 -0500
@@ -396,8 +396,8 @@ def make_shield(cpuspec, kthread):
         set.modify(USR_SET, cpuspec, memspec, False, False)
         set.modify(SYS_SET, cpuspec_inv, memspec, False, False)
         # reset cpu exlusivity
-        cset.unique_set(USR_SET).cpu_exclusive = True
-        cset.unique_set(SYS_SET).cpu_exclusive = True
+        ##cset.unique_set(USR_SET).cpu_exclusive = True
+        ##cset.unique_set(SYS_SET).cpu_exclusive = True
         log.info('--> shielding modified with:')
     # move root tasks into system set
     root_tasks = cset.unique_set('/').tasks
