diff --unified --recursive --text mesa.orig/src/gallium/frontends/clover/llvm/metadata.hpp mesa.new/src/gallium/frontends/clover/llvm/metadata.hpp
--- mesa.orig/src/gallium/frontends/clover/llvm/metadata.hpp	2021-02-14 13:15:16.994404651 +0100
+++ mesa.new/src/gallium/frontends/clover/llvm/metadata.hpp	2021-02-14 13:19:11.096855317 +0100
@@ -129,7 +129,11 @@
                      break;
                }
                if (type->isVectorTy())
-                  data += std::to_string(((::llvm::VectorType*)type)->getNumElements());
+#if LLVM_VERSION_MAJOR >= 13
+                  data += std::to_string(((::llvm::FixedVectorType*)type)->getNumElements());
+#else
+                  data += std::to_string(((::llvm::VectorType*)type)->getNumElements());
+#endif
 
             } else {
                ::llvm::raw_string_ostream os { data };
