diff --git a/neo/framework/Licensee.h b/neo/framework/Licensee.h
index 9cc20fb..fb1c882 100644
--- a/neo/framework/Licensee.h
+++ b/neo/framework/Licensee.h
@@ -90,7 +90,7 @@ If you have questions concerning this license or the applicable additional terms
 #define WIN32_CONSOLE_CLASS				"dhewm 3 WinConsole"
 
 // Linux info
-#define LINUX_DEFAULT_PATH				"/usr/local/games/doom3"
+#define LINUX_DEFAULT_PATH				"/usr/share/doom3"
 
 // CD Key file info
 // goes into BASE_GAMEDIR whatever the fs_game is set to
