--- a/SamTFE/Sources/GameMP/Map.cpp	2023-06-03 21:49:29.345299714 +0300
+++ b/SamTFE/Sources/GameMP/Map.cpp	2023-06-07 11:33:13.865159902 +0300
@@ -427,6 +427,7 @@
 BOOL ObtainMapData(void)
 {
   try {
+#ifndef FIRST_ENCOUNTER
     // the second encounter
     atoIconsSE[ 0].SetData_t(CTFILENAME("TexturesMP\\Computer\\Map\\Book.tex"));
     atoIconsSE[ 1].SetData_t(CTFILENAME("TexturesMP\\Computer\\Map\\Level00.tex"));
@@ -465,7 +466,7 @@
     ((CTextureData*)_toMapBcgLUSE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRDSE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRUSE .GetData())->Force(TEX_CONSTANT);
-
+#else
     // the first encounter
     atoIconsFE[ 0].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level00.tex"));
     atoIconsFE[ 1].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level01.tex"));
@@ -482,6 +483,7 @@
     atoIconsFE[12].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level12.tex"));
     atoIconsFE[13].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level13.tex"));
     atoIconsFE[14].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level14.tex"));
+    _toPathDot    .SetData_t(CTFILENAME("Textures\\Computer\\Map\\PathDot.tex"));
     _toMapBcgLDFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgLD.tex"));
     _toMapBcgLUFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgLU.tex"));
     _toMapBcgRDFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgRD.tex"));
@@ -502,10 +504,12 @@
     ((CTextureData*)atoIconsFE[12].GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)atoIconsFE[13].GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)atoIconsFE[14].GetData())->Force(TEX_CONSTANT);
+    ((CTextureData*)_toPathDot    .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgLDFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgLUFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRDFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRUFE .GetData())->Force(TEX_CONSTANT);
+#endif
   }
   catch (const char *strError) {
     CPrintF("%s\n", (const char *)strError);
--- a/SamTSE/Sources/GameMP/Map.cpp	2023-06-03 21:49:29.345299714 +0300
+++ b/SamTSE/Sources/GameMP/Map.cpp	2023-06-07 11:33:13.865159902 +0300
@@ -427,6 +427,7 @@
 BOOL ObtainMapData(void)
 {
   try {
+#ifndef FIRST_ENCOUNTER
     // the second encounter
     atoIconsSE[ 0].SetData_t(CTFILENAME("TexturesMP\\Computer\\Map\\Book.tex"));
     atoIconsSE[ 1].SetData_t(CTFILENAME("TexturesMP\\Computer\\Map\\Level00.tex"));
@@ -465,7 +466,7 @@
     ((CTextureData*)_toMapBcgLUSE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRDSE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRUSE .GetData())->Force(TEX_CONSTANT);
-
+#else
     // the first encounter
     atoIconsFE[ 0].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level00.tex"));
     atoIconsFE[ 1].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level01.tex"));
@@ -482,6 +483,7 @@
     atoIconsFE[12].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level12.tex"));
     atoIconsFE[13].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level13.tex"));
     atoIconsFE[14].SetData_t(CTFILENAME("Textures\\Computer\\Map\\Level14.tex"));
+    _toPathDot    .SetData_t(CTFILENAME("Textures\\Computer\\Map\\PathDot.tex"));
     _toMapBcgLDFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgLD.tex"));
     _toMapBcgLUFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgLU.tex"));
     _toMapBcgRDFE .SetData_t(CTFILENAME("Textures\\Computer\\Map\\MapBcgRD.tex"));
@@ -502,10 +504,12 @@
     ((CTextureData*)atoIconsFE[12].GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)atoIconsFE[13].GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)atoIconsFE[14].GetData())->Force(TEX_CONSTANT);
+    ((CTextureData*)_toPathDot    .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgLDFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgLUFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRDFE .GetData())->Force(TEX_CONSTANT);
     ((CTextureData*)_toMapBcgRUFE .GetData())->Force(TEX_CONSTANT);
+#endif
   }
   catch (const char *strError) {
     CPrintF("%s\n", (const char *)strError);
