From 25af5506ca2f27a89f62db8135ca6423edf78127 Mon Sep 17 00:00:00 2001
From: Matti Niemenmaa <matti.niemenmaa+git@iki.fi>
Date: Mon, 13 May 2024 17:43:15 +0300
Subject: [PATCH] meta/ubi_hx: Fix typing of chunk_size

---
 src/meta/ubi_hx.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/meta/ubi_hx.c b/src/meta/ubi_hx.c
index 76abcca4..021564d7 100644
--- a/src/meta/ubi_hx.c
+++ b/src/meta/ubi_hx.c
@@ -259,7 +259,8 @@ static int parse_header(ubi_hx_header* hx, STREAMFILE* sf, uint32_t offset, uint
     read_u32_t read_u32 = hx->big_endian ? read_u32be : read_u32le;
     read_s32_t read_s32 = hx->big_endian ? read_s32be : read_s32le;
     read_u16_t read_u16 = hx->big_endian ? read_u16be : read_u16le;
-    uint32_t riff_offset, riff_size, stream_adjust = 0, resource_size, chunk_size;
+    uint32_t riff_offset, riff_size, stream_adjust = 0, resource_size;
+    size_t chunk_size;
     off_t chunk_offset;
     int cue_flag = 0;
 
@@ -384,7 +385,7 @@ static int parse_header(ubi_hx_header* hx, STREAMFILE* sf, uint32_t offset, uint
             }
             else if ((flag_type == 0x01 || flag_type == 0x02) && /* Rayman M (not Arena) uses "data" instead */
                     find_chunk_riff_ve(sf, 0x61746164,riff_offset + 0x0c,riff_size - 0x0c, &chunk_offset,&chunk_size, hx->big_endian)) {
-                hx->stream_size     = chunk_size;
+                hx->stream_size     = (uint32_t)chunk_size;
                 hx->stream_offset   = read_u32(chunk_offset + 0x00, sf) + stream_adjust;
             }
             else {
@@ -401,7 +402,7 @@ static int parse_header(ubi_hx_header* hx, STREAMFILE* sf, uint32_t offset, uint
 
             if (chunk_size >  riff_size - (chunk_offset - riff_offset) || !chunk_size)
                 chunk_size = riff_size - (chunk_offset - riff_offset); /* just in case */
-            hx->stream_size     = chunk_size;
+            hx->stream_size     = (uint32_t)chunk_size;
         }
 
         /* can contain other RIFF stuff like "cue ", "labl" and "ump3"
-- 
2.45.0

