From 8fad54e2c1e3a978abf3c15a75014d8e5e24d226 Mon Sep 17 00:00:00 2001
From: KokaKiwi <kokakiwi+git@kokakiwi.net>
Date: Mon, 10 Jan 2022 15:47:10 +0100
Subject: [PATCH 1/3] makepkg: Remove external dependencies check

---
 CMakeLists.txt            |  1 -
 cmake/build_helpers.cmake | 11 -----------
 2 files changed, 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba7e9da..63268e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,6 @@ include("${CMAKE_SOURCE_DIR}/cmake/build_helpers.cmake")
 
 # Make sure project is configured correctly
 setDefaultBuiltTypeIfUnset()
-detectBadClone()
 
 # List plugin names here. Project name must match folder name
 set(PLUGINS
diff --git a/cmake/build_helpers.cmake b/cmake/build_helpers.cmake
index 8cbbdd9..d783265 100644
--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -266,14 +266,3 @@ macro(setDefaultBuiltTypeIfUnset)
         set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
     endif()
 endmacro()
-
-macro(detectBadClone)
-    file (GLOB EXTERNAL_DIRS "external/*")
-    foreach (EXTERNAL_DIR ${EXTERNAL_DIRS})
-        file(GLOB RESULT "${EXTERNAL_DIR}/*")
-        list(LENGTH RESULT ENTRY_COUNT)
-        if(ENTRY_COUNT LESS_EQUAL 1)
-            message(FATAL_ERROR "External dependency ${EXTERNAL_DIR} is empty!\nMake sure to correctly clone ImHex using the --recurse-submodules git option or initialize the submodules manually.")
-        endif()
-    endforeach ()
-endmacro()
\ No newline at end of file
-- 
2.35.1

