From deec944b599648b1e806a7db7f37821ca43fcba0 Mon Sep 17 00:00:00 2001
From: NyCode <nico@nycode.de>
Date: Thu, 19 Aug 2021 15:15:40 +0200
Subject: [PATCH] Add DESTDIR variable

---
 Makefile | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 815c27e..438ed74 100644
--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,18 @@
 CXX ?= clang
 
+DESTDIR =
+
 all: pridecat
 
 pridecat: main.cpp
 	$(CXX) main.cpp -o pridecat -std=c++11 -lstdc++ -Wall -Wextra -O3
 
 install: pridecat
-	cp pridecat /usr/local/bin/pridecat
+	mkdir -p ${DESTDIR}/usr/bin
+	cp pridecat ${DESTDIR}/usr/bin/pridecat
 
 uninstall:
-	rm -f /usr/local/bin/pridecat
+	rm -f ${DESTDIR}/usr/bin/pridecat
 
 clean:
 	rm -f pridecat
-- 
2.32.0

