From cc58ab8e4c5638947244b572593dfb58f6fd8036 Mon Sep 17 00:00:00 2001
From: Liao Junxuan <mikeljx@126.com>
Date: Tue, 11 Jul 2023 09:36:34 +0800
Subject: [PATCH 1/2] Include stdint.h to use intptr_t

---
 clients/drcachesim/tests/burst_replaceall.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clients/drcachesim/tests/burst_replaceall.cpp b/clients/drcachesim/tests/burst_replaceall.cpp
index ba075e8..49025dd 100644
--- a/clients/drcachesim/tests/burst_replaceall.cpp
+++ b/clients/drcachesim/tests/burst_replaceall.cpp
@@ -46,6 +46,7 @@
 #include <assert.h>
 #include <iostream>
 #include <math.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 #include <set>
-- 
2.41.0

