diff --git a/setup.py b/setup.py
index 17b13ba..83340af 100755
--- a/setup.py
+++ b/setup.py
@@ -42,12 +42,13 @@ setuptools.setup(
     package_data={
         "": ["*.pyxres", "*.png", "*.gif", "*.dylib", "*.dll", "*.so", "*.exe"]
     },
-    install_requires=["pyinstaller"],
+    install_requires=[],
+    extras_require={"packager": ["pyinstaller"]},
     python_requires=">=3.6.9",
     entry_points={
         "console_scripts": [
             "pyxeleditor=pyxel.editor:run",
-            "pyxelpackager=pyxel.packager:run",
+            "pyxelpackager=pyxel.packager:run [packager]",
             "install_pyxel_examples=pyxel.examples:install",
         ]
     },
