diff --git a/CMakeLists.txt b/CMakeLists.txt
index e165c19..7f46e22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,6 +346,8 @@ if (${SIM_PLATFORM} STREQUAL "Linux" OR ${SIM_PLATFORM} STREQUAL "Darwin")
     set(BLA_VENDOR OpenBLAS)
     find_package(BLAS REQUIRED)
     find_package(LAPACK REQUIRED)
+    # Find CBLAS; This populates 'OpenBLAS_INCLUDE_DIRS' which we use later
+    find_package(OpenBLAS CONFIG QUIET)
 endif()
 
 # What to link with the object files to produce an executable
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index c4491f6..34f135e 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -226,6 +226,7 @@ target_include_directories(simlib PUBLIC
     ${CMAKE_CURRENT_SOURCE_DIR}
     ${CMAKE_CURRENT_BINARY_DIR}
     ${FLEX_INCLUDE_DIRS}
+    ${OpenBLAS_INCLUDE_DIRS}
 )
 # target_compile_definitions(dlib PRIVATE BUILDING_SIM_DLL)
 
