*** firefox-132.0.orig/widget/gtk/GfxInfo.cpp	2024-10-21 21:51:24.000000000 +0100
--- firefox-132.0/widget/gtk/GfxInfo.cpp	2024-11-05 18:59:00.743987039 +0000
***************
*** 981,1017 ****
          nsIGfxInfo::FEATURE_BLOCKED_DEVICE, DRIVER_COMPARISON_IGNORED,
          V(0, 0, 0, 0), "FEATURE_FAILURE_WEBGL_MESA_VM", "");
  
-     ////////////////////////////////////
-     // FEATURE_WEBRENDER_COMPOSITOR
-     APPEND_TO_DRIVER_BLOCKLIST(
-         OperatingSystem::Linux, DeviceFamily::All,
-         nsIGfxInfo::FEATURE_WEBRENDER_COMPOSITOR,
-         nsIGfxInfo::FEATURE_BLOCKED_DEVICE, DRIVER_COMPARISON_IGNORED,
-         V(0, 0, 0, 0), "FEATURE_FAILURE_WEBRENDER_COMPOSITOR_DISABLED", "");
- 
-     ////////////////////////////////////
-     // FEATURE_X11_EGL
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::MesaAll, DeviceFamily::All,
-         nsIGfxInfo::FEATURE_X11_EGL, nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION,
-         DRIVER_LESS_THAN, V(17, 0, 0, 0), "FEATURE_X11_EGL_OLD_MESA",
-         "Mesa 17.0.0.0");
- 
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::MesaAll, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_X11_EGL, nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION,
-         DRIVER_LESS_THAN, V(18, 2, 0, 0), "FEATURE_X11_EGL_OLD_MESA_NOUVEAU",
-         "Mesa 18.2.0.0");
- 
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::NonMesaAll, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_X11_EGL, nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION,
-         DRIVER_LESS_THAN, V(470, 82, 0, 0),
-         "FEATURE_ROLLOUT_X11_EGL_NVIDIA_BINARY", "470.82.0");
- 
      // Disable on all AMD devices not using Mesa.
      APPEND_TO_DRIVER_BLOCKLIST_EXT(
          OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
--- 981,986 ----
***************
*** 1022,1034 ****
  
      ////////////////////////////////////
      // FEATURE_DMABUF
-     // Disabled due to high volume crash tracked in bug 1788573, fixed in the
-     // 545 driver.
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::NonMesaAll, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_DMABUF, nsIGfxInfo::FEATURE_BLOCKED_DEVICE,
-         DRIVER_LESS_THAN, V(545, 23, 6, 0), "FEATURE_FAILURE_BUG_1788573", "");
  
      // Disabled due to high volume crash tracked in bug 1913778. It appears that
      // only this version of the driver is affected.
--- 991,996 ----
***************
*** 1062,1075 ****
          V(21, 0, 0, 0), "FEATURE_HARDWARE_VIDEO_DECODING_MESA",
          "Mesa 21.0.0.0");
  
-     // Disable on all NVIDIA hardware
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::All, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_HARDWARE_VIDEO_DECODING,
-         nsIGfxInfo::FEATURE_BLOCKED_DEVICE, DRIVER_COMPARISON_IGNORED,
-         V(0, 0, 0, 0), "FEATURE_HARDWARE_VIDEO_DECODING_NO_LINUX_NVIDIA", "");
- 
      // Disable on all AMD devices not using Mesa.
      APPEND_TO_DRIVER_BLOCKLIST_EXT(
          OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
--- 1024,1029 ----
***************
*** 1119,1131 ****
          "FEATURE_HARDWARE_VIDEO_ZERO_COPY_LINUX_AMD_DISABLE", "Mesa 23.1.1.0");
  
      ////////////////////////////////////
-     // FEATURE_WEBRENDER_PARTIAL_PRESENT
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::X11, DriverVendor::NonMesaAll, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_WEBRENDER_PARTIAL_PRESENT,
-         nsIGfxInfo::FEATURE_BLOCKED_DEVICE, DRIVER_COMPARISON_IGNORED,
-         V(0, 0, 0, 0), "FEATURE_ROLLOUT_WR_PARTIAL_PRESENT_NVIDIA_BINARY", "");
  
      APPEND_TO_DRIVER_BLOCKLIST_EXT(
          OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
--- 1073,1078 ----
***************
*** 1144,1157 ****
          DRIVER_COMPARISON_IGNORED, V(0, 0, 0, 0),
          "FEATURE_FAILURE_THREADSAFE_GL_NOUVEAU", "");
  
-     // Disabled due to high volume crash tracked in bug 1788573, fixed in the
-     // 545 driver.
-     APPEND_TO_DRIVER_BLOCKLIST_EXT(
-         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-         WindowProtocol::All, DriverVendor::NonMesaAll, DeviceFamily::NvidiaAll,
-         nsIGfxInfo::FEATURE_THREADSAFE_GL, nsIGfxInfo::FEATURE_BLOCKED_DEVICE,
-         DRIVER_LESS_THAN, V(545, 23, 6, 0), "FEATURE_FAILURE_BUG_1788573", "");
- 
      // AMD R600 family does not perform well with WebRender.
      APPEND_TO_DRIVER_BLOCKLIST(
          OperatingSystem::Linux, DeviceFamily::AmdR600,
--- 1091,1096 ----
