diff -r 6af025613ee2 toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp	Fri May 01 04:28:35 2020 +0000
+++ b/toolkit/xre/nsAppRunner.cpp	Fri May 01 18:18:21 2020 -0300
@@ -3703,11 +3703,7 @@
   // consistently.
 
   // Set program name to the one defined in application.ini.
-  {
-    nsAutoCString program(gAppData->name);
-    ToLowerCase(program);
-    g_set_prgname(program.get());
-  }
+  g_set_prgname(gAppData->remotingName);
 
   // Initialize GTK here for splash.
 
diff -r 6af025613ee2 widget/gtk/nsAppShell.cpp
--- a/widget/gtk/nsAppShell.cpp	Fri May 01 04:28:35 2020 +0000
+++ b/widget/gtk/nsAppShell.cpp	Fri May 01 18:18:21 2020 -0300
@@ -24,6 +24,8 @@
 #  include "WakeLockListener.h"
 #endif
 #include "gfxPlatform.h"
+#include "nsAppRunner.h"
+#include "nsAppRunner.h"
 #include "ScreenHelperGTK.h"
 #include "HeadlessScreenHelper.h"
 #include "mozilla/widget/ScreenManager.h"
@@ -178,10 +180,8 @@
       // creating top-level windows. (At this point, a child process hasn't
       // received the list of registered chrome packages, so the
       // GetBrandShortName call would fail anyway.)
-      nsAutoString brandName;
-      mozilla::widget::WidgetUtils::GetBrandShortName(brandName);
-      if (!brandName.IsEmpty()) {
-        gdk_set_program_class(NS_ConvertUTF16toUTF8(brandName).get());
+      if (gAppData) {
+        gdk_set_program_class(gAppData->remotingName);
       }
     }
   }
