--- ddclient.orig2	2017-04-28 18:45:10.841664530 +0200
+++ ddclient	2018-03-26 17:54:21.574530003 +0200
@@ -25,6 +25,7 @@
 use Getopt::Long;
 use Sys::Hostname;
 use IO::Socket;
+use Regexp::IPv6 qw($IPv6_re);
 
 # my ($VERSION) = q$Revision: 184 $ =~ /(\d+)/;
 
@@ -2177,7 +2178,7 @@
 	$skip  =~ s/ /\\s/is;
     	$reply =~ s/^.*?${skip}//is;
     }
-    if ($reply =~ /.*? ([0-9:][^\/]*)/i) {
+    if ($reply =~ /.*?\b($IPv6_re)\b/i) {
 	$ip = $1;
     }
     if (($use ne 'ip') && (define($ip,'') eq '0.0.0.0')) {
