From 3935d459cc299e172e28e70d5ee8eb2d798ecbc0 Mon Sep 17 00:00:00 2001
From: Sandy Carter <bwrsandman@gmail.com>
Date: Sat, 30 Jan 2021 21:18:07 -0500
Subject: [PATCH 2/4] bimg: specify visibility of linked libs

---
 cmake/bimg.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/bimg.cmake b/cmake/bimg.cmake
index 56a047e..f403771 100644
--- a/cmake/bimg.cmake
+++ b/cmake/bimg.cmake
@@ -38,7 +38,7 @@ target_include_directories( bimg
 		$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
 # bimg dependencies
-target_link_libraries( bimg bx astc-codec astc edtaa3 etc1 etc2 iqa squish nvtt pvrtc )
+target_link_libraries( bimg PUBLIC bx PRIVATE astc-codec astc edtaa3 etc1 etc2 iqa squish nvtt pvrtc )
 
 # Put in a "bgfx" folder in Visual Studio
 set_target_properties( bimg PROPERTIES FOLDER "bgfx" )
\ No newline at end of file
-- 
2.30.0

