From 8323d748f52fe27ff1ba9fdc615ad23ff267db10 Mon Sep 17 00:00:00 2001
From: Jan Cholasta <grubber@grubber.cz>
Date: Thu, 18 Feb 2016 13:34:59 +0100
Subject: [PATCH] Include SHARE_DIR in IWADSearch.Directories

---
 src/gameconfigfile.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gameconfigfile.cpp b/src/gameconfigfile.cpp
index 3d396d9..c258b2d 100644
--- a/src/gameconfigfile.cpp
+++ b/src/gameconfigfile.cpp
@@ -133,6 +133,7 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", "$PROGDIR", true);
 #else
 		SetValueForKey ("Path", "~/" GAME_DIR, true);
+		SetValueForKey ("Path", SHARE_DIR, true);
 		// Arch Linux likes them in /usr/share/doom
 		// Debian likes them in /usr/share/games/doom
 		// I assume other distributions don't do anything radically different
-- 
2.7.1

