From d94409a4afdcdcc0f29fbf324eeacabc3843756a Mon Sep 17 00:00:00 2001
From: einekratzekatze <48450457+einekratzekatze@users.noreply.github.com>
Date: Mon, 25 Oct 2021 18:40:02 +0200
Subject: [PATCH 2/3] Update region.py

---
 nbt/region.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nbt/region.py b/nbt/region.py
index ada937a..d52b861 100644
--- a/nbt/region.py
+++ b/nbt/region.py
@@ -7,7 +7,7 @@ https://minecraft.gamepedia.com/Region_file_format
 
 from .nbt import NBTFile, MalformedFileError
 from struct import pack, unpack
-from collections import Mapping
+from typing import Mapping
 import zlib
 import gzip
 from io import BytesIO
-- 
2.34.1

