diff -ruN a/tests/SConstruct b/tests/SConstruct
--- a/tests/SConstruct	2023-04-10 21:09:07.963333339 -0400
+++ b/tests/SConstruct	2023-04-10 21:29:00.140000015 -0400
@@ -21,7 +21,7 @@
 tests = []
 for test in Glob('*Tests'):
     if str(test) in ('cryptoTests', 'httpTests', 'iostreamTests', 'serverTests'
-                     ) and not env.CBConfigEnabled('openssl'):
+                     ) and env.CBConfigEnabled('openssl'):
 
         for t in Glob('%s/*Test' % test):
             open('%s/disable' % t, 'w').close()
