From 4852593c91a5dad1a473c374b67e6a7a764f1a30 Mon Sep 17 00:00:00 2001
From: Jasper van Bourgognie <van.bourgognie@gmail.com>
Date: Tue, 8 Apr 2025 20:51:51 +0200
Subject: [PATCH 2/2] enable 3fg-drag by default

---
 src/evdev-mt-touchpad-gestures.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/evdev-mt-touchpad-gestures.c b/src/evdev-mt-touchpad-gestures.c
index 39880eff..a4108d83 100644
--- a/src/evdev-mt-touchpad-gestures.c
+++ b/src/evdev-mt-touchpad-gestures.c
@@ -2147,7 +2147,7 @@ tp_3fg_drag_get_enabled(struct libinput_device *device)
 static enum libinput_config_3fg_drag_state
 tp_3fg_drag_default(struct tp_dispatch *tp)
 {
-	return LIBINPUT_CONFIG_3FG_DRAG_DISABLED;
+	return LIBINPUT_CONFIG_3FG_DRAG_ENABLED_3FG;
 }
 
 static enum libinput_config_3fg_drag_state
-- 
2.47.1

