From 7f226438d12bf12e65f4e0a14173cab752266a91 Mon Sep 17 00:00:00 2001
From: rechie <502117806@qq.com>
Date: Thu, 12 Jun 2025 22:59:28 +0800
Subject: [PATCH 2/2] fix #ICETPG

---
 src/conf2/api/libkysettings.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/conf2/api/libkysettings.c b/src/conf2/api/libkysettings.c
index fe9c288..4f9b6d3 100644
--- a/src/conf2/api/libkysettings.c
+++ b/src/conf2/api/libkysettings.c
@@ -1566,7 +1566,7 @@ char **kdk_conf2_list_versions(char *app)
     AppData app_data;
     app_data.name = app;
     app_data.default_version = NULL;
-    GHashTable* app_hash = (VersionData*)g_hash_table_lookup(schemas_table, (gconstpointer)(&app_data));
+    GHashTable* app_hash = (GHashTable*)g_hash_table_lookup(schemas_table, (gconstpointer)(&app_data));
     if (NULL == app_hash)
         return NULL;
 
@@ -1578,7 +1578,7 @@ char **kdk_conf2_list_versions(char *app)
     result = calloc(length, sizeof(char*));
     for (guint i = 0; i < length; i++)
     {
-        VersionData *version = (AppData*)curr->data;
+        VersionData *version = (VersionData*)curr->data;
         result[i] = strdup(version->name);
         curr = curr->next;
     }
-- 
2.49.0

