diff --git a/build/unix/ar2/Makefile b/build/unix/ar2/Makefile
index dd8660e..390e75a 100644
--- a/build/unix/ar2/Makefile
+++ b/build/unix/ar2/Makefile
@@ -9,7 +9,7 @@ BINARY	= ar2
 OBJS	= ar.o arsup.o lz1.o o2u.o
 
 $(BINARY)$(SUFEXE):	$(OBJS)
-	$(CC) $(OBJS) -o $@ $(DEBUG)
+	$(CC) $(OBJS) -o $@ $(DEBUG) $(LDFLAGS)
 
 $(OBJS):
 
diff --git a/build/unix/makewav/Makefile b/build/unix/makewav/Makefile
index feccbc3..141c2d2 100644
--- a/build/unix/makewav/Makefile
+++ b/build/unix/makewav/Makefile
@@ -10,7 +10,7 @@ OBJS	= makewav.o
 LIBS	= -lm
 
 $(BINARY)$(SUFEXE):	$(OBJS)
-	$(CC) $(OBJS) $(LIBS) -o $@
+	$(CC) $(OBJS) $(LIBS) -o $@ $(LDFLAGS)
 
 clean:
 	$(RM) $(BINARY) $(BINARY).exe *.o
