From 4068889786749f9947409fd5f8de171d4596f51b Mon Sep 17 00:00:00 2001
From: rsp4jack <rsp4jack@outlook.com>
Date: Sun, 17 Aug 2025 18:04:17 +0800
Subject: [PATCH 2/3] Fix Wayland detection

---
 src/main.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 01d1e0f..c9689d4 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -6,6 +6,7 @@
 
 #include <plugin.h>
 #include <main.h>
+#include <core/outputbackend.h>
 #include "zones.h"
 
 using namespace KWin;
@@ -24,7 +25,10 @@ public:
 
 std::unique_ptr<KWin::Plugin> KWinZonesFactory::create() const
 {
-    return std::make_unique<Zones>();
+    if (static_cast<QObject*>(kwinApp()->outputBackend())->inherits("WaylandBackend") || static_cast<QObject*>(kwinApp()->outputBackend())->inherits("DrmBackend")) {
+    	return std::make_unique<Zones>();
+    }
+    return nullptr;
 }
 
 #include "main.moc"
-- 
2.50.1

