From 07ed900684af0e180d7370240a45ee7cc7bdfdce Mon Sep 17 00:00:00 2001
From: Brian Bidulock <bidulock@openss7.org>
Date: Sun, 29 May 2016 20:48:19 -0600
Subject: [PATCH 2/3] more gtk2 reverts

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index b58eeb1..500cb73 100644
--- a/configure.ac
+++ b/configure.ac
@@ -259,6 +259,7 @@ Build options:
   Build shared libs         $enable_shared
   Build static libs         $enable_static
   CFLAGS                    $CFLAGS
+  GTK version               $with_gtk
   Build Vala binding        $enable_vala
   Sound support             $enable_libcanberra
   Build document            $enable_gtk_doc
-- 
2.8.3

