From e6da3e24d0e80fa3e2065edc9bcbacd15807f6eb Mon Sep 17 00:00:00 2001
From: Noel Kuntze <noel.kuntze@thermi.consulting>
Date: Sun, 10 Jul 2022 18:08:58 +0200
Subject: [PATCH 2/2] lsyncd.h: #define _BSD_SOURCE 1, #define LUA_COMPAT_5_3

---
 lsyncd.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lsyncd.h b/lsyncd.h
index 9a4ec67..9ea6f72 100644
--- a/lsyncd.h
+++ b/lsyncd.h
@@ -19,6 +19,7 @@
 
 #define LUA_COMPAT_ALL
 #define LUA_COMPAT_5_1
+#define LUA_COMPAT_5_3
 
 // includes needed for headerfile
 #include "config.h"
-- 
2.37.0

