From c4300c7c12fd1c218ac9c87ca35df3b37a82e0dc Mon Sep 17 00:00:00 2001
From: Dan Elkouby <streetwalkermc@gmail.com>
Date: Fri, 27 Dec 2019 15:21:31 +0200
Subject: [PATCH 2/2] Don't send wl_pointer_leave when hide_cursor activates

---
 sway/input/cursor.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sway/input/cursor.c b/sway/input/cursor.c
index 680fe39e..86650f08 100644
--- a/sway/input/cursor.c
+++ b/sway/input/cursor.c
@@ -176,7 +176,6 @@ void cursor_rebase_all(void) {
 static void cursor_hide(struct sway_cursor *cursor) {
 	wlr_cursor_set_image(cursor->cursor, NULL, 0, 0, 0, 0, 0, 0);
 	cursor->hidden = true;
-	wlr_seat_pointer_clear_focus(cursor->seat->wlr_seat);
 }
 
 static int hide_notify(void *data) {
-- 
2.24.1

