From d1baa946929673abcaa3e8ed8103a6a37baae5d9 Mon Sep 17 00:00:00 2001
From: Andrew Apted <ajapted@gmail.com>
Date: Tue, 9 Jan 2018 15:57:35 +1100
Subject: [PATCH] Makefile : fixed "make install" not installing data/bg/*.*

---
 Makefile       | 2 ++
 Makefile.macos | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 2588f2082..27a1179da 100644
--- a/Makefile
+++ b/Makefile
@@ -248,8 +248,10 @@ install: stripped
 	install -o root -m 644 language/*.*  $(SCRIPT_DIR)/language
 	#
 	install -d $(SCRIPT_DIR)/data
+	install -d $(SCRIPT_DIR)/data/bg
 	install -d $(SCRIPT_DIR)/data/masks
 	install -o root -m 644 data/*.* $(SCRIPT_DIR)/data
+	install -o root -m 644 data/bg/*.* $(SCRIPT_DIR)/data/bg
 	install -o root -m 644 data/masks/*.* $(SCRIPT_DIR)/data/masks
 	#
 	rm -Rf $(SCRIPT_DIR)/games
diff --git a/Makefile.macos b/Makefile.macos
index e1ceee58a..d7994e451 100644
--- a/Makefile.macos
+++ b/Makefile.macos
@@ -275,8 +275,10 @@ install: stripped
 	install -o root -m 644 language/*.*  $(SCRIPT_DIR)/language
 	#
 	install -d $(SCRIPT_DIR)/data
+	install -d $(SCRIPT_DIR)/data/bg
 	install -d $(SCRIPT_DIR)/data/masks
 	install -o root -m 644 data/*.* $(SCRIPT_DIR)/data
+	install -o root -m 644 data/bg/*.* $(SCRIPT_DIR)/data/bg
 	install -o root -m 644 data/masks/*.* $(SCRIPT_DIR)/data/masks
 	#
 	rm -Rf $(SCRIPT_DIR)/games
-- 
2.16.1

