diff --git OrcaSlicer/src/clipper2/Clipper2Lib/include/clipper2/clipper.core.h OrcaSlicer/src/clipper2/Clipper2Lib/include/clipper2/clipper.core.h
index f8d6b42b7..a41bae5ce 100644
--- OrcaSlicer/src/clipper2/Clipper2Lib/include/clipper2/clipper.core.h
+++ OrcaSlicer/src/clipper2/Clipper2Lib/include/clipper2/clipper.core.h
@@ -73,7 +73,7 @@ struct Point {
 	}

 	template <typename T2>
-	explicit Point<T>(const Point<T2>& p)
+	explicit Point(const Point<T2>& p)
 	{
 		Init(p.x, p.y, p.z);
 	}
@@ -114,7 +114,7 @@ struct Point {
 	Point(const T2 x_, const T2 y_) { Init(x_, y_); }

 	template <typename T2>
-	explicit Point<T>(const Point<T2>& p) { Init(p.x, p.y); }
+	explicit Point(const Point<T2>& p) { Init(p.x, p.y); }

 	Point operator * (const double scale) const
 	{
