diff --git a/drivers/net/wireless/rtl8812au/core/rtw_sta_mgt.c b/drivers/net/wireless/rtl8812au/core/rtw_sta_mgt.c
index 73e22cb0ffe8..3bb7f241065c 100644
--- a/drivers/net/wireless/rtl8812au/core/rtw_sta_mgt.c
+++ b/drivers/net/wireless/rtl8812au/core/rtw_sta_mgt.c
@@ -207,9 +207,6 @@ void rtw_mfree_stainfo(struct sta_info *psta)
 {
 	_func_enter_;
 
-	if(&psta->lock != NULL)
-		_rtw_spinlock_free(&psta->lock);
-
 	_rtw_free_sta_xmit_priv_lock(&psta->sta_xmitpriv);
 	_rtw_free_sta_recv_priv_lock(&psta->sta_recvpriv);
 
