From f2ad417ec3260fdad18638d48bd1a26d21535287 Mon Sep 17 00:00:00 2001
From: 7Ji <pugokushin@gmail.com>
Date: Wed, 13 Apr 2022 19:50:38 +0800
Subject: [PATCH 2/2] GNUmakefile: fix install dir to /usr

---
 GNUmakefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index a27b453..027b81e 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -35,7 +35,7 @@ PACKAGE_VERSION = 1.4.12
 
 # We use GNU makefile conventions for directory names.
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
 sbindir     = $(exec_prefix)/sbin
-- 
2.35.1

