From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marko Lemmetty <marko.lemmetty@jollamobile.com>
Date: Fri, 15 May 2020 14:33:17 +0300
Subject: [PATCH] Support synchronous copyproj

---
 osc/commandline.py | 5 +++++
 osc/core.py        | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/osc/commandline.py b/osc/commandline.py
index cae02268cacca43aa7db97ace9683b841afbfe2c..481fb1c2090aa76e84aaac6e56bf74ab3f2cf811 100644
--- a/osc/commandline.py
+++ b/osc/commandline.py
@@ -3410,6 +3410,8 @@ Please submit there instead, or use --nodevelproject to force direct submission.
         )
     @cmdln.option('-b', '--with-binaries', action='store_true',
                         help='copy the built binaries over to avoid a rebuild')
+    @cmdln.option('-n', '--now', action='store_true',
+                        help='copy synchronously (may take some time to complete)')
     @cmdln.option('-x', '--with-history', action='store_true',
                         help='replicate the history of each package.')
     @cmdln.option('-o', '--make-older', action='store_true',
@@ -3427,6 +3429,8 @@ Please submit there instead, or use --nodevelproject to force direct submission.
 
         The user must be able to create DESTPRJ
 
+        Normally the copy is done asynchronously
+
         usage:
             osc copyprj SOURCEPRJ DESTPRJ
         ${cmd_option_list}
@@ -3456,6 +3460,7 @@ Please submit there instead, or use --nodevelproject to force direct submission.
                      withhistory = opts.with_history,
                      makeolder = opts.make_older,
                      resign = opts.re_sign,
+                     now = opts.now,
                      comment = comment)
         print("done cp")
         print(r)
diff --git a/osc/core.py b/osc/core.py
index fc6410974f57815ac36abc7232972da7d925b82a..83123b240217f6b02bae9447b471e424d2bc9266 100644
--- a/osc/core.py
+++ b/osc/core.py
@@ -6026,6 +6026,8 @@ def copy_prj(src_apiurl, src_project, dst_project,
         query['resign'] = '1'
     if comment:
         query['comment'] = comment
+    if now:
+        query['nodelay'] = '1'
     u = makeurl(src_apiurl, ['source', dst_project], query=query)
     print("copyprj ", u, file=sys.stderr)
     f = http_POST(u)
