From e12830055ce45b4271b7c65686c98622dd36e3c7 Mon Sep 17 00:00:00 2001
From: Aviana Cruz <gwencroft@proton.me>
Date: Wed, 23 Nov 2022 09:27:28 +0800
Subject: [PATCH 3/3] Disable Pack200

Signed-off-by: Aviana Cruz <gwencroft@proton.me>
---
 HMCL/build.gradle.kts | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/HMCL/build.gradle.kts b/HMCL/build.gradle.kts
index ecc6d792..f62e1437 100644
--- a/HMCL/build.gradle.kts
+++ b/HMCL/build.gradle.kts
@@ -176,7 +176,6 @@ tasks.getByName<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar>("sha
     }
 
     doLast {
-        repack(jarPath) // see repack()
         attachSignature(jarPath)
         createChecksum(jarPath)
     }
@@ -223,15 +222,12 @@ tasks.processResources {
     dependsOn(rootProject.tasks["generateOpenJFXDependencies"])
 }
 
-val packFile = File(jarPath.parentFile, jarPath.nameWithoutExtension + ".pack")
+val packFile = jarPath
 
 val makePack = tasks.create("makePack") {
     dependsOn(tasks.jar)
 
     doLast {
-        packFile.outputStream().use { out ->
-            JarFile(jarPath).use { jarFile -> packer.pack(jarFile, out) }
-        }
         createChecksum(packFile)
     }
 }
-- 
2.38.1

