diff --git a/base/platform/linux/base_system_media_controls_linux.cpp b/base/platform/linux/base_system_media_controls_linux.cpp
index 4ffa4e8..20abc3c 100644
--- a/base/platform/linux/base_system_media_controls_linux.cpp
+++ b/base/platform/linux/base_system_media_controls_linux.cpp
@@ -184,8 +184,9 @@ SystemMediaControls::Private::Private()
 }) {
 	set_can_quit(true);
 	set_can_raise(!::Platform::IsWayland());
-	set_desktop_entry(QGuiApplication::desktopFileName().toStdString());
-	set_identity(QGuiApplication::desktopFileName().toStdString());
+	set_desktop_entry(
+		QGuiApplication::desktopFileName().chopped(8).toStdString());
+	set_identity(QGuiApplication::desktopFileName().chopped(8).toStdString());
 	player().set_can_control(true);
 	player().set_can_seek(true);
 	player().set_maximum_rate(1.0);
diff --git a/base/platform/linux/base_url_scheme_linux.cpp b/base/platform/linux/base_url_scheme_linux.cpp
index a2efd81..06258c1 100644
--- a/base/platform/linux/base_url_scheme_linux.cpp
+++ b/base/platform/linux/base_url_scheme_linux.cpp
@@ -121,10 +121,9 @@ void RegisterUrlScheme(const UrlSchemeDescriptor &descriptor) {
 		"--",
 	}).toStdString();
 
-	const auto appId = QGuiApplication::desktopFileName().toStdString();
-	if (!appId.empty()) {
-		if (const auto appInfo = Gio::DesktopAppInfo::create(
-			appId + ".desktop")) {
+	const auto desktopId = QGuiApplication::desktopFileName().toStdString();
+	if (!desktopId.empty()) {
+		if (const auto appInfo = Gio::DesktopAppInfo::create(desktopId)) {
 			if (appInfo->get_commandline() == commandlineForCreator + " %u") {
 				appInfo->set_as_default_for_type(handlerType);
 				return;
