From 8ad450a37365799af3259d070a07405b847650b4 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Thu, 9 Jul 2020 17:55:08 +0530
Subject: [PATCH 3/4] interfaces: Remove unused win32 code

---
 agent/interfaces.c | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/agent/interfaces.c b/agent/interfaces.c
index 309ca65..9a5300e 100644
--- a/agent/interfaces.c
+++ b/agent/interfaces.c
@@ -497,42 +497,6 @@ nice_interfaces_get_ip_for_interface (gchar *interface_name)
 #define MIB_IPADDR_DELETED 0x0040
 #endif
 
-#if 0
-static gboolean started_wsa_engine = FALSE;
-
-/*
- * private function that initializes the WinSock engine and
- *  returns a prebuilt socket
- */
-SOCKET nice_interfaces_get_WSA_socket ()
-{
-  WORD wVersionRequested;
-  WSADATA wsaData;
-  int err;
-  SOCKET sock;
-
-  if (started_wsa_engine == FALSE) {
-    wVersionRequested = MAKEWORD ( 2, 0 );
-
-    err = WSAStartup ( wVersionRequested, &wsaData );
-    if ( err != 0 ) {
-      nice_debug ("Error : Could not start the winsocket engine");
-      return INVALID_SOCKET;
-    }
-    started_wsa_engine = TRUE;
-  }
-
-
-  if ((sock = socket (AF_INET, SOCK_DGRAM, 0)) == INVALID_SOCKET) {
-    nice_debug ("Error : Could not open socket to retrieve interface list,"
-        " error no : %d", WSAGetLastError ());
-    return INVALID_SOCKET;
-  }
-
-  return sock;
-}
-#endif
-
 GList * nice_interfaces_get_local_interfaces (void)
 {
   ULONG size = 0;
-- 
2.27.0.windows.1

