From c13db8fcdf05027ac9ed27780004d0c03ecf6198 Mon Sep 17 00:00:00 2001
From: Jan Cholasta <jcholast@redhat.com>
Date: Tue, 2 Aug 2016 13:49:36 +0200
Subject: [PATCH 3/3] client install: do not assume /etc/krb5.conf.d exists

---
 client/ipa-client-install | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/client/ipa-client-install b/client/ipa-client-install
index 4a263b3..ea71076 100755
--- a/client/ipa-client-install
+++ b/client/ipa-client-install
@@ -1073,8 +1073,10 @@ def configure_krb5_conf(cli_realm, cli_domain, cli_server, cli_kdc, dnsok,
     krbconf.setIndent(("","  ","    "))
 
     opts = [{'name':'comment', 'type':'comment', 'value':'File modified by ipa-client-install'},
-            {'name':'empty', 'type':'empty'},
-            {'name':'includedir', 'type':'option', 'value':paths.COMMON_KRB5_CONF_DIR, 'delim':' '}]
+            {'name':'empty', 'type':'empty'}]
+
+    if os.path.exists(paths.COMMON_KRB5_CONF_DIR):
+        opts.append({'name':'includedir', 'type':'option', 'value':paths.COMMON_KRB5_CONF_DIR, 'delim':' '})
 
     # SSSD include dir
     if options.sssd:
-- 
2.7.4

