diff --git a/drivers/net/wireless/rtl8812au/os_dep/linux/ioctl_cfg80211.c b/drivers/net/wireless/rtl8812au/os_dep/linux/ioctl_cfg80211.c
index 02072eddf4ae..bcdc5886aa48 100644
--- a/drivers/net/wireless/rtl8812au/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/rtl8812au/os_dep/linux/ioctl_cfg80211.c
@@ -2166,8 +2166,7 @@ static int cfg80211_rtw_scan(struct wiphy *wiphy
 
 #ifdef CONFIG_P2P
 	if( pwdinfo->driver_interface == DRIVER_CFG80211 ) {
-		if(ssids->ssid != NULL
-		   && _rtw_memcmp(ssids->ssid, "DIRECT-", 7)
+		if(_rtw_memcmp(ssids->ssid, "DIRECT-", 7)
 		   && rtw_get_p2p_ie((u8 *)request->ie, request->ie_len, NULL, NULL)
 		  ) {
 			if(rtw_p2p_chk_state(pwdinfo, P2P_STATE_NONE)) {
