From cd68269a531ce4e600f3a99d1687219e0959af84 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sat, 17 Sep 2016 16:51:15 +0200
Subject: [PATCH 3/3] Use fopen rather than fopen_s for XP compatibility

---
 libmariadb/dbug.c     | 4 ----
 libmariadb/my_fopen.c | 4 ----
 2 files changed, 8 deletions(-)

diff --git a/libmariadb/dbug.c b/libmariadb/dbug.c
index cd4a0f1..abca4ae 100644
--- a/libmariadb/dbug.c
+++ b/libmariadb/dbug.c
@@ -2051,11 +2051,7 @@ static void DBUGOpenFile(CODE_STATE *cs,
       }
       else
       {
-#ifdef _WIN32
-        if (fopen_s(&fp, name, append ? "a+" : "w"))
-#else
         if (!(fp= fopen(name, append ? "a+" : "w")))
-#endif
         {
           (void) fprintf(stderr, ERR_OPEN, cs->process, name);
           perror("");
diff --git a/libmariadb/my_fopen.c b/libmariadb/my_fopen.c
index c99b753..a3b485d 100644
--- a/libmariadb/my_fopen.c
+++ b/libmariadb/my_fopen.c
@@ -36,11 +36,7 @@ FILE *my_fopen(const char *FileName, int Flags, myf MyFlags)
 		   FileName, Flags, MyFlags));
 
   make_ftype(type,Flags);
-#ifdef _WIN32
-  if (fopen_s(&fd, FileName, type) == 0)
-#else
   if ((fd = fopen(FileName, type)) != 0)
-#endif
   {
     /*
       The test works if MY_NFILE < 128. The problem is that fileno() is char
-- 
2.13.0

