diff -pNaru5 a/opal/sipep.cxx b/opal/sipep.cxx
--- a/opal/sipep.cxx	2024-02-09 15:12:44.609740908 -0500
+++ b/opal/sipep.cxx	2024-02-09 15:13:09.949977096 -0500
@@ -495,11 +495,11 @@ bool MySIPEndPoint::Initialise(PArgList
         params.m_compatibility = SIPRegister::e_FullyCompliant;
         params.m_userData = NULL;
 
         if (prms.GetSize() >= 2) {
           params.m_password = prms[1];
-          output << " Password: " << params.m_password;
+          output << " Password: (" << params.m_password.GetLength() << ")";
 
           if (prms.GetSize() >= 3) {
             params.m_contactAddress = prms[2];
             output << " Contact: " << params.m_contactAddress;
 
