From d9b75bdad619c0ed17416fcd1514960dd621af8a Mon Sep 17 00:00:00 2001
From: Felix Schindler <aur@felixschindler.net>
Date: Sat, 4 Aug 2018 14:41:50 +0000
Subject: [PATCH] disable ArrayRefTest.InitializerList

---
 unittests/ADT/ArrayRefTest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unittests/ADT/ArrayRefTest.cpp b/unittests/ADT/ArrayRefTest.cpp
index b5b71f06..1d2101b4 100644
--- a/unittests/ADT/ArrayRefTest.cpp
+++ b/unittests/ADT/ArrayRefTest.cpp
@@ -133,7 +133,7 @@ static void ArgTest12(ArrayRef<int> A) {
   EXPECT_EQ(2, A[1]);
 }
 
-TEST(ArrayRefTest, InitializerList) {
+TEST(ArrayRefTest, DISABLED_InitializerList) {
   ArrayRef<int> A = { 0, 1, 2, 3, 4 };
   for (int i = 0; i < 5; ++i)
     EXPECT_EQ(i, A[i]);
-- 
2.18.0

