From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Rosenauer <wr@rosenauer.org>
Date: Sun, 27 Dec 2020 10:48:12 +0100
Subject: [PATCH] mozilla-libavcodec58_91.patch

---
 dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp b/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
index 6a7bdbce11ae51217a58718b05bc972ae351e1e8..01786d709bcb41f4d2f1d3273a92c6fb743b3624 100644
--- a/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
+++ b/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
@@ -49,6 +49,8 @@ static const char* sLibs[] = {
   "libavcodec.so.61",
   "libavcodec.so.60",
   "libavcodec.so.59",
+  "libavcodec.so.58.134",
+  "libavcodec.so.58.91",
   "libavcodec.so.58",
   "libavcodec-ffmpeg.so.58",
   "libavcodec-ffmpeg.so.57",
