From 08fc0e139509a6755489af3936ede49936896122 Mon Sep 17 00:00:00 2001
From: Andrew Comminos <andrew@comminos.com>
Date: Tue, 6 Mar 2018 02:07:31 -0800
Subject: [PATCH 4/9] wined3d: Use GL_CLIENT_STORAGE_BIT for persistent
 mappings.

---
 dlls/wined3d/buffer_heap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/wined3d/buffer_heap.c b/dlls/wined3d/buffer_heap.c
index b133bd6893..75f84b0088 100644
--- a/dlls/wined3d/buffer_heap.c
+++ b/dlls/wined3d/buffer_heap.c
@@ -169,7 +169,7 @@ HRESULT wined3d_buffer_heap_create(struct wined3d_context *context, GLsizeiptr s
     {
         access_flags |= GL_MAP_READ_BIT;
     }
-    storage_flags = access_flags;
+    storage_flags = GL_CLIENT_STORAGE_BIT | access_flags;
 
     // TODO(acomminos): where should we be checking for errors here?
     GL_EXTCALL(glGenBuffers(1, &object->buffer_object));
-- 
2.16.2

