From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marko Lemmetty <marko.lemmetty@jollamobile.com>
Date: Fri, 15 May 2020 14:47:19 +0300
Subject: [PATCH] Add -p to copyprj to enable copying of prjconf

---
 osc/commandline.py | 3 +++
 osc/core.py        | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/osc/commandline.py b/osc/commandline.py
index 584d896e83d6eb2b3db44bf8621168a7ec16a6c3..83eddac6851c23926f692e2b91946d29d74adb6e 100644
--- a/osc/commandline.py
+++ b/osc/commandline.py
@@ -2121,6 +2121,8 @@ class Osc(cmdln.Cmdln):
                         help='replicate the history of each package.')
     @cmdln.option('-o', '--make-older', action='store_true',
                         help='No idea')
+    @cmdln.option('-p', '--prjconf', action='store_true',
+                        help='copy the prjconf also')
     @cmdln.option('-r', '--re-sign', action='store_true',
                         help='re-sign the binaries')
     @cmdln.option('-m', '--message', metavar='TEXT',
@@ -2163,6 +2165,7 @@ class Osc(cmdln.Cmdln):
         r = copy_prj(src_apiurl, src_project, dst_project,
                      withbinaries = opts.with_binaries,
                      withhistory = opts.with_history,
+                     withprjconf = opts.prjconf,
                      makeolder = opts.make_older,
                      resign = opts.re_sign,
                      now = opts.now,
diff --git a/osc/core.py b/osc/core.py
index 437d49d4337628e3cc2575b87a77edde2d47a038..88a9080032560ed3e8bc0b2c70bf70361e60878b 100644
--- a/osc/core.py
+++ b/osc/core.py
@@ -3801,8 +3801,10 @@ def copy_pac(
 def copy_prj(src_apiurl, src_project, dst_project,
              withbinaries = False,
              withhistory = False,
+             withprjconf = False,
              makeolder = False,
              resign = False,
+             now = False,
              comment = None):
     """
     Create a copy of a project.
@@ -3816,6 +3818,8 @@ def copy_prj(src_apiurl, src_project, dst_project,
         query['withbinaries'] = '1'
     if withhistory:
         query['withhistory'] = '1'
+    if withprjconf:
+        query['withprjconf'] = '1'
     if makeolder:
         query['makeolder'] = '1'
     if resign:
@@ -3824,6 +3828,7 @@ def copy_prj(src_apiurl, src_project, dst_project,
         query['comment'] = comment
     if now:
         query['nodelay'] = '1'
+
     u = makeurl(src_apiurl, ['source', dst_project], query=query)
     print("copyprj ", u, file=sys.stderr)
     f = http_POST(u)
