diff --git a/drivers/net/wireless/rtl8812au/hal/OUTSRC/phydm_debug.c b/drivers/net/wireless/rtl8812au/hal/OUTSRC/phydm_debug.c
index 9fd82863b394..422e3d34b9d2 100644
--- a/drivers/net/wireless/rtl8812au/hal/OUTSRC/phydm_debug.c
+++ b/drivers/net/wireless/rtl8812au/hal/OUTSRC/phydm_debug.c
@@ -870,12 +870,10 @@ phydm_cmd_parser(
 	case PHYDM_RA:
 
 		for(i=0; i<5; i++) {
-			if(input[i+1]) {
-				PHYDM_SSCANF(input[i+1], DCMD_DECIMAL, &var1[i]);
+			PHYDM_SSCANF(input[i+1], DCMD_DECIMAL, &var1[i]);
 
-				PHYDM_SNPRINTF((output+used, out_len-used, "new SET, RA_var[%d]= (( %d ))\n", i , var1[i]));
-				input_idx++;
-			}
+			PHYDM_SNPRINTF((output+used, out_len-used, "new SET, RA_var[%d]= (( %d ))\n", i , var1[i]));
+			input_idx++;
 		}
 
 		if(input_idx>=1) {
@@ -891,12 +889,10 @@ phydm_cmd_parser(
 	case PHYDM_PATHDIV:
 
 		for(i=0; i<5; i++) {
-			if(input[i+1]) {
-				PHYDM_SSCANF(input[i+1], DCMD_HEX, &var1[i]);
+			PHYDM_SSCANF(input[i+1], DCMD_HEX, &var1[i]);
 
-				PHYDM_SNPRINTF((output+used, out_len-used, "new SET, PATHDIV_var[%d]= (( %d ))\n", i , var1[i]));
-				input_idx++;
-			}
+			PHYDM_SNPRINTF((output+used, out_len-used, "new SET, PATHDIV_var[%d]= (( %d ))\n", i , var1[i]));
+			input_idx++;
 		}
 
 		if(input_idx>=1) {
