--- Makefile.am.bak	2022-05-02 10:32:23.388685529 +0200
+++ Makefile.am	2022-05-02 10:33:34.876535389 +0200
@@ -154,7 +154,7 @@
 
 bin_SCRIPTS = curl-config
 
-SUBDIRS = lib src
+SUBDIRS = lib
 DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
 
 pkgconfigdir = $(libdir)/pkgconfig
@@ -288,7 +288,6 @@
 
 checksrc:
 	(cd lib && $(MAKE) checksrc)
-	(cd src && $(MAKE) checksrc)
 	(cd tests && $(MAKE) checksrc)
 	(cd include/curl && $(MAKE) checksrc)
 	(cd docs/examples && $(MAKE) checksrc)
@@ -606,5 +605,4 @@
 		"$$awk_code" $(srcdir)/$(VC14_30_SRCTMPL) > $(VC14_30_SRCVCXPROJ) || { exit 1; };)
 
 tidy:
-	(cd src && $(MAKE) tidy)
 	(cd lib && $(MAKE) tidy)
