From f98a18521525ebfcc3d481a6c2a6e81c4569fc0f Mon Sep 17 00:00:00 2001
From: howetuft <howetuft@gmail.com>
Date: Sat, 21 May 2022 18:18:23 +0200
Subject: [PATCH 05/13] clang-isnan-isinf

---
 include/luxrays/utils/utils.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/luxrays/utils/utils.h b/include/luxrays/utils/utils.h
index b6584b383..d48d7d93d 100644
--- a/include/luxrays/utils/utils.h
+++ b/include/luxrays/utils/utils.h
@@ -48,6 +48,11 @@ template <class T>
 int isinf(T a) { return std::isinf(a); }
 #endif
 
+#ifdef __clang__
+using std::isnan;
+using std::isinf;
+#endif
+
 #if defined(__APPLE__)
 #include <string>
 #endif
-- 
2.38.1

