From f482b19b44a708541c3300a30f8ebd58b9a0aa55 Mon Sep 17 00:00:00 2001
From: Vasiliy Stelmachenok <cabopust@yandex.ru>
Date: Sat, 17 Apr 2021 13:36:55 +0300
Subject: [PATCH] Nvidia Max Performance Powermizer without xorg.conf

Signed-off-by: Vasiliy Stelmachenok <cabopust@yandex.ru>

diff --git a/kernel/nvidia/nv-reg.h b/kernel/nvidia/nv-reg.h
index 7ea5acf..a3246e5 100644
--- a/kernel/nvidia/nv-reg.h
+++ b/kernel/nvidia/nv-reg.h
@@ -804,7 +804,7 @@ NV_DEFINE_REG_ENTRY_GLOBAL(__NV_ENABLE_PCIE_RELAXED_ORDERING_MODE, 0);
 NV_DEFINE_REG_ENTRY_GLOBAL(__NV_REGISTER_PCI_DRIVER, 1);
 
 
-NV_DEFINE_REG_STRING_ENTRY(__NV_REGISTRY_DWORDS, NULL);
+NV_DEFINE_REG_STRING_ENTRY(__NV_REGISTRY_DWORDS, "PowerMizerEnable=0x1;PerfLevelSrc=0x3333;PowerMizerDefaultAC=0x1");
 NV_DEFINE_REG_STRING_ENTRY(__NV_REGISTRY_DWORDS_PER_DEVICE, NULL);
 NV_DEFINE_REG_STRING_ENTRY(__NV_RM_MSG, NULL);
 NV_DEFINE_REG_STRING_ENTRY(__NV_GPU_BLACKLIST, NULL);
-- 
2.31.0

