From 112c2ecc87cc4769c0c4b31a7d33685bcb7bac99 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Thu, 9 Jul 2020 18:26:09 +0200
Subject: [PATCH 5/5] Arch Linux kernel v5.7.8-arch1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6163d607ca72..5112988d9e10 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 7
 SUBLEVEL = 8
-EXTRAVERSION =
+EXTRAVERSION = -arch1
 NAME = Kleptomaniac Octopus
 
 # *DOCUMENTATION*
-- 
2.27.0

