--- a/Documentation/sphinx/kernel_feat.py	2024-02-01 01:19:14.000000000 +0100
+++ b/Documentation/sphinx/kernel_feat.py.new	2024-02-01 20:08:58.896265056 +0100
@@ -93,6 +93,8 @@
         if len(self.arguments) > 1:
             args.extend(['--arch', self.arguments[1]])
 
+        fname = ' '.join(args)
+
         lines = subprocess.check_output(args, cwd=os.path.dirname(doc.current_source)).decode('utf-8')
 
         line_regex = re.compile("^\.\. FILE (\S+)$")
