--- webui/CMakeLists.txt.orig	2023-03-03 21:55:18.000000000 +0200
+++ webui/CMakeLists.txt	2023-03-03 22:01:47.835425931 +0200
@@ -66,7 +66,9 @@
 message("   BAREOS_FULL_VERSION:           ${BAREOS_FULL_VERSION} ")
 message("   LocalBuildDefinitionsFile:     ${BareosLocalBuildDefinitionsFile}")
 
-if(EXISTS ${sysconfdir}/httpd/conf.d)
+if(EXISTS ${sysconfdir}/httpd/conf/extra )
+  set(HTTPD_CONF ${sysconfdir}/httpd/conf/extra )
+elseif(EXISTS ${sysconfdir}/httpd/conf.d)
   set(HTTPD_CONF ${sysconfdir}/httpd/conf.d)
 elseif(EXISTS ${sysconfdir}/apache2/conf.d)
   set(HTTPD_CONF ${sysconfdir}/apache2/conf.d)
