diff -pNaru5 a/pith/charconv/utf8.c b/pith/charconv/utf8.c
--- a/pith/charconv/utf8.c	2008-04-02 18:09:20.000000000 -0400
+++ b/pith/charconv/utf8.c	2025-12-31 13:43:37.057305818 -0500
@@ -23,10 +23,12 @@ static char rcsid[] = "$Id: utf8.c 1019
 #ifdef _WINDOWS
 /* wingdi.h uses ERROR (!) and we aren't using the c-client ERROR so... */
 #undef ERROR
 #endif
 
+#define _XOPEN_SOURCE       /* See feature_test_macros(7) */
+#include <wchar.h>
 #include <system.h>
 
 #include "../../c-client/fs.h"
 
 /* includable WITHOUT dependency on pico */
