--- a/CMakeLists.txt	2023-09-11 19:04:13.000000000 +0300
+++ b/CMakeLists.txt	2023-09-16 16:58:53.490633834 +0300
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.14)
 
 project(Serious-Runner VERSION 1.0)
 
@@ -16,6 +16,9 @@
 find_package(Qt5Sql REQUIRED)
 find_package(Qt5 COMPONENTS Network REQUIRED)
 include(FindLibArchive)
+include(FindLibLZMA)
+include(FindBZip2)
+include(FindZLIB)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
 include(CheckCXXCompilerFlag)
@@ -280,8 +283,9 @@
     endif()
 endif()
 
-set_target_properties(Serious-Runner PROPERTIES LINK_FLAGS "-lz -lbz2 -llzma")
-target_link_libraries(Serious-Runner PRIVATE Qt${QT_VERSION_MAJOR}::Widgets  Qt5::Sql Qt5::Network LibArchive::LibArchive)
+# Link libraries.
+
+target_link_libraries(Serious-Runner PRIVATE Qt${QT_VERSION_MAJOR}::Widgets  Qt5::Sql Qt5::Network LibArchive::LibArchive LibLZMA::LibLZMA BZip2::BZip2 ZLIB::ZLIB)
 
 set_target_properties(Serious-Runner PROPERTIES
     MACOSX_BUNDLE_GUI_IDENTIFIER my.example.com
