From 9eaf6f31b7f6266bbb87f11ca6a59e47f733ea00 Mon Sep 17 00:00:00 2001
From: Michael Lass <lass@mail.uni-paderborn.de>
Date: Sat, 11 Jun 2016 22:02:30 +0200
Subject: [PATCH 6/7] Adapt to libgit2 0.24.x

---
 src/version/partschecker.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/version/partschecker.cpp b/src/version/partschecker.cpp
index a3e3967..da61560 100644
--- a/src/version/partschecker.cpp
+++ b/src/version/partschecker.cpp
@@ -121,7 +121,7 @@ bool PartsChecker::newPartsAvailable(const QString &repoPath, const QString & sh
     /**
      * Connect to the remote.
      */
-    error = git_remote_connect(remote, GIT_DIRECTION_FETCH, &callbacks);
+    error = git_remote_connect(remote, GIT_DIRECTION_FETCH, &callbacks, NULL);
     if (error) {
         partsCheckerResult.partsCheckerError = PARTS_CHECKER_ERROR_REMOTE;
         partsCheckerResult.errorMessage = QObject::tr("Unable to access network site for '%1'. %2").arg(repoPath).arg(sBoilerPlate1);
-- 
2.8.3

