--- a/CMakeLists.txt	2023-09-19 10:19:55.453999712 +0300
+++ b/CMakeLists.txt	2023-09-18 15:44:52.377869666 +0300
@@ -11,17 +11,40 @@
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
+include(CheckCXXCompilerFlag)
+
 find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets)
 find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets)
 find_package(Qt5Sql REQUIRED)
 find_package(Qt5 COMPONENTS Network REQUIRED)
-include(FindLibArchive)
+
 include(FindLibLZMA)
 include(FindBZip2)
 include(FindZLIB)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
-include(CheckCXXCompilerFlag)
+if(CMAKE_VERSION VERSION_GREATER 3.17.0)
+    include(FindLibArchive)
+else()
+    # cmake < 3.17.0 Find LIBARCHIVE
+    option(CMAKE_USE_SYSTEM_LIBARCHIVE "Use system-installed libarchive" "On")
+    if(CMAKE_USE_SYSTEM_LIBARCHIVE)
+      find_package(LibArchive 3.0.1)
+      if(NOT LibArchive_FOUND)
+        message(FATAL_ERROR "CMAKE_USE_SYSTEM_LIBARCHIVE is ON but LibArchive is not found!")
+      else()
+        message(STATUS "Find LibArchive Done.")
+        set(LibArchive_INCLUDE_DIRS ${LibArchive_INCLUDE_DIR})
+        set(LibArchive_LIBRARIES    ${LibArchive_LIBRARY})
+        if (NOT TARGET LibArchive::LibArchive)
+          add_library(LibArchive::LibArchive UNKNOWN IMPORTED)
+          set_target_properties(LibArchive::LibArchive PROPERTIES
+            IMPORTED_LOCATION "${LibArchive_LIBRARY}"
+            INTERFACE_INCLUDE_DIRECTORIES "${LibArchive_INCLUDE_DIR}")
+        endif() 
+      endif()
+    endif()
+endif()
 
 set(INTERNAL_NAME "seriousrunner")
 
@@ -32,7 +55,7 @@
     set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/")                               # set install path to project root directory since
                                                                                    # since one wasn't set during config
     set(LOCAL_INSTALL TRUE)
-    message(STATUS "Using LOCAL Imstall")
+    message(STATUS "Using LOCAL Install")
 endif()
 
 if(CMAKE_INSTALL_PREFIX STREQUAL "/usr" ) # Only works for linux since I don't know what default is for windows/macos/freebsd.
