From 84fa5b7bdb6e48635bee214daca34cee49289ef8 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sat, 5 Nov 2016 12:46:27 +0100
Subject: [PATCH 8/9] Link against ANGLE even if Qt is configured not to

---
 Source/WebCore/WebCore.pri | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/Source/WebCore/WebCore.pri b/Source/WebCore/WebCore.pri
index d6c32e80c..42aadaacc 100644
--- a/Source/WebCore/WebCore.pri
+++ b/Source/WebCore/WebCore.pri
@@ -216,14 +216,11 @@ use?(3D_GRAPHICS) {
     win32: {
         mingw: {
             # Make sure OpenGL libs are after the webcore lib so MinGW can resolve symbols
-            qtConfig(opengles2) {
-                CONFIG(debug, debug|release):qtConfig(angle) {
-                    LIBS += $$QMAKE_LIBS_OPENGL_ES2_DEBUG
-                } else {
-                    LIBS += $$QMAKE_LIBS_OPENGL_ES2
-                }
+            CONFIG(debug, debug|release):qtConfig(angle) {
+                LIBS += $$QMAKE_LIBS_OPENGL_ES2_DEBUG
             } else {
-                LIBS += $$QMAKE_LIBS_OPENGL
+                # Always link against ANGLE
+                LIBS += $$QMAKE_LIBS_OPENGL_ES2
             }
         }
     } else {
-- 
2.13.0

