--- ./src/libtcodConfig.cmake.in	2025-09-05 14:43:46.000000000 +0200
+++ ./src/libtcodConfig.cmake.in	2025-10-07 13:49:49.333330456 +0200
@@ -1,12 +1,6 @@
 @PACKAGE_INIT@

-find_package(SDL3 QUIET)
-find_package(ZLIB QUIET)
-find_package(lodepng-c QUIET)
-find_package(utf8proc QUIET)
-find_package(unofficial-utf8proc QUIET)
-
-set_and_check(LIBTCOD_INCLUDE_DIR "@CMAKE_INSTALL_INCLUDEDIR@")
+set_and_check(LIBTCOD_INCLUDE_DIR "/usr/include")

 include(${CMAKE_CURRENT_LIST_DIR}/libtcodTargets.cmake)
