--- a/configure
+++ b/configure
@@ -655,7 +655,7 @@
     check_py_lib xml     || disable_netvision "Python XML library (xml)"
     check_py_lib urllib  || disable_netvision "Python URL library (urllib)"
-    check_py_lib urllib2 || disable_netvision "Python URL library 2 (urllib2)"
-    check_py_lib oauth   || disable_netvision "Python OAuth library (oauth)"
+    check_py_lib urllib3 || disable_netvision "Python URL library 3 (urllib3)"
+    check_py_lib oauthlib   || disable_netvision "Python OAuth library (oauth)"
 
     disabled netvision && echo "Disabling MythNetvision due to missing dependencies."
 fi

