diff --git a/pipewire-confs/hardware-profiles/valve-galileo/pipewire.conf.d/filter-chain.conf b/pipewire-confs/hardware-profiles/valve-galileo/pipewire.conf.d/filter-chain.conf
index 262fa0f..d7055e2 100644
--- a/pipewire-confs/hardware-profiles/valve-galileo/pipewire.conf.d/filter-chain.conf
+++ b/pipewire-confs/hardware-profiles/valve-galileo/pipewire.conf.d/filter-chain.conf
@@ -14,7 +14,7 @@ context.modules = [
                     {
                         type = ladspa
                         name = rnnoise
-                        plugin = /usr/lib/ladspa/librnnoise_ladspa.so
+                        plugin = /usr/lib64/ladspa/librnnoise_ladspa.so
                         label = noise_suppressor_stereo
                         control = {
                             "VAD Threshold (%)" = 23.0
diff --git a/pipewire-confs/hardware-profiles/valve-jupiter/pipewire.conf.d/filter-chain.conf b/pipewire-confs/hardware-profiles/valve-jupiter/pipewire.conf.d/filter-chain.conf
index 90a1e12..43f03e0 100644
--- a/pipewire-confs/hardware-profiles/valve-jupiter/pipewire.conf.d/filter-chain.conf
+++ b/pipewire-confs/hardware-profiles/valve-jupiter/pipewire.conf.d/filter-chain.conf
@@ -14,7 +14,7 @@ context.modules = [
                     {
                         type = ladspa
                         name = rnnoise
-                        plugin = /usr/lib/ladspa/librnnoise_ladspa.so
+                        plugin = /usr/lib64/ladspa/librnnoise_ladspa.so
                         label = noise_suppressor_stereo
                         control = {
                             "VAD Threshold (%)" = 23.0
