From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marko Lemmetty <marko.lemmetty@jollamobile.com>
Date: Fri, 15 May 2020 13:46:20 +0300
Subject: [PATCH] Add sb2install support to osc

---
 osc/build.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/osc/build.py b/osc/build.py
index f59bb6520649fe1388165388e46982413029d3ac..429217916e4673cd67a954e57be04adb62b105ce 100644
--- a/osc/build.py
+++ b/osc/build.py
@@ -191,6 +191,7 @@ class Buildinfo:
                 self.urls[node.get('project') + "/" + node.get('repository')] = baseurl + '/%(arch)s/%(filename)s'
 
         self.vminstall_list = [dep.name for dep in self.deps if dep.vminstall]
+        self.sb2install_list = [dep.name for dep in self.deps if dep.sb2install]
         self.preinstall_list = [dep.name for dep in self.deps if dep.preinstall]
         self.runscripts_list = [dep.name for dep in self.deps if dep.runscripts]
         self.noinstall_list = [dep.name for dep in self.deps if dep.noinstall]
@@ -236,7 +237,8 @@ class Pac:
                   'project', 'repository', 'sysroot',
                   'preinstall', 'vminstall', 'runscripts',
                   'noinstall', 'installonly', 'notmeta',
-                  ]:
+                  'sb2install',
+                 ]:
             self.mp[i] = node.get(i)
 
         self.mp['buildarch'] = buildarch
@@ -1421,6 +1423,7 @@ def main(apiurl, store, opts, argv):
 
     rpmlist.append('preinstall: ' + ' '.join(bi.preinstall_list) + '\n')
     rpmlist.append('vminstall: ' + ' '.join(bi.vminstall_list) + '\n')
+    rpmlist.append('sb2install: ' + ' '.join(bi.sb2install_list) + '\n')
     rpmlist.append('runscripts: ' + ' '.join(bi.runscripts_list) + '\n')
     if build_type != 'kiwi' and build_type != 'docker' and build_type != 'podman' and build_type != 'fissile':
         if bi.noinstall_list:
