--- NetworkManager-strongswan-1.3.1/properties/nm-strongswan.c.orig	2016-09-07 01:29:42.639436118 +0300
+++ NetworkManager-strongswan-1.3.1/properties/nm-strongswan.c	2016-09-07 01:30:02.432769728 +0300
@@ -391,7 +391,7 @@
 	nm_setting_vpn_add_data_item (settings, "ipcomp", active ? "yes" : "no");
 
 	nm_setting_set_secret_flags (NM_SETTING (settings), "password",
-								 NM_SETTING_SECRET_FLAG_AGENT_OWNED, NULL);
+								 NM_SETTING_SECRET_FLAG_NONE, NULL);
 
 	nm_connection_add_setting (connection, NM_SETTING (settings));
 	return TRUE;
