From 3f67c969ef4a532e2fbf9a0743de56a73901ed8e Mon Sep 17 00:00:00 2001
From: Michael Cronenworth <mike@cchtml.com>
Date: Thu, 23 Jan 2020 17:05:28 -0600
Subject: [PATCH 12/15] webservices: Global variable compatibility update for
 gcc 10.

Signed-off-by: Michael Cronenworth <mike@cchtml.com>
Signed-off-by: Hans Leidekker <hans@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 dlls/webservices/webservices_private.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dlls/webservices/webservices_private.h b/dlls/webservices/webservices_private.h
index 6f2b90dad2..b352aa52d4 100644
--- a/dlls/webservices/webservices_private.h
+++ b/dlls/webservices/webservices_private.h
@@ -18,7 +18,7 @@
 
 #include "winhttp.h"
 
-_locale_t c_locale DECLSPEC_HIDDEN;
+extern _locale_t c_locale DECLSPEC_HIDDEN;
 
 #define STREAM_BUFSIZE 4096
 
@@ -50,8 +50,8 @@ struct dictionary
     ULONG              current_sequence;
     ULONG             *sequence;
 };
-struct dictionary dict_builtin DECLSPEC_HIDDEN;
-const struct dictionary dict_builtin_static DECLSPEC_HIDDEN;
+extern struct dictionary dict_builtin DECLSPEC_HIDDEN;
+extern const struct dictionary dict_builtin_static DECLSPEC_HIDDEN;
 
 int find_string( const struct dictionary *, const unsigned char *, ULONG, ULONG * ) DECLSPEC_HIDDEN;
 HRESULT insert_string( struct dictionary *, unsigned char *, ULONG, int, ULONG * ) DECLSPEC_HIDDEN;
-- 
2.26.2

