From 2e0b11f0f0ba3e756f010b5948b24089be5e3aed Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 18 Sep 2016 18:04:42 +0200
Subject: [PATCH 17/31] Use system zlib for build tools

---
 src/tools/bootstrap/bootstrap.pro | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/tools/bootstrap/bootstrap.pro b/src/tools/bootstrap/bootstrap.pro
index 24862a07db..698e6cfb95 100644
--- a/src/tools/bootstrap/bootstrap.pro
+++ b/src/tools/bootstrap/bootstrap.pro
@@ -158,7 +158,9 @@ macx {
         ../../corelib/io/qstandardpaths_win.cpp
 }
 
-!qtConfig(system-zlib)|cross_compile {
+cross_compile {
+    LIBS += -lz
+} else:!qtConfig(system-zlib) {
     include(../../3rdparty/zlib.pri)
 } else {
     CONFIG += no_core_dep
-- 
2.29.2

