From 9a0c4a937083e14ad32a8a25feb92333b5015dda Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 18 Sep 2016 18:50:21 +0200
Subject: [PATCH 25/34] Use correct pkg-config --static flag

---
 configure.pri | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.pri b/configure.pri
index 3136889a2f..601629c047 100644
--- a/configure.pri
+++ b/configure.pri
@@ -328,6 +328,9 @@ defineTest(qtConfTest_detectPkgConfig) {
             qtLog("Found pkg-config from path: $$pkgConfig")
         }
     }
+    !isEmpty(pkgConfig):qtConfig(static) {
+        pkgConfig = "$$pkgConfig --static"
+    }
 
     $$qtConfEvaluate("features.cross_compile") {
         # cross compiling, check that pkg-config is set up sanely
-- 
2.20.1

