From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Bidar?= <bjorn.bidar@jolla.com>
Date: Sat, 22 Jul 2023 18:08:57 +0300
Subject: [PATCH] Shut up warnings about future Rust version incompatibilities
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Björn Bidar <bjorn.bidar@jolla.com>
---
 Cargo.toml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index 27886bd8ff1ec0530f63498101d2fc33ce5c0cf4..7e1fb391e9fdff1b25ae20a0a249c9c7def6eb25 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -228,3 +228,8 @@ uniffi_bindgen = "=0.24.3"
 uniffi_build = "=0.24.3"
 uniffi_macros = "=0.24.3"
 weedle2 = "=4.0.0"
+
+# Package code v0.1.4 uses code "that will be rejected by a future version of Rust"
+# Shut up such messages for now to make the build succeed
+[future-incompat-report]
+frequency = "never"
