diff -Nru quarto-cli-1.4.250.original/src/render/notebook/notebook-contributor-jats.ts quarto-cli-1.4.250/src/render/notebook/notebook-contributor-jats.ts
--- quarto-cli-1.4.250.original/src/render/notebook/notebook-contributor-jats.ts	2023-07-22 08:39:00.000000000 +1200
+++ quarto-cli-1.4.250/src/render/notebook/notebook-contributor-jats.ts	2023-07-22 10:15:31.408433063 +1200
@@ -102,7 +102,6 @@
           [kOutputFile]: outputFile(nbPath),
           [kTemplate]: subarticleTemplatePath,
           [kNotebookPreserveCells]: true,
-          [kNotebookPreserveCells]: true,
         },
         quiet: false,
       },
