diff -pur dymo-cups-drivers-1.2.0/src/common/Halftoning.h dymo-cups-drivers-1.2.0-new/src/common/Halftoning.h
--- dymo-cups-drivers-1.2.0/src/common/Halftoning.h	2009-05-22 17:11:57.000000000 -0400
+++ dymo-cups-drivers-1.2.0-new/src/common/Halftoning.h	2012-01-15 01:10:55.015847914 -0500
@@ -22,6 +22,7 @@
 #ifndef h4D098F6A_47C6_4e9d_BD74_2DC6034F8EEF
 #define h4D098F6A_47C6_4e9d_BD74_2DC6034F8EEF
 
+#include <stddef.h>
 #include "CommonTypedefs.h"
 
 //namespace dymo
diff -pur dymo-cups-drivers-1.2.0/src/lw/LabelWriterDriver.h dymo-cups-drivers-1.2.0-new/src/lw/LabelWriterDriver.h
--- dymo-cups-drivers-1.2.0/src/lw/LabelWriterDriver.h	2009-05-22 17:11:57.000000000 -0400
+++ dymo-cups-drivers-1.2.0-new/src/lw/LabelWriterDriver.h	2012-01-15 01:10:24.628615257 -0500
@@ -21,6 +21,7 @@
 #ifndef h53a93353_4968_443a_a5aa_1167526295fc
 #define h53a93353_4968_443a_a5aa_1167526295fc
 
+#include <stddef.h>
 #include "PrinterDriver.h"
 
 namespace DymoPrinterDriver
