diff --unified --recursive --text a/build.gradle b/build.gradle
--- a/build.gradle	2021-10-18 12:09:38.389096998 +0000
+++ b/build.gradle	2021-10-18 12:10:04.859168754 +0000
@@ -2,7 +2,6 @@
     id 'java'
     id 'com.github.johnrengelman.shadow' version '6.0.0'
     id 'maven-publish'
-    id 'com.palantir.git-version' version '0.12.3'
     id 'com.google.cloud.tools.jib' version '2.6.0'
 }

@@ -116,7 +115,7 @@
 }

 classes {
-    dependsOn createVersionProperties
+    // dependsOn createVersionProperties
 }

 repositories {
@@ -172,7 +171,7 @@
     // Provides the EPSG coordinate reference system catalog as an HSQL database.
     implementation group: 'org.geotools', version: geotoolsVersion, name: 'gt-epsg-hsql'

-    compile 'com.wdtinc:mapbox-vector-tile:3.1.0'
+    implementation 'com.wdtinc:mapbox-vector-tile:3.1.0'

     // Legacy JTS with com.vividsolutions package name. Newer Geotools compatible with Java 11 uses a newer version of
     // JTS with the org.locationtech package name. But our MapDB format includes serialized JTS geometries with the
