diff --git a/src/ephy-header-bar.c b/src/ephy-header-bar.c
index 389c34b3b..00cb6f038 100644
--- a/src/ephy-header-bar.c
+++ b/src/ephy-header-bar.c
@@ -176,7 +177,7 @@
   column = hdy_column_new ();
   gtk_widget_set_hexpand (GTK_WIDGET (column), TRUE);
   gtk_widget_show (GTK_WIDGET (column));
-  hdy_column_set_maximum_width (column, 860);
+  hdy_column_set_maximum_width (column, gdk_screen_width ());
   hdy_column_set_linear_growth_width (column, 560);
   gtk_container_add (GTK_CONTAINER (column), GTK_WIDGET (header_bar->title_widget));
   gtk_header_bar_set_custom_title (GTK_HEADER_BAR (header_bar), GTK_WIDGET (column));
