--- server/src/analyzer/pythonPathUtils.ts.orig	2019-04-11 16:06:50.335851556 +0800
+++ server/src/analyzer/pythonPathUtils.ts	2019-04-11 16:07:47.596826850 +0800
@@ -85,7 +85,7 @@
         let pythonPaths: string[] = [];
 
         try {
-            const commandLineArgs: string[] = ['-c', 'import sys; print sys.path'];
+            const commandLineArgs: string[] = ['-c', 'import sys; print(sys.path)'];
             let execOutput: string;
 
             if (interpreterPath) {
