diff --git a/ctrtool/Makefile b/ctrtool/Makefile
index 7a534c9..245eeb4 100644
--- a/ctrtool/Makefile
+++ b/ctrtool/Makefile
@@ -4,7 +4,7 @@ OBJS = $(foreach dir,$(SRC_DIR),$(subst .c,.o,$(wildcard $(dir)/*.c))) $(foreach
 
 # Compiler Settings
 OUTPUT = ctrtool
-CXXFLAGS = -I.
-CFLAGS = -O2 -Wall -Wno-unused-variable  -Wno-unused-result -I. -std=c11
+CXXFLAGS = -O2 -Wall -I. -fPIC
+CFLAGS = -O2 -Wall -Wno-unused-variable  -Wno-unused-result -I. -std=c11 -fPIC
 CC = gcc
 CXX = g++
