From 1886d97193a279c80c24d84c21cb10cec53b31a3 Mon Sep 17 00:00:00 2001
From: David Mohammed <fossfreedom@ubuntu.com>
Date: Fri, 14 Sep 2018 19:42:18 +0100
Subject: [PATCH] Fix GTK+3.24 compilation issues #44

---
 src/Widgets/AreaSelectionWindow.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Widgets/AreaSelectionWindow.vala b/src/Widgets/AreaSelectionWindow.vala
index fa1e853..f1c84a0 100644
--- a/src/Widgets/AreaSelectionWindow.vala
+++ b/src/Widgets/AreaSelectionWindow.vala
@@ -152,8 +152,8 @@ private class AreaSelectionWindow : Gtk.Window
         style.save();
         style.add_class(Gtk.STYLE_CLASS_RUBBERBAND);
 
-        Gtk.render_background(style, ctx, 0, 0, w, h);
-        Gtk.render_frame(style, ctx, 0, 0, w, h);
+        style.render_background(ctx, 0, 0, w, h);
+        style.render_frame(ctx, 0, 0, w, h);
 
         style.restore();
 
