From d6fad3e9f9e07ee94956589c89a1d0cf96658cd5 Mon Sep 17 00:00:00 2001
From: arch1t3cht <arch1t3cht@gmail.com>
Date: Fri, 1 Jul 2022 20:23:24 +0200
Subject: [PATCH] Fix compile error on linux

---
 libaegisub/include/libaegisub/audio/provider.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libaegisub/include/libaegisub/audio/provider.h b/libaegisub/include/libaegisub/audio/provider.h
index debb560ce..70460a723 100644
--- a/libaegisub/include/libaegisub/audio/provider.h
+++ b/libaegisub/include/libaegisub/audio/provider.h
@@ -20,6 +20,7 @@
 #include <libaegisub/fs_fwd.h>
 
 #include <atomic>
+#include <memory>
 #include <vector>
 
 namespace agi {
