diff -aur burp-2.0.30-pristine/Makefile.am burp-2.0.30-patched/Makefile.am
--- burp-2.0.30-pristine/Makefile.am	2016-01-03 00:59:14.000000000 -0600
+++ burp-2.0.30-patched/Makefile.am	2016-01-03 09:10:06.010636099 -0600
@@ -358,7 +358,7 @@
 
 install-data-local:
 	$(AM_V_at)$(MKDIR_P) $(DESTDIR)$(localstatedir)/spool/burp
-	$(AM_V_at)$(MKDIR_P) $(DESTDIR)$(runstatedir)
+	$(AM_V_at)$(MKDIR_P) $(DESTDIR)/run
 	$(AM_V_at)$(MKDIR_P) $(DESTDIR)$(sysconfdir)/CA-client
 	$(AM_V_at)$(MKDIR_P) $(DESTDIR)$(sysconfdir)/autoupgrade/server/win32
 	$(AM_V_at)$(MKDIR_P) $(DESTDIR)$(sysconfdir)/autoupgrade/server/win64/$(VERSION)
