diff -ura a/Drivers/Dta/Source/Linux/Makefile b/Drivers/Dta/Source/Linux/Makefile
--- a/Drivers/Dta/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/Dta/Source/Linux/Makefile	2024-09-10 23:20:37.938769292 +0200
@@ -66,9 +66,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Include version evironment vars
diff -ura a/Drivers/DtaNw/Source/Linux/Makefile b/Drivers/DtaNw/Source/Linux/Makefile
--- a/Drivers/DtaNw/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/DtaNw/Source/Linux/Makefile	2024-09-10 23:21:11.112351300 +0200
@@ -61,9 +61,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Include version evironment vars
diff -ura a/Drivers/DtDrvCommon/Source/Linux/Makefile b/Drivers/DtDrvCommon/Source/Linux/Makefile
--- a/Drivers/DtDrvCommon/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/DtDrvCommon/Source/Linux/Makefile	2024-09-10 23:21:40.075901310 +0200
@@ -67,9 +67,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Source definitions
diff -ura a/Drivers/DtPcie/Source/Linux/Makefile b/Drivers/DtPcie/Source/Linux/Makefile
--- a/Drivers/DtPcie/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/DtPcie/Source/Linux/Makefile	2024-09-10 23:22:10.882798026 +0200
@@ -66,9 +66,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Include version evironment vars
diff -ura a/Drivers/DtSal/Source/Linux/Makefile b/Drivers/DtSal/Source/Linux/Makefile
--- a/Drivers/DtSal/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/DtSal/Source/Linux/Makefile	2024-09-10 23:22:25.512907133 +0200
@@ -60,9 +60,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Source definitions
diff -ura a/Drivers/Dtu/Source/Linux/Makefile b/Drivers/Dtu/Source/Linux/Makefile
--- a/Drivers/Dtu/Source/Linux/Makefile	2024-09-09 14:46:37.000000000 +0200
+++ b/Drivers/Dtu/Source/Linux/Makefile	2024-09-10 23:22:41.056356292 +0200
@@ -67,9 +67,7 @@
 
 # Force using inline code to mitigate against straight line speculation in stead of using
 # the current default thunk-extern which generates duplicate functions.
-ifneq ($(wildcard /etc/fedora-release),)
-	EXTRA_CFLAGS += -mindirect-branch=thunk-inline
-endif
+EXTRA_CFLAGS += -mindirect-branch=thunk-inline
 
 # ========================================================================================
 # Include version evironment vars
