--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -26,8 +26,8 @@ target_include_directories(CsCore
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtCore>
-   $<INSTALL_INTERFACE:include>
-   $<INSTALL_INTERFACE:include/QtCore>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtCore>
 )
 
 include_directories(
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -18,7 +18,7 @@ set(EXTRA_GUI_LDFLAGS)
 target_include_directories(CsGui
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtGui>
-   $<INSTALL_INTERFACE:include/QtGui>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtGui>
 )
 
 include_directories(
--- a/src/multimedia/CMakeLists.txt
+++ b/src/multimedia/CMakeLists.txt
@@ -14,7 +14,7 @@ set(EXTRA_MULTIMEDIA_LDFLAGS)
 target_include_directories(CsMultimedia
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtMultimedia>
-   $<INSTALL_INTERFACE:include/QtMultimedia>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtMultimedia>
 )
 
 include_directories(
--- a/src/network/CMakeLists.txt
+++ b/src/network/CMakeLists.txt
@@ -36,7 +36,7 @@ endif()
 target_include_directories(CsNetwork
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtNetwork>
-   $<INSTALL_INTERFACE:include/QtNetwork>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtNetwork>
 )
 
 include_directories(
--- a/src/opengl/CMakeLists.txt
+++ b/src/opengl/CMakeLists.txt
@@ -15,7 +15,7 @@ set(EXTRA_OPENGL_LDFLAGS)
 target_include_directories(CsOpenGL
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtOpenGL>
-   $<INSTALL_INTERFACE:include/QtOpenGL>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtOpenGL>
 )
 
 include_directories(
--- a/src/script/CMakeLists.txt
+++ b/src/script/CMakeLists.txt
@@ -37,7 +37,7 @@ set(EXTRA_SCRIPT_LDFLAGS)
 target_include_directories(CsScript
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtScript>
-   $<INSTALL_INTERFACE:include/QtScript>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtScript>
 )
 
 include_directories(
--- a/src/sql/CMakeLists.txt
+++ b/src/sql/CMakeLists.txt
@@ -15,7 +15,7 @@ set(EXTRA_SQL_LDFLAGS)
 target_include_directories(CsSql
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtSql>
-   $<INSTALL_INTERFACE:include/QtSql>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtSql>
 )
 
 include_directories(
--- a/src/svg/CMakeLists.txt
+++ b/src/svg/CMakeLists.txt
@@ -13,7 +13,7 @@ set(EXTRA_SVG_LDFLAGS)
 target_include_directories(CsSvg
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtSvg>
-   $<INSTALL_INTERFACE:include/QtSvg>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtSvg>
 )
 
 include_directories(
--- a/src/vulkan/CMakeLists.txt
+++ b/src/vulkan/CMakeLists.txt
@@ -14,7 +14,7 @@ set(EXTRA_VULKAN_LDFLAGS)
 target_include_directories(CsVulkan
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtVulkan>
-   $<INSTALL_INTERFACE:include/QtVulkan>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtVulkan>
 )
 
 include_directories(
--- a/src/webkit/CMakeLists.txt
+++ b/src/webkit/CMakeLists.txt
@@ -39,7 +39,7 @@ endif()
 target_include_directories(CsWebKit
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtWebKit>
-   $<INSTALL_INTERFACE:include/QtWebKit>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtWebKit>
 )
 
 include_directories(
--- a/src/xml/CMakeLists.txt
+++ b/src/xml/CMakeLists.txt
@@ -13,7 +13,7 @@ set(EXTRA_XML_LDFLAGS)
 target_include_directories(CsXml
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtXml>
-   $<INSTALL_INTERFACE:include/QtXml>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtXml>
 )
 
 include_directories(
--- a/src/xmlpatterns/CMakeLists.txt
+++ b/src/xmlpatterns/CMakeLists.txt
@@ -13,7 +13,7 @@ set(EXTRA_XMLPATTERNS_LDFLAGS)
 target_include_directories(CsXmlPatterns
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include/QtXmlPatterns>
-   $<INSTALL_INTERFACE:include/QtXmlPatterns>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/QtXmlPatterns>
 )
 
 include_directories(
