diff -Naurp a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-12-17 16:52:55.000000000 +0000
+++ b/CMakeLists.txt	2019-12-17 17:30:31.455144875 +0000
@@ -44,16 +44,10 @@ include_directories(${TC_INC_PATH})
 include_directories(${VIDEO_CODEC_SDK_INCLUDE_DIR})
 
 #Bring all the stuff together
-install(FILES $<TARGET_FILE:TC_CORE>									DESTINATION bin)
-install(FILES $<TARGET_FILE:TC>											DESTINATION bin)
+install(FILES $<TARGET_FILE:TC_CORE>									DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES $<TARGET_FILE:TC>											DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(GENERATE_PYTHON_BINDINGS)
-	install(FILES $<TARGET_FILE:PyNvCodec>								DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleDecode.py			DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleEncode.py			DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleTranscode.py		DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleFrameUpload.py		DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleColorConversion.py	DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleSufraceDownload.py	DESTINATION bin)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/SampleTranscodeOneToN.py	DESTINATION bin)
-endif(GENERATE_PYTHON_BINDINGS)
\ No newline at end of file
+	find_package(Python COMPONENTS Interpreter Development)
+	install(FILES $<TARGET_FILE:PyNvCodec>								DESTINATION ${Python_SITEARCH}/PyNvCodec)
+endif(GENERATE_PYTHON_BINDINGS)
