--- a/third_party/protobuf.cmake
+++ b/third_party/protobuf.cmake
@@ -24,7 +24,7 @@ macro(configure_protobuf VERSION)
     set(Protobuf_TARGET third_party.protobuf)
 
     set(PROTOBUF_CFLAGS "-Dgoogle=google_private")
-    set(PROTOBUF_CXXFLAGS "-Dgoogle=google_private")
+    set(PROTOBUF_CXXFLAGS "-Dgoogle=google_private -std=c++14")
 
     ExternalProject_Add(${Protobuf_TARGET}
         PREFIX ${Protobuf_TARGET}
