--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,10 +175,6 @@ if(DOTNET)
   endif()
 endif()
 
-if(EXISTS /etc/bash_completion.d)
-  install(PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/survive_autocomplete.sh DESTINATION "/etc/bash_completion.d")
-endif()
-
 install(DIRECTORY include/libsurvive DESTINATION include)
 
 file(GLOB REDIST_HEADERS
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -178,7 +178,6 @@ set(SURVIVE_BUILT_PLUGINS "${SURVIVE_BUILT_PLUGINS}" PARENT_SCOPE)
 
 if(NOT WIN32)
 install(TARGETS ${SURVIVE_BUILT_PLUGINS} DESTINATION lib/libsurvive/plugins)
-install(TARGETS survive DESTINATION lib/libsurvive)
 install(TARGETS survive DESTINATION lib)
 else()
 install(TARGETS ${SURVIVE_BUILT_PLUGINS} DESTINATION bin/plugins)
