--- a/Makefile
+++ b/Makefile
@@ -47,10 +47,10 @@ build: $(CLI)
 
 
 $(XD): $(WEBUI_CORE)
-	$(GO) build -a -ldflags "-X xd/lib/version.Git=$(GIT_VERSION)" -tags='$(TAGS)' -o $(XD)
+	$(GO) build -a -ldflags "-X xd/lib/version.Git=$(GIT_VERSION) -linkmode external -extldflags '${LDFLAGS}'" -tags='$(TAGS)' -o $(XD)
 
 dev: $(WEBUI_CORE)
-	$(GO) build -race -v -a -ldflags "-X xd/lib/version.Git=$(GIT_VERSION)" -tags='$(TAGS)' -o $(XD)
+	$(GO) build -race -v -a -ldflags "-X xd/lib/version.Git=$(GIT_VERSION) -linkmode external -extldflags '${LDFLAGS}'" -tags='$(TAGS)' -o $(XD)
 
 $(CLI): $(XD)
 	$(RM) $(CLI)
@@ -85,7 +85,7 @@ $(WEBUI_CORE): $(WEBUI_LOGO)
 webui: $(WEBUI_CORE)
 
 no-webui:
-	$(GO) build -ldflags "-X xd/lib/version.Git=$(GIT_VERSION)" -o $(XD)
+	$(GO) build -ldflags "-X xd/lib/version.Git=$(GIT_VERSION) -linkmode external -extldflags '${LDFLAGS}'" -o $(XD)
 
 install: $(XD) $(CLI)
 	$(MKDIR) $(PREFIX)/bin
