diff --git a/SConstruct b/SConstruct
index b65c61dc..c262a31a 100644
--- a/SConstruct
+++ b/SConstruct
@@ -16,7 +16,7 @@ import textwrap
 import uuid
 from glob import glob

-from pkg_resources import parse_version
+from packaging.version import parse as parse_version

 import SCons

diff --git a/buildscripts/linter/runner.py b/buildscripts/linter/runner.py
index 623a3c42..33649e42 100644
--- a/buildscripts/linter/runner.py
+++ b/buildscripts/linter/runner.py
@@ -10,7 +10,7 @@ import site
 import subprocess
 import sys
 import threading
-import pkg_resources
+import packaging.version

 from . import base

@@ -33,7 +33,7 @@ def _check_version(linter, cmd_path, args):
                 decoded_output, stderr)

         pattern = r"\b(?:(%s) )?(?P<version>\S+)\b" % (linter.cmd_name)
-        required_version = pkg_resources.parse_version(linter.required_version)
+        required_version = packaging.version.parse(linter.required_version)

         match = re.search(pattern, decoded_output)
         if match:
@@ -41,7 +41,7 @@ def _check_version(linter, cmd_path, args):
         else:
             found_version = '0.0'

-        if pkg_resources.parse_version(found_version) < required_version:
+        if packaging.version.parse(found_version) < required_version:
             logging.info(
                 "Linter %s has wrong version for '%s'. Expected >= '%s',"
                 "Standard Output:\n'%s'\nStandard Error:\n%s", linter.cmd_name, cmd,
diff --git a/site_scons/site_tools/ccache.py b/site_scons/site_tools/ccache.py
index 2a894919..227f992b 100644
--- a/site_scons/site_tools/ccache.py
+++ b/site_scons/site_tools/ccache.py
@@ -26,7 +26,7 @@ import re
 import subprocess

 import SCons
-from pkg_resources import parse_version
+from packaging.version import parse as parse_version

 # This is the oldest version of ccache that offers support for -gsplit-dwarf
 _ccache_version_min = parse_version("3.2.3")
diff --git a/site_scons/site_tools/icecream.py b/site_scons/site_tools/icecream.py
index f9e436eb..df612f2a 100644
--- a/site_scons/site_tools/icecream.py
+++ b/site_scons/site_tools/icecream.py
@@ -25,7 +25,7 @@ import re
 import subprocess
 import urllib

-from pkg_resources import parse_version
+from packaging.version import parse as parse_version

 import SCons
