--- gimp-plugin-bimp-2.0/Makefile
+++ gimp-plugin-bimp-2.0/Makefile
@@ -5,7 +5,7 @@
 
 make: 
 	which gimptool-2.0 && which pcre-config && \
-	gcc -o ./bin/bimp -Wall -O2 -Wno-unused-variable -Wno-pointer-sign src/*.c src/manipulation-gui/*.c src/images/*.c $(GIMPARGS) $(PCREARGS) -lm -DGIMP_DISABLE_DEPRECATED
+	gcc -o ./bin/bimp -Wall -O2 -Wno-unused-variable -Wno-pointer-sign -fcommon src/*.c src/manipulation-gui/*.c src/images/*.c $(GIMPARGS) $(PCREARGS) -lm -DGIMP_DISABLE_DEPRECATED
 
 makewin: 
 	which gimptool-2.0 && which pcre-config && \

