--- a/plugin/CMakeLists.txt
+++ b/plugin/CMakeLists.txt
@@ -123,7 +123,7 @@ set_target_properties(${SHARED_TARGET} PROPERTIES DEBUG_POSTFIX ${TRT_DEBUG_POST
 
 set_target_properties(${SHARED_TARGET} PROPERTIES VERSION ${TRT_VERSION} SOVERSION ${TRT_SOVERSION} )
 
-set_property(TARGET ${SHARED_TARGET} PROPERTY CUDA_STANDARD 11)
+set_property(TARGET ${SHARED_TARGET} PROPERTY CUDA_STANDARD 14)
 
 target_link_libraries(${SHARED_TARGET}
     ${CUBLAS_LIB}
@@ -163,7 +163,7 @@ set_target_properties(${STATIC_TARGET} PROPERTIES DEBUG_POSTFIX ${TRT_DEBUG_POST
 
 set_target_properties(${STATIC_TARGET} PROPERTIES VERSION ${TRT_VERSION} SOVERSION ${TRT_SOVERSION} )
 
-set_property(TARGET ${STATIC_TARGET} PROPERTY CUDA_STANDARD 11)
+set_property(TARGET ${STATIC_TARGET} PROPERTY CUDA_STANDARD 14)
 
 #########################################################################################
 
