--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -337,8 +337,8 @@ if(MSVC)
 else()
     find_path(
         PY_INCLUDE Python.h
-        HINTS ${EXT_PATH}/${PYTHON} /usr/include/${PYTHON}
-        PATH_SUFFIXES include
+        HINTS /usr/include
+        PATH_SUFFIXES ${PYTHON}
         REQUIRED
     )
 endif()
--- a/scripts/build_python_wheel.sh
+++ b/scripts/build_python_wheel.sh
@@ -35,14 +35,16 @@ cmake .. -DCMAKE_BUILD_TYPE=Release \
          -DTENSORRT_ROOT=${ROOT_PATH} \
          -DTENSORRT_MODULE=${TENSORRT_MODULE} \
          -DTENSORRT_LIBPATH=${TRT_LIBPATH} \
+         -DCMAKE_SKIP_RPATH=YES \
+         -G 'Unix Makefiles' \
          -DTRT_ONNXPARSER_NAME=nvonnxparser
-make -j12
+cmake --build .
 
 # Generate wheel
-TRT_MAJOR=$(awk '/^\#define TRT_MAJOR_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
-TRT_MINOR=$(awk '/^\#define TRT_MINOR_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
-TRT_PATCH=$(awk '/^\#define TRT_PATCH_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
-TRT_BUILD=$(awk '/^\#define TRT_BUILD_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
+TRT_MAJOR=$(awk '/^#define TRT_MAJOR_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
+TRT_MINOR=$(awk '/^#define TRT_MINOR_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
+TRT_PATCH=$(awk '/^#define TRT_PATCH_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
+TRT_BUILD=$(awk '/^#define TRT_BUILD_ENTERPRISE/ {print $3}' ${ROOT_PATH}/include/NvInferVersion.h)
 TRT_VERSION=${TRT_MAJOR}.${TRT_MINOR}.${TRT_PATCH}.${TRT_BUILD}
 TRT_MAJMINPATCH=${TRT_MAJOR}.${TRT_MINOR}.${TRT_PATCH}
 PLUGIN_DISABLED_STR="0"
@@ -51,10 +53,10 @@ echo "Generating python ${PYTHON_MAJOR_VERSION}.${PYTHON_MINOR_VERSION} bindings
 
 expand_vars_cp () {
     test -f ${1} || (echo "ERROR: File: ${1} does not exist!" && exit 1); \
-    sed -e "s|\#\#TENSORRT_VERSION\#\#|${TRT_VERSION}|g" \
-        -e "s|\#\#TENSORRT_MAJMINPATCH\#\#|${TRT_MAJMINPATCH}|g" \
-        -e "s|\#\#TENSORRT_PYTHON_VERSION\#\#|${TRT_MAJMINPATCH}|g" \
-        -e "s|\#\#TENSORRT_MODULE\#\#|${TENSORRT_MODULE}|g" \
+    sed -e "s|##TENSORRT_VERSION##|${TRT_VERSION}|g" \
+        -e "s|##TENSORRT_MAJMINPATCH##|${TRT_MAJMINPATCH}|g" \
+        -e "s|##TENSORRT_PYTHON_VERSION##|${TRT_MAJMINPATCH}|g" \
+        -e "s|##TENSORRT_MODULE##|${TENSORRT_MODULE}|g" \
         -e "s|##TENSORRT_PLUGIN_DISABLED##|${PLUGIN_DISABLED_STR}|g" \
         ${1} > ${2}
 }
@@ -65,8 +67,8 @@ for file in $(find . -type f); do expand_vars_cp $file ${WHEEL_OUTPUT_DIR}/${fil
 popd
 
 # Copy required files and directories to build the bindings.
-cp tensorrt/tensorrt.so bindings_wheel/tensorrt/tensorrt.so
-mv tensorrt_build_backend bindings_wheel/
+cp ${TENSORRT_MODULE}/${TENSORRT_MODULE}.so bindings_wheel/tensorrt/${TENSORRT_MODULE}.so
+[ ${TENSORRT_MODULE} != 'tensorrt' ] && mv bindings_wheel/{tensorrt,${TENSORRT_MODULE}}
 
 pushd ${WHEEL_OUTPUT_DIR}/bindings_wheel
 python -m build --no-isolation --wheel --config-setting=wheel-type=binding --config-setting=python-tag=cp${PYTHON_MAJOR_VERSION}${PYTHON_MINOR_VERSION} --config-setting=plat-name=linux_${TARGET}
