--- a/gcc/config/mips/linux.h
+++ b/gcc/config/mips/linux.h
@@ -58,3 +58,8 @@ along with GCC; see the file COPYING3.
 #else
 #define NEED_INDICATE_EXEC_STACK TARGET_SOFT_FLOAT
 #endif
+
+#define STARTFILE_PREFIX_SPEC \
+  "%{mabi=n32: /lib/glibc/abi-n32/} \
+   %{mabi=32: /lib/glibc/abi-32/} \
+   %{mabi=64: /lib/glibc/abi-64/}"
--- a/gcc/config/mips/t-linux64
+++ b/gcc/config/mips/t-linux64
@@ -20,7 +20,4 @@ MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi
 MULTILIB_DIRNAMES = n32 32 64
 MIPS_EL = $(if $(filter %el, $(firstword $(subst -, ,$(target)))),el)
 MIPS_SOFT = $(if $(strip $(filter MASK_SOFT_FLOAT_ABI, $(target_cpu_default)) $(filter soft, $(with_float))),soft)
-MULTILIB_OSDIRNAMES = \
-	../lib32$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabin32$(MIPS_SOFT)) \
-	../lib$(call if_multiarch,:mips$(MIPS_EL)-linux-gnu$(MIPS_SOFT)) \
-	../lib64$(call if_multiarch,:mips64$(MIPS_EL)-linux-gnuabi64$(MIPS_SOFT))
+MULTILIB_OSDIRNAMES = n32=../lib/glibc/abi-n32 32=../lib/glibc/abi-32 64=../lib/glibc/abi-64
