--- a/config/pstoedit.pc.in
+++ b/config/pstoedit.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/pstoedit
 
 Name: pstoedit
 Description: converts PostScript(TM) and PDF files to other vector graphic formats
@@ -8,4 +8,4 @@
 Version: @VERSION@.0
 Requires: 
 Libs: -L@libdir@ -lpstoedit @CXX_STD_LIB@ @CXX_RUNTIME_LIB@ @LIBLD_LDFLAGS@
-Cflags: -I@includedir@
+Cflags: -I@includedir@/pstoedit
