--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ CXX ?= g++
 FLAGS = -Wall -std=c++11 -Wno-misleading-indentation
 
 ifeq ($(UNAME),Linux)
-	FLAGS += -g
+	FLAGS +=
 else ifeq ($(UNAME),Darwin)
 	FLAGS += -g
 else ifeq ($(UNAME),FreeBSD)
@@ -16,8 +16,7 @@ else
 endif
 
 INCFLAGS = -I/usr/include/i2pd
-CXXFLAGS = $(FLAGS)
-LDFLAGS = 
+CXXFLAGS += $(FLAGS)
 LIBS = -li2pd -lboost_system$(BOOST_SUFFIX) -lboost_date_time$(BOOST_SUFFIX) -lboost_filesystem$(BOOST_SUFFIX) -lboost_program_options$(BOOST_SUFFIX) -lssl -lcrypto -lz
 
 ifeq ($(UNAME),Linux)
@@ -75,7 +74,7 @@ regaddralias: regaddralias.o
 .SUFFIXES: .c .cc .C .cpp .o
 
 %.o: %.cpp
-	$(CXX) -o $@ -c $(CXXFLAGS) $(INCFLAGS) $<
+	$(CXX) -o $@ -c $(CPPFLAGS) $(CXXFLAGS) $(INCFLAGS) $<
 
 count:
 	wc *.c *.cc *.C *.cpp *.h *.hpp
