--- a/thirdparty/ocl/CMakeLists.txt
+++ b/thirdparty/ocl/CMakeLists.txt
@@ -44,8 +44,7 @@ add_subdirectory(icd_loader EXCLUDE_FROM
 target_compile_definitions(OpenCL INTERFACE OV_GPU_USE_OPENCL_HPP
                                             OV_GPU_OPENCL_HPP_HAS_UUID)
 target_include_directories(OpenCL INTERFACE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cl_headers>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/clhpp_headers/include>)
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cl_headers>)
 
 # The following variables are needed to make find_package(OpenCL) work
 
