--- Redis-1.991/t/04-pipeline.t
+++ 04-pipeline.t
@@ -45,7 +45,7 @@
 pipeline_ok 'single-command pipeline', ([set => [foo => 'bar'], 'OK'],);
 
 pipeline_ok 'pipeline with embedded error',
-  ([set => [clunk => 'eth'], 'OK'], [oops => [], undef, q[ERR unknown command 'OOPS']], [get => ['clunk'], 'eth'],);
+  ([set => [clunk => 'eth'], 'OK'], [oops => [], undef, re(qq{^ERR unknown command .OOPS.})], [get => ['clunk'], 'eth'],);
 
 pipeline_ok 'keys in pipelined mode',
   ([keys => ['*'], bag(qw<foo clunk>)], [keys => [], undef, q[ERR wrong number of arguments for 'keys' command]],);

