--- jigdo-0.7.3.orig/scripts/jigdo-lite	2017-12-07 16:58:48.404806438 +0000
+++ jigdo-0.7.3/scripts/jigdo-lite	2017-12-07 17:00:09.493596969 +0000
@@ -251,6 +251,8 @@
     "$tmpDir" "$jigdoOpts" >>"$rcFile"
   printf "wgetOpts='%s'${nl}scanMenu='%s'${nl}" \
     "$wgetOpts" "$scanMenu" >>"$rcFile"
+  printf "filesPerFetch='%s'${nl}" \
+    "$filesPerFetch" >>"$rcFile"
 }
 #______________________________________________________________________
 
