Description: Fix narrow conversion error with GCC-6
Author: Anton Gladky <gladk@debian.org>
Bug-Debian: https://bugs.debian.org/811792
Last-Update: 2016-01-30

Index: gmsh-2.12.0-source/Fltk/FlGui.cpp
===================================================================
--- gmsh-2.12.0-source.orig/Fltk/FlGui.cpp
+++ gmsh-2.12.0-source/Fltk/FlGui.cpp
@@ -370,7 +370,7 @@ FlGui::FlGui(int argc, char **argv)
   // nothing to do here
 #else
   fl_open_display();
-  static char gmsh32x32[] = {
+  static unsigned char gmsh32x32[] = {
     0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0x01, 0x00, 0x00, 0x40, 0x03, 0x00,
     0x00, 0x40, 0x03, 0x00, 0x00, 0x20, 0x07, 0x00, 0x00, 0x20, 0x07, 0x00,
     0x00, 0x10, 0x0f, 0x00, 0x00, 0x10, 0x0f, 0x00, 0x00, 0x08, 0x1f, 0x00,
@@ -384,7 +384,7 @@ FlGui::FlGui(int argc, char **argv)
     0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00};
   graph[0]->getWindow()->icon
     ((const char*)XCreateBitmapFromData(fl_display, DefaultRootWindow(fl_display),
-                                        gmsh32x32, 32, 32));
+                                        (char*)(gmsh32x32), 32, 32));
 #endif
 
   graph[0]->getWindow()->show(argc >0 ? 1 : 0, argv);
