Description: Don't create the top panel.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: no, Ubuntu-specific

Index: gnome-screensaver-3.6.1/src/gs-window-x11.c
===================================================================
--- gnome-screensaver-3.6.1.orig/src/gs-window-x11.c	2014-01-07 18:33:03.680132713 +0100
+++ gnome-screensaver-3.6.1/src/gs-window-x11.c	2014-01-07 18:39:24.492120025 +0100
@@ -367,7 +367,8 @@
         monitor = gdk_screen_get_monitor_at_window (screen,
                                                     gtk_widget_get_window (widget));
         primary_monitor = gdk_screen_get_primary_monitor (screen);
-        gtk_widget_set_visible (window->priv->panel, monitor == primary_monitor);
+        if (window->priv->panel)
+                gtk_widget_set_visible (window->priv->panel, monitor == primary_monitor);
 }
 
 static void
@@ -2327,11 +2328,20 @@
         gtk_widget_show (window->priv->vbox);
         gtk_container_add (GTK_CONTAINER (window), window->priv->vbox);
 
-        window->priv->panel = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 12);
-        gtk_widget_show (window->priv->panel);
-        gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->panel, FALSE, FALSE, 0);
-        create_panel (window);
-
+        /* Only display top bar if we're not running under Unity */
+        if (!g_getenv ("XDG_CURRENT_DESKTOP") ||
+            strcmp (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") != 0) {
+                window->priv->panel = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 12);
+                gtk_widget_show (window->priv->panel);
+                gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->panel, FALSE, FALSE, 0);
+                create_panel (window);
+        } else {
+                window->priv->clock = gtk_label_new (NULL);
+                gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->clock, FALSE, FALSE, 0);
+                gtk_misc_set_alignment (GTK_MISC (window->priv->clock), 1.0, 0.5);
+                gtk_misc_set_padding (GTK_MISC (window->priv->clock), 10, 0);
+                gtk_widget_show(window->priv->clock);
+        }
         window->priv->drawing_area = gtk_drawing_area_new ();
         gtk_widget_show (window->priv->drawing_area);
         gtk_widget_set_app_paintable (window->priv->drawing_area, TRUE);
