diff --git a/src/podcast/podcast.pri b/src/podcast/podcast.pri
index 7e6ea9ee..e96263fa 100644
--- a/src/podcast/podcast.pri
+++ b/src/podcast/podcast.pri
@@ -69,8 +69,8 @@ linux-g++* {
             -lxcb-xfixes \
             -lxcb-render -lxcb-shape -lxcb -lX11 -lasound -lSDL -lx264 -lpthread -lvpx -lvorbisenc -lvorbis -ltheoraenc -ltheoradec -logg -lopus -lmp3lame -lfreetype -lfdk-aac -lass -llzma -lbz2 -lz -ldl -lswresample -lswscale -lavutil -lm
 
-    UBUNTU_VERSION = $$system(lsb_release -irs)
-    equals(UBUNTU_VERSION, Ubuntu 14.04) {
+    FFMPEG_VERSION = $$system(ffmpeg --version|& grep -oP "version.*?\K[0-9]\.[0-9]")
+    equals(FFMPEG_VERSION, 2.8) {
         LIBS -= -lswresample
         LIBS += -lavresample
     }
