diff --git a/daemon/gdm-local-display-factory.c b/daemon/gdm-local-display-factory.c
index 4f0b84f0b..ebc11e6a1 100644
--- a/daemon/gdm-local-display-factory.c
+++ b/daemon/gdm-local-display-factory.c
@@ -291,7 +291,8 @@ on_display_status_changed (GdmDisplay             *display,
                 if (is_local &&
                     ((g_strcmp0 (session_class, "greeter") != 0 &&
                       (!seat_active_session || g_strcmp0(session_id, seat_active_session) == 0)) ||
-                     factory->active_vt == GDM_INITIAL_VT || g_strcmp0 (seat_id, "seat0") != 0)) {
+                     (sd_seat_can_tty (seat_id) && factory->active_vt == GDM_INITIAL_VT) ||
+                     (!sd_seat_can_tty (seat_id) && g_strcmp0 (session_class, "greeter") == 0))) {
                         /* reset num failures */
                         factory->num_failures = 0;
 
@@ -366,21 +367,6 @@ lookup_prepared_display_by_seat_id (const char *id,
         return lookup_by_seat_id (id, display, user_data);
 }
 
-static gboolean
-lookup_managed_display_by_seat_id (const char *id,
-                                   GdmDisplay *display,
-                                   gpointer    user_data)
-{
-        int status;
-
-        status = gdm_display_get_status (display);
-
-        if (status != GDM_DISPLAY_MANAGED)
-                return FALSE;
-
-        return lookup_by_seat_id (id, display, user_data);
-}
-
 #ifdef HAVE_UDEV
 static gboolean
 udev_is_settled (GdmLocalDisplayFactory *factory)
@@ -513,19 +499,11 @@ static GdmDisplay *
 get_display_for_seat (GdmLocalDisplayFactory *factory,
                       const char             *seat_id)
 {
-        GdmDisplay *display;
         GdmDisplayStore *store;
-        gboolean is_seat0;
 
         store = gdm_display_factory_get_display_store (GDM_DISPLAY_FACTORY (factory));
 
-        is_seat0 = g_strcmp0 (seat_id, "seat0") == 0;
-        if (is_seat0)
-                display = gdm_display_store_find (store, lookup_prepared_display_by_seat_id, (gpointer) seat_id);
-        else
-                display = gdm_display_store_find (store, lookup_managed_display_by_seat_id, (gpointer) seat_id);
-
-        return display != NULL ? g_object_ref (display) : NULL;
+        return gdm_display_store_find (store, lookup_prepared_display_by_seat_id, (gpointer) seat_id);
 }
 
 static void
diff --git a/daemon/gdm-manager.c b/daemon/gdm-manager.c
index cf0c2c8ca..12f096855 100644
--- a/daemon/gdm-manager.c
+++ b/daemon/gdm-manager.c
@@ -1627,6 +1627,13 @@ on_start_user_session (StartUserSessionOperation *operation)
 
         g_object_unref (display);
 
+        if (gdm_session_get_display_mode (operation->session) == GDM_SESSION_DISPLAY_MODE_LOGIND_MANAGED) {
+                gdm_activate_session_by_id (operation->manager->connection,
+                                            NULL,
+                                            gdm_session_get_display_seat_id (operation->session),
+                                            session_id);
+        }
+
         start_user_session (operation->manager, operation);
 
  out:
@@ -1649,7 +1656,7 @@ queue_start_user_session (GdmManager *manager,
         g_object_set_data (G_OBJECT (session), "start-user-session-operation", operation);
 }
 
-static void
+static gboolean
 start_user_session_if_ready (GdmManager *manager,
                              GdmSession *session,
                              const char *service_name)
@@ -1663,6 +1670,8 @@ start_user_session_if_ready (GdmManager *manager,
         } else {
                 g_object_set_data (G_OBJECT (session), "waiting-to-start", GINT_TO_POINTER (TRUE));
         }
+
+        return start_when_ready;
 }
 
 static void
@@ -1704,6 +1713,8 @@ on_user_session_opened (GdmSession       *session,
                         const char       *session_id,
                         GdmManager       *manager)
 {
+        gboolean start_when_ready;
+
         manager->user_sessions = g_list_append (manager->user_sessions,
                                                       g_object_ref (session));
         if (g_strcmp0 (service_name, "gdm-autologin") == 0 &&
@@ -1713,7 +1724,20 @@ on_user_session_opened (GdmSession       *session,
                 g_object_set_data (G_OBJECT (session), "start-when-ready", GINT_TO_POINTER (TRUE));
         }
 
-        start_user_session_if_ready (manager, session, service_name);
+        start_when_ready = start_user_session_if_ready (manager, session, service_name);
+
+        if (!start_when_ready &&
+            gdm_session_get_display_mode (session) == GDM_SESSION_DISPLAY_MODE_LOGIND_MANAGED) {
+                /* systemd-logind automatically activates newly created sessions on seats
+                 * without VTs. Go back to greeter so it can run animations as well as prompt
+                 * for terminating conflicting sessions. */
+                GdmDisplay *display = get_display_for_user_session (session);
+
+                gdm_activate_session_by_id (manager->connection,
+                                            NULL,
+                                            gdm_session_get_display_seat_id (session),
+                                            gdm_display_get_session_id (display));
+        }
 }
 
 static void
diff --git a/daemon/gdm-session.c b/daemon/gdm-session.c
index 5c04ad1be..5c875823f 100644
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -3437,7 +3437,8 @@ gdm_session_get_display_mode (GdmSession *self)
                  self->is_program_session? "yes" : "no",
                  self->display_seat_id);
 
-        if (g_strcmp0 (self->display_seat_id, "seat0") != 0) {
+        if (self->display_seat_id != NULL &&
+            !sd_seat_can_tty (self->display_seat_id)) {
                 return GDM_SESSION_DISPLAY_MODE_LOGIND_MANAGED;
         }
 
