From 849dae3c2500d28c17f536c21d54a0affe1d734b Mon Sep 17 00:00:00 2001
From: Tommy Carpenter <tommyjcarpenter@gmail.com>
Date: Mon, 31 Dec 2018 08:10:46 -0500
Subject: [PATCH] Fixes for 4.20 suggested by @ddscentral

---
 exfat_oal.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/exfat_oal.c b/exfat_oal.c
index 7435442..4d5600b 100644
--- a/exfat_oal.c
+++ b/exfat_oal.c
@@ -128,13 +128,13 @@ static time_t accum_days_in_year[] = {
 
 TIMESTAMP_T *tm_current(TIMESTAMP_T *tp)
 {
-	struct timespec ts;
+	struct timespec64 ts;
 	time_t second, day, leap_day, month, year;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,8,0)
 	ts = CURRENT_TIME_SEC;
 #else
-	ktime_get_real_ts(&ts);
+	ktime_get_real_ts64(&ts);
 #endif
 
 	second = ts.tv_sec;