From 412458d281f98ea7649d2957f88ebad701b6cd18 Mon Sep 17 00:00:00 2001
From: ronys <ronys@pwsafe.org>
Date: Wed, 6 Nov 2019 22:44:17 +0200
Subject: [PATCH] Fix time64 type issue on newer glibc

---
 src/os/unix/pws_time.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/os/unix/pws_time.h b/src/os/unix/pws_time.h
index 683cb48b9..3407fe702 100644
--- a/src/os/unix/pws_time.h
+++ b/src/os/unix/pws_time.h
@@ -14,8 +14,11 @@
 #include <time.h>
 #endif
 
-#if !defined(__time64_t) && !defined(__TIME64_T_TYPE)
-typedef uint64_t __time64_t;
+#ifndef __TIME64_T_TYPE
+#define __TIME64_T_TYPE uint64_t
+#endif
+#ifndef time64_t
+typedef __TIME64_T_TYPE __time64_t;
 #endif
 
 extern int localtime64_r(const __time64_t *timep, struct tm *result);
