diff -Naur a/ac101.c b/ac101.c
--- a/ac101.c	2020-11-28 23:49:08.317226822 +0100
+++ b/ac101.c	2020-11-28 23:57:19.838831271 +0100
@@ -772,18 +772,16 @@
 
 
 static const DECLARE_TLV_DB_SCALE(dac_vol_tlv, -11925, 75, 0);
-/*
 static const DECLARE_TLV_DB_SCALE(dac_mix_vol_tlv, -600, 600, 0);
 static const DECLARE_TLV_DB_SCALE(dig_vol_tlv, -7308, 116, 0);
-*/
 static const DECLARE_TLV_DB_SCALE(speaker_vol_tlv, -4800, 150, 0);
 static const DECLARE_TLV_DB_SCALE(headphone_vol_tlv, -6300, 100, 0);
 
 static struct snd_kcontrol_new ac101_controls[] = {
 	/*DAC*/
-	SOC_DOUBLE_TLV("DAC Playback Volume", DAC_VOL_CTRL, DAC_VOL_L, DAC_VOL_R, 0xff, 0, dac_vol_tlv),
-	/* SOC_DOUBLE_TLV("DAC Mix Gain", DAC_MXR_GAIN, DACL_MXR_GAIN, DACR_MXR_GAIN, 0xf, 0, dac_mix_vol_tlv), */
-	/* SOC_SINGLE_TLV("DAC Debug Volume", DAC_DBG_CTRL, DVC, 0x3f, 1, dig_vol_tlv), */
+	SOC_DOUBLE_TLV("DAC volume", DAC_VOL_CTRL, DAC_VOL_L, DAC_VOL_R, 0xff, 0, dac_vol_tlv),
+	SOC_DOUBLE_TLV("DAC mixer gain", DAC_MXR_GAIN, DACL_MXR_GAIN, DACR_MXR_GAIN, 0xf, 0, dac_mix_vol_tlv),
+	SOC_SINGLE_TLV("digital volume", DAC_DBG_CTRL, DVC, 0x3f, 1, dig_vol_tlv),
 	SOC_SINGLE_TLV("Speaker Playback Volume", SPKOUT_CTRL, SPK_VOL, 0x1f, 0, speaker_vol_tlv),
 	SOC_SINGLE_TLV("Headphone Playback Volume", HPOUT_CTRL, HP_VOL, 0x3f, 0, headphone_vol_tlv),
 };
@@ -957,10 +955,10 @@
 
 	AC101_DBG("stream = %s, play: %d, capt: %d, active: %d\n", 
 		snd_pcm_stream_str(substream),
-		codec_dai->playback_active, codec_dai->capture_active,
-		codec_dai->active);
+		codec_dai->stream_active[SNDRV_PCM_STREAM_PLAYBACK], codec_dai->stream_active[SNDRV_PCM_STREAM_CAPTURE],
+		snd_soc_dai_active(codec_dai));
 
-	if (!codec_dai->active) {
+	if (snd_soc_dai_active(codec_dai)) {
 		ac10x->aif1_clken = 1;
 		ac101_aif1clk(codec, SND_SOC_DAPM_POST_PMD, 0);
 	} else {
@@ -1082,7 +1080,7 @@
 	freq_out = _FREQ_24_576K;
 	for (i = 0; i < ARRAY_SIZE(codec_aif1_fs); i++) {
 		if (codec_aif1_fs[i].samp_rate == params_rate(params)) {
-			if (codec_dai->capture_active && dmic_used && codec_aif1_fs[i].samp_rate == 44100) {
+			if (codec_dai->stream_active[SNDRV_PCM_STREAM_CAPTURE] && dmic_used && codec_aif1_fs[i].samp_rate == 44100) {
 				ac101_update_bits(codec, AIF_SR_CTRL, (0xf<<AIF1_FS), (0x4<<AIF1_FS));
 			} else {
 				ac101_update_bits(codec, AIF_SR_CTRL, (0xf<<AIF1_FS), ((codec_aif1_fs[i].srbit)<<AIF1_FS));
diff -Naur a/ac108_6mic.state b/ac108_6mic.state
--- a/ac108_6mic.state	2020-11-28 23:49:08.317226822 +0100
+++ b/ac108_6mic.state	2020-11-28 23:57:19.839831269 +0100
@@ -292,7 +292,7 @@
 	}
 	control.17 {
 		iface MIXER
-		name 'DAC Playback Volume'
+		name 'DAC volume'
 		value.0 0
 		value.1 0
 		comment {
@@ -308,6 +308,36 @@
 	}
 	control.18 {
 		iface MIXER
+		name 'DAC mixer gain'
+		value.0 0
+		value.1 0
+		comment {
+			access 'read write'
+			type INTEGER
+			count 2
+			range '0 - 15'
+			dbmin -600
+			dbmax 8400
+			dbvalue.0 -600
+			dbvalue.1 -600
+		}
+	}
+	control.19 {
+		iface MIXER
+		name 'digital volume'
+		value 51
+		comment {
+			access 'read write'
+			type INTEGER
+			count 1
+			range '0 - 63'
+			dbmin -7308
+			dbmax 0
+			dbvalue.0 -1392
+		}
+	}
+	control.20 {
+		iface MIXER
 		name 'Speaker Playback Volume'
 		value 25
 		comment {
@@ -320,7 +350,7 @@
 			dbvalue.0 -1050
 		}
 	}
-	control.19 {
+	control.21 {
 		iface MIXER
 		name 'Headphone Playback Volume'
 		value 52
diff -Naur a/ac108.c b/ac108.c
--- a/ac108.c	2020-11-28 23:49:08.317226822 +0100
+++ b/ac108.c	2020-11-28 23:57:19.839831269 +0100
@@ -653,7 +653,7 @@
 
 	dev_dbg(dai->dev, "%s() stream=%s play:%d capt:%d +++\n", __func__,
 			snd_pcm_stream_str(substream),
-			dai->playback_active, dai->capture_active);
+			dai->stream_active[SNDRV_PCM_STREAM_PLAYBACK], dai->stream_active[SNDRV_PCM_STREAM_CAPTURE]);
 
 	if (ac10x->i2c101) {
 		ret = ac101_hw_params(substream, params, dai);
@@ -664,8 +664,8 @@
 		}
 	}
 
-	if ((substream->stream == SNDRV_PCM_STREAM_CAPTURE && dai->playback_active)
-	 || (substream->stream == SNDRV_PCM_STREAM_PLAYBACK && dai->capture_active)) {
+	if ((substream->stream == SNDRV_PCM_STREAM_CAPTURE && dai->stream_active[SNDRV_PCM_STREAM_PLAYBACK])
+	 || (substream->stream == SNDRV_PCM_STREAM_PLAYBACK && dai->stream_active[SNDRV_PCM_STREAM_CAPTURE])) {
 		/* not configure hw_param twice */
 		/* return 0; */
 	}
@@ -1124,7 +1124,7 @@
 	}
 }
 
-int ac108_aif_mute(struct snd_soc_dai *dai, int mute) {
+int ac108_aif_mute(struct snd_soc_dai *dai, int mute, int direction) {
 	struct snd_soc_codec *codec = dai->codec;
 	struct ac10x_priv *ac10x = snd_soc_codec_get_drvdata(codec);
 
@@ -1145,12 +1145,13 @@
 	.hw_params	= ac108_hw_params,
 	.prepare	= ac108_prepare,
 	.trigger	= ac108_trigger,
-	.digital_mute	= ac108_aif_mute,
+	.mute_stream	= ac108_aif_mute,
 
 	/*DAI format configuration*/
 	.set_fmt	= ac108_set_fmt,
 
 	// .hw_free = ac108_hw_free,
+	.no_capture_mute = 1,
 };
 
 static  struct snd_soc_dai_driver ac108_dai0 = {
diff -Naur a/install.sh b/install.sh
--- a/install.sh	2020-11-28 23:49:08.318226818 +0100
+++ b/install.sh	2020-11-28 23:57:19.840831267 +0100
@@ -1,27 +1,26 @@
 #!/bin/bash
 
-FORCE_KERNEL="1.20200819-1"
-
 if [[ $EUID -ne 0 ]]; then
    echo "This script must be run as root (use sudo)" 1>&2
    exit 1
 fi
 
-# Check if enough space on /boot volume
-boot_line=$(df -BM | grep /boot | head -n 1)
-MIN_BOOT_SPC=25 # MegaBytes
+# Check for enough space on /boot volume
+boot_line=$(df -h | grep /boot | head -n 1)
 if [ "x${boot_line}" = "x" ]; then
   echo "Warning: /boot volume not found .."
 else
   boot_space=$(echo $boot_line | awk '{print $4;}')
   free_space=$(echo "${boot_space%?}")
   unit="${boot_space: -1}"
-  if [[ "$unit" != "M" ]]; then
-    echo "Warning: /boot volume not found .."
-  elif [ "$free_space" -lt "$MIN_BOOT_SPC" ]; then
+  if [[ "$unit" = "K" ]]; then
     echo "Error: Not enough space left ($boot_space) on /boot"
-    echo "       at least $MIN_BOOT_SPC MB required"
     exit 1
+  elif [[ "$unit" = "M" ]]; then
+    if [ "$free_space" -lt "25" ]; then
+      echo "Error: Not enough space left ($boot_space) on /boot"
+      exit 1
+    fi
   fi
 fi
 
@@ -48,13 +47,6 @@
     errorFound=1
   fi
 done
-
-if [ ! -x seeed-voicecard -o ! -f seeed-voicecard.service ]; then
-  echo "Please run this script in the project directory"
-  echo "which has files such as install.sh and seeed-voicecard.service"
-  errorFound=1
-fi
-
 if [ $errorFound = 1 ] ; then
   echo "Errors found, exiting." 1>&2
   exit 1
@@ -62,7 +54,6 @@
 
 ver="0.3"
 uname_r=$(uname -r)
-arch_r=$(dpkg --print-architecture)
 
 # we create a dir with this version to ensure that 'dkms remove' won't delete
 # the sources during kernel updates
@@ -96,10 +87,10 @@
 
   # echo RUN=$VER_RUN HDR=$VER_HDR
   echo " !!! Your kernel version is $VER_RUN"
-  echo "     Couldn't find *** corresponding *** kernel headers with apt-get."
-  echo "     This may happen if you ran 'rpi-update'."
-  echo " Choose  *** y *** to revert the kernel to version $VER_HDR and continue."
-  echo " Choose  *** N *** to exit without this driver support, by default."
+  echo "     Not found *** corresponding *** kernel headers with apt-get."
+  echo "     This may occur if you have ran 'rpi-update'."
+  echo " Choose  *** y *** will revert the kernel to version $VER_HDR then continue."
+  echo " Choose  *** N *** will exit without this driver support, by default."
   read -p "Would you like to proceed? (y/N)" -n 1 -r -s
   echo
   if ! [[ $REPLY =~ ^[Yy]$ ]]; then
@@ -109,128 +100,15 @@
   apt-get -y --reinstall install raspberrypi-kernel
 }
 
-function download_install_debpkg() {
-  local prefix name r pkg status _name
-  prefix=$1
-  name=$2
-  pkg=${name%%_*}
-
-  status=$(dpkg -l $pkg | tail -1)
-  _name=$(  echo "$status" | awk '{ printf "%s_%s_%s", $2, $3, $4; }')
-  status=$(echo "$status" | awk '{ printf "%s", $1; }')
-
-  if [ "X$status" == "Xii" -a "X${name%.deb}" == "X$_name" ]; then
-    echo "debian package $name already installed."
-    return 0
-  fi
-
-  for (( i = 0; i < 3; i++ )); do
-    wget $prefix$name -O /tmp/$name && break
-  done
-  dpkg -i /tmp/$name; r=$?
-  rm -f /tmp/$name
-  return $r
-}
-
-function usage() {
-  cat <<-__EOF__
-    usage: sudo ./install [ --compat-kernel | --keep-kernel ] [ -h | --help ]
-             default action is update kernel & headers to latest version.
-             --compat-kernel uses an older kernel but ensures that the driver can work.
-             --keep-kernel   don't change/update the system kernel, maybe install
-                             coressponding kernel headers.
-             --help          show this help message
-__EOF__
-  exit 1
-}
-
-compat_kernel=
-keep_kernel=
-# parse commandline options
-while [ ! -z "$1" ] ; do
-  case $1 in
-  -h|--help)
-    usage
-    ;;
-  --compat-kernel)
-    compat_kernel=Y
-    ;;
-  --keep-kernel)
-    keep_kernel=Y
-    ;;
-  esac
-  shift
-done
-
-if [ "X$keep_kernel" != "X" ]; then
-  FORCE_KERNEL=$(dpkg -s raspberrypi-kernel | awk '/^Version:/{printf "%s\n",$2;}')
-  echo -e "\n### Keep current system kernel not to change"
-elif [ "X$compat_kernel" != "X" ]; then
-  echo -e "\n### will compile with a compatible kernel..."
-else
-  FORCE_KERNEL=""
-  echo -e "\n### will compile with the latest kernel..."
-fi
-[ "X$FORCE_KERNEL" != "X" ] && {
-  echo -e "The kernel & headers use package version: $FORCE_KERNEL\r\n\r\n"
-}
-
-function install_kernel() {
-  local _url _prefix
-
-  # Instead of retrieving the lastest kernel & headers
-  [ "X$FORCE_KERNEL" == "X" ] && {
-    # Raspbian kernel packages
-    apt-get -y --force-yes install raspberrypi-kernel-headers raspberrypi-kernel || {
-      # Ubuntu kernel packages
-      apt-get -y install linux-raspi linux-headers-raspi linux-image-raspi
-    }
-  } || {
-    # We would like to a fixed version
-    KERN_NAME=raspberrypi-kernel_${FORCE_KERNEL}_${arch_r}.deb
-    HDR_NAME=raspberrypi-kernel-headers_${FORCE_KERNEL}_${arch_r}.deb
-    _url=$(apt-get download --print-uris raspberrypi-kernel | sed -nre "s/'([^']+)'.*$/\1/g;p")
-    _prefix=$(echo $_url | sed -nre 's/^(.*)raspberrypi-kernel_.*$/\1/g;p')
-
-    download_install_debpkg "$_prefix" "$KERN_NAME" && {
-      download_install_debpkg "$_prefix" "$HDR_NAME"
-    } || {
-      echo "Error: Install kernel or header failed"
-      exit 2
-    }
-  }
-}
-
-function uninstall_module {
-  src=$1
-  mod=$2
-
-  if [[ -d /var/lib/dkms/$mod/$ver/$marker ]]; then
-    rmdir /var/lib/dkms/$mod/$ver/$marker
-  fi
-
-  if [[ -e /usr/src/$mod-$ver || -e /var/lib/dkms/$mod/$ver ]]; then
-    dkms remove --force -m $mod -v $ver --all
-    rm -rf /usr/src/$mod-$ver
-  fi
-
-  return 0
-}
-
 # update and install required packages
-which apt &>/dev/null; r=$?
-if [[ $r -eq 0 ]]; then
-  echo -e "\n### Install required tool packages"
+which apt &>/dev/null
+if [[ $? -eq 0 ]]; then
   apt update -y
+  # Raspbian kernel packages
+  apt-get -y install raspberrypi-kernel-headers raspberrypi-kernel 
+  # Ubuntu kernel packages
+  apt-get -y install linux-raspi linux-headers-raspi linux-image-raspi
   apt-get -y install dkms git i2c-tools libasound2-plugins
-fi
-
-echo -e "\n### Uninstall previous dkms module"
-uninstall_module "./" "seeed-voicecard"
-
-if [[ $r -eq 0 ]]; then
-  echo -e "\n### Install required kernel package"
-  install_kernel
   # rpi-update checker
   check_kernel_headers
 fi
@@ -245,9 +123,8 @@
 # it's just been updated)
 base_ver=$(get_kernel_version)
 base_ver=${base_ver%%[-+]*}
-# kernels="${base_ver}+ ${base_ver}-v7+ ${base_ver}-v7l+"
-# select exact kernel postfix
-kernels=${base_ver}$(echo $uname_r | sed -re 's/^[0-9.]+(.*)/\1/g')
+#kernels="${base_ver}+ ${base_ver}-v7+ ${base_ver}-v7l+"
+kernels=$(uname -r)
 
 function install_module {
   local _i
@@ -255,6 +132,15 @@
   src=$1
   mod=$2
 
+  if [[ -d /var/lib/dkms/$mod/$ver/$marker ]]; then
+    rmdir /var/lib/dkms/$mod/$ver/$marker
+  fi
+
+  if [[ -e /usr/src/$mod-$ver || -e /var/lib/dkms/$mod/$ver ]]; then
+    dkms remove --force -m $mod -v $ver --all
+    rm -rf /usr/src/$mod-$ver
+  fi
+
   mkdir -p /usr/src/$mod-$ver
   cp -a $src/* /usr/src/$mod-$ver/
 
@@ -262,32 +148,26 @@
   for _i in $kernels; do
     dkms build -k $_i -m $mod -v $ver && {
       dkms install --force -k $_i -m $mod -v $ver
-    } || {
-      echo "Can't compile with this kernel, aborting"
-      echo "Please try to compile with the option --compat-kernel"
-      exit 1
     }
   done
 
   mkdir -p /var/lib/dkms/$mod/$ver/$marker
 }
 
-echo -e "\n### Install sound card driver"
 install_module "./" "seeed-voicecard"
 
+
 # install dtbos
-echo -e "\n### Install device tree overlays"
-cp -v seeed-2mic-voicecard.dtbo $OVERLAYS
-cp -v seeed-4mic-voicecard.dtbo $OVERLAYS
-cp -v seeed-8mic-voicecard.dtbo $OVERLAYS
+cp seeed-2mic-voicecard.dtbo $OVERLAYS
+cp seeed-4mic-voicecard.dtbo $OVERLAYS
+cp seeed-8mic-voicecard.dtbo $OVERLAYS
 
-# install alsa plugins
-# we don't need this plugin now
+#install alsa plugins
+# no need this plugin now
 # install -D ac108_plugin/libasound_module_pcm_ac108.so /usr/lib/arm-linux-gnueabihf/alsa-lib/
 rm -f /usr/lib/arm-linux-gnueabihf/alsa-lib/libasound_module_pcm_ac108.so
 
 #set kernel modules
-echo -e "\n### Codec driver loading at startup (in /etc/modules)"
 grep -q "^snd-soc-seeed-voicecard$" /etc/modules || \
   echo "snd-soc-seeed-voicecard" >> /etc/modules
 grep -q "^snd-soc-ac108$" /etc/modules || \
@@ -298,7 +178,6 @@
 #set dtoverlays
 CONFIG=/boot/config.txt
 [ -f /boot/firmware/usercfg.txt ] && CONFIG=/boot/firmware/usercfg.txt
-echo -e "\n### Found boot configuration file $CONFIG"
 
 sed -i -e 's:#dtparam=i2c_arm=on:dtparam=i2c_arm=on:g'  $CONFIG || true
 grep -q "^dtoverlay=i2s-mmap$" $CONFIG || \
@@ -309,13 +188,11 @@
   echo "dtparam=i2s=on" >> $CONFIG
 
 #install config files
-echo -e "\n### Install alsa and widget configuration"
 mkdir /etc/voicecard || true
-cp -v *.conf /etc/voicecard
-cp -v *.state /etc/voicecard
+cp *.conf /etc/voicecard
+cp *.state /etc/voicecard
 
 #create git repo
-echo -e "\n### Manage alsa configuration by git"
 git_email=$(git config --global --get user.email)
 git_name=$(git config --global --get user.name)
 if [ "x${git_email}" == "x" ] || [ "x${git_name}" == "x" ] ; then
@@ -330,14 +207,12 @@
 echo "git commit -m \"origin configures\""
 git --git-dir=/etc/voicecard/.git --work-tree=/etc/voicecard/ commit  -m "origin configures"
 
-echo -e "\n### Start service seeed-voicecard"
-echo -e "    see /var/log/seeed-voicecard.log for more service information"
 cp seeed-voicecard /usr/bin/
 cp seeed-voicecard.service /lib/systemd/system/
 systemctl enable  seeed-voicecard.service 
 systemctl start   seeed-voicecard
 
 echo "------------------------------------------------------"
-echo "Please reboot your device to apply all settings"
+echo "Please reboot your raspberry pi to apply all settings"
 echo "Enjoy!"
 echo "------------------------------------------------------"
diff -Naur a/seeed-voicecard b/seeed-voicecard
--- a/seeed-voicecard	2020-11-28 23:49:08.322226803 +0100
+++ b/seeed-voicecard	2020-11-28 23:57:19.843831260 +0100
@@ -21,9 +21,7 @@
 # THE SOFTWARE.
 
 set -x
-LOGF=/var/log/$(basename $0).log
-exec 1> >(tee $LOGF)
-exec 2>&1
+#exec 1>/var/log/$(basename $0).log 2>&1
 
 export PATH=$PATH:/opt/vc/bin
 OVERLAYS=/boot/overlays
@@ -124,8 +122,7 @@
     rm /etc/asound.conf
     rm /var/lib/alsa/asound.state
 
-: <<\EOF
-    kernel_ver=$(get_kernel_version)
+    kernel_ver=$(uname -r) # get_kernel_version)
     # echo kernel_ver=$kernel_ver
 
     # TODO: dynamic dtoverlay Bug of v4.19.x
@@ -140,7 +137,6 @@
             fi
         done
     fi
-EOF
     #make sure the driver loads correctly
     dtoverlay -d $OVERLAYS $overlay || true
 
@@ -163,5 +159,3 @@
 if aplay -l | grep -q "bcm2835 ALSA"; then
     amixer cset numid=3 1 || true
 fi
-
-exit 0
diff -Naur a/seeed-voicecard.c b/seeed-voicecard.c
--- a/seeed-voicecard.c	2020-11-28 23:49:08.322226803 +0100
+++ b/seeed-voicecard.c	2020-11-28 23:57:19.843831260 +0100
@@ -96,16 +96,16 @@
 	if (ret)
 		clk_disable_unprepare(dai_props->cpu_dai.clk);
 
-	if (rtd->cpu_dai->driver->playback.channels_min) {
-		priv->channels_playback_default = rtd->cpu_dai->driver->playback.channels_min;
+	if (asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_min) {
+		priv->channels_playback_default = asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_min;
 	}
-	if (rtd->cpu_dai->driver->capture.channels_min) {
-		priv->channels_capture_default = rtd->cpu_dai->driver->capture.channels_min;
+	if (asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_min) {
+		priv->channels_capture_default = asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_min;
 	}
-	rtd->cpu_dai->driver->playback.channels_min = priv->channels_playback_override;
-	rtd->cpu_dai->driver->playback.channels_max = priv->channels_playback_override;
-	rtd->cpu_dai->driver->capture.channels_min = priv->channels_capture_override;
-	rtd->cpu_dai->driver->capture.channels_max = priv->channels_capture_override;
+	asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_min = priv->channels_playback_override;
+	asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_max = priv->channels_playback_override;
+	asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_min = priv->channels_capture_override;
+	asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_max = priv->channels_capture_override;
 
 	return ret;
 }
@@ -117,10 +117,10 @@
 	struct seeed_dai_props *dai_props =
 		seeed_priv_to_props(priv, rtd->num);
 
-	rtd->cpu_dai->driver->playback.channels_min = priv->channels_playback_default;
-	rtd->cpu_dai->driver->playback.channels_max = priv->channels_playback_default;
-	rtd->cpu_dai->driver->capture.channels_min = priv->channels_capture_default;
-	rtd->cpu_dai->driver->capture.channels_max = priv->channels_capture_default;
+	asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_min = priv->channels_playback_default;
+	asoc_rtd_to_cpu(rtd, 0)->driver->playback.channels_max = priv->channels_playback_default;
+	asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_min = priv->channels_capture_default;
+	asoc_rtd_to_cpu(rtd, 0)->driver->capture.channels_max = priv->channels_capture_default;
 
 	clk_disable_unprepare(dai_props->cpu_dai.clk);
 
@@ -131,8 +131,8 @@
 				      struct snd_pcm_hw_params *params)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *codec_dai = rtd->codec_dai;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
+	struct snd_soc_dai *codec_dai = asoc_rtd_to_codec(rtd, 0);
+	struct snd_soc_dai *cpu_dai = asoc_rtd_to_cpu(rtd, 0);
 	struct seeed_card_data *priv = snd_soc_card_get_drvdata(rtd->card);
 	struct seeed_dai_props *dai_props =
 		seeed_priv_to_props(priv, rtd->num);
@@ -196,7 +196,7 @@
 static int seeed_voice_card_trigger(struct snd_pcm_substream *substream, int cmd)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *dai = rtd->codec_dai;
+	struct snd_soc_dai *dai = asoc_rtd_to_codec(rtd, 0);
 	struct seeed_card_data *priv = snd_soc_card_get_drvdata(rtd->card);
 	#if CONFIG_AC10X_TRIG_LOCK
 	unsigned long flags;
@@ -205,7 +205,7 @@
 
 	dev_dbg(rtd->card->dev, "%s() stream=%s  cmd=%d play:%d, capt:%d\n",
 		__FUNCTION__, snd_pcm_stream_str(substream), cmd,
-		dai->playback_active, dai->capture_active);
+		dai->stream_active[SNDRV_PCM_STREAM_PLAYBACK], dai->stream_active[SNDRV_PCM_STREAM_CAPTURE]);
 
 	switch (cmd) {
 	case SNDRV_PCM_TRIGGER_START:
@@ -227,7 +227,7 @@
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 		/* capture channel resync, if overrun */
-		if (dai->capture_active && substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
+		if (dai->stream_active[SNDRV_PCM_STREAM_CAPTURE] && substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
 			break;
 		}
 
@@ -337,8 +337,8 @@
 static int seeed_voice_card_dai_init(struct snd_soc_pcm_runtime *rtd)
 {
 	struct seeed_card_data *priv =	snd_soc_card_get_drvdata(rtd->card);
-	struct snd_soc_dai *codec = rtd->codec_dai;
-	struct snd_soc_dai *cpu = rtd->cpu_dai;
+	struct snd_soc_dai *codec = asoc_rtd_to_codec(rtd, 0);
+	struct snd_soc_dai *cpu = asoc_rtd_to_cpu(rtd, 0);
 	struct seeed_dai_props *dai_props =
 		seeed_priv_to_props(priv, rtd->num);
 	int ret;
@@ -636,11 +636,11 @@
 	 * Use snd_soc_dai_link_component instead of legacy style
 	 * It is codec only. but cpu/platform will be supported in the future.
 	 * see
-	 *	soc-core.c :: snd_soc_init_multicodec()
+	 *      soc-core.c :: snd_soc_init_multicodec()
 	 *
 	 * "platform" might be removed
 	 * see
-	 *	simple-card-utils.c :: asoc_simple_canonicalize_platform()
+	 *      simple-card-utils.c :: asoc_simple_canonicalize_platform()
 	 */
 	for (i = 0; i < num; i++) {
 		dai_link[i].cpus		= &dai_props[i].cpus;
diff -Naur a/sound-compatible-4.18.h b/sound-compatible-4.18.h
--- a/sound-compatible-4.18.h	2020-11-28 23:49:08.322226803 +0100
+++ b/sound-compatible-4.18.h	2020-11-28 23:57:19.844831258 +0100
@@ -31,7 +31,11 @@
 #define snd_soc_codec_get_dapm     snd_soc_component_get_dapm
 #define snd_soc_codec_get_bias_level snd_soc_component_get_bias_level
 #define snd_soc_kcontrol_codec     snd_soc_kcontrol_component
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,9,0)
+#define snd_soc_read               snd_soc_component_read
+#else
 #define snd_soc_read               snd_soc_component_read32
+#endif
 #define snd_soc_register_codec     snd_soc_register_component
 #define snd_soc_unregister_codec   snd_soc_unregister_component
 #define snd_soc_update_bits        snd_soc_component_update_bits
diff -Naur a/uninstall.sh b/uninstall.sh
--- a/uninstall.sh	2020-11-28 23:49:08.322226803 +0100
+++ b/uninstall.sh	2020-11-28 23:57:19.844831258 +0100
@@ -54,45 +54,40 @@
 RPI_HATS="seeed-2mic-voicecard seeed-4mic-voicecard seeed-8mic-voicecard"
 
 PATH=$PATH:/opt/vc/bin
-echo -e "\n### Remove dtbos"
+echo "remove dtbos"
 for i in $RPI_HATS; do
     dtoverlay -r $i
 done
 OVERLAYS=/boot/overlays
 [ -d /boot/firmware/overlays ] && OVERLAYS=/boot/firmware/overlays
 
-rm -v ${OVERLAYS}/seeed-2mic-voicecard.dtbo || true
-rm -v ${OVERLAYS}/seeed-4mic-voicecard.dtbo || true
-rm -v ${OVERLAYS}/seeed-8mic-voicecard.dtbo || true
+rm  ${OVERLAYS}/seeed-2mic-voicecard.dtbo || true
+rm  ${OVERLAYS}/seeed-4mic-voicecard.dtbo || true
+rm  ${OVERLAYS}/seeed-8mic-voicecard.dtbo || true
 
-echo -e "\n### Remove alsa configs"
-rm -rfv  /etc/voicecard/ || true
+echo "remove alsa configs"
+rm -rf  /etc/voicecard/ || true
 
-echo -e "\n### Disable seeed-voicecard.service "
+echo "disabled seeed-voicecard.service "
 systemctl stop seeed-voicecard.service
 systemctl disable seeed-voicecard.service 
 
-echo -e "\n### Remove seeed-voicecard"
-rm -v /usr/bin/seeed-voicecard || true
-rm -v /lib/systemd/system/seeed-voicecard.service || true
-
-echo -e "\n### Remove dkms"
-rm -rfv /var/lib/dkms/seeed-voicecard || true
-
-echo -e "\n### Unload codec driver by modprobe -r"
-modprobe -r snd_soc_ac108
-modprobe -r snd_soc_wm8960
-modprobe -r snd_soc_seeed_voicecard
-
-echo -e "\n### Remove kernel modules"
-rm -v /lib/modules/*/kernel/sound/soc/codecs/snd-soc-wm8960.ko || true
-rm -v /lib/modules/*/kernel/sound/soc/codecs/snd-soc-ac108.ko || true
-rm -v /lib/modules/*/kernel/sound/soc/bcm/snd-soc-seeed-voicecard.ko || true
-rm -v /lib/modules/*/updates/dkms/snd-soc-wm8960.ko || true
-rm -v /lib/modules/*/updates/dkms/snd-soc-ac108.ko || true
-rm -v /lib/modules/*/updates/dkms/snd-soc-seeed-voicecard.ko || true
+echo "remove seeed-voicecard"
+rm  /usr/bin/seeed-voicecard || true
+rm  /lib/systemd/system/seeed-voicecard.service || true
+
+echo "remove dkms"
+rm  -rf /var/lib/dkms/seeed-voicecard || true
+
+echo "remove kernel modules"
+rm  /lib/modules/*/kernel/sound/soc/codecs/snd-soc-wm8960.ko || true
+rm  /lib/modules/*/kernel/sound/soc/codecs/snd-soc-ac108.ko || true
+rm  /lib/modules/*/kernel/sound/soc/bcm/snd-soc-seeed-voicecard.ko || true
+rm  /lib/modules/*/updates/dkms/snd-soc-wm8960.ko || true
+rm  /lib/modules/*/updates/dkms/snd-soc-ac108.ko || true
+rm  /lib/modules/*/updates/dkms/snd-soc-seeed-voicecard.ko || true
 
-echo -e "\n### Remove $CONFIG configuration"
+echo "remove $CONFIG configuration"
 for i in $RPI_HATS; do
     echo Uninstall $i ...
     do_overlay $i 1
diff -Naur a/wm8960.c b/wm8960.c
--- a/wm8960.c	2020-11-28 23:49:08.322226803 +0100
+++ b/wm8960.c	2020-11-28 23:57:19.844831258 +0100
@@ -796,7 +796,7 @@
 	return 0;
 }
 
-static int wm8960_mute(struct snd_soc_dai *dai, int mute)
+static int wm8960_mute(struct snd_soc_dai *dai, int mute, int direction)
 {
 	struct snd_soc_codec *codec = dai->codec;
 
@@ -1236,11 +1236,12 @@
 static const struct snd_soc_dai_ops wm8960_dai_ops = {
 	.hw_params = wm8960_hw_params,
 	.hw_free = wm8960_hw_free,
-	.digital_mute = wm8960_mute,
+	.mute_stream = wm8960_mute,
 	.set_fmt = wm8960_set_dai_fmt,
 	.set_clkdiv = wm8960_set_dai_clkdiv,
 	.set_pll = wm8960_set_dai_pll,
 	.set_sysclk = wm8960_set_dai_sysclk,
+	.no_capture_mute = 1,
 };
 
 static struct snd_soc_dai_driver wm8960_dai = {
