--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project (
   'gala-layoutpw-plugin',
   'vala', 'c',
-  version: '6.3.0'
+  version: '6.3.1'
 )
 
 add_project_arguments ([
@@ -65,6 +65,17 @@
   vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40', '--define', 'HAS_MUTTER41']
 endif
 
+mutter42_dep = dependency('libmutter-10', version: ['>= 42', '< 43'], required: false)
+if mutter42_dep.found()
+  libmutter_dep = dependency('libmutter-10', version: '>= 42')
+  mutter_dep = [
+    libmutter_dep,
+    dependency('mutter-cogl-10'), dependency('mutter-cogl-pango-10'),
+    dependency('mutter-clutter-10')
+  ]
+  vala_flags = ['--define', 'HAS_MUTTER338', '--define', 'HAS_MUTTER40', '--define', 'HAS_MUTTER41', '--define', 'HAS_MUTTER42']
+endif
+
 if mutter_dep.length () == 0
   error ('No supported mutter library found!')
 endif
