diff -Nru genders-1.20.old/configure.ac genders-1.20/configure.ac
--- genders-1.20.old/configure.ac	2012-08-15 16:58:20.000000000 -0500
+++ genders-1.20/configure.ac	2013-05-04 23:15:03.834998392 -0500
@@ -65,6 +65,7 @@
 # Checks for programs.
 ##
 AC_PROG_CC
+AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
diff -Nru genders-1.20.old/Makefile.am genders-1.20/Makefile.am
--- genders-1.20.old/Makefile.am	2007-10-17 12:31:14.000000000 -0500
+++ genders-1.20/Makefile.am	2013-05-04 23:15:03.834998392 -0500
@@ -6,6 +6,8 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --with-perl-destdir=$$dc_install_base
 
+ACLOCAL_AMFLAGS = -I config --install
+
 SUBDIRS = \
         src \
         compat \
diff -Nru genders-1.20.old/src/extensions/perl/Genders/Makefile.am genders-1.20/src/extensions/perl/Genders/Makefile.am
--- genders-1.20.old/src/extensions/perl/Genders/Makefile.am	2011-03-10 18:07:01.000000000 -0600
+++ genders-1.20/src/extensions/perl/Genders/Makefile.am	2013-05-04 23:15:03.834998392 -0500
@@ -15,7 +15,7 @@
 
 Genders.$(PERLMAN3EXT):
 # Hacks to make distcheck work
-	if ! test -a Genders.pm; then \
+	if ! test -e Genders.pm; then \
 		cp $(srcdir)/Genders.pm .; \
 	fi
 	$(POD2MAN) Genders.pm Genders.$(PERLMAN3EXT)
diff -Nru genders-1.20.old/src/extensions/perl/Libgenders/Makefile.am genders-1.20/src/extensions/perl/Libgenders/Makefile.am
--- genders-1.20.old/src/extensions/perl/Libgenders/Makefile.am	2011-03-10 18:07:01.000000000 -0600
+++ genders-1.20/src/extensions/perl/Libgenders/Makefile.am	2013-05-04 23:15:03.834998392 -0500
@@ -32,16 +32,16 @@
 
 Makefile.xs: Makefile.PL Libgenders.xs Libgenders.pm
 # Hacks to make distcheck work
-	if ! test -a Makefile.PL; then \
+	if ! test -e Makefile.PL; then \
 		cp $(srcdir)/Makefile.PL .; \
 	fi
-	if ! test -a typemap; then \
+	if ! test -e typemap; then \
 		cp $(srcdir)/typemap .; \
 	fi
-	if ! test -a Libgenders.pm; then \
+	if ! test -e Libgenders.pm; then \
 		cp $(srcdir)/Libgenders.pm .; \
 	fi
-	if ! test -a Libgenders.xs; then \
+	if ! test -e Libgenders.xs; then \
 		cp $(srcdir)/Libgenders.xs .; \
 	fi
 	$(PERL) Makefile.PL $(MAKEMAKERFLAGS)
@@ -62,7 +62,7 @@
 	$(MAKE) -f Makefile.xs uninstall
 
 clean: 
-	if test -a Makefile.xs; then \
+	if test -e Makefile.xs; then \
 		$(MAKE) -f Makefile.xs clean; \
 	fi
 	rm -f *.c
