From 8e27756e07c8c260fcee847d8f3851181d9e3200 Mon Sep 17 00:00:00 2001
From: Yankes <Yankes@users.noreply.github.com>
Date: Sat, 11 Dec 2021 14:38:12 +0100
Subject: [PATCH] Fix bug in last commit

---
 src/Engine/Script.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Engine/Script.cpp b/src/Engine/Script.cpp
index 543d933084..62680ffa5f 100644
--- a/src/Engine/Script.cpp
+++ b/src/Engine/Script.cpp
@@ -3165,7 +3165,7 @@ void ScriptParserEventsBase::parseCode(ScriptContainerEventsBase& container, con
  */
 void ScriptParserEventsBase::load(const YAML::Node& scripts)
 {
-	ScriptParserBase::load(node);
+	ScriptParserBase::load(scripts);
 
 	if (const YAML::Node& curr = scripts[getName()])
 	{
