--- aclocal.m4	2017-10-07 12:56:34.530515973 +0200
+++ aclocal.m4.new	2017-10-07 12:57:20.510274406 +0200
@@ -48,7 +48,7 @@
 #
 AC_DEFUN([GUILE_PKG],
  [PKG_PROG_PKG_CONFIG
-  _guile_versions_to_search="m4_default([$1], [2.0 1.8])"
+  _guile_versions_to_search="m4_default([$1], [2.2 2.0 1.8])"
   if test -n "$GUILE_EFFECTIVE_VERSION"; then
     _guile_tmp=""
     for v in $_guile_versions_to_search; do
