diff -uNr adafruit-ampy-1.1.0.orig/ampy/cli.py adafruit-ampy-1.1.0/ampy/cli.py
--- adafruit-ampy-1.1.0.orig/ampy/cli.py	2021-02-13 18:23:57.000000000 +0100
+++ adafruit-ampy-1.1.0/ampy/cli.py	2025-04-29 14:27:10.801080616 +0200
@@ -48,7 +48,7 @@
     # 9 are just referred to by COM1, COM2, etc. (wacky!)  See this post for
     # more info and where this code came from:
     # http://eli.thegreenplace.net/2009/07/31/listing-all-serial-ports-on-windows-with-python/
-    m = re.match("^COM(\d+)$", portname)
+    m = re.match(r"^COM(\d+)$", portname)
     if m and int(m.group(1)) < 10:
         return portname
     else:
