diff -aur stopmotion-0.6.2/stopmotion.pro.in stopmotion-0.6.2-patched/stopmotion.pro.in
--- stopmotion-0.6.2/stopmotion.pro.in	2008-07-25 06:59:02.000000000 -0400
+++ stopmotion-0.6.2-patched/stopmotion.pro.in	2012-01-04 11:28:22.000000000 -0500
@@ -213,6 +213,7 @@
 
 LIBS += $$system(sdl-config --libs) \
         $$system(xml2-config --libs) \
+        -lX11 \
         -lSDL_image -ltar -lvorbisfile -linotifytools
 
 DISTFILES += src/config.cpp.in \
