From a5cd2164661b2b8d7bf3ffc64f0632541088a028 Mon Sep 17 00:00:00 2001
From: Didier Raboud <odyx@debian.org>
Date: Sat, 24 Sep 2016 19:04:32 +0200
Subject: Add 'include <cups/ppd.h>' as CUPS 2.2 moved some definitions to
 ppd.h

---
 rastertoptch.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rastertoptch.c b/rastertoptch.c
index 33e7193..9d4db16 100644
--- a/rastertoptch.c
+++ b/rastertoptch.c
@@ -312,6 +312,7 @@
 #include <math.h>
 #include <cups/raster.h>
 #include <cups/cups.h>
+#include <cups/ppd.h>
 
 #if STDC_HEADERS
 # include <string.h>
