diff --git a/configure.ac b/configure.ac
index cdbb138..be8fc4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -324,11 +324,16 @@ AC_CONFIG_FILES([Makefile
                  include/Makefile
                  src/Makefile
                  src/GL/Makefile
+                 src/OpenGL/opengl.pc
                  src/OpenGL/Makefile
+                 src/GLESv1/glesv1_cm.pc
                  src/GLESv1/Makefile
+                 src/GLESv2/glesv2.pc
                  src/GLESv2/Makefile
+                 src/GLX/glx.pc
                  src/GLX/Makefile
                  src/EGL/Makefile
+                 src/EGL/egl.pc
                  src/GLdispatch/Makefile
                  src/GLdispatch/vnd-glapi/Makefile
                  src/util/Makefile
diff --git a/src/EGL/Makefile.am b/src/EGL/Makefile.am
index 5c036dc..c55c83e 100644
--- a/src/EGL/Makefile.am
+++ b/src/EGL/Makefile.am
@@ -106,3 +106,6 @@ g_egldispatchstubs.c : $(GENERATE_DEPS)
 g_egldispatchstubs.h : $(GENERATE_DEPS)
 	$(AM_V_GEN)$(PYTHON) $(GENERATE_DISPATCH_SCRIPT) header $(GENERATE_LIST_FILES) > $@
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = egl.pc
diff --git a/src/EGL/egl.pc.in b/src/EGL/egl.pc.in
new file mode 100644
index 0000000..6ed6933
--- /dev/null
+++ b/src/EGL/egl.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: egl
+Description: Mesa EGL Library
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lEGL
+Cflags: -I${includedir}
diff --git a/src/GLESv1/Makefile.am b/src/GLESv1/Makefile.am
index d20017d..88e07bc 100644
--- a/src/GLESv1/Makefile.am
+++ b/src/GLESv1/Makefile.am
@@ -38,3 +38,5 @@ libGLESv1_CM_la_LIBADD = \
 	../GLdispatch/vnd-glapi/libglapi_glesv1.la \
 	../util/libutils_misc.la
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = glesv1_cm.pc
diff --git a/src/GLESv1/glesv1_cm.pc.in b/src/GLESv1/glesv1_cm.pc.in
new file mode 100644
index 0000000..c0473e8
--- /dev/null
+++ b/src/GLESv1/glesv1_cm.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glesv1_cm
+Description: OpenGL ES-CM v1 library and headers
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lGLESv1_CM
+Cflags: -I${includedir}
diff --git a/src/GLESv2/Makefile.am b/src/GLESv2/Makefile.am
index a9c2351..f40e06f 100644
--- a/src/GLESv2/Makefile.am
+++ b/src/GLESv2/Makefile.am
@@ -38,3 +38,5 @@ libGLESv2_la_LIBADD = \
 	../GLdispatch/vnd-glapi/libglapi_glesv2.la \
 	../util/libutils_misc.la
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = glesv2.pc
diff --git a/src/GLESv2/glesv2.pc.in b/src/GLESv2/glesv2.pc.in
new file mode 100644
index 0000000..d72b5ba
--- /dev/null
+++ b/src/GLESv2/glesv2.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gles2
+Description: OpenGL ES v2/v3 library and headers
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lGLESv2
+Cflags: -I${includedir}
diff --git a/src/GLX/Makefile.am b/src/GLX/Makefile.am
index 030836b..0d19674 100644
--- a/src/GLX/Makefile.am
+++ b/src/GLX/Makefile.am
@@ -69,3 +69,5 @@ libGLX_la_SOURCES = \
 	libglxmapping.c \
 	libglxproto.c
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = glx.pc
diff --git a/src/GLX/glx.pc.in b/src/GLX/glx.pc.in
new file mode 100644
index 0000000..733d4cb
--- /dev/null
+++ b/src/GLX/glx.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glx
+Description: GLX library and headers
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lGLX
+Cflags: -I${includedir}
diff --git a/src/OpenGL/Makefile.am b/src/OpenGL/Makefile.am
index 647d9e4..4b34dd5 100644
--- a/src/OpenGL/Makefile.am
+++ b/src/OpenGL/Makefile.am
@@ -47,3 +47,5 @@ libOpenGL_la_LIBADD = \
 	../GLdispatch/vnd-glapi/libglapi_opengl.la \
 	../util/libutils_misc.la
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = opengl.pc
diff --git a/src/OpenGL/opengl.pc.in b/src/OpenGL/opengl.pc.in
new file mode 100644
index 0000000..3eb2903
--- /dev/null
+++ b/src/OpenGL/opengl.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+includedir=@includedir@
+
+Name: opengl
+Description: OpenGL (without GLX) headers and interface
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lOpenGL
+Cflags: -I${includedir}
