diff --git a/avidemux_plugins/CMakeLists.txt b/avidemux_plugins/CMakeLists.txt
index afccc07bf..9fbf9ace8 100644
--- a/avidemux_plugins/CMakeLists.txt
+++ b/avidemux_plugins/CMakeLists.txt
@@ -151,7 +151,7 @@ SET(DO_CLI    0)
 SET(DO_SETTINGS    0)
 
 IF(NOT PLUGIN_UI)
-  MESSAGE(FATAL_ERROR "Please add -DPLUGIN_UI=[ALL|QT4|GTK|CLI|COMMON]")
+  MESSAGE(FATAL_ERROR "Please add -DPLUGIN_UI=[ALL|QT4|GTK|CLI|COMMON|SETTINGS]")
 ENDIF()
 
 IF(PLUGIN_UI MATCHES "ALL")
