diff --git a/src/shapes/ply/ply_parser.hpp b/src/shapes/ply/ply_parser.hpp
index d7f5dc9c..8937433e 100644
--- a/src/shapes/ply/ply_parser.hpp
+++ b/src/shapes/ply/ply_parser.hpp
@@ -27,9 +27,9 @@
 #  endif
 #endif
 
-#if !defined(ADT_WORKAROUND) && (defined(__INTEL_COMPILER) || defined(__INTELLISENSE__))
+//#if !defined(ADT_WORKAROUND) && (defined(__INTEL_COMPILER) || defined(__INTELLISENSE__))
 #define ADT_WORKAROUND 1
-#endif
+//#endif
 
 #if MTS_USE_BOOST_TR1
 #include <boost/tr1/functional.hpp>
