diff -Naur amide-1.0.5.orig/src/alignment_mutual_information.c amide-1.0.5.new/src/alignment_mutual_information.c
--- amide-1.0.5.orig/src/alignment_mutual_information.c	2012-01-22 17:25:06.000000000 +0100
+++ amide-1.0.5.new/src/alignment_mutual_information.c	2016-10-29 11:01:50.566415209 +0200
@@ -209,7 +209,7 @@
       //g_print("\t\%i", mutual_information_array[i][j] );  // for point-wise counts
       // g_print("\t\%4.3f", incremental_mi );               // for point-wise probability
       
-      if isinf(incremental_mi) {
+      if (isinf(incremental_mi)) {
         //count it (because lots and lots of zeroes mean bad registration) and go to the next loop
         mi_nan_count++;
         // this is most often a problem when the incemental MI is infinity, due to divide by zero
