Index: alire/src/alire-2.1.0/alr_env.gpr
===================================================================
--- src.orig/alire-2.1.0/alr_env.gpr
+++ src/alire-2.1.0/alr_env.gpr
@@ -9,30 +9,30 @@ aggregate project Alr_Env is
    for Create_Missing_Dirs use "True";
 
    for Project_Path use (
-                         "deps/aaa",
-                         "deps/ada-toml",
-                         "deps/ajunitgen",
-                         "deps/ansi",
-                         "deps/clic",
-                         "deps/compare_to_case",
-                         "deps/cstrings",
-                         "deps/den",
-                         "deps/dirty_booleans",
-                         "deps/diskflags",
-                         "deps/gnatcoll-slim",
-                         "deps/lml",
-                         "deps/minirest",
-                         "deps/optional",
-                         "deps/semantic_versioning",
-                         "deps/si_units",
-                         "deps/simple_logging",
-                         "deps/spdx",
-                         "deps/stopwatch",
-                         "deps/toml_slicer",
-                         "deps/umwi",
-                         "deps/uri-ada",
-                         "deps/xmlezout",
-                         "deps/yeison/yeison_12"
+                         "../aaa",
+                         "../ada-toml",
+                         "../ajunitgen",
+                         "../ansi-ada",
+                         "../clic",
+                         "../compare_to_case",
+                         "../cstrings",
+                         "../den",
+                         "../dirty_booleans",
+                         "../diskflags",
+                         "../gnatcoll-core",
+                         "../lml_ada",
+                         "../minirest",
+                         "../optional",
+                         "../semantic_versioning",
+                         "../si_units",
+                         "../simple_logging",
+                         "../spdx_ada",
+                         "../stopwatch",
+                         "../toml_slicer",
+                         "../umwi",
+                         "../uri-ada",
+                         "../xmlezout",
+                         "../yeison/yeison_12"
                         );
 
    for Project_Files use ("alr.gpr");
