Index: alire-1.1.0/alr_env.gpr
===================================================================
--- alire-1.1.0.orig/alr_env.gpr
+++ alire-1.1.0/alr_env.gpr
@@ -3,22 +3,22 @@ aggregate project Alr_Env is
    for Create_Missing_Dirs use "True";
 
    for Project_Path use (
-                         "deps/aaa",
-                         "deps/ada-toml",
-                         "deps/ajunitgen",
-                         "deps/ansi",
-                         "deps/gnatcoll-slim",
-                         "deps/minirest",
-                         "deps/optional",
-                         "deps/semantic_versioning",
-                         "deps/simple_logging",
-                         "deps/si_units",
-                         "deps/stopwatch",
-                         "deps/toml_slicer",
-                         "deps/uri-ada",
-                         "deps/xmlezout",
-                         "deps/spdx",
-                         "deps/clic");
+                         "../aaa",
+                         "../ada-toml",
+                         "../ajunitgen",
+                         "../ansi-ada",
+                         "../gnatcoll-slim",
+                         "../minirest",
+                         "../optional",
+                         "../semantic_versioning",
+                         "../simple_logging",
+                         "../si_units",
+                         "../stopwatch",
+                         "../toml_slicer",
+                         "../uri-ada",
+                         "../xmlezout",
+                         "../spdx_ada",
+                         "../clic");
 
    for Project_Files use ("alr.gpr");
 
