diff --git a/pyproject.toml b/pyproject.toml
index ebf122d..74a2d75 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 120
-target-version = ['py311']
+target-version = ['py310']
 exclude = '''
 /(
     \.eggs
@@ -21,7 +21,7 @@ authors = [{ name = "Chris Griffith", email = "chris@cdgriffith.com" }]
 readme = "README.md"
 #url = "https://fastflix.org"
 #download_url = "https://github.com/cdgriffith/FastFlix/releases"
-requires-python = ">=3.11,<3.12"
+requires-python = ">=3.10,<3.12"
 dynamic = ["version"]
 dependencies = [
     "appdirs~=1.4",
