diff -ru a/browser/components/preferences/config/appearance.mjs b/browser/components/preferences/config/appearance.mjs
--- a/browser/components/preferences/config/appearance.mjs
+++ b/browser/components/preferences/config/appearance.mjs
@@ -267,13 +267,13 @@
 
 Preferences.addSetting({
   id: "librewolfRFPWarning",
-  deps: ["librewolfRFP"],
-  visible: ({librewolfRFP}) => {
+  deps: ["librewolfRFP","librewolfRFPOverrideColor"],
+  visible: ({librewolfRFP,librewolfRFPOverrideColor}) => {
     const interval = setInterval(() => {
       const chooser = document.getElementById("web-appearance-chooser");
       if (chooser) {
         clearInterval(interval);
-        if (librewolfRFP.value) {
+        if (librewolfRFP.value && !librewolfRFPOverrideColor.value) {
           chooser.style.opacity = "0.3";
           chooser.style.pointerEvents = "none";
         } else {
@@ -282,7 +282,7 @@
         }
       }
     }, 100);
-    return librewolfRFP.value;
+    return librewolfRFP.value && !librewolfRFPOverrideColor.value;
   },
 });
 
diff -ru a/browser/components/preferences/librewolf.js b/browser/components/preferences/librewolf.js
--- a/browser/components/preferences/librewolf.js
+++ b/browser/components/preferences/librewolf.js
@@ -132,6 +132,10 @@
   pref: "privacy.resistFingerprinting",
 });
+Preferences.addSetting({
+  id: "librewolfRFPOverrideColor",
+  pref: "privacy.override_rfp_for_color_scheme",
+});
 Preferences.addSetting({
   id: "librewolfLetterboxing",
   pref: "privacy.resistFingerprinting.letterboxing",
 });
diff -ru a/dom/base/Document.cpp b/dom/base/Document.cpp
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -21103,7 +21103,8 @@
 
 ColorScheme Document::PreferredColorScheme(IgnoreRFP aIgnoreRFP) const {
   if (ShouldResistFingerprinting(RFPTarget::CSSPrefersColorScheme) &&
-      aIgnoreRFP == IgnoreRFP::No) {
+      aIgnoreRFP == IgnoreRFP::No &&
+      !StaticPrefs::privacy_override_rfp_for_color_scheme()) {
     return ColorScheme::Light;
   }
 
diff -ru a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
--- a/modules/libpref/init/StaticPrefList.yaml
+++ b/modules/libpref/init/StaticPrefList.yaml
@@ -17691,6 +17691,12 @@
   value: false
   mirror: always
 
+# Allows overriding RFP for only the dark/light pref
+- name: privacy.override_rfp_for_color_scheme
+  type: RelaxedAtomicBool
+  value: false
+  mirror: always
+
 # Enforce tracking protection in all modes.
 - name: privacy.trackingprotection.enabled
   type: bool
