--- src/amarok/src/core-impl/playlists/types/file/xspf/XSPFPlaylist.cpp	2013-08-14 22:22:52.000000000 +0200
+++ src/amarok/src/core-impl/playlists/types/file/xspf/XSPFPlaylist.cpp	2016-05-25 14:43:13.681557351 +0200
@@ -28,6 +28,7 @@
 #include "playlist/PlaylistModelStack.h"
 
 using namespace Playlists;
+using namespace Playlist;
 
 XSPFPlaylist::XSPFPlaylist( const KUrl &url, Playlists::PlaylistProvider *provider, OnLoadAction onLoad )
     : PlaylistFile( url, provider )
@@ -101,7 +102,7 @@
     //FIXME: this needs to be moved to whatever is creating the XSPFPlaylist
     if( m_autoAppendAfterLoad )
         The::playlistController()->insertPlaylist(
-                    ::Playlist::ModelStack::instance()->bottom()->rowCount(),
+                    ModelStack::instance()->bottom()->rowCount(),
                     Playlists::PlaylistPtr( this )
                 );
 }

