diff --git a/meson.build b/meson.build
index 1bcb424..4d636b6 100644
--- a/meson.build
+++ b/meson.build
@@ -363,6 +363,7 @@ if host_machine.system() != 'gba' and host_machine.system() != 'n3ds'
             dependencies : dependencies,
             link_args : link_args,
             win_subsystem : 'windows',
+            install_dir : 'lib/apotris',
             install : true
         )
         if host_machine.system() == 'darwin'
@@ -372,9 +373,8 @@ if host_machine.system() != 'gba' and host_machine.system() != 'n3ds'
             install_subdir(audio_dir, install_dir : 'Contents/Resources')
             install_subdir('license', install_dir : 'Contents/Resources')
         else
-            install_subdir(audio_dir, install_dir : '.')
-            install_subdir(shaders_dir, install_dir : '.')
-            install_subdir('license', install_dir : '.')
+            install_subdir(audio_dir, install_dir : 'share/apotris')
+            install_subdir(shaders_dir, install_dir : 'share/apotris/assets')
         endif
     endif
 elif host_machine.system() == 'n3ds'
diff --git a/source/liba_window.cpp b/source/liba_window.cpp
index 6bbcbd3..1f8ded5 100644
--- a/source/liba_window.cpp
+++ b/source/liba_window.cpp
@@ -103,7 +103,7 @@ void windowInit() {
 
     refreshWindowSize();
 
-    loadAudio("");
+    loadAudio("/usr/share/apotris/");
 
     nanotime_step_init(&stepper, (uint64_t)(NANOTIME_NSEC_PER_SEC / FPS_TARGET),
                        nanotime_now_max(), nanotime_now, nanotime_sleep);
diff --git a/source/shader.cpp b/source/shader.cpp
index 5bb2b7e..523580d 100644
--- a/source/shader.cpp
+++ b/source/shader.cpp
@@ -328,7 +328,7 @@ static bool hasSuffix(const std::string& s, const std::string& suffix) {
 }
 
 std::vector<std::string> findShaders() {
-    std::string shaderPath = "assets/shaders";
+    std::string shaderPath = "/usr/share/apotris/assets/shaders";
 
     DIR* dir = opendir(shaderPath.c_str());
     if (!dir) {
