diff -Nurwa cardano-wallet-2021-01-12/cabal.project cardano-wallet-2021-01-12-archghc/cabal.project
--- cardano-wallet-2021-01-12/cabal.project	2021-01-21 08:33:38.666197087 +0100
+++ cardano-wallet-2021-01-12-archghc/cabal.project	2021-01-12 13:28:17.000000000 +0100
@@ -2,7 +2,7 @@
 
 index-state: 2021-01-05T01:46:07Z
 
-with-compiler: ghc-8.6.5
+with-compiler: /opt/ghc-8.6/bin/ghc
 
 packages:
     lib/core/
