diff --git a/src/comp/Makefile b/src/comp/Makefile
index 5e38ec2..b832d80 100644
--- a/src/comp/Makefile
+++ b/src/comp/Makefile
@@ -63,6 +63,8 @@ TCL_LIBS  = -ltcl$(TCL_VER) -ltk$(TCL_VER) -litcl$(ITCL_VER) \
 ifeq ($(OSTYPE), Darwin)
 TCL_ARGS += -I$(shell xcrun --show-sdk-path)/System/Library/Frameworks/Tk.framework/Headers
 TCL_LIBS += -L"$(shell dirname $(shell $(FIND) /System/Library/Tcl -type f -name libitcl$(ITCL_VER).dylib))"
+else
+TCL_LIBS += -L/usr/lib/itcl$(ITCL_VER) -optl-Wl,-rpath,/usr/lib/itcl$(ITCL_VER)
 endif
 
 # STP
@@ -94,6 +96,8 @@ WISHFLAGS = -litk$(ITK_VER) $(shell pkg-config --libs x11)
 WISHFLAGS += $(EXTRAWISHLIBS)
 ifeq ($(OSTYPE), Darwin)
 WISHFLAGS += -L"$(shell dirname $(shell $(FIND) /System/Library/Tcl -type f -name libitk$(ITK_VER).dylib))"
+else
+WISHFLAGS += -L/usr/lib/itk$(ITK_VER) -optl-Wl,-rpath,/usr/lib/itk$(ITK_VER)
 endif
 
 # -----
diff --git a/src/vendor/htcl/Makefile b/src/vendor/htcl/Makefile
index a5deb63..ec51a6b 100644
--- a/src/vendor/htcl/Makefile
+++ b/src/vendor/htcl/Makefile
@@ -1,9 +1,9 @@
-CFLAGS += -Wall $(shell pkg-config --silence-errors --cflags-only-I tcl || echo -I/usr/include/tcl)
+GHCFLAGS += -Wall $(shell pkg-config --silence-errors --cflags-only-I tcl || echo -I/usr/include/tcl)
 GHC ?= ghc
 
 # We use GHC to compile this, so it has the proper RTS includes
 %.o: %.c
-	$(GHC) $(CFLAGS) -c $<
+	$(GHC) $(GHCFLAGS) -c $<
 
 libhtcl.a: haskell.o
 	ar -r $@ $(filter %.o, $+)
diff --git a/src/vendor/stp/src/sat/Makefile b/src/vendor/stp/src/sat/Makefile
index 0d0937d..912f7b1 100644
--- a/src/vendor/stp/src/sat/Makefile
+++ b/src/vendor/stp/src/sat/Makefile
@@ -13,6 +13,7 @@ export COPTIMIZE=$(CFLAGS_M32) $(CFLAGS_FPIC) -O3
 core: $(LIB)
 
 # $(LIB) depends on */lib$(SUB)_release.a and will be rebuilt only if they have been updated
+.NOTPARALLEL:
 $(LIB): core/libcore_release.a core_prop/libcore_prop_release.a simp/libsimp_release.a utils/libutils_release.a cryptominisat2/libminisat.a $(OBJS)
 	$(RM) $@
 	$(call arcat,$@,$(filter %.a,$^))
