diff --git a/CMakeLists.txt b/CMakeLists.txt
index c99a6c7af..352d72f24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14 -lSDL2main -lSDL2")
 
 set(UNFUCK_GCC_FLAGS "")
 set(UNFUCK_GXX_FLAGS "")
diff --git a/source/in_sdl.cpp b/source/in_sdl.cpp
index c0024a8ed..363db9197 100644
--- a/source/in_sdl.cpp
+++ b/source/in_sdl.cpp
@@ -24,7 +24,7 @@
 //**  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //**
 //**************************************************************************
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #include "gamedefs.h"
 #include "drawer.h"
 #include "ui/ui.h"
diff --git a/source/neoui/neoui_main.cpp b/source/neoui/neoui_main.cpp
index d7aa78e37..bcd1e77cd 100644
--- a/source/neoui/neoui_main.cpp
+++ b/source/neoui/neoui_main.cpp
@@ -26,7 +26,7 @@
 #include "../gamedefs.h"
 
 #ifdef CLIENT
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #ifdef USE_GLAD
 # include "glad.h"
 #else
diff --git a/source/render/hwgl/gl_sdl.cpp b/source/render/hwgl/gl_sdl.cpp
index 6d1993889..bdf9f9585 100644
--- a/source/render/hwgl/gl_sdl.cpp
+++ b/source/render/hwgl/gl_sdl.cpp
@@ -24,7 +24,7 @@
 //**  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //**
 //**************************************************************************
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #include "gl_local.h"
 
 
diff --git a/source/sys_sdl.cpp b/source/sys_sdl.cpp
index b737bab95..ed9f1f6d3 100644
--- a/source/sys_sdl.cpp
+++ b/source/sys_sdl.cpp
@@ -31,7 +31,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <dirent.h>
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #if !defined(_WIN32) && !defined(__SWITCH__)
 # include <execinfo.h>
 #elif defined(__SWITCH__)
diff --git a/vccrun/modules/sdlgl/mod_sdlgl.cpp b/vccrun/modules/sdlgl/mod_sdlgl.cpp
index b648891c5..f5dc0db52 100644
--- a/vccrun/modules/sdlgl/mod_sdlgl.cpp
+++ b/vccrun/modules/sdlgl/mod_sdlgl.cpp
@@ -34,7 +34,7 @@
 
 #include <sys/time.h>
 
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #include <GL/gl.h>
 #include <GL/glext.h>
 static SDL_Window *hw_window = nullptr;
diff --git a/vccrun/modules/sdlgl/mod_sdlgl.h b/vccrun/modules/sdlgl/mod_sdlgl.h
index 24b769202..d779a01c9 100644
--- a/vccrun/modules/sdlgl/mod_sdlgl.h
+++ b/vccrun/modules/sdlgl/mod_sdlgl.h
@@ -30,7 +30,7 @@
 #include "../../vcc_run.h"
 
 #if defined (VCCRUN_HAS_SDL) && defined(VCCRUN_HAS_OPENGL)
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #ifdef USE_GLAD
 #include "glad.h"
 #else
