--- a/Makefile	2023-04-01 23:03:38.896807602 +0530
+++ b/Makefile	2023-04-02 00:26:38.886519015 +0530
@@ -3,10 +3,10 @@
 #
 
 # Installation destinations
-PREFIX := /usr/local
+PREFIX := /usr
 
 # Default compilation flags
-CXXFLAGS := ${CXXFLAGS} -std=c++11 -O3
+CXXFLAGS := ${CXXFLAGS} -std=c++11 -O2
 
 ## Optional features; comment out or set to value other than 'yes' to disable
 
@@ -20,7 +20,7 @@
 DEBUG_BUILD := no
 
 # Include coverage instrumentation in build
-COVERAGE := no
+COVERAGE := yes
 
 ###############################################################################
 # Makefile internals. Normally you do not need to touch these.
@@ -111,20 +111,20 @@
 install: build/$(PROG)
 	@echo $(COLOR_GREEN)"Installing AdapterRemoval .."$(COLOR_END)
 	@echo $(COLOR_GREEN)"  .. binary into ${PREFIX}/bin/"$(COLOR_END)
-	$(QUIET) $(MKDIR) ${PREFIX}/bin/
-	$(QUIET) $(INSTALLEXE) build/$(PROG) ${PREFIX}/bin/
+	$(QUIET) $(MKDIR) $(DESTDIR)${PREFIX}/bin/
+	$(QUIET) $(INSTALLEXE) build/$(PROG) $(DESTDIR)${PREFIX}/bin/
 
 	@echo $(COLOR_GREEN)"  .. man-page into ${PREFIX}/share/man/man1/"$(COLOR_END)
-	$(QUIET) $(MKDIR) ${PREFIX}/share/man/man1/
-	$(QUIET) $(INSTALLDOC) $(PROG).1 ${PREFIX}/share/man/man1/
+	$(QUIET) $(MKDIR) $(DESTDIR)${PREFIX}/share/man/man1/
+	$(QUIET) $(INSTALLDOC) $(PROG).1 $(DESTDIR)${PREFIX}/share/man/man1/
 
 	@echo $(COLOR_GREEN)"  .. README into ${PREFIX}/share/adapterremoval/"$(COLOR_END)
-	$(QUIET) $(MKDIR) ${PREFIX}/share/adapterremoval/
-	$(QUIET) $(INSTALLDOC) README.md ${PREFIX}/share/adapterremoval/
+	$(QUIET) $(MKDIR) $(DESTDIR)${PREFIX}/share/doc/adapterremoval/
+	$(QUIET) $(INSTALLDOC) README.md $(DESTDIR)${PREFIX}/share/doc/adapterremoval/
 
 	@echo $(COLOR_GREEN)"  .. examples into ${PREFIX}/share/adapterremoval/examples/"$(COLOR_END)
-	$(QUIET) $(MKDIR) ${PREFIX}/share/adapterremoval/examples/
-	$(QUIET) $(INSTALLDAT) examples/*.* ${PREFIX}/share/adapterremoval/examples/
+	$(QUIET) $(MKDIR) $(DESTDIR)${PREFIX}/share/adapterremoval/examples/
+	$(QUIET) $(INSTALLDAT) examples/*.* $(DESTDIR)${PREFIX}/share/adapterremoval/examples/
 
 static: build/$(LIBNAME).a
 
