diff --git a/Telegram/SourceFiles/qt_static_plugins.cpp b/Telegram/SourceFiles/qt_static_plugins.cpp
index b975c7a8..5ad461a4 100644
--- a/Telegram/SourceFiles/qt_static_plugins.cpp
+++ b/Telegram/SourceFiles/qt_static_plugins.cpp
@@ -30,6 +30,7 @@ Q_IMPORT_PLUGIN(QGenericEnginePlugin)
 #elif defined Q_OS_LINUX // Q_OS_WIN | Q_OS_MAC
 Q_IMPORT_PLUGIN(QWebpPlugin)
 Q_IMPORT_PLUGIN(QXcbIntegrationPlugin)
+Q_IMPORT_PLUGIN(QWaylandIntegrationPlugin)
 Q_IMPORT_PLUGIN(QConnmanEnginePlugin)
 Q_IMPORT_PLUGIN(QGenericEnginePlugin)
 Q_IMPORT_PLUGIN(QNetworkManagerEnginePlugin)
diff --git a/Telegram/ThirdParty/libtgvoip b/Telegram/ThirdParty/libtgvoip
--- a/Telegram/ThirdParty/libtgvoip
+++ b/Telegram/ThirdParty/libtgvoip
@@ -1 +1 @@
-Subproject commit 9f78cb85e5db9b39a6c4c3f7847ea3e98c9f544a
+Subproject commit 9f78cb85e5db9b39a6c4c3f7847ea3e98c9f544a-dirty
diff --git a/Telegram/gyp/Telegram.gyp b/Telegram/gyp/Telegram.gyp
index 6fa545e5..dc8383fa 100644
--- a/Telegram/gyp/Telegram.gyp
+++ b/Telegram/gyp/Telegram.gyp
@@ -86,27 +86,23 @@
     ],
 
     'defines': [
-      'AL_LIBTYPE_STATIC',
       'AL_ALEXT_PROTOTYPES',
       'TGVOIP_USE_CXX11_LIB',
       '<!@(python -c "for s in \'<(build_defines)\'.split(\',\'): print(s)")',
+      '__STDC_FORMAT_MACROS',
+      'TDESKTOP_DISABLE_AUTOUPDATE',
+      'TDESKTOP_DISABLE_REGISTER_CUSTOM_SCHEME',
     ],
 
     'include_dirs': [
       '<(src_loc)',
       '<(SHARED_INTERMEDIATE_DIR)',
-      '<(libs_loc)/breakpad/src',
-      '<(libs_loc)/lzma/C',
-      '<(libs_loc)/libexif-0.6.20',
-      '<(libs_loc)/zlib',
-      '<(libs_loc)/ffmpeg',
-      '<(libs_loc)/openal-soft/include',
-      '<(libs_loc)/opus/include',
       '<(minizip_loc)',
       '<(sp_media_key_tap_loc)',
       '<(emoji_suggestions_loc)',
       '<(submodules_loc)/GSL/include',
       '<(submodules_loc)/variant/include',
+      '/usr/include/breakpad',
     ],
     'sources': [
       '<@(qrc_files)',
diff --git a/Telegram/gyp/qt.gypi b/Telegram/gyp/qt.gypi
index 67f7976e..8c960313 100644
--- a/Telegram/gyp/qt.gypi
+++ b/Telegram/gyp/qt.gypi
@@ -38,7 +38,6 @@
             'Qt5Network',
             'Qt5Widgets',
             'Qt5Gui',
-            'qtharfbuzzng',
           ],
           'qt_version%': '<(qt_version)',
           'conditions': [
@@ -97,20 +96,9 @@
               '<@(qt_libs)',
               'Qt5DBus',
               'Qt5Core',
-              'qtpcre',
-              'Xi',
-              'Xext',
-              'Xfixes',
-              'SM',
-              'ICE',
-              'fontconfig',
-              'expat',
-              'freetype',
-              'z',
-              'xcb-shm',
-              'xcb-xfixes',
-              'xcb-render',
-              'xcb-static',
+              'Qt5WaylandClient',
+              'qwayland-generic',
+              'qwayland-egl',
             ],
           }],
         ],
@@ -201,6 +189,7 @@
     '<(qt_loc)/include/QtGui/<(qt_version)',
     '<(qt_loc)/include/QtCore/<(qt_version)/QtCore',
     '<(qt_loc)/include/QtGui/<(qt_version)/QtGui',
+    '/usr/include/openssl-1.0',
   ],
   'library_dirs': [
     '<(qt_loc)/lib',
@@ -221,7 +210,6 @@
         '<(qt_loc)/plugins/platforminputcontexts',
       ],
       'libraries': [
-        '<(linux_path_xkbcommon)/lib/libxkbcommon.a',
         '<@(qt_libs_release)',
         '<(linux_lib_ssl)',
         '<(linux_lib_crypto)',
@@ -234,12 +222,30 @@
         '-lgthread-2.0',
         '-lglib-2.0',
         '-lpthread',
-      ],
+        '-lXi',
+        '-lXext',
+        '-lXfixes',
+        '-lSM',
+        '-lICE',
+        '-lfontconfig',
+        '-lexpat',
+        '-lfreetype',
+        '<!(pkg-config 2> /dev/null --libs xkbcommon xkbcommon-x11)',
+        '<!(pkg-config 2> /dev/null --libs harfbuzz)',
+        '<!(pkg-config 2> /dev/null --libs libpcre16)',
+        '<!(pkg-config 2> /dev/null --libs zlib)',
+        '<!(pkg-config 2> /dev/null --libs libpng16)',
+        '<!(pkg-config 2> /dev/null --libs libjpeg)',
+        '<!(pkg-config 2> /dev/null --libs libwebp)',
+        '<!(pkg-config 2> /dev/null --libs xcb-shm xcb-xfixes xcb-render xcb-renderutil xcb-sync xcb-randr xcb-xinerama xcb-xkb xcb-icccm xcb-image xcb-shape xcb-keysyms xcb-util)',
+        '<!(pkg-config 2> /dev/null --libs wayland-client wayland-cursor)',
+        '-lproxy',
+        '-lGL',
+        ],
       'include_dirs': [
         '<(qt_loc)/mkspecs/linux-g++',
       ],
       'ldflags': [
-        '-static-libstdc++',
         '-pthread',
         '-g',
         '-rdynamic',
diff --git a/Telegram/gyp/settings_linux.gypi b/Telegram/gyp/settings_linux.gypi
index fde88dcd..1fd1b9cb 100644
--- a/Telegram/gyp/settings_linux.gypi
+++ b/Telegram/gyp/settings_linux.gypi
@@ -25,7 +25,6 @@
           '-pipe',
           '-g',
           '-Wall',
-          '-Werror',
           '-W',
           '-fPIC',
           '-Wno-unused-variable',
diff --git a/Telegram/gyp/telegram_linux.gypi b/Telegram/gyp/telegram_linux.gypi
index 721d4f6c..b30c2da7 100644
--- a/Telegram/gyp/telegram_linux.gypi
+++ b/Telegram/gyp/telegram_linux.gypi
@@ -22,11 +22,20 @@
     'variables': {
       'not_need_gtk%': '<!(python -c "print(\'TDESKTOP_DISABLE_GTK_INTEGRATION\' in \'<(build_defines)\')")',
       'pkgconfig_libs': [
-# In order to work libxkbcommon must be linked statically,
-# PKGCONFIG links it like "-L/usr/local/lib -lxkbcommon"
-# which makes a dynamic link which leads to segfault in
-# QApplication() -> createPlatformIntegration -> QXcbIntegrationPlugin::create
-        #'xkbcommon',
+        'liblzma',
+        'openal',
+        'libavformat',
+        'libavcodec',
+        'libswresample',
+        'libswscale',
+        'libavutil',
+        'opus',
+        'libva-x11',
+        'libva-drm',
+        'libva',
+        'vdpau',
+        'libdrm',
+        'zlib',
       ],
       'linux_path_ffmpeg%': '/usr/local',
       'linux_path_openal%': '/usr/local',
@@ -37,20 +46,11 @@
       'linux_path_opus_include%': '<(libs_loc)/opus/include',
     },
     'include_dirs': [
-      '/usr/local/include',
-      '<(linux_path_ffmpeg)/include',
-      '<(linux_path_openal)/include',
-      '<(linux_path_breakpad)/include/breakpad',
       '<(linux_path_opus_include)',
+      '/usr/include',
     ],
     'library_dirs': [
-      '/usr/local/lib',
-      '<(linux_path_ffmpeg)/lib',
-      '<(linux_path_openal)/lib',
-      '<(linux_path_libexif_lib)',
-      '<(linux_path_va)/lib',
-      '<(linux_path_vdpau)/lib',
-      '<(linux_path_breakpad)/lib',
+      '/usr/lib',
     ],
     'libraries': [
       'breakpad_client',
@@ -58,21 +58,7 @@
       'ibusplatforminputcontextplugin',
       'fcitxplatforminputcontextplugin',
       'himeplatforminputcontextplugin',
-      'liblzma.a',
-      'libopenal.a',
-      'libavformat.a',
-      'libavcodec.a',
-      'libswresample.a',
-      'libswscale.a',
-      'libavutil.a',
-      'libopus.a',
-      'libva-x11.a',
-      'libva-drm.a',
-      'libva.a',
-      'libvdpau.a',
-      'libdrm.a',
-      'libz.a',
-#      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
+      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
     ],
     'conditions': [['not_need_gtk!="True"', {
         'cflags_cc': [
@@ -80,6 +66,7 @@
             '<!(pkg-config 2> /dev/null --cflags gtk+-2.0)',
             '<!(pkg-config 2> /dev/null --cflags glib-2.0)',
             '<!(pkg-config 2> /dev/null --cflags dee-1.0)',
+            '<!(pkg-config 2> /dev/null --cflags <@(pkgconfig_libs))',
          ],
     }]],
     'configurations': {
