diff -rupN pantheon-default-settings.orig/debian/elementary-default-settings.gsettings-override pantheon-default-settings/debian/elementary-default-settings.gsettings-override
--- pantheon-default-settings.orig/debian/elementary-default-settings.gsettings-override	2016-02-14 17:09:27.093570959 +0100
+++ pantheon-default-settings/debian/elementary-default-settings.gsettings-override	2016-02-14 17:11:00.074100778 +0100
@@ -1,13 +1,12 @@
 [net.launchpad.plank.dock.settings]
-dock-items=['gala-multitaskingview.dockitem','midori.dockitem','pantheon-mail.dockitem','maya-calendar.dockitem','noise.dockitem','audience.dockitem','pantheon-photos.dockitem','switchboard.dockitem','appcenter.dockitem']
+dock-items=['gala-multitaskingview.dockitem','midori.dockitem','pantheon-mail.dockitem','maya-calendar.dockitem','noise.dockitem','audience.dockitem','pantheon-photos.dockitem','switchboard.dockitem']
 hide-mode='window-dodge'
 show-dock-item=false
 theme='Gtk+'
 
 [org.gnome.desktop.background]
-draw-background=true
 picture-options='zoom'
-picture-uri='file:///usr/share/backgrounds/elementaryos-default'
+picture-uri='file:///usr/share/backgrounds/pantheon/elementaryos-default'
 primary-color='#000000'
 show-desktop-icons=false
 
@@ -21,7 +20,6 @@ font-name='Droid Sans 9'
 gtk-theme='elementary'
 icon-theme='elementary'
 monospace-font-name='Droid Sans Mono 10'
-show-unicode-menu=false
 toolbar-style='icons'
 
 [org.gnome.desktop.peripherals.touchpad]
@@ -102,7 +100,6 @@ active=false
 
 [org.gnome.settings-daemon.plugins.media-keys]
 screensaver='<Super>l'
-terminal='<Super>t'
 
 [org.gnome.settings-daemon.plugins.power]
 ambient-enabled=false
diff -rupN pantheon-default-settings.orig/plank/dock1/launchers/appcenter.dockitem pantheon-default-settings/plank/dock1/launchers/appcenter.dockitem
--- pantheon-default-settings.orig/plank/dock1/launchers/appcenter.dockitem	2016-02-14 17:09:27.093570959 +0100
+++ pantheon-default-settings/plank/dock1/launchers/appcenter.dockitem	1970-01-01 01:00:00.000000000 +0100
@@ -1,4 +0,0 @@
-[PlankDockItemPreferences]
-Launcher=file:///usr/share/applications/appcenter.desktop
-
-
diff -rupN pantheon-default-settings.orig/policykit.desktop pantheon-default-settings/policykit.desktop
--- pantheon-default-settings.orig/policykit.desktop	2016-02-14 17:09:27.093570959 +0100
+++ pantheon-default-settings/policykit.desktop	2016-02-14 17:11:00.074100778 +0100
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Authentication Dialog
-Exec=/usr/lib/policykit-1-gnome/polkit-gnome-authentication-agent-1
+Exec=/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
 Icon=dialog-password
 Terminal=false
 Type=Application
diff -rupN pantheon-default-settings.orig/settings.ini pantheon-default-settings/settings.ini
--- pantheon-default-settings.orig/settings.ini	2016-02-14 17:09:27.093570959 +0100
+++ pantheon-default-settings/settings.ini	2016-02-14 17:11:00.074100778 +0100
@@ -2,6 +2,6 @@
 gtk-theme-name = elementary
 gtk-icon-theme-name = elementary
 gtk-fallback-icon-theme = gnome
-gtk-sound-theme-name = ubuntu
+gtk-sound-theme-name = freedesktop
 gtk-icon-sizes = panel-menu-bar=24,24
 gtk-font-name = Droid Sans 9
