diff -uNr zarafa-7.1.7.old/caldav/CalDAV.cpp zarafa-7.1.7/caldav/CalDAV.cpp
--- zarafa-7.1.7.old/caldav/CalDAV.cpp	2013-09-27 16:58:27.000000000 -0600
+++ zarafa-7.1.7/caldav/CalDAV.cpp	2014-03-19 14:46:09.360454187 -0600
@@ -191,7 +191,7 @@
     struct sigaction act = {{0}};
 
 	// Configuration
-	char opt = '\0';
+	int opt = '\0';
 	const char *lpszCfg = ECConfig::GetDefaultPath("ical.cfg");
 	const configsetting_t lpDefaults[] = {
 		{ "run_as_user", "" },
