diff --git a/pynndescent/tests/test_distances.py b/pynndescent/tests/test_distances.py
index 101530b..9ad88b0 100644
--- a/pynndescent/tests/test_distances.py
+++ b/pynndescent/tests/test_distances.py
@@ -109,7 +109,7 @@ def test_binary_check(binary_data, metric):
 def test_sparse_spatial_check(sparse_spatial_data, metric, decimal=6):
     if metric in spdist.sparse_named_distances:
         dist_matrix = pairwise_distances(
-            sparse_spatial_data.todense().astype(np.float32), metric=metric
+            np.asarray(sparse_spatial_data.todense().astype(np.float32)), metric=metric
         )
     if metric in ("braycurtis", "dice", "sokalsneath", "yule"):
         dist_matrix[np.where(~np.isfinite(dist_matrix))] = 0.0
@@ -174,7 +174,7 @@ def test_sparse_spatial_check(sparse_spatial_data, metric, decimal=6):
 )
 def test_sparse_binary_check(sparse_binary_data, metric):
     if metric in spdist.sparse_named_distances:
-        dist_matrix = pairwise_distances(sparse_binary_data.todense(), metric=metric)
+        dist_matrix = pairwise_distances(np.asarray(sparse_binary_data.todense()), metric=metric)
     if metric in ("jaccard", "dice", "sokalsneath"):
         dist_matrix[np.where(~np.isfinite(dist_matrix))] = 0.0
     if metric in ("kulsinski", "russellrao"):
