--- a/usr/share/applications/atom-beta.desktop
+++ b/usr/share/applications/atom-beta.desktop
@@ -2,7 +2,7 @@
 Name=Atom Beta
 Comment=A hackable text editor for the 21st Century.
 GenericName=Text Editor
-Exec=env ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=false /usr/bin/atom-beta %F
+Exec=env PYTHON=python2 /usr/bin/atom-beta %U
 Icon=atom-beta
 Type=Application
 StartupNotify=true


--- a/usr/share/atom-beta/resources/app/apm/bin/apm
+++ b/usr/share/atom-beta/resources/app/apm/bin/apm
@@ -2,6 +2,8 @@

 set -e

+export PYTHON=python2
+
 initialCwd=`pwd -P`

 apmPath=$0
