--- autogen.sh	2011-10-14 15:41:37.000000000 +0200
+++ autogen.sh.new	2015-03-27 16:19:29.696853619 +0100
@@ -4,16 +4,16 @@
 # configure program.  This must be done FIRST; note well that there is
 # no Makefile to run until it is done.
 
-# echo "Running autoheader..."
-# autoheader
-# echo "Running aclocal..."
-# aclocal
-# echo "Running libtoolize..."
-# libtoolize
-# rm -f Makefile Makefile.in dieharder/Makefile dieharder/Makefile.in \
-#    libdieharder/Makefile libdieharder/Makefile.in
-# echo "Running automake..."
-# automake --add-missing --copy --gnu
+echo "Running autoheader..."
+autoheader
+echo "Running aclocal..."
+aclocal
+echo "Running libtoolize..."
+libtoolize
+rm -f Makefile Makefile.in dieharder/Makefile dieharder/Makefile.in \
+   libdieharder/Makefile libdieharder/Makefile.in
+echo "Running automake..."
+automake --add-missing --copy --gnu
 
 echo "Running autoreconf..."
 autoreconf
