--- a/avidemux_plugins/ADM_demuxers/VapourSynth/ADM_vsProxy.h     2016-03-05 02:22:54.000000000 -0600
+++ b/avidemux_plugins/ADM_demuxers/VapourSynth/ADM_vsProxy.h        2016-08-01 21:14:33.275996239 -0500
@@ -17,13 +17,9 @@
 #pragma once 
 
 
-extern "C"
-{
-#undef __cplusplus
 #include "VSScript.h"
 #include "VSHelper.h"
-#define __cplusplus
-}
+
 #include "../../ADM_coreSocket/include/ADM_coreAvsProtocol.h"
 
 /**