diff --git a/AviSynth/meson.build b/AviSynth/meson.build
index b20c0d4..97c0255 100644
--- a/AviSynth/meson.build
+++ b/AviSynth/meson.build
@@ -57,6 +57,7 @@ sources = [
 includes = include_directories('../include')
 
 deps = [
+  dependency('avisynth'),
   dependency('liblsmash'),
   dependency('libavcodec', version : '>=58.18.0'),
   dependency('libavformat', version : '>=58.12.0'),
@@ -78,5 +79,5 @@ shared_module('lsmashsource', sources,
   include_directories : includes,
   install : true,
   install_dir : join_paths(get_option('libdir'), 'avisynth'),
-  gnu_symbol_visibility : 'hidden'
+  gnu_symbol_visibility : 'default'
 )
