diff --git a/autogen.sh b/autogen.sh
index 8695fcb..d01cd95 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -259,7 +259,7 @@ if [ "$myhost" = "find-version" ]; then
     matchexcl="--exclude $package-*beta*"
 
     beta=no
-    if [ -e .git ]; then
+    if false; then
       ingit=yes
       tmp=$(git describe --match "${matchstr1}" $matchexcl --long 2>/dev/null)
       if [ -n "$tmp" ]; then
@@ -285,8 +285,8 @@ if [ "$myhost" = "find-version" ]; then
       rvd=$((0x$(echo ${rev} | dd bs=1 count=4 2>/dev/null)))
     else
       ingit=no
-      beta=yes
-      tmp="-unknown"
+      beta=no
+      tmp=""
       cid="0000000"
       rev="0000000"
       rvd="0"
