diff --git a/autogen.sh b/autogen.sh
index 4b511bf95..bc92cc322 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -229,7 +229,7 @@ if [ "$myhost" = "find-version" ]; then
     esac
 
     beta=no
-    if [ -e .git ]; then
+    if false; then
       ingit=yes
       tmp=$(git describe --match "${matchstr1}" --long 2>/dev/null)
       tmp=$(echo "$tmp" | sed s/^"$package"//)
@@ -245,8 +245,8 @@ if [ "$myhost" = "find-version" ]; then
       rvd=$((0x$(echo ${rev} | dd bs=1 count=4 2>/dev/null)))
     else
       ingit=no
-      beta=yes
-      tmp="-unknown"
+      beta=no
+      tmp=""
       rev="0000000"
       rvd="0"
     fi
