diff --git a/factor/avx-ecm/avx_ecm_main.c b/factor/avx-ecm/avx_ecm_main.c
index 660da8a..a7bcc6f 100644
--- a/factor/avx-ecm/avx_ecm_main.c
+++ b/factor/avx-ecm/avx_ecm_main.c
@@ -643,6 +643,7 @@ void vec_ecm_main(fact_obj_t* fobj, uint32_t numcurves, uint64_t B1,
                     printf("Using 1040-bit mul/sqr core\n");
                 }
             }
+#ifdef USE_AVX512F
             else if (tdata[0].MAXBITS > 16400)
             {
                 // faster starting around 16000 bits on Xeon 6254
@@ -654,6 +655,7 @@ void vec_ecm_main(fact_obj_t* fobj, uint32_t numcurves, uint64_t B1,
                     printf("Using karatsuba mul/sqr core\n");
                 }
             }
+#endif
             else
             {
                 vecmulmod_ptr = &vecmulmod52;
