From b2f08adb19fcb18fea7cda9908fa52e2b9db5e7f Mon Sep 17 00:00:00 2001
From: Mika Kuoppala <mika.kuoppala@linux.intel.com>
Date: Mon, 8 Feb 2016 23:36:40 +0200
Subject: drm/i915/vlv: Disable rps when changing frequencies

Signed-off-by: Mika Kuoppala <mika.kuoppala@intel.com>

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index d4f7f46..6780dda 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4466,10 +4466,20 @@ static void valleyview_set_rps(struct drm_device *dev, u8 val)
 	I915_WRITE(GEN6_PMINTRMSK, gen6_rps_pm_mask(dev_priv, val));
 
 	if (val != dev_priv->rps.cur_freq) {
+		u32 ctrl;
+
 		intel_uncore_forcewake_get(dev_priv, FORCEWAKE_ALL);
+
+		ctrl = 	I915_READ(GEN6_RC_CONTROL);
+		I915_WRITE(GEN6_RC_CONTROL, 0);
+
 		vlv_punit_write(dev_priv, PUNIT_REG_GPU_FREQ_REQ, val);
 		if (!IS_CHERRYVIEW(dev_priv))
 			gen6_set_rps_thresholds(dev_priv, val);
+
+		I915_WRITE(GEN6_RC_CONTROL, ctrl);
+		POSTING_READ(GEN6_RC_CONTROL);
+
 		intel_uncore_forcewake_put(dev_priv, FORCEWAKE_ALL);
 	}
 
-- 
cgit v0.10.2

