--- fbthrift-0.24.0/thrift/lib/cpp2/protocol/BinaryProtocol.h	2015-03-27 07:17:13.000000000 +0900
+++ fbthrift.patched/src/fbthrift-0.24.0/thrift/lib/cpp2/protocol/BinaryProtocol.h	2015-03-29 22:35:47.387552550 +0900
@@ -22,7 +22,7 @@
 #include <folly/io/Cursor.h>
 #include <thrift/lib/cpp/protocol/TProtocol.h>
 #include <thrift/lib/cpp2/protocol/Protocol.h>
-
+#include <gflags/gflags.h>
 DECLARE_int32(thrift_cpp2_protocol_reader_string_limit);
 DECLARE_int32(thrift_cpp2_protocol_reader_container_limit);
 
