diff --git a/src/proofofwork.py b/src/proofofwork.py
index df6ed295..a033e8d1 100644
--- a/src/proofofwork.py
+++ b/src/proofofwork.py
@@ -270,7 +270,9 @@ def init():
                 bso = None
     else:
         try:
-            bso = ctypes.CDLL(os.path.join(paths.codePath(), "bitmsghash", bitmsglib))
+            # Modified by Arch Linux PKGBUILD
+            # https://aur.archlinux.org/packages/pybitmessage-git/
+            bso = ctypes.CDLL(os.path.join("/usr/lib/pybitmessage-git", bitmsglib))
         except OSError:
             import glob
             try:
