diff --git a/src/cmp/cmpmain.lsp b/src/cmp/cmpmain.lsp
index 51da7a6..0951787 100644
--- a/src/cmp/cmpmain.lsp
+++ b/src/cmp/cmpmain.lsp
@@ -158,7 +158,7 @@
     (dolist (lib mkcl-libraries)
       (push (mkcl:bstr+ "\"" mkcl-libdir lib "\" ") out))
 
-    (unless use-mkcl-shared-libs (push "-Wl,--no-whole-archive" out))
+    (unless use-mkcl-shared-libs (push "-Wl,--no-whole-archive " out))
 
     (if use-external-shared-libs
         (push *syslibs-&-ld-flags-tail* out)
@@ -185,7 +185,7 @@
       (dolist (lib *mkcl-static-libs*)
         (push (mkcl:bstr+ "\"" mkcl-libdir lib "\" ") out)))
 
-    (push "-Wl,--no-whole-archive" out)
+    (push "-Wl,--no-whole-archive " out)
 
     (push *static-program-ld-flags-tail* out)