--- gtborg-gtsam-03f518eadb35/gtsam/linear/HessianFactor.cpp.orig	2019-05-07 02:51:51.695383978 +0800
+++ gtborg-gtsam-03f518eadb35/gtsam/linear/HessianFactor.cpp	2019-05-07 02:52:01.481994514 +0800
@@ -273,7 +273,7 @@
     boost::optional<const Scatter&> scatter)
 {
   gttic(HessianFactor_MergeConstructor);
-  boost::optional<Scatter> computedScatter;
+  boost::optional<const Scatter&> computedScatter;
   if(!scatter) {
     computedScatter = Scatter(factors);
     scatter = computedScatter;
--- gtborg-gtsam-03f518eadb35/gtsam/linear/JacobianFactor.cpp.orig	2019-05-07 02:51:29.435511635 +0800
+++ gtborg-gtsam-03f518eadb35/gtsam/linear/JacobianFactor.cpp	2019-05-07 02:51:38.215461284 +0800
@@ -224,7 +224,7 @@
   gttic(JacobianFactor_combine_constructor);
 
   // Compute VariableSlots if one was not provided
-  boost::optional<VariableSlots> computedVariableSlots;
+  boost::optional<const VariableSlots&> computedVariableSlots;
   if (!variableSlots) {
     computedVariableSlots = VariableSlots(graph);
     variableSlots = computedVariableSlots; // Binds reference, does not copy VariableSlots
