diff -Naur domoticz-3.8153-orig/webserver/proxyclient.cpp domoticz-3.8153/webserver/proxyclient.cpp
--- domoticz-3.8153-orig/webserver/proxyclient.cpp	2018-01-11 16:09:55.976665686 -0500
+++ domoticz-3.8153/webserver/proxyclient.cpp	2018-01-11 16:12:24.747525079 -0500
@@ -639,7 +639,7 @@
 		void CProxyManager::StartThread()
 		{
 			try {
-				boost::asio::ssl::context ctx(io_service, boost::asio::ssl::context::sslv23);
+				boost::asio::ssl::context ctx(boost::asio::ssl::context::sslv23);
 				ctx.set_verify_mode(boost::asio::ssl::verify_none);
 
 				proxyclient.reset(new CProxyClient(io_service, ctx, m_pWebEm));
diff -Naur domoticz-3.8153-orig/webserver/server.cpp domoticz-3.8153/webserver/server.cpp
--- domoticz-3.8153-orig/webserver/server.cpp	2018-01-11 16:09:55.976665686 -0500
+++ domoticz-3.8153/webserver/server.cpp	2018-01-11 16:12:24.747525079 -0500
@@ -148,7 +148,7 @@
 ssl_server::ssl_server(const ssl_server_settings & ssl_settings, request_handler & user_request_handler) :
 		server_base(ssl_settings, user_request_handler),
 		settings_(ssl_settings),
-		context_(io_service_, ssl_settings.get_ssl_method())
+		context_(ssl_settings.get_ssl_method())
 {
 #ifdef DEBUG_WWW
 	_log.Log(LOG_STATUS, "[web:%s] create ssl_server using ssl_server_settings : %s", ssl_settings.listening_port.c_str(), ssl_settings.to_string().c_str());
@@ -161,7 +161,7 @@
 ssl_server::ssl_server(const server_settings & settings, request_handler & user_request_handler) :
 		server_base(settings, user_request_handler),
 		settings_(dynamic_cast<ssl_server_settings const &>(settings)),
-		context_(io_service_, dynamic_cast<ssl_server_settings const &>(settings).get_ssl_method()) {
+		context_(dynamic_cast<ssl_server_settings const &>(settings).get_ssl_method()) {
 #ifdef DEBUG_WWW
 	_log.Log(LOG_STATUS, "[web:%s] create ssl_server using server_settings : %s", settings.listening_port.c_str(), settings.to_string().c_str());
 #endif
