--- libs/context/build/Jamfile.v2.orig	2013-03-03 13:39:59.684868916 +0100
+++ libs/context/build/Jamfile.v2	2013-03-03 15:09:09.893232829 +0100
@@ -45,10 +45,11 @@
 local rule default_binary_format ( )
 {
     local tmp = elf ;
-    if [ os.name ] = "NT" { tmp = pe ; }
-    else if [ os.name ] = "CYGWIN" { tmp = pe ; }
-    else if [ os.name ] = "AIX" { tmp = xcoff ; }
-    else if [ os.name ] = "MACOSX" { tmp = mach-o ; }
+    #if [ os.name ] = "NT" { tmp = pe ; }
+    #else if [ os.name ] = "CYGWIN" { tmp = pe ; }
+    #else if [ os.name ] = "AIX" { tmp = xcoff ; }
+    #else if [ os.name ] = "MACOSX" { tmp = mach-o ; }
+    tmp = pe ;
     return $(tmp) ;
 }
 
@@ -65,10 +66,11 @@
 local rule default_abi ( )
 {
     local tmp = sysv ;
-    if [ os.name ] = "NT" { tmp = ms ; }
-    else if [ os.name ] = "CYGWIN" { tmp = ms ; }
-    else if [ os.platform ] = "ARM" { tmp = aapcs ; }
-    else if [ os.platform ] = "MIPS" { tmp = o32 ; }
+    #if [ os.name ] = "NT" { tmp = ms ; }
+    #else if [ os.name ] = "CYGWIN" { tmp = ms ; }
+    #else if [ os.platform ] = "ARM" { tmp = aapcs ; }
+    #else if [ os.platform ] = "MIPS" { tmp = o32 ; }
+    tmp = ms ;
     return $(tmp) ;
 }
 
