diff --git a/brave/components/debounce/browser/debounce_navigation_throttle.cc b/brave/components/debounce/browser/debounce_navigation_throttle.cc
index ebdbfb38c6..85669affc3 100644
--- a/brave/components/debounce/browser/debounce_navigation_throttle.cc
+++ b/brave/components/debounce/browser/debounce_navigation_throttle.cc
@@ -56,7 +56,7 @@ class DebounceTabHelper
  private:
   friend class content::WebContentsUserData<DebounceTabHelper>;
 
-  std::vector<const std::string> redirects_;
+  std::vector<std::string> redirects_;
   WEB_CONTENTS_USER_DATA_KEY_DECL();
 };
 
