--- 7kaa-2.15.2/src/LocaleRes.cpp
+++ 7kaa-2.15.2/src/LocaleRes.cpp
@@ -107,10 +107,12 @@
 		strcpy(lang, "??");
 		strcpy(codeset, "ISO-8859-1");
 	}
-	strcat(codeset, "//TRANSLIT");
 
-	cd = iconv_open(codeset, "");
+	String tocode(codeset);
+	tocode += "//TRANSLIT";
+	cd = iconv_open(tocode, "");
 	cd_latin = iconv_open("ISO-8859-1", "");
+
 	in_buf = mem_add(INIT_BUF_SIZE+1);
 	in_buf_size = INIT_BUF_SIZE;
 	out_buf = mem_add(INIT_BUF_SIZE+1);

