--- asgiref/server.py	2021-11-16 15:13:28.090329487 +0000
+++ asgiref/server.py.new	2021-11-16 15:13:21.830245153 +0000
@@ -5,6 +5,8 @@
 
 from .compatibility import get_running_loop, guarantee_single_callable, run_future
 
+from asgiref.sync import async_to_sync
+
 logger = logging.getLogger(__name__)
 
 
@@ -56,10 +58,11 @@
         """
         Runs the asyncio event loop with our handler loop.
         """
-        event_loop = get_running_loop()
+#        event_loop = get_running_loop()
         asyncio.ensure_future(self.application_checker())
         try:
-            event_loop.run_until_complete(self.handle())
+            async_to_sync(self.handle())
+#            event_loop.run_until_complete(self.handle())
         except KeyboardInterrupt:
             logger.info("Exiting due to Ctrl-C/interrupt")
 
