From 0811afcfc831d6ed1f31b36718148c884a07e1ec Mon Sep 17 00:00:00 2001
From: Chloe Fontenot <foley2431@gmail.com>
Date: Mon, 14 Oct 2024 14:05:40 -0500
Subject: Fix build issue

---
 .gitignore   | 3 ++-
 src/proc.hpp | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index 6a8bc10..45bfd2e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 .vscode
-build
\ No newline at end of file
+.cache
+build
diff --git a/src/proc.hpp b/src/proc.hpp
index 07c67fb..109aab9 100644
--- a/src/proc.hpp
+++ b/src/proc.hpp
@@ -6,6 +6,7 @@
 #include <algorithm>
 #include <fstream>
 #include <iostream>
+#include <cinttypes>
 #include <limits>
 #include <map>
 #include <sstream>
@@ -279,4 +280,4 @@ class proc
         config_->nic_name(selected_nic_name());
     }
 };
-} // namespace gambal
\ No newline at end of file
+} // namespace gambal
-- 
2.47.0

