--- a/package.json
+++ b/package.json
@@ -2 +2 @@
-    "name": "VRCX",
+    "name": "vrcx",

# .no-updater doesn't work when after building
# https://github.com/vrcx-team/VRCX/issues/1490
--- a/src-electron/main.js
+++ b/src-electron/main.js
@@ -59,3 +59 @@ const startup = args.includes('--startup');
-const noUpdater =
-    args.includes('--no-updater') ||
-    fs.existsSync(path.join(rootDir, '.no-updater'));
+const noUpdater = true;

# Login.vue remains updater when use --no-updater
# https://github.com/vrcx-team/VRCX/issues/1489
--- a/src/views/Login/Login.vue
+++ b/src/views/Login/Login.vue
@@ -3,3 +2,0 @@
-        <el-tooltip placement="top" :content="t('view.login.updater')">
-            <el-button type="default" size="small" :icon="Download" circle @click="showVRCXUpdateDialog"></el-button>
-        </el-tooltip>
@@ -162 +158,0 @@
-    const { showVRCXUpdateDialog } = useVRCXUpdaterStore();
