--- qt-linux/ts3soundboard_linux.pro_orig	2017-06-25 13:04:17.367485782 +0200
+++ qt-linux/ts3soundboard_linux.pro	2017-06-25 13:05:46.234153803 +0200
@@ -20,7 +20,12 @@
 CONFIG(debug, release|debug):DEFINES += _DEBUG
 CONFIG(release, release|debug):DEFINES += NDEBUG
 
-INCLUDEPATH += ../include
+INCLUDEPATH += /opt/qt5.6/include
+INCLUDEPATH += /opt/qt5.6/include/QtWidgets
+INCLUDEPATH += /opt/qt5.6/include/QtGui
+INCLUDEPATH += /opt/qt5.6/include/QtNetwork
+INCLUDEPATH += /opt/qt5.6/include/QtCore
+INCLUDEPATH += /usr/include/teamspeak3
 
 LIBS += -L$(TS3DIR)
 
@@ -29,7 +34,7 @@
     -lavutil \
     -lswresample
 
-QMAKE_POST_LINK += "cp $${DESTDIR}/lib$${TARGET}.so $(TS3DIR)/plugins"
+#QMAKE_POST_LINK += "cp $${DESTDIR}/lib$${TARGET}.so $(TS3DIR)/plugins"
 
 
 include(./ts3soundboard.pri)
--- src/main.cpp_orig	2017-06-25 13:05:05.209847399 +0200
+++ src/main.cpp	2017-06-25 13:05:22.744978748 +0200
@@ -169,8 +169,8 @@
 
 	configModel->notifyAllEvents();
 
-	updateChecker = new UpdateChecker();
-	updateChecker->startCheck();
+//	updateChecker = new UpdateChecker();
+//	updateChecker->startCheck();
 }
 
 
--- src/peakmeter.h_orig	2017-06-25 13:09:23.183737983 +0200
+++ src/peakmeter.h	2017-06-25 13:10:14.383142304 +0200
@@ -5,6 +5,7 @@
 // All rights reserved
 // Contact: rp_soundboard@mgraefe.de
 //----------------------------------
+#include <math.h>
 
 #pragma once
 #ifndef rpsbsrc__peakmeter_H__
