--- pyproject.toml	2025-01-05 15:08:42.000000000 +0530
+++ pyproject.toml	2025-01-06 09:18:18.470071688 +0530
@@ -135,7 +135,7 @@
 path = "package.json"
 pattern = '"version":\s*"(?P<version>[^"]+)"'

-[tool.hatch.build.hooks.custom]  # keep this for reading hooks from `hatch_build.py`
+#[tool.hatch.build.hooks.custom]  # keep this for reading hooks from `hatch_build.py`

 [tool.hatch.build.targets.wheel]
 sources = ["backend"]
@@ -150,7 +150,7 @@
     "webui.db",
     "chroma.sqlite3",
 ]
-force-include = { "CHANGELOG.md" = "open_webui/CHANGELOG.md", build = "open_webui/frontend" }
+force-include = { "CHANGELOG.md" = "open_webui/CHANGELOG.md" }

 [tool.codespell]
 # Ref: https://github.com/codespell-project/codespell#using-a-config-file
