diff --git a/js/src/builtin/TestingFunctions.cpp b/js/src/builtin/TestingFunctions.cpp
index 9189ceb214..eee87893e4 100644
--- a/js/src/builtin/TestingFunctions.cpp
+++ b/js/src/builtin/TestingFunctions.cpp
@@ -7382,16 +7382,6 @@ JS_FOR_WASM_FEATURES(WASM_FEATURE, WASM_FEATURE)
 "helperThreadCount()",
 "  Returns the number of helper threads available for off-thread tasks."),
 
-    JS_FN_HELP("createShapeSnapshot", CreateShapeSnapshot, 1, 0,
-"createShapeSnapshot(obj)",
-"  Returns an object containing a shape snapshot for use with\n"
-"  checkShapeSnapshot.\n"),
-
-    JS_FN_HELP("checkShapeSnapshot", CheckShapeSnapshot, 2, 0,
-"checkShapeSnapshot(snapshot, [obj])",
-"  Check shape invariants based on the given snapshot and optional object.\n"
-"  If there's no object argument, the snapshot's object is used.\n"),
-
     JS_FN_HELP("enableShapeConsistencyChecks", EnableShapeConsistencyChecks, 0, 0,
 "enableShapeConsistencyChecks()",
 "  Enable some slow Shape assertions.\n"),
