--- vrc-get-gui-v1.1.2-patched/vrc-get-gui/Tauri.toml	2025-06-30 16:10:58.312827953 +0900
+++ vrc-get-gui-v1.1.2/vrc-get-gui/Tauri.toml	2025-06-30 16:11:17.682821820 +0900
@@ -34,8 +34,6 @@
 resources = []
 publisher = "anatawa12"
 
-createUpdaterArtifacts = "v1Compatible"  # remove if ci # we do not generate updater artifacts in CI
-
 [[bundle.fileAssociations]]
 # note: for macOS we directory use info.plist for registering file association.
 description = "ALCOM Project Template"
