diff -Naurp a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
--- a/cmake/Dependencies.cmake	2018-08-14 16:43:38.000000000 +0000
+++ b/cmake/Dependencies.cmake	2018-08-14 16:30:23.000000000 +0000
@@ -327,11 +327,11 @@ endif()
 # ---[ EIGEN
 # Due to license considerations, we will only use the MPL2 parts of Eigen.
 set(EIGEN_MPL2_ONLY 1)
-find_package(Eigen3)
+#find_package(Eigen3)
 if(EIGEN3_FOUND)
   message(STATUS "Found system Eigen at " ${EIGEN3_INCLUDE_DIR})
 else()
-  message(STATUS "Did not find system Eigen. Using third party subdirectory.")
+  message(STATUS "Using Eigen third party subdirectory for compatibility.")
   set(EIGEN3_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/../third_party/eigen)
 endif()
 include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
