From 7e1c5ca6d5b1aa97ffe1f35acf5d4a2ece6c0d47 Mon Sep 17 00:00:00 2001
From: xkszltl <xkszltl@users.noreply.github.com>
Date: Thu, 19 Apr 2018 20:46:09 -0700
Subject: [PATCH] Add missing #include for CAFFE2_MODULE macro. (#6790)

---
 modules/rocksdb/rocksdb.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/rocksdb/rocksdb.cc b/modules/rocksdb/rocksdb.cc
index 8f21c5fc2a1..64fdde4fff5 100644
--- a/modules/rocksdb/rocksdb.cc
+++ b/modules/rocksdb/rocksdb.cc
@@ -16,6 +16,7 @@
 
 #include "caffe2/core/db.h"
 #include "caffe2/core/logging.h"
+#include "caffe2/core/module.h"
 #include "caffe2/core/flags.h"
 #include "rocksdb/db.h"
 #include "rocksdb/utilities/leveldb_options.h"
