diff --git a/src/sdl_milton.cc b/src/sdl_milton.cc
index 2384abf..84921c1 100644
--- a/src/sdl_milton.cc
+++ b/src/sdl_milton.cc
@@ -794,7 +794,7 @@ milton_main(bool is_fullscreen, char* file_to_open)
         milton_log("ImGUI setup\n");
         ImGuiIO& io = ImGui::GetIO();
         io.IniFilename = NULL;  // Don't save any imgui.ini file
-        PATH_CHAR fname[MAX_PATH] = TO_PATH_STR("Carlito.ttf");
+        PATH_CHAR fname[MAX_PATH] = TO_PATH_STR("/usr/share/fonts/TTF/Carlito-Regular.ttf");
         platform_fname_at_exe(fname, MAX_PATH);
         FILE* fd = platform_fopen(fname, TO_PATH_STR("rb"));
 
