--- lastfm-desktop-2.1.30/app/client/Services/RadioService/RadioService.cpp
+++ lastfm-desktop-2.1.30/app/client/Services/RadioService/RadioService.cpp
@@ -309,7 +309,7 @@
 void
 RadioService::onPhononStateChanged( Phonon::State newstate, Phonon::State oldstate )
 {
-    qDebug() << oldstate << " -> " << newstate;
+    qDebug() << (int)oldstate << " -> " << (int)newstate;
     if (m_mediaObject == 0) {
         qDebug() << "m_mediaObject is null!";
         return;

