--- CCPhysicsBody.cpp	2016-12-29 13:42:38.000000000 -0400
+++ /opt/cocos2d-x-git/cocos/physics/CCPhysicsBody.cpp	2017-01-04 13:22:53.445318965 -0400
@@ -575,12 +575,6 @@
 
 void PhysicsBody::setVelocity(const Vec2& velocity)
 {
-    if (!_dynamic)
-    {
-        CCLOG("physics warning: your can't set velocity for a static body.");
-        return;
-    }
-    
     cpBodySetVelocity(_cpBody, PhysicsHelper::point2cpv(velocity));
 }
 
@@ -601,12 +595,6 @@
 
 void PhysicsBody::setAngularVelocity(float velocity)
 {
-    if (!_dynamic)
-    {
-        CCLOG("physics warning: your can't set angular velocity for a static body.");
-        return;
-    }
-    
     cpBodySetAngularVelocity(_cpBody, velocity);
 }
 
