--- /opt/cocos2d-x/cocos/renderer/ccShader_3D_Skybox.frag	2016-12-29 13:42:38.000000000 -0400
+++ /tmp/ccShader_3D_Skybox.frag	2017-01-04 15:03:03.586738888 -0400
@@ -1,9 +1,9 @@
-const char* cc3D_Skybox_frag = R"(
-#ifdef GL_ES
+const char* cc3D_Skybox_frag = STRINGIFY(
+\n#ifdef GL_ES\n
 varying mediump vec3        v_reflect;
-#else
+\n#else\n
 varying vec3        v_reflect;
-#endif
+\n#endif\n
 uniform samplerCube u_Env;
 uniform vec4 u_color;
 
@@ -11,4 +11,4 @@
 {
     gl_FragColor = textureCube(u_Env, v_reflect) * u_color;
 }
-)";
+);
