--- /opt/cocos2d-x/cocos/renderer/ccShader_3D_Particle.frag	2016-12-29 13:42:38.000000000 -0400
+++ /tmp/ccShader_3D_Particle.frag	2017-01-04 15:03:03.356765365 -0400
@@ -1,20 +1,20 @@
 
-const char* cc3D_Particle_tex_frag = R"(
+const char* cc3D_Particle_tex_frag = STRINGIFY(
 
-#ifdef GL_ES
+\n#ifdef GL_ES\n
 varying mediump vec2 TextureCoordOut;
 varying mediump vec4 ColorOut;
-#else
+\n#else\n
 varying vec4 ColorOut;
 varying vec2 TextureCoordOut;
-#endif
+\n#endif\n
 uniform vec4 u_color;
 
 void main(void)
 {
     gl_FragColor = texture2D(CC_Texture0, TextureCoordOut) * ColorOut * u_color;
 }
-)";
+);
 
 const char* cc3D_Particle_color_frag = R"(
 
