From cdc7bd2f15c89fdc58bed2208392dd857cd479d5 Mon Sep 17 00:00:00 2001
From: Andrey Pavlenko <andrey.a.pavlenko@gmail.com>
Date: Thu, 27 May 2021 03:07:54 +0300
Subject: [PATCH] Fixed web module build

---
 web/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/web/CMakeLists.txt b/web/CMakeLists.txt
index 71136967cf..bc2e6e742d 100644
--- a/web/CMakeLists.txt
+++ b/web/CMakeLists.txt
@@ -68,6 +68,7 @@ add_custom_target(
   trweb ALL
   COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/package.json" "${CMAKE_CURRENT_BINARY_DIR}/package.json"
   COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/yarn.lock" "${CMAKE_CURRENT_BINARY_DIR}/yarn.lock"
+  COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_CURRENT_BINARY_DIR}/node_modules" "${CMAKE_CURRENT_SOURCE_DIR}/node_modules"
   COMMAND yarn install
   COMMAND yarn webpack --config "${CMAKE_CURRENT_SOURCE_DIR}/webpack.config.js" --context "${CMAKE_CURRENT_SOURCE_DIR}"
   BYPRODUCTS
