diff --git a/whipper/program/cdparanoia.py b/whipper/program/cdparanoia.py
index 160646e..e2c45aa 100644
--- a/whipper/program/cdparanoia.py
+++ b/whipper/program/cdparanoia.py
@@ -277,10 +277,10 @@ class ReadTrackTask(task.Task):
 
         bufsize = 1024
         if self._overread:
-            argv = ["cd-paranoia", "--stderr-progress",
+            argv = ["cdparanoia", "--stderr-progress",
                     "--sample-offset=%d" % self._offset, "--force-overread", ]
         else:
-            argv = ["cd-paranoia", "--stderr-progress",
+            argv = ["cdparanoia", "--stderr-progress",
                     "--sample-offset=%d" % self._offset, ]
         if self._device:
             argv.extend(["--force-cdrom-device", self._device, ])
@@ -298,7 +298,7 @@ class ReadTrackTask(task.Task):
                                          close_fds=True)
         except OSError as e:
             if e.errno == errno.ENOENT:
-                raise common.MissingDependencyException('cd-paranoia')
+                raise common.MissingDependencyException('cdparanoia')
 
             raise
 
@@ -557,8 +557,8 @@ _VERSION_RE = re.compile(
 
 
 def getCdParanoiaVersion():
-    getter = common.VersionGetter('cd-paranoia',
-                                  ["cd-paranoia", "-V"],
+    getter = common.VersionGetter('cdparanoia',
+                                  ["cdparanoia", "-V"],
                                   _VERSION_RE,
                                   "%(version)s %(release)s")
 
@@ -584,12 +584,12 @@ class AnalyzeTask(ctask.PopenTask):
     def __init__(self, device=None):
         # cdparanoia -A *always* writes cdparanoia.log
         self.cwd = tempfile.mkdtemp(suffix='.whipper.cache')
-        self.command = ['cd-paranoia', '-A']
+        self.command = ['cdparanoia', '-A']
         if device:
             self.command += ['-d', device]
 
     def commandMissing(self):
-        raise common.MissingDependencyException('cd-paranoia')
+        raise common.MissingDependencyException('cdparanoia')
 
     def readbyteserr(self, bytes_stderr):
         self._output.append(bytes_stderr)
