diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c63a6b..a1a6639 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         -Wno-unused-parameter
         -Wno-missing-field-initializers
         -Wno-unknown-pragmas
+        -Wno-shift-negative-value
+        -Wno-strict-aliasing
         -Wno-reorder)
     if(CMAKE_COMPILER_IS_GNUCXX)
         list(APPEND warning_flags -Wno-unused-but-set-variable)
