diff --git a/src/dsd_ncurses.c b/src/dsd_ncurses.c
index 96ab846..f498bcd 100644
--- a/src/dsd_ncurses.c
+++ b/src/dsd_ncurses.c
@@ -203,10 +203,10 @@ void beeper (dsd_opts * opts, dsd_state * state, int type)
   FILE *beep;
   char wav_name[1024] = {0};

-  if (opts->pulse_digi_rate_out == 8000) strncpy(wav_name, "/usr/share/tone8.wav", 1023);
-  if (opts->pulse_digi_rate_out == 48000) strncpy(wav_name, "/usr/share/tone48.wav", 1023);
-  if (opts->pulse_digi_rate_out == 24000) strncpy(wav_name, "/usr/share/tone24.wav", 1023);
-  if (opts->audio_out_type == 5) strncpy(wav_name, "/usr/share/tone24.wav", 1023);
+  if (opts->pulse_digi_rate_out == 8000) strncpy(wav_name, "/usr/share/dsd-fme/tone8.wav", 1023);
+  if (opts->pulse_digi_rate_out == 48000) strncpy(wav_name, "/usr/share/dsd-fme/tone48.wav", 1023);
+  if (opts->pulse_digi_rate_out == 24000) strncpy(wav_name, "/usr/share/dsd-fme/tone24.wav", 1023);
+  if (opts->audio_out_type == 5) strncpy(wav_name, "/usr/share/dsd-fme/tone24.wav", 1023);
   wav_name[1023] = '\0';
   struct stat stat_buf;
   if (stat(wav_name, &stat_buf) == 0)
