diff --git a/configure.ac b/configure.ac
index 4c2644b11d..ab3b78f863 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2524,7 +2524,7 @@ AC_DEFUN
     ## 6.3.5 is the earliest version known to work; see Bug#17339.
     ## 6.8.2 makes Emacs crash; see Bug#13867.
     ## 7 and later have not been ported to; See Bug#25967.
-    IMAGEMAGICK_MODULE="Wand >= 6.3.5 Wand != 6.8.2 Wand < 7"
+    IMAGEMAGICK_MODULE="Wand-6.Q16HDRI"
     EMACS_CHECK_MODULES([IMAGEMAGICK], [$IMAGEMAGICK_MODULE])
 
     if test $HAVE_IMAGEMAGICK = yes; then
