--- include/chipmunk/chipmunk_ffi.h.orig	2011-10-13 17:32:21.383255385 +0200
+++ include/chipmunk/chipmunk_ffi.h	2011-10-13 17:32:27.493226698 +0200
@@ -160,7 +160,6 @@
 MAKE_PROPERTIES_REF(cpSpace, Damping);
 MAKE_PROPERTIES_REF(cpSpace, IdleSpeedThreshold);
 MAKE_PROPERTIES_REF(cpSpace, SleepTimeThreshold);
-MAKE_PROPERTIES_REF(cpSpace, SleepTimeThreshold);
 MAKE_PROPERTIES_REF(cpSpace, CollisionSlop);
 MAKE_PROPERTIES_REF(cpSpace, CollisionBias);
 MAKE_PROPERTIES_REF(cpSpace, CollisionPersistence);
