--- patch-4.19.25-ck1	2019-04-22 19:37:07.000000000 +0800
+++ patch-4.19.34-ck1	2019-04-22 21:25:52.000000000 +0800
@@ -16165,10 +16165,10 @@
 index cc02050fd0c4..7dac6885128e 100644
 --- a/kernel/sysctl.c
 +++ b/kernel/sysctl.c
-@@ -126,9 +126,17 @@ static int __maybe_unused one = 1;
- static int __maybe_unused two = 2;
+@@ -127,9 +127,17 @@ static int __maybe_unused one = 1;
  static int __maybe_unused four = 4;
  static unsigned long one_ul = 1;
+ static unsigned long long_max = LONG_MAX;
 -static int one_hundred = 100;
 -static int one_thousand = 1000;
 -#ifdef CONFIG_PRINTK
