# set client ID. Generated own ID from twitch
diff --git a/src/chatty/Chatty.java b/src/chatty/Chatty.java
index 618985c..bb54ccf 100644
--- a/src/chatty/Chatty.java
+++ b/src/chatty/Chatty.java
@@ -37,7 +37,7 @@ public class Chatty {
      * If you compile this program yourself, you should create your own client
      * id on http://www.twitch.tv/kraken/oauth2/clients/new
      */
-    public static final String CLIENT_ID = "spyiu9jqdnfjtwv6l1xjk5zgt8qb91l";
+    public static final String CLIENT_ID = "q1b6bqsamkie9jl3917hsvaq388psra";
     
     /**
      * The redirect URI for getting a token.
-- 
2.5.0

