--- a/bin/clion.sh	2021-02-10 02:08:05.000000000 -0800
+++ b/bin/clion.sh	2021-02-11 17:14:41.753248506 -0800
@@ -38,7 +38,7 @@ OS_ARCH=$(uname -m)
 # ---------------------------------------------------------------------
 # Ensure $IDE_HOME points to the directory where the IDE is installed.
 # ---------------------------------------------------------------------
-cd "$(dirname "$0")" || exit 2
+cd "$(dirname "$(realpath "$0")")" || exit 2
 IDE_BIN_HOME=$(pwd -P)
 IDE_HOME=$(dirname "$IDE_BIN_HOME")
 cd "${OLDPWD}" || exit 2
