diff --unified --recursive --text cloudflare-blog.orig/2017-06-29-ssdp/mmhistogram cloudflare-blog.new/2017-06-29-ssdp/mmhistogram
--- cloudflare-blog.orig/2017-06-29-ssdp/mmhistogram	2023-09-06 00:55:02.952366159 +0200
+++ cloudflare-blog.new/2017-06-29-ssdp/mmhistogram	2023-09-06 00:50:10.223048641 +0200
@@ -1,8 +1,9 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
+
 import argparse
 import itertools
-import math
 import sys
+import statistics
 
 parser = argparse.ArgumentParser(description='Print log-2 histogram, like systemtap')
 parser.add_argument('-t', '--title', default="Values",
@@ -31,10 +32,10 @@
 totalcount = len(M)
 minval = M[-1]
 maxval = M[0]
-avgval = sum(M) / float(len(M))
-devval = math.sqrt(sum([(m - avgval)**2 for m in M]) / float(len(M)))
+avgval = statistics.mean(M)
+devval = statistics.stdev(M)
+medval = statistics.median(M)
 
-medval = M[len(M)/2]
 
 KV = []
 
@@ -65,35 +66,41 @@
 maxbound = KV[-1][0]
 boundl = max(len(str(maxbound)), len('value'))
 
-print "%s min:%.2f avg:%.2f med=%.2f max:%.2f dev:%.2f count:%d" % (
+print_values = "%s min:%.2f avg:%.2f med=%.2f max:%.2f dev:%.2f count:%d" % (
     args.title,
     minval,
     avgval,
     medval,
     maxval,
     devval,
-    totalcount,
+    totalcount
 )
-print "%s:" % (
+print(print_values)
+
+print_title = "%s:" % (
     args.title,
 )
-print "%*s |%*s %s" % (
+print(print_title)
+
+print_column = "%*s |%*s %s" % (
     boundl+1,
     "value",
     args.columns,
     "-" * args.columns,
     "count"
 )
+print(print_column)
 
 for boundb, bound, c in KV:
     if args.percentage:
         cp = "%5.2f%%" % ((c / float(totalcount))*100.0,)
     else:
         cp = "%d" % (c,)
-    print "%*d |%*s %s" % (
+    print_bound = "%*d |%*s %s" % (
         boundl + 1,
         boundb,
         args.columns,
         "*" * int(args.columns * (c/maxcount)),
         cp
     )
+    print(print_bound)
diff --unified --recursive --text cloudflare-blog.orig/2017-06-29-ssdp/mmsum cloudflare-blog.new/2017-06-29-ssdp/mmsum
--- cloudflare-blog.orig/2017-06-29-ssdp/mmsum	2023-09-06 00:55:02.952366159 +0200
+++ cloudflare-blog.new/2017-06-29-ssdp/mmsum	2023-09-05 22:01:48.306100235 +0200
@@ -6,4 +6,4 @@
 for line in sys.stdin:
     s += float(line)
 
-print s
+print(s)
Only in cloudflare-blog.orig/2017-06-29-ssdp: mmwatch
