diff --git /partio/CMakeLists.txt /partio/CMakeLists.txt
index fdab418..c6a5a4b 100644
--- /partio/CMakeLists.txt
+++ /partio/CMakeLists.txt
@@ -103,11 +103,11 @@ SET( PARTIO_LIBRARIES partio ${ZLIB_LIBRARY} )
 # make it so partio can be found
 INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/src/lib )
 
-SET(SEEXPR_BASE "c:/aselle/seexpr-inst" CACHE STRING "path to SeExpr")
+SET(SEEXPR_BASE "/usr" CACHE STRING "path to SeExpr")
 SET(SEEXPR_INCLUDE_PATH ${SEEXPR_BASE}/include)
 SET(SEEXPR_LIBDIR ${CMAKE_INSTALL_LIBDIR})
 SET(SEEXPR_LINK_PATH ${SEEXPR_BASE}/${SEEXPR_LIBDIR})
-SET(SEEXPR_LIBS "SeExpr")
+SET(SEEXPR_LIBS "SeExpr2")
 
 
 ## Traverse subdirectories
