diff --git a/src/cmake/FindMAGMA.cmake b/src/cmake/FindMAGMA.cmake
index 51cd4b30..3f5fc72d 100644
--- a/src/cmake/FindMAGMA.cmake
+++ b/src/cmake/FindMAGMA.cmake
@@ -17,6 +17,8 @@
 #If environment variable MAGMA_ROOT is specified, it has same effect as MAGMA_ROOT
 if( NOT MAGMA_ROOT AND NOT $ENV{MAGMA_ROOT} STREQUAL "" )
     set( MAGMA_ROOT $ENV{MAGMA_ROOT} )
+endif()
+if(MAGMA_ROOT)
     # set library directories
     set(MAGMA_LIBRARY_DIRS ${MAGMA_ROOT}/lib)
     # set include directories
@@ -32,4 +34,4 @@ if( NOT MAGMA_ROOT AND NOT $ENV{MAGMA_ROOT} STREQUAL "" )
     set(MAGMA_FOUND TRUE)
 else()
     set(MAGMA_FOUND FALSE)
-endif()
\ No newline at end of file
+endif()
