--- CMakeLists.txt	2025-08-26 03:54:43.000000000 +0800
+++ CMakeLists.txt.new	2025-09-25 17:35:21.056855979 +0800
@@ -51,17 +51,7 @@
   add_definitions(-DWITH_CUDA)
   set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-relaxed-constexpr -allow-unsupported-compiler")
 
-  if (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
-    set(CMAKE_CUDA_ARCHITECTURES "75;80;86;90;100;120")
-  elseif (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 12.8)
-    set(CMAKE_CUDA_ARCHITECTURES "70;75;80;86;90;100;120")
-  elseif (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.8)
-    set(CMAKE_CUDA_ARCHITECTURES "60;70;75;80;86;90")
-  elseif (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.3)
-    set(CMAKE_CUDA_ARCHITECTURES "60;70;75;80;86")
-  else()
-    set(CMAKE_CUDA_ARCHITECTURES "60;70;75")
-  endif()
+  set(CMAKE_CUDA_ARCHITECTURES native)
 
   if (NOT "$ENV{EXTERNAL_CUTLASS_INCLUDE_DIR}" STREQUAL "")
     include_directories($ENV{EXTERNAL_CUTLASS_INCLUDE_DIR})
@@ -71,8 +61,6 @@
     set(CUTLASS_UTIL_DIR third_party/cutlass/tools/util/include)
     include_directories(${CUTLASS_UTIL_DIR})
   endif()
-  set(CUCOLLECTIONS_DIR third_party/cuCollections/include)
-  include_directories(${CUCOLLECTIONS_DIR})
 endif()
 
 set(CSRC pyg_lib/csrc)
@@ -123,10 +111,10 @@
 endif()
 
 if(WITH_CUDA)
-  target_include_directories(${PROJECT_NAME} PRIVATE
-                             third_party/cccl/thrust
-                             third_party/cccl/cub
-                             third_party/cccl/libcudacxx/include)
+  find_package(CCCL REQUIRED)
+  find_package(cuco REQUIRED)
+  target_link_libraries(${PROJECT_NAME} PRIVATE CCCL::CCCL)
+  target_link_libraries(${PROJECT_NAME} PRIVATE cuco::cuco)
 endif()
 
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0")
