--- CMakeLists.txt.orig	2017-02-13 20:37:27.549331001 +0100
+++ CMakeLists.txt	2017-02-13 20:37:39.109879899 +0100
@@ -116,18 +116,6 @@
         set(REMMINA_VERSION "${REMMINA_VERSION}-${REMMINA_VERSION_SUFFIX}")
 endif()
 
-
-include(GetGitRevisionDescription)
-git_get_exact_tag(GIT_REVISION --tags --always)
-if (${GIT_REVISION} STREQUAL "n/a")
-        git_rev_parse(GIT_REVISION --short)
-endif()
-
-if(NOT GIT_REVISION)
-    set(GIT_REVISION ${REMMINA_VERSION_SUFFIX})
-endif()
-message(STATUS "Git Revision ${GIT_REVISION}")
-
 if(NOT UNIQUE_APPNAME)
     set(UNIQUE_APPNAME "org.Remmina")
 endif()
