diff -ruN aom/build/cmake/aom_configure.cmake patched/build/cmake/aom_configure.cmake
--- aom/build/cmake/aom_configure.cmake	2020-05-25 10:32:02.443333045 +0200
+++ patched/build/cmake/aom_configure.cmake	2020-05-25 10:37:34.206666378 +0200
@@ -67,6 +67,7 @@
                     "      CMAKE_GENERATOR=${CMAKE_GENERATOR}\n")
     endif()
   elseif("${cpu_lowercase}" STREQUAL "i386"
+         OR "${cpu_lowercase}" STREQUAL "i686"
          OR "${cpu_lowercase}" STREQUAL "x86")
     set(AOM_TARGET_CPU "x86")
   elseif("${cpu_lowercase}" MATCHES "^arm"
