--- CMakeLists.txt	2025-07-27 11:58:32.000000000 -0500
+++ CMakeLists.new	2025-07-28 04:08:01.761112955 -0500
@@ -8,6 +8,7 @@
 set(PACKAGE_STRING "${PROJECT_NAME} ${PROJECT_VERSION}")
 set(PACKAGE_COPYRIGHT "Copyright (C) 1996-2025")
 set(PACKAGE_LICENSE "GNU General Public License, version 3")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=format-security")
 
 # Find math library
 find_library(MATH m)
