diff -upr src-old/client-2.4.1/cmake/modules/QtVersionAbstraction.cmake src/client-2.4.1/cmake/modules/QtVersionAbstraction.cmake
--- src-old/client-2.4.1/cmake/modules/QtVersionAbstraction.cmake	2018-06-15 23:41:00.778417109 +0200
+++ src/client-2.4.1/cmake/modules/QtVersionAbstraction.cmake	2018-06-16 12:25:32.250294653 +0200
@@ -9,6 +9,7 @@ find_package(Qt5Core REQUIRED)
 find_package(Qt5Network REQUIRED)
 find_package(Qt5Xml REQUIRED)
 find_package(Qt5Concurrent REQUIRED)
+find_package(Qt5Sql REQUIRED)
 if(UNIT_TESTING)
     find_package(Qt5Test REQUIRED)
 endif()
diff -upr src-old/client-2.4.1/CMakeLists.txt src/client-2.4.1/CMakeLists.txt
--- src-old/client-2.4.1/CMakeLists.txt	2018-06-15 23:41:00.571751406 +0200
+++ src/client-2.4.1/CMakeLists.txt	2018-06-16 12:00:28.329726348 +0200
@@ -1,5 +1,5 @@
 
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8.12)
 cmake_policy(VERSION 2.8.0)
 if(POLICY CMP0020)
   cmake_policy(SET CMP0020 NEW)
diff -upr src-old/client-2.4.1/src/cmd/CMakeLists.txt src/client-2.4.1/src/cmd/CMakeLists.txt
--- src-old/client-2.4.1/src/cmd/CMakeLists.txt	2018-06-15 23:41:01.605079928 +0200
+++ src/client-2.4.1/src/cmd/CMakeLists.txt	2018-06-16 11:59:45.359770661 +0200
@@ -27,7 +27,7 @@ endif()
 
 if(NOT BUILD_LIBRARIES_ONLY)
    add_executable(${cmd_NAME}  ${cmd_SRC})
-	qt5_use_modules(${cmd_NAME} Network )
+	target_link_libraries(${cmd_NAME} Qt5::Network )
 	set_target_properties(${cmd_NAME} PROPERTIES
 	        RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY} )
         set_target_properties(${cmd_NAME} PROPERTIES
diff -upr src-old/client-2.4.1/src/csync/CMakeLists.txt src/client-2.4.1/src/csync/CMakeLists.txt
--- src-old/client-2.4.1/src/csync/CMakeLists.txt	2018-06-15 23:41:01.658413013 +0200
+++ src/client-2.4.1/src/csync/CMakeLists.txt	2018-06-15 23:43:19.384535393 +0200
@@ -134,7 +134,7 @@ if(ZLIB_FOUND)
 endif(ZLIB_FOUND)
 
 find_package(Qt5Core REQUIRED)
-qt5_use_modules(${CSYNC_LIBRARY} Core Concurrent)
+target_link_libraries(${CSYNC_LIBRARY} Qt5::Core Qt5::Concurrent)
 
 # For src/common/utility_mac.cpp
 if (APPLE)
diff -upr src-old/client-2.4.1/src/csync/std/CMakeLists.txt src/client-2.4.1/src/csync/std/CMakeLists.txt
--- src-old/client-2.4.1/src/csync/std/CMakeLists.txt	2018-06-15 23:41:01.698412827 +0200
+++ src/client-2.4.1/src/csync/std/CMakeLists.txt	2018-06-15 23:42:23.114735762 +0200
@@ -36,4 +36,4 @@ add_library(${CSTDLIB_LIBRARY} STATIC ${
 if(NOT WIN32)
     add_definitions( -fPIC )
 endif()
-qt5_use_modules(${CSTDLIB_LIBRARY} Core)
+target_link_libraries(${CSTDLIB_LIBRARY} Qt5::Core)
diff -upr src-old/client-2.4.1/src/gui/CMakeLists.txt src/client-2.4.1/src/gui/CMakeLists.txt
--- src-old/client-2.4.1/src/gui/CMakeLists.txt	2018-06-15 23:41:01.735079323 +0200
+++ src/client-2.4.1/src/gui/CMakeLists.txt	2018-06-16 12:24:22.400232017 +0200
@@ -236,7 +236,7 @@ list(APPEND final_src ${ownCloud})
 set(ownCloud ${ownCloud_old})
 
 if (WITH_DBUS)
-    set(ADDITIONAL_APP_MODULES DBus)
+    set(ADDITIONAL_APP_MODULES Qt5::DBus)
 endif(WITH_DBUS)
 if (NOT NO_SHIBBOLETH)
     list(APPEND ADDITIONAL_APP_MODULES WebKitWidgets)
@@ -265,14 +265,14 @@ if(NOT BUILD_OWNCLOUD_OSX_BUNDLE)
 
     # add_executable( ${APPLICATION_EXECUTABLE} main.cpp ${final_src})
     add_executable( ${APPLICATION_EXECUTABLE} WIN32 main.cpp ${final_src})
-    qt5_use_modules(${APPLICATION_EXECUTABLE} Widgets Network Xml Sql ${ADDITIONAL_APP_MODULES})
+    target_link_libraries(${APPLICATION_EXECUTABLE} Qt5::Widgets Qt5::Network Qt5::Xml Qt5::Sql ${ADDITIONAL_APP_MODULES})
 else()
     # set(CMAKE_INSTALL_PREFIX ".") # Examples use /Applications. hurmpf.
     set(MACOSX_BUNDLE_ICON_FILE "ownCloud.icns")
 
     # we must add MACOSX_BUNDLE only if building a bundle
     add_executable( ${APPLICATION_EXECUTABLE} WIN32 MACOSX_BUNDLE main.cpp ${final_src})
-    qt5_use_modules(${APPLICATION_EXECUTABLE} Widgets Network Xml Sql  ${ADDITIONAL_APP_MODULES})
+    target_link_libraries(${APPLICATION_EXECUTABLE} Qt5::Widgets Qt5::Network Qt5::Xml Qt5::Sql  ${ADDITIONAL_APP_MODULES})
 
     set (QM_DIR ${OWNCLOUD_OSX_BUNDLE}/Contents/Resources/Translations)
     install(FILES ${client_I18N} DESTINATION ${QM_DIR})
@@ -291,7 +291,7 @@ endif()
 
 add_library(updater STATIC ${updater_SRCS} ${updaterMoc})
 target_link_libraries(updater ${synclib_NAME})
-qt5_use_modules(updater Widgets Network Xml)
+target_link_libraries(updater Qt5::Widgets Qt5::Network Qt5::Xml)
 
 set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY}
diff -upr src-old/client-2.4.1/src/libsync/CMakeLists.txt src/client-2.4.1/src/libsync/CMakeLists.txt
--- src-old/client-2.4.1/src/libsync/CMakeLists.txt	2018-06-15 23:41:02.001744749 +0200
+++ src/client-2.4.1/src/libsync/CMakeLists.txt	2018-06-15 23:44:08.254383204 +0200
@@ -120,9 +120,9 @@ GENERATE_EXPORT_HEADER( ${synclib_NAME}
 )
 
 if(TOKEN_AUTH_ONLY)
-    qt5_use_modules(${synclib_NAME} Network)
+    target_link_libraries(${synclib_NAME} Qt5::Network)
 else()
-    qt5_use_modules(${synclib_NAME} Widgets Network)
+    target_link_libraries(${synclib_NAME} Qt5::Widgets Qt5::Network)
 endif()
 
 set_target_properties( ${synclib_NAME}  PROPERTIES
