diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3297673..602909f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,7 @@ qt_standard_project_setup(I18N_TRANSLATED_LANGUAGES en de cs es ko zh_CN)
 find_package(OpenSSL REQUIRED)
 find_package(ZLIB REQUIRED)
 find_package(Freetype REQUIRED)
-find_package(OpenJPEG CONFIG REQUIRED)
+find_package(OpenJPEG REQUIRED)
 find_package(JPEG REQUIRED)
 find_package(PNG REQUIRED)
 find_package(blend2d CONFIG REQUIRED)
 
