diff -ru mygui-MyGUI3.2.2/CMakeLists.txt patched/CMakeLists.txt
--- mygui-MyGUI3.2.2/CMakeLists.txt	2015-01-26 16:32:35.000000000 +0100
+++ patched/CMakeLists.txt	2015-02-06 17:16:09.016820193 +0100
@@ -24,6 +24,8 @@
 set(MYGUI_TEMPLATES_DIR "${MYGUI_SOURCE_DIR}/CMake/Templates")
 set(MYGUI_WORK_DIR ${MYGUI_BINARY_DIR})
 
+find_package(Boost REQUIRED COMPONENTS system)
+
 
 #####################################################################
 # Set up the basic build environment
