diff --unified --recursive --text qiskit-aer-0.4.0/CMakeLists.txt qiskit-aer-0.4.0-patched/CMakeLists.txt
--- qiskit-aer-0.4.0/CMakeLists.txt	2020-02-06 17:43:27.000000000 +0100
+++ qiskit-aer-0.4.0-patched/CMakeLists.txt	2020-02-07 22:19:39.729434756 +0100
@@ -265,8 +265,6 @@
 
 # Cython build is only enabled if building through scikit-build.
 if(SKBUILD) # Terra Addon build
-	add_muparserx_lib()
-
 	add_subdirectory(qiskit/providers/aer/pulse/cy)
 	add_subdirectory(qiskit/providers/aer/pulse/qutip_lite/cy)
 	add_subdirectory(qiskit/providers/aer/backends/wrappers)
@@ -291,7 +289,8 @@
 		RUNTIME_OUTPUT_DIRECTORY_RELEASE Release)
 	target_include_directories(qasm_simulator
 		PRIVATE ${AER_SIMULATOR_CPP_SRC_DIR}
-		PRIVATE ${AER_SIMULATOR_CPP_EXTERNAL_LIBS})
+		PRIVATE ${AER_SIMULATOR_CPP_EXTERNAL_LIBS}
+		PRIVATE "/usr/include/muparserx/")
 	target_compile_definitions(qasm_simulator
 		PRIVATE ${AER_COMPILER_DEFINITIONS})
 	install(TARGETS qasm_simulator DESTINATION bin)
diff --unified --recursive --text qiskit-aer-0.4.0/src/open_pulse/eval_hamiltonian.hpp qiskit-aer-0.4.0-patched/src/open_pulse/eval_hamiltonian.hpp
--- qiskit-aer-0.4.0/src/open_pulse/eval_hamiltonian.hpp	2020-02-06 17:43:27.000000000 +0100
+++ qiskit-aer-0.4.0-patched/src/open_pulse/eval_hamiltonian.hpp	2020-02-07 22:20:19.945499052 +0100
@@ -20,7 +20,7 @@
 #include <complex>
 #include <string>
 #include <numpy/arrayobject.h>
-#include <muparserx/parser/mpParser.h>
+#include <muparserx/mpParser.h>
 
 struct ParserValues {
     ParserValues(std::unique_ptr<mup::ParserX> parser, const std::string& expr):
