diff -Naur webissues-1.1.4-orig/src/data/data.pri webissues-1.1.4/src/data/data.pri
--- webissues-1.1.4-orig/src/data/data.pri	2015-06-16 21:23:58.615926000 +0500
+++ webissues-1.1.4/src/data/data.pri	2017-03-25 09:10:20.279126344 +0500
@@ -23,7 +23,7 @@
            data/query.cpp \
            data/updateevent.cpp
 
-contains( QT_CONFIG, openssl ) | contains( QT_CONFIG, openssl-linked ) {
+#contains( QT_CONFIG, openssl ) | contains( QT_CONFIG, openssl-linked ) {
     HEADERS += data/certificatesstore.h
     SOURCES += data/certificatesstore.cpp
-}
+#}
diff -Naur webissues-1.1.4-orig/src/dialogs/dialogs.pri webissues-1.1.4/src/dialogs/dialogs.pri
--- webissues-1.1.4-orig/src/dialogs/dialogs.pri	2013-04-09 04:31:06.581388000 +0600
+++ webissues-1.1.4/src/dialogs/dialogs.pri	2017-03-25 09:10:06.662459933 +0500
@@ -50,7 +50,7 @@
 
 FORMS   += dialogs/settingsdialog.ui
 
-contains( QT_CONFIG, openssl ) | contains( QT_CONFIG, openssl-linked ) {
+#contains( QT_CONFIG, openssl ) | contains( QT_CONFIG, openssl-linked ) {
     HEADERS += dialogs/ssldialogs.h
     SOURCES += dialogs/ssldialogs.cpp
-}
+#}
