diff --git a/nitter.conf b/nitter.conf
index 93526b4..c11e819 100644
--- a/nitter.conf
+++ b/nitter.conf
@@ -1,11 +1,11 @@
 [Server]
-address = "0.0.0.0"
-port = 8080
+address = "localhost"  # Change this?
+port = 13500  # Change this?
 https = false  # disable to enable cookies when not using https
 httpMaxConnections = 100
-staticDir = "./public"
-title = "nitter"
-hostname = "nitter.net"
+staticDir = "/usr/share/nitter/public"
+title = "nitter"  # Change this?
+hostname = "nitter.net"  # Change this
 
 [Cache]
 listMinutes = 240  # how long to cache list info (not the tweets, so keep it high)
@@ -20,8 +20,8 @@ redisMaxConnections = 30
 # you receive tons of requests per second
 
 [Config]
-hmacKey = "secretkey"  # random key for cryptographic signing of video urls
-base64Media = false  # use base64 encoding for proxied media urls
+hmacKey = "43179cf305cda132874f7a4d3fa4831a01528e516780157bebd564b0ca0297c2"  # Regenerate e.g. `head -c 25 </dev/urandom | sha256sum`
+base64Media = true  # use base64 encoding for proxied media urls
 enableRSS = true  # set this to false to disable RSS feeds
 enableDebug = false # enable request logs and debug endpoints
 proxy = "" # proxy type http/https
@@ -34,10 +34,10 @@ tokenCount = 10
 
 # Change default preferences here, see src/prefs_impl.nim for a complete list
 [Preferences]
-theme = "Nitter"
-replaceTwitter = "nitter.net"
-replaceYouTube = "piped.video"
-replaceReddit = "teddit.net"
+theme = "twitter_dark"  # Change this?
+replaceTwitter = "nitter.net"  # Use the same "hostname" as above
+replaceYouTube = ""  # Default: piped.kavin.rocks
+replaceReddit = ""   # Default: teddit.net
 replaceInstagram = ""
 proxyVideos = true
 hlsPlayback = false
