--- config.default.orig	2020-12-30 13:56:52.910168682 +0100
+++ config.default	2020-12-30 14:22:16.056031632 +0100
@@ -12,7 +12,7 @@
 #BACKEND = gdi
 #BACKEND = sdl
 #BACKEND = sdl2
-#BACKEND = mixer_sdl
+BACKEND = mixer_sdl2
 #BACKEND = mixer_sdl2
 #BACKEND = posix
 
@@ -21,41 +21,41 @@
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
 #DEBUG = 1    # Level 1-3, higher number means more debug-friendly but slower, see Makefile
 #MSG_LEVEL = 1 # Level 1-4, more runtime debug messages (without only warnings and errors)
-#OPTIMISE = 1 # Add umpteen optimisation flags
+OPTIMISE = 1 # Add umpteen optimisation flags
 #PROFILE = 1  # Enable profiling
 #PROFILE = 2  # Enable profiling with optimisation flags, can be used with `OPTIMISE = 1'
 
 #STATIC = 1   # Enable static linkage, currently mingw only
 #AV_FOUNDATION = 1  # Use AVFoundation instead of QTKit. If you are using macOS 10.12 or later, this must be enabled.
 
-#WITH_REVISION = 1 # adds the revision from svn; required for networkgames
+WITH_REVISION = 1 # adds the revision from svn; required for networkgames
 # if you do not use SVN, add -DREVISION="1234" to the FLAGS below
 
 #WIN32_CONSOLE = 1 # adds a console for windows debugging
 
-#MULTI_THREAD = 1 # Enable multithreading
+MULTI_THREAD = 1 # Enable multithreading
 
 # using freetype for Truetype font support
-USE_FREETYPE = 0
+USE_FREETYPE = 1
 
 # using UPnP for easy server hosting behind routers
-#USE_UPNP = 0
+USE_UPNP = 1
 
 # using zstd compression library
-#USE_ZSTD = 0
+USE_ZSTD = 1
 
 # Define these as empty strings, if you don't have the respective config program
 #ALLEGRO_CONFIG = allegro-config
 #PNG_CONFIG     = pkg-config libpng
 #SDL_CONFIG     = sdl-config
 #SDL2_CONFIG    = sdl2-config
-#FREETYPE_CONFIG    = freetype-config
+FREETYPE_CONFIG    = pkg-config freetype2
 
 #VERBOSE = 1
 
