diff --git a/redbot/core/data_manager.py b/redbot/core/data_manager.py
index 1472d9dc..5fa6274e 100644
--- a/redbot/core/data_manager.py
+++ b/redbot/core/data_manager.py
@@ -40,7 +40,7 @@ config_dir = None
 appdir = appdirs.AppDirs("Red-DiscordBot")
 if sys.platform == "linux":
     if 0 < os.getuid() < 1000:  # pylint: disable=no-member
-        config_dir = Path(appdir.site_data_dir)
+        config_dir = Path(appdir.site_config_dir)
 if not config_dir:
     config_dir = Path(appdir.user_config_dir)
 config_file = config_dir / "config.json"
diff --git a/redbot/setup.py b/redbot/setup.py
index 20e9ba2a..f460c034 100644
--- a/redbot/setup.py
+++ b/redbot/setup.py
@@ -3,6 +3,7 @@ import asyncio
 import json
 import logging
 import os
+import pwd
 import sys
 import re
 from copy import deepcopy
@@ -23,7 +24,7 @@ config_dir = None
 appdir = appdirs.AppDirs("Red-DiscordBot")
 if sys.platform == "linux":
     if 0 < os.getuid() < 1000:  # pylint: disable=no-member  # Non-exist on win
-        config_dir = Path(appdir.site_data_dir)
+        config_dir = Path(appdir.site_config_dir)
 if not config_dir:
     config_dir = Path(appdir.user_config_dir)
 try:
@@ -61,7 +62,13 @@ def save_config(name, data, remove=False):
 
 
 def get_data_dir(instance_name: str):
-    data_path = Path(appdir.user_data_dir) / "data" / instance_name
+    # Patch for red-discordbot-git AUR package -xiretza
+    if pwd.getpwuid(os.getuid()).pw_name == "redbot":
+        data_basedir = Path("/var/lib/Red-DiscordBot")
+    else:
+        data_basedir = Path(appdir.user_data_dir) / "data"
+
+    data_path = data_basedir / instance_name
 
     print()
     print(
