diff --git a/configure.ac b/configure.ac
index 67cb476..d97e9c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -97,6 +97,14 @@ done
 dnl And for gtk-doc
 gtk_CHECK_GTK_DOC
 
+LIBENCA_LDFLAGS=""
+case "$host_os" in
+  mingw*)
+    LIBENCA_LDFLAGS="-no-undefined ${wl}-no-undefined ${wl}-export-all-symbols" ;;
+esac
+AC_MSG_RESULT([$LIBENCA_LDFLAGS])
+AC_SUBST([LIBENCA_LDFLAGS])
+
 dnl Checks for libraries.
 ye_CHECK_LIBM
 
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 9771adb..bbbe344 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -6,7 +6,7 @@ lib_LTLIBRARIES = libenca.la
 # Any iface change    C++: 0:   A
 # Adding ifaces       C:   R:   A++
 # Changing ifaces     C:   R:   0
-libenca_la_LDFLAGS = -version-info 5:1:5
+libenca_la_LDFLAGS = @LIBENCA_LDFLAGS@ -version-info 5:1:5
 libenca_la_SOURCES = \
   common.c \
   ctype.c \
