diff --git a/configuration/configuration.go b/configuration/configuration.go
index ac3947c..d65da20 100644
--- a/configuration/configuration.go
+++ b/configuration/configuration.go
@@ -40,8 +40,8 @@ func NewHideGuardConfiguration() *HideGuardConfiguration {
 			Host:           		"",											// command line option "-n"
 			Port:					432,										// command line option "-p"
 			Domain:					"hide.me",									// Not configurable
-			CA:						"CA.pem",									// command line option "-ca"
-			AccessTokenFile:		"accessToken.txt",							// command line option "-t"
+			CA:						"/usr/share/hide.me/CA.pem",									// command line option "-ca"
+			AccessTokenFile:		"/etc/hide.me/accessToken.txt",							// command line option "-t"
 			Username:       		"",											// command line option "-u"
 			Password:				"",											// Only configurable through the config file
 			ConnectTimeout: 		10 * time.Second,							// Only configurable through the config file
@@ -120,4 +120,4 @@ func ( c *HideGuardConfiguration ) InteractiveCredentials() ( err error ) {
 func ( c *HideGuardConfiguration ) Print() {
 	if out, err := yaml.Marshal( c ); err != nil { fmt.Println( err ) } else { fmt.Print( string( out ) ) }
 	return
-}
\ No newline at end of file
+}
