*** configurableMenus.js.bak	2018-12-19 16:25:08.000000000 +0200
--- configurableMenus.js	2018-12-19 16:41:25.000000000 +0200
***************
*** 714,720 ****
          let margin = (4 * borderRadius + borderWidth + arrowBase);
          let halfMargin = margin / 2;
  
!         let themeNode = this.actor.get_theme_node();
          let gap = themeNode.get_length('-boxpointer-gap');
  
          let resX, resY;
--- 714,720 ----
          let margin = (4 * borderRadius + borderWidth + arrowBase);
          let halfMargin = margin / 2;
  
!         //let themeNode = this.actor.get_theme_node();
          let gap = themeNode.get_length('-boxpointer-gap');
  
          let resX, resY;
***************
*** 5593,5599 ****
        //return portSize;
        let spaceBox = new Clutter.ActorBox();
        x = box.x1 + leftEmptySpace;
!       let spaceBox = this._calculateSpaceBox(this._spaceActor, x, y, box);
        this._spaceActor.allocate(spaceBox, flags);
        this.box.set_skip_paint(this._spaceActor, false);
        //Main.notify("" + spaceBox.y2 + "-" + spaceBox.y1 + "-" + spaceBox.x2 + "-" + spaceBox.x1);
--- 5593,5599 ----
        //return portSize;
        let spaceBox = new Clutter.ActorBox();
        x = box.x1 + leftEmptySpace;
!       spaceBox = this._calculateSpaceBox(this._spaceActor, x, y, box);
        this._spaceActor.allocate(spaceBox, flags);
        this.box.set_skip_paint(this._spaceActor, false);
        //Main.notify("" + spaceBox.y2 + "-" + spaceBox.y1 + "-" + spaceBox.x2 + "-" + spaceBox.x1);
***************
*** 6811,6817 ****
                PopupMenu using a non instance of the class PopupMenuAbstractFactory");
        }
        // The shell menu
!       let shellItem = this._createShellItem(factoryMenu, launcher, orientation, menuManager);
        this._attachToMenu(shellItem, factoryMenu);
        this._menuManager.push(menuManager);
        return shellItem;
--- 6811,6817 ----
                PopupMenu using a non instance of the class PopupMenuAbstractFactory");
        }
        // The shell menu
!       shellItem = this._createShellItem(factoryMenu, launcher, orientation, menuManager);
        this._attachToMenu(shellItem, factoryMenu);
        this._menuManager.push(menuManager);
        return shellItem;
