diff -ruN libbluray-0.8.0/configure.ac patched/configure.ac
--- libbluray-0.8.0/configure.ac	2015-04-10 08:48:23.000000000 +0200
+++ patched/configure.ac	2015-05-05 19:06:33.431632721 +0200
@@ -41,6 +41,7 @@
         AC_DEFINE([_WIN32_WINNT], 0x0502, [Define to '0x0502' for Windows XP SP2 APIs.])
         AC_DEFINE([_WIN32_IE], 0x0501, [Define to '0x0501' for IE 5.01.])
         CPPFLAGS="${CPPFLAGS} -D__USE_MINGW_ANSI_STDIO=1"
+        lt_cv_deplibs_check_method='pass_all'
         ;;
     esac
     ;;
