--- config.default.orig	2021-01-22 22:00:17.324949087 +0100
+++ config.default	2021-01-22 23:32:23.111701888 +0100
@@ -10,19 +10,19 @@
 
 #BACKEND = gdi
 #BACKEND = sdl
-#BACKEND = sdl2
+BACKEND = sdl2
 #BACKEND = mixer_sdl
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw32
 #OSTYPE = mingw64
 #OSTYPE = mac
@@ -44,10 +44,10 @@
 MULTI_THREAD = 1 # Enable multithreading
 
 # using freetype for Truetype font support
-USE_FREETYPE = 0
+USE_FREETYPE = 1
 
 # using UPnP for easy server hosting behind routers
-#USE_UPNP = 0
+USE_UPNP = 1
 
 # using zstd compression library
 USE_ZSTD = 1
@@ -57,6 +57,7 @@
 #SDL_CONFIG     = sdl-config
 #SDL2_CONFIG    = sdl2-config
 #FREETYPE_CONFIG = freetype-config
+FREETYPE_CONFIG    = pkg-config freetype2
 
 #VERBOSE = 1
 
