diff --git a/configure.py b/configure.py
index 76ab661..f8b9fb8 100644
--- a/configure.py
+++ b/configure.py
@@ -22,9 +22,9 @@ makefile.extra_lib_dirs=["/usr/lib", "/usr/lib/x86_64-linux-gnu/"]
 # Add the library we are wrapping.  The name doesn't include any platform
 # specific prefixes or extensions (e.g. the "lib" prefix on UNIX, or the
 # ".dll" extension on Windows).
-makefile.extra_libs = ["opencv_core", "opencv_highgui", "opencv_imgproc", 
-                        "opencv_legacy", "opencv_contrib", "opencv_features2d", 
-                        "opencv_ml", "opencv_gpu", "oqapy-iproc"]
+makefile.extra_libs = ["opencv_core", "opencv_highgui", "opencv_imgproc",
+                        "opencv_features2d",
+                        "opencv_ml", "oqapy-iproc"]
 
 # Generate the Makefile itself.
 makefile.generate()
