diff --git a/packages/core/src/IntifaceBackendManager.ts b/packages/core/src/IntifaceBackendManager.ts
index 35346b2..aa36775 100644
--- a/packages/core/src/IntifaceBackendManager.ts
+++ b/packages/core/src/IntifaceBackendManager.ts
@@ -9,7 +9,7 @@ import { IntifaceUtils } from "./Utils";
 import { IApplicationUpdater } from "./IApplicationUpdater";
 import { IntifaceBackendLogger } from "./IntifaceBackendLogger";
 import { EventEmitter } from "events";
-import isOnline from "is-online";
+import * as isOnline from "is-online";
 import { promisify } from "util";
 import * as winston from "winston";
 import * as rimraf from "rimraf";
