--- src/radium-7.0.00/crashreporter/backtrace-symbols.c	2022-12-19 11:03:23.579772161 +0100
+++ src/radium-7.0.00/crashreporter/backtrace-symbols.c	2022-12-19 11:04:24.236437340 +0100
@@ -114,10 +114,10 @@
 	if ((bfd_get_file_flags(abfd) & HAS_SYMS) == 0)
 		return;
 
-	symcount = bfd_read_minisymbols(abfd, false, (PTR) & syms, &size);
+	symcount = bfd_read_minisymbols(abfd, false, (void **) & syms, &size);
 	if (symcount == 0)
 		symcount = bfd_read_minisymbols(abfd, true /* dynamic */ ,
-						(PTR) & syms, &size);
+						(void **) & syms, &size);
 
 	if (symcount < 0)
 		bfd_fatal(bfd_get_filename(abfd));
@@ -184,7 +184,7 @@
 
 		found = false;
 		bfd_map_over_sections(abfd, find_address_in_section,
-		(PTR) NULL);
+		(void **) NULL);
 
 		if (!found) {
 			printf("[%s] \?\?() \?\?:0\n",addr[naddr-1]);
@@ -244,7 +244,7 @@
 
 		found = false;
 		bfd_map_over_sections(abfd, find_address_in_section,
-		(PTR) NULL);
+		(void **) NULL);
 
 		if (!found) {
 			total += snprintf(buf, len, "[0x%llx] \?\?() \?\?:0",(long long unsigned int) addr[naddr-1]) + 1;
