diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d84a02..72e309b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,13 +65,14 @@ if(OpenMVS_USE_OPENMP)
 	endif()
 endif()
 
-SET(CUDA_CUDA_LIBRARY "")
+#SET(CUDA_CUDA_LIBRARY "")
 if(OpenMVS_USE_CUDA)
 	FIND_PACKAGE(CUDA)
 	if(CUDA_FOUND)
 		INCLUDE_DIRECTORIES(${CUDA_INCLUDE_DIRS})
 		ADD_DEFINITIONS(-D_USE_CUDA)
 		SET(_USE_CUDA TRUE)
+		SET(CUDA_LIBRARIES "${CUDA_LIBRARIES};-lcuda")
 	else()
 		MESSAGE("-- Can't find CUDA. Continuing without it.")
 	endif()
diff --git a/libs/MVS/CMakeLists.txt b/libs/MVS/CMakeLists.txt
index 309ea32..0a80d05 100644
--- a/libs/MVS/CMakeLists.txt
+++ b/libs/MVS/CMakeLists.txt
@@ -37,7 +37,7 @@ cxx_library_with_type_no_pch(MVS "Libs" "" "${cxx_default}"
 set_target_pch(MVS Common.h)
 
 # Link its dependencies
-TARGET_LINK_LIBRARIES(MVS Common Math IO ${CERES_LIBS} ${CGAL_LIBS} ${CUDA_CUDA_LIBRARY})
+TARGET_LINK_LIBRARIES(MVS Common Math IO ${CERES_LIBS} ${CGAL_LIBS} ${CUDA_LIBRARIES})
 
 # Install
 SET_TARGET_PROPERTIES(MVS PROPERTIES
