diff -ruN cnijfilter-source-4.00-1.orig/backend/src/cnij_backend_common.c cnijfilter-source-4.00-1/backend/src/cnij_backend_common.c
--- cnijfilter-source-4.00-1.orig/backend/src/cnij_backend_common.c	2013-07-17 06:03:45.000000000 +0430
+++ cnijfilter-source-4.00-1/backend/src/cnij_backend_common.c	2015-08-04 12:29:43.528967207 +0430
@@ -38,6 +38,7 @@
 // CUPS Header
 #include <cups/cups.h>
 #include <cups/ipp.h>
+#include <cups/ppd.h>
 
 // Header file for CANON
 #include "cnij_backend_common.h"
diff -ruN cnijfilter-source-4.00-1.orig/cngpij/cngpij/bjcups.c cnijfilter-source-4.00-1/cngpij/cngpij/bjcups.c
--- cnijfilter-source-4.00-1.orig/cngpij/cngpij/bjcups.c	2013-07-17 06:03:45.000000000 +0430
+++ cnijfilter-source-4.00-1/cngpij/cngpij/bjcups.c	2015-08-04 12:26:36.445633660 +0430
@@ -21,6 +21,7 @@
 #include	<config.h>
 #endif	// HAVE_CONFIG_H
 
+#define _IPP_PRIVATE_STRUCTURES 1
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
diff -ruN cnijfilter-source-4.00-1.orig/cngpijmnt/src/main.c cnijfilter-source-4.00-1/cngpijmnt/src/main.c
--- cnijfilter-source-4.00-1.orig/cngpijmnt/src/main.c	2013-07-17 06:03:45.000000000 +0430
+++ cnijfilter-source-4.00-1/cngpijmnt/src/main.c	2015-08-04 12:28:27.062300453 +0430
@@ -21,6 +21,7 @@
 #include	<config.h>
 #endif	// HAVE_CONFIG_H
 
+#define _IPP_PRIVATE_STRUCTURES 1
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
