--- squashfs-root/cura.desktop
+++ squashfs-root/cura.desktop
@@ -4,7 +4,7 @@
 GenericName=3D Printing Software
 GenericName[de]=3D-Druck-Software
 Comment=Cura converts 3D models into paths for a 3D printer. It prepares your print for maximum accuracy, minimum printing time and good reliability with many extra features that make your print come out great.
-Exec=Ultimaker-Cura %F
+Exec=/usr/bin/cura %F
 Icon=cura-icon
 Terminal=false
 Type=Application
