--- a/CxImage/ximacfg.h	2016-12-29 17:26:16.227749167 +0100
+++ b/CxImage/ximacfg.h	2016-12-29 17:34:03.346609818 +0100
@@ -36,10 +36,10 @@
 
 #define CXIMAGE_SUPPORT_JBG 0		// GPL'd see ../jbig/copying.txt & ../jbig/patents.htm
 
-#define CXIMAGE_SUPPORT_MNG 1
+#define CXIMAGE_SUPPORT_MNG 0
 #define CXIMAGE_SUPPORT_SKA 1
-#define CXIMAGE_SUPPORT_RAW 1
-#define CXIMAGE_SUPPORT_PSD 1
+#define CXIMAGE_SUPPORT_RAW 0
+#define CXIMAGE_SUPPORT_PSD 0
 
 /////////////////////////////////////////////////////////////////////////////
 #define	CXIMAGE_MAX_MEMORY 268435456
