diff -urN line_profiler-3.3.1-orig/setup.py line_profiler-3.3.1/setup.py
--- line_profiler-3.3.1-orig/setup.py	2021-12-17 14:50:32.739504227 -0500
+++ line_profiler-3.3.1/setup.py	2021-12-17 14:50:41.046174297 -0500
@@ -215,13 +215,16 @@
 
 
 if __name__ == '__main__':
-    if '--universal' in sys.argv:
-        # Dont use scikit-build for universal wheels
-        # if 'develop' in sys.argv:
-        sys.argv.remove('--universal')
-        from setuptools import setup  # NOQA
-    else:
-        from skbuild import setup
+    from setuptools import setup  # NOQA
+    from setuptools.extension import Extension
+    from Cython.Build import cythonize
+    extension = Extension('line_profiler._line_profiler',
+                          sources=["line_profiler/_line_profiler.pyx",
+                                   'line_profiler/timers.c',
+                                   'line_profiler/unset_trace.c'],
+                          depends=['python25.pxd'],
+                          extra_compile_args=["-DHAVE_GETTIMEOFDAY"])
+    extension = cythonize([extension])
     setupkw = dict(
         name=NAME,
         version=VERSION,
@@ -251,6 +254,7 @@
             'Topic :: Software Development',
         ],
         # py_modules=find_packages(),
+        ext_modules=extension,
         packages=list(find_packages()),
         py_modules=['kernprof', 'line_profiler'],
         entry_points={
