From da02dfeac7f008cb7b30a5487add19401bd94b9e Mon Sep 17 00:00:00 2001
From: Corentin Le Molgat <corentinl@google.com>
Date: Tue, 16 Jan 2024 15:09:40 +0100
Subject: [PATCH] Fix pybind11_* commit sha1sum (Fix #4027)

---
 WORKSPACE                         | 6 ++----
 cmake/dependencies/CMakeLists.txt | 4 ++--
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/WORKSPACE b/WORKSPACE
index 86377227f8..e361770963 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -219,7 +219,7 @@ install_notebook_deps()
 
 git_repository(
     name = "pybind11_bazel",
-    commit = "fc56ce8a8b51e3dd941139d329b63ccfea1d304b",
+    commit = "23926b00e2b2eb2fc46b17e587cf0c0cfd2f2c4b", # 2023/11/29
     patches = ["//patches:pybind11_bazel.patch"],
     patch_args = ["-p1"],
     remote = "https://github.com/pybind/pybind11_bazel.git",
@@ -234,9 +234,7 @@ new_git_repository(
 
 new_git_repository(
     name = "pybind11_protobuf",
-    #build_file = "@pybind11_bazel//:pybind11.BUILD",
-    #tag = "v2.10.3",
-    commit = "5baa2dc9d93e3b608cde86dfa4b8c63aeab4ac78",
+    commit = "b713501f1da56d9b76c42f89efd00b97c26c9eac", # 2023/11/29
     remote = "https://github.com/pybind/pybind11_protobuf.git",
 )
 
diff --git a/cmake/dependencies/CMakeLists.txt b/cmake/dependencies/CMakeLists.txt
index af3a28c113..f6ec06b2d0 100644
--- a/cmake/dependencies/CMakeLists.txt
+++ b/cmake/dependencies/CMakeLists.txt
@@ -164,7 +164,7 @@ if(BUILD_PYTHON AND BUILD_pybind11)
   FetchContent_Declare(
     pybind11
     GIT_REPOSITORY "https://github.com/pybind/pybind11.git"
-    GIT_TAG "v2.10.3"
+    GIT_TAG "v2.11.1"
     PATCH_COMMAND git apply --ignore-whitespace "${CMAKE_CURRENT_LIST_DIR}/../../patches/pybind11.patch"
   )
   FetchContent_MakeAvailable(pybind11)
@@ -178,7 +178,7 @@ if(BUILD_PYTHON AND BUILD_pybind11_protobuf)
   FetchContent_Declare(
     pybind11_protobuf
     GIT_REPOSITORY "https://github.com/pybind/pybind11_protobuf.git"
-    GIT_TAG "main"
+    GIT_TAG "b713501f1da56d9b76c42f89efd00b97c26c9eac" # 2023/11/29
     PATCH_COMMAND git apply --ignore-whitespace "${CMAKE_CURRENT_LIST_DIR}/../../patches/pybind11_protobuf.patch"
   )
   FetchContent_MakeAvailable(pybind11_protobuf)
