diff -ur a/data/Makefile.am b/data/Makefile.am
--- a/data/Makefile.am	2017-03-14 15:53:25.000000000 +0300
+++ b/data/Makefile.am	2017-04-23 16:36:55.033679497 +0300
@@ -6,7 +6,7 @@
 dbusservice_DATA     = $(dbusservice_in_files:.service.in=.service)
 
 $(dbusservice_DATA): $(dbusservice_in_files) Makefile
-	@sed -e "s|\@udisksprivdir\@|$(libexecdir)/udisks2|" $< > $@
+	@sed -e "s|\@udisksdprivdir\@|$(libexecdir)/udisks2|" $< > $@
 
 dbusconfdir = $(sysconfdir)/dbus-1/system.d
 dbusconf_in_files = org.freedesktop.UDisks2.conf.in
