diff -u -r -N ddb_discord_presence/Makefile ddb_discord_presence-makefile/Makefile
--- ddb_discord_presence/Makefile	2020-01-15 09:47:59.792849482 +0100
+++ ddb_discord_presence-makefile/Makefile	2020-01-15 09:49:29.572231291 +0100
@@ -14,8 +14,8 @@
 CC=gcc
 CXX=g++
 STD=gnu99
-CFLAGS+=-fPIC -I /usr/local/include -I discord-rpc/include -I . -Wall
-CXXFLAGS+=-fPIC -I /usr/local/include -I . -Wall
+CFLAGS+=$(COPT) -fPIC -I /usr/local/include -I discord-rpc/include -I . -Wall
+CXXFLAGS+=$(CXXOPT) -fPIC -I /usr/local/include -I . -Wall
 ifeq ($(UNAME_S),Darwin)
     CFLAGS+=-I $(DEADBEEF_OSX)/Contents/Headers
     CXXFLAGS+=-I $(DEADBEEF_OSX)/Contents/Headers
