diff --git a/Makefile b/Makefile
index 8f067fc..5037306 100644
--- a/Makefile
+++ b/Makefile
@@ -194,7 +194,7 @@ log :
 
 # The Debian version gets derived from the package version $(version) as well
 # as the date and serial number of the last commit.
-debversion = $(version)+git$(shell git log -1 --format=%cd --date=short | sed -e 's/-//g')+$(shell git rev-list --count HEAD)
+debversion = $(version)+git$(shell git log -1 --format=%cd --date=short 2>/dev/null | sed -e 's/-//g')+$(shell git rev-list --count HEAD 2>/dev/null)
 # Debian revision number of the package.
 debrevision = 1
 # Source tarball and folder.
