diff -Naur buildscripts/build-devkit.sh buildscripts.no-libs/build-devkit.sh
--- buildscripts/build-devkit.sh	2015-02-16 01:53:37.000000000 +0100
+++ buildscripts.no-libs/build-devkit.sh	2015-02-22 11:19:40.545502408 +0100
@@ -226,7 +226,7 @@
 fi
 
 cd "$SRCDIR"
-for archive in $archives $targetarchives $hostarchives
+for archive in $archives $hostarchives
 do
 	echo $archive
 	if [ ! -f $archive ]; then
@@ -244,16 +244,6 @@
 extract_and_patch newlib $NEWLIB_VER gz
 extract_and_patch gdb $GDB_VER bz2
 
-for archive in $targetarchives
-do
-	destdir=$(echo $archive | sed -e 's/\(.*\)-src-\(.*\)\.tar\.bz2/\1-\2/' )
-	echo $destdir
-	if [ ! -d $destdir ]; then
-		mkdir -p $destdir
-		bzip2 -cd "$SRCDIR/$archive" | tar -xf - -C $destdir || { echo "Error extracting "$archive; exit 1; }
-	fi
-done
-
 for archive in $hostarchives
 do
 	destdir=$(echo $archive | sed -e 's/\(.*\)-src-\(.*\)\.tar\.bz2/\1-\2/' )
diff -Naur buildscripts/dkarm-eabi/scripts/build-crtls.sh buildscripts.no-libs/dkarm-eabi/scripts/build-crtls.sh
--- buildscripts/dkarm-eabi/scripts/build-crtls.sh	2015-02-15 17:25:19.000000000 +0100
+++ buildscripts.no-libs/dkarm-eabi/scripts/build-crtls.sh	2015-02-22 11:21:07.943233856 +0100
@@ -22,40 +22,3 @@
 $MAKE CRT=ds_arm9
 $MAKE CRT=ds_cart
 $target-gcc -march=armv6k -mfloat-abi=hard -c 3dsx_crt0.s -o armv6k/fpu/3dsx_crt0.o
-
-cd $BUILDDIR/libgba-$LIBGBA_VER
-$MAKE || { echo "error building libgba"; exit 1; }
-$MAKE install || { echo "error installing libgba"; exit 1; }
-
-cd $BUILDDIR/libnds-$LIBNDS_VER
-$MAKE || { echo "error building libnds"; exit 1; }
-$MAKE install || { echo "error installing libnds"; exit 1; }
-
-cd $BUILDDIR/dswifi-$DSWIFI_VER
-$MAKE || { echo "error building dswifi"; exit 1; }
-$MAKE install || { echo "error installing dswifi"; exit 1; }
-
-cd $BUILDDIR/maxmod-$MAXMOD_VER
-$MAKE || { echo "error building maxmod"; exit 1; }
-$MAKE install || { echo "error installing maxmod"; exit 1; }
-
-cd $BUILDDIR/default_arm7-$DEFAULT_ARM7_VER
-$MAKE || { echo "error building default arm7"; exit 1; }
-$MAKE install || { echo "error installing default arm7"; exit 1; }
-
-cd $BUILDDIR/libfat-$LIBFAT_VER
-$MAKE nds-install || { echo "error building nds libfat"; exit 1; }
-$MAKE gba-install || { echo "error installing gba libfat"; exit 1; }
-
-cd $BUILDDIR/libmirko-$LIBMIRKO_VER
-$MAKE || { echo "error building libmirko"; exit 1; }
-$MAKE install || { echo "error installing libmirko"; exit 1; }
-
-cd $BUILDDIR/libfilesystem-$FILESYSTEM_VER
-$MAKE || { echo "error building libfilesystem"; exit 1; }
-$MAKE install || { echo "error installing libfilesystem"; exit 1; }
-
-cd $BUILDDIR/libctru-$LIBCTRU_VER
-$MAKE || { echo "error building libctru"; exit 1; }
-$MAKE install || { echo "error installing libctru"; exit 1; }
-
