diff -Naur buildscripts.old/build-devkit.sh buildscripts/build-devkit.sh
--- buildscripts.old/build-devkit.sh	2014-04-02 23:16:25.000000000 +0200
+++ buildscripts/build-devkit.sh	2014-08-19 02:43:59.867864270 +0200
@@ -218,7 +218,7 @@
 fi
 
 cd "$SRCDIR"
-for archive in $archives $targetarchives $hostarchives
+for archive in $archives $hostarchives
 do
 	echo $archive
 	if [ ! -f $archive ]; then
@@ -236,15 +236,6 @@
 extract_and_patch newlib $NEWLIB_VER gz
 extract_and_patch gdb $GDB_VER bz2
 
-for archive in $targetarchives
-do
-	destdir=$(echo $archive | sed -e 's/\(.*\)-src-\(.*\)\.tar\.bz2/\1-\2/' )
-	echo $destdir
-	if [ ! -d $destdir ]; then
-		mkdir -p $destdir
-		bzip2 -cd "$SRCDIR/$archive" | tar -xf - -C $destdir || { echo "Error extracting "$archive; exit 1; }
-	fi
-done
 
 for archive in $hostarchives
 do
diff -Naur buildscripts.old/dkarm-eabi/scripts/build-crtls.sh buildscripts/dkarm-eabi/scripts/build-crtls.sh
--- buildscripts.old/dkarm-eabi/scripts/build-crtls.sh	2013-01-05 14:10:23.000000000 +0100
+++ buildscripts/dkarm-eabi/scripts/build-crtls.sh	2014-08-19 02:43:51.807763871 +0200
@@ -22,34 +22,3 @@
 $MAKE CRT=ds_arm9
 $MAKE CRT=ds_cart
 
-cd $BUILDDIR/libgba-$LIBGBA_VER
-$MAKE || { echo "error building libgba"; exit 1; }
-$MAKE install || { echo "error installing libgba"; exit 1; }
-
-cd $BUILDDIR/libnds-$LIBNDS_VER
-$MAKE || { echo "error building libnds"; exit 1; }
-$MAKE install || { echo "error installing libnds"; exit 1; }
-
-cd $BUILDDIR/dswifi-$DSWIFI_VER
-$MAKE || { echo "error building dswifi"; exit 1; }
-$MAKE install || { echo "error installing dswifi"; exit 1; }
-
-cd $BUILDDIR/maxmod-$MAXMOD_VER
-$MAKE || { echo "error building maxmod"; exit 1; }
-$MAKE install || { echo "error installing maxmod"; exit 1; }
-
-cd $BUILDDIR/default_arm7-$DEFAULT_ARM7_VER
-$MAKE || { echo "error building default arm7"; exit 1; }
-$MAKE install || { echo "error installing default arm7"; exit 1; }
-
-cd $BUILDDIR/libfat-$LIBFAT_VER
-$MAKE nds-install || { echo "error building nds libfat"; exit 1; }
-$MAKE gba-install || { echo "error installing gba libfat"; exit 1; }
-
-cd $BUILDDIR/libmirko-$LIBMIRKO_VER
-$MAKE || { echo "error building libmirko"; exit 1; }
-$MAKE install || { echo "error installing libmirko"; exit 1; }
-
-cd $BUILDDIR/libfilesystem-$FILESYSTEM_VER
-$MAKE || { echo "error building libfilesystem"; exit 1; }
-$MAKE install || { echo "error installing libfilesystem"; exit 1; }
