diff -Naur buildscripts.old/dkarm-eabi/scripts/build-tools.sh buildscripts/dkarm-eabi/scripts/build-tools.sh
--- buildscripts.old/dkarm-eabi/scripts/build-tools.sh	2016-11-27 21:58:43.449895511 +0100
+++ buildscripts/dkarm-eabi/scripts/build-tools.sh	2016-11-27 23:43:07.699791078 +0100
@@ -1,6 +1,9 @@
 #!/bin/bash
 cd $BUILDDIR
 
+patch -Np1 -d grit-0.8.13 < ../../fix-grit.patch
+patch -Np0 < ../../fix-gp32tools.patch
+
 for archive in $hostarchives
 do
 	dir=$(echo $archive | sed -e 's/\(.*\)\.tar\.bz2/\1/' )
