--- didjvu-0.8/setup.py.old	2016-03-04 07:32:53.263736257 +0100
+++ didjvu-0.8/setup.py	2016-03-04 07:36:43.670778498 +0100
@@ -156,8 +156,8 @@
     url='http://jwilk.net/software/didjvu',
     author='Jakub Wilk',
     author_email='jwilk@jwilk.net',
-    packages=['didjvu'],
-    package_dir=dict(didjvu='lib'),
+    packages=['didjvu','didjvu.xmp'],
+    package_dir={'didjvu':'lib'},
     scripts=['didjvu'],
     data_files=[('share/man/man1', glob.glob('doc/*.1'))],
     cmdclass=dict(
