diff --git a/browser/ethereum_remote_client/buildflags/buildflags.gni b/browser/ethereum_remote_client/buildflags/buildflags.gni
index 253a07e..a85a97d 100644
--- a/browser/ethereum_remote_client/buildflags/buildflags.gni
+++ b/browser/ethereum_remote_client/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  ethereum_remote_client_enabled = !is_ios && !is_android
+  ethereum_remote_client_enabled = false
 }
diff --git a/components/binance/browser/buildflags/buildflags.gni b/components/binance/browser/buildflags/buildflags.gni
index d4bc81a..7b26be8 100644
--- a/components/binance/browser/buildflags/buildflags.gni
+++ b/components/binance/browser/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  binance_enabled = is_mac || is_linux || is_win || is_android
+  binance_enabled = false
 }
diff --git a/components/brave_ads/browser/buildflags/buildflags.gni b/components/brave_ads/browser/buildflags/buildflags.gni
index 763a7d5..44a73d7 100644
--- a/components/brave_ads/browser/buildflags/buildflags.gni
+++ b/components/brave_ads/browser/buildflags/buildflags.gni
@@ -1,6 +1,5 @@
 import("//brave/components/brave_rewards/browser/buildflags/buildflags.gni")
 
 declare_args() {
-  brave_ads_enabled =
-      brave_rewards_enabled && (is_mac || is_linux || is_win || is_android)
+  brave_ads_enabled = true
 }
diff --git a/components/brave_referrals/buildflags/buildflags.gni b/components/brave_referrals/buildflags/buildflags.gni
index 3848636..a500258 100755
--- a/components/brave_referrals/buildflags/buildflags.gni
+++ b/components/brave_referrals/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  enable_brave_referrals = is_mac || is_linux || is_win || is_android
+  enable_brave_referrals = false
 }
diff --git a/components/brave_rewards/browser/buildflags/buildflags.gni b/components/brave_rewards/browser/buildflags/buildflags.gni
index 18ad294..1a031ad 100644
--- a/components/brave_rewards/browser/buildflags/buildflags.gni
+++ b/components/brave_rewards/browser/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  brave_rewards_enabled = is_mac || is_linux || is_win || is_android
+  brave_rewards_enabled = false
 }
diff --git a/components/brave_sync/buildflags/buildflags.gni b/components/brave_sync/buildflags/buildflags.gni
index debaffc..aa59fa8 100755
--- a/components/brave_sync/buildflags/buildflags.gni
+++ b/components/brave_sync/buildflags/buildflags.gni
@@ -1,5 +1,5 @@
 import("//brave/build/config.gni")
 
 declare_args() {
-  enable_brave_sync = true
+  enable_brave_sync = false
 }
diff --git a/components/brave_vpn/buildflags/buildflags.gni b/components/brave_vpn/buildflags/buildflags.gni
index e8cbfc9..cdf09ce 100644
--- a/components/brave_vpn/buildflags/buildflags.gni
+++ b/components/brave_vpn/buildflags/buildflags.gni
@@ -5,5 +5,5 @@
 
 declare_args() {
   # On macOS, vpn is not available w/o signing.
-  enable_brave_vpn = is_win || is_android || (is_mac && is_official_build)
+  enable_brave_vpn = false
 }
diff --git a/components/brave_wallet/common/buildflags/buildflags.gni b/components/brave_wallet/common/buildflags/buildflags.gni
index 213b51c..a38ea6b 100644
--- a/components/brave_wallet/common/buildflags/buildflags.gni
+++ b/components/brave_wallet/common/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  brave_wallet_enabled = true
+  brave_wallet_enabled = false
 }
diff --git a/components/crypto_dot_com/browser/buildflags/buildflags.gni b/components/crypto_dot_com/browser/buildflags/buildflags.gni
index e0ebbbf..b23ccdb 100644
--- a/components/crypto_dot_com/browser/buildflags/buildflags.gni
+++ b/components/crypto_dot_com/browser/buildflags/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  crypto_dot_com_enabled = is_mac || is_linux || is_win
+  crypto_dot_com_enabled = false
 }
diff --git a/components/p3a/buildflags.gni b/components/p3a/buildflags.gni
index b8fca0c..d606873 100644
--- a/components/p3a/buildflags.gni
+++ b/components/p3a/buildflags.gni
@@ -1,3 +1,3 @@
 declare_args() {
-  brave_p3a_enabled = true
+  brave_p3a_enabled = false
 }
