diff --git a/ProcessList.c b/ProcessList.c
--- a/ProcessList.c	(revision 7a7c6938aad999f33ceb1e4ec5ebaada4c461f29)
+++ b/ProcessList.c	(date 1681664316383)
@@ -132,6 +132,8 @@
 }

 void ProcessList_printHeader(const ProcessList* this, RichString* header) {
+   return;
+
    RichString_rewind(header, RichString_size(header));

    const Settings* settings = this->settings;
@@ -451,6 +453,7 @@
 }

 void ProcessList_scan(ProcessList* this, bool pauseProcessUpdate) {
+   pauseProcessUpdate = true;
    // in pause mode only gather global data for meters (CPU/memory/...)
    if (pauseProcessUpdate) {
       ProcessList_goThroughEntries(this, true);
