diff --no-dereference -urpN a/api/cpp/CMakeLists.txt b/api/cpp/CMakeLists.txt
--- a/api/cpp/CMakeLists.txt	2025-05-04 06:43:38.074573213 +0200
+++ b/api/cpp/CMakeLists.txt	2025-05-04 06:45:28.063219022 +0200
@@ -151,10 +151,6 @@ if(SLINT_FEATURE_COMPILER AND NOT SLINT_
     if (SLINT_FEATURE_SDF_FONTS)
         list(APPEND slint_compiler_features "sdf-fonts")
     endif()
-    # Enable jemalloc, except when cross-compiling. (#7463)
-    if(NOT CMAKE_CROSSCOMPILING)
-        list(APPEND slint_compiler_features "jemalloc")
-    endif()
     set_property(
         TARGET slint-compiler
         PROPERTY CORROSION_FEATURES
