--- a/services/settings/Utils.jsm
+++ b/services/settings/Utils.jsm
@@ -59,9 +59,7 @@
       Ci.nsIEnvironment
     );
     const isXpcshell = env.exists("XPCSHELL_TEST_PROFILE_DIR");
-    return AppConstants.RELEASE_OR_BETA && !Cu.isInAutomation && !isXpcshell
-      ? "https://firefox.settings.services.mozilla.com/v1"
-      : gServerURL;
+    return gServerURL;
   },
 
   CHANGES_PATH: "/buckets/monitor/collections/changes/records",
