diff --git a/src/routes/__root.tsx b/src/routes/__root.tsx
index 3013454..2c6fa94 100644
--- a/src/routes/__root.tsx
+++ b/src/routes/__root.tsx
@@ -117,13 +117,7 @@ function RootLayout() {
   const checkForUpdates = useCallback(async () => {
     const update = await check();
     if (update) {
-      const yes = await ask("Do you want to install them now?", {
-        title: "New version available",
-      });
-      if (yes) {
-        await update.downloadAndInstall();
-        await relaunch();
-      }
+      await message("New version available. Check the AUR for updates.");
     } else {
       await message("No updates available");
     }
