diff --git a/src/Tests/Makefile.am b/src/Tests/Makefile.am
index 4543e32..b01d616 100644
--- a/src/Tests/Makefile.am
+++ b/src/Tests/Makefile.am
@@ -57,11 +57,7 @@ TESTS=\
 	test-regression \
 	USB/test-descriptor-parser.sh \
 	Packaging/spell-check.sh \
-	Rules/test-rules.sh \
-	UseCase/000_executable.sh \
-	UseCase/001_cli_policy.sh \
-	UseCase/002_cli_devices.sh \
-	UseCase/003_cli_devices_dummy.sh
+	Rules/test-rules.sh
 
 check_PROGRAMS=\
 	test-unit \
