--- a/Makefile	2023-03-11 21:10:36.362000000 +0100
+++ b/Makefile	2023-03-11 21:11:29.114000000 +0100
@@ -109,8 +109,7 @@
 PROGRAMS += test-show-type
 PROGRAMS += test-unssa
 
-INST_PROGRAMS = smatch sparse cgcc
-INST_MAN1 = sparse.1 cgcc.1
+INST_PROGRAMS = smatch
 INST_ASSETS = $(wildcard smatch_data/db/*.schema)
 INST_ASSETS += $(wildcard smatch_data/*)
 INST_ASSETS += $(wildcard smatch_data/db/kernel.*)
@@ -133,112 +133,6 @@
 smatch_datadir := $(DESTDIR)$(SHAREDIR)/smatch
 
 ########################################################################
-# target specificities
-
-compile: compile-i386.o
-EXTRA_OBJS += compile-i386.o
-
-# Can we use GCC's generated dependencies?
-HAVE_GCC_DEP:=$(shell touch .gcc-test.c && 				\
-		$(CC) -c -Wp,-MP,-MMD,.gcc-test.d .gcc-test.c 2>/dev/null && \
-		echo 'yes'; rm -f .gcc-test.d .gcc-test.o .gcc-test.c)
-ifeq ($(HAVE_GCC_DEP),yes)
-cflags += -Wp,-MP,-MMD,$(@D)/.$(@F).d
-endif
-
-# Can we use libxml (needed for c2xml)?
-HAVE_LIBXML:=$(shell $(PKG_CONFIG) --exists libxml-2.0 2>/dev/null && echo 'yes')
-ifeq ($(HAVE_LIBXML),yes)
-PROGRAMS+=c2xml
-INST_PROGRAMS+=c2xml
-c2xml-ldlibs := $(shell $(PKG_CONFIG) --libs libxml-2.0)
-c2xml-cflags := $(shell $(PKG_CONFIG) --cflags libxml-2.0)
-else
-$(warning Your system does not have libxml, disabling c2xml)
-endif
-
-HAVE_SQLITE := $(shell $(PKG_CONFIG) --exists sqlite3 2>/dev/null && echo 'yes')
-ifeq ($(HAVE_SQLITE),yes)
-SQLITE_VERSION:=$(shell $(PKG_CONFIG) --modversion sqlite3)
-SQLITE_VNUMBER:=$(shell printf '%d%02d%02d' $(subst ., ,$(SQLITE_VERSION)))
-ifeq ($(shell expr "$(SQLITE_VNUMBER)" '>=' 32400),1)
-PROGRAMS += semind
-INST_PROGRAMS += semind
-INST_MAN1 += semind.1
-semind-ldlibs := $(shell $(PKG_CONFIG) --libs sqlite3)
-semind-cflags := $(shell $(PKG_CONFIG) --cflags sqlite3)
-semind-cflags += -std=gnu99
-else
-$(warning Your SQLite3 version ($(SQLITE_VERSION)) is too old, 3.24.0 or later is required.)
-endif
-else
-$(warning Your system does not have sqlite3, disabling semind)
-endif
-
-# Can we use gtk (needed for test-inspect)
-GTK_VERSION:=3.0
-HAVE_GTK:=$(shell $(PKG_CONFIG) --exists gtk+-$(GTK_VERSION) 2>/dev/null && echo 'yes')
-ifneq ($(HAVE_GTK),yes)
-GTK_VERSION:=2.0
-HAVE_GTK:=$(shell $(PKG_CONFIG) --exists gtk+-$(GTK_VERSION) 2>/dev/null && echo 'yes')
-endif
-ifeq ($(HAVE_GTK),yes)
-GTK_CFLAGS := $(shell $(PKG_CONFIG) --cflags gtk+-$(GTK_VERSION))
-ast-view-cflags := $(GTK_CFLAGS)
-ast-model-cflags := $(GTK_CFLAGS)
-ast-inspect-cflags := $(GTK_CFLAGS)
-test-inspect-cflags := $(GTK_CFLAGS)
-test-inspect-ldlibs := $(shell $(PKG_CONFIG) --libs gtk+-$(GTK_VERSION))
-test-inspect: ast-model.o ast-view.o ast-inspect.o
-EXTRA_OBJS += ast-model.o ast-view.o ast-inspect.o
-PROGRAMS += test-inspect
-INST_PROGRAMS += test-inspect
-else
-$(warning Your system does not have gtk3/gtk2, disabling test-inspect)
-endif
-
-# Can we use LLVM (needed for ... sparse-llvm)?
-LLVM_CONFIG:=llvm-config
-HAVE_LLVM:=$(shell $(LLVM_CONFIG) --version >/dev/null 2>&1 && echo 'yes')
-ifeq ($(HAVE_LLVM),yes)
-arch := $(shell uname -m)
-ifeq (${MULTIARCH_TRIPLET},x86_64-linux-gnux32)
-arch := x32
-endif
-ifneq ($(filter ${arch},i386 i486 i586 i686 x86_64 amd64),)
-LLVM_VERSION:=$(shell $(LLVM_CONFIG) --version)
-LLVM_VERSION_MAJOR:=$(firstword $(subst ., ,$(LLVM_VERSION)))
-ifeq ($(shell expr "$(LLVM_VERSION_MAJOR)" '>=' 3),1)
-LLVM_PROGS := sparse-llvm
-$(LLVM_PROGS): LD := $(CXX)
-LLVM_LDFLAGS := $(shell $(LLVM_CONFIG) --ldflags)
-LLVM_CFLAGS := $(shell $(LLVM_CONFIG) --cppflags)
-LLVM_LIBS := $(shell $(LLVM_CONFIG) --libs)
-LLVM_LIBS += $(shell $(LLVM_CONFIG) --system-libs 2>/dev/null)
-LLVM_LIBS += $(shell $(LLVM_CONFIG) --cxxflags | grep -F -q -e '-stdlib=libc++' && echo -lc++)
-PROGRAMS += $(LLVM_PROGS)
-INST_PROGRAMS += sparse-llvm sparsec
-sparse-llvm-cflags := $(LLVM_CFLAGS) -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS
-sparse-llvm-ldflags := $(LLVM_LDFLAGS)
-sparse-llvm-ldlibs := $(LLVM_LIBS)
-else
-$(warning LLVM 3.0 or later required. Your system has version $(LLVM_VERSION) installed.)
-endif
-else
-$(warning sparse-llvm disabled on ${arch})
-endif
-else
-$(warning Your system does not have llvm, disabling sparse-llvm)
-endif
-
-ifeq ($(HAVE_BOOLECTOR),yes)
-PROGRAMS += scheck
-scheck-cflags  := -I${BOOLECTORDIR}/include/boolector
-scheck-ldflags := -L${BOOLECTORDIR}/lib
-scheck-ldlibs  := -lboolector -llgl -lbtor2parser
-endif
-
-########################################################################
 LIBS := libsparse.a
 OBJS := $(LIB_OBJS) $(EXTRA_OBJS) $(PROGRAMS:%=%.o)
 
@@ -390,7 +389,7 @@
 	smatch_constants.h avl.h
 
 ########################################################################
-all: $(PROGRAMS) smatch
+all: smatch
 
 ldflags += $($(@)-ldflags) $(LDFLAGS)
 ldlibs  += $($(@)-ldlibs)  $(LDLIBS) -lm

