--- a/man/manpage.xsl
+++ b/man/manpage.xsl
@@ -1,6 +1,6 @@
 <!-- Set parameters for manpage xsl -->
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-	<xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl"/>
+	<xsl:import href="/usr/share/xml/docbook/xsl-stylesheets-1.79.1/manpages/docbook.xsl"/>
 	<!-- Don't display notes list of link urls. -->
 	<xsl:param name="man.endnotes.list.enabled">0</xsl:param>
 	<xsl:param name="man.endnotes.are.numbered">0</xsl:param>
