diff -Naur gnuplot.orig/docs/doc2texi.el gnuplot/docs/doc2texi.el
--- gnuplot.orig/docs/doc2texi.el	2017-05-18 18:31:58.000000000 +0200
+++ gnuplot/docs/doc2texi.el	2017-05-19 13:54:53.368051644 +0200
@@ -760,7 +760,7 @@
     (while (re-search-forward "^#end" (point-max) "to_end")
       (replace-match "@end itemize" t))
     (goto-char (point-min))
-    (while (re-search-forward "^#b" (point-max) "to_end")
+    (while (re-search-forward "^#b " (point-max) "to_end")
       (replace-match "@item\n" t))
     (goto-char (point-min))
     (while (re-search-forward "^##" (point-max) "to_end")
