diff --git a/src/main.rs b/src/main.rs
index 868c8d1..4dc05f4 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -198,7 +198,7 @@ fn run<P: Protocol>(config: Config) {
             daemonize = daemonize.group(&group as &str);
         }
         if let Some(pid_file) = config.pid_file {
-            daemonize = daemonize.pid_file(pid_file).chown_pid_file(true);
+            daemonize = daemonize.pid_file(pid_file).chown_pid_file(false);
             // Give child process some time to write PID file
             daemonize = daemonize.exit_action(|| thread::sleep(std::time::Duration::from_millis(10)));
         }
