--- Authen-PAM-0.16/Makefile.PL.orig    2017-06-07 17:12:32.243856974 +0530
+++ Authen-PAM-0.16/Makefile.PL 2017-06-07 17:12:41.920801639 +0530
@@ -10,7 +10,7 @@
 
     # returns a reference to anonymous hash which is then interpreted as
     # additional options to the WriteMakeFile
-    $options = require "pam.cfg";
+    $options = require "./pam.cfg";
 
     if ( $Config{'osname'} eq 'solaris' && $Config{'osvers'} eq '2.6') {
       print "Adding a workaround for a bug in the Solaris 2.6 pam library\n";
