diff --git a/Makefile b/Makefile
index cad130e..edd7b50 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ all: $(DEBS)
 .PHONY: submodule
 submodule:
 ifeq ($(shell test -f "$(SRCDIR)/configure" && echo 1 || echo 0), 0)
-	git submodule update --init --recursive
+	git -c protocol.file.allow=always submodule update --init --recursive
 	cd $(SRCDIR); meson subprojects download
 endif
 
@@ -56,8 +56,8 @@ $(BUILDDIR): submodule
 deb kvm: $(DEBS)
 $(DEB_DBG): $(DEB)
 $(DEB): $(BUILDDIR)
-	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc -j
-	lintian $(DEBS)
+	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc -j -d
+#	lintian $(DEBS)
 
 sbuild: $(DSC)
 	sbuild $(DSC)
diff --git a/debian/rules b/debian/rules
index 51f56c5..7475b09 100755
--- a/debian/rules
+++ b/debian/rules
@@ -69,7 +69,7 @@ endif
 	    --enable-linux-io-uring \
 	    --enable-numa \
 	    --enable-opengl \
-	    --enable-rbd \
+	    --disable-rbd \
 	    --enable-seccomp \
 	    --enable-slirp \
 	    --enable-spice \
@@ -156,7 +156,7 @@ binary-arch: build install
 	dh_installdeb
 	# Debian >= Buster has libglusterfs0 and lots of libgfFOO, upstream has the libs in glusterfs-common
 	# pass -x to dpkg-shlibdeps and specify dependencies in d/control instead
-	dh_shlibdeps -- -xlibglusterfs0 -xlibgfxdr0 -xlibgfrpc0 -xlibgfdb0 -xlibgfchangelog0 -xlibgfapi0
+#	dh_shlibdeps -- -xlibglusterfs0 -xlibgfxdr0 -xlibgfrpc0 -xlibgfdb0 -xlibgfchangelog0 -xlibgfapi0
 	dh_gencontrol
 	dh_md5sums
 	dh_builddeb
