diff --git a/Makefile b/Makefile
index 8be4721..7933bdb 100644
--- a/Makefile
+++ b/Makefile
@@ -33,8 +33,7 @@ $(BUILDDIR): keycodemapdb | submodule
 deb kvm: $(DEBS)
 $(DEB_DBG): $(DEB)
 $(DEB): $(BUILDDIR)
-	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc -j
-	lintian $(DEBS)
+	cd $(BUILDDIR); dpkg-buildpackage -b -us -uc -j -d
 
 .PHONY: update
 update:
diff --git a/debian/rules b/debian/rules
index 6814f42..6c47867 100755
--- a/debian/rules
+++ b/debian/rules
@@ -180,7 +180,6 @@ binary-arch: build install
 	dh_installdeb
 	# Debian >= Buster has libglusterfs0 and lots of libgfFOO, upstream has the libs in glusterfs-common
 	# pass -x to dpkg-shlibdeps and specify dependencies in d/control instead
-	dh_shlibdeps -- -xlibglusterfs0 -xlibgfxdr0 -xlibgfrpc0 -xlibgfdb0 -xlibgfchangelog0 -xlibgfapi0
 	dh_gencontrol
 	dh_md5sums
 	dh_builddeb
