diff -Naur gnome-blog-0.9.2.orig/blog_applet.py gnome-blog-0.9.2/blog_applet.py
--- gnome-blog-0.9.2.orig/blog_applet.py	2009-11-04 14:08:22.000000000 +0100
+++ gnome-blog-0.9.2/blog_applet.py	2012-03-07 18:18:15.935465366 +0100
@@ -4,7 +4,6 @@
 
 import gtk
 import gobject
-import gnome
 import gnomeapplet
 import gconf
 import string  # maybe someone can do this trick without string?
diff -Naur gnome-blog-0.9.2.orig/gnome-blog-poster gnome-blog-0.9.2/gnome-blog-poster
--- gnome-blog-0.9.2.orig/gnome-blog-poster	2012-03-07 18:17:33.000000000 +0100
+++ gnome-blog-0.9.2/gnome-blog-poster	2012-03-07 18:17:43.937229318 +0100
@@ -3,8 +3,6 @@
 pygtk.require('2.0')
 
 import gtk
-import gnome
-import gnome.ui
 import gconf
 import gettext
 _ = gettext.gettext
diff -Naur gnome-blog-0.9.2.orig/rich_entry.py gnome-blog-0.9.2/rich_entry.py
--- gnome-blog-0.9.2.orig/rich_entry.py	2012-03-07 18:17:33.000000000 +0100
+++ gnome-blog-0.9.2/rich_entry.py	2012-03-07 18:17:43.937229319 +0100
@@ -2,7 +2,6 @@
 import gtk.gdk
 import pango
 import gobject
-import gnome
 import gio
 
 import gettext
