diff -rudp rendercheck-1.5.orig/t_blend.c rendercheck-1.5/t_blend.c
--- rendercheck-1.5.orig/t_blend.c	2015-06-30 23:20:52.000000000 -0400
+++ rendercheck-1.5/t_blend.c	2015-06-30 23:22:07.000000000 -0400
@@ -125,7 +125,7 @@ blend_test(Display *dpy, picture_info *w
 							   dst_color[k]->color.g,
 							   dst_color[k]->color.b,
 							   dst_color[k]->color.a);
-						    printf("src: %s, dst: %s\n", src_color[j]->name, dst->name);
+						    printf("src: %s, dst: %s\n", src_color[j]->name, dst_color[k]->name);
 						    free(srcformat);
 						    return FALSE;
 					    }
diff -rudp rendercheck-1.5.orig/t_composite.c rendercheck-1.5/t_composite.c
--- rendercheck-1.5.orig/t_composite.c	2015-06-30 23:20:52.000000000 -0400
+++ rendercheck-1.5/t_composite.c	2015-06-30 23:21:11.000000000 -0400
@@ -155,7 +155,7 @@ composite_test(Display *dpy, picture_inf
 				printf("src: %s, mask: %s, dst: %s\n",
 				       src_color[s]->name,
 				       mask_color[m]->name,
-				       dst->name);
+				       dst_color[d]->name);
 				XDestroyImage(image);
 				return FALSE;
 			    }
