diff --git a/src/core/renderer_vk/renderer_vk.cpp b/src/core/renderer_vk/renderer_vk.cpp
index d05a070f..56d93f72 100644
--- a/src/core/renderer_vk/renderer_vk.cpp
+++ b/src/core/renderer_vk/renderer_vk.cpp
@@ -888,7 +888,7 @@ void RendererVK::display() {
 void RendererVK::initGraphicsContext(SDL_Window* window) {
 	targetWindow = window;
 	// Resolve all instance function pointers
-	static vk::DynamicLoader dl;
+	static vk::detail::DynamicLoader dl;
 	VULKAN_HPP_DEFAULT_DISPATCHER.init(dl.getProcAddress<PFN_vkGetInstanceProcAddr>("vkGetInstanceProcAddr"));
 
 	// Create Instance
@@ -1588,4 +1588,4 @@ void RendererVK::deinitGraphicsContext() {
 
 	// TODO: Make it so that depth and colour buffers get written back to 3DS memory
 	printf("RendererVK::DeinitGraphicsContext called\n");
-}
\ No newline at end of file
+}
