From ef26cd0e8014e09b3671500f92c8f1b0d0ebe340 Mon Sep 17 00:00:00 2001
From: Gianluca Boiano <morf3089@gmail.com>
Date: Thu, 20 Sep 2018 14:24:39 +0200
Subject: [PATCH] menu.popup options object should be optional (fix #1560)

---
 main.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.js b/main.js
index 6d222a7da..ddc4d807b 100644
--- a/main.js
+++ b/main.js
@@ -389,7 +389,7 @@ function createWindow() {
   Menu.setApplicationMenu(menu);
 
   ipcMain.on('contextmenu-click', () => {
-    menu.popup();
+    menu.popup({});
   });
 
   // put logic here to set tray icon based on OS
