--- emacspeak-55.0/lisp/emacspeak-preamble.el	2021-10-22 19:11:58.000000000 +0300
+++ emacspeak-55.0/lisp/emacspeak-preamble.el	2022-01-10 16:12:47.935076751 +0300
@@ -54,7 +54,7 @@
 
 ;;;###autoload
 (defvar emacspeak-directory
-  (expand-file-name "../" (file-name-directory load-file-name))
+  (file-name-as-directory "/usr/share/emacs/site-lisp/emacspeak/")
   "emacspeak directory")
 
 ;;;###autoload
@@ -64,7 +64,7 @@
 
 ;;;###autoload
 (defvar emacspeak-sounds-directory
-  (expand-file-name  "sounds/" emacspeak-directory)
+  (file-name-as-directory "/usr/share/sounds/emacspeak/")
   "Auditory icons directory.")
 
 ;;;###autoload

diff --no-dereference --ignore-all-space --unified --recursive --text emacspeak.orig/servers/espeak emacspeak.new/servers/espeak
--- emacspeak.orig/servers/espeak	2021-07-12 18:44:02.147914519 +0300
+++ emacspeak.new/servers/espeak	2021-07-12 19:22:34.098431019 +0300
@@ -526,9 +526,7 @@
 if {[info exists server_p]} {
     set tts(input) sock0
 }
-set servers [file dirname $argv0]
-set tclTTS $servers/native-espeak
-load $tclTTS/tclespeak[info sharedlibextension]
+load /usr/lib/emacspeak/tclespeak.so
 if {[file exists /proc/asound]} {
     set tts(play) /usr/bin/aplay
 }
