diff -aur vale-1.0.6-old/Makefile vale-1.0.6-new/Makefile
--- vale-1.0.6-old/Makefile	2018-10-23 18:12:29.833646925 +0100
+++ vale-1.0.6-new/Makefile	2018-10-23 18:13:54.895016355 +0100
@@ -5,6 +5,7 @@
 CURR_SHA=$(shell git rev-parse --verify HEAD)
 
 LDFLAGS=-ldflags "-s -w -X main.version=$(LAST_TAG)"
+RUBY_CUSTOM_PATH=$(GEM_HOME)/bin
 
 .PHONY: data test lint ci cross install bump rules setup bench compare release
 
@@ -64,8 +65,10 @@
 	go get github.com/aclements/go-misc/benchmany
 	go get -u github.com/alecthomas/gometalinter
 	go get -u github.com/jteeuwen/go-bindata/...
-	bundle install
-	gem specific_install -l https://github.com/jdkato/aruba.git -b d-win-fix
+	gem install --no-rdoc --no-ri bundle
+	bundle install --path ${RUBY_CUSTOM_PATH}
+	gem install --no-rdoc --no-ri specific_install
+	gem specific_install -l https://github.com/jdkato/aruba.git
 
 rules:
 	go-bindata -ignore=\\.DS_Store -pkg="rule" -o rule/rule.go rule/**/*.yml
