diff -Naur notifier/octopi-notifier/octopi-notifier.pro-orig notifier/octopi-notifier/octopi-notifier.pro
--- notifier/octopi-notifier/octopi-notifier.pro-orig	2016-04-01 16:34:55.541500593 -0400
+++ notifier/octopi-notifier/octopi-notifier.pro	2016-04-01 16:34:59.948205892 -0400
@@ -8,7 +8,7 @@
 
 # This controls whether octopi-notifier uses KStatusNotifier lib
 # You SHOULD REALLY enable the following define in plasma 5 desktops!
-#DEFINES += KSTATUS
+DEFINES += KSTATUS
 
 contains(DEFINES, KSTATUS){
   QT += KNotifications
