diff -Naurp a/caddy/caddymain/run.go b/caddy/caddymain/run.go
--- a/caddy/caddymain/run.go	2017-05-02 13:02:28.000000000 -0500
+++ b/caddy/caddymain/run.go	2017-05-15 20:56:53.736272531 -0500
@@ -21,6 +21,21 @@ import (
 
 	"github.com/mholt/caddy/caddytls"
 	// This is where other plugins get plugged in (imported)
+	_ "github.com/caddyserver/dnsproviders/cloudflare"
+	_ "github.com/caddyserver/dnsproviders/digitalocean"
+	_ "github.com/caddyserver/dnsproviders/dnsimple"
+	_ "github.com/caddyserver/dnsproviders/dnspod"
+	_ "github.com/caddyserver/dnsproviders/dyn"
+	_ "github.com/caddyserver/dnsproviders/exoscale"
+	_ "github.com/caddyserver/dnsproviders/gandi"
+	_ "github.com/caddyserver/dnsproviders/googlecloud"
+	_ "github.com/caddyserver/dnsproviders/linode"
+	_ "github.com/caddyserver/dnsproviders/namecheap"
+	_ "github.com/caddyserver/dnsproviders/ovh"
+	_ "github.com/caddyserver/dnsproviders/rackspace"
+	_ "github.com/caddyserver/dnsproviders/rfc2136"
+	_ "github.com/caddyserver/dnsproviders/route53"
+	_ "github.com/caddyserver/dnsproviders/vultr"
 )
 
 func init() {
