diff -Naur laditools-1.1.0.orig/INSTALL laditools-1.1.0/INSTALL
--- laditools-1.1.0.orig/INSTALL	2015-09-13 14:38:51.000000000 -0700
+++ laditools-1.1.0/INSTALL	2018-01-27 10:23:41.606495560 -0800
@@ -29,7 +29,7 @@
 Plus, the following dependencies must be satisfied:
 
  - GTK+3            (>= 3.2.2,  http://www.gtk.org/)
- - python-enum      (>= 0.4.4,  http://pypi.python.org/pypi/enum/)
+ - python-enum34    (>= 1.1.2,  https://pypi.python.org/pypi/enum34/)
  - VTE              (>= 0.30.1, http://ftp.gnome.org/pub/GNOME/sources/vte/)
 
 Here is a list of optional dependencies:
diff -Naur laditools-1.1.0.orig/laditools/ladish.py laditools-1.1.0/laditools/ladish.py
--- laditools-1.1.0.orig/laditools/ladish.py	2015-09-13 14:38:51.000000000 -0700
+++ laditools-1.1.0/laditools/ladish.py	2018-01-27 10:23:41.629828859 -0800
@@ -28,10 +28,8 @@
 studio_obj_path = "/org/ladish/Studio"
 service_name = name_base
 
-LadishStatusType = Enum("STUDIO_STOPPED",
-                        "NOT_AVAILABLE",
-                        "NO_STUDIO_LOADED",
-                        "STUDIO_RUNNING")
+LadishStatusType = Enum("LadishStatusType",
+                        "STUDIO_STOPPED NOT_AVAILABLE NO_STUDIO_LOADED STUDIO_RUNNING")
 
 class LadishProxyError(Exception): pass
 class LadishStudioException(Exception): pass
