diff --git a/src/main.cpp b/src/main.cpp
index e1f43ad0..ef2b74be 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -34,6 +34,8 @@ int main(int argc, char** argv) {
     setenv("_JAVA_AWT_WM_NONREPARENTING", "1", 1);
     setenv("MOZ_ENABLE_WAYLAND", "1", 1);
     setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
+    setenv("ASAN_OPTIONS", "detect_odr_violation=0,log_path=asan.log", 1);
+    setenv("LD_PRELOAD", "/usr/lib/libasan.so.8.0.0", 1);
 
     // parse some args
     std::string              configPath;
