--- etc-update.orig	2012-01-29 12:57:16.311869407 +1000
+++ etc-update	2012-01-29 13:37:15.892883106 +1000
@@ -3,6 +3,8 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Id: etc-update 13723 2009-06-28 16:12:28Z vapier $
 
+# Ported to Arch Linux by Adam Nielsen <malvineous@shikadi.net>
+
 # Author Brandon Low <lostlogic@gentoo.org>
 #
 # Previous version (from which I've borrowed a few bits) by:
@@ -51,19 +53,19 @@
 	for path in ${CONFIG_PROTECT} ; do
 		path="${ROOT}${path}"
 		# Do not traverse hidden directories such as .svn or .git.
-		find_opts="-name .* -type d -prune -o -name ._cfg????_*"
+		find_opts="-name .* -type d -prune -o -name *.pacnew"
 		if [ ! -d "${path}" ]; then
 			[ ! -f "${path}" ] && continue
 			my_basename="${path##*/}"
 			path="${path%/*}"
-			find_opts="-maxdepth 1 -name ._cfg????_${my_basename}"
+			find_opts="-maxdepth 1 -name *.pacnew"
 		fi
 
-		ofile=""
+		ofile=".pacnew"
 		# The below set -f turns off file name globbing in the ${find_opts} expansion.
 		for file in $(set -f ; find ${path}/ ${find_opts} \
 		       ! -name '.*~' ! -iname '.*.bak' -print |
-			   sed -e "s:\(^.*/\)\(\._cfg[0-9]*_\)\(.*$\):\1\2\3\%\1%\2\%\3:" |
+			   sed -e "s:\(^.*/\)\(.*$\)\(\.pacnew\):\1\2\3\%\1%\2\%\3:" |
 			   sort -t'%' -k2,2 -k4,4 -k3,3 | LANG=POSIX LC_ALL=POSIX cut -f1 -d'%'); do
 
 			rpath=$(echo "${file/\/\///}" | sed -e "s:/[^/]*$::")
@@ -72,36 +74,37 @@
 				mpath="${ROOT}${mpath}"
 				mpath=$(echo "${mpath/\/\///}")
 				if [[ "${rpath}" == "${mpath}"* ]]; then
-					mv ${rpath}/${rfile} ${rpath}/${rfile:10}
+					mv ${rpath}/${rfile} ${rpath}/${rfile::-7}
 					break
 				fi
 			done
+
 			if [[ ! -f ${file} ]] ; then
 				echo "Skipping non-file ${file} ..."
 				continue
 			fi
 
-			if [[ "${ofile:10}" != "${rfile:10}" ]] ||
+			if [[ "${ofile::-7}" != "${rfile::-7}" ]] ||
 			   [[ ${opath} != ${rpath} ]]; then
 				MATCHES=0
 				if [[ "${EU_AUTOMERGE}" == "yes" ]]; then
-					if [ ! -e "${rpath}/${rfile}" ] || [ ! -e "${rpath}/${rfile:10}" ]; then
+					if [ ! -e "${rpath}/${rfile}" ] || [ ! -e "${rpath}/${rfile::-7}" ]; then
 						MATCHES=0
 					else
-						diff -Bbua ${rpath}/${rfile} ${rpath}/${rfile:10} | egrep '^[+-]' | egrep -v '^[+-][\t ]*#|^--- |^\+\+\+ ' | egrep -qv '^[-+][\t ]*$'
+						diff -Bbua ${rpath}/${rfile} ${rpath}/${rfile::-7} | egrep '^[+-]' | egrep -v '^[+-][\t ]*#|^--- |^\+\+\+ ' | egrep -qv '^[-+][\t ]*$'
 						MATCHES=$?
 					fi
-				elif [[ -z $(diff -Nua ${rpath}/${rfile} ${rpath}/${rfile:10}|
+				elif [[ -z $(diff -Nua ${rpath}/${rfile} ${rpath}/${rfile::-7}|
 							  grep "^[+-][^+-]"|grep -v '# .Header:.*') ]]; then
 					MATCHES=1
 				fi
 				if [[ "${MATCHES}" == "1" ]]; then
-					echo "Automerging trivial changes in: ${rpath}/${rfile:10}"
-					mv ${rpath}/${rfile} ${rpath}/${rfile:10}
+					echo "Automerging trivial changes in: ${rpath}/${rfile::-7}"
+					mv ${rpath}/${rfile} ${rpath}/${rfile::-7}
 					continue
 				else
 					count=${count}+1
-					echo "${rpath}/${rfile:10}" > ${TMP}/files/${count}
+					echo "${rpath}/${rfile::-7}" > ${TMP}/files/${count}
 					echo "${rpath}/${rfile}" >> ${TMP}/files/${count}
 					ofile="${rfile}"
 					opath="${rpath}"
@@ -494,10 +497,16 @@
 done
 ${SET_X} && set -x
 
-type portageq > /dev/null || exit $?
-eval $(portageq envvar -v CONFIG_PROTECT \
-	CONFIG_PROTECT_MASK PORTAGE_CONFIGROOT PORTAGE_TMPDIR ROOT USERLAND)
-export PORTAGE_TMPDIR
+# These paths are relative to ${ROOT}
+export CONFIG_PROTECT="/etc /usr/lib /usr/share/config"
+export CONFIG_PROTECT_MASK="/etc/arch-release /etc/udev/rules.d"
+# Location of etc-update.conf
+export PORTAGE_CONFIGROOT="/"
+# Place to store temp files while merging
+export PORTAGE_TMPDIR="/var/tmp"
+# System root directory
+export ROOT="/"
+export USERLAND='GNU'
 
 TMP="${PORTAGE_TMPDIR}/etc-update-$$"
 trap "die terminated 1" SIGTERM
@@ -563,7 +572,7 @@
 
 declare -i count=0
 declare input=0
-declare title="Gentoo's etc-update tool!"
+declare title="Arch Linux etc-update tool!"
 
 scan
 
