--- a/ETL/_pen.h	2017-09-08 05:51:26.000000000 -0400
+++ b/ETL/_pen.h	2018-07-27 12:56:43.833337339 -0400
@@ -136,7 +136,8 @@
 		typedef int value_type;
 		value_type x,y;
 		difference_type(value_type x, value_type y):x(x),y(y) { }
-		value_type &operator[](int i)const { return i?y:x; }
+		const value_type &operator[](int i) const { return i?y:x; }
+		value_type &operator[](int i) { return i?y:x; }
 	};
 
 protected:
