diff --git a/src/software/SfM/CMakeLists.txt b/src/software/SfM/CMakeLists.txt
index cc103147..68b040d4 100644
--- a/src/software/SfM/CMakeLists.txt
+++ b/src/software/SfM/CMakeLists.txt
@@ -241,6 +241,7 @@ endif(OpenMVG_USE_OPENCV)
 ###
 add_executable(openMVG_main_evalQuality main_evalQuality.cpp)
 target_link_libraries(openMVG_main_evalQuality
+  openMVG_image
   openMVG_system
   openMVG_features
   openMVG_sfm
