diff --git a/fs/exfat/exfat_super.c b/fs/exfat/exfat_super.c
index a003a72..464de88 100644
--- a/fs/exfat/exfat_super.c
+++ b/fs/exfat/exfat_super.c
@@ -1375,8 +1375,7 @@ static void *exfat_follow_link(struct dentry *dentry, struct nameidata *nd)
 #endif
 
 const struct inode_operations exfat_symlink_inode_operations = {
-	.readlink    = generic_readlink,
-	.follow_link = exfat_follow_link,
+	.readlink    = generic_readlink
 };
 
 static int exfat_file_release(struct inode *inode, struct file *filp)
