diff -Naur package-rebuild/main/createWindow.js package-rebuild2/main/createWindow.js
--- package-rebuild/main/createWindow.js	2021-04-15 01:41:05.592749192 +0200
+++ package-rebuild2/main/createWindow.js	2021-04-15 01:39:12.321576964 +0200
@@ -78,7 +78,7 @@
     }
     window.on("close", () => {
         windowState.saveState(window);
-        if (process.platform === "win32") {
+        if (process.platform !== "darwin") {
             const currentWindows = electron_1.BrowserWindow.getAllWindows();
             const hasNoOtherOpenWindows = currentWindows.every(currentWindow => Boolean(currentWindow.id === window.id ||
                 (warmWindowState.warmedWindow &&
