--- a/ts/scripts/get-expire-time.ts	2021-12-15 03:06:07.198974886 -0500
+++ b/ts/scripts/get-expire-time.ts	2021-12-15 03:07:10.951561856 -0500
@@ -2,15 +2,12 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 
 import { join } from 'path';
-import { execSync } from 'child_process';
 import { writeFileSync } from 'fs';
 
 import { DAY } from '../util/durations';
 
-const unixTimestamp = parseInt(
-  execSync('git show -s --format=%ct').toString('utf8'),
-  10
-);
+const unixTimestamp = parseInt(process.env.SOURCE_DATE_EPOCH!.toString(), 10) * 1000;
+
 const buildCreation = unixTimestamp * 1000;
 
 const buildExpiration = buildCreation + DAY * 90;
