--- a/Makefile
+++ b/Makefile
@@ -70,6 +70,14 @@
    LIBGPR_TYPES=static
 endif
 
+ifneq ($(strip $(CFLAGS)),)
+   TOOLFLAGS += -cargs $(CFLAGS)
+endif
+
+ifneq ($(strip $(LDFLAGS)),)
+   TOOLFLAGS += -largs $(LDFLAGS)
+endif
+
 # Used to pass extra options to GPRBUILD, like -d for instance
 GPRBUILD_OPTIONS=
 
@@ -92,27 +100,27 @@
 .PHONY: all distall gprbuild gprconfig gprclean gprinstall gprname gprls
 
 build all:
-	$(GPRBUILD_BUILDER)
+	$(GPRBUILD_BUILDER) $(TOOLFLAGS)
 
 distall: all install
 
 gprbuild:
-	$(GPRBUILD_BUILDER) gprbuild-main.adb
+	$(GPRBUILD_BUILDER) gprbuild-main.adb $(TOOLFLAGS)
 
 gprinstall:
-	$(GPRBUILD_BUILDER) gprinstall-main.adb
+	$(GPRBUILD_BUILDER) gprinstall-main.adb $(TOOLFLAGS)
 
 gprclean:
-	$(GPRBUILD_BUILDER) gprclean-main.adb
+	$(GPRBUILD_BUILDER) gprclean-main.adb $(TOOLFLAGS)
 
 gprconfig:
-	$(GPRBUILD_BUILDER) gprconfig-main.adb
+	$(GPRBUILD_BUILDER) gprconfig-main.adb $(TOOLFLAGS)
 
 gprname:
-	$(GPRBUILD_BUILDER) gprname-main.adb
+	$(GPRBUILD_BUILDER) gprname-main.adb $(TOOLFLAGS)
 
 gprls:
-	$(GPRBUILD_BUILDER) gprls-main.adb
+	$(GPRBUILD_BUILDER) gprls-main.adb $(TOOLFLAGS)
 
 #################################
 # Gprbuild installation targets #
@@ -140,15 +148,15 @@
 
 libgpr.build.shared:
 	${BUILDER} -XLIBRARY_TYPE=relocatable \
-		-XXMLADA_BUILD=relocatable $(GPR_GPR)
+		-XXMLADA_BUILD=relocatable $(GPR_GPR) $(TOOLFLAGS)
 
 libgpr.build.static:
 	${BUILDER} -XLIBRARY_TYPE=static \
-		-XXMLADA_BUILD=static $(GPR_GPR)
+		-XXMLADA_BUILD=static $(GPR_GPR) $(TOOLFLAGS)
 
 libgpr.build.static-pic:
 	${BUILDER} -XLIBRARY_TYPE=static-pic \
-		-XXMLADA_BUILD=static-pic $(GPR_GPR)
+		-XXMLADA_BUILD=static-pic $(GPR_GPR) $(TOOLFLAGS)
 
 libgpr.install: libgpr.uninstall $(foreach t, $(LIBGPR_TYPES), libgpr.install.$(t))
 
