diff --git a/src/tracer/xml-parse.c b/src/tracer/xml-parse.c
index f0b77bb..4a89edf 100644
--- a/src/tracer/xml-parse.c
+++ b/src/tracer/xml-parse.c
@@ -1710,9 +1710,9 @@ static void Parse_XML_Merge (int rank, xmlDocPtr xmldoc, xmlNodePtr current_tag,
 
 	sort_trace =  xmlGetProp_env(rank, current_tag, TRACE_MERGE_SORT_TRACE);
 	if (sort_trace != NULL && !xmlStrcasecmp(sort_trace, xmlNO))
-		set_option_merge_sort_trace(FALSE);
+		set_option_merge_SortTrace(FALSE);
 	else
-		set_option_merge_sort_trace(TRUE);
+		set_option_merge_SortTrace(TRUE);
 
 	filename = (char*) xmlNodeListGetString_env (rank, xmldoc, current_tag->xmlChildrenNode, 1);
 	if (filename == NULL || strlen(filename) == 0)
