diff --git a/setup.py b/setup.py
index cca7cc3..b5617e3 100644
--- a/setup.py
+++ b/setup.py
@@ -18,9 +18,10 @@ from _setup.upload_products import upload_products
 from _setup.release import release
 
 # bootstrap setuptools for users that don't already have it
-import ez_setup
+# commented because it breaks build process
+#import ez_setup
 
-ez_setup.use_setuptools()
+#ez_setup.use_setuptools()
 
 try:
     FILE = open("README.rst", "r")
