diff --git a/far2l/CMakeLists.txt b/far2l/CMakeLists.txt
index 7213211..3aa507d 100644
--- a/far2l/CMakeLists.txt
+++ b/far2l/CMakeLists.txt
@@ -187,6 +187,6 @@ add_custom_command(TARGET far2l POST_BUILD
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 #TODO
 else()
-    install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/DE/icons" DESTINATION "/usr/share" USE_SOURCE_PERMISSIONS FILES_MATCHING PATTERN "*")
-    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l.desktop" DESTINATION "/usr/share/applications")
+    install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/DE/icons" DESTINATION "${CMAKE_INSTALL_PREFIX}/share" USE_SOURCE_PERMISSIONS FILES_MATCHING PATTERN "*")
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l.desktop" DESTINATION "${CMAKE_INSTALL_PREFIX}/share/applications")
 endif()
[mmozeiko@BHQKCX1 far2l]$ git diff CMakeLists.txt  far2l/CMakeLists.txt
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ca2753..52d1442 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,9 +6,9 @@ if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-#if(NOT CMAKE_INSTALL_PREFIX)
+if(NOT CMAKE_INSTALL_PREFIX)
    set(CMAKE_INSTALL_PREFIX "/usr")
-#endif()
+endif()
