diff --color --unified --recursive --text '--exclude=.git' '--exclude=generated' '--exclude=build' '--exclude=_build' OpenRV.clean/cmake/dependencies/ffmpeg.cmake OpenRV/cmake/dependencies/ffmpeg.cmake
--- OpenRV.clean/cmake/dependencies/ffmpeg.cmake	2024-07-19 23:50:43.935913713 -0700
+++ OpenRV/cmake/dependencies/ffmpeg.cmake	2024-07-19 23:51:22.256940466 -0700
@@ -34,6 +34,14 @@
     "6dfc27fcb6da6f653c6ec025c2cd9b00"
 )
 
+SET(_git_repository
+    "https://git.ffmpeg.org/ffmpeg.git"
+)
+
+SET(_git_tag
+    "e38092ef9395d7049f871ef4d5411eb410e283e0"
+)
+
 SET(_base_dir
     ${RV_DEPS_BASE_DIR}/${_target}
 )
@@ -167,6 +175,8 @@
 
 # Make a list of common FFmpeg config options
 LIST(APPEND RV_FFMPEG_COMMON_CONFIG_OPTIONS "--enable-shared")
+LIST(APPEND RV_FFMPEG_COMMON_CONFIG_OPTIONS "--enable-pic")
+LIST(APPEND RV_FFMPEG_EXTRA_C_OPTIONS "--extra-cflags=-fno-lto")
 LIST(APPEND RV_FFMPEG_COMMON_CONFIG_OPTIONS "--disable-static")
 LIST(APPEND RV_FFMPEG_COMMON_CONFIG_OPTIONS "--disable-iconv")
 LIST(APPEND RV_FFMPEG_COMMON_CONFIG_OPTIONS "--disable-outdevs")
@@ -297,12 +306,14 @@
 EXTERNALPROJECT_ADD(
   ${_target}
   DEPENDS ${RV_FFMPEG_DEPENDS}
-  DOWNLOAD_NAME ${_target}_${_version}.zip
-  DOWNLOAD_DIR ${RV_DEPS_DOWNLOAD_DIR}
-  DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+  # DOWNLOAD_NAME ${_target}_${_version}.zip
+  # DOWNLOAD_DIR ${RV_DEPS_DOWNLOAD_DIR}
+  # DOWNLOAD_EXTRACT_TIMESTAMP TRUE
   INSTALL_DIR ${_install_dir}
-  URL ${_download_url}
-  URL_MD5 ${_download_hash}
+  # URL ${_download_url}
+  # URL_MD5 ${_download_hash}
+  GIT_REPOSITORY ${_git_repository}
+  GIT_TAG ${_git_tag}
   SOURCE_DIR ${RV_DEPS_BASE_DIR}/${_target}/src
   PATCH_COMMAND ${RV_FFMPEG_PATCH_COMMAND_STEP}
   CONFIGURE_COMMAND
