diff -r --unified rtw88.orig/rtw8703b.c rtw88.new/rtw8703b.c
--- rtw88.orig/rtw8703b.c	2024-10-14 03:28:28.996045616 +0000
+++ rtw88.new/rtw8703b.c	2024-10-14 03:29:40.942798097 +0000
@@ -1955,7 +1955,7 @@
 	.ops = &rtw8703b_ops,
 	.id = RTW_CHIP_TYPE_8703B,
 
-	.fw_name = "rtw88/rtw8703b_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8703b_fw.bin",
 	.wlan_cpu = RTW_WCPU_11N,
 	.tx_pkt_desc_sz = 40,
 	.tx_buf_desc_sz = 16,
@@ -2015,7 +2015,7 @@
 	.iqk_threshold = 8,
 
 	/* WOWLAN firmware exists, but not implemented yet */
-	.wow_fw_name = "rtw88/rtw8703b_wow_fw.bin",
+	.wow_fw_name = "rtw88-fixed/rtw8703b_wow_fw.bin",
 	.wowlan_stub = NULL,
 	.max_scan_ie_len = IEEE80211_MAX_DATA_LEN,
 
@@ -2068,8 +2068,8 @@
 };
 EXPORT_SYMBOL(rtw8703b_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8703b_fw.bin");
-MODULE_FIRMWARE("rtw88/rtw8703b_wow_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8703b_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8703b_wow_fw.bin");
 
 MODULE_AUTHOR("Fiona Klute <fiona.klute@gmx.de>");
 MODULE_DESCRIPTION("Realtek 802.11n wireless 8703b driver");
diff -r --unified rtw88.orig/rtw8723d.c rtw88.new/rtw8723d.c
--- rtw88.orig/rtw8723d.c	2024-10-14 03:28:28.996045616 +0000
+++ rtw88.new/rtw8723d.c	2024-10-14 03:29:40.952798559 +0000
@@ -2114,7 +2114,7 @@
 const struct rtw_chip_info rtw8723d_hw_spec = {
 	.ops = &rtw8723d_ops,
 	.id = RTW_CHIP_TYPE_8723D,
-	.fw_name = "rtw88/rtw8723d_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8723d_fw.bin",
 	.wlan_cpu = RTW_WCPU_11N,
 	.tx_pkt_desc_sz = 40,
 	.tx_buf_desc_sz = 16,
@@ -2198,7 +2198,7 @@
 };
 EXPORT_SYMBOL(rtw8723d_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8723d_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8723d_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11n wireless 8723d driver");
diff -r --unified rtw88.orig/rtw8812a.c rtw88.new/rtw8812a.c
--- rtw88.orig/rtw8812a.c	2024-10-14 03:28:28.999379113 +0000
+++ rtw88.new/rtw8812a.c	2024-10-14 03:29:40.939464611 +0000
@@ -1016,7 +1016,7 @@
 const struct rtw_chip_info rtw8812a_hw_spec = {
 	.ops = &rtw8812a_ops,
 	.id = RTW_CHIP_TYPE_8812A,
-	.fw_name = "rtw88/rtw8812a_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8812a_fw.bin",
 	.wlan_cpu = RTW_WCPU_11N,
 	.tx_pkt_desc_sz = 40,
 	.tx_buf_desc_sz = 16,
@@ -1095,7 +1095,7 @@
 };
 EXPORT_SYMBOL(rtw8812a_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8812a_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8812a_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11ac wireless 8812a driver");
diff -r --unified rtw88.orig/rtw8821a.c rtw88.new/rtw8821a.c
--- rtw88.orig/rtw8821a.c	2024-10-14 03:28:28.999379113 +0000
+++ rtw88.new/rtw8821a.c	2024-10-14 03:29:40.932797636 +0000
@@ -1110,7 +1110,7 @@
 const struct rtw_chip_info rtw8821a_hw_spec = {
 	.ops = &rtw8821a_ops,
 	.id = RTW_CHIP_TYPE_8821A,
-	.fw_name = "rtw88/rtw8821a_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8821a_fw.bin",
 	.wlan_cpu = RTW_WCPU_11N,
 	.tx_pkt_desc_sz = 40,
 	.tx_buf_desc_sz = 16,
@@ -1190,7 +1190,7 @@
 };
 EXPORT_SYMBOL(rtw8821a_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8821a_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8821a_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11ac wireless 8821a/8811a driver");
diff -r --unified rtw88.orig/rtw8821c.c rtw88.new/rtw8821c.c
--- rtw88.orig/rtw8821c.c	2024-10-14 03:28:28.999379113 +0000
+++ rtw88.new/rtw8821c.c	2024-10-14 03:29:40.939464611 +0000
@@ -1951,7 +1951,7 @@
 const struct rtw_chip_info rtw8821c_hw_spec = {
 	.ops = &rtw8821c_ops,
 	.id = RTW_CHIP_TYPE_8821C,
-	.fw_name = "rtw88/rtw8821c_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8821c_fw.bin",
 	.wlan_cpu = RTW_WCPU_11AC,
 	.tx_pkt_desc_sz = 48,
 	.tx_buf_desc_sz = 16,
@@ -2034,7 +2034,7 @@
 };
 EXPORT_SYMBOL(rtw8821c_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8821c_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8821c_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11ac wireless 8821c driver");
diff -r --unified rtw88.orig/rtw8822b.c rtw88.new/rtw8822b.c
--- rtw88.orig/rtw8822b.c	2024-10-14 03:28:28.999379113 +0000
+++ rtw88.new/rtw8822b.c	2024-10-14 03:29:40.952798559 +0000
@@ -2490,7 +2490,7 @@
 const struct rtw_chip_info rtw8822b_hw_spec = {
 	.ops = &rtw8822b_ops,
 	.id = RTW_CHIP_TYPE_8822B,
-	.fw_name = "rtw88/rtw8822b_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8822b_fw.bin",
 	.wlan_cpu = RTW_WCPU_11AC,
 	.tx_pkt_desc_sz = 48,
 	.tx_buf_desc_sz = 16,
@@ -2580,7 +2580,7 @@
 };
 EXPORT_SYMBOL(rtw8822b_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8822b_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8822b_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11ac wireless 8822b driver");
diff -r --unified rtw88.orig/rtw8822c.c rtw88.new/rtw8822c.c
--- rtw88.orig/rtw8822c.c	2024-10-14 03:28:29.002712609 +0000
+++ rtw88.new/rtw8822c.c	2024-10-14 03:29:40.942798097 +0000
@@ -5310,7 +5310,7 @@
 const struct rtw_chip_info rtw8822c_hw_spec = {
 	.ops = &rtw8822c_ops,
 	.id = RTW_CHIP_TYPE_8822C,
-	.fw_name = "rtw88/rtw8822c_fw.bin",
+	.fw_name = "rtw88-fixed/rtw8822c_fw.bin",
 	.wlan_cpu = RTW_WCPU_11AC,
 	.tx_pkt_desc_sz = 48,
 	.tx_buf_desc_sz = 16,
@@ -5372,7 +5372,7 @@
 	.ampdu_density = IEEE80211_HT_MPDU_DENSITY_2,
 
 #ifdef CONFIG_PM
-	.wow_fw_name = "rtw88/rtw8822c_wow_fw.bin",
+	.wow_fw_name = "rtw88-fixed/rtw8822c_wow_fw.bin",
 	.wowlan_stub = &rtw_wowlan_stub_8822c,
 	.max_sched_scan_ssids = 4,
 #endif
@@ -5413,8 +5413,8 @@
 };
 EXPORT_SYMBOL(rtw8822c_hw_spec);
 
-MODULE_FIRMWARE("rtw88/rtw8822c_fw.bin");
-MODULE_FIRMWARE("rtw88/rtw8822c_wow_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8822c_fw.bin");
+MODULE_FIRMWARE("rtw88-fixed/rtw8822c_wow_fw.bin");
 
 MODULE_AUTHOR("Realtek Corporation");
 MODULE_DESCRIPTION("Realtek 802.11ac wireless 8822c driver");
