diff -Naurp gtk-theme.orig/extra/Makefile.am gtk-theme/extra/Makefile.am
--- gtk-theme.orig/extra/Makefile.am	2017-04-27 16:26:47.000000000 -0500
+++ gtk-theme/extra/Makefile.am	2017-05-08 21:13:52.053126399 -0500
@@ -35,7 +35,6 @@ if ENABLE_CHROME
 	$(MKDIR_P) $(adaptadir)/chrome
 
 	cp -Rv $(chrome_file) $(adaptadir)/chrome
-	cp -Rv $(chrome_nokto_file) $(noktodir)/chrome
 
 	cd $(adaptadir)
 	ln -sf ../Pop/chrome $(adaptaetadir)/chrome
@@ -43,6 +42,7 @@ if ENABLE_CHROME
 	
 if ENABLE_NOKTO
 	$(MKDIR_P) $(noktodir)/chrome
+	cp -Rv $(chrome_nokto_file) $(noktodir)/chrome
 	cd $(adaptadir)
 	ln -sf ../Pop-Nokto/chrome $(noktoetadir)/chrome
 endif
