diff --git a/build/main.js b/build/main.js
index e7a2cc5..2e1ebac 100644
--- a/build/main.js
+++ b/build/main.js
@@ -1975,7 +1975,7 @@
           tags: Object.assign(
             { country: user.country, offerId: user.offerId },
             {
-              environment: isDev ? "development" : "production",
+              environment: isDev() ? "development" : "production",
               os: `${external_os_default().type()} ${external_os_default().release()}`,
               "os.name": external_os_default().type(),
               arch: process.arch,
@@ -3255,10 +3255,10 @@
         processType: isProcessMain() ? "main" : process.type,
       }),
       (external_electron_log_default().transports.console.format = `{h}:{i}:{s}.{ms} ({processType}/{level}) ${isPlatform(PLATFORM.WINDOWS) ? ">" : "›"}  {text}`),
-      (external_electron_log_default().transports.console.level = isDev
+      (external_electron_log_default().transports.console.level = isDev()
         ? "silly"
         : "warn"),
-      (external_electron_log_default().transports.file.level = isDev
+      (external_electron_log_default().transports.file.level = isDev()
         ? "silly"
         : "warn"),
       (external_electron_log_default().transports.sentry = function (msg) {
diff --git a/build/preload.js b/build/preload.js
index a8bb8d6..73dc8b4 100644
--- a/build/preload.js
+++ b/build/preload.js
@@ -490,7 +490,7 @@
         }
         getSentryTags() {
           return {
-            environment: isDev ? "development" : "production",
+            environment: isDev() ? "development" : "production",
             os: `${external_os_default().type()} ${external_os_default().release()}`,
             "os.name": external_os_default().type(),
             arch: process.arch,
-- 
2.34.1
