--- a/src/main.c	2017-06-25 22:56:31.000000000 +0300
+++ b/src/main.c	2020-08-12 00:42:15.857946873 +0300
@@ -67,7 +67,7 @@
 		g_object_ref(G_OBJECT(applet->scroll_pix));
 		gtk_container_remove(GTK_CONTAINER(applet->scroll_pix->parent), applet->scroll_pix);
 	}
-#elif #HAVE_GTK3
+#elif HAVE_GTK3
 	if (gtk_widget_get_parent(applet->num_pix)) {
 		g_object_ref(G_OBJECT(applet->num_pix));
 		gtk_container_remove(GTK_CONTAINER(gtk_widget_get_parent(applet->num_pix)), applet->num_pix);
