diff --git a/apps/studio/src/common/platform_info/mainPlatformInfo.ts b/apps/studio/src/common/platform_info/mainPlatformInfo.ts
index beaed500..81b2cf72 100644
--- a/apps/studio/src/common/platform_info/mainPlatformInfo.ts
+++ b/apps/studio/src/common/platform_info/mainPlatformInfo.ts
@@ -59,7 +59,15 @@ export function mainPlatformInfo(): IPlatformInfo {
 
   const sessionType = p.env.XDG_SESSION_TYPE
 
-  const slice = isDevEnv ? 2 : 1
+  let slice = 0
+
+  for (const arg of p.argv) {
+    slice++
+    if (arg.includes('app.asar') === true) {
+      break
+    }
+  }
+
   const parsedArgs = yargs(p.argv.slice(slice))
   const appVersion = testMode ? '0.0.0' : e.app.getVersion()
 
