diff --git a/VariantAnnotation/src/vcffile.c b/VariantAnnotation/src/vcffile.c
index 6504ba7..079233d 100644
--- a/VariantAnnotation/src/vcffile.c
+++ b/VariantAnnotation/src/vcffile.c
@@ -43,7 +43,7 @@ static void vcfwarn(khash_t(WARNINGS) *warnings,
     va_start(argp, fmt);
     (void) vsnprintf(buf, bufsize, fmt, argp);
     va_end(argp);
-    Rf_warning(buf);
+    Rf_warning("%s", buf);
     Free(buf);
 
     return;
