diff --git a/src/software/utils/main_computeUncertainty.cpp b/src/software/utils/main_computeUncertainty.cpp
index 23d2ab9ee..5bba4d7b2 100644
--- a/src/software/utils/main_computeUncertainty.cpp
+++ b/src/software/utils/main_computeUncertainty.cpp
@@ -41,7 +41,8 @@ int aliceVision_main(int argc, char **argv)
   std::string algorithm = cov::EAlgorithm_enumToString(cov::eAlgorithmSvdTaylorExpansion);
   bool debug = false;
 
-  params.add_options()
+  po::options_description requiredParams("Required parameters");
+  requiredParams.add_options()
     ("input,i", po::value<std::string>(&sfmDataFilename)->required(),
       "SfMData file to align.")
   ("output,o", po::value<std::string>(&outSfMDataFilename)->required(),
@@ -51,12 +52,10 @@ int aliceVision_main(int argc, char **argv)
   ("algorithm,a", po::value<std::string>(&algorithm)->default_value(algorithm),
     "Algorithm.")
   ("debug,d", po::value<bool>(&debug)->default_value(debug),
-    "Enable creation of debug files in the current folder.")
-    ("verboseLevel,v", po::value<std::string>(&verboseLevel)->default_value(verboseLevel),
-      "verbosity level (fatal,  error, warning, info, debug, trace).");
+    "Enable creation of debug files in the current folder.");
 
   CmdLine cmdline("AliceVision computeUncertainty");
-  cmdline.add(params);
+  cmdline.add(requiredParams);
   if (!cmdline.execute(argc, argv))
   {
       return EXIT_FAILURE;
