diff --git a/Makefile.in b/Makefile.in
index 3c5cf07..3642dc2 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -13,11 +13,10 @@
 # (enclosed in the file LGPL).
 
 # where to install the binaries
-DESTDIR=@prefix@
 # prefix=@prefix@
 # exec_prefix=@exec_prefix@
 # BINDIR=$(DESTDIR)@bindir@
-LIBDIR=$(DESTDIR)/lib/
+LIBDIR ?= $(shell @OCAMLC@ -where)
 # DATADIR=$(DESTDIR)@datadir@/ocplib-simplex
 
 # where to install the man page
@@ -112,7 +111,8 @@ META: config.status
 ##############
 
 install: all META
-	OCAMLFIND_DESTDIR=$(LIBDIR) \
+	install -dm0755 $(DESTDIR)$(LIBDIR)/ocplib-simplex
+	OCAMLFIND_DESTDIR=$(DESTDIR)$(LIBDIR) \
 	  ocamlfind install ocplib-simplex src/$(LIBNAME).* src/*.mli META
