diff -aur --no-dereference '--exclude=.git' radium-5.9.99.pristine/bin/packages/build.sh radium-5.9.99.new/bin/packages/build.sh
--- radium-5.9.99.pristine/bin/packages/build.sh	2020-08-15 08:11:06.000000000 +0100
+++ radium-5.9.99.new/bin/packages/build.sh	2020-08-17 00:04:25.066598881 +0100
@@ -100,6 +100,8 @@
 patch -p0 <../faust.patch
 patch -p1 <../faust_setlocale.patch
 patch -p0 <../faust_svgfix.patch
+curl -sSLo ../faust_HTTPDServer_fix.patch "https://github.com/grame-cncm/faust/commit/8ccdc687572ec8de0e525b668c86611a3033e3fc.patch"
+patch -p1 < ../faust_HTTPDServer_fix.patch
 if env |grep INCLUDE_FAUSTDEV_BUT_NOT_LLVM ; then
     patch -p0 <../faust_nollvm.patch
 fi
@@ -173,7 +175,7 @@
 tar xvzf fluidsynth-1.1.6.tar.gz
 cd fluidsynth-1.1.6
 make clean
-CFLAGS="-fno-strict-aliasing -O3 -DDEFAULT_SOUNDFONT=\\\"\\\"" CPPFLAGS="-fno-strict-aliasing -O3" CXXFLAGS="-fno-strict-aliasing -O3" ./configure --enable-static --disable-aufile-support --disable-pulse-support --disable-alsa-support --disable-libsndfile-support --disable-portaudio-support --disable-oss-support --disable-midishare --disable-jack-support --disable-coreaudio --disable-coremidi --disable-dart --disable-lash --disable-ladcca --disable-aufile-support --disable-dbus-support --without-readline
+CFLAGS="-fPIC -fno-strict-aliasing -O3 -DDEFAULT_SOUNDFONT=\\\"\\\"" CPPFLAGS="-fPIC -fno-strict-aliasing -O3" CXXFLAGS="-fPIC -fno-strict-aliasing -O3" ./configure --enable-static --disable-aufile-support --disable-pulse-support --disable-alsa-support --disable-libsndfile-support --disable-portaudio-support --disable-oss-support --disable-midishare --disable-jack-support --disable-coreaudio --disable-coremidi --disable-dart --disable-lash --disable-ladcca --disable-aufile-support --disable-dbus-support --without-readline
 # --enable-debug
 make -j`nproc`
 cd ..
