diff --git a/common/lifecyleMgr.go b/common/lifecyleMgr.go
index 57ba87d7..ed0a5f9e 100644
--- a/common/lifecyleMgr.go
+++ b/common/lifecyleMgr.go
@@ -372,7 +372,9 @@ func (lcm *lifecycleMgr) Exit(o OutputBuilder, applicationExitCode ExitCode) {
 		exitCode:   applicationExitCode,
 	}
 
-	AzcopyCurrentJobLogger.CloseLog()
+  if AzcopyCurrentJobLogger != nil {
+	  AzcopyCurrentJobLogger.CloseLog()
+  }
 	if applicationExitCode != EExitCode.NoExit() {
 		// stall forever until the success message is printed and program exits
 		lcm.SurrenderControl()
