diff -Nurp terminus-font-4.49.1/75-yes-terminus.conf terminus-font-4.49.1-fix/75-yes-terminus.conf
--- terminus-font-4.49.1/75-yes-terminus.conf	2014-10-12 21:26:56.000000000 +0200
+++ terminus-font-4.49.1-fix/75-yes-terminus.conf	2025-06-30 01:32:01.115180075 +0200
@@ -7,6 +7,9 @@
    <pattern>
      <patelt name="family"><string>Terminus</string></patelt>
    </pattern>
+   <pattern>
+     <patelt name="family"><string>xos4 Terminus</string></patelt>
+   </pattern>
   </acceptfont>
  </selectfont>
 </fontconfig>
