--- gcc/gcc/ubsan.c	2018-02-04 21:13:39.853992701 +0100
+++ gcc/gcc/ubsan.c	2018-02-04 21:12:44.957195790 +0100
@@ -1469,7 +1469,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
