diff --git a/include/message_filters/synchronizer.h b/include/message_filters/synchronizer.h
index 7891890..1c14a6f 100644
--- a/include/message_filters/synchronizer.h
+++ b/include/message_filters/synchronizer.h
@@ -355,7 +355,7 @@ private:
   template<int i>
   void cb(const typename mpl::at_c<Events, i>::type& evt)
   {
-    this->add<i>(evt);
+    this->template add<i>(evt);
   }
 
   uint32_t queue_size_;
