From dbbc4fa3b09a4a64f73e783d726699107139fb1a Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Thu, 15 Dec 2016 17:33:41 +0100
Subject: [PATCH] Fix import_array return type error

---
 python/lcmaes.cc | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/python/lcmaes.cc b/python/lcmaes.cc
index ac95acb..d8774b6 100644
--- a/python/lcmaes.cc
+++ b/python/lcmaes.cc
@@ -224,13 +224,29 @@ BOOST_PYTHON_FUNCTION_OVERLOADS(make_parameters_default_pwqb,make_parameters<Gen
 //BOOST_PYTHON_FUNCTION_OVERLOADS(make_parameters_default_ls,make_parameters<GenoPheno<NoBoundStrategy,linScalingStrategy>>,3,5) // prevented by bug in Boost Python
 //BOOST_PYTHON_FUNCTION_OVERLOADS(make_parameters_default_pwb_ls,make_parameters<GenoPheno<pwqBoundStrategy,linScalingStrategy>>,3,5)
 
+
+inline
+#if PY_VERSION_HEX >= 0x03000000
+int
+#else
+void
+#endif
+init_numpy()
+{
+  import_array();
+#if PY_VERSION_HEX >= 0x03000000
+  return 0;
+#endif
+}
+
+
 BOOST_PYTHON_MODULE(lcmaes)
 {
   // disables C++ signatures in the Python module documentation
   docstring_options local_docstring_options(true,true,false);
 
-  import_array(); // numpy.
-  
+  init_numpy();
+
   /*- parameters object and maker -*/
   class_<CMAParameters<GenoPheno<NoBoundStrategy>>>("CMAParametersNB","CMAParameters object for problems with unbounded function parameters")
     .def("initialize_parameters", &CMAParameters<GenoPheno<NoBoundStrategy>>::initialize_parameters,"initialize required CMA parameters based on dim, lambda, x0 and sigma")
