Index: doxyparse-upstream-1.5.9/src/code.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/code.l
+++ doxyparse-upstream-1.5.9/src/code.l
@@ -3494,7 +3494,7 @@ void parseCCode(CodeOutputInterface &od,
 
 void codeFreeScanner()
 {
-#if defined(YY_FLEX_SUBMINOR_VERSION) 
+#if defined(YY_FLEX_SUBMINOR_VERSION) || defined(YY_FLEX_MINOR_VERSION)
   if (g_lexInit)
   {
     codeYYlex_destroy();
@@ -3504,11 +3504,11 @@ void codeFreeScanner()
 
 
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
   void codeYYdummy() { yy_flex_realloc(0,0); } 
 }
-#elif YY_FLEX_SUBMINOR_VERSION<33
+#elif YY_FLEX_SUBMINOR_VERSION<33 && YY_FLEX_MINOR_VERSION < 6
 #error "You seem to be using a version of flex newer than 2.5.4 but older than 2.5.33. These versions do NOT work with doxygen! Please use version <=2.5.4 or >=2.5.33 or expect things to be parsed wrongly!"
 #endif
 
Index: doxyparse-upstream-1.5.9/src/commentcnv.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/commentcnv.l
+++ doxyparse-upstream-1.5.9/src/commentcnv.l
@@ -785,7 +785,7 @@ void convertCppComments(BufStr *inBuf,Bu
 
 
 //----------------------------------------------------------------------------
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
     void commentcnvYYdummy() { yy_flex_realloc(0,0); } 
 }
Index: doxyparse-upstream-1.5.9/src/commentscan.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/commentscan.l
+++ doxyparse-upstream-1.5.9/src/commentscan.l
@@ -949,7 +949,7 @@ RCSTAG    "$"{ID}":"[^\n$]+"$"
 					      // but we need to know the position in the input buffer where this 
 					      // rule matched.
 					      // for flex 2.5.33+ we should use YY_CURRENT_BUFFER_LVALUE
-#if YY_FLEX_MINOR_VERSION>=5 && YY_FLEX_SUBMINOR_VERSION>=33
+#if (YY_FLEX_MINOR_VERSION==5 && YY_FLEX_SUBMINOR_VERSION>=33) || YY_FLEX_MINOR_VERSION > 5
 					      inputPosition=prevPosition + yy_bp - YY_CURRENT_BUFFER_LVALUE->yy_ch_buf;
 #else
 					      inputPosition=prevPosition + yy_bp - yy_current_buffer->yy_ch_buf;
@@ -2631,7 +2631,7 @@ static void groupAddDocs(Entry *e,const
 }
 
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 //----------------------------------------------------------------------------
 extern "C" { // some bogus code to keep the compiler happy
   void commentScanYYdummy() { yy_flex_realloc(0,0); } 
Index: doxyparse-upstream-1.5.9/src/declinfo.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/declinfo.l
+++ doxyparse-upstream-1.5.9/src/declinfo.l
@@ -346,7 +346,7 @@ int main()
 }
 #endif
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 //----------------------------------------------------------------------------
 extern "C" { // some bogus code to keep the compiler happy
   void declinfoYYdummy() { yy_flex_realloc(0,0); } 
Index: doxyparse-upstream-1.5.9/src/defargs.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/defargs.l
+++ doxyparse-upstream-1.5.9/src/defargs.l
@@ -470,7 +470,7 @@ void stringToArgumentList(const char *ar
   //printf("stringToArgumentList(%s) result=%s\n",argsString,argListToString(al).data());
 }
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
   void defargsYYdummy() { yy_flex_realloc(0,0); } 
 }
Index: doxyparse-upstream-1.5.9/src/fortrancode.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/fortrancode.l
+++ doxyparse-upstream-1.5.9/src/fortrancode.l
@@ -1006,11 +1006,11 @@ void parseFortranCode(CodeOutputInterfac
   return;
 }
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
   void fcodeYYdummy() { yy_flex_realloc(0,0); } 
 }
-#elif YY_FLEX_SUBMINOR_VERSION<33
+#elif YY_FLEX_SUBMINOR_VERSION<33 && YY_FLEX_MINOR_VERSION < 6
 #error "You seem to be using a version of flex newer than 2.5.4 but older than 2.5.33. These versions do NOT work with doxygen! Please use version <=2.5.4 or >=2.5.33 or expect things to be parsed wrongly!"
 #else
 extern "C" { // some bogus code to keep the compiler happy
Index: doxyparse-upstream-1.5.9/src/fortranscanner.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/fortranscanner.l
+++ doxyparse-upstream-1.5.9/src/fortranscanner.l
@@ -1758,7 +1758,7 @@ static void scanner_abort()
 
 //----------------------------------------------------------------------------
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 //----------------------------------------------------------------------------
 extern "C" { // some bogus code to keep the compiler happy
   void fscannerYYdummy() { yy_flex_realloc(0,0); yy_top_state(); } 
Index: doxyparse-upstream-1.5.9/src/pre.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/pre.l
+++ doxyparse-upstream-1.5.9/src/pre.l
@@ -2517,7 +2517,7 @@ void preprocessFile(const char *fileName
 
 void preFreeScanner()
 {
-#if defined(YY_FLEX_SUBMINOR_VERSION) 
+#if defined(YY_FLEX_SUBMINOR_VERSION) || defined(YY_FLEX_MINOR_VERSION)
   if (g_lexInit)
   {
     preYYlex_destroy();
@@ -2525,7 +2525,7 @@ void preFreeScanner()
 #endif
 }
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
 //  int  preYYwrap() { return 1 ; }
   void preYYdummy() { yy_flex_realloc(0,0); } 
Index: doxyparse-upstream-1.5.9/src/pycode.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/pycode.l
+++ doxyparse-upstream-1.5.9/src/pycode.l
@@ -1450,11 +1450,11 @@ void parsePythonCode(CodeOutputInterface
 }
 
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
   void pycodeYYdummy() { yy_flex_realloc(0,0); } 
 }
-#elif YY_FLEX_SUBMINOR_VERSION<33
+#elif YY_FLEX_SUBMINOR_VERSION<33 && YY_FLEX_MINOR_VERSION < 6
 #error "You seem to be using a version of flex newer than 2.5.4. These are currently incompatible with 2.5.4, and do NOT work with doxygen! Please use version 2.5.4 or expect things to be parsed wrongly! A bug report has been submitted (#732132)."
 #endif
 
Index: doxyparse-upstream-1.5.9/src/pyscanner.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/pyscanner.l
+++ doxyparse-upstream-1.5.9/src/pyscanner.l
@@ -1650,7 +1650,7 @@ void PythonLanguageScanner::resetCodePar
 
 //----------------------------------------------------------------------------
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 //----------------------------------------------------------------------------
 extern "C" { // some bogus code to keep the compiler happy
   void pyscannerYYdummy() { yy_flex_realloc(0,0); } 
Index: doxyparse-upstream-1.5.9/src/scanner.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/scanner.l
+++ doxyparse-upstream-1.5.9/src/scanner.l
@@ -5720,7 +5720,7 @@ void CLanguageScanner::parsePrototype(co
 
 //----------------------------------------------------------------------------
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 //----------------------------------------------------------------------------
 extern "C" { // some bogus code to keep the compiler happy
   void scannerYYdummy() { yy_flex_realloc(0,0); } 
Index: doxyparse-upstream-1.5.9/src/vhdlcode.l
===================================================================
--- doxyparse-upstream-1.5.9.orig/src/vhdlcode.l
+++ doxyparse-upstream-1.5.9/src/vhdlcode.l
@@ -1586,7 +1586,7 @@ void parseVhdlCode(CodeOutputInterface &
 
 void codeFreeVhdlScanner()
 {
-#if defined(YY_FLEX_SUBMINOR_VERSION) 
+#if defined(YY_FLEX_SUBMINOR_VERSION) || defined(YY_FLEX_MINOR_VERSION)
   if (g_lexInit)
   {
     vhdlcodeYYlex_destroy();
@@ -1594,11 +1594,11 @@ void codeFreeVhdlScanner()
 #endif
 }
 
-#if !defined(YY_FLEX_SUBMINOR_VERSION) 
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || !defined(YY_FLEX_MINOR_VERSION)
 extern "C" { // some bogus code to keep the compiler happy
   void vhdlcodeYYdummy() { yy_flex_realloc(0,0); } 
 }
-#elif YY_FLEX_SUBMINOR_VERSION<33
+#elif YY_FLEX_SUBMINOR_VERSION<33 && YY_FLEX_MINOR_VERSION < 6
 #error "You seem to be using a version of flex newer than 2.5.4 but older than 2.5.33. These versions do NOT work with doxygen! Please use version <=2.5.4 or >=2.5.33 or expect things to be parsed wrongly!"
 #endif
 
