diff --git a/desktop-app/package.json b/desktop-app/package.json
index 9ff6d1c..15827c5 100644
--- a/desktop-app/package.json
+++ b/desktop-app/package.json
@@ -147,8 +147,7 @@
     },
     "linux": {
       "target": [
-        "AppImage",
-        "rpm"
+        "AppImage"
       ],
       "category": "Development"
     },
diff --git a/desktop-app/scripts/generate-checksums.js b/desktop-app/scripts/generate-checksums.js
index 382b54a..77dcdc9 100644
--- a/desktop-app/scripts/generate-checksums.js
+++ b/desktop-app/scripts/generate-checksums.js
@@ -5,12 +5,7 @@ const pkg = require('../package.json');
 
 const version = pkg.version;
 const requiredFiles = [
-  `Responsively-App-${version}.x86_64.rpm`,
-  `ResponsivelyApp-${version}-mac.zip`,
   `ResponsivelyApp-${version}.AppImage`,
-  `ResponsivelyApp-${version}.dmg`,
-  `ResponsivelyApp ${version}.exe`,
-  `ResponsivelyApp Setup ${version}.exe`,
 ];
 
 function hashFile(file, algorithm = 'sha512', encoding = 'hex', options = {}) {
