diff -r -U 0 a/nuxhash/nuxhash/gui/benchmarks.py b/nuxhash/nuxhash/gui/benchmarks.py
--- a/nuxhash/nuxhash/gui/benchmarks.py	2022-01-11 09:27:48.765355026 -0500
+++ b/nuxhash/nuxhash/gui/benchmarks.py	2022-01-11 09:30:56.294923624 -0500
@@ -7 +7 @@
-from wx.lib.pubsub import pub
+from pubsub import pub
diff -r -U 0 a/nuxhash/nuxhash/gui/main.py b/nuxhash/nuxhash/gui/main.py
--- a/nuxhash/nuxhash/gui/main.py	2022-01-11 09:27:48.765355026 -0500
+++ b/nuxhash/nuxhash/gui/main.py	2022-01-11 09:31:02.677207091 -0500
@@ -10 +10 @@
-from wx.lib.pubsub import pub
+from pubsub import pub
diff -r -U 0 a/nuxhash/nuxhash/gui/mining.py b/nuxhash/nuxhash/gui/mining.py
--- a/nuxhash/nuxhash/gui/mining.py	2022-01-11 09:27:48.765355026 -0500
+++ b/nuxhash/nuxhash/gui/mining.py	2022-01-11 09:31:10.756562457 -0500
@@ -12 +12 @@
-from wx.lib.pubsub import pub
+from pubsub import pub
@@ -91,2 +91 @@
-        bottomSizer.Add(self._StartStop, wx.SizerFlags().Expand()
-                                                        .Center())
+        bottomSizer.Add(self._StartStop, wx.SizerFlags().Center())
diff -r -U 0 a/nuxhash/nuxhash/gui/settings.py b/nuxhash/nuxhash/gui/settings.py
--- a/nuxhash/nuxhash/gui/settings.py	2022-01-11 09:27:48.765355026 -0500
+++ b/nuxhash/nuxhash/gui/settings.py	2022-01-11 09:31:20.455984024 -0500
@@ -5 +5 @@
-from wx.lib.pubsub import pub
+from pubsub import pub
