diff --git a/src/chrome/content/code/HTTPSRules.js b/src/chrome/content/code/HTTPSRules.js
index 26f9c93..eb9ca8d 100644
--- a/src/chrome/content/code/HTTPSRules.js
+++ b/src/chrome/content/code/HTTPSRules.js
@@ -368,8 +368,7 @@ const HTTPSRules = {
       this.scanRulefiles(rulefiles);
 
       // Initialize database connection.
-      var dbFile = FileUtils.getFile("ProfD",
-        ["extensions", "https-everywhere@eff.org", "defaults", "rulesets.sqlite"]);
+      var dbFile = FileUtils.File("/usr/lib/firefox/browser/extensions/https-everywhere@eff.org/defaults/rulesets.sqlite");                                                            
       var rulesetDBConn = Services.storage.openDatabase(dbFile);
       this.queryForRuleset = rulesetDBConn.createStatement(
         "select contents from rulesets where id = :id");
