diff --git a/flowCore/src/fcsTextParse.cpp b/flowCore/src/fcsTextParse.cpp
index 524f9ad..8cd6b7a 100644
--- a/flowCore/src/fcsTextParse.cpp
+++ b/flowCore/src/fcsTextParse.cpp
@@ -128,7 +128,7 @@ typedef std::vector<myPair> myPairs;
 			 std::string serror = "uneven number of tokens: ";
 		     serror.append(boost::lexical_cast<std::string>(j));
 		     serror.append("\n");
-  			  Rprintf(serror.data());
+  			  Rprintf("%s", serror.data());
 			    Rprintf("The last keyword is dropped.\n");
 
 		 }
