diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,7 +10,7 @@ log = "0.3"
 env_logger = "0.3"
 rustc-serialize = "0.3"
 docopt = "0.6"
-ctrlc = { git = "https://github.com/tomusdrw/rust-ctrlc.git" }
+ctrlc = { git = "https://github.com/quininer/rust-ctrlc.git", branch = "parity" }
 clippy = { version = "0.0.42", optional = true }
 ethcore-util = { path = "util" }
 ethcore = { path = "ethcore" }
