diff --git a/example/oomd.json b/example/oomd.json
index 664aed5..859f4ba 100644
--- a/example/oomd.json
+++ b/example/oomd.json
@@ -7,7 +7,7 @@
                 {"sshd.service": { "max_usage": "inf" } }
             ],
             "oomdetector": "default",
-            "oomkiller": "fbje"
+            "oomkiller": "default"
         },
         {
             "target": "workload.slice",
diff --git a/meson.build b/meson.build
index f12f067..2842453 100644
--- a/meson.build
+++ b/meson.build
@@ -39,14 +39,16 @@ srcs = files('''
 
 deps = [dependency('jsoncpp'),
         dependency('threads')]
-oomd_lib = library('oomd',
+oomd_lib = shared_library('oomd',
                    srcs,
                    include_directories : inc,
+                   install: true,
                    dependencies : deps)
 executable('oomd_bin',
            files('Main.cpp'),
            include_directories : inc,
            dependencies : deps,
+           install: true,
            link_with : oomd_lib)
 
 gtest_dep = dependency('gtest', main : true, required : false)
