diff --git a/CMakeLists.txt b/CMakeLists.txt
index df44a93..594cb80 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,9 @@ if (MSVC)
   
   set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS   _CRT_SECURE_NO_WARNINGS)
 else(MSVC)
-  list(APPEND CMAKE_CXX_FLAGS "-Wall -Wunused-variable -Wno-long-long")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wunused-variable -Wno-long-long")
    #-pedantic 
-  list(APPEND CMAKE_C_FLAGS "-Wall -Wunused-variable -Wno-long-long")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wunused-variable -Wno-long-long")
 endif(MSVC)
 
 # Include all the necessary files for macros
@@ -49,6 +49,7 @@ option(LITESQL_WITH_DOCS "generate Documentation" OFF)
 option(LITESQL_WITH_SAX_PARSER "use new SAX parser" ON)
 option(LITESQL_WITH_UI "build also GUI-Version of generator" OFF)
 option(LITESQL_WITH_TESTS "enable testing" ON)
+option(LITESQL_WITH_EXAMPLE "enable example" ON)
 option(CMAKE_SUPPRESS_REGENERATION "suppress rules to re-run cmake on changes (warning: no dependency checks!)?" OFF)
 
 if(LITESQL_WITH_TESTS)
