From cc7c6ac5400bac438bdd12c54dcedb503446fde6 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Wed, 26 Feb 2020 17:18:52 +0100
Subject: [PATCH] fix-link

---
 benchmark/CMakeLists.txt | 1 +
 unittest/CMakeLists.txt  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/benchmark/CMakeLists.txt b/benchmark/CMakeLists.txt
index 0e5160e3..e1a0b2eb 100644
--- a/benchmark/CMakeLists.txt
+++ b/benchmark/CMakeLists.txt
@@ -15,6 +15,7 @@ FOREACH(BENCHMARK_NAME ${${PROJECT_NAME}_BENCHMARK})
   PKG_CONFIG_USE_DEPENDENCY(${BENCHMARK_NAME} pinocchio)
   PKG_CONFIG_USE_DEPENDENCY(${BENCHMARK_NAME} example-robot-data)
   TARGET_LINK_LIBRARIES(${BENCHMARK_NAME} ${PROJECT_NAME})
+  TARGET_LINK_LIBRARIES(${BENCHMARK_NAME} ${PYTHON_LIBRARIES})
 ENDFOREACH(BENCHMARK_NAME ${${PROJECT_NAME}_BENCHMARK})
 
 FOREACH(BENCHMARK_NAME ${${PROJECT_NAME}_BENCHMARK})
diff --git a/unittest/CMakeLists.txt b/unittest/CMakeLists.txt
index ef2c298d..6b5ce795 100644
--- a/unittest/CMakeLists.txt
+++ b/unittest/CMakeLists.txt
@@ -52,6 +52,7 @@ MACRO(ADD_CPP_UNIT_TEST NAME PKGS)
 
   TARGET_LINK_LIBRARIES(${unittest_name} ${PROJECT_NAME})
   TARGET_LINK_LIBRARIES(${unittest_name} ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
+  TARGET_LINK_LIBRARIES(${unittest_name} ${PYTHON_LIBRARIES})
 
   ADD_TEST_CFLAGS(${unittest_name} '-DCROCODDYL_SOURCE_DIR=\\\"${${PROJECT_NAME}_SOURCE_DIR}\\\"')
 ENDMACRO(ADD_CPP_UNIT_TEST NAME PKGS)
-- 
2.25.1

