--- cpp/include/rmm/device_uvector.hpp	2025-10-09 00:21:15.000000000 +0800
+++ cpp/include/rmm/device_uvector.hpp.new	2025-10-09 18:36:30.308616855 +0800
@@ -76,8 +76,6 @@
  */
 template <typename T>
 class device_uvector {
-  static_assert(std::is_trivially_copyable_v<T>,
-                "device_uvector only supports types that are trivially copyable.");
 
  public:
   using value_type = T;            ///< T; stored value type
