diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23b2230..c9b1bed 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -102,6 +102,7 @@ set(LUXMARK_SRCS
 ADD_EXECUTABLE(luxmark WIN32 ${LUXMARK_SRCS})
 
 TARGET_LINK_LIBRARIES(luxmark ${ALL_LUXCORE_LIBRARIES} ${Boost_LIBRARIES} ${Qt5_LIBRARIES} ${OPENGL_gl_LIBRARY} ${OPENCL_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxmark openvdb.a OpenImageIO python3)
 if (WIN32)
 	# This is needed by Boost 1.67 but is not found automatically
     TARGET_LINK_LIBRARIES(luxmark bcrypt.lib)

