--- a/src/vs/code/electron-main/app.ts	2020-10-29 17:51:27.330706163 +0800
+++ b/src/vs/code/electron-main/app.ts	2020-11-24 14:11:33.058005352 +0800
@@ -322,6 +322,7 @@
 	}

 	async startup(): Promise<void> {
+		await new Promise((resolve) => setTimeout(resolve, 400));
 		this.logService.debug('Starting VS Code');
 		this.logService.debug(`from: ${this.environmentService.appRoot}`);
 		this.logService.debug('args:', this.environmentService.args);
