diff --git a/meshroom/ui/qml/Viewer3D/MediaLoader.qml b/meshroom/ui/qml/Viewer3D/MediaLoader.qml
index 4c5606ea..a1323e72 100644
--- a/meshroom/ui/qml/Viewer3D/MediaLoader.qml
+++ b/meshroom/ui/qml/Viewer3D/MediaLoader.qml
@@ -76,6 +76,7 @@ import Utils 1.0
 
         // Media loader available
         if (component) {
+            console.debug(component.errorString())
             object = component.createObject(root, {"source": source})
         }
     }
