diff --git a/doc/hddsupertool-p1.texi b/doc/hddsupertool-p1.texi
index 8ecec4f..b34a9f0 100644
--- a/doc/hddsupertool-p1.texi
+++ b/doc/hddsupertool-p1.texi
@@ -376,7 +376,7 @@ sudo /home/hddsupertool
 
 You may notice that examples did not include a drive or script. As long as the
 hddscripts folder is either in the current working directory, or has been
-copied to /usr/local/bin/hddscripts, hddsupertool will start with a menu driven
+copied to /usr/bin/hddscripts, hddsupertool will start with a menu driven
 system. You can still run hddsupertool with manual options to declare which
 drive and script with a command such as the following:
 @example
@@ -528,7 +528,7 @@ with the BIOS set to IDE. You are stuck using passthrough mode in this case.
 The script file that contains the commands to be sent to the drive. If this is
 not set, then it will look for a file named "hddmenu" first in the current
 directory, then in the sub directory "hddscripts/", and then in
-"/usr/local/bin/hddscripts/". All subscripts will be loaded from the directory
+"/usr/bin/hddscripts/". All subscripts will be loaded from the directory
 where hddmenu was found. If no script file is found then the program will exit.
 
 @item -i @var{spaces}
diff --git a/doc/hddsupertool.texi b/doc/hddsupertool.texi
index b178357..313ae26 100644
--- a/doc/hddsupertool.texi
+++ b/doc/hddsupertool.texi
@@ -376,7 +376,7 @@ sudo /home/hddsupertool
 
 You may notice that examples did not include a drive or script. As long as the
 hddscripts folder is either in the current working directory, or has been
-copied to /usr/local/bin/hddscripts, hddsupertool will start with a menu driven
+copied to /usr/bin/hddscripts, hddsupertool will start with a menu driven
 system. You can still run hddsupertool with manual options to declare which
 drive and script with a command such as the following:
 @example
@@ -528,7 +528,7 @@ with the BIOS set to IDE. You are stuck using passthrough mode in this case.
 The script file that contains the commands to be sent to the drive. If this is
 not set, then it will look for a file named "hddmenu" first in the current
 directory, then in the sub directory "hddscripts/", and then in
-"/usr/local/bin/hddscripts/". All subscripts will be loaded from the directory
+"/usr/bin/hddscripts/". All subscripts will be loaded from the directory
 where hddmenu was found. If no script file is found then the program will exit.
 
 @item -i @var{spaces}
diff --git a/src/clone_gui2.c b/src/clone_gui2.c
index 3cefba7..a5eefa2 100644
--- a/src/clone_gui2.c
+++ b/src/clone_gui2.c
@@ -1318,9 +1318,9 @@ void import_language_file_ccc(void)
                                         GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
                                         GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
                                         NULL);
-  if ( access( "/usr/local/share/doc/hddsuperclone/Language/English", F_OK ) == 0 )
+  if ( access( "/usr/share/doc/hddsuperclone/Language/English", F_OK ) == 0 )
   {
-    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/local/share/doc/hddsuperclone/Language");
+    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/share/doc/hddsuperclone/Language");
     g_print ("found installed languages\n");
   }
   else if ( access( "Language/English", F_OK ) == 0 )
@@ -5901,7 +5901,7 @@ void help_html_ccc(void)
   if (pid == 0)
   {
     // make sure this exits and does not return
-    system("cp -f /usr/local/share/doc/hddsuperclone/hddsuperclone.html /tmp/");
+    system("cp -f /usr/share/doc/hddsuperclone/hddsuperclone.html /tmp/");
     system("sudo -E -H -P -u $(who | head -1 | awk '{print $1}') xdg-open /tmp/hddsuperclone.html > /dev/null 2>&1");
     exit(0);
   }
@@ -5919,7 +5919,7 @@ void help_text_ccc(void)
   if (pid == 0)
   {
     // make sure this exits and does not return
-    system("cp -f /usr/local/share/doc/hddsuperclone/hddsuperclone.txt /tmp/");
+    system("cp -f /usr/share/doc/hddsuperclone/hddsuperclone.txt /tmp/");
     system("sudo -E -H -P -u $(who | head -1 | awk '{print $1}') xdg-open /tmp/hddsuperclone.txt > /dev/null 2>&1");
     exit(0);
   }
diff --git a/src/clone_gui3.c b/src/clone_gui3.c
index ca1c9bf..012dd19 100644
--- a/src/clone_gui3.c
+++ b/src/clone_gui3.c
@@ -1309,9 +1309,9 @@ void import_language_file_ccc(void)
                                         GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
                                         GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
                                         NULL);
-  if ( access( "/usr/local/share/doc/hddsuperclone/Language/English", F_OK ) == 0 )
+  if ( access( "/usr/share/doc/hddsuperclone/Language/English", F_OK ) == 0 )
   {
-    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/local/share/doc/hddsuperclone/Language");
+    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/share/doc/hddsuperclone/Language");
     g_print ("found installed languages\n");
   }
   else if ( access( "Language/English", F_OK ) == 0 )
@@ -5863,7 +5863,7 @@ void help_html_ccc(void)
   if (pid == 0)
   {
     // make sure this exits and does not return
-    system("cp -f /usr/local/share/doc/hddsuperclone/hddsuperclone.html /tmp/");
+    system("cp -f /usr/share/doc/hddsuperclone/hddsuperclone.html /tmp/");
     system("sudo -E -H -P -u $(who | head -1 | awk '{print $1}') xdg-open /tmp/hddsuperclone.html > /dev/null 2>&1");
     exit(0);
   }
@@ -5881,7 +5881,7 @@ void help_text_ccc(void)
   if (pid == 0)
   {
     // make sure this exits and does not return
-    system("cp -f /usr/local/share/doc/hddsuperclone/hddsuperclone.txt /tmp/");
+    system("cp -f /usr/share/doc/hddsuperclone/hddsuperclone.txt /tmp/");
     system("sudo -E -H -P -u $(who | head -1 | awk '{print $1}') xdg-open /tmp/hddsuperclone.txt > /dev/null 2>&1");
     exit(0);
   }
diff --git a/src/hddscviewer2.c b/src/hddscviewer2.c
index fc171fa..9e7541c 100644
--- a/src/hddscviewer2.c
+++ b/src/hddscviewer2.c
@@ -3635,9 +3635,9 @@ void import_language_file(void)
                                         GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
                                         GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
                                         NULL);
-  if ( access( "/usr/local/share/doc/hddscviewer/Language/English", F_OK ) == 0 )
+  if ( access( "/usr/share/doc/hddscviewer/Language/English", F_OK ) == 0 )
   {
-    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/local/share/doc/hddscviewer/Language");
+    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/share/doc/hddscviewer/Language");
     g_print ("found installed languages\n");
   }
   else if ( access( "Language/English", F_OK ) == 0 )
diff --git a/src/hddscviewer3.c b/src/hddscviewer3.c
index b080de7..47e50d2 100644
--- a/src/hddscviewer3.c
+++ b/src/hddscviewer3.c
@@ -2603,9 +2603,9 @@ void import_language_file(void)
                                         GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
                                         GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
                                         NULL);
-  if ( access( "/usr/local/share/doc/hddscviewer/Language/English", F_OK ) == 0 )
+  if ( access( "/usr/share/doc/hddscviewer/Language/English", F_OK ) == 0 )
   {
-    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/local/share/doc/hddscviewer/Language");
+    gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (dialog), "/usr/share/doc/hddscviewer/Language");
     g_print ("found installed languages\n");
   }
   else if ( access( "Language/English", F_OK ) == 0 )
diff --git a/src/hddsuperclone.c b/src/hddsuperclone.c
index d8a9651..d9fcf27 100644
--- a/src/hddsuperclone.c
+++ b/src/hddsuperclone.c
@@ -18906,14 +18906,14 @@ int install_license_ccc (void)
   }
   memcpy (temp_data + 14, &checksum, 2);
 
-  DIR* dir = opendir("/usr/local/etc");
+  DIR* dir = opendir("/etc");
   if (dir)
   {
     closedir(dir);
   }
   else
   {
-    system("mkdir /usr/local/etc");
+    system("mkdir /etc");
   }
   FILE *writefile = fopen(license_file_name_ccc, "wb");
   if (writefile == NULL)
diff --git a/src/hddsuperclone.h b/src/hddsuperclone.h
index 952b49b..f427465 100644
--- a/src/hddsuperclone.h
+++ b/src/hddsuperclone.h
@@ -512,10 +512,10 @@ char *get_smart_attribute_name_ccc(int id);
 
 
 unsigned char ccc_mac_address_ccc[6];
-char *install_directory_ccc = "/usr/local/bin/";
-//char *license_file_name_ccc = "/usr/local/etc/hddsuperclone.lic";
-char *license_file_name_ccc = "/usr/local/etc/dhsdpurelceno.cil";
-char *time_check_file_name_ccc = "/usr/local/etc/dhsdpurelceno.kct";
+char *install_directory_ccc = "/usr/bin/";
+//char *license_file_name_ccc = "/etc/hddsuperclone.lic";
+char *license_file_name_ccc = "/etc/dhsdpurelceno.cil";
+char *time_check_file_name_ccc = "/etc/dhsdpurelceno.kct";
 long long blacklist_check_interval_ccc = 128000;    // 128000 seconds is about 1 1/2 day
 bool path_fail_ccc = false;
 char start_working_directory_ccc[PATH_MAX];
diff --git a/src/tool.h b/src/tool.h
index dc2f731..21281f2 100644
--- a/src/tool.h
+++ b/src/tool.h
@@ -9,10 +9,10 @@ unsigned int number_variable_rows_ccc = 100;
 unsigned int string_variable_rows_ccc = 100;
 char *menufile1_ccc = "hddmenu";
 char *menufile2_ccc = "hddscripts/hddmenu";
-char *menufile3_ccc = "/usr/local/bin/hddscripts/hddmenu";
+char *menufile3_ccc = "/usr/bin/hddscripts/hddmenu";
 char script_directory1_ccc[PATH_MAX] = "";
 char script_directory2_ccc[PATH_MAX] = "hddscripts/";
-char script_directory3_ccc[PATH_MAX] = "/usr/local/bin/hddscripts/";
+char script_directory3_ccc[PATH_MAX] = "/usr/bin/hddscripts/";
 char script_directory4_ccc[PATH_MAX] = "";
 char *current_script_directory_ccc = "";
 char full_script_path_ccc[PATH_MAX];
