diff -aur a/deps/juce/modules/juce_gui_basics/juce_gui_basics.h b/deps/juce/modules/juce_gui_basics/juce_gui_basics.h
--- a/deps/juce/modules/juce_gui_basics/juce_gui_basics.h	2022-04-26 01:54:23.000000000 +0200
+++ b/deps/juce/modules/juce_gui_basics/juce_gui_basics.h	2022-05-25 16:33:14.962989597 +0200
@@ -54,6 +54,7 @@
 #pragma once
 #define JUCE_GUI_BASICS_H_INCLUDED
 
+#include <utility>
 #include <juce_graphics/juce_graphics.h>
 #include <juce_data_structures/juce_data_structures.h>
 
