diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,7 @@ add_definitions( -DUNIX )
 
 include( CheckCSourceCompiles )
 include( CheckLibraryExists )
+include_directories(BEFORE SYSTEM "/usr/include/openssl-1.0")
 
 set(
 	RELVER "2.2.1" )
@@ -270,7 +271,7 @@ endif( NOT PATH_INC_CRYPTO )
 
 find_library(
 	PATH_LIB_CRYPTO
-	NAMES "crypto"
+	NAMES "libcrypto.so.1.0.0"
 	PATHS ${SEARCH_LIB} )
 
 if( NOT PATH_LIB_CRYPTO )
diff --git a/source/ikec/CMakeLists.txt b/source/ikec/CMakeLists.txt
--- a/source/ikec/CMakeLists.txt
+++ b/source/ikec/CMakeLists.txt
@@ -32,7 +32,7 @@ target_link_libraries(
 	ss_idb
 	ss_ith
 	ss_log
-	crypto
+	${PATH_LIB_CRYPTO}
 	pthread
 	edit )
 
diff --git a/source/iked/CMakeLists.txt b/source/iked/CMakeLists.txt
--- a/source/iked/CMakeLists.txt
+++ b/source/iked/CMakeLists.txt
@@ -73,7 +73,7 @@ target_link_libraries(
 	ss_ip
 	ss_log
 	ss_pfk
-	crypto
+	${PATH_LIB_CRYPTO}
 	pthread )
 
 if( FUNC_LIB_CRYPT )
diff --git a/source/libike/CMakeLists.txt b/source/libike/CMakeLists.txt
--- a/source/libike/CMakeLists.txt
+++ b/source/libike/CMakeLists.txt
@@ -29,7 +29,7 @@ target_link_libraries(
 	ss_ike
 	ss_idb
 	ss_ith
-	crypto )
+	${PATH_LIB_CRYPTO} )
 
 set_target_properties(
 	ss_ike PROPERTIES
diff --git a/source/qikea/CMakeLists.txt b/source/qikea/CMakeLists.txt
--- a/source/qikea/CMakeLists.txt
+++ b/source/qikea/CMakeLists.txt
@@ -55,7 +55,7 @@ target_link_libraries(
 	ss_idb
 	ss_ith
 	ss_log
-	crypto
+	${PATH_LIB_CRYPTO}
 	pthread
 	${QT_LIBRARIES} )
 
diff --git a/source/qikec/CMakeLists.txt b/source/qikec/CMakeLists.txt
--- a/source/qikec/CMakeLists.txt
+++ b/source/qikec/CMakeLists.txt
@@ -53,7 +53,7 @@ target_link_libraries(
 	ss_idb
 	ss_ith
 	ss_log
-	crypto
+	${PATH_LIB_CRYPTO}
 	pthread
 	${QT_LIBRARIES} )
 
-- 
2.12.2

