diff --git a/api/api_gui_proc.h b/api/api_gui_proc.h
index e7d0d85..7d1600e 100644
--- a/api/api_gui_proc.h
+++ b/api/api_gui_proc.h
@@ -35,9 +35,9 @@ int64_t API_get_gui_from_existing_widget(QWidget *widget);
 
 #if defined(QREGION_H)
 void API_run_paint_event_for_custom_widget(QWidget *widget, QPaintEvent *ev, const QRegion &already_painted_areas);
+bool API_run_custom_gui_paint_function(QWidget *widget, QPainter *p, const QRegion *region, std::function<void(void)> func);
 #endif
 
-bool API_run_custom_gui_paint_function(QWidget *widget, QPainter *p, const QRegion *region, std::function<void(void)> func);
 
 bool API_run_mouse_press_event_for_custom_widget(QWidget *widget, QMouseEvent *ev);
 bool API_run_mouse_move_event_for_custom_widget(QWidget *widget, QMouseEvent *ev);
