diff --git a/pyproject.toml b/pyproject.toml
index 89e4950..5afcf65 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,9 +24,9 @@ packages = [
 ]
 include = [
   # all is an object -> prevent parse issue with dependabot
-  { path="README.md", format =["sdist","wheel"] },
-  { path="LICENSE", format=["sdist","wheel"] },
-  { path="NOTICE", format=["sdist","wheel"] },
+  { path="README.md", format =["sdist"] },
+  { path="LICENSE", format=["sdist"] },
+  { path="NOTICE", format=["sdist"] },
   { path="tests", format=["sdist"] },
   { path="CHANGELOG.md", format=["sdist"] },
   { path="docs", format=["sdist"] },
