diff --git a/func.py b/Spongebobcli.py
similarity index 100%
rename from func.py
rename to Spongebobcli.py
diff --git a/setup.py b/setup.py
index f6376d1..3cdb8a0 100644
--- a/setup.py
+++ b/setup.py
@@ -3,9 +3,11 @@ from setuptools import setup
 setup(
    name='spongebob-cli',
    version='1.1',
-   install_requires=["termcolor", "BeautifulSoup4", "prettytable", "youtube-dl", "halo", "requests"], 
+   install_requires=["termcolor", "BeautifulSoup4", "prettytable", "youtube-dl", "halo", "requests"],
+   py_modules=[
+            'Spongebobcli'
+           ],
    scripts=[
-            'spongebob-cli',
-            'func.py',
+            'spongebob-cli'
            ]
 )
diff --git a/spongebob-cli b/spongebob-cli
index 35f7622..bde798a 100644
--- a/spongebob-cli
+++ b/spongebob-cli
@@ -12,7 +12,7 @@ try:
     from halo import Halo
    
     # local file, func.py
-    from func import httperror_assess
+    from Spongebobcli import httperror_assess
     
 except ImportError or ModuleNotFoundError:
     # Using python color codes instead of termcolor if it isn't found
