diff -aur odamex-src-0.8.3.org/CMakeLists.txt odamex-src-0.8.3/CMakeLists.txt
--- odamex-src-0.8.3.org/CMakeLists.txt	2020-05-10 05:29:36.000000000 +0200
+++ odamex-src-0.8.3/CMakeLists.txt	2020-06-13 22:56:47.027614254 +0200
@@ -28,7 +28,7 @@
 set(PROJECT_COPYRIGHT "2006-2020")
 
 # Use C++ 98/03 for all targets
-set(CMAKE_CXX_STANDARD 98)
+set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # identify the target CPU
