Index: iodev/pcidev.cc
===================================================================
--- iodev/pcidev.cc	(revision 14201)
+++ iodev/pcidev.cc	(working copy)
@@ -242,7 +242,7 @@
   // Check if the device is disabled or not configured
   if (SIM->get_param_num(BXPN_PCIDEV_VENDOR)->get() == 0xffff) {
     BX_INFO(("Host PCI device mapping disabled"));
-    BX_UNREGISTER_DEVICE_DEVMODEL("pcidev");
+    BX_UNREGISTER_DEVICE_DEVMODEL("pcidev", (int)PLUGTYPE_OPTIONAL);
     return;
   }
 
