diff --git a/coz b/coz
index 7e9a1ff..2a1e3b4 100755
--- a/coz
+++ b/coz
@@ -34,8 +34,7 @@ def _coz_run(args):
     exit(1)
 
   env = copy.deepcopy(os.environ)
-  coz_prefix = dirname(dirname(abspath(sys.argv[0])))
-  coz_runtime = coz_prefix + path_sep + 'coz' + path_sep + 'libcoz' + path_sep + 'libcoz.so'
+  coz_runtime = '/usr/share/coz/libcoz.so'
 
   if 'LD_PRELOAD' in env:
     env['LD_PRELOAD'] += ':' + coz_runtime
