diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5303458..1369e97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,6 +122,7 @@ target_compile_definitions(${PLUGIN_NAME} PUBLIC
         )
 
 find_library(pcap REQUIRED)
+find_library(procps REQUIRED)
 
 # 设置目标要使用的链接库
 # 变量 ${DtkWidget_LIBRARIES} 和 ${Qt5Widgets_LIBRARIES} 是在前面执行 find_package 命令时引入的
@@ -132,6 +133,7 @@ target_link_libraries(${PLUGIN_NAME} PRIVATE
         ${DdeDockInterface_LIBRARIES}
         ${CMAKE_CURRENT_SOURCE_DIR}/nethogs/src/libnethogs.a
         pcap
+        procps
         dde_istate_netspeed
         dde_istate_datetime
         dde_istate_cpu
diff --git a/utils/desktop_entry_stat.cpp b/utils/desktop_entry_stat.cpp
index d2bedc6..aac43ce 100644
--- a/utils/desktop_entry_stat.cpp
+++ b/utils/desktop_entry_stat.cpp
@@ -35,7 +35,7 @@ DCORE_USE_NAMESPACE
 
 static const int kUpdateInterval = 1000 * 300; // 5 minutes interval
 
-auto print_err = [](decltype(errno) e, const QString &msg)
+auto print_err_entry = [](decltype(errno) e, const QString &msg)
 {
     qDebug() << QString("Error: [%1] %2, ").arg(e).arg(strerror(e)) << msg;
 };
@@ -98,7 +98,7 @@ void DesktopEntryStat::refresh()
     errno = 0;
     dir = opendir(DESKTOP_ENTRY_PATH);
     if (!dir) {
-        print_err(errno, QString("open %1 failed").arg(DESKTOP_ENTRY_PATH));
+        print_err_entry(errno, QString("open %1 failed").arg(DESKTOP_ENTRY_PATH));
         return;
     }
 
@@ -121,7 +121,7 @@ void DesktopEntryStat::refresh()
         }
     }
     if (errno && !cache.size()) {
-        print_err(errno, QString("read %1 failed").arg(DESKTOP_ENTRY_PATH));
+        print_err_entry(errno, QString("read %1 failed").arg(DESKTOP_ENTRY_PATH));
     }
     closedir(dir);
 
diff --git a/utils/process_stat.cpp b/utils/process_stat.cpp
index 048b2f4..6e9f5fe 100644
--- a/utils/process_stat.cpp
+++ b/utils/process_stat.cpp
@@ -42,7 +42,7 @@
 #define PROC_FD_NAME_PATH       "/proc/%u/fd/%s"
 #define PROC_SCHEDSTAT_PATH     "/proc/%u/schedstat"
 
-auto print_err = [](decltype(errno) e, const QString &msg)
+auto print_err_process = [](decltype(errno) e, const QString &msg)
 {
 //    qDebug() << QString("Error: [%1] %2, ").arg(e).arg(strerror(e)) << msg;
 };
@@ -91,7 +91,7 @@ bool ProcessStat::readProcStats(ProcIterateCallback pfnCallback, void *context)
     errno = 0;
     dp = opendir(PROC_PATH);
     if (!dp) {
-        print_err(errno, "open /proc failed");
+        print_err_process(errno, "open /proc failed");
         return b;
     }
 
@@ -116,7 +116,7 @@ bool ProcessStat::readProcStats(ProcIterateCallback pfnCallback, void *context)
         }
     }
     if (errno) {
-        print_err(errno, "read /proc failed");
+        print_err_process(errno, "read /proc failed");
         closedir(dp);
         return b;
     }
@@ -143,13 +143,13 @@ bool ProcessStat::readStat(ProcStat &ps)
     errno = 0;
     sprintf(path, PROC_STAT_PATH, ps->pid);
     if ((fd = open(path, O_RDONLY)) < 0) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
     sz = read(fd, buf.data(), 1024);
     if (sz < 0) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
         close(fd);
         return b;
     }
@@ -216,13 +216,13 @@ bool ProcessStat::readCmdline(ProcStat &ps)
 
     errno = 0;
     if (!(fp = fopen(path, "r"))) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
     nb = fread(cmd.data(), 1, bsiz - 1, fp);
     if (ferror(fp)) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
         fclose(fp);
         return b;
     }
@@ -263,7 +263,7 @@ void ProcessStat::readEnviron(ProcStat &ps)
     errno = 0;
     fd = open(path, O_RDONLY);
     if (fd < 0) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return;
     }
 
@@ -272,7 +272,7 @@ void ProcessStat::readEnviron(ProcStat &ps)
         sbuf.append(buf, int(nb));
     }
     if (nb == 0 && errno != 0) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
     }
     close(fd);
 
@@ -302,13 +302,13 @@ bool ProcessStat::readSchedStat(ProcStat &ps, struct stat_context &ctx)
 
     errno = 0;
     if ((fd = open(path, O_RDONLY)) < 0) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
     n = read(fd, buf.data(), bsiz - 1);
     if (n < 0) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
         close(fd);
         return b;
     }
@@ -336,7 +336,7 @@ bool ProcessStat::readStatus(ProcStat &ps)
 
     errno = 0;
     if (!(fp = fopen(path, "r)"))) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
@@ -360,7 +360,7 @@ bool ProcessStat::readStatus(ProcStat &ps)
         }
     }
     if (ferror(fp)) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
     } else {
         b = true;
     }
@@ -381,13 +381,13 @@ bool ProcessStat::readStatm(ProcStat &ps, struct stat_context &ctx)
 
     errno = 0;
     if ((fd = open(path, O_RDONLY)) < 0) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
     nr = read(fd, buf, bsiz);
     if (nr < 0) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
         close(fd);
         return b;
     }
@@ -398,7 +398,7 @@ bool ProcessStat::readStatm(ProcStat &ps, struct stat_context &ctx)
     if (nr != 2) {
         ps->rss = 0;
         ps->shm = 0;
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
     } else {
         ps->rss <<= ctx.kb_shift;
         ps->shm <<= ctx.kb_shift;
@@ -418,7 +418,7 @@ bool ProcessStat::readIO(ProcStat &ps)
 
     errno = 0;
     if (!(fp = fopen(path, "r"))) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
@@ -432,7 +432,7 @@ bool ProcessStat::readIO(ProcStat &ps)
         }
     }
     if (ferror(fp)) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
     } else {
         b = true;
     }
@@ -453,7 +453,7 @@ bool ProcessStat::readSockInodes(ProcStat &ps)
 
     errno = 0;
     if (!(dir = opendir(path))) {
-        print_err(errno, QString("open %1 failed").arg(path));
+        print_err_process(errno, QString("open %1 failed").arg(path));
         return b;
     }
 
@@ -469,7 +469,7 @@ bool ProcessStat::readSockInodes(ProcStat &ps)
         }
     }
     if (errno) {
-        print_err(errno, QString("read %1 failed").arg(path));
+        print_err_process(errno, QString("read %1 failed").arg(path));
     } else {
         b = true;
     }
diff --git a/utils/system_stat.cpp b/utils/system_stat.cpp
index 90395fb..bfea2c4 100644
--- a/utils/system_stat.cpp
+++ b/utils/system_stat.cpp
@@ -43,7 +43,7 @@
 
 #define MAX_NAME_LEN 128
 
-auto print_err = [](decltype(errno) e, const QString &msg)
+auto print_err_system = [](decltype(errno) e, const QString &msg)
 {
     qDebug() << QString("Error: [%1] %2, ").arg(e).arg(strerror(e)) << msg;
 };
@@ -58,7 +58,7 @@ bool SystemStat::readLoadAvg(qreal &loadAvg1, qreal &loadAvg5, qreal &loadAvg15)
 
     errno = 0;
     if ((fp = fopen(PROC_PATH_LOADAVG, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_LOADAVG));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_LOADAVG));
         return b;
     }
 
@@ -73,7 +73,7 @@ bool SystemStat::readLoadAvg(qreal &loadAvg1, qreal &loadAvg5, qreal &loadAvg15)
     }
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_LOADAVG));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_LOADAVG));
     }
 
     return b;
@@ -90,7 +90,7 @@ bool SystemStat::readUpTime(qulonglong &uptime)
 
     errno = 0;
     if ((fp = fopen(PROC_PATH_UPTIME, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_UPTIME));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_UPTIME));
         return b;
     }
 
@@ -103,7 +103,7 @@ bool SystemStat::readUpTime(qulonglong &uptime)
     }
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_UPTIME));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_UPTIME));
     }
 
     return b;
@@ -119,7 +119,7 @@ bool SystemStat::readBootTime(time_t &btime)
 
     errno = 0;
     if (!(fp = fopen(PROC_PATH_STAT, "r"))) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_STAT));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_STAT));
         return b;
     }
 
@@ -130,7 +130,7 @@ bool SystemStat::readBootTime(time_t &btime)
         }
     }
     if (ferror(fp)) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_STAT));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_STAT));
     }
     fclose(fp);
 
@@ -152,7 +152,7 @@ bool SystemStat::readCPUStats(CPUStat &cpuStat, CPUStatMap &cpuStatMap)
     cpuStatMap.clear();
 
     if ((fp = fopen(PROC_PATH_CPU, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_CPU));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_CPU));
         return false;
     }
 
@@ -198,7 +198,7 @@ bool SystemStat::readCPUStats(CPUStat &cpuStat, CPUStatMap &cpuStatMap)
     b = !ferror(fp) && b;
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_CPU));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_CPU));
     }
 
     return b;
@@ -213,7 +213,7 @@ bool SystemStat::readMemStats(MemStat &memStat)
     bool b = true;
 
     if ((fp = fopen(PROC_PATH_MEM, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_MEM));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_MEM));
         return false;
     }
 
@@ -283,7 +283,7 @@ bool SystemStat::readMemStats(MemStat &memStat)
     b = !ferror(fp) && b;
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_MEM));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_MEM));
     }
 
     return b;
@@ -315,7 +315,7 @@ bool SystemStat::readDiskIOStats(DiskIOStat &statSum, DiskIOStatMap &statIOMap)
     };
 
     if ((fp = fopen(PROC_PATH_DISK, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_DISK));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_DISK));
         return b;
     }
 
@@ -360,7 +360,7 @@ bool SystemStat::readDiskIOStats(DiskIOStat &statSum, DiskIOStatMap &statIOMap)
     b = !ferror(fp) && b;
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_DISK));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_DISK));
     }
 
     return b;
@@ -376,7 +376,7 @@ bool SystemStat::readNetIfStats(NetIFStat &statSum, NetIFStatMap &statNetIfMap)
     int rc;
 
     if ((fp = fopen(PROC_PATH_NET, "r")) == nullptr) {
-        print_err(errno, QString("open %1 failed").arg(PROC_PATH_NET));
+        print_err_system(errno, QString("open %1 failed").arg(PROC_PATH_NET));
         return b;
     }
 
@@ -427,7 +427,7 @@ bool SystemStat::readNetIfStats(NetIFStat &statSum, NetIFStatMap &statNetIfMap)
     b = !ferror(fp) && b;
     fclose(fp);
     if (!b) {
-        print_err(errno, QString("read %1 failed").arg(PROC_PATH_NET));
+        print_err_system(errno, QString("read %1 failed").arg(PROC_PATH_NET));
     }
 
     return b;
diff --git a/utils/utils.h b/utils/utils.h
index caa4e49..b58732f 100644
--- a/utils/utils.h
+++ b/utils/utils.h
@@ -32,7 +32,10 @@
 #include <QString>
 #include <unordered_set>
 #include "find_window_title.h"
+#include <QtGlobal>
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
 #include "hashqstring.h"
+#endif
 #include <QDebug>
 #include <QPainterPath>
 
