diff -uprNEBZ --suppress-blank-empty a/cmake/modules/FindPythonQt.cmake b/cmake/modules/FindPythonQt.cmake
--- a/cmake/modules/FindPythonQt.cmake	2017-06-08 10:41:37.661363000 -0500
+++ b/cmake/modules/FindPythonQt.cmake	2017-06-08 10:43:09.778030233 -0500
@@ -5,7 +5,7 @@

 find_path(PYTHONQT_INSTALL_DIR include/PythonQt/PythonQt.h DOC "Directory where PythonQt was installed.")
 find_path(PYTHONQT_INCLUDE_DIR PythonQt.h "${PYTHONQT_INSTALL_DIR}/include/PythonQt" DOC "Path to the PythonQt include directory")
-find_library(PYTHONQT_LIBRARY NAMES PythonQt QtPython PATHS "${PYTHONQT_INSTALL_DIR}/lib" DOC "The PythonQt library.")
+find_library(PYTHONQT_LIBRARY NAMES PythonQt QtPython PythonQt-Qt5-Python3 PATHS "${PYTHONQT_INSTALL_DIR}/lib" DOC "The PythonQt library.")

 mark_as_advanced(PYTHONQT_INSTALL_DIR)
 mark_as_advanced(PYTHONQT_INCLUDE_DIR)
diff -uprNEBZ --suppress-blank-empty a/cmake/modules/FindPythonQt_QtAll.cmake b/cmake/modules/FindPythonQt_QtAll.cmake
--- a/cmake/modules/FindPythonQt_QtAll.cmake	2017-06-08 10:41:37.661363000 -0500
+++ b/cmake/modules/FindPythonQt_QtAll.cmake	2017-06-08 10:44:16.514697198 -0500
@@ -6,7 +6,7 @@

 find_path(PYTHONQT_QTALL_INSTALL_DIR NAMES include/PythonQt/PythonQt_QtAll.h include/PythonQt/extensions/PythonQt_QtAll/PythonQt_QtAll.h DOC "Directory where PythonQt_QtAll.h was installed.")
 find_path(PYTHONQT_QTALL_INCLUDE_DIR PythonQt_QtAll.h PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt/extensions/PythonQt_QtAll" ${CMAKE_INCLUDE_PATH} ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include/PythonQt ${CMAKE_INSTALL_PREFIX}/include/PythonQt/extensions/PythonQt_QtAll DOC "Path to the PythonQt_QtAll include directory")
-find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" DOC "The PythonQt_QtAll library.")
+find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll PythonQt_QtAll-Qt5-Python3 PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" DOC "The PythonQt_QtAll library.")

 mark_as_advanced(PYTHONQT_QTALL_INSTALL_DIR)
 mark_as_advanced(PYTHONQT_QTALL_INCLUDE_DIR)
