diff --git a/src/formula_function.cpp b/src/formula_function.cpp
index 6ad07c92..40ccb0b8 100644
--- a/src/formula_function.cpp
+++ b/src/formula_function.cpp
@@ -28,7 +28,7 @@
 
 #include <boost/format.hpp>
 #include <boost/lexical_cast.hpp>
-#include <boost/uuid/sha1.hpp>
+#include <boost/uuid/detail/sha1.hpp>
 #include <boost/algorithm/string.hpp>
 #include <boost/exception/diagnostic_information.hpp>
 #include <boost/exception_ptr.hpp>
diff --git a/src/preferences.cpp b/src/preferences.cpp
index d8a0cbbd..3c4afcb2 100644
--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -28,7 +28,7 @@
 
 #include <boost/regex.hpp>
 #include <boost/lexical_cast.hpp>
-#include <boost/uuid/sha1.hpp>
+#include <boost/uuid/detail/sha1.hpp>
 
 #include "asserts.hpp"
 #include "controls.hpp"
