diff -uprNEBZ --suppress-blank-empty a/urbanlifestyle/Main.qml b/urbanlifestyle/Main.qml
--- a/urbanlifestyle/Main.qml	2016-04-21 15:12:02.000000000 -0500
+++ b/urbanlifestyle/Main.qml	2016-04-21 15:20:15.058646634 -0500
@@ -204,6 +204,15 @@ Rectangle {
 								hoverColor: "#ebaf1d"
 								tooltipBG: "lightgrey"

+								// Fix password focus
+								// https://quickgit.kde.org/?p=plasma-workspace.git&a=commitdiff&h=dfc4b8b2a0e2b012f68f0192e29081ee230e8c03
+								Timer {
+									interval: 200
+									running: true
+									repeat: false
+									onTriggered: password.forceActiveFocus()
+								}
+
 								KeyNavigation.backtab: name
 								KeyNavigation.tab: session

