diff -Naur a/scangearmp/configure.in b/scangearmp/configure.in
--- a/scangearmp/configure.in   2012-02-10 00:53:01.000000000 +0000
+++ b/scangearmp/configure.in   2025-04-09 00:02:31.157520901 +0000
@@ -21,14 +21,12 @@
 AC_SUBST(enable_libpath)

 AM_INIT_AUTOMAKE(scangearmp, 2.00-1)
-AM_CONFIG_HEADER(config.h)
 AM_PROG_LIBTOOL
 AC_PROG_LIBTOOL
 AM_MAINTAINER_MODE

 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB

@@ -37,8 +35,12 @@
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)

-gimp_modules="gimp-2.0 >= 2.0.0"
-PKG_CHECK_MODULES(GIMP, [$gimp_modules])
+AC_MSG_CHECKING([for GIMP 3 or GIMP 2])
+PKG_CHECK_MODULES(GIMP, [gimp-3.0 >= 3.0.0], [AC_MSG_RESULT([GIMP 3 found])], [
+  PKG_CHECK_MODULES(GIMP, [gimp-2.0 >= 2.0.0], [AC_MSG_RESULT([GIMP 2 found])], [
+    AC_MSG_ERROR([Could not find GIMP 3 or GIMP 2])
+  ])
+])
 AC_SUBST(GIMP_CFLAGS)
 AC_SUBST(GIMP_LIBS)
