--- src/joystick_list_widget.cpp	2016-01-03 02:39:09.387377631 -0200
+++ src/joystick_list_widget.cpp	2016-01-03 02:38:52.483930756 -0200
@@ -133,7 +133,7 @@
 	//General icon for the rest
 	else icon_filename = "generic.png";
 
-	(*it)[DeviceListColumns::instance().icon] = Gdk::Pixbuf::create_from_file(Main::current()->get_data_directory() + icon_filename);
+    (*it)[DeviceListColumns::instance().icon] = Gdk::Pixbuf::create_from_file("/usr/share/jstest-gtk/data/" + icon_filename);
     (*it)[DeviceListColumns::instance().path] = i->filename;
 
     std::ostringstream out;
