--- a/torch/utils/cpp_extension.py	2024-10-19 09:30:29.462048147 +0200
+++ b/torch/utils/cpp_extension.py	2024-10-19 09:31:18.237982324 +0200
@@ -2185,7 +2185,7 @@
         common_cflags += [f'-I{include}' for include in user_includes + system_includes]
     else:
         common_cflags += [f'-I{shlex.quote(include)}' for include in user_includes]
-        common_cflags += [f'-isystem {shlex.quote(include)}' for include in system_includes]
+        common_cflags += [f'-I{shlex.quote(include)}' for include in system_includes]
 
     common_cflags += [f"{x}" for x in _get_glibcxx_abi_build_flags()]
 
