=== modified file 'common/config.py'
--- common/config.py	2015-09-28 20:19:24 +0000
+++ common/config.py	2015-12-13 19:32:09 +0000
@@ -28,8 +28,8 @@
     import getpass
     pwd = None
 
+import tools
 import configfile
-import tools
 import logger
 import mount
 import sshtools

=== modified file 'common/tools.py'
--- common/tools.py	2015-08-23 22:21:04 +0000
+++ common/tools.py	2015-12-13 19:31:42 +0000
@@ -1169,5 +1169,13 @@
             return
         self.iface.clean()
 
+def __log_keyring_warning():
+    from time import sleep
+    sleep(0.1)
+    logger.warning('import keyring failed')
+
 if keyring is None and keyring_warn:
-    logger.warning('import keyring failed')
+    #delay warning to give logger some time to import
+    import _thread
+    _thread.start_new_thread(__log_keyring_warning, ())
+    # logger.warning('import keyring failed')
