From 1ced7a71a68bd2da82b5e0031eed2b671c0e8523 Mon Sep 17 00:00:00 2001
From: Thomas Haider <t.haider@deprecate.de>
Date: Mon, 20 Mar 2017 08:48:41 +0100
Subject: [PATCH] fix getoutput call

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 0785e68..ba17f5f 100755
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@ def systemd_unit_path():
     try:
         command = ["pkg-config", "--variable=systemdsystemunitdir", "systemd"]
         try:
-            path = subprocess.getoutput(command)
+            path = subprocess.getoutput(' '.join(command))
         except AttributeError:
             path = subprocess.check_output(command, stderr=subprocess.STDOUT)
         return path.strip()
--
2.12.0
