diff --git a/setup.py b/setup.py
index 39450ae..da50ec0 100644
--- a/setup.py
+++ b/setup.py
@@ -86,7 +86,7 @@ setup(
     name='pyreadr',
     version='0.5.3',
     ext_modules=cythonize([librdata], force=True),
-    packages=["pyreadr"],
+    packages=setuptools.find_namespace_packages(include=("pyreadr*",)),
     include_package_data=include_package_data,
     exclude_package_data=exclude_package_data,
     install_requires=['pandas>=1.2.0'],
