diff -r -u a/FreeFileSync/Source/base/log_file.cpp b/FreeFileSync/Source/base/log_file.cpp
--- a/FreeFileSync/Source/base/log_file.cpp	2020-02-14 19:02:05.000000000 +0100
+++ b/FreeFileSync/Source/base/log_file.cpp	2020-02-15 12:52:40.340265109 +0100
@@ -6,6 +6,8 @@
 
 #include "log_file.h"
 #include <zen/file_io.h>
+#include <zen/http.h>
+#include <zen/system.h>
 #include <wx/datetime.h>
 #include "ffs_paths.h"
 #include "../afs/concrete.h"
diff -r -u a/FreeFileSync/Source/Makefile b/FreeFileSync/Source/Makefile
--- a/FreeFileSync/Source/Makefile	2020-02-15 12:50:45.429040498 +0100
+++ b/FreeFileSync/Source/Makefile	2020-02-15 12:58:39.131372522 +0100
@@ -93,6 +93,7 @@
 CPP_FILES+=../../zen/shutdown.cpp
 CPP_FILES+=../../zen/thread.cpp
 CPP_FILES+=../../zen/zlib_wrap.cpp
+CPP_FILES+=../../zen/system.cpp
 CPP_FILES+=../../wx+/file_drop.cpp
 CPP_FILES+=../../wx+/grid.cpp
 CPP_FILES+=../../wx+/image_tools.cpp
