From bff6b7fd34aee7642842da74981f55b6216c3716 Mon Sep 17 00:00:00 2001
From: Boris Yumankulov <boria138@altlinux.org>
Date: Tue, 9 Sep 2025 10:59:41 +0500
Subject: [PATCH] chore(build): update setuptools

Signed-off-by: Boris Yumankulov <boria138@altlinux.org>
---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index ba965b2..0613f99 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=61.0"]
+requires = ["setuptools >= 77.0.3"]
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -22,7 +22,7 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
   "Programming Language :: Python :: 3.13",
-  "Operating System :: Linux"
+  "Operating System :: POSIX :: Linux"
 ]
 requires-python = ">=3.10"
 dependencies = [
