diff -Naur 4kvideodownloader.orig/4kvideodownloader.sh 4kvideodownloader/4kvideodownloader.sh
--- 4kvideodownloader.orig/4kvideodownloader.sh	2019-11-29 14:29:11.000000000 +0100
+++ 4kvideodownloader/4kvideodownloader.sh	2019-12-02 15:56:55.198273114 +0100
@@ -1,7 +1,6 @@
 #!/bin/sh
 
-SCRIPT_DIR=$(dirname "$0")
-SCRIPT_DIR=`cd "$SCRIPT_DIR"; pwd` #make path absolute
+SCRIPT_DIR=$(dirname $(readlink -f "$0"))
 
 LD_LIBRARY_PATH=$SCRIPT_DIR
 export LD_LIBRARY_PATH
