diff --git a/java/java/build.gradle b/java/java/build.gradle
index 2f31542..bd52473 100644
--- a/java/java/build.gradle
+++ b/java/java/build.gradle
@@ -6,7 +6,6 @@ buildscript {
 }
 
 apply plugin: 'java'
-apply plugin: 'maven'
 apply plugin: 'signing'
 
 sourceCompatibility = 1.7
@@ -28,7 +27,7 @@ sourceSets {
 }
 
 dependencies {
-    testCompile ('junit:junit:4.12')
+    testImplementation ('junit:junit:4.12')
 }
 
 test {
@@ -67,54 +66,6 @@ signing {
     sign configurations.archives
 }
 
-uploadArchives {
-    configuration = configurations.archives
-    repositories.mavenDeployer {
-        beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
-
-        repository(url: getReleaseRepositoryUrl()) {
-            authentication(userName: getRepositoryUsername(), password: getRepositoryPassword())
-        }
-
-        pom.project {
-            name 'signal-client-java'
-            packaging 'jar'
-            description 'Signal Protocol cryptography library for Java'
-            url 'https://github.com/signalapp/libsignal-client'
-
-            scm {
-                url 'scm:git@github.com:signalapp/libsignal-client.git'
-                connection 'scm:git@github.com:signalapp/libsignal-client.git'
-                developerConnection 'scm:git@github.com:signalapp/libsignal-client.git'
-            }
-
-            licenses {
-                license {
-                    name 'AGPLv3'
-                    url 'https://www.gnu.org/licenses/agpl-3.0.txt'
-                    distribution 'repo'
-                }
-            }
-
-            developers {
-                developer {
-                    name 'Signal Messenger LLC'
-                }
-            }
-        }
-    }
-}
-
-task installArchives(type: Upload) {
-    description "Installs the artifacts to the local Maven repository."
-    configuration = configurations['archives']
-    repositories {
-        mavenDeployer {
-            repository url: "file://${System.properties['user.home']}/.m2/repository"
-        }
-    }
-}
-
 task packageJavadoc(type: Jar, dependsOn: 'javadoc') {
     from javadoc.destinationDir
     classifier = 'javadoc'
