diff -up src/ypbind-mt-ypbind-mt-2.4/configure.ac.orig src/ypbind-mt-ypbind-mt-2.4/configure.ac
--- src/ypbind-mt-ypbind-mt-2.4/configure.ac.orig	2016-03-02 07:12:27.000000000 -0700
+++ src/ypbind-mt-ypbind-mt-2.4/configure.ac	2018-04-28 03:27:33.000000000 -0600
@@ -26,7 +26,7 @@ AC_CHECK_LIB(pthread, pthread_create)
 
 PKG_CHECK_MODULES([SYSTEMD], [libsystemd >= 209], [USE_SD_NOTIFY=1], [USE_SD_NOTIFY=0])
 AC_SUBST(USE_SD_NOTIFY)
-#PKG_CHECK_MODULES([NSL], [libnsl], [], [AC_CHECK_LIB([nsl],[yp_match],[LIBNSL_LIBS="-lnsl"],[LIBNSL_LIBS=""])])
+#PKG_CHECK_MODULES([NSL], [libnsl], [], [AC_CHECK_LIB([nsl],[yp_match],[NSL_LIBS="-lnsl"],[NSL_LIBS=""])])
 PKG_CHECK_MODULES([NSL], [libnsl])
 PKG_CHECK_MODULES([TIRPC], [libtirpc], [], [TIRPC_LIBS=""])
 if test -n "$TIRPC_LIBS"; then
diff -up src/ypbind-mt-ypbind-mt-2.4/src/ypbind-mt.c.orig src/ypbind-mt-ypbind-mt-2.4/src/ypbind-mt.c
--- src/ypbind-mt-ypbind-mt-2.4/src/ypbind-mt.c.orig	2018-04-28 03:28:56.000000000 -0600
+++ src/ypbind-mt-ypbind-mt-2.4/src/ypbind-mt.c	2018-04-28 03:29:49.000000000 -0600
@@ -278,7 +278,7 @@ create_pidfile (void)
   struct flock lock;
   int left, written;
   pid_t pid;
-  char pbuf[10], *ptr;
+  char pbuf[14], *ptr;
   int flags;
 
   lock_fd = open (_YPBIND_PIDFILE, O_CREAT | O_RDWR,
