--- a/contrib/bin/fusiondirectory-insert-schema	2017-04-28 14:58:02.002760246 +0200
+++ b/contrib/bin/fusiondirectory-insert-schema	2017-04-28 14:58:31.590835519 +0200
@@ -31,7 +31,7 @@
 use 5.008;
 
 my $ldapSchemaManager = 'ldap-schema-manager';
-my $defaultpath       = "/etc/ldap/schema/fusiondirectory/";
+my $defaultpath       = "/etc/openldap/schema/fusiondirectory/";
 my @defaultschemas    = ("core-fd.schema","core-fd-conf.schema","ldapns.schema","template-fd.schema");
 
 exec (
