diff -aur src/EmptyEpsilon/src/spaceObjects/spaceship.cpp src.patched/EmptyEpsilon/src/spaceObjects/spaceship.cpp
--- src/EmptyEpsilon/src/spaceObjects/spaceship.cpp	2016-05-13 17:57:56.566666915 +0200
+++ src.patched/EmptyEpsilon/src/spaceObjects/spaceship.cpp	2016-05-13 17:59:01.846667135 +0200
@@ -554,7 +554,7 @@
         weapon_tube[n].update(delta);
     }
 
-    model_info.engine_scale = std::min(1.0, std::max(fabs(getAngularVelocity() / turn_speed), fabs(current_impulse)));
+    model_info.engine_scale = std::min(1.0f, std::max(fabs(getAngularVelocity() / turn_speed), fabs(current_impulse)));
     if (has_jump_drive && jump_delay > 0.0f)
         model_info.warp_scale = (10.0f - jump_delay) / 10.0f;
     else
diff -aur src/SeriousProton/src/networkRecorder.cpp src.patched/SeriousProton/src/networkRecorder.cpp
--- src/SeriousProton/src/networkRecorder.cpp	2016-05-13 10:25:07.899749832 +0200
+++ src.patched/SeriousProton/src/networkRecorder.cpp	2016-05-13 17:58:19.280000325 +0200
@@ -1,4 +1,5 @@
 #include <string.h>
+#include <math.h>
 
 #include "networkRecorder.h"
 #include "multiplayer.h"
