--- src/plink2/2.0/build_dynamic/Makefile	2025-12-27 16:52:53.591939278 +0530
+++ src/plink2-patched/2.0/build_dynamic/Makefile	2025-12-27 17:00:29.089812189 +0530
@@ -17,24 +17,24 @@
 #     FORCE_32BIT (warning: you may need to add a zconf.h symlink to make that
 #     work)
 #   Debug symbols: change DEBUG to -g (still incompatible with -flto?)
-NO_AVX2 = 1
-NO_SSE42 = 1
-CPU_CHECK = 1
-NO_LAPACK =
-ZSTD_O2 = 1
-STATIC_ZLIB =
-STATIC_ZSTD = 1
-DYNAMIC_MKL =
-MKLROOT = /opt/intel/oneapi/mkl/2025.1
-MKL_IOMP5_DIR = /opt/intel/oneapi/compiler/2025.1/lib
-STATIC_AOCL =
-AOCLROOT = /home/admin/aocl/5.1.0/gcc
-FORCE_32BIT =
+NO_AVX2 ?=
+NO_SSE42 ?=
+CPU_CHECK ?= 1
+NO_LAPACK ?=
+ZSTD_O2 ?= 1
+STATIC_ZLIB ?=
+STATIC_ZSTD ?= 1
+DYNAMIC_MKL ?=
+MKLROOT ?= /opt/intel/oneapi/mkl/latest
+MKL_IOMP5_DIR ?= /opt/intel/oneapi/compiler/latest/lib
+STATIC_AOCL ?=
+AOCLROOT ?=
+FORCE_32BIT ?=
 # Warning: I have observed intermittent (i.e. has always disappeared on retry
 # in my testing so far) gcc 10.2 crashes with -flto.  I'm leaving -flto here
 # for now since I have yet to see a miscompilation when gcc doesn't crash, but
 # this requires further investigation.
-DEBUG = -flto
+DEBUG ?= -flto
 CC ?= gcc
 CXX ?= g++
 
@@ -55,7 +55,7 @@
   STATIC_ZLIB = 1
   ZSTD_O2 = 1
   ARCH32 = -m32 -march=i686
-  CXXFLAGS = -std=c++0x
+  CXXFLAGS += -std=c++0x
 else
   ifdef NO_AVX2
     ifndef NO_SSE42
@@ -72,7 +72,7 @@
       CPUCHECK_FLAGS = ${DEBUG} -O2 -DCPU_CHECK_AVX2 ${CXXWARN2}
     endif
   endif
-  CXXFLAGS = -std=c++11
+  CXXFLAGS += -std=c++11
 endif
 BASEFLAGS += ${ARCH32}
 
@@ -188,6 +188,13 @@
 
 SFX ?= ""
 
+# Installation defaults
+MKDIR     ?= mkdir
+INSTALL   ?= install
+STRIP_CMD ?= strip
+PREFIX    ?= /usr/local
+DESTDIR   ?= .
+
 all: plink2$(SFX) pgen_compress$(SFX)
 
 plink2$(SFX): $(CSRC2) $(ZCSRC2) $(ZSSRC2) $(CCSRC2) ../plink2_cpu.cc
@@ -216,12 +223,16 @@
 static_plink2lib_test: plink2lib.a ../pgen_compress.cc
 	$(CXX) $(CXXFLAGS) -o $@ ../pgen_compress.cc -L. plink2lib.a
 
-.PHONY: clean
+install-strip: install
+	$(STRIP_CMD) $(DESTDIR)$(PREFIX)/bin/*
+
+install:
+	$(MKDIR) -p $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) plink2 pgen_compress $(DESTDIR)$(PREFIX)/bin
+
 clean:
 	rm -f *.o
-	rm -f plink2
-	rm -f pgen_compress
-	rm -f pgenlib.a
-	rm -f plink2lib.a
-	rm -f static_pgenlib_test
-	rm -f static_plink2lib_test
+	rm -f plink2*
+	rm -f pgen_compress*
+	rm -f *.a
+	rm -f static_*_test
