From 6dd346daefbe6133ebd3a81489880d94b1a87a65 Mon Sep 17 00:00:00 2001
From: Florian Freund <florian88freund@gmail.com>
Date: Wed, 21 Nov 2018 01:08:04 +0100
Subject: Change lib installation to poppler 0.69.0 and fix ignoring DESTDIR


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2715e91b..93a9c8eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -322,11 +322,11 @@ install (DIRECTORY ui
 
 if (BUILD_POPPLER)
 	# The Path /usr/lib is hardcoded, fix if there is a better solution
-	set(POPPLER_LIB ${CMAKE_INSTALL_PREFIX}/lib/libpoppler.so.72)
+	set(POPPLER_LIB ${CMAKE_INSTALL_PREFIX}/lib/libpoppler.so.80)
 
 	install(CODE "MESSAGE(\"Install Poppler workaround to ${POPPLER_LIB}\")")
-	install(CODE "configure_file(${CMAKE_CURRENT_BINARY_DIR}/poppler-prefix/src/poppler-build/libpoppler.so.72.0.0 ${POPPLER_LIB} COPYONLY)")
-	install(CODE "file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/install_manifest_workaround.txt \"${POPPLER_LIB}\")")
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/poppler-prefix/src/poppler-build/libpoppler.so.80.0.0 DESTINATION lib)
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/poppler-prefix/src/poppler-build/libpoppler.so.80 DESTINATION lib)
 endif()
 
 # Uninstall target
