diff --git a/PKGBUILD b/PKGBUILD
index ca9019d..73a7de1 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -108,7 +108,7 @@ build() {
     -DFEATURE_openssl_linked=ON \
     -DFEATURE_system_sqlite=ON \
     -DFEATURE_system_xcb_xinput=ON \
-    -DFEATURE_no_direct_extern_access=ON \
+    -DFEATURE_no_direct_extern_access=OFF \
     -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
     -DCMAKE_MESSAGE_LOG_LEVEL=STATUS
   cmake --build build
