diff --git a/wine-staging-nine.org/dlls/wined3d/glsl_shader.c b/wine-staging-nine/dlls/wined3d/glsl_shader.c
index b3952fa..c293e4c 100644
--- a/wine-staging-nine.org/dlls/wined3d/glsl_shader.c
+++ b/wine-staging-nine/dlls/wined3d/glsl_shader.c
@@ -6709,6 +6709,12 @@ static void shader_glsl_setup_sm3_rasterizer_input(struct shader_glsl_priv *priv
             shader_addline(buffer, "gl_Layer = floatBitsToInt(outputs[%u])%s;\n",
                     output->register_idx, reg_mask);
         }
+        else if (output->sysval_semantic == WINED3D_SV_CLIP_DISTANCE)
+        {
+            gl_info->gl_ops.gl.p_glEnable(GL_CLIP_DISTANCE0+semantic_idx);
+            shader_addline(buffer, "gl_ClipDistance[%u] = outputs[%u]%s;\n",
+                    semantic_idx, output->register_idx, reg_mask);
+        }
         else if (output->sysval_semantic)
         {
             FIXME("Unhandled sysval semantic %#x.\n", output->sysval_semantic);
