diff --git a/src/Makefile b/src/Makefile
index 13b3610..3ef07f0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -30,7 +30,7 @@ checkghcpkg=$(if $(shell ghc-pkg list --simple-output $(1)),,$(error GHC package
 $(foreach pkg,$(GHC_PKGS),$(call checkghcpkg,$(pkg)))
 
 # Check all submodules are initialized
-in_git_repo=$(shell git rev-parse --is-inside-work-tree 2> /dev/null )
+in_git_repo=false # To ignore AUR git
 ifeq ($(in_git_repo),true)
 define SUBMOD_MSG
 Submodule $(1) missing. Initialize with
