--- a/qt/aqt/qt/__init__.py
+++ b/qt/aqt/qt/__init__.py
@@ -9,16 +9,7 @@ import sys
 import traceback
 from typing import Callable, TypeVar, Union

-try:
-    import PyQt6
-except:
-    from .qt5 import *  # type: ignore
-else:
-    if os.getenv("ENABLE_QT5_COMPAT"):
-        print("Running with temporary Qt5 compatibility shims.")
-        from . import qt5_compat  # needs to be imported first
-    from .qt6 import *
-
+from .qt5 import *  # type: ignore
 from anki.utils import is_mac, is_win

 # fix buggy ubuntu12.04 display of language selector
