diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d2580f..f7f8026 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,7 +165,7 @@ if (DEFINED SDL2_INCLUDE_DIRS AND DEFINED SDL2_LIBRARIES)
 	endif()
 else()
 	# Only try to autodetect if both SDL2 variables aren't explicitly set
-	find_package(SDL2 CONFIG)
+	include(/usr/lib32/cmake/SDL2/SDL2Config.cmake)
 	if (TARGET SDL2::SDL2)
 		message(STATUS "using TARGET SDL2::SDL2")
 		target_link_libraries(FAudio PUBLIC SDL2::SDL2)
