diff -ru3 ffmpeg-1.2.1/configure ffmpeg-1.2.1-p/configure
--- ffmpeg-1.2.1/configure	2013-05-10 02:16:22.000000000 +0200
+++ ffmpeg-1.2.1-p/configure	2014-01-05 12:17:12.004338758 +0100
@@ -3903,7 +3903,7 @@
 enabled libfdk_aac && require  libfdk_aac fdk-aac/aacenc_lib.h aacEncOpen -lfdk-aac
 flite_libs="-lflite_cmu_time_awb -lflite_cmu_us_awb -lflite_cmu_us_kal -lflite_cmu_us_kal16 -lflite_cmu_us_rms -lflite_cmu_us_slt -lflite_usenglish -lflite_cmulex -lflite"
 enabled libflite   && require2 libflite "flite/flite.h" flite_init $flite_libs
-enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype/freetype.h" FT_Init_FreeType
+enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype2/freetype.h" FT_Init_FreeType
 enabled libgsm     && { for gsm_hdr in "gsm.h" "gsm/gsm.h"; do
                             check_lib "${gsm_hdr}" gsm_create -lgsm && break;
                         done || die "ERROR: libgsm not found"; }
diff -ru3 ffmpeg-1.2.1/libavfilter/vf_drawtext.c ffmpeg-1.2.1-p/libavfilter/vf_drawtext.c
--- ffmpeg-1.2.1/libavfilter/vf_drawtext.c	2013-05-10 02:16:06.000000000 +0200
+++ ffmpeg-1.2.1-p/libavfilter/vf_drawtext.c	2014-01-05 12:18:33.674284457 +0100
@@ -48,7 +48,7 @@
 #include "video.h"
 
 #include <ft2build.h>
-#include <freetype/config/ftheader.h>
+#include <freetype2/config/ftheader.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 #if CONFIG_FONTCONFIG
