--- Makefile.install.directories	2016-01-23 13:53:43.678049616 +0100
+++ Makefile.install.directories	2016-01-23 13:57:24.928993539 +0100
@@ -4,13 +4,13 @@
 # the different directories for the files
 
 # installing in /usr/local
-MAIN_DIRECTORY         = /usr/local
-BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
-DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
-HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
-MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
-ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
-DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+#MAIN_DIRECTORY         = /usr/local
+#BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
+#DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
+#HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+#MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+#ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
+#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 # debian
 #MAIN_DIRECTORY         = /usr
@@ -48,14 +48,14 @@
 #ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
 #DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
-#arch linux
-#MAIN_DIRECTORY         = /usr
-#BIN_DIRECTORY = $(MAIN_DIRECTORY)/bin
-#DATA_DIRECTORY = $(MAIN_DIRECTORY)/share/freedoko
-#HELP_DIRECTORY = $(MAIN_DIRECTORY)/share/doc/freedoko
-#MAN_DIRECTORY = $(MAIN_DIRECTORY)/share/man/man6
-#ICON_DIRECTORY = $(MAIN_DIRECTORY)/share/pixmaps/
-#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+# Arch Linux
+MAIN_DIRECTORY         = /usr
+BIN_DIRECTORY          = $(MAIN_DIRECTORY)/bin
+DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/freedoko
+HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/freedoko
+MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
+DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 
 # release

--- Makefile	2016-03-03 09:37:30.163436058 +0100
+++ Makefile	2016-03-03 09:40:36.958111227 +0100
@@ -289,22 +289,11 @@
 ifeq ($(OPERATING_SYSTEM), Linux)
 .PHONY: install
 install :
-	$(MAKE) \
-	  --always-make \
-	  "USE_UI_TEXT=false" \
-	  "USE_UI_GTKMM=true" \
-	  "USE_NETWORK=true" \
-	  "CXXFLAGS=-Wall -Werror -Wno-parentheses -pipe -Os -std=c++11" \
-	  "CPPFLAGS=-DRELEASE -DVERSION_DESCRIPTION='\"$(OPERATING_SYSTEM_NAME) binary\"' -DPUBLIC_DATA_DIRECTORY_VALUE='\"$(DATA_DIRECTORY)\"' -DMANUAL_DIRECTORY_VALUE='\"$(HELP_DIRECTORY)/manual\"'" \
-	  -C src \
-	  $(PROGRAM)
-	$(MAKE) -C src strip
 #	Copying the program
 	echo "Copying the program"
 	mkdir -p $(DESTDIR)$(BIN_DIRECTORY)
 	cp src/FreeDoko $(DESTDIR)$(BIN_DIRECTORY)
-	-chgrp games $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
-	chmod 750 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
+	chmod 755 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
 	cd $(DESTDIR)$(BIN_DIRECTORY) && ln -sf FreeDoko freedoko
 #	Copying the data files
 	echo "Copying the data files"
@@ -321,15 +310,13 @@
 	a2x -f manpage doc/freedoko.asciidoc
 	mkdir -p $(DESTDIR)$(MAN_DIRECTORY)
 	cp doc/freedoko.6 $(DESTDIR)$(MAN_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
-	chmod 640 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
+	chmod 644 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
 	cd $(DESTDIR)$(MAN_DIRECTORY)/ && ln -sf freedoko.6 FreeDoko.6
 #	copy the icon
 	echo "copy the icon"
 	mkdir -p $(DESTDIR)$(ICON_DIRECTORY)
 	cp src/FreeDoko.png $(DESTDIR)$(ICON_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
-	chmod 640 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
+	chmod 644 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
 
 .PHONY: uninstall
 uninstall :

--- data/Makefile	2016-01-23 13:54:18.967086099 +0100
+++ data/Makefile	2016-01-23 13:59:16.287086218 +0100
@@ -40,8 +40,7 @@
 	for d in cardsets iconsets backgrounds sounds; do \
 	  $(MAKE) -C $$d install; \
 	done
-	-chgrp games $(DESTDIR)$(DATA_DIRECTORY) && chgrp -R games $(DESTDIR)$(DATA_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(DATA_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(DATA_DIRECTORY)
 endif
 
 ifeq ($(SHELLTYPE), sh)

--- doc/Makefile	2016-01-23 13:54:26.600146704 +0100
+++ doc/Makefile	2016-01-23 13:59:38.710069282 +0100
@@ -36,8 +36,7 @@
 	fi
 	cp -a Bugs GPL Roadmap hpux.required_libs-ia64.txt $(DESTDIR)$(HELP_DIRECTORY)/
 	for d in de en; do $(MAKE) -C $$d install; done
-	-chgrp games $(DESTDIR)$(HELP_DIRECTORY) && chgrp -R games $(DESTDIR)$(HELP_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(HELP_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(HELP_DIRECTORY)
 	cd $(DESTDIR)$(DATA_DIRECTORY) && ln -sf $(HELP_DIRECTORY)/ doc
 endif

--- src/Makefile.local.template	2016-03-03 09:44:06.736272619 +0100
+++ src/Makefile.local.template	2016-03-03 09:46:38.268785905 +0100
@@ -2,6 +2,8 @@
 # Just rename it to 'Makefile.local' to get rid of the make error
 
 include $(DEPTH)/../Makefile.version
+include $(DEPTH)/../Makefile.os
+include $(DEPTH)/../Makefile.install.directories
 
 # which modules to use (see Makefile.modules)
 # thread support -- experimental
@@ -28,13 +30,13 @@
 
 
 # Where is the directory with the public data?
-CPPFLAGS += -DPUBLIC_DATA_DIRECTORY_VALUE='"../data"'
+CPPFLAGS += -DPUBLIC_DATA_DIRECTORY_VALUE='"$(DATA_DIRECTORY)"'
 
 # Where is the directory with the manual?
-#CPPFLAGS += -DMANUAL_DIRECTORY_VALUE='"../manual"'
+CPPFLAGS += -DMANUAL_DIRECTORY_VALUE='"$(HELP_DIRECTORY)/manual"'
 
 # a description for the version information
-CPPFLAGS += -DVERSION_DESCRIPTION='"self compiled"'
+CPPFLAGS += -DVERSION_DESCRIPTION='"$(OPERATING_SYSTEM_NAME) binary"'
 
 # no debugging (speeds up the program but does not check for errors)
 #CPPFLAGS += -DDEBUG_NO
 
--- bin/FreeDoko.desktop	2016-01-23 13:55:07.632168568 +0100
+++ bin/FreeDoko.desktop	2016-01-23 14:00:16.109500816 +0100
@@ -6,7 +6,7 @@
 Comment=game of Doppelkopf
 Comment[de]=Doppelkopf-Spiel
 Comment[fr]=un jeu de Doppelkopf
-Exec=/usr/games/freedoko
+Exec=/usr/bin/freedoko
 Path=
 Icon=FreeDoko
 Categories=Application;Game;CardGame;
