From d4b6a6a8d1a5dbc30003f1f1c07b519d51b95d92 Mon Sep 17 00:00:00 2001
From: 0x9fff00 <0x9fff00+git@protonmail.ch>
Date: Sat, 20 Jun 2020 20:35:10 +0200
Subject: [PATCH 2/2] Update libplist and libirecovery names for 2.2.0/1.0.0

---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index f0b2683..a647cdb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,15 +42,15 @@ AC_CONFIG_SRCDIR([futurerestore])
 
 # Checks the libraries.
 LIBIPATCHER_REQUIRES_STR="libipatcher >= 61"
-LIBPLIST_REQUIRES_STR="libplist >= 2.0.0"
+LIBPLIST_REQUIRES_STR="libplist-2.0 >= 2.2.0"
 LIBZIP_REQUIRES_STR="libzip >= 0.10"
 LIBIMOBILEDEVICE_REQUIRES_STR="libimobiledevice-1.0 >= 1.2.1"
 LIBFRAGMENTZIP_REQUIRES_STR="libfragmentzip >= 47"
-LIBIRECOVERY_REQUIRES_STR="libirecovery >= 0.2.0"
+LIBIRECOVERY_REQUIRES_STR="libirecovery-1.0 >= 1.0.0"
 IMG4TOOL_REQUIRES_STR="libimg4tool >= 162"
 LIBGENERAL_REQUIRES_STR="libgeneral >= 26"
 
-PKG_CHECK_MODULES(libplist, libplist >= 2.0.0)
+PKG_CHECK_MODULES(libplist, $LIBPLIST_REQUIRES_STR)
 PKG_CHECK_MODULES(libzip, $LIBZIP_REQUIRES_STR)
 PKG_CHECK_MODULES(libimobiledevice, $LIBIMOBILEDEVICE_REQUIRES_STR)
 PKG_CHECK_MODULES(libfragmentzip, $LIBFRAGMENTZIP_REQUIRES_STR)
-- 
2.27.0

