--- src/Magics-2.33.0-Source/src/boost/geometry/geometries/point.hpp	2017-05-17 10:35:00.000000000 +0200
+++ src/Magics-2.33.0-Source/src/boost/geometry/geometries/point.hpp.new	2017-06-06 10:54:25.052432745 +0200
@@ -92,6 +92,16 @@
         m_values[K] = value;
     }
 
+    /// @brief Set a coordinate
+    /// @tparam K coordinate to set
+    /// @param value value to set
+    template <std::size_t K>
+    inline void xset(CoordinateType const& value)
+    {
+        BOOST_STATIC_ASSERT(K < DimensionCount);
+        m_values[K] = value;
+    }
+
 private:
 
     CoordinateType m_values[DimensionCount];
@@ -166,7 +176,7 @@
         model::point<CoordinateType, DimensionCount, CoordinateSystem>& p,
         CoordinateType const& value)
     {
-        p.template set<Dimension>(value);
+        p.template xset<Dimension>(value);
     }
 };
 
--- src/Magics-2.33.0-Source/src/boost/geometry/geometries/point_xy.hpp	2017-05-17 10:35:00.000000000 +0200
+++ src/Magics-2.33.0-Source/src/boost/geometry/geometries/point_xy.hpp.new	2017-06-06 10:54:37.669171037 +0200
@@ -65,11 +65,11 @@
 
     /// Set x-value
     inline void x(CoordinateType const& v)
-    { this->template set<0>(v); }
+    { this->template xset<0>(v); }
 
     /// Set y-value
     inline void y(CoordinateType const& v)
-    { this->template set<1>(v); }
+    { this->template xset<1>(v); }
 };
 
 
@@ -116,7 +116,7 @@
     static inline void set(model::d2::point_xy<CoordinateType, CoordinateSystem>& p,
         CoordinateType const& value)
     {
-        p.template set<Dimension>(value);
+        p.template xset<Dimension>(value);
     }
 };
 
