diff -Naur a/src/main.c b/src/main.c
--- a/src/main.c	2017-06-25 21:52:26.000000000 +0200
+++ b/src/main.c	2025-04-25 07:28:30.310478900 +0200
@@ -171,7 +171,7 @@
 
 	gtk_about_dialog_set_copyright (GTK_ABOUT_DIALOG(about), "Copyleft ?-2016. See License for details.");
 
-	gchar *authors[3];
+	const gchar *authors[3];
 	authors[0] = "Joergen Scheibengruber <mfcn@gmx.de>";
 	authors[1] = "Assen Totin <assen.totin@gmail.com>";
 	authors[2] = NULL;
