diff --color -ruN 1/plugins/DebuggerCore/unix/linux/FeatureDetect.cpp 2/plugins/DebuggerCore/unix/linux/FeatureDetect.cpp
--- 1/plugins/DebuggerCore/unix/linux/FeatureDetect.cpp	2020-12-14 02:09:07.000000000 +0100
+++ 2/plugins/DebuggerCore/unix/linux/FeatureDetect.cpp	2023-05-26 00:06:41.230343297 +0200
@@ -19,6 +19,7 @@
 #include "FeatureDetect.h"
 #include "version.h"
 
+#include <cstdint>
 #include <fcntl.h>
 #include <iomanip>
 #include <iostream>
