diff --git a/ImageCache.cpp b/ImageCache.cpp
index c48844a..ba5a22a 100644
--- a/ImageCache.cpp
+++ b/ImageCache.cpp
@@ -257,7 +257,7 @@ namespace ImageCache {
 	{
 	    __gnu_cxx::stdio_filebuf<char> buffer(fd, std::ios::out);
 	    std::ostream out(&buffer);
-	    success = (out << *image);
+	    success = (bool)(out << *image);
 	    // stdio_filebuf destructor will close fd
 	}
 	if (success) {
