diff -u Template/LO/SubProcesses/genps.f Template/LO/SubProcesses/genps.f
--- a/Template/LO/SubProcesses/genps.f
+++ b/Template/LO/SubProcesses/genps.f
@@ -41,7 +41,7 @@
       integer mincfig,maxcfig                  !Range of configurations
       integer invar
       double precision wgt                     !(input and output)
-      double precision x(maxdim),p(maxdim)     !x,p (output) [p(0:3,nexternal)]
+      double precision x(maxdim),p(*)          !x,p (output) [p(0:3,nexternal)]
 c
 c     Local
 c

Diff finished.  Tue Jun 26 11:32:30 2018
