diff -rcB geany-zc-orig/configure.ac geany-zencoding/configure.ac
*** geany-zc-orig/configure.ac	2011-11-14 20:59:28.746554254 +0545
--- geany-zencoding/configure.ac	2011-11-14 20:56:50.613230459 +0545
***************
*** 15,21 ****
  AC_PROG_SED
  
  # Check for Python to pre-compile python sources
! AC_PATH_PROG([PYTHON], [python], [no])
  AC_SUBST([PYTHON])
  if [[ x"$PYTHON" = "xno" ]]; then
  	AC_MSG_ERROR([Python not found, is it installed?])
--- 15,21 ----
  AC_PROG_SED
  
  # Check for Python to pre-compile python sources
! AC_PATH_PROG([PYTHON], [python2], [no])
  AC_SUBST([PYTHON])
  if [[ x"$PYTHON" = "xno" ]]; then
  	AC_MSG_ERROR([Python not found, is it installed?])
diff -rcB geany-zc-orig/zencoding/actions/basic.py geany-zencoding/zencoding/actions/basic.py
*** geany-zc-orig/zencoding/actions/basic.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/actions/basic.py	2011-11-14 20:58:38.286557305 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  """
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  """
diff -rcB geany-zc-orig/zencoding/actions/__init__.py geany-zencoding/zencoding/actions/__init__.py
*** geany-zc-orig/zencoding/actions/__init__.py	2011-11-14 20:59:28.746554254 +0545
--- geany-zencoding/zencoding/actions/__init__.py	2011-11-14 20:58:38.166557314 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  # import all actions
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  # import all actions
***************
*** 12,15 ****
  		
  __import__(__name__, globals(), locals(), __sub_modules)
  
! del __sub_modules
\ No newline at end of file
--- 12,15 ----
  		
  __import__(__name__, globals(), locals(), __sub_modules)
  
! del __sub_modules
diff -rcB geany-zc-orig/zencoding/actions/token.py geany-zencoding/zencoding/actions/token.py
*** geany-zc-orig/zencoding/actions/token.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/actions/token.py	2011-11-14 20:58:38.039890653 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
diff -rcB geany-zc-orig/zencoding/filters/comment.py geany-zencoding/zencoding/filters/comment.py
*** geany-zc-orig/zencoding/filters/comment.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/comment.py	2011-11-14 20:58:38.999890624 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 44,47 ****
  			add_comments(item, i)
  		process(item, profile)
  	
! 	return tree
\ No newline at end of file
--- 44,47 ----
  			add_comments(item, i)
  		process(item, profile)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/css.py geany-zencoding/zencoding/filters/css.py
*** geany-zc-orig/zencoding/filters/css.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/css.py	2011-11-14 20:58:38.886557271 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  '''
  Process CSS properties: replaces snippets, augumented with ! char, with 
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  '''
  Process CSS properties: replaces snippets, augumented with ! char, with 
***************
*** 20,23 ****
  		
  		process(item, profile)
  	
! 	return tree
\ No newline at end of file
--- 20,23 ----
  		
  		process(item, profile)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/escape.py geany-zencoding/zencoding/filters/escape.py
*** geany-zc-orig/zencoding/filters/escape.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/escape.py	2011-11-14 20:58:38.586557288 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 28,31 ****
  		
  		process(item)
  	
! 	return tree
\ No newline at end of file
--- 28,31 ----
  		
  		process(item)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/format-css.py geany-zencoding/zencoding/filters/format-css.py
*** geany-zc-orig/zencoding/filters/format-css.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/format-css.py	2011-11-14 20:58:39.126557254 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 21,24 ****
  		
  		process(item, profile)
  		
! 	return tree
\ No newline at end of file
--- 21,24 ----
  		
  		process(item, profile)
  		
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/format.py geany-zencoding/zencoding/filters/format.py
*** geany-zc-orig/zencoding/filters/format.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/format.py	2011-11-14 20:58:39.489890569 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  """
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  """
***************
*** 179,182 ****
  			
  		process(item, profile, level + 1)
  	
! 	return tree
\ No newline at end of file
--- 179,182 ----
  			
  		process(item, profile, level + 1)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/haml.py geany-zencoding/zencoding/filters/haml.py
*** geany-zc-orig/zencoding/filters/haml.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/haml.py	2011-11-14 20:58:39.376557242 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 157,160 ****
  		
  		process(item, profile, level + 1)
  		
! 	return tree
\ No newline at end of file
--- 157,160 ----
  		
  		process(item, profile, level + 1)
  		
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/html.py geany-zencoding/zencoding/filters/html.py
*** geany-zc-orig/zencoding/filters/html.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/html.py	2011-11-14 20:58:38.389890632 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
diff -rcB geany-zc-orig/zencoding/filters/__init__.py geany-zencoding/zencoding/filters/__init__.py
*** geany-zc-orig/zencoding/filters/__init__.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/__init__.py	2011-11-14 20:58:38.683223948 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  import os.path
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  import os.path
***************
*** 14,17 ****
  		
  __import__(__name__, globals(), locals(), __sub_modules)
  
! del __sub_modules
\ No newline at end of file
--- 14,17 ----
  		
  __import__(__name__, globals(), locals(), __sub_modules)
  
! del __sub_modules
diff -rcB geany-zc-orig/zencoding/filters/single-line.py geany-zencoding/zencoding/filters/single-line.py
*** geany-zc-orig/zencoding/filters/single-line.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/single-line.py	2011-11-14 20:58:38.786557275 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  '''
  Output abbreviation on a single line (i.e. no line breaks)
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  '''
  Output abbreviation on a single line (i.e. no line breaks)
***************
*** 26,29 ****
  		
  		process(item, profile)
  	
! 	return tree
\ No newline at end of file
--- 26,29 ----
  		
  		process(item, profile)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/trim.py geany-zencoding/zencoding/filters/trim.py
*** geany-zc-orig/zencoding/filters/trim.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/trim.py	2011-11-14 20:58:39.249890589 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  '''
  Trim filter: removes characters at the beginning of the text
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  '''
  Trim filter: removes characters at the beginning of the text
***************
*** 19,22 ****
  		
  		process(item, profile)
  	
! 	return tree
\ No newline at end of file
--- 19,22 ----
  		
  		process(item, profile)
  	
! 	return tree
diff -rcB geany-zc-orig/zencoding/filters/xsl.py geany-zencoding/zencoding/filters/xsl.py
*** geany-zc-orig/zencoding/filters/xsl.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/filters/xsl.py	2011-11-14 20:58:38.489890626 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 30,33 ****
  		if item.type == 'tag' and item.name.lower() in tags and item.children:
  			trim_attribute(item)
  		
! 		process(item, profile)
\ No newline at end of file
--- 30,33 ----
  		if item.type == 'tag' and item.name.lower() in tags and item.children:
  			trim_attribute(item)
  		
! 		process(item, profile)
diff -rcB geany-zc-orig/zencoding/html_matcher.py geany-zencoding/zencoding/html_matcher.py
*** geany-zc-orig/zencoding/html_matcher.py	2011-11-14 20:59:28.749887587 +0545
--- geany-zencoding/zencoding/html_matcher.py	2011-11-14 20:58:37.916557326 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  
  '''
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  
  '''
***************
*** 270,273 ****
  				
  			ix += 1
  	
! 	return action(opening_tag, closing_tag, start_ix)
\ No newline at end of file
--- 270,273 ----
  				
  			ix += 1
  	
! 	return action(opening_tag, closing_tag, start_ix)
diff -rcB geany-zc-orig/zencoding/resources.py geany-zencoding/zencoding/resources.py
*** geany-zc-orig/zencoding/resources.py	2011-11-14 20:59:28.753220920 +0545
--- geany-zencoding/zencoding/resources.py	2011-11-14 20:58:39.589890567 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  '''
  @author Sergey Chikuyonok (serge.che@gmail.com)
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  '''
  @author Sergey Chikuyonok (serge.che@gmail.com)
***************
*** 367,370 ****
  		pass
  	
  if user_settings:
! 	set_vocabulary(user_settings, VOC_USER)
\ No newline at end of file
--- 367,370 ----
  		pass
  	
  if user_settings:
! 	set_vocabulary(user_settings, VOC_USER)
diff -rcB geany-zc-orig/zencoding/utils.py geany-zencoding/zencoding/utils.py
*** geany-zc-orig/zencoding/utils.py	2011-11-14 20:59:28.753220920 +0545
--- geany-zencoding/zencoding/utils.py	2011-11-14 20:58:34.886557511 +0545
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  '''
  Core Zen Coding library. Contains various text manipulation functions:
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  '''
  Core Zen Coding library. Contains various text manipulation functions:
