diff --u a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -51,7 +51,18 @@
 webkit2gtk = dependency('webkit2gtk-4.0', version: '>=' + target_webkit)
 
 # Secondary deps - keep sorted alphabetically
-enchant = dependency('enchant', version: '>= 1.6')
+enchant = dependency('enchant', version: '>= 1.6', required: false)
+if not enchant.found()
+  dependency('enchant-2', version: '>= 1.6')
+  enchant = valac.find_library(
+    'enchant',
+    dirs: '/usr/share/vala-0.40/vapi/',
+    required: true
+  )
+  # TODO These arguments should be applied more specifically
+  add_project_arguments('-I/usr/include/enchant-2/', language : 'c')
+  add_project_link_arguments('-lenchant-2', language : 'c')
+endif
 gcr = dependency('gcr-3', version: '>= 3.10.1')
 gdk = dependency('gdk-3.0', version: '>=' + target_gtk)
 gee = dependency('gee-0.8', version: '>= 0.8.5')

