diff -up src/windowmaker-crm-git/wrlib/load_gif.c.orig src/windowmaker-crm-git/wrlib/load_gif.c
--- src/windowmaker-crm-git/wrlib/load_gif.c.orig	2014-06-18 04:27:12.000000000 -0600
+++ src/windowmaker-crm-git/wrlib/load_gif.c	2014-06-18 04:27:37.000000000 -0600
@@ -81,7 +81,7 @@ RImage *RLoadGIF(const char *file, int i
 	}
 
 	if (gif->SWidth < 1 || gif->SHeight < 1) {
-		DGifCloseFile(gif);
+		DGifCloseFile(gif, NULL);
 		RErrorCode = RERR_BADIMAGEFILE;
 		return NULL;
 	}
@@ -216,7 +216,7 @@ RImage *RLoadGIF(const char *file, int i
 		free(buffer);
 
 	if (gif)
-		DGifCloseFile(gif);
+		DGifCloseFile(gif, NULL);
 
 	return image;
 }
