diff --git a/examples/Makefile.am b/examples/Makefile.am
index 20e925a..93fbebb 100644
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -41,7 +41,7 @@ webpmux_LDADD = libexampleutil.la ../src/mux/libwebpmux.la ../src/libwebp.la
 
 vwebp_SOURCES = vwebp.c
 vwebp_CPPFLAGS = $(AM_CPPFLAGS) $(USE_EXPERIMENTAL_CODE) $(GL_INCLUDES)
-vwebp_LDADD = libexampleutil.la ../src/demux/libwebpdemux.la $(GL_LIBS)
+vwebp_LDADD = libexampleutil.la ../src/demux/libwebpdemux.la $(GL_LIBS) -lGL
 
 if BUILD_LIBWEBPDECODER
   dwebp_LDADD += ../src/libwebpdecoder.la
