diff -ru a/sview-17_04/include/StStrings/StStringStream.h b/sview-17_04/include/StStrings/StStringStream.h
--- a/sview-17_04/include/StStrings/StStringStream.h	2017-04-16 19:38:46.000000000 +0200
+++ b/sview-17_04/include/StStrings/StStringStream.h	2017-11-20 11:58:06.030845513 +0100
@@ -21,7 +21,7 @@
         #define ST_NO_XLOCALE
         #warning xlocale is not supported by compiler!
     #else
-        #include <xlocale.h>
+        #include <bits/types/__locale_t.h>
     #endif
 #endif
 
