--- CMakeLists.txt
+++ CMakeLists.txt
@@ -70,15 +70,6 @@ option(BUILD_LAYERS "Build layers" ON)
 option(BUILD_DEMOS "Build demos" ON)
 option(BUILD_VKJSON "Build vkjson" ON)
 
-if (BUILD_TESTS)
-    # Hard code our glslang path for now
-    get_filename_component(GLSLANG_PREFIX ../glslang ABSOLUTE)
-
-    if(NOT EXISTS ${GLSLANG_PREFIX})
-        message(FATAL_ERROR "Necessary glslang components do not exist: " ${GLSLANG_PREFIX})
-    endif()
-endif()
-
 if(NOT WIN32)
     include(GNUInstallDirs)
     add_definitions(-DSYSCONFDIR="${CMAKE_INSTALL_SYSCONFDIR}")
