diff -aur old/configure new/configure
--- old/configure	2019-02-27 14:33:09.773385526 +0100
+++ new/configure	2019-02-27 14:38:40.619816260 +0100
@@ -2945,7 +2945,7 @@
 
 ac_config_headers="$ac_config_headers config.h"
 
-am__api_version='1.15'
+am__api_version='1.14'
 
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
diff -aur old/g10/keygen.c new/g10/keygen.c
--- old/g10/keygen.c	2019-02-27 14:33:09.533386670 +0100
+++ new/g10/keygen.c	2019-02-27 14:38:40.627816221 +0100
@@ -1639,7 +1639,7 @@
   int err;
   char *keyparms;
   char nbitsstr[35];
-  const unsigned maxsize = (opt.flags.large_rsa ? 8192 : 4096);
+  const unsigned maxsize = (opt.flags.large_rsa ? 16384 : 4096);
 
   log_assert (is_RSA(algo));
 
@@ -2114,6 +2114,12 @@
       def=255;
       break;
 
+    case PUBKEY_ALGO_RSA:
+      *min = opt.compliance == CO_DE_VS ? 2048 : 1024;
+      *max = (opt.flags.large_rsa ? 16384 : 4096);
+      def = 2048;
+      break;
+
     default:
       *min = opt.compliance == CO_DE_VS ? 2048: 1024;
       *max = 4096;
