--- a/deploy/go-carbon.conf	2018-03-10 09:29:38.000000000 +0100
+++ b/deploy/go-carbon.conf	2018-04-20 22:57:21.412327373 +0200
@@ -196,7 +196,7 @@
 listen = "127.0.0.1:8080"
 # Carbonserver support is still experimental and may contain bugs
 # Or be incompatible with github.com/grobian/carbonserver
-enabled = false
+enabled = true
 # Buckets to track response times
 buckets = 10
 # carbonserver-specific metrics will be sent as counters
