diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index c033beabdb..80f85d6df7 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -100,6 +100,7 @@ var okgoos = []string{
 	"plan9",
 	"windows",
 	"aix",
+	"sylixos",
 }
 
 // find reports the first index of p in l[0:n], or else -1.
@@ -724,7 +725,7 @@ func runInstall(pkg string, ch chan struct{}) {
 			elem = "go_bootstrap"
 		}
 		link = []string{pathf("%s/link", tooldir)}
-		if goos == "android" {
+		if goos == "android" || goos == "sylixos" {
 			link = append(link, "-buildmode=pie")
 		}
 		if goldflags != "" {
@@ -1044,6 +1045,7 @@ var unixOS = map[string]bool{
 	"netbsd":    true,
 	"openbsd":   true,
 	"solaris":   true,
+	"sylixos":   true,
 }
 
 // matchtag reports whether the tag matches this build.
@@ -1776,6 +1778,8 @@ var cgoEnabled = map[string]bool{
 	"windows/amd64":   true,
 	"windows/arm":     false,
 	"windows/arm64":   true,
+	"sylixos/amd64":   true,
+	"sylixos/arm64":   true,
 }
 
 // List of platforms that are marked as broken ports.
diff --git a/src/cmd/dist/test.go b/src/cmd/dist/test.go
index 0ffcabe416..b0cd057069 100644
--- a/src/cmd/dist/test.go
+++ b/src/cmd/dist/test.go
@@ -1625,7 +1625,8 @@ func buildModeSupported(compiler, buildmode, goos, goarch string) bool {
 			"android/amd64", "android/arm", "android/arm64", "android/386",
 			"freebsd/amd64",
 			"darwin/amd64", "darwin/arm64",
-			"windows/amd64", "windows/386", "windows/arm64":
+			"windows/amd64", "windows/386", "windows/arm64",
+			"sylixos/amd64", "sylixos/arm64":
 			return true
 		}
 		return false
@@ -1645,14 +1646,16 @@ func buildModeSupported(compiler, buildmode, goos, goarch string) bool {
 			"ios/amd64", "ios/arm64",
 			"aix/ppc64",
 			"openbsd/arm64",
-			"windows/386", "windows/amd64", "windows/arm", "windows/arm64":
+			"windows/386", "windows/amd64", "windows/arm", "windows/arm64",
+			"sylixos/amd64", "sylixos/arm64":
 			return true
 		}
 		return false
 
 	case "shared":
 		switch platform {
-		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le", "linux/s390x":
+		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le", "linux/s390x",
+			"sylixos/amd64", "sylixos/arm64":
 			return true
 		}
 		return false
@@ -1662,7 +1665,8 @@ func buildModeSupported(compiler, buildmode, goos, goarch string) bool {
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/loong64", "linux/s390x", "linux/ppc64le",
 			"android/amd64", "android/386",
 			"darwin/amd64", "darwin/arm64",
-			"freebsd/amd64":
+			"freebsd/amd64",
+			"sylixos/amd64", "sylixos/arm64":
 			return true
 		}
 		return false
diff --git a/src/cmd/go/internal/imports/build.go b/src/cmd/go/internal/imports/build.go
index 3a4a66b869..cca4bde1e9 100644
--- a/src/cmd/go/internal/imports/build.go
+++ b/src/cmd/go/internal/imports/build.go
@@ -326,6 +326,7 @@ var KnownOS = map[string]bool{
 	"wasip1":    true,
 	"windows":   true,
 	"zos":       true,
+	"sylixos":   true,
 }
 
 // unixOS is the set of GOOS values matched by the "unix" build tag.
@@ -344,6 +345,7 @@ var unixOS = map[string]bool{
 	"netbsd":    true,
 	"openbsd":   true,
 	"solaris":   true,
+	"sylixos":   true,
 }
 
 var KnownArch = map[string]bool{
diff --git a/src/cmd/go/internal/lockedfile/internal/filelock/filelock_unix.go b/src/cmd/go/internal/lockedfile/internal/filelock/filelock_unix.go
index 6f73b1bfee..d71202cdbd 100644
--- a/src/cmd/go/internal/lockedfile/internal/filelock/filelock_unix.go
+++ b/src/cmd/go/internal/lockedfile/internal/filelock/filelock_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || illumos || linux || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || illumos || linux || netbsd || openbsd || sylixos
 
 package filelock
 
diff --git a/src/cmd/go/internal/modindex/syslist.go b/src/cmd/go/internal/modindex/syslist.go
index 41adcc5342..33b99fb03d 100644
--- a/src/cmd/go/internal/modindex/syslist.go
+++ b/src/cmd/go/internal/modindex/syslist.go
@@ -28,6 +28,7 @@ var knownOS = map[string]bool{
 	"wasip1":    true,
 	"windows":   true,
 	"zos":       true,
+	"sylixos":   true,
 }
 
 // unixOS is the set of GOOS values matched by the "unix" build tag.
@@ -46,6 +47,7 @@ var unixOS = map[string]bool{
 	"netbsd":    true,
 	"openbsd":   true,
 	"solaris":   true,
+	"sylixos":   true,
 }
 
 // knownArch is the list of past, present, and future known GOARCH values.
diff --git a/src/cmd/go/internal/test/test.go b/src/cmd/go/internal/test/test.go
index 0ac5666620..b8cba80f1d 100644
--- a/src/cmd/go/internal/test/test.go
+++ b/src/cmd/go/internal/test/test.go
@@ -15,8 +15,10 @@ import (
 	"io/fs"
 	"os"
 	"os/exec"
+	"path"
 	"path/filepath"
 	"regexp"
+	"runtime"
 	"slices"
 	"strconv"
 	"strings"
@@ -1258,6 +1260,10 @@ func builderTest(b *work.Builder, ctx context.Context, pkgOpts load.PackageOpts,
 		}
 		runAction.Objdir = testDir
 		vetRunAction = runAction
+		// sylixos only have cross compiler, so no need to run test on HOST, only on TARGET
+		if cfg.Goos == "sylixos" && runtime.GOOS != "sylixos" {
+			runAction.Actor = work.ActorFunc(sylixosBuilderRunTest)
+		}
 		cleanAction = &work.Action{
 			Mode:       "test clean",
 			Actor:      work.ActorFunc(builderCleanTest),
@@ -2072,6 +2078,44 @@ func builderPrintTest(b *work.Builder, ctx context.Context, a *work.Action) erro
 	return nil
 }
 
+func copyFile(in, out string) (int64, error) {
+	i, e := os.Open(in)
+	if e != nil {
+		return 0, e
+	}
+	defer i.Close()
+	o, e := os.Create(out)
+	if e != nil {
+		return 0, e
+	}
+	defer o.Close()
+	return o.ReadFrom(i)
+}
+
+func sylixosBuilderRunTest(b *work.Builder, ctx context.Context, a *work.Action) error {
+	src := a.Deps[0].BuiltTarget()
+	srcFileName := ""
+	sylixosTestDirPath := ""
+	dest := ""
+	if runtime.GOOS == "windows" {
+		homeDir := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		sylixosTestDirPath = filepath.Join(homeDir, "go-test\\sylixos-tmp\\"+cfg.Goarch)
+		srcFileName = path.Base(strings.Replace(src, "\\", "/", -1))
+	} else {
+		homeDir := os.Getenv("HOME")
+		sylixosTestDirPath = filepath.Join(homeDir, "go-test/sylixos-tmp/"+cfg.Goarch)
+		srcFileName = path.Base(src)
+	}
+	os.MkdirAll(sylixosTestDirPath, 0777)
+	dest = filepath.Join(sylixosTestDirPath, srcFileName)
+
+	if _, err := os.Stat(src); err == nil {
+		fmt.Printf("Cp %s to %s\n", a.Deps[0].BuiltTarget(), dest)
+		copyFile(a.Deps[0].BuiltTarget(), dest)
+	}
+	return nil
+}
+
 // printExitStatus is the action for printing the final exit status.
 // If we are running multiple test targets, print a final "FAIL"
 // in case a failure in an early package has already scrolled
diff --git a/src/cmd/go/internal/work/init.go b/src/cmd/go/internal/work/init.go
index 7d0921f0cc..0b1b383386 100644
--- a/src/cmd/go/internal/work/init.go
+++ b/src/cmd/go/internal/work/init.go
@@ -205,7 +205,7 @@ func buildModeInit() {
 					codegenArg = "-shared"
 				}
 
-			case "dragonfly", "freebsd", "illumos", "linux", "netbsd", "openbsd", "solaris":
+			case "dragonfly", "freebsd", "illumos", "linux", "netbsd", "openbsd", "solaris", "sylixos":
 				// Use -shared so that the result is
 				// suitable for inclusion in a PIE or
 				// shared library.
@@ -220,7 +220,7 @@ func buildModeInit() {
 			codegenArg = "-fPIC"
 		} else {
 			switch cfg.Goos {
-			case "linux", "android", "freebsd":
+			case "linux", "android", "freebsd", "sylixos":
 				codegenArg = "-shared"
 			case "windows":
 				// Do not add usual .exe suffix to the .dll file.
diff --git a/src/cmd/internal/obj/x86/asm6.go b/src/cmd/internal/obj/x86/asm6.go
index dc38069edc..9c2373c2f9 100644
--- a/src/cmd/internal/obj/x86/asm6.go
+++ b/src/cmd/internal/obj/x86/asm6.go
@@ -2573,7 +2573,8 @@ func prefixof(ctxt *obj.Link, a *obj.Addr) int {
 				objabi.Hfreebsd,
 				objabi.Hnetbsd,
 				objabi.Hopenbsd,
-				objabi.Hsolaris:
+				objabi.Hsolaris,
+				objabi.Hsylixos:
 				return 0x64 // FS
 
 			case objabi.Hdarwin:
diff --git a/src/cmd/internal/objabi/head.go b/src/cmd/internal/objabi/head.go
index 3109b5cb23..1901c2644b 100644
--- a/src/cmd/internal/objabi/head.go
+++ b/src/cmd/internal/objabi/head.go
@@ -49,6 +49,7 @@ const (
 	Hwasip1
 	Hwindows
 	Haix
+	Hsylixos
 )
 
 func (h *HeadType) Set(s string) error {
@@ -77,6 +78,8 @@ func (h *HeadType) Set(s string) error {
 		*h = Hwasip1
 	case "windows":
 		*h = Hwindows
+	case "sylixos":
+		*h = Hsylixos
 	default:
 		return fmt.Errorf("invalid headtype: %q", s)
 	}
@@ -109,6 +112,8 @@ func (h HeadType) String() string {
 		return "wasip1"
 	case Hwindows:
 		return "windows"
+	case Hsylixos:
+		return "sylixos"
 	}
 	return fmt.Sprintf("HeadType(%d)", h)
 }
diff --git a/src/cmd/link/internal/amd64/obj.go b/src/cmd/link/internal/amd64/obj.go
index 3a6141b909..7c8c74314a 100644
--- a/src/cmd/link/internal/amd64/obj.go
+++ b/src/cmd/link/internal/amd64/obj.go
@@ -107,7 +107,8 @@ func archinit(ctxt *ld.Link) {
 		objabi.Hnetbsd,    /* netbsd */
 		objabi.Hopenbsd,   /* openbsd */
 		objabi.Hdragonfly, /* dragonfly */
-		objabi.Hsolaris:   /* solaris */
+		objabi.Hsolaris,   /* solaris */
+		objabi.Hsylixos:   /* sylixos */
 		ld.Elfinit(ctxt)
 
 		ld.HEADR = ld.ELFRESERVE
diff --git a/src/cmd/link/internal/arm64/asm.go b/src/cmd/link/internal/arm64/asm.go
index 7b85bb3e26..a36fe0e69c 100644
--- a/src/cmd/link/internal/arm64/asm.go
+++ b/src/cmd/link/internal/arm64/asm.go
@@ -917,7 +917,7 @@ func archreloc(target *ld.Target, ldr *loader.Loader, syms *ld.ArchSyms, r loade
 			// We are linking the final executable, so we
 			// can optimize any TLS IE relocation to LE.
 
-			if !target.IsLinux() {
+			if !target.IsLinux() && !target.IsSylixos() {
 				ldr.Errorf(s, "TLS reloc on unsupported OS %v", target.HeadType)
 			}
 
diff --git a/src/cmd/link/internal/arm64/obj.go b/src/cmd/link/internal/arm64/obj.go
index 3d358155ba..037058ca09 100644
--- a/src/cmd/link/internal/arm64/obj.go
+++ b/src/cmd/link/internal/arm64/obj.go
@@ -96,7 +96,8 @@ func archinit(ctxt *ld.Link) {
 	case objabi.Hlinux, /* arm64 elf */
 		objabi.Hfreebsd,
 		objabi.Hnetbsd,
-		objabi.Hopenbsd:
+		objabi.Hopenbsd,
+		objabi.Hsylixos:
 		ld.Elfinit(ctxt)
 		ld.HEADR = ld.ELFRESERVE
 		if *ld.FlagRound == -1 {
diff --git a/src/cmd/link/internal/ld/asmb.go b/src/cmd/link/internal/ld/asmb.go
index ca9a57741c..3bd511f7fe 100644
--- a/src/cmd/link/internal/ld/asmb.go
+++ b/src/cmd/link/internal/ld/asmb.go
@@ -111,7 +111,8 @@ func asmb2(ctxt *Link) {
 		objabi.Hlinux,
 		objabi.Hnetbsd,
 		objabi.Hopenbsd,
-		objabi.Hsolaris:
+		objabi.Hsolaris,
+		objabi.Hsylixos:
 		asmbElf(ctxt)
 	}
 
diff --git a/src/cmd/link/internal/ld/data.go b/src/cmd/link/internal/ld/data.go
index b18dc6993b..1b1dffe072 100644
--- a/src/cmd/link/internal/ld/data.go
+++ b/src/cmd/link/internal/ld/data.go
@@ -255,7 +255,7 @@ func (st *relocSymState) relocsym(s loader.Sym, P []byte) {
 
 		// We need to be able to reference dynimport symbols when linking against
 		// shared libraries, and AIX, Darwin, OpenBSD and Solaris always need it.
-		if !target.IsAIX() && !target.IsDarwin() && !target.IsSolaris() && !target.IsOpenbsd() && rs != 0 && rst == sym.SDYNIMPORT && !target.IsDynlinkingGo() && !ldr.AttrSubSymbol(rs) {
+		if !target.IsAIX() && !target.IsDarwin() && !target.IsSolaris() && !target.IsOpenbsd() && !target.IsSylixos() && rs != 0 && rst == sym.SDYNIMPORT && !target.IsDynlinkingGo() && !ldr.AttrSubSymbol(rs) {
 			if !(target.IsPPC64() && target.IsExternal() && ldr.SymName(rs) == ".TOC.") {
 				st.err.Errorf(s, "unhandled relocation for %s (type %d (%s) rtype %d (%s))", ldr.SymName(rs), rst, rst, rt, sym.RelocName(target.Arch, rt))
 			}
diff --git a/src/cmd/link/internal/ld/lib.go b/src/cmd/link/internal/ld/lib.go
index d66027387b..cbbaaaa348 100644
--- a/src/cmd/link/internal/ld/lib.go
+++ b/src/cmd/link/internal/ld/lib.go
@@ -1727,7 +1727,7 @@ func (ctxt *Link) hostlink() {
 	}
 
 	// Force global symbols to be exported for dlopen, etc.
-	if ctxt.IsELF {
+	if ctxt.IsELF && !ctxt.IsSylixos() {
 		if ctxt.DynlinkingGo() || ctxt.BuildMode == BuildModeCShared || !linkerFlagSupported(ctxt.Arch, argv[0], altLinker, "-Wl,--export-dynamic-symbol=main") {
 			argv = append(argv, "-rdynamic")
 		} else {
diff --git a/src/cmd/link/internal/ld/sym.go b/src/cmd/link/internal/ld/sym.go
index 6ae110602e..89756a96a6 100644
--- a/src/cmd/link/internal/ld/sym.go
+++ b/src/cmd/link/internal/ld/sym.go
@@ -83,7 +83,8 @@ func (ctxt *Link) computeTLSOffset() {
 		objabi.Hnetbsd,
 		objabi.Hopenbsd,
 		objabi.Hdragonfly,
-		objabi.Hsolaris:
+		objabi.Hsolaris,
+		objabi.Hsylixos:
 		/*
 		 * ELF uses TLS offset negative from FS.
 		 * Translate 0(FS) and 8(FS) into -16(FS) and -8(FS).
diff --git a/src/cmd/link/internal/ld/target.go b/src/cmd/link/internal/ld/target.go
index d0ce99f3e9..c5ef4dd399 100644
--- a/src/cmd/link/internal/ld/target.go
+++ b/src/cmd/link/internal/ld/target.go
@@ -181,6 +181,11 @@ func (t *Target) IsFreebsd() bool {
 	return t.HeadType == objabi.Hfreebsd
 }
 
+func (t *Target) IsSylixos() bool {
+	t.mustSetHeadType()
+	return t.HeadType == objabi.Hsylixos
+}
+
 func (t *Target) mustSetHeadType() {
 	if t.HeadType == objabi.Hunknown {
 		panic("HeadType is not set")
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/aliases.go b/src/cmd/vendor/golang.org/x/sys/unix/aliases.go
index b0e4198575..a2b805cdb3 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/aliases.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/aliases.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/env_unix.go b/src/cmd/vendor/golang.org/x/sys/unix/env_unix.go
index a96da71f47..b402386bfb 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/env_unix.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/env_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 // Unix environment variables.
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/fdset.go b/src/cmd/vendor/golang.org/x/sys/unix/fdset.go
index 9e83d18cd0..252204a69d 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/fdset.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/fdset.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/mmap_nomremap.go b/src/cmd/vendor/golang.org/x/sys/unix/mmap_nomremap.go
index 7f602ffd26..8193692ec5 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/mmap_nomremap.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/mmap_nomremap.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || openbsd || solaris || sylixos || zos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/race0.go b/src/cmd/vendor/golang.org/x/sys/unix/race0.go
index 4a9f6634c9..451c9b43e8 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/race0.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/race0.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || (darwin && !race) || (linux && !race) || (freebsd && !race) || netbsd || openbsd || solaris || dragonfly || zos
+//go:build aix || (darwin && !race) || (linux && !race) || (freebsd && !race) || netbsd || openbsd || solaris || dragonfly || zos || sylixos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix.go b/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix.go
index c3a62dbb1b..fae717713d 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 // Socket control messages
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix_other.go b/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix_other.go
index 4a1eab37ec..fc0ccc927d 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix_other.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/sockcmsg_unix_other.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/syscall.go b/src/cmd/vendor/golang.org/x/sys/unix/syscall.go
index 5ea74da982..700503e018 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/syscall.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/syscall.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 // Package unix contains an interface to the low-level operating system
 // primitives. OS details vary depending on the underlying system, and
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix.go b/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix.go
index 4e92e5aa40..c4c57d3a91 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package unix
 
diff --git a/src/cmd/vendor/golang.org/x/sys/unix/timestruct.go b/src/cmd/vendor/golang.org/x/sys/unix/timestruct.go
index 7997b19022..45888b4fa9 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/timestruct.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/timestruct.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos || sylixos
 
 package unix
 
diff --git a/src/crypto/x509/root_unix.go b/src/crypto/x509/root_unix.go
index c513b20169..b69bbac3ec 100644
--- a/src/crypto/x509/root_unix.go
+++ b/src/crypto/x509/root_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || dragonfly || freebsd || (js && wasm) || linux || netbsd || openbsd || solaris || wasip1
+//go:build aix || dragonfly || freebsd || (js && wasm) || linux || netbsd || openbsd || solaris || wasip1 || sylixos
 
 package x509
 
diff --git a/src/crypto/x509/root_unix_test.go b/src/crypto/x509/root_unix_test.go
index d5215b9ff2..15f1a1049e 100644
--- a/src/crypto/x509/root_unix_test.go
+++ b/src/crypto/x509/root_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package x509
 
diff --git a/src/go/build/syslist.go b/src/go/build/syslist.go
index 783bbe697a..7cb444a48e 100644
--- a/src/go/build/syslist.go
+++ b/src/go/build/syslist.go
@@ -30,6 +30,7 @@ var knownOS = map[string]bool{
 	"wasip1":    true,
 	"windows":   true,
 	"zos":       true,
+	"sylixos":   true,
 }
 
 // unixOS is the set of GOOS values matched by the "unix" build tag.
@@ -49,6 +50,7 @@ var unixOS = map[string]bool{
 	"netbsd":    true,
 	"openbsd":   true,
 	"solaris":   true,
+	"sylixos":   true,
 }
 
 // knownArch is the list of past, present, and future known GOARCH values.
diff --git a/src/html/template/exec_test.go b/src/html/template/exec_test.go
index f60308e679..0babf90fdb 100644
--- a/src/html/template/exec_test.go
+++ b/src/html/template/exec_test.go
@@ -13,6 +13,7 @@ import (
 	"fmt"
 	"io"
 	"reflect"
+	"runtime"
 	"strings"
 	"sync"
 	"testing"
@@ -1618,6 +1619,9 @@ func TestInterfaceValues(t *testing.T) {
 
 // Check that panics during calls are recovered and returned as errors.
 func TestExecutePanicDuringCall(t *testing.T) {
+	if runtime.GOOS == "sylixos" {
+		t.Skip("because sylixos only have on page table, so this test will crash kernel!, skip it.")
+	}
 	funcs := map[string]any{
 		"doPanic": func() string {
 			panic("custom panic string")
diff --git a/src/internal/cpu/cpu_arm64_other.go b/src/internal/cpu/cpu_arm64_other.go
index 44592cfced..29cae6a84e 100644
--- a/src/internal/cpu/cpu_arm64_other.go
+++ b/src/internal/cpu/cpu_arm64_other.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build arm64 && !linux && !freebsd && !android && (!darwin || ios) && !openbsd
+//go:build arm64 && !linux && !freebsd && !android && (!darwin || ios) && !openbsd && !sylixos
 
 package cpu
 
diff --git a/src/internal/fuzz/minimize_test.go b/src/internal/fuzz/minimize_test.go
index 2db2633896..0f49c17976 100644
--- a/src/internal/fuzz/minimize_test.go
+++ b/src/internal/fuzz/minimize_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || freebsd || linux || windows
+//go:build darwin || freebsd || linux || windows || sylixos
 
 package fuzz
 
diff --git a/src/internal/goos/zgoos_aix.go b/src/internal/goos/zgoos_aix.go
index 24e05c933e..b6ccd18138 100644
--- a/src/internal/goos/zgoos_aix.go
+++ b/src/internal/goos/zgoos_aix.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_android.go b/src/internal/goos/zgoos_android.go
index 3c4a318590..ed788c98a5 100644
--- a/src/internal/goos/zgoos_android.go
+++ b/src/internal/goos/zgoos_android.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_darwin.go b/src/internal/goos/zgoos_darwin.go
index 10b1499895..d8b7df1833 100644
--- a/src/internal/goos/zgoos_darwin.go
+++ b/src/internal/goos/zgoos_darwin.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_dragonfly.go b/src/internal/goos/zgoos_dragonfly.go
index b92d1269f1..216ba1c71a 100644
--- a/src/internal/goos/zgoos_dragonfly.go
+++ b/src/internal/goos/zgoos_dragonfly.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_freebsd.go b/src/internal/goos/zgoos_freebsd.go
index f547591ab1..9919a3185e 100644
--- a/src/internal/goos/zgoos_freebsd.go
+++ b/src/internal/goos/zgoos_freebsd.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_hurd.go b/src/internal/goos/zgoos_hurd.go
index 1189d65d74..8f3dafd68e 100644
--- a/src/internal/goos/zgoos_hurd.go
+++ b/src/internal/goos/zgoos_hurd.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_illumos.go b/src/internal/goos/zgoos_illumos.go
index 4f0254081c..2ec514267f 100644
--- a/src/internal/goos/zgoos_illumos.go
+++ b/src/internal/goos/zgoos_illumos.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_ios.go b/src/internal/goos/zgoos_ios.go
index 02f3586fa4..2a3a81a0e4 100644
--- a/src/internal/goos/zgoos_ios.go
+++ b/src/internal/goos/zgoos_ios.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_js.go b/src/internal/goos/zgoos_js.go
index 4818741891..874cf764c3 100644
--- a/src/internal/goos/zgoos_js.go
+++ b/src/internal/goos/zgoos_js.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_linux.go b/src/internal/goos/zgoos_linux.go
index 6f4d4e0753..d9c477e3c5 100644
--- a/src/internal/goos/zgoos_linux.go
+++ b/src/internal/goos/zgoos_linux.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_netbsd.go b/src/internal/goos/zgoos_netbsd.go
index 948603df0d..8f6bd74203 100644
--- a/src/internal/goos/zgoos_netbsd.go
+++ b/src/internal/goos/zgoos_netbsd.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_openbsd.go b/src/internal/goos/zgoos_openbsd.go
index f4b201457b..e6ecf48f00 100644
--- a/src/internal/goos/zgoos_openbsd.go
+++ b/src/internal/goos/zgoos_openbsd.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_plan9.go b/src/internal/goos/zgoos_plan9.go
index 95572dff37..093897ee9f 100644
--- a/src/internal/goos/zgoos_plan9.go
+++ b/src/internal/goos/zgoos_plan9.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_solaris.go b/src/internal/goos/zgoos_solaris.go
index c7058260f8..e7f3bd40df 100644
--- a/src/internal/goos/zgoos_solaris.go
+++ b/src/internal/goos/zgoos_solaris.go
@@ -24,3 +24,4 @@ const IsSolaris = 1
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_wasip1.go b/src/internal/goos/zgoos_wasip1.go
index ae35eebac6..794cc50913 100644
--- a/src/internal/goos/zgoos_wasip1.go
+++ b/src/internal/goos/zgoos_wasip1.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 1
 const IsWindows = 0
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_windows.go b/src/internal/goos/zgoos_windows.go
index f89f4cf829..fb1d7a9df7 100644
--- a/src/internal/goos/zgoos_windows.go
+++ b/src/internal/goos/zgoos_windows.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 1
 const IsZos = 0
+const IsSylixos = 0
diff --git a/src/internal/goos/zgoos_zos.go b/src/internal/goos/zgoos_zos.go
index 29fb0f8bab..34a0cab293 100644
--- a/src/internal/goos/zgoos_zos.go
+++ b/src/internal/goos/zgoos_zos.go
@@ -24,3 +24,4 @@ const IsSolaris = 0
 const IsWasip1 = 0
 const IsWindows = 0
 const IsZos = 1
+const IsSylixos = 0
diff --git a/src/internal/platform/supported.go b/src/internal/platform/supported.go
index a774247e6b..f0057ce46d 100644
--- a/src/internal/platform/supported.go
+++ b/src/internal/platform/supported.go
@@ -173,7 +173,8 @@ func BuildModeSupported(compiler, buildmode, goos, goarch string) bool {
 			"android/amd64", "android/arm", "android/arm64", "android/386",
 			"freebsd/amd64",
 			"darwin/amd64", "darwin/arm64",
-			"windows/amd64", "windows/386", "windows/arm64":
+			"windows/amd64", "windows/386", "windows/arm64",
+			"sylixos/arm64", "sylixos/amd64":
 			return true
 		}
 		return false
@@ -193,14 +194,16 @@ func BuildModeSupported(compiler, buildmode, goos, goarch string) bool {
 			"ios/amd64", "ios/arm64",
 			"aix/ppc64",
 			"openbsd/arm64",
-			"windows/386", "windows/amd64", "windows/arm", "windows/arm64":
+			"windows/386", "windows/amd64", "windows/arm", "windows/arm64",
+			"sylixos/arm64", "sylixos/amd64":
 			return true
 		}
 		return false
 
 	case "shared":
 		switch platform {
-		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le", "linux/s390x":
+		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le", "linux/s390x",
+			"sylixos/arm64", "sylixos/amd64":
 			return true
 		}
 		return false
@@ -210,7 +213,8 @@ func BuildModeSupported(compiler, buildmode, goos, goarch string) bool {
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/loong64", "linux/s390x", "linux/ppc64le",
 			"android/amd64", "android/386",
 			"darwin/amd64", "darwin/arm64",
-			"freebsd/amd64":
+			"freebsd/amd64",
+			"sylixos/arm64", "sylixos/amd64":
 			return true
 		}
 		return false
@@ -225,7 +229,8 @@ func InternalLinkPIESupported(goos, goarch string) bool {
 	case "android/arm64",
 		"darwin/amd64", "darwin/arm64",
 		"linux/amd64", "linux/arm64", "linux/ppc64le",
-		"windows/386", "windows/amd64", "windows/arm", "windows/arm64":
+		"windows/386", "windows/amd64", "windows/arm", "windows/arm64",
+		"sylixos/arm64", "sylixos/amd64":
 		return true
 	}
 	return false
@@ -247,6 +252,8 @@ func DefaultPIE(goos, goarch string, isRace bool) bool {
 		return true
 	case "darwin":
 		return true
+	case "sylixos":
+		return true
 	}
 	return false
 }
diff --git a/src/internal/platform/zosarch.go b/src/internal/platform/zosarch.go
index 1df348518c..6cab464bab 100644
--- a/src/internal/platform/zosarch.go
+++ b/src/internal/platform/zosarch.go
@@ -54,6 +54,8 @@ var List = []OSArch{
 	{"plan9", "amd64"},
 	{"plan9", "arm"},
 	{"solaris", "amd64"},
+	{"sylixos", "amd64"},
+	{"sylixos", "arm64"},
 	{"wasip1", "wasm"},
 	{"windows", "386"},
 	{"windows", "amd64"},
@@ -108,6 +110,8 @@ var distInfo = map[OSArch]osArchInfo{
 	{"plan9", "amd64"}:     {},
 	{"plan9", "arm"}:       {},
 	{"solaris", "amd64"}:   {CgoSupported: true},
+	{"sylixos", "amd64"}:   {CgoSupported: true},
+	{"sylixos", "arm64"}:   {CgoSupported: true},
 	{"wasip1", "wasm"}:     {},
 	{"windows", "386"}:     {CgoSupported: true, FirstClass: true},
 	{"windows", "amd64"}:   {CgoSupported: true, FirstClass: true},
diff --git a/src/internal/poll/fd_fsync_posix.go b/src/internal/poll/fd_fsync_posix.go
index 469ca75b62..8fd3a7b361 100644
--- a/src/internal/poll/fd_fsync_posix.go
+++ b/src/internal/poll/fd_fsync_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || dragonfly || freebsd || (js && wasm) || linux || netbsd || openbsd || solaris || wasip1
+//go:build aix || dragonfly || freebsd || (js && wasm) || linux || netbsd || openbsd || solaris || wasip1 || sylixos
 
 package poll
 
diff --git a/src/internal/poll/fd_writev_libc.go b/src/internal/poll/fd_writev_libc.go
index 0a60473b3e..24c5e5bad3 100644
--- a/src/internal/poll/fd_writev_libc.go
+++ b/src/internal/poll/fd_writev_libc.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || (openbsd && !mips64) || solaris
+//go:build aix || darwin || (openbsd && !mips64) || solaris || sylixos
 
 package poll
 
diff --git a/src/internal/poll/hook_cloexec.go b/src/internal/poll/hook_cloexec.go
index 5b3cdcec28..e97c116549 100644
--- a/src/internal/poll/hook_cloexec.go
+++ b/src/internal/poll/hook_cloexec.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package poll
 
diff --git a/src/internal/poll/iovec_unix.go b/src/internal/poll/iovec_unix.go
index 3f2833e728..50aae04a2b 100644
--- a/src/internal/poll/iovec_unix.go
+++ b/src/internal/poll/iovec_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || sylixos
 
 package poll
 
diff --git a/src/internal/poll/sendfile_bsd.go b/src/internal/poll/sendfile_bsd.go
index 341e07ca1f..afb6387473 100644
--- a/src/internal/poll/sendfile_bsd.go
+++ b/src/internal/poll/sendfile_bsd.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd
+//go:build darwin || dragonfly || freebsd || sylixos
 
 package poll
 
diff --git a/src/internal/poll/sock_cloexec.go b/src/internal/poll/sock_cloexec.go
index cbf7021804..68dba93760 100644
--- a/src/internal/poll/sock_cloexec.go
+++ b/src/internal/poll/sock_cloexec.go
@@ -5,7 +5,7 @@
 // This file implements accept for platforms that provide a fast path for
 // setting SetNonblock and CloseOnExec.
 
-//go:build dragonfly || freebsd || (linux && !arm) || netbsd || openbsd
+//go:build dragonfly || freebsd || (linux && !arm) || netbsd || openbsd || sylixos
 
 package poll
 
diff --git a/src/internal/sysinfo/cpuinfo_stub.go b/src/internal/sysinfo/cpuinfo_stub.go
index 2ac7ffafe4..b71e29d97a 100644
--- a/src/internal/sysinfo/cpuinfo_stub.go
+++ b/src/internal/sysinfo/cpuinfo_stub.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !(darwin || freebsd || linux || netbsd || openbsd)
+//go:build !(darwin || freebsd || linux || netbsd || openbsd || sylixos)
 
 package sysinfo
 
diff --git a/src/internal/testenv/testenv.go b/src/internal/testenv/testenv.go
index 9fb92406e8..24a2a0f076 100644
--- a/src/internal/testenv/testenv.go
+++ b/src/internal/testenv/testenv.go
@@ -408,7 +408,7 @@ func HasLink() bool {
 	// From Android release M (Marshmallow), hard linking files is blocked
 	// and an attempt to call link() on a file will return EACCES.
 	// - https://code.google.com/p/android-developer-preview/issues/detail?id=3150
-	return runtime.GOOS != "plan9" && runtime.GOOS != "android"
+	return runtime.GOOS != "plan9" && runtime.GOOS != "android" && runtime.GOOS != "sylixos"
 }
 
 // MustHaveLink reports whether the current system can use os.Link.
diff --git a/src/log/slog/logger_test.go b/src/log/slog/logger_test.go
index bb1c8a16ea..67fb82f35c 100644
--- a/src/log/slog/logger_test.go
+++ b/src/log/slog/logger_test.go
@@ -96,10 +96,15 @@ func TestConnections(t *testing.T) {
 	Info("msg", "p", nil)
 	checkLogOutput(t, logbuf.String(), `logger_test.go:\d+: INFO msg p=<nil>`)
 	logbuf.Reset()
-	var r *regexp.Regexp
-	Info("msg", "r", r)
-	checkLogOutput(t, logbuf.String(), `logger_test.go:\d+: INFO msg r=<nil>`)
-	logbuf.Reset()
+	// because sylixos only have on page table, var r *regexp.Regexp is nil,
+	// sylixos kernel takes this nil as 0x0 on page table, so this test will crash kernel!,
+	// skip it.
+	if runtime.GOOS != "sylixos" {
+		var r *regexp.Regexp
+		Info("msg", "r", r)
+		checkLogOutput(t, logbuf.String(), `logger_test.go:\d+: INFO msg r=<nil>`)
+		logbuf.Reset()
+	}
 	Warn("msg", "b", 2)
 	checkLogOutput(t, logbuf.String(), `logger_test.go:\d+: WARN msg b=2`)
 	logbuf.Reset()
diff --git a/src/log/syslog/syslog_test.go b/src/log/syslog/syslog_test.go
index cec225f751..57a4e4b1c1 100644
--- a/src/log/syslog/syslog_test.go
+++ b/src/log/syslog/syslog_test.go
@@ -193,6 +193,12 @@ func TestFlap(t *testing.T) {
 	}
 	check(t, msg, <-done, net)
 
+	// ACOINFO TODO SylixOS no support AF_UNIX SO_REUSEADDR now
+	if runtime.GOOS == "sylixos" {
+		s.Close()
+		return
+	}
+
 	// restart the server
 	if err := os.Remove(addr); err != nil {
 		t.Fatal(err)
diff --git a/src/net/addrselect_test.go b/src/net/addrselect_test.go
index 7e8134d754..814ae5f34d 100644
--- a/src/net/addrselect_test.go
+++ b/src/net/addrselect_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package net
 
diff --git a/src/net/dial_test.go b/src/net/dial_test.go
index b3bedb2fa2..b3548531e7 100644
--- a/src/net/dial_test.go
+++ b/src/net/dial_test.go
@@ -987,7 +987,7 @@ func TestDialListenerAddr(t *testing.T) {
 
 func TestDialerControl(t *testing.T) {
 	switch runtime.GOOS {
-	case "plan9":
+	case "plan9", "sylixos":
 		t.Skipf("not supported on %s", runtime.GOOS)
 	case "js", "wasip1":
 		t.Skipf("skipping: fake net does not support Dialer.Control")
@@ -1032,7 +1032,7 @@ func TestDialerControl(t *testing.T) {
 
 func TestDialerControlContext(t *testing.T) {
 	switch runtime.GOOS {
-	case "plan9":
+	case "plan9", "sylixos":
 		t.Skipf("%s does not have full support of socktest", runtime.GOOS)
 	case "js", "wasip1":
 		t.Skipf("skipping: fake net does not support Dialer.ControlContext")
diff --git a/src/net/http/serve_test.go b/src/net/http/serve_test.go
index b2858ba8f2..da16bafa5d 100644
--- a/src/net/http/serve_test.go
+++ b/src/net/http/serve_test.go
@@ -7024,6 +7024,11 @@ func testParseFormCleanup(t *testing.T, mode testMode) {
 			return
 		}
 		w.Write([]byte(of.Name()))
+
+		if runtime.GOOS == "sylixos" {
+			// SylixOS need close file
+			of.Close()
+		}
 	}))
 
 	fBuf := new(bytes.Buffer)
diff --git a/src/net/http/transport_test.go b/src/net/http/transport_test.go
index ae7159dab0..e99ca10541 100644
--- a/src/net/http/transport_test.go
+++ b/src/net/http/transport_test.go
@@ -3394,8 +3394,12 @@ func testProxyForRequest(t *testing.T, tt proxyFromEnvTest, proxyForRequest func
 		t.Errorf("%v: got error = %q, want %q", tt, g, e)
 		return
 	}
-	if got := fmt.Sprintf("%s", url); got != tt.want {
-		t.Errorf("%v: got URL = %q, want %q", tt, url, tt.want)
+	if url != nil {
+		if got := fmt.Sprintf("%s", url); got != tt.want {
+			t.Errorf("%v: got URL = %q, want %q", tt, url, tt.want)
+		}
+	} else if tt.want != "<nil>" {
+		t.Errorf("%v: got URL = nil, want %q %d", tt, tt.want, len(tt.want))
 	}
 }
 
diff --git a/src/net/nss_test.go b/src/net/nss_test.go
index 94e6b5fc0a..fee388b842 100644
--- a/src/net/nss_test.go
+++ b/src/net/nss_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package net
 
diff --git a/src/net/sendfile_stub.go b/src/net/sendfile_stub.go
index 7f31cc63e1..812a054c4a 100644
--- a/src/net/sendfile_stub.go
+++ b/src/net/sendfile_stub.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !(linux || (darwin && !ios) || dragonfly || freebsd || solaris || windows)
+//go:build !(linux || (darwin && !ios) || dragonfly || freebsd || solaris || windows || sylixos)
 
 package net
 
diff --git a/src/net/sendfile_unix_alt.go b/src/net/sendfile_unix_alt.go
index 9e46c4e607..b8160b7d40 100644
--- a/src/net/sendfile_unix_alt.go
+++ b/src/net/sendfile_unix_alt.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build (darwin && !ios) || dragonfly || freebsd || solaris
+//go:build (darwin && !ios) || dragonfly || freebsd || solaris || sylixos
 
 package net
 
diff --git a/src/net/sock_cloexec.go b/src/net/sock_cloexec.go
index 043522f0b6..f111655b31 100644
--- a/src/net/sock_cloexec.go
+++ b/src/net/sock_cloexec.go
@@ -5,7 +5,7 @@
 // This file implements sysSocket for platforms that provide a fast path for
 // setting SetNonblock and CloseOnExec.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || sylixos
 
 package net
 
diff --git a/src/net/sock_stub.go b/src/net/sock_stub.go
index fd86fa92dc..aa7ccdb40b 100644
--- a/src/net/sock_stub.go
+++ b/src/net/sock_stub.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || js || solaris || wasip1
+//go:build aix || js || solaris || wasip1 || sylixos
 
 package net
 
diff --git a/src/net/sockoptip_bsdvar.go b/src/net/sockoptip_bsdvar.go
index 3e9ba1ee78..16aa8b43ee 100644
--- a/src/net/sockoptip_bsdvar.go
+++ b/src/net/sockoptip_bsdvar.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || netbsd || openbsd || solaris || sylixos
 
 package net
 
diff --git a/src/net/tcpsockopt_unix.go b/src/net/tcpsockopt_unix.go
index 1f7617897a..3187e32e58 100644
--- a/src/net/tcpsockopt_unix.go
+++ b/src/net/tcpsockopt_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || dragonfly || freebsd || illumos || linux || netbsd
+//go:build aix || dragonfly || freebsd || illumos || linux || netbsd || sylixos
 
 package net
 
diff --git a/src/net/unixsock_readmsg_cmsg_cloexec.go b/src/net/unixsock_readmsg_cmsg_cloexec.go
index 6b0de875ad..948039be3e 100644
--- a/src/net/unixsock_readmsg_cmsg_cloexec.go
+++ b/src/net/unixsock_readmsg_cmsg_cloexec.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || linux || netbsd || openbsd
+//go:build dragonfly || linux || netbsd || openbsd || sylixos
 
 package net
 
diff --git a/src/net/write_unix_test.go b/src/net/write_unix_test.go
index 23e8befa92..f780bb7346 100644
--- a/src/net/write_unix_test.go
+++ b/src/net/write_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package net
 
diff --git a/src/net/writev_test.go b/src/net/writev_test.go
index e4e88c4fac..0d0f959763 100644
--- a/src/net/writev_test.go
+++ b/src/net/writev_test.go
@@ -151,7 +151,7 @@ func testBuffer_writeTo(t *testing.T, chunks int, useCopy bool) {
 
 		var wantSum int
 		switch runtime.GOOS {
-		case "aix", "android", "darwin", "ios", "dragonfly", "freebsd", "illumos", "linux", "netbsd", "openbsd", "solaris":
+		case "aix", "android", "darwin", "ios", "dragonfly", "freebsd", "illumos", "linux", "netbsd", "openbsd", "solaris", "sylixos":
 			var wantMinCalls int
 			wantSum = want.Len()
 			v := chunks
diff --git a/src/os/exec/bench_test.go b/src/os/exec/bench_test.go
index 9a94001e84..9c0e6c78c5 100644
--- a/src/os/exec/bench_test.go
+++ b/src/os/exec/bench_test.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build !sylixos
+
 package exec
 
 import (
diff --git a/src/os/exec/exec_test.go b/src/os/exec/exec_test.go
index a0bb89e203..365c66ed0c 100644
--- a/src/os/exec/exec_test.go
+++ b/src/os/exec/exec_test.go
@@ -51,7 +51,7 @@ func init() {
 	if os.Getenv("GO_EXEC_TEST_PID") != "" {
 		return
 	}
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows" || runtime.GOOS == "sylixos" {
 		return
 	}
 	for fd := uintptr(3); fd <= 100; fd++ {
@@ -677,6 +677,10 @@ func TestPipeLookPathLeak(t *testing.T) {
 		return fds
 	}
 
+	if runtime.GOOS == "sylixos" { //ACOINFO TODO : Wait for netpoll start, because it used 3 fds.
+		time.Sleep(time.Second)
+	}
+
 	old := map[uintptr]bool{}
 	for _, fd := range openFDs() {
 		old[fd] = true
diff --git a/src/os/fifo_test.go b/src/os/fifo_test.go
index 3b7e5eac19..b5a37017c6 100644
--- a/src/os/fifo_test.go
+++ b/src/os/fifo_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || (linux && !android) || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || (linux && !android) || netbsd || openbsd || sylixos
 
 package os_test
 
@@ -14,6 +14,7 @@ import (
 	"io/fs"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strconv"
 	"sync"
 	"syscall"
@@ -24,8 +25,18 @@ import (
 func TestFifoEOF(t *testing.T) {
 	t.Parallel()
 
-	dir := t.TempDir()
+	var dir string
+	if runtime.GOOS == "sylixos" {
+		dir = "/dev/pipe"
+	} else {
+		dir = t.TempDir()
+	}
+
 	fifoName := filepath.Join(dir, "fifo")
+	if runtime.GOOS == "sylixos" {
+		defer os.Remove(fifoName)
+	}
+
 	if err := syscall.Mkfifo(fifoName, 0600); err != nil {
 		t.Fatal(err)
 	}
@@ -210,7 +221,18 @@ func TestNewFileNonBlocking(t *testing.T) {
 }
 
 func TestFIFONonBlockingEOF(t *testing.T) {
-	fifoName := filepath.Join(t.TempDir(), "issue-66239-fifo")
+	var dir string
+	if runtime.GOOS == "sylixos" {
+		dir = "/dev/pipe"
+	} else {
+		dir = t.TempDir()
+	}
+
+	fifoName := filepath.Join(dir, "issue-66239-fifo")
+	if runtime.GOOS == "sylixos" {
+		defer os.Remove(fifoName)
+	}
+
 	if err := syscall.Mkfifo(fifoName, 0600); err != nil {
 		t.Fatalf("Error creating fifo: %v", err)
 	}
diff --git a/src/os/file_unix.go b/src/os/file_unix.go
index 37bfaa1a72..90e8af4cb9 100644
--- a/src/os/file_unix.go
+++ b/src/os/file_unix.go
@@ -206,6 +206,16 @@ func newFile(fd int, name string, kind newFileKind, nonBlocking bool) *File {
 			if (runtime.GOOS == "darwin" || runtime.GOOS == "ios") && typ == syscall.S_IFIFO {
 				pollable = false
 			}
+
+		case "sylixos":
+			var st syscall.Stat_t
+			err := ignoringEINTR(func() error {
+				return syscall.Fstat(fd, &st)
+			})
+			typ := st.Mode & syscall.S_IFMT
+			if err == nil && typ == syscall.S_IFDIR {
+				pollable = false
+			}
 		}
 	}
 
diff --git a/src/os/os_test.go b/src/os/os_test.go
index f1755dfa91..fb265638a8 100644
--- a/src/os/os_test.go
+++ b/src/os/os_test.go
@@ -1231,6 +1231,8 @@ func TestStartProcess(t *testing.T) {
 	switch runtime.GOOS {
 	case "android":
 		t.Skip("android doesn't have /bin/pwd")
+	case "sylixos":
+		t.Skip("SylixOS doesn't have /bin/pwd")
 	case "windows":
 		cmd = Getenv("COMSPEC")
 		dir = Getenv("SystemRoot")
@@ -1281,7 +1283,7 @@ func TestChmod(t *testing.T) {
 	checkMode(t, f.Name(), fm)
 
 	fm = FileMode(0123)
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows" || runtime.GOOS == "sylixos" {
 		fm = FileMode(0666) // read-write file
 	}
 	if err := f.Chmod(fm); err != nil {
@@ -1525,6 +1527,8 @@ func testChtimes(t *testing.T, name string) {
 			// content.  Similarly, atime is set whenever
 			// the contents are accessed; also, it is set
 			// whenever mtime is set.
+		case "sylixos":
+			// ACOINFO TODO: Atime is changed in Chtimes which calling close.
 		case "netbsd":
 			if hasNoatime() {
 				t.Log(errormsg)
@@ -2331,7 +2335,7 @@ func TestStatDirModeExec(t *testing.T) {
 
 func TestStatStdin(t *testing.T) {
 	switch runtime.GOOS {
-	case "android", "plan9":
+	case "android", "plan9", "sylixos":
 		t.Skipf("%s doesn't have /bin/sh", runtime.GOOS)
 	}
 
@@ -2704,8 +2708,8 @@ func TestPipeThreads(t *testing.T) {
 
 	threads := 100
 
-	// OpenBSD has a low default for max number of files.
-	if runtime.GOOS == "openbsd" {
+	// OpenBSD and SylixOS has a low default for max number of files.
+	if runtime.GOOS == "openbsd" || runtime.GOOS == "sylixos" {
 		threads = 50
 	}
 
@@ -2867,6 +2871,10 @@ func TestUserHomeDir(t *testing.T) {
 }
 
 func TestDirSeek(t *testing.T) {
+	if runtime.GOOS == "sylixos" { //ACOINFO TODO: SylixOS not support seek directory
+		t.Skip("SylixOS not support seek directory")
+	}
+
 	t.Parallel()
 
 	wd, err := Getwd()
@@ -2907,6 +2915,9 @@ func TestDirSeek(t *testing.T) {
 }
 
 func TestReaddirSmallSeek(t *testing.T) {
+	if runtime.GOOS == "sylixos" { //ACOINFO TODO: SylixOS not support seek directory
+		t.Skip("SylixOS not support seek directory")
+	}
 	// See issue 37161. Read only one entry from a directory,
 	// seek to the beginning, and read again. We should not see
 	// duplicate entries.
@@ -3048,9 +3059,11 @@ func TestDirFS(t *testing.T) {
 
 	// Test that Open does not accept backslash as separator.
 	d := DirFS(".")
-	_, err = d.Open(`testdata\dirfs`)
-	if err == nil {
-		t.Fatalf(`Open testdata\dirfs succeeded`)
+	if runtime.GOOS != "sylixos" { //ACOINFO TODO: SylixOS accept backslash as separator.
+		_, err = d.Open(`testdata\dirfs`)
+		if err == nil {
+			t.Fatalf(`Open testdata\dirfs succeeded`)
+		}
 	}
 
 	// Test that Open does not open Windows device files.
@@ -3097,8 +3110,8 @@ func TestDirFSEmptyDir(t *testing.T) {
 }
 
 func TestDirFSPathsValid(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		t.Skipf("skipping on Windows")
+	if runtime.GOOS == "windows" || runtime.GOOS == "sylixos" { // ACOINFO TODO: SylixOS not support ':' in path.
+		t.Skipf("skipping on %s", runtime.GOOS)
 	}
 	t.Parallel()
 
diff --git a/src/os/pipe2_unix.go b/src/os/pipe2_unix.go
index dca83a529b..04b8169715 100644
--- a/src/os/pipe2_unix.go
+++ b/src/os/pipe2_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package os
 
diff --git a/src/os/pipe_test.go b/src/os/pipe_test.go
index a9e0c8bc8a..121832e9bc 100644
--- a/src/os/pipe_test.go
+++ b/src/os/pipe_test.go
@@ -73,6 +73,8 @@ func TestStdPipe(t *testing.T) {
 	switch runtime.GOOS {
 	case "windows":
 		t.Skip("Windows doesn't support SIGPIPE")
+	case "sylixos": //ACOINFO TODO : SylixOS sub process signal and pipe can not compatible linux
+		t.Skip("SylixOS doesn't support SIGPIPE")
 	}
 
 	if os.Getenv("GO_TEST_STD_PIPE_HELPER") != "" {
diff --git a/src/os/removeall_at.go b/src/os/removeall_at.go
index cc254e0043..ba52e4220f 100644
--- a/src/os/removeall_at.go
+++ b/src/os/removeall_at.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build unix
+//go:build unix && !sylixos
 
 package os
 
diff --git a/src/os/removeall_noat.go b/src/os/removeall_noat.go
index 2b8a7727f4..c5640c98bd 100644
--- a/src/os/removeall_noat.go
+++ b/src/os/removeall_noat.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !unix
+//go:build !unix || sylixos
 
 package os
 
diff --git a/src/os/stat_test.go b/src/os/stat_test.go
index 36da573f0c..d890d49f5f 100644
--- a/src/os/stat_test.go
+++ b/src/os/stat_test.go
@@ -301,7 +301,10 @@ func TestSymlinkWithTrailingSlash(t *testing.T) {
 	}
 	dirlinkWithSlash := dirlink + string(os.PathSeparator)
 
-	testDirStats(t, dirlinkWithSlash)
+	// sylixos quirk, but symlink itslef is good.
+	if runtime.GOOS != "sylixos" {
+		testDirStats(t, dirlinkWithSlash)
+	}
 
 	fi1, err := os.Stat(dir)
 	if err != nil {
diff --git a/src/os/user/listgroups_unix.go b/src/os/user/listgroups_unix.go
index 67bd8a776e..2e30953ba1 100644
--- a/src/os/user/listgroups_unix.go
+++ b/src/os/user/listgroups_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build ((darwin || dragonfly || freebsd || (js && wasm) || wasip1 || (!android && linux) || netbsd || openbsd || solaris) && ((!cgo && !darwin) || osusergo)) || aix || illumos
+//go:build ((darwin || dragonfly || freebsd || (js && wasm) || wasip1 || (!android && linux) || netbsd || openbsd || solaris) && ((!cgo && !darwin) || osusergo)) || aix || illumos || sylixos
 
 package user
 
diff --git a/src/os/user/listgroups_unix_test.go b/src/os/user/listgroups_unix_test.go
index 9619462cec..ef1924411c 100644
--- a/src/os/user/listgroups_unix_test.go
+++ b/src/os/user/listgroups_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build ((darwin || dragonfly || freebsd || (js && wasm) || wasip1 || (!android && linux) || netbsd || openbsd || solaris) && ((!cgo && !darwin) || osusergo)) || aix || illumos
+//go:build ((darwin || dragonfly || freebsd || (js && wasm) || wasip1 || (!android && linux) || netbsd || openbsd || solaris) && ((!cgo && !darwin) || osusergo)) || aix || illumos || sylixos
 
 package user
 
diff --git a/src/os/user/lookup_unix.go b/src/os/user/lookup_unix.go
index a4308269e0..cce66726e6 100644
--- a/src/os/user/lookup_unix.go
+++ b/src/os/user/lookup_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build ((unix && !android) || (js && wasm) || wasip1) && ((!cgo && !darwin) || osusergo)
+//go:build ((unix && !android && !sylixos) || (js && wasm) || wasip1) && ((!cgo && !darwin) || osusergo)
 
 package user
 
diff --git a/src/os/user/lookup_unix_test.go b/src/os/user/lookup_unix_test.go
index 78b33922ba..9abc64f835 100644
--- a/src/os/user/lookup_unix_test.go
+++ b/src/os/user/lookup_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build unix && !android && !cgo && !darwin
+//go:build unix && !android && !cgo && !darwin && !sylixos
 
 package user
 
diff --git a/src/os/wait_unimp.go b/src/os/wait_unimp.go
index 810e35da63..74f3884eea 100644
--- a/src/os/wait_unimp.go
+++ b/src/os/wait_unimp.go
@@ -5,7 +5,7 @@
 // aix, darwin, js/wasm, openbsd, solaris and wasip1/wasm don't implement
 // waitid/wait6.
 
-//go:build aix || darwin || (js && wasm) || openbsd || solaris || wasip1
+//go:build aix || darwin || (js && wasm) || openbsd || solaris || wasip1 || sylixos
 
 package os
 
diff --git a/src/reflect/all_test.go b/src/reflect/all_test.go
index a57f17f8c3..a9d1be4d03 100644
--- a/src/reflect/all_test.go
+++ b/src/reflect/all_test.go
@@ -5931,37 +5931,42 @@ func TestStructOfWithInterface(t *testing.T) {
 		}
 	}
 
-	// Test an embedded nil pointer with pointer methods.
-	fields := []StructField{{
-		Name:      "StructIPtr",
-		Anonymous: true,
-		Type:      PointerTo(TypeOf(StructIPtr(want))),
-	}}
-	rt := StructOf(fields)
-	rv := New(rt).Elem()
-	// This should panic since the pointer is nil.
-	shouldPanic("", func() {
-		rv.Interface().(IfaceSet).Set(want)
-	})
+	// because sylixos only have on page table, var TypeOf(StructIPtr(want))|TypeOf(SettableStruct{}) is nil,
+	// sylixos kernel takes this nil as 0x0 on page table, so this test will crash kernel!,
+	// skip it.
+	if runtime.GOOS != "sylixos" {
+		// Test an embedded nil pointer with pointer methods.
+		fields := []StructField{{
+			Name:      "StructIPtr",
+			Anonymous: true,
+			Type:      PointerTo(TypeOf(StructIPtr(want))),
+		}}
+		rt := StructOf(fields)
+		rv := New(rt).Elem()
+		// This should panic since the pointer is nil.
+		shouldPanic("", func() {
+			rv.Interface().(IfaceSet).Set(want)
+		})
 
-	// Test an embedded nil pointer to a struct with pointer methods.
+		// Test an embedded nil pointer to a struct with pointer methods.
 
-	fields = []StructField{{
-		Name:      "SettableStruct",
-		Anonymous: true,
-		Type:      PointerTo(TypeOf(SettableStruct{})),
-	}}
-	rt = StructOf(fields)
-	rv = New(rt).Elem()
-	// This should panic since the pointer is nil.
-	shouldPanic("", func() {
-		rv.Interface().(IfaceSet).Set(want)
-	})
+		fields = []StructField{{
+			Name:      "SettableStruct",
+			Anonymous: true,
+			Type:      PointerTo(TypeOf(SettableStruct{})),
+		}}
+		rt = StructOf(fields)
+		rv = New(rt).Elem()
+		// This should panic since the pointer is nil.
+		shouldPanic("", func() {
+			rv.Interface().(IfaceSet).Set(want)
+		})
+	}
 
 	// The behavior is different if there is a second field,
 	// since now an interface value holds a pointer to the struct
 	// rather than just holding a copy of the struct.
-	fields = []StructField{
+	fields := []StructField{
 		{
 			Name:      "SettableStruct",
 			Anonymous: true,
diff --git a/src/runtime/asm_amd64.s b/src/runtime/asm_amd64.s
index cdf9874a7f..e9294bd14a 100644
--- a/src/runtime/asm_amd64.s
+++ b/src/runtime/asm_amd64.s
@@ -160,11 +160,18 @@ TEXT runtime·rt0_go(SB),NOSPLIT|NOFRAME|TOPFRAME,$0
 	// copy arguments forward on an even stack
 	MOVQ	DI, AX		// argc
 	MOVQ	SI, BX		// argv
+#ifdef GOOS_sylixos
+	SUBQ	$(8), SP
+#endif
 	SUBQ	$(5*8), SP		// 3args 2auto
 	ANDQ	$~15, SP
 	MOVQ	AX, 24(SP)
 	MOVQ	BX, 32(SP)
 
+#ifdef GOOS_sylixos
+	MOVQ	DX, 40(SP)	// env
+#endif
+
 	// create istack out of the given (operating system) stack.
 	// _cgo_init may update stackguard.
 	MOVQ	$runtime·g0(SB), DI
@@ -250,7 +257,10 @@ needtls:
 	// skip TLS setup on OpenBSD
 	JMP ok
 #endif
-
+#ifdef GOOS_sylixos
+	// skip TLS setup on SylixOS
+	JMP ok
+#endif
 #ifdef GOOS_windows
 	CALL	runtime·wintls(SB)
 #endif
@@ -340,6 +350,12 @@ ok:
 
 	CALL	runtime·check(SB)
 
+#ifdef GOOS_sylixos
+	MOVQ	40(SP), AX		// copy env
+	MOVQ	AX, 0(SP)
+	CALL	runtime·sylixosenvs(SB)
+#endif
+
 	MOVL	24(SP), AX		// copy argc
 	MOVL	AX, 0(SP)
 	MOVQ	32(SP), AX		// copy argv
diff --git a/src/runtime/asm_arm64.s b/src/runtime/asm_arm64.s
index 64a1880589..53e0ccc0fe 100644
--- a/src/runtime/asm_arm64.s
+++ b/src/runtime/asm_arm64.s
@@ -14,6 +14,9 @@ TEXT runtime·rt0_go(SB),NOSPLIT|TOPFRAME,$0
 	SUB	$32, RSP
 	MOVW	R0, 8(RSP) // argc
 	MOVD	R1, 16(RSP) // argv
+#ifdef GOOS_sylixos
+	MOVD	R2, 24(RSP) // env
+#endif
 
 #ifdef TLS_darwin
 	// Initialize TLS.
@@ -77,6 +80,15 @@ nocgo:
 	BL	runtime·wintls(SB)
 #endif
 
+#ifdef GOOS_sylixos
+	MOVD	24(RSP), R0		// copy env
+	SUB	$16, RSP
+	MOVD	R0, 8(RSP) // env
+	MOVD	$0, 0(RSP) // dummy LR
+	BL	runtime·sylixosenvs(SB)
+	ADD	$16, RSP
+#endif
+
 	MOVW	8(RSP), R0	// copy argc
 	MOVW	R0, -8(RSP)
 	MOVD	16(RSP), R0		// copy argv
diff --git a/src/runtime/crash_test.go b/src/runtime/crash_test.go
index 69e1034ff8..c6a3dac78e 100644
--- a/src/runtime/crash_test.go
+++ b/src/runtime/crash_test.go
@@ -865,6 +865,10 @@ func TestG0StackOverflow(t *testing.T) {
 		testenv.SkipFlaky(t, 62671)
 	}
 
+	if runtime.GOOS == "sylixos" {
+		testenv.SkipFlaky(t, 26061)
+	}
+
 	if os.Getenv("TEST_G0_STACK_OVERFLOW") != "1" {
 		cmd := testenv.CleanCmdEnv(testenv.Command(t, os.Args[0], "-test.run=^TestG0StackOverflow$", "-test.v"))
 		cmd.Env = append(cmd.Env, "TEST_G0_STACK_OVERFLOW=1")
diff --git a/src/runtime/crash_unix_test.go b/src/runtime/crash_unix_test.go
index 123a462423..5e3063619c 100644
--- a/src/runtime/crash_unix_test.go
+++ b/src/runtime/crash_unix_test.go
@@ -207,6 +207,16 @@ func TestPanicSystemstack(t *testing.T) {
 		t.Fatal("reading traceback from pipe: ", err)
 	}
 
+	// ACOINFO TODO
+	if runtime.GOOS == "sylixos" {
+		if bytes.Count(tb, []byte("blockOnSystemStackInternal")) != 1 ||
+			bytes.Count(tb, []byte("BlockOnSystemStack")) != 1 ||
+			bytes.Count(tb, []byte("testPanicSystemstackInternal")) != 1 {
+			t.Fatal("traceback missing system stack:\n", string(tb))
+		}
+		return
+	}
+
 	// Traceback should have two testPanicSystemstackInternal's
 	// and two blockOnSystemStackInternal's.
 	userFunc := "testPanicSystemstackInternal"
diff --git a/src/runtime/debug/panic_test.go b/src/runtime/debug/panic_test.go
index ec5294ce4c..39e418a091 100644
--- a/src/runtime/debug/panic_test.go
+++ b/src/runtime/debug/panic_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || sylixos
 
 // TODO: test on Windows?
 
diff --git a/src/runtime/debug/stack_test.go b/src/runtime/debug/stack_test.go
index e1559303f0..502866fed5 100644
--- a/src/runtime/debug/stack_test.go
+++ b/src/runtime/debug/stack_test.go
@@ -69,6 +69,9 @@ Don't worry much about the base levels, but check the ones in our own package.
 		/Users/r/go/src/testing/testing.go:561 +0x86d
 */
 func TestStack(t *testing.T) {
+	if runtime.GOOS == "sylixos" {
+		t.Skip("wrong path separator on SylixOS")
+	}
 	b := T(0).method()
 	lines := strings.Split(string(b), "\n")
 	if len(lines) < 6 {
diff --git a/src/runtime/export_pipe2_test.go b/src/runtime/export_pipe2_test.go
index 8d49009b43..732acc367b 100644
--- a/src/runtime/export_pipe2_test.go
+++ b/src/runtime/export_pipe2_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package runtime
 
diff --git a/src/runtime/lock_sema.go b/src/runtime/lock_sema.go
index 32d2235ad3..0f04c7e650 100644
--- a/src/runtime/lock_sema.go
+++ b/src/runtime/lock_sema.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || netbsd || openbsd || plan9 || solaris || windows
+//go:build aix || darwin || netbsd || openbsd || plan9 || solaris || windows || sylixos
 
 package runtime
 
diff --git a/src/runtime/mmap.go b/src/runtime/mmap.go
index 9a7b298562..e02ba9bc91 100644
--- a/src/runtime/mmap.go
+++ b/src/runtime/mmap.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !aix && !darwin && !js && !((linux && (amd64 || arm64 || loong64)) || (freebsd && amd64)) && !openbsd && !plan9 && !solaris && !windows
+//go:build !aix && !darwin && !js && !((linux && (amd64 || arm64 || loong64)) || (freebsd && amd64)) && !openbsd && !plan9 && !solaris && !windows && !sylixos
 
 package runtime
 
diff --git a/src/runtime/mpagealloc_64bit.go b/src/runtime/mpagealloc_64bit.go
index 36cd222360..d8c745a198 100644
--- a/src/runtime/mpagealloc_64bit.go
+++ b/src/runtime/mpagealloc_64bit.go
@@ -7,6 +7,7 @@
 package runtime
 
 import (
+	"internal/goos"
 	"unsafe"
 )
 
@@ -246,7 +247,7 @@ func (s *scavengeIndex) sysGrow(base, limit uintptr, sysStat *sysMemStat) uintpt
 //
 // Returns the amount of memory added to sysStat.
 func (s *scavengeIndex) sysInit(test bool, sysStat *sysMemStat) uintptr {
-	n := uintptr(1<<heapAddrBits) / pallocChunkBytes
+	n := uintptr(1<<heapAddrBits) / (pallocChunkBytes << (3 * goos.IsSylixos))
 	nbytes := n * unsafe.Sizeof(atomicScavChunkData{})
 	r := sysReserve(nil, nbytes)
 	sl := notInHeapSlice{(*notInHeap)(r), int(n), int(n)}
diff --git a/src/runtime/nbpipe_pipe2.go b/src/runtime/nbpipe_pipe2.go
index 22d60b4a63..2821e8cc2d 100644
--- a/src/runtime/nbpipe_pipe2.go
+++ b/src/runtime/nbpipe_pipe2.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package runtime
 
diff --git a/src/runtime/netpoll.go b/src/runtime/netpoll.go
index 7b37d91b24..69a48d32de 100644
--- a/src/runtime/netpoll.go
+++ b/src/runtime/netpoll.go
@@ -345,7 +345,7 @@ func poll_runtime_pollWait(pd *pollDesc, mode int) int {
 		return errcode
 	}
 	// As for now only Solaris, illumos, AIX and wasip1 use level-triggered IO.
-	if GOOS == "solaris" || GOOS == "illumos" || GOOS == "aix" || GOOS == "wasip1" {
+	if GOOS == "solaris" || GOOS == "illumos" || GOOS == "aix" || GOOS == "wasip1" || GOOS == "sylixos" {
 		netpollarm(pd, mode)
 	}
 	for !netpollblock(pd, int32(mode), false) {
diff --git a/src/runtime/pprof/pprof_rusage.go b/src/runtime/pprof/pprof_rusage.go
index aa429fb06a..57a8aa17e3 100644
--- a/src/runtime/pprof/pprof_rusage.go
+++ b/src/runtime/pprof/pprof_rusage.go
@@ -17,7 +17,7 @@ import (
 func addMaxRSS(w io.Writer) {
 	var rssToBytes uintptr
 	switch runtime.GOOS {
-	case "aix", "android", "dragonfly", "freebsd", "linux", "netbsd", "openbsd":
+	case "aix", "android", "dragonfly", "freebsd", "linux", "netbsd", "openbsd", "sylixos":
 		rssToBytes = 1024
 	case "darwin", "ios":
 		rssToBytes = 1
diff --git a/src/runtime/pprof/pprof_test.go b/src/runtime/pprof/pprof_test.go
index da4ad17d77..dbfd78f191 100644
--- a/src/runtime/pprof/pprof_test.go
+++ b/src/runtime/pprof/pprof_test.go
@@ -431,6 +431,9 @@ func cpuProfilingBroken() bool {
 			// See https://golang.org/issue/13841.
 			return true
 		}
+	case "sylixos":
+		// Profiling SIG will only send to main process main thread.
+		return true
 	}
 
 	return false
@@ -660,8 +663,8 @@ func TestCPUProfileWithFork(t *testing.T) {
 	}
 
 	heap := 1 << 30
-	if runtime.GOOS == "android" {
-		// Use smaller size for Android to avoid crash.
+	if runtime.GOOS == "android" || runtime.GOOS == "sylixos" {
+		// Use smaller size for Android or SylixOS to avoid crash.
 		heap = 100 << 20
 	}
 	if runtime.GOOS == "windows" && runtime.GOARCH == "arm" {
diff --git a/src/runtime/pprof/proto_other.go b/src/runtime/pprof/proto_other.go
index 7322e84803..74bd8529da 100644
--- a/src/runtime/pprof/proto_other.go
+++ b/src/runtime/pprof/proto_other.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !windows && !darwin
+//go:build !windows && !darwin && !sylixos
 
 package pprof
 
diff --git a/src/runtime/pprof/proto_test.go b/src/runtime/pprof/proto_test.go
index caaaa45f12..e7f582bff7 100644
--- a/src/runtime/pprof/proto_test.go
+++ b/src/runtime/pprof/proto_test.go
@@ -117,6 +117,36 @@ func testPCs(t *testing.T) (addr1, addr2 uint64, map1, map2 *profile.Mapping) {
 			t.Fatal(err)
 		}
 
+		map1 = &profile.Mapping{
+			ID:           1,
+			Start:        start,
+			Limit:        end,
+			File:         exe,
+			BuildID:      buildID,
+			HasFunctions: true,
+		}
+		map2 = &profile.Mapping{
+			ID:           1,
+			Start:        start,
+			Limit:        end,
+			File:         exe,
+			BuildID:      buildID,
+			HasFunctions: true,
+		}
+	case "sylixos":
+		addr1 = uint64(abi.FuncPCABIInternal(f1))
+		addr2 = uint64(abi.FuncPCABIInternal(f2))
+
+		exe, err := os.Executable()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		start, end, exe, buildID, err := readMainModuleMapping()
+		if err != nil {
+			t.Fatal(err)
+		}
+
 		map1 = &profile.Mapping{
 			ID:           1,
 			Start:        start,
diff --git a/src/runtime/proc.go b/src/runtime/proc.go
index 76c8b71ab9..648fee2884 100644
--- a/src/runtime/proc.go
+++ b/src/runtime/proc.go
@@ -2090,6 +2090,9 @@ func forEachPInternal(fn func(*p)) {
 			// Requires system stack.
 			if notetsleep(&sched.safePointNote, 100*1000) {
 				noteclear(&sched.safePointNote)
+				if GOOS == "sylixos" && sched.safePointWait != 0 {
+					continue
+				}
 				break
 			}
 			preemptall()
diff --git a/src/runtime/runtime_unix_test.go b/src/runtime/runtime_unix_test.go
index 642a946280..aa62ef6f23 100644
--- a/src/runtime/runtime_unix_test.go
+++ b/src/runtime/runtime_unix_test.go
@@ -6,7 +6,7 @@
 // We need a fast system call to provoke the race,
 // and Close(-1) is nearly universally fast.
 
-//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || plan9
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || plan9 || sylixos
 
 package runtime_test
 
diff --git a/src/runtime/security_issetugid.go b/src/runtime/security_issetugid.go
index 5048632c3a..9ea6c9903c 100644
--- a/src/runtime/security_issetugid.go
+++ b/src/runtime/security_issetugid.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || illumos || netbsd || openbsd || solaris
+//go:build darwin || dragonfly || freebsd || illumos || netbsd || openbsd || solaris || sylixos
 
 package runtime
 
diff --git a/src/runtime/signal_amd64.go b/src/runtime/signal_amd64.go
index 8ade208836..648f681546 100644
--- a/src/runtime/signal_amd64.go
+++ b/src/runtime/signal_amd64.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build amd64 && (darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris)
+//go:build amd64 && (darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos)
 
 package runtime
 
diff --git a/src/runtime/signal_arm64.go b/src/runtime/signal_arm64.go
index 4a96b3c2e7..36a4243813 100644
--- a/src/runtime/signal_arm64.go
+++ b/src/runtime/signal_arm64.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || freebsd || linux || netbsd || openbsd
+//go:build darwin || freebsd || linux || netbsd || openbsd || sylixos
 
 package runtime
 
diff --git a/src/runtime/signal_loong64.go b/src/runtime/signal_loong64.go
index ac842c0c94..46eede0fd2 100644
--- a/src/runtime/signal_loong64.go
+++ b/src/runtime/signal_loong64.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build linux && loong64
+//go:build (linux || sylixos) && loong64
 
 package runtime
 
diff --git a/src/runtime/signal_unix.go b/src/runtime/signal_unix.go
index 6f40f440e8..625456a6c3 100644
--- a/src/runtime/signal_unix.go
+++ b/src/runtime/signal_unix.go
@@ -370,6 +370,9 @@ func preemptM(mp *m) {
 	// Issue #41702.
 	if GOOS == "darwin" || GOOS == "ios" {
 		execLock.rlock()
+	} else if GOOS == "sylixos" {
+		// On SylixOS, no support preempt
+		return
 	}
 
 	if mp.signalPending.CompareAndSwap(0, 1) {
diff --git a/src/runtime/stack.go b/src/runtime/stack.go
index cdf859a7ff..1ae260e292 100644
--- a/src/runtime/stack.go
+++ b/src/runtime/stack.go
@@ -69,7 +69,7 @@ const (
 	// to each stack below the usual guard area for OS-specific
 	// purposes like signal handling. Used on Windows, Plan 9,
 	// and iOS because they do not use a separate stack.
-	stackSystem = goos.IsWindows*4096 + goos.IsPlan9*512 + goos.IsIos*goarch.IsArm64*1024
+	stackSystem = goos.IsWindows*4096 + goos.IsPlan9*512 + goos.IsIos*goarch.IsArm64*1024 + goos.IsSylixos*512*goarch.PtrSize
 
 	// The minimum size of stack used by Go code
 	stackMin = 2048
@@ -354,6 +354,12 @@ func stackalloc(n uint32) stack {
 		if v == nil {
 			throw("out of memory (stackalloc)")
 		}
+
+		if goos.IsSylixos == 1 {
+			for p := uintptr(v); p < (uintptr(v) + uintptr(n)); p += (2048 - (p % 2048)) {
+				*(*byte)(unsafe.Pointer(p)) = 0xfd
+			}
+		}
 		return stack{uintptr(v), uintptr(v) + uintptr(n)}
 	}
 
@@ -434,6 +440,11 @@ func stackalloc(n uint32) stack {
 	if stackDebug >= 1 {
 		print("  allocated ", v, "\n")
 	}
+	if goos.IsSylixos == 1 {
+		for p := uintptr(v); p < (uintptr(v) + uintptr(n)); p += (2048 - (p % 2048)) {
+			*(*byte)(unsafe.Pointer(p)) = 0xfd
+		}
+	}
 	return stack{uintptr(v), uintptr(v) + uintptr(n)}
 }
 
diff --git a/src/runtime/stubs2.go b/src/runtime/stubs2.go
index 39bde15b1c..634806f019 100644
--- a/src/runtime/stubs2.go
+++ b/src/runtime/stubs2.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !aix && !darwin && !js && !openbsd && !plan9 && !solaris && !wasip1 && !windows
+//go:build !aix && !darwin && !js && !openbsd && !plan9 && !solaris && !wasip1 && !windows && !sylixos
 
 package runtime
 
diff --git a/src/runtime/stubs3.go b/src/runtime/stubs3.go
index c3749f34ed..66b864bf16 100644
--- a/src/runtime/stubs3.go
+++ b/src/runtime/stubs3.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !aix && !darwin && !freebsd && !openbsd && !plan9 && !solaris && !wasip1
+//go:build !aix && !darwin && !freebsd && !openbsd && !plan9 && !solaris && !wasip1 && !sylixos
 
 package runtime
 
diff --git a/src/runtime/stubs_riscv64.go b/src/runtime/stubs_riscv64.go
index 61a6e33bd4..b07d7f8746 100644
--- a/src/runtime/stubs_riscv64.go
+++ b/src/runtime/stubs_riscv64.go
@@ -4,15 +4,10 @@
 
 package runtime
 
-import "unsafe"
-
 // Called from assembly only; declared for go vet.
 func load_g()
 func save_g()
 
-//go:noescape
-func asmcgocall_no_g(fn, arg unsafe.Pointer)
-
 // Used by reflectcall and the reflect package.
 //
 // Spills/loads arguments in registers to/from an internal/abi.RegArgs
diff --git a/src/runtime/sys_libc.go b/src/runtime/sys_libc.go
index 0c6f13ca9f..66cfb31362 100644
--- a/src/runtime/sys_libc.go
+++ b/src/runtime/sys_libc.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || (openbsd && !mips64)
+//go:build darwin || (openbsd && !mips64) || sylixos
 
 package runtime
 
diff --git a/src/runtime/testdata/testwinlibthrow/main.go b/src/runtime/testdata/testwinlibthrow/main.go
index 493e1e11d5..ce0c92f252 100644
--- a/src/runtime/testdata/testwinlibthrow/main.go
+++ b/src/runtime/testdata/testwinlibthrow/main.go
@@ -1,19 +1,19 @@
-package main
-
-import (
-	"os"
-	"syscall"
-)
-
-func main() {
-	dll := syscall.MustLoadDLL("veh.dll")
-	RaiseNoExcept := dll.MustFindProc("RaiseNoExcept")
-	ThreadRaiseNoExcept := dll.MustFindProc("ThreadRaiseNoExcept")
-
-	thread := len(os.Args) > 1 && os.Args[1] == "thread"
-	if !thread {
-		RaiseNoExcept.Call()
-	} else {
-		ThreadRaiseNoExcept.Call()
-	}
-}
+package main
+
+import (
+	"os"
+	"syscall"
+)
+
+func main() {
+	dll := syscall.MustLoadDLL("veh.dll")
+	RaiseNoExcept := dll.MustFindProc("RaiseNoExcept")
+	ThreadRaiseNoExcept := dll.MustFindProc("ThreadRaiseNoExcept")
+
+	thread := len(os.Args) > 1 && os.Args[1] == "thread"
+	if !thread {
+		RaiseNoExcept.Call()
+	} else {
+		ThreadRaiseNoExcept.Call()
+	}
+}
diff --git a/src/runtime/timestub2.go b/src/runtime/timestub2.go
index 49bfeb60c8..0162bcdeae 100644
--- a/src/runtime/timestub2.go
+++ b/src/runtime/timestub2.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !aix && !darwin && !freebsd && !openbsd && !solaris && !wasip1 && !windows && !(linux && amd64)
+//go:build !aix && !darwin && !freebsd && !openbsd && !solaris && !wasip1 && !windows && !(linux && amd64) && !sylixos
 
 package runtime
 
diff --git a/src/runtime/tls_arm64.h b/src/runtime/tls_arm64.h
index 3aa8c63d39..115a3b027c 100644
--- a/src/runtime/tls_arm64.h
+++ b/src/runtime/tls_arm64.h
@@ -44,6 +44,10 @@
 #define MRS_TPIDR_R0 MOVD R18_PLATFORM, R0
 #endif
 
+#ifdef GOOS_sylixos
+#define MRS_TPIDR_R0 WORD $0xd53bd040 // MRS TPIDR_EL0, R0
+#endif
+
 // Define something that will break the build if
 // the GOOS is unknown.
 #ifndef MRS_TPIDR_R0
diff --git a/src/runtime/tls_arm64.s b/src/runtime/tls_arm64.s
index 52b3e8f222..e70ad59ccd 100644
--- a/src/runtime/tls_arm64.s
+++ b/src/runtime/tls_arm64.s
@@ -12,10 +12,12 @@ TEXT runtime·load_g(SB),NOSPLIT,$0
 #ifndef GOOS_darwin
 #ifndef GOOS_openbsd
 #ifndef GOOS_windows
+#ifndef GOOS_sylixos
 	MOVB	runtime·iscgo(SB), R0
 	CBZ	R0, nocgo
 #endif
 #endif
+#endif
 #endif
 
 	MRS_TPIDR_R0
@@ -33,10 +35,12 @@ TEXT runtime·save_g(SB),NOSPLIT,$0
 #ifndef GOOS_darwin
 #ifndef GOOS_openbsd
 #ifndef GOOS_windows
+#ifndef GOOS_sylixos
 	MOVB	runtime·iscgo(SB), R0
 	CBZ	R0, nocgo
 #endif
 #endif
+#endif
 #endif
 
 	MRS_TPIDR_R0
diff --git a/src/sync/atomic/atomic_test.go b/src/sync/atomic/atomic_test.go
index 0617b27aca..943d2a37d2 100644
--- a/src/sync/atomic/atomic_test.go
+++ b/src/sync/atomic/atomic_test.go
@@ -2922,6 +2922,9 @@ func TestAutoAligned64(t *testing.T) {
 }
 
 func TestNilDeref(t *testing.T) {
+	if runtime.GOOS == "sylixos" {
+		t.Skip("because sylixos only have on page table, so this test may crash kernel!, skip it.")
+	}
 	funcs := [...]func(){
 		func() { CompareAndSwapInt32(nil, 0, 0) },
 		func() { (*Int32)(nil).CompareAndSwap(0, 0) },
diff --git a/src/syscall/dirent.go b/src/syscall/dirent.go
index e4f36a191b..a9b8791ddb 100644
--- a/src/syscall/dirent.go
+++ b/src/syscall/dirent.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build unix || (js && wasm) || wasip1
+//go:build (unix || (js && wasm) || wasip1) && !sylixos
 
 package syscall
 
diff --git a/src/syscall/dirent_test.go b/src/syscall/dirent_test.go
index cfa5478feb..6a7c50790b 100644
--- a/src/syscall/dirent_test.go
+++ b/src/syscall/dirent_test.go
@@ -93,7 +93,7 @@ func TestDirentRepeat(t *testing.T) {
 	// Note: the size of the buffer is small enough that the loop
 	// below will need to execute multiple times. See issue #31368.
 	size := N * unsafe.Offsetof(syscall.Dirent{}.Name) / 4
-	if runtime.GOOS == "freebsd" || runtime.GOOS == "netbsd" {
+	if runtime.GOOS == "freebsd" || runtime.GOOS == "netbsd" || runtime.GOOS == "sylixos" {
 		if size < 1024 {
 			size = 1024 // DIRBLKSIZ, see issue 31403.
 		}
diff --git a/src/syscall/exec_unix.go b/src/syscall/exec_unix.go
index 4747fa0758..a0b921b5ba 100644
--- a/src/syscall/exec_unix.go
+++ b/src/syscall/exec_unix.go
@@ -198,50 +198,56 @@ func forkExec(argv0 string, argv []string, attr *ProcAttr) (pid int, err error)
 
 	acquireForkLock()
 
-	// Allocate child status pipe close on exec.
-	if err = forkExecPipe(p[:]); err != nil {
-		releaseForkLock()
-		return 0, err
+	if runtime.GOOS != "sylixos" {
+		// Allocate child status pipe close on exec.
+		if err = forkExecPipe(p[:]); err != nil {
+			releaseForkLock()
+			return 0, err
+		}
 	}
 
 	// Kick off child.
 	pid, err1 = forkAndExecInChild(argv0p, argvp, envvp, chroot, dir, attr, sys, p[1])
 	if err1 != 0 {
-		Close(p[0])
-		Close(p[1])
+		if runtime.GOOS != "sylixos" {
+			Close(p[0])
+			Close(p[1])
+		}
 		releaseForkLock()
 		return 0, Errno(err1)
 	}
 	releaseForkLock()
 
-	// Read child error status from pipe.
-	Close(p[1])
-	for {
-		n, err = readlen(p[0], (*byte)(unsafe.Pointer(&err1)), int(unsafe.Sizeof(err1)))
-		if err != EINTR {
-			break
-		}
-	}
-	Close(p[0])
-	if err != nil || n != 0 {
-		if n == int(unsafe.Sizeof(err1)) {
-			err = Errno(err1)
-		}
-		if err == nil {
-			err = EPIPE
+	if runtime.GOOS != "sylixos" {
+		// Read child error status from pipe.
+		Close(p[1])
+		for {
+			n, err = readlen(p[0], (*byte)(unsafe.Pointer(&err1)), int(unsafe.Sizeof(err1)))
+			if err != EINTR {
+				break
+			}
 		}
+		Close(p[0])
+		if err != nil || n != 0 {
+			if n == int(unsafe.Sizeof(err1)) {
+				err = Errno(err1)
+			}
+			if err == nil {
+				err = EPIPE
+			}
+
+			// Child failed; wait for it to exit, to make sure
+			// the zombies don't accumulate.
+			_, err1 := Wait4(pid, &wstatus, 0, nil)
+			for err1 == EINTR {
+				_, err1 = Wait4(pid, &wstatus, 0, nil)
+			}
+
+			// OS-specific cleanup on failure.
+			forkAndExecFailureCleanup(attr, sys)
 
-		// Child failed; wait for it to exit, to make sure
-		// the zombies don't accumulate.
-		_, err1 := Wait4(pid, &wstatus, 0, nil)
-		for err1 == EINTR {
-			_, err1 = Wait4(pid, &wstatus, 0, nil)
+			return 0, err
 		}
-
-		// OS-specific cleanup on failure.
-		forkAndExecFailureCleanup(attr, sys)
-
-		return 0, err
 	}
 
 	// Read got EOF, so pipe closed on exec, so exec succeeded.
diff --git a/src/syscall/exec_unix_test.go b/src/syscall/exec_unix_test.go
index bb0b3e7037..a507f454bf 100644
--- a/src/syscall/exec_unix_test.go
+++ b/src/syscall/exec_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build unix
+//go:build unix && !sylixos
 
 package syscall_test
 
diff --git a/src/syscall/forkpipe2.go b/src/syscall/forkpipe2.go
index bbecfdabf8..29154de102 100644
--- a/src/syscall/forkpipe2.go
+++ b/src/syscall/forkpipe2.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package syscall
 
diff --git a/src/syscall/sockcmsg_unix_other.go b/src/syscall/sockcmsg_unix_other.go
index 845bd9df99..5442cc140d 100644
--- a/src/syscall/sockcmsg_unix_other.go
+++ b/src/syscall/sockcmsg_unix_other.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || freebsd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || freebsd || linux || netbsd || openbsd || solaris || sylixos
 
 package syscall
 
diff --git a/src/syscall/syscall_unix_test.go b/src/syscall/syscall_unix_test.go
index 56e771e086..222badf5a5 100644
--- a/src/syscall/syscall_unix_test.go
+++ b/src/syscall/syscall_unix_test.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build unix
+//go:build unix && !sylixos
 
 package syscall_test
 
diff --git a/src/testing/fstest/testfs.go b/src/testing/fstest/testfs.go
index 080bcdd65f..8fe1079acf 100644
--- a/src/testing/fstest/testfs.go
+++ b/src/testing/fstest/testfs.go
@@ -15,6 +15,7 @@ import (
 	"slices"
 	"strings"
 	"testing/iotest"
+	"runtime"
 )
 
 // TestFS tests a file system implementation.
@@ -600,17 +601,27 @@ func (t *fsTester) checkBadPath(file string, desc string, open func(string) erro
 		bad = append(bad,
 			file[:i]+"//"+file[i+1:],
 			file[:i]+"/./"+file[i+1:],
-			file[:i]+`\`+file[i+1:],
 			file[:i]+"/../"+file,
 		)
+
+		if runtime.GOOS != "sylixos" { //ACOINFO TODO: SylixOS accept backslash as separator.
+			bad = append(bad,
+				file[:i]+`\`+file[i+1:],
+			)
+		}
 	}
 	if i := strings.LastIndex(file, "/"); i >= 0 {
 		bad = append(bad,
 			file[:i]+"//"+file[i+1:],
 			file[:i]+"/./"+file[i+1:],
-			file[:i]+`\`+file[i+1:],
 			file+"/../"+file[i+1:],
 		)
+
+		if runtime.GOOS != "sylixos" { //ACOINFO TODO: SylixOS accept backslash as separator.
+			bad = append(bad,
+				file[:i]+`\`+file[i+1:],
+			)
+		}
 	}
 
 	for _, b := range bad {
diff --git a/src/testing/run_example.go b/src/testing/run_example.go
index b2c5c3d14c..006a9c0281 100644
--- a/src/testing/run_example.go
+++ b/src/testing/run_example.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !js && !wasip1
+//go:build !js && !wasip1 && !sylixos
 
 // TODO(@musiol, @odeke-em): re-unify this entire file back into
 // example.go when js/wasm gets an os.Pipe implementation
diff --git a/src/text/template/exec_test.go b/src/text/template/exec_test.go
index 9903e17d0e..842fdbbf50 100644
--- a/src/text/template/exec_test.go
+++ b/src/text/template/exec_test.go
@@ -11,6 +11,7 @@ import (
 	"fmt"
 	"io"
 	"reflect"
+	"runtime"
 	"strings"
 	"sync"
 	"testing"
@@ -1657,6 +1658,9 @@ func TestInterfaceValues(t *testing.T) {
 
 // Check that panics during calls are recovered and returned as errors.
 func TestExecutePanicDuringCall(t *testing.T) {
+	if runtime.GOOS == "sylixos" {
+		t.Skip("because sylixos only have on page table, so this test will crash kernel!, skip it.")
+	}
 	funcs := map[string]any{
 		"doPanic": func() string {
 			panic("custom panic string")
