diff -ru cinelerra-4.4.orig/quicktime/thirdparty/x264.052906/configure cinelerra-4.4/quicktime/thirdparty/x264.052906/configure
--- cinelerra-4.4.orig/quicktime/thirdparty/x264.052906/configure	2013-01-02 15:42:12.870084505 -0800
+++ cinelerra-4.4/quicktime/thirdparty/x264.052906/configure	2013-01-02 15:43:59.639182303 -0800
@@ -290,7 +290,7 @@
     esac
 fi
 
-MP4_LDFLAGS="-lgpac_static"
+MP4_LDFLAGS="-lgpac_static -lz"
 if [ $SYS = CYGWIN -o $SYS = MINGW ]; then
     MP4_LDFLAGS="$MP4_LDFLAGS -lwinmm"
 fi
