diff --unified --recursive --text a/snmpb-1.0/app/graph.cpp b/snmpb-1.0/app/graph.cpp
--- snmpb-1.0/app/graph.cpp	2021-01-31 18:27:22.000000000 +0100
+++ snmpb-1.0/app/graph.cpp	2023-04-12 00:01:11.429865965 +0200
@@ -27,6 +27,7 @@
 
 #include <qwt_plot_zoomer.h>
 #include <qwt_plot_layout.h>
+#include <qwt_scale_widget.h>
 
 #define ui s->MainUI()
 
diff --unified --recursive --text a/snmpb-1.0/app/graph.h b/snmpb-1.0/app/graph.h
--- snmpb-1.0/app/graph.h	2021-01-31 18:27:22.000000000 +0100
+++ snmpb-1.0/app/graph.h	2023-04-12 00:01:26.963227083 +0200
@@ -24,6 +24,7 @@
 #include <qtabwidget.h>
 #include <qwt_plot.h>
 #include <qwt_plot_curve.h>
+#include <qwt_scale_widget.h>
 #include <QTimerEvent>
 #include <qlistwidget.h>
 #include <qlist.h>
