--- src.pristine/n64/src/libgs/makefile.in	2019-06-15 21:50:50.656063865 -0400
+++ src.new/n64/src/libgs/makefile.in	2019-07-06 22:17:44.897606509 -0400
@@ -8,7 +8,7 @@
 LDFLAGS                       = -Wl,-rpath,\$$ORIGIN/../lib @LDFLAGS@
 LDLIBS                        = @LIBUSB@ @LIBM@
 SRC                           = $(srcdir)/src/lib/*.c $(srcdir)/src/exe/*.c $(srcdir)/../../include/vector/vector.c
-PROGNAME                      = gs
+PROGNAME                      = gs.n64
 INSTALL                       = @INSTALL@
 MKDIR_P                       = @MKDIR_P@
 