diff --git src/CMakeLists.txt src/CMakeLists.txt
index 177cb77..b62239b 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -20,8 +20,8 @@ INCLUDE_DIRECTORIES(
         ${SQLITE3_INCLUDE_DIRS}
         ${LIBCURL_INCLUDE_DIRS}
         ${LIBTAG_INCLUDE_DIRS}
-        ${LIBDBUS_INCLUDE_DIRS} 
-        ${LIBGIO_INCLUDE_DIRS} 
+        ${LIBDBUS_INCLUDE_DIRS}
+        ${LIBGIO_INCLUDE_DIRS}
 	${LIBGPOD_INCLUDE_DIRS}
 	${LIBINDICATE_INCLUDE_DIRS}
 	${LIBINDICATE06_INCLUDE_DIRS}
@@ -40,7 +40,7 @@ ENDIF( ADD_WXSQLITE_SUPPORT )
 
 SET( CURL_SOURCES
 	curl/base.cpp
-	curl/http.cpp 
+	curl/http.cpp
 )
 
 SET( DBUS_SOURCES
@@ -319,6 +319,7 @@ TARGET_LINK_LIBRARIES( guayadeque
 	${LIBINDICATE07_LIBRARIES}
 	${LIBAPPINDICATOR_LIBRARIES}
 	${LIBWXSQLITE3_LIBRARIES}
+	-lgdk_pixbuf-2.0
 	)
 
 INSTALL( TARGETS guayadeque
