diff --git po/guayadeque.pot po/guayadeque.pot
index 2c6b8f0..734d89f 100644
--- po/guayadeque.pot
+++ po/guayadeque.pot
@@ -8,7 +8,7 @@ msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2014-02-25 21:21+0000\n"
+"POT-Creation-Date: 2014-04-22 08:36+0100\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
diff --git src/Config.h src/Config.h
index ea5a27d..a518fb4 100644
--- src/Config.h
+++ src/Config.h
@@ -139,6 +139,7 @@ wxString inline escape_configlist_str( const wxString &val )
     wxString RetVal = val;
     RetVal.Replace( wxT( ":" ), wxT( "_$&" ) );
     RetVal.Replace( wxT( ";" ), wxT( "_&$" ) );
+    RetVal.Replace( wxT( "," ), wxT( "$_&" ) );
     return RetVal;
 }
 
@@ -148,6 +149,7 @@ wxString inline unescape_configlist_str( const wxString &val )
     wxString RetVal = val;
     RetVal.Replace( wxT( "_$&" ), wxT( ":" ) );
     RetVal.Replace( wxT( "_&$" ), wxT( ";" ) );
+    RetVal.Replace( wxT( "$_&" ), wxT( "," ) );
     return RetVal;
 }
 
diff --git src/PlayerPanel.cpp src/PlayerPanel.cpp
index ac5136b..326314f 100644
--- src/PlayerPanel.cpp
+++ src/PlayerPanel.cpp
@@ -1596,7 +1596,7 @@ void  guPlayerPanel::OnMediaPosition( guMediaEvent &event )
             !m_NextTrackId &&
             ( m_MediaSong.m_Type != guTRACK_TYPE_RADIOSTATION ) &&
             ( CurPos > 0 ) && ( m_LastLength > 0 ) &&
-            ( ( CurPos + 1000 + ( !m_ForceGapless ? m_FadeOutTime : 0 ) ) >= m_LastLength )
+            ( ( CurPos + 4000 + ( !m_ForceGapless ? m_FadeOutTime : 0 ) ) >= m_LastLength )
           )
         {
             //if( !m_ForceGapless && m_FadeOutTime && !m_MediaSong.m_Offset && ( GetState() == guMEDIASTATE_PLAYING ) )
