diff --git a/gulpfile.js b/gulpfile.js
index a3fdd9d7..2f442484 100644
--- a/gulpfile.js
+++ b/gulpfile.js
@@ -142,8 +142,8 @@ const nw = new nwBuilder({
   macIcns: './src/app/images/butter.icns',
   version: nwVersion,
   flavor: nwFlavor,
-  manifestUrl: 'https://popcorntime.app/version.json',
-  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
+  manifestUrl: 'https://popcorn-ru.tk/version.json',
+  downloadUrl: 'https://popcorn-ru.tk/nw/',
   platforms: parsePlatforms()
 }).on('log', console.log);
 
