diff --git a/configure.ac b/configure.ac
index 90573e2..891a153 100644
--- a/configure.ac
+++ b/configure.ac
@@ -109,6 +109,7 @@ AS_IF([test "x$enable_manpages" != xno], [
         AC_CHECK_FILES([
                 /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl
                 /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
+                /usr/share/xml/docbook/xsl-stylesheets-1.78.0/manpages/docbook.xsl
                 ],[have_docbook=yes])
         AS_IF([test "x$enable_manpages" = xyes -a "x$have_docbook" = x], [
                 AC_MSG_ERROR([*** Manpages requested but docbook not found])
