diff --git a/packages/hadron-build/commands/release.js b/packages/hadron-build/commands/release.js
index 1b4611457..9295a2a3a 100644
--- a/packages/hadron-build/commands/release.js
+++ b/packages/hadron-build/commands/release.js
@@ -543,9 +543,6 @@ exports.run = (argv, done) => {
     task('write license file', writeLicenseFile),
     task('remove development files', removeDevelopmentFiles),
     !noAsar && task('create application asar', createApplicationAsar),
-    !skipInstaller && task('create branded installer', createBrandedInstaller),
-    task('create application zip', createApplicationZip),
-    task('store build configuration as json', writeConfigToJson)
   ].filter(Boolean));
 
   return async.series(tasks, (_err) => {
@@ -568,9 +565,5 @@ exports.run = (argv, done) => {
 exports.handler = (argv) => {
   exports.run(argv, (_err, CONFIG) => {
     cli.abortIfError(_err);
-    cli.ok(`${CONFIG.assets.length} assets successfully built`);
-    CONFIG.assets.map(function(asset) {
-      cli.info(asset.path);
-    });
   });
 };
diff --git a/packages/hadron-build/lib/target.js b/packages/hadron-build/lib/target.js
index 39ed422ab..959ce16c9 100644
--- a/packages/hadron-build/lib/target.js
+++ b/packages/hadron-build/lib/target.js
@@ -10,7 +10,6 @@ const path = require('path');
 const { promisify } = require('util');
 const normalizePkg = require('normalize-package-data');
 const parseGitHubRepoURL = require('parse-github-repo-url');
-const ffmpegAfterExtract = require('electron-packager-plugin-non-proprietary-codecs-ffmpeg').default;
 const windowsInstallerVersion = require('./windows-installer-version');
 const debug = require('debug')('hadron-build:target');
 const execFile = promisify(childProcess.execFile);
@@ -227,7 +226,6 @@ class Target {
       arch: this.arch,
       electronVersion: this.electronVersion,
       sign: null,
-      afterExtract: [ffmpegAfterExtract]
     };
 
     if (this.platform === 'win32') {
