diff --git a/packages/hadron-build/commands/release.js b/packages/hadron-build/commands/release.js
index 3fee4a5b8..0bb22f348 100644
--- a/packages/hadron-build/commands/release.js
+++ b/packages/hadron-build/commands/release.js
@@ -272,6 +272,10 @@ const transformPackageJson = async(CONFIG, done) => {
     'optionalDependencies'
   ]) {
     for (const depName of Object.keys(contents[depType] || {})) {
+      if (depName === 'os-dns-native') {
+        contents[depType][depName] = '1.2.1';
+        continue;
+      }
       const depEdge = packageNode.edgesOut.get(depName);
       if (!depEdge.to && !depEdge.optional) {
         throw new Error(
