Author: Andreas Tille <tille@debian.org>
Last-Update: Mon, 11 May 2020 16:16:46 +0200
Description: Propagate hardening flags

--- a/c/Makefile
+++ b/c/Makefile
@@ -6,7 +6,7 @@ CC = gcc -O3 -DALLELECOUNTER_VERSION='"$
 #compiler flags
 # -g adds debug info to the executable file
 # -Wall turns on most warnings from compiler
-CFLAGS = -g -Wall
+CFLAGS += -g -Wall
 
 #Location of samtools/htslib libraries
 HTSLOC?=$(HTSLIB)
@@ -67,7 +67,7 @@ all: clean make_bin make_htslib_tmp $(CO
 	@echo  Binaries have been compiled.
 
 $(COUNTER_TARGET): $(OBJS)
-	$(CC) $(JOIN_INCLUDES) $(INCLUDES) $(CFLAGS) -o $(COUNTER_TARGET) $(OBJS) $(LFLAGS) $(CAT_LFLAGS) $(LIBS) ./src/alleleCounter.c
+	$(CC) $(JOIN_INCLUDES) $(INCLUDES) $(CFLAGS) -o $(COUNTER_TARGET) $(OBJS) $(LFLAGS) $(CAT_LFLAGS) $(LIBS) ./src/alleleCounter.c $(LDFLAGS)
 
 #Unit Tests
 test: $(COUNTER_TARGET)