--- ezthumb-3.6.7/external/iup/tecmake.mak	2017-04-19 01:06:14.000000000 -0300
+++ ezthumb-3.6.7/external/iup/tecmake.mak	2019-10-12 15:33:21.877326940 -0300
@@ -1354,6 +1354,7 @@
     
     STDINCS += $(GTK)/include/atk-1.0 $(GTK)/include/gtk-$(GTKSFX).0 $(GTK)/include/gdk-pixbuf-2.0 
     STDINCS += $(GTK)/include/cairo $(GTK)/include/pango-1.0 $(GTK)/include/glib-2.0
+    STDINCS += /usr/include/harfbuzz
 
     ifeq ($(TEC_SYSARCH), x64)
       STDINCS += $(GTK)/lib64/glib-2.0/include 
