diff --unified --recursive --text hdrview/CMakeLists.txt hdrview/CMakeLists.txt
--- hdrview/CMakeLists.txt	2019-03-29 12:02:37.897333401 -0400
+++ hdrview/CMakeLists.txt	2019-03-29 12:05:52.409958962 -0400
@@ -266,14 +266,11 @@
 else()
     install(TARGETS HDRView RUNTIME DESTINATION "bin")
     install(FILES resources/hdrview.desktop DESTINATION "/usr/share/applications")
-    install(FILES resources/icon-512.png DESTINATION "/usr/share/icons/hicolor/1024x1024/apps" RENAME hdrview.png)
+    install(FILES resources/icon-1024.png DESTINATION "/usr/share/icons/hicolor/1024x1024/apps" RENAME hdrview.png)
     install(FILES resources/icon-512.png DESTINATION "/usr/share/icons/hicolor/512x512/apps" RENAME hdrview.png)
     install(FILES resources/icon-256.png DESTINATION "/usr/share/icons/hicolor/256x256/apps" RENAME hdrview.png)
     install(FILES resources/icon-128.png DESTINATION "/usr/share/icons/hicolor/128x128/apps" RENAME hdrview.png)
-    install(FILES resources/icon-96.png DESTINATION "/usr/share/icons/hicolor/96x96/apps" RENAME hdrview.png)
     install(FILES resources/icon-64.png DESTINATION "/usr/share/icons/hicolor/64x64/apps" RENAME hdrview.png)
-    install(FILES resources/icon-48.png DESTINATION "/usr/share/icons/hicolor/48x48/apps" RENAME hdrview.png)
     install(FILES resources/icon-32.png DESTINATION "/usr/share/icons/hicolor/32x32/apps" RENAME hdrview.png)
-    install(FILES resources/icon-24.png DESTINATION "/usr/share/icons/hicolor/24x24/apps" RENAME hdrview.png)
     install(FILES resources/icon-16.png DESTINATION "/usr/share/icons/hicolor/16x16/apps" RENAME hdrview.png)
 endif()
