diff --git a/TTS/tts/utils/text/phonemizers/espeak_wrapper.py b/TTS/tts/utils/text/phonemizers/espeak_wrapper.py
index 328e52f3..57a87b82 100644
--- a/TTS/tts/utils/text/phonemizers/espeak_wrapper.py
+++ b/TTS/tts/utils/text/phonemizers/espeak_wrapper.py
@@ -58,7 +58,7 @@ def _espeak_exe(espeak_lib: str, args: List, sync=False) -> List[str]:
     with subprocess.Popen(
         cmd,
         stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT,
+        # stderr=subprocess.STDOUT,
     ) as p:
         res = iter(p.stdout.readline, b"")
         if not sync:
