--- a/steamos-manager/src/lib.rs
+++ b/steamos-manager/src/lib.rs
@@ -27,6 +27,7 @@
 
 mod ds_inhibit;
 mod error;
+mod hhd;
 mod inputplumber;
 mod job;
 mod manager;
--- a/steamos-manager/src/power.rs
+++ b/steamos-manager/src/power.rs
@@ -169,6 +169,9 @@
 }
 
 pub(crate) async fn tdp_limit_manager(system: &Connection) -> Result<Box<dyn TdpLimitManager>> {
+    if let Some(manager) = crate::hhd::tdp_limit_manager().await? {
+        return Ok(manager);
+    }
     let config = device_config().await?;
     if let Some(config) = config.as_ref().and_then(|config| config.tdp_limit.as_ref()) {
         Ok(match &config.method {
--- a/steamos-manager/src/gpu.rs
+++ b/steamos-manager/src/gpu.rs
@@ -151,6 +151,9 @@
 }
 
 pub(crate) async fn gpu_performance_level_driver() -> Result<Box<dyn GpuPerformanceLevelDriver>> {
+    if let Some(driver) = crate::hhd::gpu_performance_level_driver().await? {
+        return Ok(driver);
+    }
     let config = device_config().await?;
     let config = config
         .as_ref()
--- a/steamos-manager/src/manager/user.rs
+++ b/steamos-manager/src/manager/user.rs
@@ -1738,10 +1738,6 @@
     object_server: &ObjectServer,
     tdp_manager: Option<UnboundedSender<TdpManagerCommand>>,
 ) -> Result<()> {
-    let Some(config) = device_config().await? else {
-        return Ok(());
-    };
-
     let performance_profile = PerformanceProfile1 {
         proxy: proxy.clone(),
         tdp_limit_manager: tdp_manager.clone(),
@@ -1751,14 +1747,8 @@
         let low_power_mode = LowPowerMode1 {
             manager: manager.clone(),
         };
-        if config
-            .tdp_limit
-            .as_ref()
-            .and_then(|config| config.download_mode_limit)
-            .is_some()
-        {
-            object_server.at(MANAGER_PATH, low_power_mode).await?;
-        }
+
+        object_server.at(MANAGER_PATH, low_power_mode).await?;
 
         let object_server = object_server.clone();
         spawn(async move {
@@ -1775,6 +1765,14 @@
         });
     }
 
+    if !crate::hhd::enable_power_features().await {
+        return Ok(());
+    }
+
+    let Some(config) = device_config().await? else {
+        return Ok(());
+    };
+
     if let Some(config) = config.performance_profile.as_ref()
         && !get_available_platform_profiles(&config.platform_profile_name)
             .await
@@ -1891,17 +1889,21 @@
         object_server.at(MANAGER_PATH, wifi_debug_dump).await?;
     }
 
-    if get_max_charge_level().await.is_ok() {
-        object_server.at(MANAGER_PATH, battery_charge_limit).await?;
-    }
+    let power_features = crate::hhd::enable_power_features().await;
 
-    if get_cpu_boost_state().await.is_ok() {
-        object_server.at(MANAGER_PATH, cpu_boost).await?;
-    }
+    if power_features {
+        if get_max_charge_level().await.is_ok() {
+            object_server.at(MANAGER_PATH, battery_charge_limit).await?;
+        }
+
+        if get_cpu_boost_state().await.is_ok() {
+            object_server.at(MANAGER_PATH, cpu_boost).await?;
+        }
 
-    object_server.at(MANAGER_PATH, cpu_scaling).await?;
-    if CpuSchedulerManager::is_supported().await? {
-        object_server.at(MANAGER_PATH, cpu_scheduler).await?;
+        object_server.at(MANAGER_PATH, cpu_scaling).await?;
+        if CpuSchedulerManager::is_supported().await? {
+            object_server.at(MANAGER_PATH, cpu_scheduler).await?;
+        }
     }
 
     match gpu_performance_level_driver().await {
@@ -1920,20 +1922,22 @@
         Err(e) => warn!("Can't add GpuPerformanceLevel1 interface: {e}"),
     }
 
-    match gpu_power_profile_driver().await {
-        Ok(driver) => {
-            object_server
-                .at(
-                    MANAGER_PATH,
-                    GpuPowerProfile1 {
-                        proxy: proxy.clone(),
-                        driver,
-                        order: SerialOrderValidator::default(),
-                    },
-                )
-                .await?;
+    if power_features {
+        match gpu_power_profile_driver().await {
+            Ok(driver) => {
+                object_server
+                    .at(
+                        MANAGER_PATH,
+                        GpuPowerProfile1 {
+                            proxy: proxy.clone(),
+                            driver,
+                            order: SerialOrderValidator::default(),
+                        },
+                    )
+                    .await?;
+            }
+            Err(e) => warn!("Can't add GpuPowerProfile1 interface: {e}"),
         }
-        Err(e) => warn!("Can't add GpuPowerProfile1 interface: {e}"),
     }
 
     if let Ok(hdmi_cec) = hdmi_cec {
--- a/data/system/steamos-manager.service
+++ b/data/system/steamos-manager.service
@@ -2,6 +2,10 @@
 Description=SteamOS Manager Daemon
 Wants=steamos-log-submitter.service
 After=steamos-log-submitter.service
+Wants=hhd.service
+Wants=hhdl.service
+After=hhd.service
+After=hhdl.service
 StartLimitIntervalSec=120
 StartLimitBurst=5
 

--- a/data/user/steamos-manager.service
+++ b/data/user/steamos-manager.service
@@ -1,5 +1,9 @@
 [Unit]
 Description=SteamOS Manager Daemon
+Before=graphical-session.target
+After=hhd.service
+After=hhdl.service
+Before=gamescope-session-plus@steam.service
 StartLimitIntervalSec=120
 StartLimitBurst=5
 
--- a/Makefile
+++ b/Makefile
@@ -44,4 +44,3 @@
 	install -m644 "data/user/steamos-manager.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-session-cleanup.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-configure-cecd.service" "$(DESTDIR)/usr/lib/systemd/user/"
-	install -m644 "data/user/orca.service" "$(DESTDIR)/usr/lib/systemd/user/"
