--- a/steamos-manager/src/power.rs
+++ b/steamos-manager/src/power.rs
@@ -21,10 +21,11 @@
 use tokio::fs::{self, File, read_dir, read_to_string, try_exists};
 use tokio::io::{AsyncWriteExt, ErrorKind, Interest};
 use tokio::net::unix::pipe;
+use tokio::process::Command;
 use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
 use tokio::sync::oneshot;
 use tokio::task::JoinSet;
-use tracing::{debug, error, warn};
+use tracing::{debug, error, info, warn};
 use zbus::names::OwnedBusName;
 use zbus::zvariant::OwnedObjectPath;
 use zbus::{Connection, ObjectServer, fdo};
@@ -168,7 +169,111 @@
     }
 }
 
+#[derive(PartialEq, Debug)]
+pub(crate) enum HhdStatus {
+    Inactive,
+    Active,
+    Conflicts,
+}
+
+pub(crate) const HHD_CMD: &str = "hhd.steamos";
+
+async fn get_hhd_status() -> HhdStatus {
+    let output = Command::new(HHD_CMD)
+        .arg("steamos-tdp")
+        .arg("get")
+        .output()
+        .await;
+
+    if output.is_err() {
+        return HhdStatus::Inactive;
+    }
+
+    let code = output.unwrap().status.code().unwrap_or(0);
+    match code {
+        1 => HhdStatus::Inactive,
+        2 => HhdStatus::Conflicts,
+        _ => HhdStatus::Active,
+    }
+}
+
+pub(crate) async fn enable_power_features() -> bool {
+    let hhd_status = get_hhd_status().await;
+    info!("Handheld daemon TDP status: {:?}", hhd_status);
+    hhd_status == HhdStatus::Inactive
+}
+
+pub(crate) struct HhdTdpManager {}
+
+#[async_trait]
+impl TdpLimitManager for HhdTdpManager {
+    async fn get_tdp_limit(&self) -> Result<u32> {
+        anyhow::bail!("Getting TDP from handheld daemon is not implemented.");
+    }
+
+    async fn set_tdp_limit(&self, limit: u32) -> Result<()> {
+        let output = Command::new(HHD_CMD)
+            .arg("steamos-tdp")
+            .arg(limit.to_string())
+            .output()
+            .await
+            .map_err(|e| anyhow!("Failed to execute hhd.steamos steamos-tdp {limit}: {e}"))?;
+
+        ensure!(
+            output.status.success(),
+            "hhd.steamos steamos-tdp exited with status {status}",
+            status = output.status
+        );
+
+        Ok(())
+    }
+
+    async fn get_tdp_limit_range(&self) -> Result<RangeInclusive<u32>> {
+        let output = Command::new(HHD_CMD)
+            .arg("steamos-tdp")
+            .arg("get")
+            .output()
+            .await
+            .map_err(|e| anyhow!("Failed to execute hhd.steamos steamos-tdp get: {e}"))?;
+
+        ensure!(
+            output.status.success(),
+            "hhd.steamos steamos-tdp get exited with status {status}",
+            status = output.status
+        );
+
+        let stdout = String::from_utf8(output.stdout)
+            .map_err(|e| anyhow!("Invalid UTF-8 from hhd.steamos steamos-tdp get: {e}"))?;
+
+        let mut parts = stdout.split_whitespace();
+        let min: u32 = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing min value"))?
+            .parse()
+            .map_err(|e| anyhow!("Failed to parse min value: {e}"))?;
+        let max: u32 = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing max value"))?
+            .parse()
+            .map_err(|e| anyhow!("Failed to parse max value: {e}"))?;
+        Ok(min..=max)
+    }
+
+    async fn is_active(&self) -> Result<bool> {
+        Ok(true)
+    }
+}
+
 pub(crate) async fn tdp_limit_manager(system: &Connection) -> Result<Box<dyn TdpLimitManager>> {
+    let hhd_status = get_hhd_status().await;
+
+    if hhd_status == HhdStatus::Active {
+        debug!("Using handheld daemon for TDP limiting");
+        return Ok(Box::new(HhdTdpManager {}));
+    } else if hhd_status == HhdStatus::Conflicts {
+        bail!("Conflicting TDP limiting method found");
+    }
+
     let config = device_config().await?;
     if let Some(config) = config.as_ref().and_then(|config| config.tdp_limit.as_ref()) {
         Ok(match &config.method {
--- a/steamos-manager/src/gpu.rs
+++ b/steamos-manager/src/gpu.rs
@@ -18,10 +18,11 @@
 use strum::{Display, EnumString, VariantNames};
 use tokio::fs::{self, File, try_exists};
 use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::process::Command;
 use tracing::{debug, error};
 
 use crate::hardware::{device_config, device_type};
-use crate::power::find_hwmon;
+use crate::power::{HhdStatus, HHD_CMD, find_hwmon};
 use crate::{path, write_synced};
 
 pub(crate) const AMDGPU_HWMON_NAME: &str = "amdgpu";
@@ -150,7 +151,134 @@
     })
 }
 
+async fn get_hhd_status() -> HhdStatus {
+    let output = Command::new(HHD_CMD)
+        .arg("steamos-gpu")
+        .arg("get")
+        .output()
+        .await;
+
+    if output.is_err() {
+        return HhdStatus::Inactive;
+    }
+
+    let code = output.unwrap().status.code().unwrap_or(0);
+    match code {
+        1 => HhdStatus::Inactive,
+        2 => HhdStatus::Conflicts,
+        _ => HhdStatus::Active,
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct HhdPerformanceLevelDriver {}
+
+impl AmdgpuGpuPerfDriver for HhdPerformanceLevelDriver {}
+
+#[async_trait]
+impl GpuPerformanceLevelDriver for HhdPerformanceLevelDriver {
+    fn performance_level_from_str(&self, value: &str) -> Result<GpuPerformanceLevel> {
+        Ok(GpuPerformanceLevel::Intel(IntelPerformanceLevel::from_str(
+            value,
+        )?))
+    }
+
+    async fn get_available_performance_levels(&self) -> Result<Vec<GpuPerformanceLevel>> {
+        Ok(vec![
+            GpuPerformanceLevel::Intel(IntelPerformanceLevel::Auto),
+            GpuPerformanceLevel::Intel(IntelPerformanceLevel::Manual),
+        ])
+    }
+
+    async fn get_performance_level(&self) -> Result<GpuPerformanceLevel> {
+        // Just lie so that we always get a trigger to auto
+        Ok(GpuPerformanceLevel::Intel(IntelPerformanceLevel::Manual))
+    }
+
+    async fn get_clocks(&self) -> Result<u32> {
+        // Dummy value
+        Ok(1200)
+    }
+
+    async fn set_performance_level(&self, level: GpuPerformanceLevel) -> Result<()> {
+        if level == GpuPerformanceLevel::Intel(IntelPerformanceLevel::Manual) {
+            return Ok(());
+        }
+        let output = Command::new(HHD_CMD)
+            .arg("steamos-gpu")
+            .arg("clear")
+            .output()
+            .await
+            .map_err(|e| anyhow!("Failed to clear gpu limit: {e}"))?;
+
+        ensure!(
+            output.status.success(),
+            "hhd.steamos steamos-tdp exited with status {status}",
+            status = output.status
+        );
+
+        Ok(())
+    }
+
+    async fn get_clocks_range(&self) -> Result<RangeInclusive<u32>> {
+        let output = Command::new(HHD_CMD)
+            .arg("steamos-gpu")
+            .arg("get")
+            .output()
+            .await
+            .map_err(|e| anyhow!("Failed to execute hhd.steamos steamos-tdp get: {e}"))?;
+
+        ensure!(
+            output.status.success(),
+            "hhd.steamos get exited with status {status}",
+            status = output.status
+        );
+
+        let stdout = String::from_utf8(output.stdout)
+            .map_err(|e| anyhow!("Invalid UTF-8 from hhd.steamos steamos-tdp get: {e}"))?;
+
+        let mut parts = stdout.split_whitespace();
+        let min: u32 = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing min value"))?
+            .parse()
+            .map_err(|e| anyhow!("Failed to parse min value: {e}"))?;
+        let max: u32 = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing max value"))?
+            .parse()
+            .map_err(|e| anyhow!("Failed to parse max value: {e}"))?;
+        Ok(min..=max)
+    }
+
+    async fn set_clocks(&self, clocks: u32) -> Result<()> {
+        let output = Command::new(HHD_CMD)
+            .arg("steamos-gpu")
+            .arg(clocks.to_string())
+            .output()
+            .await
+            .map_err(|e| anyhow!("Failed to execute hhd.steamos steamos-tdp {clocks}: {e}"))?;
+
+        ensure!(
+            output.status.success(),
+            "hhd.steamos steamos-tdp exited with status {status}",
+            status = output.status
+        );
+
+        Ok(())
+    }
+}
+
 pub(crate) async fn gpu_performance_level_driver() -> Result<Box<dyn GpuPerformanceLevelDriver>> {
+    let hhd_status = get_hhd_status().await;
+
+    if hhd_status == HhdStatus::Active {
+        debug!("Using handheld daemon for TDP limiting");
+        return Ok(Box::new(HhdPerformanceLevelDriver {}));
+    } else if hhd_status == HhdStatus::Conflicts {
+        bail!("Conflicting GPU controls found");
+    }
+
     let config = device_config().await?;
     let config = config
         .as_ref()
--- a/steamos-manager/src/manager/user.rs
+++ b/steamos-manager/src/manager/user.rs
@@ -50,9 +50,9 @@
 use crate::platform::platform_config;
 use crate::power::{
     BATTERY_DEFAULT_SUGGESTED_MINIMUM_LIMIT, CpuSchedulerManager, TdpManagerCommand,
-    get_available_cpu_scaling_governors, get_available_platform_profiles, get_cpu_boost_state,
-    get_cpu_scaling_governor, get_max_charge_level, get_platform_profile, register_tdp_limit1,
-    unregister_tdp_limit1,
+    enable_power_features, get_available_cpu_scaling_governors, get_available_platform_profiles,
+    get_cpu_boost_state, get_cpu_scaling_governor, get_max_charge_level, get_platform_profile,
+    register_tdp_limit1, unregister_tdp_limit1,
 };
 use crate::proxy::{
     BatteryChargeLimit1Proxy, CpuBoost1Proxy, FactoryReset1Proxy, FanControl1Proxy,
@@ -1738,10 +1738,6 @@
     object_server: &ObjectServer,
     tdp_manager: Option<UnboundedSender<TdpManagerCommand>>,
 ) -> Result<()> {
-    let Some(config) = device_config().await? else {
-        return Ok(());
-    };
-
     let performance_profile = PerformanceProfile1 {
         proxy: proxy.clone(),
         tdp_limit_manager: tdp_manager.clone(),
@@ -1751,14 +1747,8 @@
         let low_power_mode = LowPowerMode1 {
             manager: manager.clone(),
         };
-        if config
-            .tdp_limit
-            .as_ref()
-            .and_then(|config| config.download_mode_limit)
-            .is_some()
-        {
-            object_server.at(MANAGER_PATH, low_power_mode).await?;
-        }
+
+        object_server.at(MANAGER_PATH, low_power_mode).await?;
 
         let object_server = object_server.clone();
         spawn(async move {
@@ -1775,6 +1765,14 @@
         });
     }
 
+    if !enable_power_features().await {
+        return Ok(());
+    }
+
+    let Some(config) = device_config().await? else {
+        return Ok(());
+    };
+
     if let Some(config) = config.performance_profile.as_ref()
         && !get_available_platform_profiles(&config.platform_profile_name)
             .await
@@ -1891,17 +1889,21 @@
         object_server.at(MANAGER_PATH, wifi_debug_dump).await?;
     }
 
-    if get_max_charge_level().await.is_ok() {
-        object_server.at(MANAGER_PATH, battery_charge_limit).await?;
-    }
+    let power_features = enable_power_features().await;
 
-    if get_cpu_boost_state().await.is_ok() {
-        object_server.at(MANAGER_PATH, cpu_boost).await?;
-    }
+    if power_features {
+        if get_max_charge_level().await.is_ok() {
+            object_server.at(MANAGER_PATH, battery_charge_limit).await?;
+        }
+
+        if get_cpu_boost_state().await.is_ok() {
+            object_server.at(MANAGER_PATH, cpu_boost).await?;
+        }
 
-    object_server.at(MANAGER_PATH, cpu_scaling).await?;
-    if CpuSchedulerManager::is_supported().await? {
-        object_server.at(MANAGER_PATH, cpu_scheduler).await?;
+        object_server.at(MANAGER_PATH, cpu_scaling).await?;
+        if CpuSchedulerManager::is_supported().await? {
+            object_server.at(MANAGER_PATH, cpu_scheduler).await?;
+        }
     }
 
     match gpu_performance_level_driver().await {
@@ -1920,20 +1922,22 @@
         Err(e) => warn!("Can't add GpuPerformanceLevel1 interface: {e}"),
     }
 
-    match gpu_power_profile_driver().await {
-        Ok(driver) => {
-            object_server
-                .at(
-                    MANAGER_PATH,
-                    GpuPowerProfile1 {
-                        proxy: proxy.clone(),
-                        driver,
-                        order: SerialOrderValidator::default(),
-                    },
-                )
-                .await?;
+    if power_features {
+        match gpu_power_profile_driver().await {
+            Ok(driver) => {
+                object_server
+                    .at(
+                        MANAGER_PATH,
+                        GpuPowerProfile1 {
+                            proxy: proxy.clone(),
+                            driver,
+                            order: SerialOrderValidator::default(),
+                        },
+                    )
+                    .await?;
+            }
+            Err(e) => warn!("Can't add GpuPowerProfile1 interface: {e}"),
         }
-        Err(e) => warn!("Can't add GpuPowerProfile1 interface: {e}"),
     }
 
     if let Ok(hdmi_cec) = hdmi_cec {
--- a/data/system/steamos-manager.service
+++ b/data/system/steamos-manager.service
@@ -2,6 +2,10 @@
 Description=SteamOS Manager Daemon
 Wants=steamos-log-submitter.service
 After=steamos-log-submitter.service
+Wants=hhd.service
+Wants=hhdl.service
+After=hhd.service
+After=hhdl.service
 StartLimitIntervalSec=120
 StartLimitBurst=5
 

--- a/data/user/steamos-manager.service
+++ b/data/user/steamos-manager.service
@@ -1,5 +1,9 @@
 [Unit]
 Description=SteamOS Manager Daemon
+Before=graphical-session.target
+After=hhd.service
+After=hhdl.service
+Before=gamescope-session-plus@steam.service
 StartLimitIntervalSec=120
 StartLimitBurst=5
 
--- a/Makefile
+++ b/Makefile
@@ -44,4 +44,3 @@
 	install -m644 "data/user/steamos-manager.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-session-cleanup.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-configure-cecd.service" "$(DESTDIR)/usr/lib/systemd/user/"
-	install -m644 "data/user/orca.service" "$(DESTDIR)/usr/lib/systemd/user/"
