diff --git a/Makefile b/Makefile
index 699d781..e77c4f6 100644
--- a/Makefile
+++ b/Makefile
@@ -57,14 +57,18 @@ ccflags-y +=	-D COREFREQ_MAJOR=$(COREFREQ_MAJOR) \
 ccflags-y += $(WARNING)
 
 ifeq ($(OPTIM_LVL),0)
-OPTIM_FLG = -O$(OPTIM_LVL)
+CFLAGS += -O$(OPTIM_LVL)
+LDFLAGS += -O$(OPTIM_LVL)
 ccflags-y += -fno-inline
 else ifneq ($(OPTIM_LVL),)
-OPTIM_FLG = -O$(OPTIM_LVL)
+CFLAGS += -O$(OPTIM_LVL)
+LDFLAGS += -O$(OPTIM_LVL)
 ccflags-y += -D OPTIM_LVL=$(OPTIM_LVL)
-ccflags-y += $(OPTIM_FLG)
+ccflags-y += $(OPTIM_LVL)
 endif
 
+CFLAGS +=  $(WARNING)
+
 DEFINITIONS =	-D COREFREQ_MAJOR=$(COREFREQ_MAJOR) \
 		-D COREFREQ_MINOR=$(COREFREQ_MINOR) \
 		-D COREFREQ_REV=$(COREFREQ_REV) \
@@ -285,17 +289,17 @@ clean:
 
 $(BUILD)/corefreqm.o: $(HW)/corefreqm.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) -pthread $(DEFINITIONS) \
+	$(CC)) $(CFLAGS) $(WARNING) -pthread $(DEFINITIONS) \
 	  -c $(HW)/corefreqm.c -o $(BUILD)/corefreqm.o
 
 $(BUILD)/corefreqd.o: $(HW)/corefreqd.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) -pthread $(DEFINITIONS) \
+	$(CC)) $(CFLAGS) $(WARNING) -pthread $(DEFINITIONS) \
 	  -c $(HW)/corefreqd.c -o $(BUILD)/corefreqd.o
 
 $(BUILD)/corefreqd: prepare $(BUILD)/corefreqd.o $(BUILD)/corefreqm.o
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  LD [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) -o $(BUILD)/corefreqd \
+	$(CC)) $(CFLAGS) -o $(BUILD)/corefreqd \
 	  $(BUILD)/corefreqd.o $(BUILD)/corefreqm.o -lpthread -lm -lrt -lc
 
 .PHONY: corefreqd
@@ -303,22 +307,22 @@ corefreqd: $(BUILD)/corefreqd
 
 $(BUILD)/corefreq-ui.o: $(HW)/corefreq-ui.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) $(DEFINITIONS) \
+	$(CC)) $(CFLAGS) $(WARNING) $(DEFINITIONS) \
 	  -c $(HW)/corefreq-ui.c -o $(BUILD)/corefreq-ui.o
 
 $(BUILD)/corefreq-cli.o: $(HW)/corefreq-cli.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) $(DEFINITIONS) $(LAYOUT) \
+	$(CC)) $(CFLAGS) $(WARNING) $(DEFINITIONS) $(LAYOUT) \
 	  -c $(HW)/corefreq-cli.c -o $(BUILD)/corefreq-cli.o
 
 $(BUILD)/corefreq-cli-rsc.o: $(HW)/corefreq-cli-rsc.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) $(DEFINITIONS) $(LAYOUT) \
+	$(CC)) $(CFLAGS) $(WARNING) $(DEFINITIONS) $(LAYOUT) \
 	  -c $(HW)/corefreq-cli-rsc.c -o $(BUILD)/corefreq-cli-rsc.o
 
 $(BUILD)/corefreq-cli-json.o: $(HW)/corefreq-cli-json.c
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  CC [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) $(WARNING) $(DEFINITIONS) \
+	$(CC)) $(CFLAGS) $(WARNING) $(DEFINITIONS) \
 	  -c $(HW)/corefreq-cli-json.c -o $(BUILD)/corefreq-cli-json.o
 
 $(BUILD)/corefreq-cli-extra.o: $(HW)/corefreq-cli-extra.c
@@ -333,7 +337,7 @@ $(BUILD)/corefreq-cli:	prepare \
 			$(BUILD)/corefreq-cli-json.o \
 			$(BUILD)/corefreq-cli-extra.o
 	$(if $(V), $(CC), @if [ ${SILENT} -eq 0 ]; then echo "  LD [$@]"; fi; \
-	$(CC)) $(OPTIM_FLG) -o $(BUILD)/corefreq-cli \
+	$(CC)) $(LDFLAGS) -o $(BUILD)/corefreq-cli \
 	  $(BUILD)/corefreq-cli.o $(BUILD)/corefreq-ui.o \
 	  $(BUILD)/corefreq-cli-rsc.o $(BUILD)/corefreq-cli-json.o \
 	  $(BUILD)/corefreq-cli-extra.o -lm -lrt -lc
@@ -345,6 +349,8 @@ corefreq-cli: $(BUILD)/corefreq-cli
 info:
 	$(info HW [$(HW)])
 	$(info CC [$(shell whereis -b $(CC))])
+	$(info CFLAGS [$(CFLAGS)])
+	$(info LDFLAGS [$(LDFLAGS)])
 	$(info WARNING [$(WARNING)])
 	$(info PWD [$(PWD)])
 	$(info BUILD [$(BUILD)])
