diff -Naur gri-2.12.23.orig/doc/gri.texi gri-2.12.23.new/doc/gri.texi
--- gri-2.12.23.orig/doc/gri.texi	2018-02-18 20:31:30.170994925 +0100
+++ gri-2.12.23.new/doc/gri.texi	2018-02-18 20:44:55.136236169 +0100
@@ -18486,7 +18486,7 @@
 (Actually, the windows version is just the MSDOS version, which can be
 run inside an msdos window within windows-95, windows-NT, etc.)
 
-@subsubsection MSDOS Operating System
+@subsection MSDOS Operating System
 @cindex compilation under MSDOS
 @cindex MSDOS compilation
 
diff -Naur gri-2.12.23.orig/doc/texinfo2HTML gri-2.12.23.new/doc/texinfo2HTML
--- gri-2.12.23.orig/doc/texinfo2HTML	2018-02-18 20:31:30.172994915 +0100
+++ gri-2.12.23.new/doc/texinfo2HTML	2018-02-18 21:04:04.482173015 +0100
@@ -120,7 +120,7 @@
 	next;
     }
     # Process image commands (NOT...it's there already) 
-    if (/\@image{(.*)}/) {
+    if (/\@image\{(.*)\}/) {
 	#print "<img src=\"$1.gif\" border=\"1\">\n";
 	next;
     }
@@ -208,12 +208,12 @@
     s,<,&lt;,og;
     # Special tweak to make 'Gri' look cooler.
 	#    s, Gri , G<FONT SIZE=-1>RI</FONT> ,og;
-    s,\@code{\@\@},\@code{TEXINFO2HTML-AT-AT},og; 
+    s,\@code\{\@\@\},\@code\{TEXINFO2HTML-AT-AT\},og; 
     s,\@},TEXINFO2HTML-CLOSE-BRACE,og; # retain inside e.g. @code{}
     s,\@\@,TEXINFO2HTML-AT-AT,og;
-    s,\@{,{,og;
-    s,\@TeX{},TeX,og;
-    s,\@dots{},...,og;
+    s,\@\{,\{,og;
+    s,\@TeX\{\},TeX,og;
+    s,\@dots\{\},...,og;
     # Put in place-holders for some accents.  I should check for all
     # of them, but for now, I'm just kludging in a couple, 
     # to solve an immediate problem and to serve as a place-holder
@@ -237,18 +237,18 @@
 	}
 	next;
     }
-    while (/\@url{([^}]*)}/) {
+    while (/\@url\{([^}]*)\}/) {
 	$the_url = $1;
-        s:\@url{[^}]*}:<a href="$the_url">\@code{$the_url}</a>:;
+        s:\@url\{[^}]*\}:<a href="$the_url">\@code{$the_url}</a>:;
     }
-    while (/\@uref{([^}]*)}/) {
+    while (/\@uref\{([^}]*)\}/) {
 	@items = split(/,/, $1);
         if ($#items == 0) {
-            s:\@uref{[^}]*}:<a href="$items[0]">$items[0]</a>:;
+            s:\@uref\{[^}]*\}:<a href="$items[0]">$items[0]</a>:;
         } elsif ($#items == 1) {
-            s:\@uref{[^}]*}:<a href="$items[0]">$items[1]</a>:;
+            s:\@uref\{[^}]*\}:<a href="$items[0]">$items[1]</a>:;
         } elsif ($#items == 2) {
-            s:\@uref{[^}]*}:\@code{$items[2]}:;
+            s:\@uref\{[^}]*\}:\@code{$items[2]}:;
         } else {
             die "Cannot have more than 3 items in a 'uref' at \"$_\"";
         }
@@ -330,7 +330,7 @@
 	while(get_a_line()) {
 	    next if /\@sp/;
 	    next if /\@cindex/;
-            s|\@anchor{([^}]*)}|<a name=\"$1\"></a>|g;
+            s|\@anchor\{([^}]*)\}|<a name=\"$1\"></a>|g;
 	    &process_examples();
 	    # Handle HTML inserts
 	    if (/^\@c HTML (.*)/o) {
@@ -346,10 +346,10 @@
 	    }
 	    if (/\s*\@item\s*(.*)/o) {
                 $the_item = $1;
-		$the_item =~ s:\@{:{:og;
-		$the_item =~ s:\@}:}:og;
+		$the_item =~ s:\@\{:{:og;
+		$the_item =~ s:\@\}:}:og;
 		$the_item =~ s:\@\@:\@:og;
-		$the_item =~ s:\@code{([^}]*)}:`<font color="$ex_color"><code>$1</code></font>':og;
+		$the_item =~ s:\@code\{([^}]*)\}:`<font color="$ex_color"><code>$1</code></font>':og;
 		print "<dt> $start_item$the_item$end_item\n<dd>";
 	    } else {
 		print "<p>" if (/^$/o);
@@ -401,39 +401,39 @@
 	$in_example = 0;
     } else {
         s,\@value\{([^}]*)\},$value{$1},g; # Substitute set/value pair
-        s,\@},},og;
-        s,\@{,{,og;
+        s,\@\},},og;
+        s,\@\{,{,og;
     }
 }
 
 
 sub sub_refs {
-    die "line $. of file: cannot have multiple refs on one line" if (/\@[px]*ref{(.*)}(.*)\@[px]*ref{(.*)}/);
+    die "line $. of file: cannot have multiple refs on one line" if (/\@[px]*ref\{(.*)\}(.*)\@[px]*ref\{(.*)\}/);
     # anchors
-    s|\@anchor{([^}]*)}|<a name=\"$1\"></a>|g;
+    s|\@anchor\{([^}]*)\}|<a name=\"$1\"></a>|g;
     # Change e.g. 
     #     @xref{Viewing}
     # into
     #     <a href="#Viewing">see Viewing</a>
 
     #if (/\@ref/){print "AAA[$_]AAA\n";
-    s|\@ref{([^}]*)}|see <a href="#$1">$1</a>|g;
+    s|\@ref\{([^}]*)\}|see <a href="#$1">$1</a>|g;
     #print "BBB[$_]BBB\n";}
 
-    s|\@xref{([^}]*)}|see <a href="#$1">$1</a>|g;
-    s|\@pxref{([^}]*)}|see <a href="#$1">$1.</a>|g;
-    while (/\@url{([^}]*)}/) {
+    s|\@xref\{([^}]*)\}|see <a href="#$1">$1</a>|g;
+    s|\@pxref\{([^}]*)\}|see <a href="#$1">$1.</a>|g;
+    while (/\@url\{([^}]*)\}/) {
 	$the_url = $1;
-        s:\@url{[^}]*}:<a href="$the_url">\@code{$the_url}</a>:;
+        s:\@url\{[^}]*\}:<a href="$the_url">\@code{$the_url}</a>:;
     }
-    while (/\@uref{([^}]*)}/) {
+    while (/\@uref\{([^}]*)\}/) {
 	@items = split(/,/, $1);
         if ($#items == 0) {
-            s:\@uref{[^}]*}:<a href="$items[0]">$items[0]</a>:;
+            s:\@uref\{[^}]*\}:<a href="$items[0]">$items[0]</a>:;
         } elsif ($#items == 1) {
-            s:\@uref{[^}]*}:<a href="$items[0]">$items[1]</a>:;
+            s:\@uref\{[^}]*\}:<a href="$items[0]">$items[1]</a>:;
         } elsif ($#items == 2) {
-            s:\@uref{[^}]*}:\@code{$items[2]}:;
+            s:\@uref\{[^}]*\}:\@code{$items[2]}:;
         } else {
             die "Cannot have more than 3 items in a 'uref' at \"$_\"";
         }
@@ -466,17 +466,17 @@
 sub sub_emphasis {
     s,<<,&lt&lt,g;
     s,>>,&gt&gt,g;
-    s,\@emph{([^}]*)},<em>$1</em>,g;
-    s,\@strong{([^}]*)},<b>$1</b>,g;
-    s,\@footnote{([^}]*)}, [$1],g;
-    s,\@b{([^}]*)},<b>$1</b>,g;
+    s,\@emph\{([^}]*)\},<em>$1</em>,g;
+    s,\@strong\{([^}]*)\},<b>$1</b>,g;
+    s,\@footnote\{([^}]*)\}, [$1],g;
+    s,\@b\{([^}]*)\},<b>$1</b>,g;
 
-    s,\@code{([^}]*)},`<font color="$ex_color"><code>$1</code></font>',g;
+    s,\@code\{([^}]*)\},`<font color="$ex_color"><code>$1</code></font>',g;
     s,\@\@,\@,g;
-    s,\@samp{([^}]*)},`<font color="$ex_color"><samp>$1</samp></font>',g;
-    s,\@key{([^}]*)},`<font color="$ex_color"><kbd>$1</kbd></font>',g;
-    s,\@kbd{([^}]*)},`<font color="$ex_color"><kbd>$1</kbd></font>',g;
-    s,\@file{([^}]*)},`<font color="$ex_color"><samp>$1</samp></font>',g;
+    s,\@samp\{([^}]*)\},`<font color="$ex_color"><samp>$1</samp></font>',g;
+    s,\@key\{([^}]*)\},`<font color="$ex_color"><kbd>$1</kbd></font>',g;
+    s,\@kbd\{([^}]*)\},`<font color="$ex_color"><kbd>$1</kbd></font>',g;
+    s,\@file\{([^}]*)\},`<font color="$ex_color"><samp>$1</samp></font>',g;
     s,TEXINFO2HTML-CLOSE-BRACE,},g;
     s,TEXINFO2HTML-AT-AT,\@,g;
     s,TEXINFO2HTML-ACCENT-ACUTE-a,&#225,g;
