--- http.js	2015-12-09 16:01:12.418503712 +0100
+++ http.js.new	2015-12-09 16:02:41.956760729 +0100
@@ -14,9 +14,9 @@
     url = require("url"),
     path = require("path"),
     iniparser = require('iniparser'),
-    php = require("./lib/php.js"),
-    out = require("./lib/out.js"),
-    fsd = require("./lib/fsaccess.js"),
+    php = require("/var/lib/httpjs/js/php.js"),
+    out = require("/var/lib/httpjs/js/out.js"),
+    fsd = require("/var/lib/httpjs/js/fsaccess.js"),
     zlib = require("zlib"),
     httpserver = null,
     phpserver = null,
