diff -rpu --color old/usr/bin/ice new/usr/bin/ice
--- old/usr/bin/ice	2021-05-10 15:47:34.278040167 +0200
+++ new/usr/bin/ice	2021-05-10 15:48:35.098265219 +0200
@@ -44,8 +44,8 @@ _EPIPHANY_PROFILES_DIR = "{0}/epiphany".
 _ICE_ICON = "/usr/share/pixmaps/ice.png"
 _ICON_DIR = "{0}/icons".format(_ICE_DIR)
 _BRAVE_BIN = "/usr/bin/brave-browser"
-_CHROME_BIN = "/usr/bin/google-chrome"
-_CHROMIUM_BIN = "/usr/bin/chromium-browser"
+_CHROME_BIN = "/usr/bin/google-chrome-stable"
+_CHROMIUM_BIN = "/usr/bin/chromium"
 _VIVALDI_BIN = "/usr/bin/vivaldi-stable"
 _FIREFOX_BIN = "/usr/bin/firefox"
 _EPIPHANY_BIN = "/usr/bin/epiphany"
@@ -699,7 +699,7 @@ class Ice(Gtk.Window):
         if self.chrome.get_active() is True:
             self.browser = "google-chrome"
         elif self.chromium.get_active() is True:
-            self.browser = "chromium-browser"
+            self.browser = "chromium"
         elif self.brave.get_active() is True:
             self.browser = "brave"
         elif self.vivaldi.get_active() is True:
