diff --unified --recursive --text a/tests/test_authentication.py b/tests/test_authentication.py
--- a/tests/test_authentication.py	2025-09-02 02:53:56.000000000 +0200
+++ b/tests/test_authentication.py	2026-01-09 12:42:30.727664351 +0100
@@ -111,6 +111,7 @@
         self.assertIn("Authentication completed successfully", result.output)
         self.assertEqual(result.exit_code, 0, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_password_prompt_2sa(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -147,6 +148,7 @@
         )
         self.assertEqual(result.exit_code, 0, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_password_prompt_2fa(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
diff --unified --recursive --text a/tests/test_email_notifications.py b/tests/test_email_notifications.py
--- a/tests/test_email_notifications.py	2025-09-02 02:53:56.000000000 +0200
+++ b/tests/test_email_notifications.py	2026-01-09 12:46:13.705742480 +0100
@@ -17,6 +17,7 @@
         self.vcr_path = os.path.join(self.root_path, "vcr_cassettes")
 
     @freeze_time("2018-01-01")
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_required_email_notification(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -65,6 +66,7 @@
             )
 
     @freeze_time("2018-01-01")
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_notification_without_smtp_login_and_tls(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -109,6 +111,7 @@
             )
 
     @freeze_time("2018-01-01")
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_required_notification_script(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -138,6 +141,7 @@
             subprocess_patched.assert_called_once_with(["test_script.sh"])
 
     @freeze_time("2018-01-01")
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_required_email_notification_from(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
diff --unified --recursive --text a/tests/test_two_step_auth.py b/tests/test_two_step_auth.py
--- a/tests/test_two_step_auth.py	2025-09-02 02:53:56.000000000 +0200
+++ b/tests/test_two_step_auth.py	2026-01-09 12:48:28.064592617 +0100
@@ -21,6 +21,7 @@
         self.fixtures_path = os.path.join(self.root_path, "fixtures")
         self.vcr_path = calc_vcr_dir(self.root_path)
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_flow_invalid_code(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = calc_cookie_dir(base_dir)
@@ -49,6 +50,7 @@
 
         self.assertEqual(result.exit_code, 1, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_flow_valid_code(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -85,6 +87,7 @@
         )
         self.assertEqual(result.exit_code, 0, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2sa_flow_failed_send_code(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -121,6 +124,7 @@
             )
             self.assertEqual(result.exit_code, 1, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2fa_flow_invalid_code(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -149,6 +153,7 @@
 
         self.assertEqual(result.exit_code, 1, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2fa_flow_valid_code(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
@@ -186,6 +191,7 @@
         )
         self.assertEqual(result.exit_code, 0, "exit code")
 
+    @pytest.mark.skip(reason="requires interactive input")
     def test_2fa_flow_valid_code_zero_lead(self) -> None:
         base_dir = os.path.join(self.fixtures_path, inspect.stack()[0][3])
         cookie_dir = os.path.join(base_dir, "cookie")
