diff -Naur word-sys-pdf-editor-1.11.1.orig/word_sys_pdf_editor/welcome_view.py word-sys-pdf-editor-1.11.1/word_sys_pdf_editor/welcome_view.py
--- word-sys-pdf-editor-1.11.1.orig/word_sys_pdf_editor/welcome_view.py	2026-10-02 14:50:22.000000000 +0200
+++ word-sys-pdf-editor-1.11.1/word_sys_pdf_editor/welcome_view.py	2026-10-03 20:34:33.780195678 +0200
@@ -32,7 +32,7 @@
         main_box.set_margin_bottom(40)
 
         try:
-            app_icon = Gtk.Image.new_from_icon_name("f-pv1")
+            app_icon = Gtk.Image.new_from_icon_name("word-sys-pdf-editor")
             app_icon.set_pixel_size(200)
             app_icon.set_valign(Gtk.Align.END)
             app_icon.set_halign(Gtk.Align.CENTER)
@@ -292,4 +292,4 @@
             self.parent_window.load_document(row._file_path)
         elif hasattr(row, '_uri') and row._uri:
             gfile = Gio.File.new_for_uri(row._uri)
-            self.parent_window.load_document(gfile.get_path())
\ Manca newline alla fine del file
+            self.parent_window.load_document(gfile.get_path())
diff -Naur word-sys-pdf-editor-1.11.1.orig/word_sys_pdf_editor/window.py word-sys-pdf-editor-1.11.1/word_sys_pdf_editor/window.py
--- word-sys-pdf-editor-1.11.1.orig/word_sys_pdf_editor/window.py	2026-10-02 14:50:22.000000000 +0200
+++ word-sys-pdf-editor-1.11.1/word_sys_pdf_editor/window.py	2026-10-03 20:34:57.192060907 +0200
@@ -43,7 +43,7 @@
         super().__init__(*args, **kwargs)
         self.set_title(constants.APP_NAME)
         self.set_default_size(1200, 800)
-        self.set_icon_name("f-pv1")
+        self.set_icon_name("word-sys-pdf-editor")
 
         self.sessions = []
         self._active_session = DocumentSession(undo_manager=UndoManager(self))
@@ -1762,7 +1762,7 @@
         about_dialog.set_comments(_("about_comments"))
 
         try:
-            about_dialog.set_logo_icon_name("f-pv1")
+            about_dialog.set_logo_icon_name("word-sys-pdf-editor")
         except Exception as e:
             print(f"Warning: Could not load application icon for About dialog: {e}")
             about_dialog.set_logo_icon_name("application-x-executable")
