From c60bb81d33c760356e97385ec3209a838221d9b6 Mon Sep 17 00:00:00 2001
From: 0x9fff00 <0x9fff00+git@protonmail.ch>
Date: Sat, 20 Jun 2020 20:51:41 +0200
Subject: [PATCH 2/2] Update libirecovery and libplist names for 1.0.0/2.2.0

---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 99d22c1..d234c71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,9 +14,9 @@ AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
 # Minimum package versions
-LIBIRECOVERY_VERSION=0.2.0
+LIBIRECOVERY_VERSION=1.0.0
 LIBIMOBILEDEVICE_VERSION=1.2.1
-LIBPLIST_VERSION=1.12
+LIBPLIST_VERSION=2.2.0
 LIBZIP_VERSION=0.8
 LIBCURL_VERSION=7.0
 OPENSSL_VERSION=0.9.8
@@ -37,9 +37,9 @@ AC_PROG_LIBTOOL
 
 # Checks for libraries.
 AC_SEARCH_LIBS([pthread_create], [pthread])
-PKG_CHECK_MODULES(libirecovery, libirecovery >= $LIBIRECOVERY_VERSION)
+PKG_CHECK_MODULES(libirecovery, libirecovery-1.0 >= $LIBIRECOVERY_VERSION)
 PKG_CHECK_MODULES(libimobiledevice, libimobiledevice-1.0 >= $LIBIMOBILEDEVICE_VERSION)
-PKG_CHECK_MODULES(libplist, libplist >= $LIBPLIST_VERSION)
+PKG_CHECK_MODULES(libplist, libplist-2.0 >= $LIBPLIST_VERSION)
 PKG_CHECK_MODULES(libzip, libzip >= $LIBZIP_VERSION)
 PKG_CHECK_MODULES(libcurl, libcurl >= $LIBCURL_VERSION)
 PKG_CHECK_MODULES(openssl, openssl >= $OPENSSL_VERSION)
-- 
2.27.0

