diff --git a/ayatanawebmail/idler.py b/ayatanawebmail/idler.py
index 8f956ea..b6b63b4 100755
--- a/ayatanawebmail/idler.py
+++ b/ayatanawebmail/idler.py
@@ -22,6 +22,12 @@ class Idler(object):
 
     def stop(self):
 
+        if self.oConnection.oImap is not None:
+            try:
+                # Send a NOOP command to interrupt the IDLE mode and free the blocked thread
+                self.oConnection.oImap.noop()
+            except:
+                pass
         self.oEvent.set()
 
     def join(self):
