diff --git a/tox.ini b/tox.ini
index d8c2c29..fcdc474 100644
--- a/tox.ini
+++ b/tox.ini
@@ -10,7 +10,7 @@ commands =
                f.write('import coverage\ncoverage.process_startup()')"
     # nose ignores __main__.py by default but we want to cover this file. Only
     # ignore files whose name starts with a dot.
-    nosetests -v --ignore-files=^\.$ --with-coverage --cover-erase \
+    nosetests -v --ignore-files=(^\.|^setup\.py$) --with-coverage --cover-erase \
               --cover-package=jedihttp --cover-html --cover-inclusive {posargs}
 [testenv:py26]
 deps =
