diff -ruN a/ffmpegthumbnailer/imagewriter.cpp b/ffmpegthumbnailer/imagewriter.cpp
--- a/ffmpegthumbnailer/imagewriter.cpp	2016-05-29 20:37:24.000000000 +0200
+++ b/ffmpegthumbnailer/imagewriter.cpp	2018-04-21 22:57:55.608039514 +0200
@@ -62,7 +62,7 @@
         if (q1.exists()){
             QImage matteImage(QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/ffmpegthumbs-mattepaint/quadrant1");
             QPainter painter(&image);
-            if ((image.height() <= 128) and (image.width() <= 128)) {	
+            if ((image.height() <= 128) && (image.width() <= 128)) {	
                 painter.drawImage( 0,  0, matteImage.scaled(128, 128, Qt::KeepAspectRatio, Qt::FastTransformation));
             } else {
                     painter.drawImage( 0,  0, matteImage);
@@ -73,7 +73,7 @@
             QImage matteImage(QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/ffmpegthumbs-mattepaint/quadrant2");
             mirror = image.mirrored(true, false);
             QPainter painter(&mirror);
-            if ((image.height() <= 128) and (image.width() <= 128)) {
+            if ((image.height() <= 128) && (image.width() <= 128)) {
                 painter.drawImage( 0,  0, matteImage.scaled(128, 128, Qt::KeepAspectRatio, Qt::FastTransformation)); 
             } else {
                 painter.drawImage( 0,  0, matteImage);
@@ -85,7 +85,7 @@
             QImage matteImage(QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/ffmpegthumbs-mattepaint/quadrant3");
             mirror = image.mirrored(false, true);
             QPainter painter(&mirror);
-            if ((image.height() <= 128) and (image.width() <= 128)) {
+            if ((image.height() <= 128) && (image.width() <= 128)) {
                 painter.drawImage( 0,  0, matteImage.scaled(128, 128, Qt::KeepAspectRatio, Qt::FastTransformation)); 
             } else {
                 painter.drawImage( 0,  0, matteImage);
@@ -97,7 +97,7 @@
             QImage matteImage(QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/ffmpegthumbs-mattepaint/quadrant4");
             mirror = image.mirrored(true, true);
             QPainter painter(&mirror);
-            if ((image.height() <= 128) and (image.width() <= 128)) {
+            if ((image.height() <= 128) && (image.width() <= 128)) {
                 painter.drawImage( 0,  0, matteImage.scaled(128, 128, Qt::KeepAspectRatio, Qt::FastTransformation)); 
             } else {
                 painter.drawImage( 0,  0, matteImage);
