diff --git a/wrap/io_trimesh/import_nvm.h b/wrap/io_trimesh/import_nvm.h
index bebe7c36..f797da9d 100644
--- vcglib/wrap/io_trimesh/import_nvm.h
+++ b/wrap/io_trimesh/import_nvm.h
@@ -85,14 +85,6 @@ static bool ReadHeader(FILE *fp,unsigned int &num_cams){
     return true;
 }
 
-static bool ReadHeader(const char * filename,unsigned int &/*num_cams*/, unsigned int &/*num_points*/){
-    FILE *fp = fopen(filename, "r");
-    if(!fp) return false;
-    ReadHeader(fp);
-    fclose(fp);
-    return true;
-}
-
 
 static int Open( OpenMeshType &m, std::vector<Shot<ScalarType> >  & shots,
                  std::vector<std::string > & image_filenames,
diff --git a/wrap/io_trimesh/import_out.h b/wrap/io_trimesh/import_out.h
index 01331bc5..c2cd86fe 100644
--- vcglib/wrap/io_trimesh/import_out.h
+++ b/wrap/io_trimesh/import_out.h
@@ -85,14 +85,6 @@ static bool ReadHeader(FILE *fp,unsigned int &num_cams, unsigned int &num_points
     return true;
 }
 
-static bool ReadHeader(const char * filename,unsigned int &/*num_cams*/, unsigned int &/*num_points*/){
-    FILE *fp = fopen(filename, "r");
-    if(!fp) return false;
-    ReadHeader(fp);
-    fclose(fp);
-    return true;
-}
-
 
 static int Open( OpenMeshType &m, std::vector<Shot<ScalarType> >  & shots,
                  std::vector<std::string > & image_filenames,
