--- configure.ac	2018-08-17 13:18:28.010088365 +0200
+++ configure.ac.new	2018-08-17 18:14:06.049959957 +0200
@@ -6,6 +6,7 @@
 m4_define([abi_version], [abi_version_major.abi_version_minor.abi_version_micro])
 
 AC_INIT([abiword],[abi_version],[http://www.abisource.com/])
+m4_pattern_allow([AC_MSG_ERROR])
 AC_CANONICAL_HOST
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
@@ -561,61 +562,6 @@
 fi
 AC_SUBST(ABIWORD_ICONDIR, "$abi_cv_icondir")
 
-dnl ***********************************************************************
-dnl GCC/Clang sanitizer support
-dnl ***********************************************************************
-AC_ARG_WITH([sanitizer],
-            [AS_HELP_STRING([--with-sanitizer=@<:@address/undefined/no@:>@],
-                            [Use specific GCC/Clang analyzer])],
-            [with_sanitizer=$withval],
-            [with_sanitizer=no])
-AS_IF([test "x$with_sanitizer" != "xno"],[
-	AX_CHECK_COMPILE_FLAG([-fsanitize=$withval],
-	                      [CFLAGS="$CFLAGS -fsanitize=$withval"
-			       CXXFLAGS="$CXXFLAGS -fsanitize=$withval"],
-	                      [AC_MSG_ERROR([-fsanitize=$withval is not supported])])
-        AX_APPEND_COMPILE_FLAGS([-fno-omit-frame-pointer])
-        AC_LANG_PUSH(C++)
-        AX_APPEND_COMPILE_FLAGS([-fno-omit-frame-pointer])
-        AC_LANG_POP
-])
-
-dnl ***************************************************************
-dnl Excessive warnings
-dnl ***************************************************************
-AC_LANG_PUSH(C++)
-AX_CHECK_COMPILE_FLAG([-Werror=unknown-warning-option], [
-	ax_compiler_flags_test="-Werror=unknown-warning-option"
-], [
-	ax_compiler_flags_test=""
-])
-dnl XXX enable these
-dnl -Wformat-nonliteral
-AX_APPEND_COMPILE_FLAGS([ \
-   -Wcast-align \
-   -Wchar-subscripts \
-   -Wconst-qual \
-   -Wextra \
-   -Wformat \
-   -Wformat-security \
-   -Wformat-overflow=2 \
-   -Wheader-guard \
-   -Wimplicit-fallthrough=3 \
-   -Wlogical-not-parentheses \
-   -Wmisleading-indentation \
-   -Wmissing-noreturn \
-   -Wno-overloaded-virtual \
-   -Wpointer-arith \
-   -Wpointer-bool-conversion \
-   -Wredundant-decls \
-   -Wshadow \
-   -Wsign-compare \
-   -Wunreachable-code \
-   -Wunused \
-   -Wwrite-strings \
-], [], [$ax_compiler_flags_test])
-AC_LANG_POP
-
 #
 # Dependency tests
 #
