--- a/python/pyqt/defs.bzl
+++ b/python/pyqt/defs.bzl
@@ -1,7 +1,7 @@
 # based off https://github.com/ali5h/rules_pip/blob/master/defs.bzl

 def _execute(repository_ctx, arguments, quiet = False):
-    return repository_ctx.execute(arguments, environment = {}, quiet = quiet)
+    return repository_ctx.execute(arguments, environment = {}, quiet = quiet, timeout=6000)

 def _install_pyqt_impl(repository_ctx):
     python_interpreter = repository_ctx.attr.python_interpreter
