diff -x .sh -aur radium-5.6.5.pristine/bin/packages/build.sh radium-5.6.5.new/bin/packages/build.sh
--- radium-5.6.5.pristine/bin/packages/build.sh	2018-05-09 21:55:12.740084438 +0100
+++ radium-5.6.5.new/bin/packages/build.sh	2018-05-09 23:14:07.360090134 +0100
@@ -97,6 +97,7 @@
 tar xvzf libpd-master.tar.gz
 cd libpd-master/
 make clean
+sed -i '/define CFLAGS/ s|")| -I/usr/include/tirpc ")|' make.scm
 make -j  `nproc`
 cd ..
 
