diff -aur src.orig/spivak-1.4/src/karaokepainter.cpp src/spivak-1.4/src/karaokepainter.cpp
--- src.orig/spivak-1.4/src/karaokepainter.cpp	2020-10-26 14:47:46.513213500 +0100
+++ src/spivak-1.4/src/karaokepainter.cpp	2020-10-26 14:49:17.683212991 +0100
@@ -16,6 +16,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  **************************************************************************/
 
+#include <QPainterPath>
 #include "karaokepainter.h"
 #include "settings.h"
 
@@ -66,7 +67,7 @@
 int KaraokePainter::largestFontSize(const QFont &font, int maxsize, int width, const QString &textline, int maxheight )
 {
     int minsize = 8;
-    int cursize;
+    int cursize = 10;
     QFont testfont( font );
 
     // We are trying to find the maximum font size which fits by doing the binary search
