diff --git a/scripts/ingen.py b/scripts/ingen.py
index 594a7c01..9e757504 100644
--- a/scripts/ingen.py
+++ b/scripts/ingen.py
@@ -99,7 +99,8 @@ class Remote(Interface):
         self.ns_manager  = rdflib.namespace.NamespaceManager(self.model)
         self.ns_manager.bind('server', self.server_base)
         for (k, v) in NS.__dict__.items():
-            self.ns_manager.bind(k, v)
+            if not k.startswith('__'):
+                self.ns_manager.bind(k, v)
         if uri.startswith('unix://'):
             self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
             self.sock.connect(uri[len('unix://'):])
diff --git a/scripts/ingenish b/scripts/ingenish
index 97640645..b0911021 100755
--- a/scripts/ingenish
+++ b/scripts/ingenish
@@ -76,7 +76,8 @@ def run(cmd):
     elif cmd[0] == 'exit':
         sys.exit(0)
     elif cmd[0] == 'get' and len(cmd) == 2:
-        print(ingen.get(cmd[1]).serialize(format='n3'))
+        print(ingen.get(cmd[1]).serialize(
+            format='n3', encoding='utf-8').decode('utf-8'))
     elif cmd[0] == 'put' and len(cmd) == 3:
         return ingen.put(cmd[1], cmd[2])
     elif cmd[0] == 'patch' and len(cmd) == 4:
