diff --git a/vhdl_lang/Cargo.toml b/vhdl_lang/Cargo.toml
index 166724f..4dbbff8 100644
--- a/vhdl_lang/Cargo.toml
+++ b/vhdl_lang/Cargo.toml
@@ -33,3 +33,4 @@ assert_matches = "^1"
 [features]
 default = []
 packaged = []
+installed = []
diff --git a/vhdl_lang/src/config.rs b/vhdl_lang/src/config.rs
index be9076a..6b61ed3 100644
--- a/vhdl_lang/src/config.rs
+++ b/vhdl_lang/src/config.rs
@@ -215,7 +215,9 @@ impl Config {
         let config_relative_path = if cfg!(feature = "packaged") {
             // relative to packaged bin folder
             "../vhdl_libraries"
-        } else {
+        } else if cfg!(feature = "installed") {
+            "/usr/lib/rust_hdl/vhdl_libraries"
+        }else {
             // relative to target/debug or target/release
             "../../vhdl_libraries"
         };
diff --git a/vhdl_ls/Cargo.toml b/vhdl_ls/Cargo.toml
index feb906a..d98db65 100644
--- a/vhdl_ls/Cargo.toml
+++ b/vhdl_ls/Cargo.toml
@@ -30,4 +30,5 @@ pretty_assertions = "^0"
 
 [features]
 default = []
-packaged = ["vhdl_lang/packaged"]
\ No newline at end of file
+packaged = ["vhdl_lang/packaged"]
+installed = ["vhdl_lang/installed"]
