diff -urN fontMath-0.8.1-orig/setup.py fontMath-0.8.1/setup.py
--- fontMath-0.8.1-orig/setup.py	2021-07-19 18:10:45.934845200 -0400
+++ fontMath-0.8.1/setup.py	2021-07-19 18:10:57.344850599 -0400
@@ -27,7 +27,6 @@
         "write_to": 'Lib/fontMath/_version.py',
         "write_to_template": '__version__ = "{version}"',
     },
-    setup_requires=pytest_runner + wheel + ['setuptools_scm'],
     tests_require=[
         'pytest>=3.0.3',
     ],
