diff --git a/sources/mainapp/mainapp.pro b/sources/mainapp/mainapp.pro
index 823902f..8ffc5ad 100644
--- a/sources/mainapp/mainapp.pro
+++ b/sources/mainapp/mainapp.pro
@@ -33,9 +33,8 @@ unix:!macx {
 
 	QMAKE_LFLAGS += -Wl,-rpath=.
 
-	INSTALLDIR = /tmp/ubpm.AppDir
+	INSTALLDIR = /opt/ubpm
 
-	!system(command -v linuxdeployqt-continuous-x86_64.AppImage >/dev/null) { warning("linuxdeployqt (https://github.com/probonopd/linuxdeployqt/releases) missing, cant't build package!") }
 	!exists($$[QT_INSTALL_PLUGINS]/sqldrivers/libqsqlcipher.so) { warning("SQLCipher (https://github.com/sjemens/qsqlcipher-qt5) missing, encryption not available!") }
 	exists($$INSTALLDIR) { system(rm -r $$INSTALLDIR) }
 
@@ -67,7 +66,7 @@ equals(QT_MAJOR_VERSION, 5) {
 				export VERSION=$$VERSION &&\
 				linuxdeployqt-continuous-x86_64.AppImage $$INSTALLDIR/ubpm.desktop -appimage -no-copy-copyright-files -no-translations &&\
 				mv Universal_Blood_Pressure_Manager-*-x86_64.AppImage ubpm-$$VERSION-qt$$QT_MAJOR_VERSION-x86_64.AppImage
-	INSTALLS += target translations themes devices help ssl lin
+	INSTALLS += target translations themes devices help
 }
 
 win32 {
