diff -rupN ipbt-20220403.d4e7fcd/unix/ipbt.c ipbt-20220403.d4e7fcdb/unix/ipbt.c
--- ipbt-20220403.d4e7fcd/unix/ipbt.c	2022-04-02 21:10:02.000000000 -0400
+++ ipbt-20220403.d4e7fcdb/unix/ipbt.c	2022-04-05 20:35:38.918153147 -0400
@@ -16,18 +16,7 @@
 
 #define _XOPEN_SOURCE_EXTENDED
 
-#if CURSES_HAVE_NCURSES_CURSES_H
-#  include <ncursesw/curses.h>
-#elif CURSES_HAVE_NCURSES_CURSES_H
-#  include <ncurses/curses.h>
-#elif CURSES_HAVE_NCURSES_H
-#  include <ncurses.h>
-#elif CURSES_HAVE_CURSES_H
-#  include <curses.h>
-#else
-#  error "SysV or X/Open-compatible Curses header file required"
-#endif
-
+#include <ncurses.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
