diff -Naupr ipt-netflow-2.5/configure ipt-netflow/configure
--- ipt-netflow-2.5/configure	2020-04-24 14:05:11.000000000 +0700
+++ ipt-netflow/configure	2020-06-18 15:45:20.866195767 +0700
@@ -641,6 +641,8 @@ iptables_modules	#IPTLIB
 snmp_check
 dkms_check
 
+rm -f compat_def.h
+
 REPLACE="\
 s!@CARGS@!$CARGS!;\
 s!@KVERSION@!$KVERSION!;\
diff -Naupr ipt-netflow-2.5/ipt_NETFLOW.c ipt-netflow/ipt_NETFLOW.c
--- ipt-netflow-2.5/ipt_NETFLOW.c	2020-04-24 14:05:11.000000000 +0700
+++ ipt-netflow/ipt_NETFLOW.c	2020-06-18 15:45:20.866195767 +0700
@@ -3515,11 +3515,13 @@ static inline __u8 hook2dir(const __u8 h
 }
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,7,0)
 static inline void put_unaligned_be24(u32 val, unsigned char *p)
 {
 	*p++ = val >> 16;
 	put_unaligned_be16(val, p);
 }
+#endif
 
 static struct {
 	s64		ms;	 /* this much abs milliseconds */
