diff --git a/src/bsdfs/SConscript b/src/bsdfs/SConscript
index 3b5b724c..42f38f9b 100644
--- a/src/bsdfs/SConscript
+++ b/src/bsdfs/SConscript
@@ -31,6 +31,7 @@ plugins += env.SharedLibrary('thindielectric', ['thindielectric.cpp'])
 # The Irawan-Marschner plugin uses a Boost::Spirit parser, which makes it
 # pretty heavy stuff to compile. Go easy on the compiler flags:
 irawanEnv = env.Clone()
+irawanEnv.RemoveFlags(['-std=gnu\+\+11'])
 irawanEnv.RelaxCompilerSettings()
 plugins += irawanEnv.SharedLibrary('irawan', ['irawan.cpp'])
 
