=== modified file 'midori/midori-browser.c'
--- midori/midori-browser.c	2016-01-23 19:21:31 +0000
+++ midori/midori-browser.c	2017-03-04 11:02:54 +0000
@@ -4026,7 +4026,7 @@
     {
         const gchar* keywords = NULL;
         const gchar* search_uri = NULL;
-        KatzeItem* item;
+        KatzeItem* item = NULL;
 
         /* Do we have a keyword and a string? */
         if (browser->search_engines

