diff -Naur kcrap-0.2.3-a/server/modules/ntlm.c kcrap-0.2.3-b/server/modules/ntlm.c
--- kcrap-0.2.3-a/server/modules/ntlm.c	2018-07-23 21:20:58.255936921 -0400
+++ kcrap-0.2.3-b/server/modules/ntlm.c	2018-07-23 21:22:09.645708232 -0400
@@ -19,7 +19,7 @@
 
 static krb5_error_code lanman_hash1(unsigned char *pw, unsigned char *chal, unsigned char *resp) {
     DES_cblock key8;
-    des_key_schedule ks;
+    DES_key_schedule ks;
 
     key8[0] = pw[0] & 0xFE;
     key8[1] = ((pw[0]<<7 &0x80)|(pw[1]>>1 &0x7E));
diff -Naur kcrap-0.2.3-a/server/modules/ntlmv2.c kcrap-0.2.3-b/server/modules/ntlmv2.c
--- kcrap-0.2.3-a/server/modules/ntlmv2.c	2018-07-23 21:20:58.255936921 -0400
+++ kcrap-0.2.3-b/server/modules/ntlmv2.c	2018-07-23 21:27:57.204598767 -0400
@@ -21,7 +21,7 @@
     int retval;
     int nkeys;
     struct keyblocks keyblocks[5];
-    HMAC_CTX ctx;
+    HMAC_CTX *ctx = HMAC_CTX_new();
     unsigned char *tmpbuf;
     int i, j;
     unsigned char v2hash[EVP_MAX_MD_SIZE];
@@ -84,11 +84,11 @@
 	dump_data(v2hash, v2hlen);
 #endif
 
-	HMAC_Init(&ctx, v2hash, v2hlen, EVP_md5());
-	HMAC_Update(&ctx, (unsigned char*)req->server_challenge.data, req->server_challenge.length);
-	HMAC_Update(&ctx, (unsigned char*)req->client_challenge.data, req->client_challenge.length);
-	HMAC_Final(&ctx, resp, &resplen);
-	HMAC_cleanup(&ctx);
+	HMAC_CTX_reset(ctx);
+	HMAC_Update(ctx, (unsigned char*)req->server_challenge.data, req->server_challenge.length);
+	HMAC_Update(ctx, (unsigned char*)req->client_challenge.data, req->client_challenge.length);
+	HMAC_Final(ctx, resp, &resplen);
+	HMAC_CTX_free(ctx);
     
 	if (resplen != 16) {
 	    if (tmpbuf) free(tmpbuf);
diff -Naur kcrap-0.2.3-a/server/modules/ntlmv2s.c kcrap-0.2.3-b/server/modules/ntlmv2s.c
--- kcrap-0.2.3-a/server/modules/ntlmv2s.c	2018-07-23 21:20:58.255936921 -0400
+++ kcrap-0.2.3-b/server/modules/ntlmv2s.c	2018-07-23 21:22:36.145623418 -0400
@@ -20,7 +20,7 @@
 
 static krb5_error_code lanman_hash1(unsigned char *pw, unsigned char *chal, unsigned char *resp) {
     DES_cblock key8;
-    des_key_schedule ks;
+    DES_key_schedule ks;
      
     key8[0] = pw[0] & 0xFE;
     key8[1] = ((pw[0]<<7 &0x80)|(pw[1]>>1 &0x7E));
