diff -Naur kcrap-0.2.3/lib/Makefile.in kcrap-0.2.3-patched/lib/Makefile.in
--- kcrap-0.2.3/lib/Makefile.in	2009-08-27 18:08:41.000000000 -0400
+++ kcrap-0.2.3-patched/lib/Makefile.in	2011-11-09 15:22:42.000000000 -0500
@@ -18,7 +18,7 @@
 all: libkcrap.@SOEXT@ libkcrap.a
 
 libkcrap.$(SOEXT): $(OBJS)
-@MKSO@	$(LDSH) $(LDSHFLAGS) -soname=libkcrap.$(SOEXTVER) -o libkcrap.$(SOEXTVER) $(OBJS) $(LDFLAGS) $(LIBS)
+@MKSO@	$(LDSH) $(LDSHFLAGS) -o libkcrap.$(SOEXTVER) $(OBJS) $(LDFLAGS) $(LIBS)
 @MKSO@	-[ "$(SOEXT)" != "$(SOEXTVER)" ] && \
 @MKSO@		rm -f libkcrap.$(SOEXT) && \
 @MKSO@		ln -s libkcrap.$(SOEXTVER) libkcrap.$(SOEXT)
