diff --git a/src/gst-plugins/kmselements.c b/src/gst-plugins/kmselements.c
index 51283aa..8d3f00f 100644
--- a/src/gst-plugins/kmselements.c
+++ b/src/gst-plugins/kmselements.c
@@ -65,7 +65,7 @@ kurento_init (GstPlugin * kurento)
 
 GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
     GST_VERSION_MINOR,
-    kmselements,
+    kmselementsplugins,
     "Kurento elements",
     kurento_init, VERSION, GST_LICENSE_UNKNOWN, "Kurento Elements",
     "http://kurento.com/")
