diff -Naurp a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2019-06-05 00:24:38.000000000 +0000
+++ b/tests/Makefile.am	2019-06-05 00:25:15.735223946 +0000
@@ -1,6 +1,5 @@
 
 TESTS = $(check_PROGRAMS) \
-    test_external_symbols.sh \
     test_gop.sh \
     test_interlace.sh \
     test_intra.sh \
@@ -14,7 +13,6 @@ TESTS = $(check_PROGRAMS) \
     test_weird_shapes.sh
 
 EXTRA_DIST = \
-    test_external_symbols.sh \
     test_gop.sh \
     test_interlace.sh \
     test_intra.sh \
