--- src/ossim-1.8.18/ossim_package_support/cmake/CMakeLists.txt 2015-08-13 09:17:39.811546639 +0000
+++ src/ossim-1.8.18/ossim_package_support/cmake/CMakeLists.txt 2015-08-13 09:09:28.608232907 +0000
@@ -257,2 +257,2 @@
-  INSTALL(FILES ${${PROJECT_NAME}_BINARY_DIR}/packaging/pkgconfig/${PKGCONFIG_FILE}.pc DESTINATION lib${LIBSUFFIX}/pkgconfig COMPONENT ossim)
+  INSTALL(FILES ${${PROJECT_NAME}_BINARY_DIR}/packaging/pkgconfig/${PKGCONFIG_FILE}.pc DESTINATION lib/pkgconfig COMPONENT ossim)
