diff -ur cinelerra-4.4/quicktime/thirdparty/ffmpeg-0.6.1/libavcodec/vaapi_mpeg4.c cinelerra-4.4new/quicktime/thirdparty/ffmpeg-0.6.1/libavcodec/vaapi_mpeg4.c
--- cinelerra-4.4/quicktime/thirdparty/ffmpeg-0.6.1/libavcodec/vaapi_mpeg4.c	2010-03-30 16:30:55.000000000 -0700
+++ cinelerra-4.4new/quicktime/thirdparty/ffmpeg-0.6.1/libavcodec/vaapi_mpeg4.c	2013-06-18 09:02:32.953720213 -0700
@@ -20,6 +20,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "h263.h"
 #include "vaapi_internal.h"
 
 /** Reconstruct bitstream intra_dc_vlc_thr */
