diff -rauN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2025-11-24 14:45:17.604862891 +0100
+++ b/Makefile.am	2025-11-24 14:45:55.975047912 +0100
@@ -1114,7 +1114,7 @@
 bsdtar_DEPENDENCIES= libarchive.la libarchive_fe.la
 
 if STATIC_BSDTAR
-bsdtar_ldstatic= -static
+bsdtar_ldstatic= -all-static
 bsdtar_ccstatic= -DLIBARCHIVE_STATIC
 else
 bsdtar_ldstatic=
@@ -1287,7 +1287,7 @@
 
 
 if STATIC_BSDCPIO
-bsdcpio_ldstatic= -static
+bsdcpio_ldstatic= -all-static
 bsdcpio_ccstatic= -DLIBARCHIVE_STATIC
 else
 bsdcpio_ldstatic=
@@ -1442,7 +1442,7 @@
 
 
 if STATIC_BSDCAT
-bsdcat_ldstatic= -static
+bsdcat_ldstatic= -all-static
 bsdcat_ccstatic= -DLIBARCHIVE_STATIC
 else
 bsdcat_ldstatic=
@@ -1551,7 +1551,7 @@
 
 
 if STATIC_BSDUNZIP
-bsdunzip_ldstatic= -static
+bsdunzip_ldstatic= -all-static
 bsdunzip_ccstatic= -DLIBARCHIVE_STATIC
 else
 bsdunzip_ldstatic=
