--- installpanda.py.orig	2020-12-25 15:36:36.061173319 +0100
+++ installpanda.py	2020-12-25 15:37:11.540946811 +0100
@@ -119,7 +119,7 @@
     something like "lib" or "lib64" or in some cases, something
     similar to "lib/x86_64-linux-gnu". """
 
-    if sys.platform in ("darwin", "win32", "cygwin"):
+    if sys.platform in ("darwin", "win32", "cygwin", "linux"):
         return "lib"
 
     # This one's a bit tricky.  Some systems require us to install
