diff -ru a/ext/libhidapi/configure.ac b/ext/libhidapi/configure.ac
--- a/ext/libhidapi/configure.ac	2020-12-08 11:58:13.000000000 +0100
+++ b/ext/libhidapi/configure.ac	2021-01-19 18:31:21.034233203 +0100
@@ -18,7 +18,6 @@
 lt_age="0"
 LTLDFLAGS="-version-info ${lt_current}:${lt_revision}:${lt_age}"
 
-AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([foreign -Wall -Werror])
 AC_CONFIG_MACRO_DIR([m4])

