diff --unified --recursive --text package.orig/valgrind/mpi/libmpiwrap.c package.new/valgrind/mpi/libmpiwrap.c
--- package.orig/valgrind/mpi/libmpiwrap.c	2019-03-16 08:18:40.172019645 +0200
+++ package.new/valgrind/mpi/libmpiwrap.c	2019-03-16 08:13:54.000000000 +0200
@@ -278,8 +278,12 @@
    else if (ty == MPI_LONG_INT)       fprintf(f,"LONG_INT");
    else if (ty == MPI_SHORT_INT)      fprintf(f,"SHORT_INT");
    else if (ty == MPI_2INT)           fprintf(f,"2INT");
+#  if defined(MPI_UB)
    else if (ty == MPI_UB)             fprintf(f,"UB");
+#  endif
+#  if defined(MPI_LB)
    else if (ty == MPI_LB)             fprintf(f,"LB");
+#  endif
 #  if defined(MPI_WCHAR)
    else if (ty == MPI_WCHAR)          fprintf(f,"WCHAR");
 #  endif
@@ -733,8 +737,10 @@
          f(base + offsetof(Ty,loc), sizeof(int));
          return;
       }
+#     if defined(MPI_LB) && defined(MPI_UB)
       if (ty == MPI_LB || ty == MPI_UB)
          return; /* have zero size, so nothing needs to be done */
+#     endif
       goto unhandled;
       /*NOTREACHED*/
    }
