diff -aur scangearmp-source-1.80-1.old/scangearmp/src/scanfile.c scangearmp-source-1.80-1.new/scangearmp/src/scanfile.c
--- scangearmp-source-1.80-1.old/scangearmp/src/scanfile.c	2011-02-14 07:19:36.000000000 +0100
+++ scangearmp-source-1.80-1.new/scangearmp/src/scanfile.c	2012-03-02 22:20:03.013463968 +0100
@@ -34,6 +34,7 @@
 //#include <sys/types.h>
 //#include <unistd.h>
 #include <limits.h>
+#include <zlib.h>
 
 #include "cnmsfunc.h"
 #include "png.h"
@@ -386,7 +387,7 @@
 
 void write_data_for_png( png_structp png_ptr, png_bytep data, png_size_t length )
 {
-	FileControlWriteFile( *( (int *)png_ptr->io_ptr ), (CNMSLPSTR)data, length );
+	FileControlWriteFile( *( (int *)png_get_io_ptr(png_ptr)), (CNMSLPSTR)data, length );
 
 	return;
 }
