diff -ura libseekthermal.orig/examples/seek_viewer.cpp libseekthermal.new/examples/seek_viewer.cpp
--- libseekthermal.orig/examples/seek_viewer.cpp	2019-01-20 15:52:50.116639260 -0500
+++ libseekthermal.new/examples/seek_viewer.cpp	2019-01-20 16:52:42.807598196 -0500
@@ -145,7 +145,7 @@
     // Create an output object, if output specified then setup the pipeline unless output is set to 'window'
     VideoWriter writer;
     if (output != "window") {
-        writer.open(output, CV_FOURCC('F', 'M', 'P', '4'), fps, Size(outframe.cols, outframe.rows));
+        writer.open(output, VideoWriter::fourcc('F', 'M', 'P', '4'), fps, Size(outframe.cols, outframe.rows));
         if (!writer.isOpened()) {
             std::cerr << "Error can't create video writer" << std::endl;
             return 1;
