From 6444109a12480cc40d6169111ea6491332d754cb Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Sun, 3 Sep 2017 09:54:27 +0200
Subject: [PATCH 1/2] [cmake] install pkgconfig file

---
 CMakeLists.txt | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eea9be1..6644816 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,4 @@
 # TODO:
-# 1. pkgconfig file not yet built.
 # 2. installation not tested.
 # 3. Problems generating html pages.
 # 4. create usable libtool .la file
@@ -124,7 +123,14 @@ ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(regress)
 ADD_SUBDIRECTORY(examples)
 
-# TODO: pkgconfig file
+# pkgconfig file
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${prefix})
+SET(libdir ${prefix}/lib)
+SET(includedir ${prefix}/lib)
+SET(LIBS -lz)
+CONFIGURE_FILE(libzip.pc.in libzip.pc @ONLY)
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libzip.pc DESTINATION lib/pkgconfig)
 
 # write out config file
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/cmake-config.h.in

From fdbe7ce30a5ee31b7e520d5cf1a6b19db7ffd962 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Sun, 3 Sep 2017 09:57:06 +0200
Subject: [PATCH 2/2] [cmake] install man pages in share/man

---
 man/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/man/CMakeLists.txt b/man/CMakeLists.txt
index 4934f8b..d43c10f 100644
--- a/man/CMakeLists.txt
+++ b/man/CMakeLists.txt
@@ -105,7 +105,7 @@ ENDif(CMAKE_SYSTEM_NAME MATCHES BSD)
 
 # installation
 
-SET(MAN_PATH man)
+SET(MAN_PATH share/man)
 
 FOREACH(MAN_PAGE ${MAN1_PAGES})
   INSTALL(FILES ${MAN_PAGE}.${MANFMT} DESTINATION ${MAN_PATH}/man1
