diff --git a/lieer/gmailieer.py b/lieer/gmailieer.py
index 2a90258..8631877 100755
--- a/lieer/gmailieer.py
+++ b/lieer/gmailieer.py
@@ -278,7 +278,7 @@ class Gmailieer:
         raise NotADirectoryError("error: %s is not a valid path!" % args.path)
 
     self.dry_run          = dry_run
-    self.verbose          = args.verbose
+    self.verbose          = False
     self.HAS_TQDM         = (not args.no_progress)
     self.credentials_file = args.credentials
 
