diff -aur --no-dereference package.pristine/phpdoc package.new/phpdoc
--- package.pristine/phpdoc	2016-11-29 00:37:36.309025526 +0100
+++ package.new/phpdoc	2016-11-29 00:41:18.460676550 +0100
@@ -2,7 +2,7 @@
 if [ -z "$PHP" ]; then
    PHP=`which php`
 fi
-(exec $PHP -C -q -d output_buffering=1 "$0" "$@") 2>/dev/null
+(exec $PHP -C -q -d output_buffering=1 -d open_basedir= "$0" "$@") 2>/dev/null
 if [ "$?" -ne "0" ]; then
 	echo "FAILED:  Bad environment variable \$PHP (set to \"$PHP\")"
 	exit 1
