diff -rupN pantheon-greeter-2.0.0.orig/CMakeLists.txt pantheon-greeter-2.0.0/CMakeLists.txt
--- pantheon-greeter-2.0.0.orig/CMakeLists.txt	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/CMakeLists.txt	2015-04-14 12:43:10.943603059 +0200
@@ -8,7 +8,7 @@ list (APPEND CMAKE_MODULE_PATH ${CMAKE_S
 
 set (CONF_DIR "/etc/lightdm")
 set (DATADIR "${CMAKE_INSTALL_PREFIX}/share")
-set (PKGDATADIR "${DATADIR}/pantheon-greeter")
+set (PKGDATADIR "${DATADIR}/lightdm-pantheon-greeter")
 set (CMAKE_INSTALL_PREFIX /usr)
 set (VERSION "2.0.0")
 
@@ -62,9 +62,9 @@ add_subdirectory (po)
 
 include (GSettings)
 
-add_executable (pantheon-greeter ${VALA_C})
-target_link_libraries(pantheon-greeter m)
+add_executable (lightdm-pantheon-greeter ${VALA_C})
+target_link_libraries(lightdm-pantheon-greeter m)
 
-install (TARGETS pantheon-greeter RUNTIME DESTINATION sbin)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/pantheon-greeter.desktop DESTINATION share/xgreeters)
-install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/pantheon-greeter.conf DESTINATION ${CONF_DIR})
+install (TARGETS lightdm-pantheon-greeter RUNTIME DESTINATION bin)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/lightdm-pantheon-greeter.desktop DESTINATION share/xgreeters)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/lightdm-pantheon-greeter.conf DESTINATION ${CONF_DIR})
diff -rupN pantheon-greeter-2.0.0.orig/data/lightdm-pantheon-greeter.conf pantheon-greeter-2.0.0/data/lightdm-pantheon-greeter.conf
--- pantheon-greeter-2.0.0.orig/data/lightdm-pantheon-greeter.conf	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/data/lightdm-pantheon-greeter.conf	1970-01-01 01:00:00.000000000 +0100
@@ -0,0 +1,6 @@
+[greeter]
+#activate-numlock=true
+#default-wallpaper=/usr/share/backgrounds/elementaryos-default
+#high-contrast=false
+#onscreen-keyboard=false
+#screensaver-timeout=60
diff -rupN pantheon-greeter-2.0.0.orig/data/lightdm-pantheon-greeter.desktop pantheon-greeter-2.0.0/data/lightdm-pantheon-greeter.desktop
--- pantheon-greeter-2.0.0.orig/data/lightdm-pantheon-greeter.desktop	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/data/lightdm-pantheon-greeter.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Name=Pantheon Greeter
+Comment=Pantheon Greeter
+Exec=lightdm-pantheon-greeter
+Type=Application
+X-Ubuntu-Gettext-Domain=pantheon-greeter
+Name[en_US]=lightdm-pantheon-greeter
diff -rupN pantheon-greeter-2.0.0.orig/data/pantheon-greeter.conf pantheon-greeter-2.0.0/data/pantheon-greeter.conf
--- pantheon-greeter-2.0.0.orig/data/pantheon-greeter.conf	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/data/pantheon-greeter.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-[greeter]
-#activate-numlock=true
-#default-wallpaper=/usr/share/backgrounds/elementaryos-default
-#high-contrast=false
-#onscreen-keyboard=false
-#screensaver-timeout=60
diff -rupN pantheon-greeter-2.0.0.orig/data/pantheon-greeter.desktop pantheon-greeter-2.0.0/data/pantheon-greeter.desktop
--- pantheon-greeter-2.0.0.orig/data/pantheon-greeter.desktop	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/data/pantheon-greeter.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,7 +0,0 @@
-[Desktop Entry]
-Name=Pantheon Greeter
-Comment=Pantheon Greeter
-Exec=pantheon-greeter
-Type=Application
-X-Ubuntu-Gettext-Domain=pantheon-greeter
-Name[en_US]=pantheon-greeter
diff -rupN pantheon-greeter-2.0.0.orig/src/PantheonGreeter.vala pantheon-greeter-2.0.0/src/PantheonGreeter.vala
--- pantheon-greeter-2.0.0.orig/src/PantheonGreeter.vala	2015-04-06 23:37:23.000000000 +0200
+++ pantheon-greeter-2.0.0/src/PantheonGreeter.vala	2015-04-20 11:57:44.000000000 +0000
@@ -81,7 +81,7 @@ public class PantheonGreeter : Gtk.Windo
 
         settings = new KeyFile ();
         try {
-            settings.load_from_file (Constants.CONF_DIR+"/pantheon-greeter.conf",
+            settings.load_from_file (Constants.CONF_DIR+"/lightdm-pantheon-greeter.conf",
                     KeyFileFlags.KEEP_COMMENTS);
         } catch (Error e) {
             warning (e.message);
