--- a/src/rxvttoolkit.C
+++ b/src/rxvttoolkit.C
@@ -834,7 +834,7 @@
   // parse the nonstandard "[alphapercent]" prefix
   if (1 <= sscanf (name, "[%hd]%n", &c.a, &skip))
     {
-      c.a = lerp<int, int, int> (0, rgba::MAX_CC, c.a);
+      c.a = lerpint<int, int, int> (0, rgba::MAX_CC, c.a);
       name += skip;
     }
 
@@ -981,10 +981,10 @@
   result.set (
     screen,
     rgba (
-      lerp (c.r, to.r, percent),
-      lerp (c.g, to.g, percent),
-      lerp (c.b, to.b, percent),
-      lerp (c.a, to.a, percent)
+      lerpint (c.r, to.r, percent),
+      lerpint (c.g, to.g, percent),
+      lerpint (c.b, to.b, percent),
+      lerpint (c.a, to.a, percent)
     )
   );
 }
--- a/src/rxvtutil.h
+++ b/src/rxvtutil.h
@@ -21,7 +21,7 @@
 // linear interpolation
 template<typename T, typename U, typename P>
 static inline T
-lerp (T a, U b, P p)
+lerpint (T a, U b, P p)
 {
   return (long(a) * long(100 - p) + long(b) * long(p) + 50) / 100;
 }
