diff -aur --no-dereference package.pristine/tools/ocp-autoconf/build.ocp2 package.new/tools/ocp-autoconf/build.ocp2
--- package.pristine/tools/ocp-autoconf/build.ocp2	2017-05-05 13:13:32.000000000 +0200
+++ package.new/tools/ocp-autoconf/build.ocp2	2017-10-28 19:40:06.491675230 +0200
@@ -62,6 +62,7 @@
     "ocplib-file";
     "ocplib-config";
   ] + string_compat;
+  ocaml.cclib = [ "-ltinfo"; ];
 
    new_package("ocp-autoconf","program", ocaml);
 end
diff -aur --no-dereference package.pristine/tools/ocp-build/build.ocp2 package.new/tools/ocp-build/build.ocp2
--- package.pristine/tools/ocp-build/build.ocp2	2017-05-05 13:13:32.000000000 +0200
+++ package.new/tools/ocp-build/build.ocp2	2017-10-28 20:18:03.391482086 +0200
@@ -133,6 +133,7 @@
                      "ocp-build-oasis";
                      "ocp-build-actions";
   ];
+  ocaml.cclib = [ "-ltinfo"; ];
   OCaml.program( "ocp-build", ocaml);
 end
 
@@ -140,6 +141,7 @@
 begin
   ocaml.files = [ "projectMain.ml"  ];
   ocaml.requires = [ "ocp-build-project" ];
+  ocaml.cclib = [ "-ltinfo"; ];
   OCaml.program( "ocp-project", ocaml);
 end
 
diff -aur --no-dereference package.pristine/tools/ocp-build/lang-ocp2/build.ocp2 package.new/tools/ocp-build/lang-ocp2/build.ocp2
--- package.pristine/tools/ocp-build/lang-ocp2/build.ocp2	2017-05-05 13:13:32.000000000 +0200
+++ package.new/tools/ocp-build/lang-ocp2/build.ocp2	2017-10-28 19:38:55.652190822 +0200
@@ -27,5 +27,6 @@
    ocaml.install = false;
    ocaml.files = [ "eval.ml" ];
    ocaml.requires = [ "ocp-build-lang-ocp2" ];
+   ocaml.cclib = [ "-ltinfo"; ];
    new_package( "ocp2-eval", "program", ocaml);
 end
diff -aur --no-dereference package.pristine/tools/ocp-build/ocaml/build.ocp2 package.new/tools/ocp-build/ocaml/build.ocp2
--- package.pristine/tools/ocp-build/ocaml/build.ocp2	2017-05-05 13:13:32.000000000 +0200
+++ package.new/tools/ocp-build/ocaml/build.ocp2	2017-10-28 20:11:29.574389616 +0200
@@ -25,6 +25,7 @@
 begin
    ocaml.files = [ "metaMain.ml" ];
    ocaml.requires = [ "ocplib-ocamlfind"];
+   ocaml.cclib = [ "-ltinfo"; ];
    OCaml.program ("test-meta", ocaml);
 end
 
