diff --color --unified --recursive --text '--exclude=.git' '--exclude=generated' '--exclude=build' '--exclude=_build' OpenRV.clean/cmake/dependencies/CMakeLists.txt OpenRV/cmake/dependencies/CMakeLists.txt
--- OpenRV.clean/cmake/dependencies/CMakeLists.txt	2023-08-31 22:15:18.786591021 -0700
+++ OpenRV/cmake/dependencies/CMakeLists.txt	2023-09-01 17:45:22.500642500 -0700
@@ -58,6 +58,7 @@
 INCLUDE(ffmpeg)
 INCLUDE(doctest)
 INCLUDE(spdlog)
+INCLUDE(fmt)
 
 LIST(REMOVE_DUPLICATES RV_DEPS_LIST)
 SET(RV_DEPS_LIST
diff --color --unified --recursive --text '--exclude=.git' '--exclude=generated' '--exclude=build' '--exclude=_build' OpenRV.clean/src/bin/apps/rv/CMakeLists.txt OpenRV/src/bin/apps/rv/CMakeLists.txt
--- OpenRV.clean/src/bin/apps/rv/CMakeLists.txt	2023-08-31 22:15:18.869924353 -0700
+++ OpenRV/src/bin/apps/rv/CMakeLists.txt	2023-09-01 17:54:06.037289665 -0700
@@ -64,7 +64,7 @@
 IF(NOT RV_TARGET_WINDOWS)
   TARGET_LINK_LIBRARIES(
     ${_target}
-    PRIVATE Qt::OpenGL
+    PRIVATE Qt::OpenGL FMT::FMT
   )
 ENDIF()
 
