diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9d0501d..ac919122 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,7 @@ endif ()
 set(CUSTOM_FLAGS "${CUSTOM_FLAGS} -DQT_NO_EXCEPTIONS")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CUSTOM_FLAGS}")
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 # See: http://stackoverflow.com/a/1372836
 if (WIN32)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_WINSOCKAPI_")
