diff --git a/CMakeLists.txt b/CMakeLists.txt
index 60350bc42..5c1dc4bf3 100644
--- a/submodules/linphone/CMakeLists.txt
+++ b/submodules/linphone/CMakeLists.txt
@@ -37,7 +37,7 @@ string(REPLACE ";" " " LINPHONE_ALL_LANGS "${LINPHONE_ALL_LANGS_LIST}")
 include(CMakeDependentOption)
 
 option(ENABLE_SHARED "Build shared library." YES)
-option(ENABLE_STATIC "Build static library." YES)
+option(ENABLE_STATIC "Build static library." NO)
 
 option(ENABLE_CONSOLE_UI "Turn on or off compilation of console interface." YES)
 option(ENABLE_CSHARP_WRAPPER "Build the C# wrapper for Liblinphone." OFF)
@@ -50,20 +50,20 @@ option(ENABLE_GTK_UI "Turn on or off compilation of gtk interface." NO)
 option(ENABLE_JAVADOC "Add a target to generate documentation for Java API" NO)
 option(ENABLE_LDAP "Enable LDAP support." NO)
 option(ENABLE_NLS "Build with internationalisation support" YES)
-option(ENABLE_RELATIVE_PREFIX "Find resources relatively to the installation directory." NO)
-option(ENABLE_ROOTCA_DOWNLOAD "Download rootca.pem at build time." YES)
+option(ENABLE_RELATIVE_PREFIX "Find resources relatively to the installation directory." YES)
+option(ENABLE_ROOTCA_DOWNLOAD "Download rootca.pem at build time." NO)
 option(ENABLE_SOCI_STORAGE "Turn on compilation soci storage, for messages, contacts, history" YES)
 option(ENABLE_SQLITE_STORAGE "Turn on compilation sqlite storage, for messages, contacts, history" YES)
-option(ENABLE_STRICT "Build with strict compile options." YES)
+option(ENABLE_STRICT "Build with strict compile options." NO)
 option(ENABLE_TOOLS "Turn on or off compilation of tools." YES)
 option(ENABLE_TUNNEL "Turn on compilation of tunnel support." NO)
 option(ENABLE_TUTORIALS "Enable compilation of tutorials." YES)
-option(ENABLE_UNIT_TESTS "Enable compilation of unit tests." YES)
+option(ENABLE_UNIT_TESTS "Enable compilation of unit tests." NO)
 option(ENABLE_UPDATE_CHECK "Enable update check." NO)
 option(ENABLE_VCARD "Turn on compilation of vcard4 support." YES)
 option(ENABLE_VIDEO "Build with video support." YES)
 
-cmake_dependent_option(ENABLE_LIME "Enable Instant Messaging Encryption." YES "ENABLE_SQLITE_STORAGE" NO)
+cmake_dependent_option(ENABLE_LIME "Enable Instant Messaging Encryption." YES "ENABLE_SQLITE_STORAGE" YES)
 cmake_dependent_option(ENABLE_NOTIFY "Enable libnotify support." YES "ENABLE_GTK_UI;NOT APPLE" NO)
 cmake_dependent_option(ENABLE_ASSISTANT "Turn on assistant compiling." YES "ENABLE_GTK_UI" NO)
 cmake_dependent_option(ENABLE_SPHINX_DOC "Turn on the generation of the multi-language API documentation based on Sphinx." NO "ENABLE_DOC" NO)
