Index: Makefile.in
===================================================================
--- Makefile.in	(revision 993)
+++ Makefile.in	(working copy)
@@ -51,7 +51,7 @@
     LIBMACOSX = -framework IOKit -framework CoreFoundation
 endif
 
-CFL	= -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
+CFL	= $(CFLAGS) -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
 -DHAVE_SHM=@LIBXEXT32_OK@ -Wimplicit -Wbad-function-cast -DOSSD=@OSSHDR32@ \
 -DHAVE_X11=@X11_32_ALL_OK@ -DIA64=0 -Wnested-externs -DHAVE_OSS=@OSS32_H@ \
 @DARWIN@ @OCLPROGS32@ -DSERVER=@MAKE_SERVER@ -DHAVE_SVGALIB=@CONSOLE32_OK@ \
@@ -60,7 +60,7 @@
 -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror -ffast-math \
 -Wundef -fomit-frame-pointer -c -o
 
-CFLN	= -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
+CFLN	= $(CFLAGS) -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
 -DHAVE_SHM=@LIBXEXT32_OK@ -DHAVE_SVGALIB=@CONSOLE32_OK@ -Wimplicit \
 -DHAVE_X11=@X11_32_ALL_OK@ -DIA64=0 -DHAVE_OSS=@OSS32_H@ -DOSSD=@OSSHDR32@ \
 @DARWIN@ @OCLPROGS32@ -DSERVER=@MAKE_SERVER@ -DHAVE_CUFFT=@CUFFT32_OK@ \
@@ -69,7 +69,7 @@
 -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror -ffast-math \
 -Wundef -fomit-frame-pointer -c -o
 
-CFLOCL	= -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
+CFLOCL	= $(CFLAGS) -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
 -DHAVE_SHM=@LIBXEXT32_OK@ -Wimplicit -Wbad-function-cast -DOSSD=@OSSHDR32@ \
 -DHAVE_X11=@X11_32_ALL_OK@ -DIA64=0 -Wnested-externs -DHAVE_OSS=@OSS32_H@ \
 @DARWIN@ -DSERVER=@MAKE_SERVER@ -DHAVE_SVGALIB=@CONSOLE32_OK@ \
@@ -78,7 +78,7 @@
 -Wuninitialized -Wparentheses -Werror -ffast-math \
 -Wundef -fomit-frame-pointer -c -o
 
-CFW	= -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=0 -DHAVE_CUFFT=0 \
+CFW	= $(CFLAGS) -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=0 -DHAVE_CUFFT=0 \
 -DCPU=CPU_INTEL -DSERVER=@MAKE_SERVER@ -DHAVE_OSS=0 -DOPENCL_PRESENT=0 \
 @M32@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
@@ -86,7 +86,7 @@
 
 # -fsyntax-only -pedantic -pedantic-errors -w -Wextra -Wall -Waddress -Waggrega
 
-CFWN	= -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=0 -DHAVE_CUFFT=0 \
+CFWN	= $(CFLAGS) -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=0 -DHAVE_CUFFT=0 \
 -DCPU=CPU_INTEL -DSERVER=@MAKE_SERVER@ -DHAVE_OSS=0 -DOPENCL_PRESENT=0 \
 @M32@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wconversion \
@@ -93,13 +93,13 @@
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
 -ffast-math -Wundef -fomit-frame-pointer -c -o
 
-CFW64	= -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=1 -DHAVE_CUFFT=0 \
+CFW64	= $(CFLAGS) -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=1 -DHAVE_CUFFT=0 \
 -DCPU=CPU_INTEL -DSERVER=@MAKE_SERVER@ -DHAVE_OSS=0 -DOPENCL_PRESENT=0 \
 @M64@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
 -ffast-math -Wundef -fomit-frame-pointer -c -o
 
-CFL64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
+CFL64	= $(CFLAGS) -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
 -DCPU=@CPU@ -DHAVE_SHM=@LIBXEXT64_OK@ -DHAVE_X11=@X11_64_ALL_OK@ \
 -DIA64=1 @DARWIN@ -DSERVER=@MAKE_SERVER@ -DHAVE_SVGALIB=@CONSOLE64_OK@ \
 @OCLPROGS64@ -DHAVE_CUFFT=@CUFFT64_OK@ \
@@ -108,7 +108,7 @@
 -Wstrict-prototypes -Werror -ffast-math -Wundef -fomit-frame-pointer \
 -c -o
 
-CFLN64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
+CFLN64	= $(CFLAGS) -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
 -DCPU=@CPU@ -DHAVE_SHM=@LIBXEXT64_OK@ -DHAVE_SVGALIB=@CONSOLE64_OK@ \
 -DIA64=1 @DARWIN@ -DSERVER=@MAKE_SERVER@ -DHAVE_X11=@X11_64_ALL_OK@ \
 -DHAVE_CUFFT=@CUFFT64_OK@ -DOSSD=@OSSHDR64@ @OSSINC64@ @OCLPROGS64@ \
@@ -117,7 +117,7 @@
 -Wstrict-prototypes -Werror -ffast-math -Wundef -fomit-frame-pointer \
 -c -o
 
-CFLOCL64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
+CFLOCL64	= $(CFLAGS) -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
 -DCPU=@CPU@ -DHAVE_SHM=@LIBXEXT64_OK@ -DHAVE_X11=@X11_64_ALL_OK@ \
 -DIA64=1 @DARWIN@ -DSERVER=@MAKE_SERVER@ -DHAVE_SVGALIB=@CONSOLE64_OK@ \
 @OCLPROGS64@ -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DHAVE_CUFFT=@CUFFT64_OK@ \
@@ -1707,7 +1707,7 @@
 bladerf.ol mirics.ol sdrplay2.ol openhpsdr.ol oclprogs.ol loadalsa.ol \
 cloudiq.ol lfxsys.ol airspyhf.ol siganal_graph.ol
 
-	${CC} @M32@ -o xlinrad @ST@ \
+	${CC} $(LDFLAGS) @M32@ -o xlinrad @ST@ \
 xmain.ol xvar.ol ui.ol lsetad.ol fft1.ol buf.ol lvar.ol radar.ol \
 mouse.ol wide_graph.ol fft0.ol adtest.ol hires_graph.ol elad.ol \
 baseb_graph.ol mix2.ol afc_graph.ol hwaredriver.ol affinity.ol \
@@ -2057,7 +2057,7 @@
 oclprogs.om loadalsa.om cloudiq.om lfxsys.om loadusb.om airspyhf.om \
 siganal_graph.om
 
-	${CC} @M64@ -o xlinrad64 @ST@ \
+	${CC} $(LDFLAGS) @M64@ -o xlinrad64 @ST@ \
 xmain.om xvar.om ui.om lsetad.om fft1.om buf.om lvar.om radar.om \
 mouse.om wide_graph.om fft0.om adtest.om hires_graph.om elad.om \
 baseb_graph.om mix2.om afc_graph.om hwaredriver.om loadusb.om \
