diff --git a/include/rtw_debug.h b/include/rtw_debug.h
index bf36357..86d0728 100644
--- a/include/rtw_debug.h
+++ b/include/rtw_debug.h
@@ -227,7 +227,7 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 		if (sel == RTW_DBGDUMP)\
 			_DBG_871X_LEVEL(_drv_always_, fmt, ##arg); \
 		else {\
-			if(_seqdump(sel, fmt, ##arg)) /*rtw_warn_on(1)*/; \
+			_seqdump(sel, fmt, ##arg); /*rtw_warn_on(1)*/ \
 		} \
 	}while(0)
 
@@ -237,7 +237,7 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 		if (sel == RTW_DBGDUMP)\
 			DBG_871X_LEVEL(_drv_always_, fmt, ##arg); \
 		else {\
-			if(_seqdump(sel, fmt, ##arg)) /*rtw_warn_on(1)*/; \
+			_seqdump(sel, fmt, ##arg); /*rtw_warn_on(1)*/ \
 		} \
 	}while(0)
 
