--- r8152.c	2019-01-07 16:09:18.763539084 +0000
+++ r8152.c	2019-01-07 16:09:12.533474396 +0000
@@ -11505,8 +11505,10 @@
 
 static const struct ethtool_ops ops = {
 	.get_drvinfo = rtl8152_get_drvinfo,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,20,0)
 	.get_settings = rtl8152_get_settings,
 	.set_settings = rtl8152_set_settings,
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(4,20,0) */
 	.get_link = ethtool_op_get_link,
 	.nway_reset = rtl8152_nway_reset,
 	.get_msglevel = rtl8152_get_msglevel,
