diff -Naur nv.orig/nv.c nv/nv.c
--- nv.orig/nv.c	2016-01-20 04:11:03.974819000 +0100
+++ nv/nv.c	2016-01-20 04:11:48.095900342 +0100
@@ -574,7 +574,8 @@
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
diff -Naur nv.orig/nv-linux.h nv/nv-linux.h
--- nv.orig/nv-linux.h	2016-01-20 04:11:03.958152000 +0100
+++ nv/nv-linux.h	2016-01-20 04:11:48.095900342 +0100
@@ -143,6 +143,10 @@
 #include <linux/seq_file.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,3,0)
+#undef CONFIG_MTRR
+#endif
+
 #ifdef CONFIG_MTRR
 #include <asm/mtrr.h>
 #endif
