diff -up src/wminet-3.0.0/src/list.h.orig src/wminet-3.0.0/src/list.h
--- src/wminet-3.0.0/src/list.h.orig	2016-05-01 20:55:22.000000000 -0600
+++ src/wminet-3.0.0/src/list.h	2016-05-01 20:55:27.000000000 -0600
@@ -29,11 +29,7 @@ Boston, MA 02111-1307, USA.  */
 #ifndef __LIST_H_
 #define __LIST_H_
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-# define INLINE inline
-#else
 # define INLINE
-#endif
 
 typedef struct LinkedList {
   void *head;
