diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,10 +66,8 @@ add_subdirectory(cmake)
 add_subdirectory(ui)
 add_subdirectory(po)
 add_subdirectory(plugins)
-add_subdirectory(doc)
 add_subdirectory(tools)

-add_subdirectory(debian)

 livewallpaper_plugin_post_install()

diff --git a/plugins/circles/circles.py b/plugins/circles/circles.py
--- a/plugins/circles/circles.py
+++ b/plugins/circles/circles.py
@@ -1,3 +1,4 @@
+#!/usr/bin/python3
 #
 #
 # Livewallpaper
diff --git a/plugins/photoslide/photoslide.py b/plugins/photoslide/photoslide.py
--- a/plugins/photoslide/photoslide.py
+++ b/plugins/photoslide/photoslide.py
@@ -1,3 +1,4 @@
+#!/usr/bin/python3
 #
 #
 # Livewallpaper Photoslide Plugins
@@ -205,8 +206,8 @@ class PhotoslidePlugin(GObject.Object, LW.Wallpaper):
        if pixbuf:
            width = pixbuf.get_width()
            height = pixbuf.get_height()
-           img = Image.fromstring("RGBA", (width, height), pixbuf.get_pixels())
-           imgarr = array.array("B", img.convert("RGBA").tostring("raw", "BGRA"))
+           img = Image.frombytes("RGBA", (width, height), pixbuf.get_pixels())
+           imgarr = array.array("B", img.convert("RGBA").tobytes("raw", "BGRA"))
            imgsurface = cairo.ImageSurface.create_for_data(imgarr, cairo.FORMAT_ARGB32, width, height, width * 4)
            cr.set_source_surface(imgsurface, (self.img_width - width) / 2, (self.img_height - height) / 2)
            cr.paint()
diff --git a/ui/config/files-selector.vala b/ui/config/files-selector.vala
--- a/ui/config/files-selector.vala
+++ b/ui/config/files-selector.vala
@@ -55,7 +55,7 @@ public class FilesSelector : Box {
         if (ts.count_selected_rows () > 0) {
             TreeIter it;
             ts.get_selected (null, out it);
-            model.remove (it);
+            model.remove(ref it);
             value_changed (this.get_values());
         }
     }
