diff --git a/src/piclone.c b/src/piclone.c
index 50abaa9..253eef0 100644
--- a/src/piclone.c
+++ b/src/piclone.c
@@ -35,6 +35,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ctype.h>
 #include <dirent.h>
 
+#include <locale.h>
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
@@ -236,7 +237,7 @@ static gpointer backup_thread (gpointer data)
     FILE *fp;
 
     // get a new partition UUID
-    get_string ("uuid | cut -f1 -d-", npuuid);
+    get_string ("uuidgen | cut -f1 -d-", npuuid);
 
     // check the source has an msdos partition table
     sprintf (buffer, "parted %s unit s print | tail -n +4 | head -n 1", src_dev);
