From 6d220b0a083b3adf7fd4ae65ca011b29488493b7 Mon Sep 17 00:00:00 2001
From: jdoe0 <jdoe@mailinator.com>
Date: Wed, 18 Jul 2018 02:29:58 -0700
Subject: [PATCH] libtransport: add missing semicolon

---
 libtransport/NetworkPluginServer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libtransport/NetworkPluginServer.cpp b/libtransport/NetworkPluginServer.cpp
index bff05a6a..c4e19f70 100644
--- a/libtransport/NetworkPluginServer.cpp
+++ b/libtransport/NetworkPluginServer.cpp
@@ -1815,7 +1815,7 @@ void NetworkPluginServer::wrapIncomingImage(Swift::Message* msg, const pbnetwork
             msg->addPayload(oob_payload);
             msg->setBody(image_url);
         } else {
-            LOG4CXX_WARN(logger, "xhtml seems to contain an image, but doesn't match: " + payload.xhtml())
+            LOG4CXX_WARN(logger, "xhtml seems to contain an image, but doesn't match: " + payload.xhtml());
         }
     }
 }
