diff -uprNEBZ --suppress-blank-empty a/vokoscreen.pro b/vokoscreen.pro
--- a/vokoscreen.pro	2016-03-22 19:42:41.311844000 -0600
+++ b/vokoscreen.pro	2016-03-22 20:00:55.593940662 -0600
@@ -17,7 +17,7 @@ FORMS += vokoscreen.ui
 # Das Script/Macro siehe weiter unten "# language packages" muß weiter bestehen bleiben damit "make clean" die *.qm Dateien löscht.

 equals(QT_MAJOR_VERSION, "4") {
-  system(lrelease language/vokoscreen_*.ts)
+  system(lrelease-qt4 language/vokoscreen_*.ts)
   message("")
   message("[vokoscreen] lrelease --> qt4")
   message("")
@@ -40,7 +40,7 @@ equals(QT_MAJOR_VERSION, "4") {
 !isEmpty(TRANSLATIONS) {
   isEmpty(QMAKE_LRELEASE) {
     win32: QMAKE_LRELEASE = $$[QT_INSTALL_BINS]\lrelease.exe
-      else: QMAKE_LRELEASE = $$[QT_INSTALL_BINS]/lrelease
+	  else: QMAKE_LRELEASE = $$[QT_INSTALL_PREFIX]/bin/lrelease-qt4
   }
   isEmpty(TS_DIR):TS_DIR = language
   TSQM.name = lrelease ${QMAKE_FILE_IN}
@@ -61,7 +61,7 @@ equals(QT_MAJOR_VERSION, "5") {
 !isEmpty(TRANSLATIONS) {
   isEmpty(QMAKE_LRELEASE) {
     win32: QMAKE_LRELEASE = $$[QT_INSTALL_BINS]\lrelease.exe
-      else: QMAKE_LRELEASE = $$[QT_INSTALL_BINS]/lrelease-qt5
+      else: QMAKE_LRELEASE = $$[QT_INSTALL_PREFIX]/bin/lrelease-qt5
   }
   isEmpty(TS_DIR):TS_DIR = language
   TSQM.name = lrelease-qt5 ${QMAKE_FILE_IN}
