diff -ur src/binutils-2.29.1.orig/ld/scripttempl/i386go32.sc src/binutils-2.29.1/ld/scripttempl/i386go32.sc
--- src/binutils-2.29.1.orig/ld/scripttempl/i386go32.sc	2017-12-28 16:44:53.887247994 +0200
+++ src/binutils-2.29.1/ld/scripttempl/i386go32.sc	2017-12-28 16:54:10.917217558 +0200
@@ -84,7 +84,7 @@
     ${RELOCATING+ . = ALIGN(${SEGMENT_SIZE});}
   }
   /* Discard LTO sections.  */
-  /DISCARD/ : { *(gnu.lto_*) } 
+  /DISCARD/ : { *(.note.GNU-stack) *(.gnu_debuglink) *(gnu.lto_*) }
   /* Stabs debugging sections.  */
   .stab 0 : { *(.stab) }
   .stabstr 0 : { *(.stabstr) }
