--- a/src/qtluastate.cc	2013-03-17 16:14:27.000000000 -0700
+++ b/src/qtluastate.cc	2015-06-27 20:39:35.371045361 -0700
@@ -439,7 +439,7 @@ void State::lua_pgettable(lua_State *st,
   if (lua_type(st, index) == LUA_TTABLE)
     {
       if (!lua_getmetatable(st, index))
-	return lua_rawget(st, index);
+	return (void)lua_rawget(st, index);
       lua_pop(st, 1);
     }
 
--- a/src/qtluavaluebase.cc	2013-03-17 16:14:27.000000000 -0700
+++ b/src/qtluavaluebase.cc	2015-06-27 20:44:12.166864362 -0700
@@ -657,7 +657,7 @@ QByteArray ValueBase::to_bytecode() cons
   if (lua_type(lst, -1) == LUA_TFUNCTION)
     {
       QByteArray bytecode;
-      int status = lua_dump(lst, &lua_writer, &bytecode);	
+      int status = lua_dump(lst, &lua_writer, &bytecode, 0);	
       lua_pop(lst, 1);
       if (status)
 	QTLUA_THROW(QtLua::ValueBase, "Unable to dump function bytecode (status=%)", .arg(status));
