--- Makefile.in	2018-02-23 07:33:00.804174657 -0500
+++ Makefile.in	2018-02-23 07:33:53.433167622 -0500
@@ -34,15 +34,9 @@
 MONGOOSE_HOME=${PWD}/third-party/mongoose
 MONGOOSE_INC=-I$(MONGOOSE_HOME)
 ######
-HAS_LUAJIT=$(shell pkg-config --exists luajit; echo $$?)
-ifeq ($(HAS_LUAJIT), 0)
-	LUAJIT_INC = $(shell pkg-config --cflags luajit)
-	LUAJIT_LIB = $(shell pkg-config --libs luajit)
-else
 	LUAJIT_HOME=${PWD}/third-party/LuaJIT-2.1.0-beta3
 	LUAJIT_INC=-I$(LUAJIT_HOME)/src
 	LUAJIT_LIB=$(LUAJIT_HOME)/src/libluajit.a
-endif
 ######
 LIBRRDTOOL_HOME=${PWD}/third-party/rrdtool-1.4.8
 HAS_LIBRRDTOOL=$(shell pkg-config --atleast-version=1.4.8 librrd; echo $$?)
