Only in lswt-v1.0.1: lswt-v1.0.1.orig
diff --unified --recursive --text --color lswt-v1.0.1/Makefile lswt-v1.0.1.new/Makefile
--- lswt-v1.0.1/Makefile	2021-04-22 12:00:26.000000000 +0200
+++ lswt-v1.0.1.new/Makefile	2021-11-19 13:59:57.549487866 +0100
@@ -11,7 +11,7 @@
 GEN=wlr-foreign-toplevel-management-unstable-v1.c wlr-foreign-toplevel-management-unstable-v1.h xdg-output-unstable-v1.c xdg-output-unstable-v1.h
 
 lswt: $(OBJ)
-	$(CC)$ $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
 
 $(OBJ): $(GEN)
 
