--- Makefile
+++ Makefile
@@ -331,7 +331,7 @@
 
 ANALYZE_DEP:=ghdl_$(backend)$(EXEEXT)
 GHDL=$(PWD)/$(ANALYZE_DEP)
-ANALYZE_OPTS:=
+#ANALYZE_OPTS:=
 ANALYZE:=$(GHDL) -a $(ANALYZE_OPTS) $(LIB_CFLAGS)
 
 # TODO?: don't include, make it separate
@@ -341,7 +341,7 @@
 ifeq "$(backend)" "gcc"
   # ghdl with gcc backend is already installed, no need to specify ghdl1
   # (and we don't know where it is).
-  STD_GHDL_FLAGS=
+  #STD_GHDL_FLAGS=
 else
   # Specify ghdl1 path, as it is spawned by ghdl.
   STD_GHDL_FLAGS=--GHDL1=$(PWD)/ghdl1-$(backend)$(EXEEXT)
