--- src/pastix_5.2.2.22/src/Makefile	2015-10-04 13:20:26.331860284 +0100
+++ src/pastix_5.2.2.22/src/Makefile	2015-10-04 13:20:36.168419483 +0100
@@ -364,15 +364,15 @@
 DYLIB_OPT=${LDFLAGS} -L__LIBDIR__ -lpastix
 endif
 %/libmatrix_driver${LIB_SO}: $(OBJ_DRIVERS)
-	$(MPCCPROG) $(subst __SO_NAME__,$@, ${SHARED_FLAGS}) -o $(dir $@)libmatrix_driver${VERSIONNAME}${LIB_SO} $(^) $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT})
+	$(MPCCPROG) $(subst __SO_NAME__,$(subst $(LIBDIR),,$@), ${SHARED_FLAGS}) -o $(dir $@)libmatrix_driver${VERSIONNAME}${LIB_SO} $(^) $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT})
 	(cd $(dir $@) && ln -sf libmatrix_driver${VERSIONNAME}${LIB_SO} libmatrix_driver${LIB_SO})
 
 %/libpastix${LIB_SO}: $(OBJ_LIB)
-	$(MPCCPROG)  $(subst __SO_NAME__,$@, ${SHARED_FLAGS}) -o $(dir $@)libpastix${VERSIONNAME}${LIB_SO} $(^) $(filter-out -lpastix, $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT}))
+	$(MPCCPROG)  $(subst __SO_NAME__,$(subst $(LIBDIR),,$@), ${SHARED_FLAGS}) -o $(dir $@)libpastix${VERSIONNAME}${LIB_SO} $(^) $(filter-out -lpastix, $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT}))
 	(cd $(dir $@) && ln -sf libpastix${VERSIONNAME}${LIB_SO} libpastix${LIB_SO})
 
 %/libpastix_murge${LIB_SO}: $(OBJ_MURGE) %/libpastix${LIB}
-	$(MPCCPROG)  $(subst __SO_NAME__,$@, ${SHARED_FLAGS}) -o $(dir $@)libpastix_murge${VERSIONNAME}${LIB_SO} $(^) $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT})
+	$(MPCCPROG)  $(subst __SO_NAME__,$(subst $(LIBDIR),,$@), ${SHARED_FLAGS}) -o $(dir $@)libpastix_murge${VERSIONNAME}${LIB_SO} $(^) $(subst __LIBDIR__,$(dir $@),${DYLIB_OPT})
 	(cd $(dir $@) && ln -sf libpastix_murge${VERSIONNAME}${LIB_SO} libpastix_murge${LIB_SO})
 
 endif
@@ -418,8 +418,8 @@
 	cp utils/src/pastix-conf.sh $@
 	echo "MM${MPCXXPROG}MM"
 	cat $@								   | \
-	   sed -e s%\_LIBS\_%"-L$(LIBDIR) -lpastix ${EXTRALIB}"%	   | \
-	   sed -e s%\_INC\_%"-I$(INCLUDEDIR)"%		        	   | \
+	   sed -e s%\_LIBS\_%"-L$(subst $(ROOT),,$(LIBDIR)) -lpastix ${EXTRALIB}"% | \
+	   sed -e s%\_INC\_%"-I$(subst $(ROOT),,$(INCLUDEDIR))"%	   | \
 	   sed -e s%\_CC\_%"${MPCCPROG}"%				   | \
 	   sed -e s%\_CCOPT\_%"${CCOPT}"%				   | \
 	   sed -e s%\_CXX\_%"${MPCXXPROG}"%				   | \
