diff -Naur cacti-spine-0.8.8g/configure.ac cacti-spine-0.8.8g-patched/configure.ac
--- cacti-spine-0.8.8g/configure.ac	2016-02-15 20:21:01.000000000 +0100
+++ cacti-spine-0.8.8g-patched/configure.ac	2016-03-02 10:11:49.885091992 +0100
@@ -9,7 +9,7 @@
 AC_PREFIX_DEFAULT(/usr/local/spine)
 AC_LANG(C)
 
-AM_INIT_AUTOMAKE(spine, 0.8.8g)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config/config.h)
 
 # static libraries
@@ -269,7 +269,12 @@
 )
 AC_DEFINE_UNQUOTED(SNMP_LOCALNAME, $havelocalname, If snmp localname session structure member exists)
 
-AC_CHECK_LIB(netsnmp, snmp_timeout)
+AC_CHECK_LIB(netsnmp, snmp_timeout,
+  [ LIBS="-lnetsnmp $LIBS"
+    AC_DEFINE(USE_NET_SNMP, 1, New Net SNMP Version)
+  ],
+  AC_MSG_RESULT(Cannot find NET-SNMP libraries(snmp))
+)
 
 # ****************** Spine Result Buffer Check ***********************
 # Check for the default spine output buffer size
diff -Naur cacti-spine-0.8.8g/Makefile.am cacti-spine-0.8.8g-patched/Makefile.am
--- cacti-spine-0.8.8g/Makefile.am	2016-02-22 02:49:22.000000000 +0100
+++ cacti-spine-0.8.8g-patched/Makefile.am	2016-03-02 10:10:12.858219813 +0100
@@ -7,6 +7,8 @@
 
 bin_PROGRAMS = spine
 
+ACLOCAL_AMFLAGS = -I m4
+
 man_MANS = spine.1
 
 spine.1: spine.c 
