--- ../apertium-is-en-0.1.0/Makefile.am	2010-03-05 13:33:18.000000000 +0100
+++ apertium-is-en-0.1.0/Makefile.am	2010-09-15 13:54:17.000000000 +0200
@@ -119,10 +119,10 @@
 install-data-local:
 	apertium-gen-modes modes.xml apertium-$(PREFIX1)
 	cp *.mode modes/
-	$(INSTALL_DATA) $(PREFIX1).mode $(apertium_is_modesdir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(apertium_is_endir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t2x $(apertium_is_endir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(apertium_is_endir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t4x $(apertium_is_endir)
+	$(INSTALL_DATA) $(PREFIX1).mode $(DESTDIR)$(apertium_is_modesdir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(DESTDIR)$(apertium_is_endir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t2x $(DESTDIR)$(apertium_is_endir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(DESTDIR)$(apertium_is_endir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t4x $(DESTDIR)$(apertium_is_endir)
 
 CLEANFILES = -rf $(TARGETS_COMMON) .deps modes
