--- Makefile	2014-02-20 14:39:50.000000000 +0100
+++ Makefile	2014-03-31 15:12:30.424985665 +0200
@@ -25,12 +25,14 @@
 	rm -f *.o $(LIBNAME) $(DEPENDFILE) *Test
 
 install: $(LIBNAME)
-	sudo cp reader.conf /etc/reader.conf.d/persoSim
-	sudo cp $(LIBNAME) /usr/lib/pcsc/drivers/serial/
+	mkdir -p $(PREFIX)/etc/reader.conf.d/
+	cp reader.conf $(PREFIX)/etc/reader.conf.d/persoSim
+	mkdir -p $(PREFIX)/usr/lib/pcsc/drivers/serial/
+	cp $(LIBNAME) $(PREFIX)/usr/lib/pcsc/drivers/serial/
 
 uninstall: 
-	sudo rm -f /etc/reader.conf.d/persoSim
-	sudo rm -f /usr/lib/pcsc/drivers/serial/$(LIBNAME)
+	rm -f $(PREFIX)/etc/reader.conf.d/persoSim
+	rm -f $(PREFIX)/usr/lib/pcsc/drivers/serial/$(LIBNAME)
 
 test: hexStringTest
 	./hexStringTest
