diff --git a/Makefile b/Makefile
index 3b06f103..60bb737f 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ HFILES = sipAPIoqapy_iproc.h
 CC = gcc
 CXX = g++
 LINK = g++
-CPPFLAGS = -DNDEBUG -I. -I/usr/include/python2.7
+CPPFLAGS = -DNDEBUG -I. -I/usr/include/python2.7 -I../src
 CFLAGS = -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fPIC -O2 -Wall -W
 CXXFLAGS = -pipe -fPIC -O2 -Wall -W
 LFLAGS = -Wl,-O1,--sort-common,--as-needed,-z,relro -shared -Wl,-rpath,/usr/lib -Wl,-rpath,/usr/lib/x86_64-linux-gnu/ -Wl,--version-script=oqapy_iproc.exp
