*** Makefile.cvs	2020-12-01 20:09:51.767207562 +0000
--- Makefile.cvs	2020-12-01 20:10:49.539725687 +0000
***************
*** 8,14 ****
  	./bootstrap.sh ; \
  	mkdir build ; \
  	cd build ; \
! 	cmake -DCMAKE_BUILD_TYPE=RELEASE ..
  
  install: configure
  	cd build ; \
--- 8,14 ----
  	./bootstrap.sh ; \
  	mkdir build ; \
  	cd build ; \
! 	cmake -DCMAKE_BUILD_TYPE=RELEASE $(EXTRA_PARAMS) ..
  
  install: configure
  	cd build ; \
