--- Makefile.install.directories.orig	2011-12-18 19:19:39.677451914 +0100
+++ Makefile.install.directories	2011-12-18 19:20:41.820782593 +0100
@@ -4,13 +4,13 @@
 # the different directories for the files
 
 # installing in /usr/local
-MAIN_DIRECTORY         = /usr/local
-BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
-DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
-HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
-MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
-ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
-DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+#MAIN_DIRECTORY         = /usr/local
+#BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
+#DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
+#HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+#MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+#ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
+#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 # debian
 #MAIN_DIRECTORY         = /usr
@@ -48,14 +48,14 @@
 #ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps/
 #DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
-#arch linux
-#MAIN_DIRECTORY         = /usr
-#BIN_DIRECTORY = $(MAIN_DIRECTORY)/bin
-#DATA_DIRECTORY = $(MAIN_DIRECTORY)/share/freedoko
-#HELP_DIRECTORY = $(MAIN_DIRECTORY)/share/doc/freedoko
-#MAN_DIRECTORY = $(MAIN_DIRECTORY)/share/man/man6
-#ICON_DIRECTORY = $(MAIN_DIRECTORY)/share/pixmaps/
-#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+#arch linux
+MAIN_DIRECTORY         = /usr
+BIN_DIRECTORY = $(MAIN_DIRECTORY)/bin
+DATA_DIRECTORY = $(MAIN_DIRECTORY)/share/freedoko
+HELP_DIRECTORY = $(MAIN_DIRECTORY)/share/doc/freedoko
+MAN_DIRECTORY = $(MAIN_DIRECTORY)/share/man/man6
+ICON_DIRECTORY = $(MAIN_DIRECTORY)/share/pixmaps/
+DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 
 # release
