diff -aur samtools-1.2/Makefile samtools-1.2-patched/Makefile
--- samtools-1.2/Makefile	2015-02-02 16:18:41.000000000 +0100
+++ samtools-1.2-patched/Makefile	2015-06-30 15:48:41.006563956 +0200
@@ -83,7 +83,9 @@
 
 # Adjust $(HTSDIR) to point to your top-level htslib directory
 HTSDIR = ../htslib
+ifneq ("$(wildcard $(HTSDIR)/htslib.mk)","")
 include $(HTSDIR)/htslib.mk
+endif
 HTSLIB = $(HTSDIR)/libhts.a
 BGZIP  = $(HTSDIR)/bgzip
 
