--- pachi-pachi-12.82/Makefile	2023-07-05 04:25:05.000000000 +0200
+++ pachi-pachi-12.82.new/Makefile	2023-08-18 23:50:06.538581419 +0200
@@ -207,6 +207,7 @@
 
 ifeq ($(DCNN_DETLEF), 1)
 	COMMON_FLAGS += -DDCNN_DETLEF
+	EXTRA_DATAFILES += detlef54.prototxt detlef54.trained
 endif
 
 ifeq ($(DCNN_DARKFOREST), 1)
@@ -284,7 +285,7 @@
 
 # Low-level dependencies last
 SUBDIRS   = $(EXTRA_SUBDIRS) pattern joseki uct uct/policy t-unit t-predict engines playout tactics
-DATAFILES = $(EXTRA_DATAFILES) detlef54.prototxt detlef54.trained joseki19.gtp opening.dat patterns_mm.gamma patterns_mm.spat 
+DATAFILES = $(EXTRA_DATAFILES) joseki19.gtp patterns_mm.gamma patterns_mm.spat 
 
 
 ############################################################################################################
@@ -310,7 +311,7 @@
 	@CC="$(CC)" CFLAGS="$(CFLAGS)" ./spudfrog
 
 # Build info
-build.h: .git/HEAD .git/index Makefile
+build.h: Makefile
 	+@make spudfrog
 	@echo "[make] build.h"
 	@CC="$(CC)" CFLAGS="$(CFLAGS)" ./genbuild > $@
