--- Makefile	2014-10-10 13:44:34.610873684 -0400
+++ Makefile	2014-10-10 13:46:17.175468831 -0400
@@ -104,8 +104,8 @@
 LIBGCRYPT_CFLAGS=
 LIBGCRYPT_LDFLAGS=-lgcrypt
 
-LIBJSONC_CFLAGS=$(shell pkg-config --cflags json)
-LIBJSONC_LDFLAGS=$(shell pkg-config --libs json)
+LIBJSONC_CFLAGS=$(shell pkg-config --cflags json-c)
+LIBJSONC_LDFLAGS=$(shell pkg-config --libs json-c)
 
 ifeq (${DISABLE_ID3TAG}, 1)
 	LIBID3TAG_CFLAGS=
