diff --git a/.gitmodules b/.gitmodules
index beda934f6..01f7cbb83 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -91,3 +91,6 @@
 [submodule "Telegram/lib_webview"]
 	path = Telegram/lib_webview
 	url = https://github.com/kotatogram/lib_webview.git
+[submodule "Telegram/ThirdParty/mallocng"]
+	path = Telegram/ThirdParty/mallocng
+	url = https://github.com/desktop-app/mallocng.git
diff --git a/Telegram/CMakeLists.txt b/Telegram/CMakeLists.txt
index 0e00fcee1..fed1f9634 100644
--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -89,6 +89,7 @@ elseif (LINUX)
     PRIVATE
         desktop-app::external_glibmm
         desktop-app::external_glib
+        desktop-app::external_mallocng
     )
 
     if (NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
