diff --git a/include/mapnik/geometry/fusion_adapted.hpp b/include/mapnik/geometry/fusion_adapted.hpp
index fcf574567..69cbde29f 100644
--- a/include/mapnik/geometry/fusion_adapted.hpp
+++ b/include/mapnik/geometry/fusion_adapted.hpp
@@ -26,6 +26,8 @@
 #include <mapnik/geometry/polygon.hpp>
 #include <boost/fusion/include/adapt_struct.hpp>
 
+#include <cstdint>
+
 BOOST_FUSION_ADAPT_STRUCT(mapnik::geometry::point<double>, (double, x)(double, y))
 
 BOOST_FUSION_ADAPT_STRUCT(mapnik::geometry::point<std::int64_t>, (std::int64_t, x)(std::int64_t, y))
diff --git a/include/mapnik/util/singleton.hpp b/include/mapnik/util/singleton.hpp
index 897fcee69..4a379f737 100644
--- a/include/mapnik/util/singleton.hpp
+++ b/include/mapnik/util/singleton.hpp
@@ -39,7 +39,7 @@
 namespace mapnik {
 
 template<typename T>
-class CreateUsingNew
+class MAPNIK_DECL CreateUsingNew
 {
   public:
     static T* create() { return new T; }
@@ -47,7 +47,7 @@ class CreateUsingNew
 };
 
 template<typename T>
-class CreateStatic
+class MAPNIK_DECL CreateStatic
 {
   private:
     using storage_type = typename std::aligned_storage<sizeof(T), alignof(T)>::type;
