diff --git a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
index fb38ffe34fe..c26c447ad3d 100644
--- a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
+++ b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/LinuxToolChain.cs
@@ -57,6 +57,11 @@ namespace UnrealBuildTool
 		/// When we support larger the 4GB files with objcopy.exe this can be removed!
 		/// </summary>
 		DisableSplitDebugInfoWithObjCopy = 0x40,
+		
+		/// <summary>
+		/// Enable Native Instruction sets for the cpu
+		/// </summary>
+		EnableNativeInstructionSet = 0x80
 	}
 
 	class LinuxToolChain : ISPCToolChain
@@ -789,6 +794,11 @@ namespace UnrealBuildTool
 				}
 			}
 
+			if (Options.HasFlag(LinuxToolChainOptions.EnableNativeInstructionSet))
+			{
+				Result += " -march=native";
+			}
+
 			if (!CompileEnvironment.bUseInlining)
 			{
 				Result += " -fno-inline-functions";
diff --git a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/UEBuildLinux.cs b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/UEBuildLinux.cs
index 2f0acc23d7a..5503cb87646 100644
--- a/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/UEBuildLinux.cs
+++ b/Engine/Source/Programs/UnrealBuildTool/Platform/Linux/UEBuildLinux.cs
@@ -73,6 +73,12 @@ namespace UnrealBuildTool
 		[CommandLine("-ThinLTO")]
 		public bool bEnableThinLTO = false;
 
+		/// <summary>
+		/// Enables Native cpu instruction set
+		/// </summary>
+		[CommandLine("-EnableNativeInstructionSet")]
+		public bool bEnableNativeInstructionSet = false;
+
 		/// <summary>
 		/// Whether or not to preserve the portable symbol file produced by dump_syms
 		/// </summary>
@@ -137,6 +143,11 @@ namespace UnrealBuildTool
 			get { return Inner.bEnableThinLTO; }
 		}
 
+		public bool bEnableNativeInstructionSet
+		{
+			get { return Inner.bEnableNativeInstructionSet; }
+		}
+
 		#if !__MonoCS__
 		#pragma warning restore CS1591
 		#endif
@@ -617,6 +628,11 @@ namespace UnrealBuildTool
 				Options |= LinuxToolChainOptions.EnableThinLTO;
 			}
 
+			if (Target.LinuxPlatform.bEnableNativeInstructionSet)
+			{
+				Options |= LinuxToolChainOptions.EnableNativeInstructionSet;
+			}
+
 			// When building a monolithic editor we have to avoid using objcopy.exe as it cannot handle files
 			// larger then 4GB. This is only an issue with our binutils objcopy.exe.
 			// llvm-objcopy.exe does not have this issue and once we switch over to using that in clang 10.0.1 we can remove this!
