diff -ur mayavi-4.8.1.orig/mayavi/__init__.py mayavi-4.8.1/mayavi/__init__.py
--- mayavi-4.8.1.orig/mayavi/__init__.py	2022-10-18 11:18:00.000000000 +0200
+++ mayavi-4.8.1/mayavi/__init__.py	2023-04-21 09:32:29.003917103 +0200
@@ -16,7 +16,6 @@
     'traits>=6.0.0',
     'traitsui>=7.0.0',
     'packaging',
-    'vtk'
 ]

 __extras_require__ = {
diff -ur mayavi-4.8.1.orig/mayavi/plugins/app.py mayavi-4.8.1/mayavi/plugins/app.py
--- mayavi-4.8.1.orig/mayavi/plugins/app.py	2020-04-29 21:28:22.000000000 +0200
+++ mayavi-4.8.1/mayavi/plugins/app.py	2023-04-21 10:41:20.952497313 +0200
@@ -23,6 +23,9 @@
 # GLOBALS
 logger = logging.getLogger()

+# The application ID
+ID = 'mayavi_e3'
+
 ######################################################################
 # Useful functions.
 ######################################################################
@@ -183,6 +186,11 @@

         plugins += get_custom_plugins()

+        # Set correct application home and user data
+        path = os.path.join(ETSConfig.application_data, ID)
+        ETSConfig.application_home = path
+        ETSConfig.user_data = path
+
         # Create the application
         prefs = preference_manager.preferences
         app = MayaviWorkbenchApplication(plugins=plugins,
diff -ur mayavi-4.8.1.orig/mayavi.egg-info/requires.txt mayavi-4.8.1/mayavi.egg-info/requires.txt
--- mayavi-4.8.1.orig/mayavi.egg-info/requires.txt	2022-10-18 11:49:21.000000000 +0200
+++ mayavi-4.8.1/mayavi.egg-info/requires.txt	2023-04-21 09:32:37.457290549 +0200
@@ -6,7 +6,6 @@
 traits>=6.0.0
 traitsui>=7.0.0
 packaging
-vtk

 [app]
 envisage
