diff -aur menumaker-0.99.7/MenuMaker/__init__.py menumaker-0.99.7.mod/MenuMaker/__init__.py
--- menumaker-0.99.7/MenuMaker/__init__.py	2005-12-13 01:01:42.000000000 -0800
+++ menumaker-0.99.7.mod/MenuMaker/__init__.py	2012-12-19 14:28:52.723592607 -0800
@@ -36,7 +36,9 @@
 	Kw("Deskmenu")		: KwS("Deskmenu"),
 	Kw("Xfce4")			: KwS("Xfce", "Xfce4"),
 	Kw("OpenBox3")		: KwS("OpenBox", "OpenBox3"),
-	Kw("WindowMaker")	: KwS("WindowMaker", "WMaker")
+	Kw("WindowMaker")	: KwS("WindowMaker", "WMaker"),
+	Kw("Compiz")		: KwS("Compiz", "Compiz-deskmenu"),
+	Kw("MyGTKMenu")		: KwS("MyGTKMenu")
 }
 
 
@@ -47,12 +49,15 @@
 terms = [
 	(T.xterm, KwS("Xterm")),
 	(T._xfterm, KwS("Xfterm")),
-	(T.xvt, KwS("Rxvt", "Xvt")),
+	(T.xvt, KwS("Rxvt", "Xvt", "Urxvt")),
 	(T.aterm, KwS("Aterm")),
 	(T.wterm, KwS("Wterm")),
 	(T.Eterm, KwS("Eterm")),
 	(T._gterm, KwS("GNOME-terminal", "Gterm")),
-	(T._konsole, KwS("Konsole"))
+	(T._konsole, KwS("Konsole")),
+	(T.roxterm, KwS("ROXterm")),
+	(T.lilyterm, KwS("Lilyterm")),
+	(T.terminator, KwS("Terminator"))
 ]
 
 
diff -aur menumaker-0.99.7/MenuMaker/PekWM.py menumaker-0.99.7.mod/MenuMaker/PekWM.py
--- menumaker-0.99.7/MenuMaker/PekWM.py	2005-12-12 19:51:23.000000000 -0800
+++ menumaker-0.99.7.mod/MenuMaker/PekWM.py	2012-12-19 14:56:48.274171175 -0800
@@ -95,8 +95,8 @@
 	def __init__(self) :
 		subs = []
 		try :
-			prefix = pekwm().prefix
-			themeset = os.path.join(prefix, "share/pekwm/scripts/pekwm_themeset.pl")
+			prefix = pekwm().prefix.replace("/bin","")
+			themeset = os.path.join(prefix, "share/pekwm/scripts/pekwm_themeset.sh")
 			if Prophet.isExe(themeset) :
 				for dir in (os.path.join(prefix, "share/pekwm/themes"), "~/.pekwm/themes") :
 					if os.path.isdir(os.path.expanduser(dir)) :
diff -aur menumaker-0.99.7/Prophet/Legacy/Shell.py menumaker-0.99.7.mod/Prophet/Legacy/Shell.py
--- menumaker-0.99.7/Prophet/Legacy/Shell.py	2005-12-13 00:50:13.000000000 -0800
+++ menumaker-0.99.7.mod/Prophet/Legacy/Shell.py	2012-12-19 14:27:44.473569342 -0800
@@ -128,7 +128,20 @@
 	comment = "RISCOS-like file manager"
 	keywords = KwS(FileManager, Shell)
 
+class roxterm(_Terminal) :
+	name = "ROXterm"
+	comment = "X terminal emulator for ROX"
+	keywords = KwS(TerminalEmulator)
+
+class lilyterm(_Terminal) :
+	name = "Lilyterm"
+	comment = "A light and easy to use libvte based X terminal emulator"
+	keywords = KwS(TerminalEmulator)
 
+class terminator(_Terminal):
+	name = "Terminator"
+	comment = "An application that provides lots of terminals in a single window"
+	keywords = KwS(TerminalEmulator)
 
 
 class tuxcmd(_App, _X11App) :
@@ -200,7 +213,7 @@
 class xvt(_Terminal) : # FIXME : do we need a separate entry for rxvt?
 	name = "Rxvt"
 	comment = "X terminal emulator"
-	exes = ["rxvt", "xvt"]
+	exes = ["rxvt", "xvt", "urxvt"]
 	keywords = KwS(TerminalEmulator)
 
 
