--- a/build/main.js
+++ b/build/main.js
@@ -4541,6 +4541,8 @@
                 .setAutoResize({ width: !0, height: !0 }))
             : this.appService.setWindow(this.window, void 0),
           this.appService.setUserAgent(),
+          this.window.setAutoHideMenuBar(true),
+          this.window.setMenuBarVisibility(false),
           isPlatform(PLATFORM_DARWIN)
             ? this.window.on("moved", () => {
                 this.windowState.save();
