diff -rupN noise.orig/data/CMakeLists.txt noise/data/CMakeLists.txt
--- noise.orig/data/CMakeLists.txt	2016-02-14 17:19:41.832028621 +0100
+++ noise/data/CMakeLists.txt	2016-02-14 17:30:04.621096326 +0100
@@ -2,16 +2,16 @@ include (Translations)
 
 # We need to create two temporary desktop files in order to have all translations available.
 set (APP_NAME "Music")
-configure_file(noise.desktop.in.in ${CMAKE_CURRENT_BINARY_DIR}/noise.desktop.in)
+configure_file(noise-player.desktop.in.in ${CMAKE_CURRENT_BINARY_DIR}/noise-player.desktop.in)
 set (APP_NAME "Noise")
-configure_file(noise.desktop.in.in ${CMAKE_CURRENT_BINARY_DIR}/noise-original.desktop.in)
+configure_file(noise-player.desktop.in.in ${CMAKE_CURRENT_BINARY_DIR}/noise-player-original.desktop.in)
 if (BUILD_FOR_ELEMENTARY)
-    configure_file_translation(${CMAKE_CURRENT_BINARY_DIR}/noise.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/noise.desktop ${CMAKE_SOURCE_DIR}/po/)
+    configure_file_translation(${CMAKE_CURRENT_BINARY_DIR}/noise-player.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/noise-player.desktop ${CMAKE_SOURCE_DIR}/po/)
 else ()
-    configure_file_translation(${CMAKE_CURRENT_BINARY_DIR}/noise-original.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/noise.desktop ${CMAKE_SOURCE_DIR}/po/)
+    configure_file_translation(${CMAKE_CURRENT_BINARY_DIR}/noise-player-original.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/noise-player.desktop ${CMAKE_SOURCE_DIR}/po/)
 endif ()
 
 configure_file_translation(${CMAKE_CURRENT_SOURCE_DIR}/noise.appdata.xml.in ${CMAKE_CURRENT_BINARY_DIR}/noise.appdata.xml ${CMAKE_SOURCE_DIR}/po/)
 
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/noise.appdata.xml DESTINATION share/appdata)
-install (FILES ${CMAKE_CURRENT_BINARY_DIR}/noise.desktop DESTINATION share/applications)
+install (FILES ${CMAKE_CURRENT_BINARY_DIR}/noise-player.desktop DESTINATION share/applications)
diff -rupN noise.orig/data/noise.appdata.xml.in noise/data/noise.appdata.xml.in
--- noise.orig/data/noise.appdata.xml.in	2016-02-14 17:19:41.832028621 +0100
+++ noise/data/noise.appdata.xml.in	2016-02-14 17:30:19.570444585 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2015 elementary, LLC <contact@elementary.io> -->
 <component type="desktop">
-  <id>noise.desktop</id>
+  <id>noise-player.desktop</id>
   <metadata_license>CC0-1.0</metadata_license>
   <project_group>elementary</project_group>
   <project_license>GPL-3.0+</project_license>
@@ -25,7 +25,7 @@
     <release timestamp="1320168563" version="0.1.0"/>
   </releases>
   <provides>
-    <binary>noise</binary>
+    <binary>noise-player</binary>
     <library>libnoise-core.so.0</library>
   </provides>
   <developer_name>elementary LLC.</developer_name>
diff -rupN noise.orig/data/noise.desktop.in.in noise/data/noise.desktop.in.in
--- noise.orig/data/noise.desktop.in.in	2016-02-14 17:19:41.832028621 +0100
+++ noise/data/noise.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,19 +0,0 @@
-[Desktop Entry]
-Type=Application
-_Name=@APP_NAME@
-_GenericName=Music Player
-_Comment=Listen to music
-Exec=noise %U
-Icon=multimedia-audio-player
-Terminal=false
-Categories=Audio;Music;Player;AudioVideo;GNOME;GTK;
-MimeType=x-content/audio-player;x-content/audio-cdda;application/ogg;application/x-extension-m4a;application/x-extension-mp4;application/x-flac;application/x-ogg;audio/3gpp;audio/aac;audio/ac3;audio/AMR;audio/AMR-WB;audio/basic;audio/flac;audio/midi;audio/mp2;audio/mp4;audio/mpeg;audio/ogg;audio/vnd.rn-realaudio;audio/x-aiff;audio/x-ape;audio/x-flac;audio/x-gsm;audio/x-it;audio/x-m4a;audio/x-matroska;audio/x-mod;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-asf;audio/x-ms-asx;audio/x-ms-wax;audio/x-ms-wma;audio/x-musepack;audio/x-pn-aiff;audio/x-pn-au;audio/x-pn-realaudio;audio/x-pn-realaudio-plugin;audio/x-pn-wav;audio/x-pn-windows-acm;audio/x-realaudio;audio/x-real-audio;audio/x-sbc;audio/x-scpls;audio/x-speex;audio/x-tta;audio/x-vorbis;audio/x-vorbis+ogg;audio/x-wav;audio/x-wavpack;audio/x-xm;
-StartupNotify=true
-_Keywords=Noise;Audio;Player;MP3;iPod;Play;Playlist;Media;CD;Phone;Songs;
-X-GNOME-UsesNotifications=true
-X-PulseAudio-Properties=media.role=music
-Actions=AboutDialog;
-
-[Desktop Action AboutDialog]
-Exec=noise --about
-_Name=About @APP_NAME@
diff -rupN noise.orig/data/noise-player.desktop.in.in noise/data/noise-player.desktop.in.in
--- noise.orig/data/noise-player.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
+++ noise/data/noise-player.desktop.in.in	2016-02-14 17:19:48.261652683 +0100
@@ -0,0 +1,19 @@
+[Desktop Entry]
+Type=Application
+_Name=@APP_NAME@
+_GenericName=Music Player
+_Comment=Listen to music
+Exec=noise-player %U
+Icon=multimedia-audio-player
+Terminal=false
+Categories=Audio;Music;Player;AudioVideo;GNOME;GTK;
+MimeType=x-content/audio-player;x-content/audio-cdda;application/ogg;application/x-extension-m4a;application/x-extension-mp4;application/x-flac;application/x-ogg;audio/3gpp;audio/aac;audio/ac3;audio/AMR;audio/AMR-WB;audio/basic;audio/flac;audio/midi;audio/mp2;audio/mp4;audio/mpeg;audio/ogg;audio/vnd.rn-realaudio;audio/x-aiff;audio/x-ape;audio/x-flac;audio/x-gsm;audio/x-it;audio/x-m4a;audio/x-matroska;audio/x-mod;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-asf;audio/x-ms-asx;audio/x-ms-wax;audio/x-ms-wma;audio/x-musepack;audio/x-pn-aiff;audio/x-pn-au;audio/x-pn-realaudio;audio/x-pn-realaudio-plugin;audio/x-pn-wav;audio/x-pn-windows-acm;audio/x-realaudio;audio/x-real-audio;audio/x-sbc;audio/x-scpls;audio/x-speex;audio/x-tta;audio/x-vorbis;audio/x-vorbis+ogg;audio/x-wav;audio/x-wavpack;audio/x-xm;
+StartupNotify=true
+_Keywords=Noise;Audio;Player;MP3;iPod;Play;Playlist;Media;CD;Phone;Songs;
+X-GNOME-UsesNotifications=true
+X-PulseAudio-Properties=media.role=music
+Actions=AboutDialog;
+
+[Desktop Action AboutDialog]
+Exec=noise-player --about
+_Name=About @APP_NAME@
diff -rupN noise.orig/po/CMakeLists.txt noise/po/CMakeLists.txt
--- noise.orig/po/CMakeLists.txt	2016-02-14 17:19:41.835361759 +0100
+++ noise/po/CMakeLists.txt	2016-02-14 17:28:26.395450634 +0100
@@ -5,7 +5,7 @@ add_translations_catalog (${GETTEXT_PACK
     ../src
     ../plugins
     DESKTOP_FILES
-        ${CMAKE_BINARY_DIR}/data/noise.desktop.in
-        ${CMAKE_BINARY_DIR}/data/noise-original.desktop.in
+        ${CMAKE_BINARY_DIR}/data/noise-player.desktop.in
+        ${CMAKE_BINARY_DIR}/data/noise-player-original.desktop.in
     APPDATA_FILES
         ../data/noise.appdata.xml.in)
diff -rupN noise.orig/src/CMakeLists.txt noise/src/CMakeLists.txt
--- noise.orig/src/CMakeLists.txt	2016-02-14 17:19:41.845361173 +0100
+++ noise/src/CMakeLists.txt	2016-02-14 17:19:48.261652683 +0100
@@ -88,7 +88,7 @@ vala_precompile (CLIENT_VALA_C ${SRC_TAR
   ${CLIENT_VAPI}
 )
 
-set (EXEC_NAME ${SRC_TARGET})
+set (EXEC_NAME ${SRC_TARGET}-player)
 
 add_definitions (${DEPS_CFLAGS})
 link_directories (${DEPS_LIBRARY_DIRS})
