--- meson.build	2020-06-30 09:04:50.000000000 +0200
+++ meson.build.new	2020-06-30 09:04:45.000000000 +0200
@@ -78,8 +78,8 @@
 is_openbsd = host_machine.system() == 'openbsd'
 
 thread_dep = dependency('threads')
-fmt = dependency('fmt', version : ['>=5.3.0'], fallback : ['fmt', 'fmt_dep'])
-spdlog = dependency('spdlog', version : ['>=1.8.0'], fallback : ['spdlog', 'spdlog_dep'])
+fmt = subproject('fmt').get_variable('fmt_dep')
+spdlog = subproject('spdlog').get_variable('spdlog_dep')
 wayland_client = dependency('wayland-client')
 wayland_cursor = dependency('wayland-cursor')
 wayland_protos = dependency('wayland-protocols')
