diff --git a/meson.build b/meson.build
index 871c5304..0cf9a082 100644
--- a/meson.build
+++ b/meson.build
@@ -21,8 +21,7 @@ sass = find_program('sass', required: false)
 sass_opts = ['--no-source-map']
 
 if sass.found()
-  sass_full_version = run_command(sass, '--version',
-                                  version:'>= 1.23.0').stdout()
+  sass_full_version = run_command(sass, '--version', check: true).stdout()
   sass_is_ruby_sass = sass_full_version.contains('Ruby Sass')
   sass_has_module_system = sass_full_version.version_compare('>= 1.23.0')
 endif
@@ -30,7 +29,7 @@ endif
 if not sass.found() or sass_is_ruby_sass or not sass_has_module_system
   message('dart-sass >= 1.23.0 not found, installing it locally via npm')
   npm = find_program('npm')
-  run_command(npm, 'install')
+  run_command(npm, 'install', check: true)
   sass = find_program('./node_modules/.bin/sass')
 endif
 
