--- Messages.h	2018-10-13 13:38:00.289005712 -0500
+++ Messages_new.h	2018-10-14 00:07:54.826558760 -0500
@@ -19,7 +19,9 @@
 	TArray<uint8> AsData()
 	{
 		FBufferArchive MemAr;
-		return MemAr;
+		typedef FBufferArchive &FBufferArchiveRef;
+		FBufferArchiveRef &&MemAr_ref = MemAr;
+		return (FBufferArchive&&)MemAr_ref; 
 	}
 };
 
@@ -47,7 +49,9 @@
 	{
 		FBufferArchive MemAr;
 		MemAr << Param1 << Param2;
-		return MemAr;
+		typedef FBufferArchive &FBufferArchiveRef;
+		FBufferArchiveRef &&MemAr_ref = MemAr;
+		return (FBufferArchive&&)MemAr_ref; 
 	}
 };
 
@@ -79,7 +83,9 @@
 	{
 		FBufferArchive MemAr;
 		MemAr << Param1 << Param2 << Param3;
-		return MemAr;
+		typedef FBufferArchive &FBufferArchiveRef;
+		FBufferArchiveRef &&MemAr_ref = MemAr;
+		return (FBufferArchive&&)MemAr_ref; 
 	}
 };
 
@@ -115,7 +121,9 @@
 	{
 		FBufferArchive MemAr;
 		MemAr << Param1 << Param2 << Param3 << Param4;
-		return MemAr;
+		typedef FBufferArchive &FBufferArchiveRef;
+		FBufferArchiveRef &&MemAr_ref = MemAr;
+		return (FBufferArchive&&)MemAr_ref; 
 	}
 };
 
@@ -155,6 +163,8 @@
 	{
 		FBufferArchive MemAr;
 		MemAr << Param1 << Param2 << Param3 << Param4 << Param5;
-		return MemAr;
+		typedef FBufferArchive &FBufferArchiveRef;
+		FBufferArchiveRef &&MemAr_ref = MemAr;
+		return (FBufferArchive&&)MemAr_ref; 
 	}
 };
