diff -rupN trunk/terminatorlib/terminal.py trunk.middle_mouse/terminatorlib/terminal.py
--- trunk/terminatorlib/terminal.py	2015-07-29 16:45:55.154412000 +0000
+++ trunk.middle_mouse/terminatorlib/terminal.py	2015-07-29 16:47:20.470794351 +0000
@@ -895,14 +895,14 @@ class Terminal(gtk.VBox):
                 if url:
                     self.open_url(url, prepare=True)
         elif event.button == 2:
-            # middleclick should paste the clipboard
-            self.paste_clipboard(True)
-            return(True)
-        elif event.button == 3:
-            # rightclick should display a context menu if Ctrl is not pressed
+            # middleclick should display a context menu if Ctrl is not pressed
             if event.state & gtk.gdk.CONTROL_MASK == 0:
                 self.popup_menu(widget, event)
                 return(True)
+        elif event.button == 3:
+            # rightclick should paste the clipboard
+            self.paste_clipboard(True)
+            return(True)
 
         return(False)
     
