diff -ruN AviSynthPlus-3.7.1/avs_core/CMakeLists.txt patched/avs_core/CMakeLists.txt
--- AviSynthPlus-3.7.1/avs_core/CMakeLists.txt	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/CMakeLists.txt	2022-01-04 18:40:22.879994562 +0100
@@ -126,7 +126,7 @@
 if (MSVC)
   target_link_libraries("AvsCore" "uuid" "winmm" "vfw32" "msacm32" "gdi32" "user32" "advapi32" "ole32" "imagehlp")
 elseif (MINGW)
-  target_link_libraries("AvsCore" "uuid" "winmm" "msacm32" "gdi32" "ole32" "vfw32" "imagehlp" "${SYSLIB}" "${FSLIB}" -static)
+  target_link_libraries("AvsCore" "uuid" "winmm" "msacm32" "gdi32" "ole32" "vfw32" "imagehlp" "${SYSLIB}" "${FSLIB}" "ssp" -static)
 else()
   target_link_libraries("AvsCore" "${SYSLIB}" "${FSLIB}")
 endif()
diff -ruN AviSynthPlus-3.7.1/avs_core/core/BufferPool.h patched/avs_core/core/BufferPool.h
--- AviSynthPlus-3.7.1/avs_core/core/BufferPool.h	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/core/BufferPool.h	2022-01-04 18:40:04.139994555 +0100
@@ -11,12 +11,12 @@
 private:
 
   struct BufferDesc;
-  typedef std::multimap<size_t, BufferDesc*> MapType;
+  typedef std::multimap<std::size_t, BufferDesc*> MapType;
 
   InternalEnvironment* Env;
   MapType Map;
 
-  void* PrivateAlloc(size_t nBytes, size_t alignment, void* user);
+  void* PrivateAlloc(std::size_t nBytes, std::size_t alignment, void* user);
   void PrivateFree(void* buffer);
 
 public:
@@ -24,7 +24,7 @@
   BufferPool(InternalEnvironment* env);
   ~BufferPool();
 
-  void* Allocate(size_t nBytes, size_t alignment, bool pool);
+  void* Allocate(std::size_t nBytes, std::size_t alignment, bool pool);
   void Free(void* ptr);
 
 };
diff -ruN AviSynthPlus-3.7.1/avs_core/core/internal.h patched/avs_core/core/internal.h
--- AviSynthPlus-3.7.1/avs_core/core/internal.h	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/core/internal.h	2022-01-04 18:40:04.139994555 +0100
@@ -47,7 +47,7 @@
 #include <limits.h>
 #endif
 #include "InternalEnvironment.h"
-#ifdef INTEL_INTRINSICS
+#if defined(INTEL_INTRINSICS) && !defined(__MINGW32__)
 #ifdef AVS_WINDOWS
 #include <intrin.h>
 #else
diff -ruN AviSynthPlus-3.7.1/avs_core/core/ThreadPool.h patched/avs_core/core/ThreadPool.h
--- AviSynthPlus-3.7.1/avs_core/core/ThreadPool.h	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/core/ThreadPool.h	2022-01-04 18:40:04.143327888 +0100
@@ -42,7 +42,7 @@
     return ret;
   }
 
-  virtual ~JobCompletion()
+  virtual __stdcall ~JobCompletion()
   {
     Wait();
     delete [] pairs;
diff -ruN AviSynthPlus-3.7.1/avs_core/filters/AviSource/avi_source.cpp patched/avs_core/filters/AviSource/avi_source.cpp
--- AviSynthPlus-3.7.1/avs_core/filters/AviSource/avi_source.cpp	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/filters/AviSource/avi_source.cpp	2022-01-04 18:40:04.143327888 +0100
@@ -294,7 +294,7 @@
     return result;
 }
 
-#ifndef MSVC
+#if !defined(MSVC) && !defined(__MINGW32__)
 static __inline LRESULT
 ICDecompressEx(HIC hic,DWORD dwFlags,LPBITMAPINFOHEADER lpbiSrc,LPVOID lpSrc,INT xSrc,INT ySrc,INT dxSrc,INT dySrc,LPBITMAPINFOHEADER lpbiDst,LPVOID lpDst,INT xDst,INT yDst,INT dxDst,INT dyDst)
 {
diff -ruN AviSynthPlus-3.7.1/avs_core/include/avisynth.h patched/avs_core/include/avisynth.h
--- AviSynthPlus-3.7.1/avs_core/include/avisynth.h	2021-12-28 13:00:08.000000000 +0100
+++ patched/avs_core/include/avisynth.h	2022-01-04 18:40:04.143327888 +0100
@@ -1635,7 +1635,7 @@
 {
 public:
 
-  virtual ~IJobCompletion() {}
+  virtual __stdcall ~IJobCompletion() {}
   virtual void __stdcall Wait() = 0;
   virtual AVSValue __stdcall Get(size_t i) = 0;
   virtual size_t __stdcall Size() const = 0;
diff -ruN AviSynthPlus-3.7.1/plugins/ImageSeq/CMakeLists.txt patched/plugins/ImageSeq/CMakeLists.txt
--- AviSynthPlus-3.7.1/plugins/ImageSeq/CMakeLists.txt	2021-12-28 13:00:08.000000000 +0100
+++ patched/plugins/ImageSeq/CMakeLists.txt	2022-01-04 18:40:04.146661222 +0100
@@ -43,14 +43,14 @@
     # Specify include directories
     target_include_directories(${ProjectName} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x86 ${AvsCore_SOURCE_DIR})
     # Windows DLL dependencies
-    target_link_libraries(${ProjectName} "${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x86/DevIL.lib")
+    target_link_libraries(${ProjectName} "${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x86/DevIL.lib" "ssp")
     # Copy binary dependencies to a common folder for easy deployment
     configure_file("${PROJECT_SOURCE_DIR}/lib/DevIL_x86/DevIL.dll" "${CMAKE_BINARY_DIR}/Output/system/DevIL.dll" COPYONLY)
   else()   # 64-bit
     # Specify include directories
     target_include_directories(${ProjectName} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x64 ${AvsCore_SOURCE_DIR})
     # Windows DLL dependencies
-    target_link_libraries(${ProjectName} "${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x64/DevIL.lib")
+    target_link_libraries(${ProjectName} "${CMAKE_CURRENT_SOURCE_DIR}/lib/DevIL_x64/DevIL.lib" "ssp")
     # Copy binary dependencies to a common folder for easy deployment
     configure_file("${PROJECT_SOURCE_DIR}/lib/DevIL_x64/DevIL.dll" "${CMAKE_BINARY_DIR}/Output/system/DevIL.dll" COPYONLY)
   endif()
diff -ruN AviSynthPlus-3.7.1/plugins/VDubFilter/CMakeLists.txt patched/plugins/VDubFilter/CMakeLists.txt
--- AviSynthPlus-3.7.1/plugins/VDubFilter/CMakeLists.txt	2021-12-28 13:00:08.000000000 +0100
+++ patched/plugins/VDubFilter/CMakeLists.txt	2022-01-04 18:40:04.146661222 +0100
@@ -22,7 +22,7 @@
 endif()
 
 # Library dependencies
-target_link_libraries(${ProjectName} "user32")
+target_link_libraries(${ProjectName} "user32" "ssp")
 
 # Include directories
 target_include_directories(${ProjectName} PRIVATE ${AvsCore_SOURCE_DIR})
