diff -ruN libssh-0.8.1/CMakeLists.txt patched/CMakeLists.txt
--- libssh-0.8.1/CMakeLists.txt	2018-08-13 22:21:19.000000000 +0200
+++ patched/CMakeLists.txt	2018-08-26 00:27:16.718587338 +0200
@@ -98,7 +98,7 @@
 add_subdirectory(src)
 
 # pkg-config file
-if (UNIX)
+if (UNIX OR MINGW)
 configure_file(libssh.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libssh.pc)
 install(
   FILES
@@ -108,7 +108,7 @@
   COMPONENT
     pkgconfig
 )
-endif (UNIX)
+endif (UNIX OR MINGW)
 
 # cmake config files
 set(LIBSSH_LIBRARY_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}ssh${CMAKE_SHARED_LIBRARY_SUFFIX})
