diff --git a/configure.ac b/configure.ac
index e89ad854f..7dc2fba2f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,18 +121,13 @@ AC_ARG_ENABLE([documentation],
 	[], [enable_documentation=yes])
 AM_CONDITIONAL(ENABLE_DOCUMENTATION, test x$enable_documentation = xyes)
 
-LIBGNOME_DESKTOP_REQUIRED=2.91.0
 GDK_PIXBUF_REQUIRED=2.26.0
 PANGO_REQUIRED=1.15.4
 GLIB_REQUIRED=2.67.1
 GTK_REQUIRED=3.22.0
 LIBWNCK_REQUIRED=40.0
-LIBGNOME_MENU_REQUIRED=3.7.90
-LIBECAL_REQUIRED=3.33.2
-LIBEDATASERVER_REQUIRED=3.5.3
 LIBSYSTEMD_REQUIRED=230
 CAIRO_REQUIRED=1.0.0
-GWEATHER_REQUIRED=3.91.0
 DCONF_REQUIRED=0.13.4
 XRANDR_REQUIRED=1.3.0
 
@@ -144,13 +139,10 @@ PKG_CHECK_MODULES([PANEL], [
   pango >= $PANGO_REQUIRED
   gtk+-3.0 >= $GTK_REQUIRED
   cairo-xlib
-  gdm
   glib-2.0 >= $GLIB_REQUIRED
   gmodule-2.0 >= $GLIB_REQUIRED
   gio-2.0 >= $GLIB_REQUIRED
   gio-unix-2.0 >= $GLIB_REQUIRED
-  gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
-  libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
   dconf >= $DCONF_REQUIRED
   x11
   xi
@@ -163,29 +155,9 @@ PKG_CHECK_MODULES([LIBGNOME_PANEL], [
   gtk+-3.0 >= $GTK_REQUIRED
 ])
 
-PKG_CHECK_MODULES([ACTION_BUTTON], [
-  gio-unix-2.0 >= $GLIB_REQUIRED
-  gtk+-3.0 >= $GTK_REQUIRED
-  libsystemd >= $LIBSYSTEMD_REQUIRED
-  x11
-])
-
-PKG_CHECK_MODULES(FISH, gtk+-3.0 >= $GTK_REQUIRED cairo >= $CAIRO_REQUIRED)
-AC_SUBST(FISH_CFLAGS)
-AC_SUBST(FISH_LIBS)
-
 PKG_CHECK_MODULES([LAUNCHER], [
   gio-unix-2.0 >= $GLIB_REQUIRED
   gtk+-3.0 >= $GTK_REQUIRED
-  libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
-  libsystemd >= $LIBSYSTEMD_REQUIRED
-])
-
-PKG_CHECK_MODULES([MENU], [
-  gdm
-  gio-unix-2.0 >= $GLIB_REQUIRED
-  gtk+-3.0 >= $GTK_REQUIRED
-  libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
   libsystemd >= $LIBSYSTEMD_REQUIRED
 ])
 
@@ -198,11 +170,6 @@ PKG_CHECK_MODULES([SEPARATOR], [
   gtk+-3.0 >= $GTK_REQUIRED
 ])
 
-PKG_CHECK_MODULES([STATUS_NOTIFIER], [
-  gio-unix-2.0 >= $GLIB_REQUIRED
-  gtk+-3.0 >= $GTK_REQUIRED
-])
-
 PKG_CHECK_MODULES(WNCKLET, gtk+-3.0 >= $GTK_REQUIRED libwnck-3.0 >= $LIBWNCK_REQUIRED)
 AC_SUBST(WNCKLET_CFLAGS)
 AC_SUBST(WNCKLET_LIBS)
@@ -210,45 +177,6 @@ AC_SUBST(WNCKLET_LIBS)
 AC_CHECK_HEADERS(langinfo.h)
 AC_CHECK_FUNCS(nl_langinfo)
 
-PKG_CHECK_MODULES([CLOCK], [
-  dconf >= $DCONF_REQUIRED
-  gio-2.0 >= $GLIB_REQUIRED
-  gio-unix-2.0 >= $GLIB_REQUIRED
-  glib-2.0 >= $GLIB_REQUIRED
-  gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
-  gtk+-3.0 >= $GTK_REQUIRED
-  gweather4 >= $GWEATHER_REQUIRED
-  pango >= $PANGO_REQUIRED
-  polkit-gobject-1
-])
-
-# Check whether to enable e-d-s support for clock applet
-AC_MSG_CHECKING([whether to enable evolution-data-server support])
-AC_ARG_ENABLE([eds],
-    AS_HELP_STRING([--enable-eds], [Whether to enable evolution-data-server support @<:@default=auto@:>@]),
-    [], [enable_eds=auto])
-AC_MSG_RESULT([$enable_eds])
-CLOCK_EDS_ICONDIR=
-if test "$enable_eds" != "no"; then
-    PKG_CHECK_MODULES(CLOCK_EDS,[
-        libecal-2.0 >= $LIBECAL_REQUIRED
-        libedataserver-1.2 >= $LIBEDATASERVER_REQUIRED
-     ], [have_eds=yes], [have_eds=no])
-
-    if test "$enable_eds" = "yes" -a "$have_eds" = "no"; then
-        AC_MSG_ERROR([evolution-data-server support was requested but required dependencies were not found: $CLOCK_EDS_PKG_ERRORS])
-    elif test "$have_eds" = "yes"; then
-        enable_eds=yes
-        AC_DEFINE([HAVE_EDS], 1, [Define to enable evolution-data-server support])
-        CLOCK_EDS_ICONDIR=`$PKG_CONFIG --variable=datadir evolution-data-server-1.2`/pixmaps/evolution-data-server
-    else
-        enable_eds=no
-    fi
-fi
-
-AM_CONDITIONAL(ENABLE_EDS, test "$enable_eds" = "yes")
-AC_SUBST(CLOCK_EDS_ICONDIR)
-
 dnl **************************************************************************
 dnl Check if struct dirent contains a d_type member, for run dialog
 dnl **************************************************************************
@@ -296,21 +224,12 @@ AC_CONFIG_FILES([
   docs/reference/libgnome-panel/Makefile
 
   help/Makefile
-  help/clock/Makefile
-  help/fish/Makefile
 
   man/Makefile
 
   modules/Makefile
-  modules/action-button/Makefile
-  modules/clock/Makefile
-  modules/clock/pixmaps/Makefile
-  modules/fish/Makefile
-  modules/launcher/Makefile
-  modules/menu/Makefile
   modules/notification-area/Makefile
   modules/separator/Makefile
-  modules/status-notifier/Makefile
   modules/wncklet/Makefile
 
   po/Makefile.in
diff --git a/data/16x16/Makefile.am b/data/16x16/Makefile.am
index 709c3a768..421a74246 100644
--- a/data/16x16/Makefile.am
+++ b/data/16x16/Makefile.am
@@ -4,7 +4,6 @@ iconsdir = $(datadir)/icons/hicolor/$(size)/apps
 
 icons_DATA =					\
 	gnome-panel-clock.png			\
-	gnome-panel-fish.png			\
 	gnome-panel-force-quit.png		\
 	gnome-panel-hibernate.png		\
 	gnome-panel-launcher.png		\
@@ -18,7 +17,6 @@ icons_DATA =					\
 
 icons_SOURCE =					\
 	gnome-panel-clock.xcf.bz2		\
-	gnome-panel-fish.xcf.bz2		\
 	gnome-panel-force-quit.svg		\
 	gnome-panel-launcher.svg		\
 	gnome-panel-notification-area.svg	\
diff --git a/data/22x22/Makefile.am b/data/22x22/Makefile.am
index 689cbdc26..4d404e771 100644
--- a/data/22x22/Makefile.am
+++ b/data/22x22/Makefile.am
@@ -4,7 +4,6 @@ iconsdir = $(datadir)/icons/hicolor/$(size)/apps
 
 icons_DATA =					\
 	gnome-panel-clock.png			\
-	gnome-panel-fish.png			\
 	gnome-panel-force-quit.png		\
 	gnome-panel-hibernate.png		\
 	gnome-panel-launcher.png		\
@@ -18,7 +17,6 @@ icons_DATA =					\
 
 icons_SOURCE =					\
 	gnome-panel-clock.svg			\
-	gnome-panel-fish.xcf.bz2		\
 	gnome-panel-force-quit.xcf.bz2		\
 	gnome-panel-launcher.svg		\
 	gnome-panel-notification-area.svg	\
diff --git a/data/24x24/Makefile.am b/data/24x24/Makefile.am
index d870db2df..9bc1c31ce 100644
--- a/data/24x24/Makefile.am
+++ b/data/24x24/Makefile.am
@@ -4,7 +4,6 @@ iconsdir = $(datadir)/icons/hicolor/$(size)/apps
 
 icons_DATA =					\
 	gnome-panel-clock.png			\
-	gnome-panel-fish.png			\
 	gnome-panel-force-quit.png		\
 	gnome-panel-hibernate.png		\
 	gnome-panel-launcher.png		\
diff --git a/data/32x32/Makefile.am b/data/32x32/Makefile.am
index 2c6cc33f5..671f58ff9 100644
--- a/data/32x32/Makefile.am
+++ b/data/32x32/Makefile.am
@@ -4,7 +4,6 @@ iconsdir = $(datadir)/icons/hicolor/$(size)/apps
 
 icons_DATA =					\
 	gnome-panel-clock.png			\
-	gnome-panel-fish.png			\
 	gnome-panel-force-quit.png		\
 	gnome-panel-hibernate.png		\
 	gnome-panel-launcher.png		\
@@ -18,7 +17,6 @@ icons_DATA =					\
 
 icons_SOURCE =					\
 	gnome-panel-clock.svg			\
-	gnome-panel-fish.svg			\
 	gnome-panel-force-quit.svg		\
 	gnome-panel-launcher.svg		\
 	gnome-panel-notification-area.svg	\
diff --git a/data/Makefile.am b/data/Makefile.am
index 67339eae9..879905876 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -10,10 +10,8 @@ gsettings_ENUM_NAMESPACE = org.gnome.gnome-panel
 gsettings_ENUM_FILES = $(top_srcdir)/gnome-panel/panel-enums-gsettings.h
 
 gsettings_SCHEMAS = \
-	org.gnome.gnome-panel.applet.fish.gschema.xml \
 	org.gnome.gnome-panel.applet.window-list.gschema.xml \
 	org.gnome.gnome-panel.applet.workspace-switcher.gschema.xml \
-	org.gnome.gnome-panel.applet.clock.gschema.xml \
 	org.gnome.gnome-panel.applet.initial-settings.gschema.xml \
 	org.gnome.gnome-panel.gschema.xml \
 	org.gnome.gnome-panel.object.gschema.xml \
diff --git a/data/default.layout b/data/default.layout
index 8e16f764a..2f15b30a6 100644
--- a/data/default.layout
+++ b/data/default.layout
@@ -1,48 +1,29 @@
-[Toplevel top-panel]
-expand=true
-orientation=top
-size=24
-
 [Toplevel bottom-panel]
 expand=true
 orientation=bottom
-size=24
-
-[Object menu-bar]
-object-iid=org.gnome.gnome-panel.menu::menu-bar
-toplevel-id=top-panel
-pack-index=0
-
-[Object clock]
-object-iid=org.gnome.gnome-panel.clock::clock
-toplevel-id=top-panel
-pack-type=end
-pack-index=1
+alignment=center
+size=28
 
 [Object notification-area]
 object-iid=org.gnome.gnome-panel.notification-area::notification-area
-toplevel-id=top-panel
+toplevel-id=bottom-panel
 pack-type=end
-pack-index=2
+pack-index=0
 
 [Object show-desktop]
 object-iid=org.gnome.gnome-panel.wncklet::show-desktop
 toplevel-id=bottom-panel
-pack-index=0
-
-[Object user-menu]
-object-iid=org.gnome.gnome-panel.menu::user-menu
-toplevel-id=top-panel
-pack-type=end
+pack-type=start
 pack-index=0
 
 [Object window-list]
 object-iid=org.gnome.gnome-panel.wncklet::window-list
 toplevel-id=bottom-panel
-pack-index=1
+pack-type=start
+pack-index=2
 
 [Object workspace-switcher]
 object-iid=org.gnome.gnome-panel.wncklet::workspace-switcher
 toplevel-id=bottom-panel
-pack-type=end
-pack-index=0
+pack-type=start
+pack-index=1
diff --git a/data/org.gnome.gnome-panel.gschema.xml b/data/org.gnome.gnome-panel.gschema.xml
index 08c4bbfa3..f835fb6ef 100644
--- a/data/org.gnome.gnome-panel.gschema.xml
+++ b/data/org.gnome.gnome-panel.gschema.xml
@@ -4,7 +4,6 @@
     <child name="general" schema="org.gnome.gnome-panel.general"/>
     <child name="layout" schema="org.gnome.gnome-panel.layout"/>
     <child name="lockdown" schema="org.gnome.gnome-panel.lockdown"/>
-    <child name="run-dialog" schema="org.gnome.gnome-panel.run-dialog"/>
   </schema>
 
   <enum id='org.gnome.gnome-panel.menu-icon-size'>
@@ -95,27 +94,4 @@
     </key>
   </schema>
 
-  <schema id="org.gnome.gnome-panel.run-dialog" path="/org/gnome/gnome-panel/run-dialog/">
-    <key type="as" name="history">
-      <default>[]</default>
-      <summary>History of "Run Application" dialog</summary>
-      <description>The list of commands used in the past in the "Run Application" dialog.</description>
-    </key>
-    <key name="enable-program-list" type="b">
-      <default>true</default>
-      <summary>Enable program list in "Run Application" dialog</summary>
-      <description>If true, the "Known Applications" listing in the "Run Application" dialog is made available. Whether or not the listing is expanded when the dialog is shown is controlled by the show_program_list key.</description>
-    </key>
-    <key name="show-program-list" type="b">
-      <default>false</default>
-      <summary>Expand program list in "Run Application" dialog</summary>
-      <description>If true, the "Known Applications" listing in the "Run Application" dialog is expanded when the dialog is opened. This key is only relevant if the enable_program_list key is true.</description>
-    </key>
-    <key name="enable-autocompletion" type="b">
-      <default>true</default>
-      <summary>Enable autocompletion in "Run Application" dialog</summary>
-      <description>If true, autocompletion in the "Run Application" dialog is made available.</description>
-    </key>
-  </schema>
-
 </schemalist>
diff --git a/data/scalable/Makefile.am b/data/scalable/Makefile.am
index 3933b3000..3acbaa593 100644
--- a/data/scalable/Makefile.am
+++ b/data/scalable/Makefile.am
@@ -4,7 +4,6 @@ iconsdir = $(datadir)/icons/hicolor/$(size)/apps
 
 icons_DATA =					\
 	gnome-panel-clock.svg			\
-	gnome-panel-fish.svg			\
 	gnome-panel-force-quit.svg		\
 	gnome-panel-hibernate.svg		\
 	gnome-panel-launcher.svg		\
diff --git a/gnome-panel/Makefile.am b/gnome-panel/Makefile.am
index 976d37f0b..0b3b3e58d 100644
--- a/gnome-panel/Makefile.am
+++ b/gnome-panel/Makefile.am
@@ -36,7 +36,6 @@ panel_sources =			\
 	panel.c			\
 	applet.c		\
 	panel-util.c		\
-	panel-run-dialog.c	\
 	panel-context-menu.c	\
 	panel-applet-frame.c	\
 	panel-applets-manager.c	\
@@ -58,7 +57,6 @@ panel_headers =			\
 	panel.h			\
 	applet.h		\
 	panel-util.h		\
-	panel-run-dialog.h	\
 	panel-context-menu.h	\
 	panel-applet-frame.h	\
 	panel-applets-manager.h	\
@@ -155,8 +153,7 @@ BUILT_SOURCES = \
 	$(NULL)
 
 ui_FILES =					\
-	gp-properties-dialog.ui \
-	panel-run-dialog.ui
+	gp-properties-dialog.ui
 
 panel-resources.c: panel.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/panel.gresource.xml)
 	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate --c-name panel $<
diff --git a/gnome-panel/panel-action-protocol.c b/gnome-panel/panel-action-protocol.c
index 97f63850a..35efabd7d 100644
--- a/gnome-panel/panel-action-protocol.c
+++ b/gnome-panel/panel-action-protocol.c
@@ -32,11 +32,9 @@
 #include "applet.h"
 #include "panel-toplevel.h"
 #include "panel-util.h"
-#include "panel-run-dialog.h"
 
 static Atom atom_gnome_panel_action            = None;
 static Atom atom_gnome_panel_action_main_menu  = None;
-static Atom atom_gnome_panel_action_run_dialog = None;
 
 static void
 menu_destroy_cb (GtkWidget   *widget,
@@ -95,13 +93,6 @@ panel_action_protocol_main_menu (GdkScreen *screen,
 	g_signal_connect (menu, "loaded", G_CALLBACK (menu_loaded_cb), panels->data);
 }
 
-static void
-panel_action_protocol_run_dialog (GdkScreen *screen,
-				  guint32    activate_time)
-{
-	panel_run_dialog_present (screen, activate_time);
-}
-
 static GdkFilterReturn
 panel_action_protocol_filter (GdkXEvent *gdk_xevent,
 			      GdkEvent  *event,
@@ -132,8 +123,6 @@ panel_action_protocol_filter (GdkXEvent *gdk_xevent,
 
 	if (atom == atom_gnome_panel_action_main_menu)
 		panel_action_protocol_main_menu (screen, xevent->xclient.data.l [1]);
-	else if (atom == atom_gnome_panel_action_run_dialog)
-		panel_action_protocol_run_dialog (screen, xevent->xclient.data.l [1]);
 	else
 		return GDK_FILTER_CONTINUE;
 
@@ -155,10 +144,6 @@ panel_action_protocol_init (void)
 		XInternAtom (GDK_DISPLAY_XDISPLAY (display),
 			     "_GNOME_PANEL_ACTION_MAIN_MENU",
 			     FALSE);
-	atom_gnome_panel_action_run_dialog =
-		XInternAtom (GDK_DISPLAY_XDISPLAY (display),
-			     "_GNOME_PANEL_ACTION_RUN_DIALOG",
-			     FALSE);
 
 	/* We'll filter event sent on non-root windows later */
 	gdk_window_add_filter (NULL, panel_action_protocol_filter, NULL);
diff --git a/gnome-panel/panel.gresource.xml b/gnome-panel/panel.gresource.xml
index e97864aed..87efe461a 100644
--- a/gnome-panel/panel.gresource.xml
+++ b/gnome-panel/panel.gresource.xml
@@ -12,6 +12,5 @@
   </gresource>
   <gresource prefix="/org/gnome/panel">
     <file compressed="true">gp-properties-dialog.ui</file>
-    <file compressed="true">panel-run-dialog.ui</file>
   </gresource>
 </gresources>
diff --git a/help/Makefile.am b/help/Makefile.am
index 8c086344e..9bd400e46 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -1,4 +1,4 @@
-DIST_SUBDIRS = clock fish
+DIST_SUBDIRS =
 SUBDIRS =
 if ENABLE_DOCUMENTATION
 SUBDIRS += $(DIST_SUBDIRS)
diff --git a/modules/Makefile.am b/modules/Makefile.am
index ee1ae0bf6..a488ae9aa 100644
--- a/modules/Makefile.am
+++ b/modules/Makefile.am
@@ -1,14 +1,8 @@
 NULL =
 
 SUBDIRS = \
-	action-button \
-	clock \
-	fish \
-	launcher \
-	menu \
 	notification-area \
 	separator \
-	status-notifier \
 	wncklet \
 	$(NULL)
 
diff --git a/po/POTFILES.in b/po/POTFILES.in
index fd6cfd5cb..b587dfa59 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,7 +1,5 @@
 # List of source files containing translatable strings.
 # Please keep this file sorted alphabetically.
-data/org.gnome.gnome-panel.applet.clock.gschema.xml
-data/org.gnome.gnome-panel.applet.fish.gschema.xml
 data/org.gnome.gnome-panel.applet.window-list.gschema.xml
 data/org.gnome.gnome-panel.applet.workspace-switcher.gschema.xml
 data/org.gnome.gnome-panel.gschema.xml
@@ -23,60 +21,14 @@ gnome-panel/panel-bindings.c
 gnome-panel/panel.c
 gnome-panel/panel-context-menu.c
 gnome-panel/panel-layout.c
-gnome-panel/panel-run-dialog.c
-gnome-panel/panel-run-dialog.ui
 gnome-panel/panel-toplevel.c
 gnome-panel/panel-util.c
 libgnome-panel/gp-initial-setup-dialog.c
 libgnome-panel/gp-module.c
-modules/action-button/gp-action-button-module.c
-modules/action-button/gp-force-quit-applet.c
-modules/action-button/gp-lock-screen-applet.c
-modules/action-button/gp-lock-screen-menu.ui
-modules/action-button/gp-logout-applet.c
-modules/action-button/gp-run-applet.c
-modules/action-button/gp-shutdown-applet.c
-modules/action-button/panel-force-quit.c
-modules/clock/calendar-client.c
-modules/clock/calendar-window.c
-modules/clock/clock-applet.c
-modules/clock/clock-location.c
-modules/clock/clock-location-entry.c
-modules/clock/clock-location-tile.c
-modules/clock/clock-module.c
-modules/clock/clock-menu.ui
-modules/clock/clock.ui
-modules/clock/clock-utils.c
-modules/fish/fish-applet.c
-modules/fish/fish-module.c
-modules/fish/fish-menu.ui
-modules/fish/fish.ui
-modules/launcher/custom-launcher-menu.ui
-modules/launcher/gp-editor.c
-modules/launcher/gp-icon-name-chooser.c
-modules/launcher/gp-icon-name-chooser.ui
-modules/launcher/gp-launcher-applet.c
-modules/launcher/gp-launcher-module.c
-modules/launcher/gp-launcher-properties.c
-modules/launcher/gp-launcher-utils.c
-modules/launcher/launcher-menu.ui
-modules/launcher/org.gnome.gnome-panel.applet.launcher.gschema.xml
-modules/menu/gp-bookmarks.c
-modules/menu/gp-lock-logout.c
-modules/menu/gp-menu-bar-applet.c
-modules/menu/gp-menu-button-applet.c
-modules/menu/gp-menu-module.c
-modules/menu/gp-menu-utils.c
-modules/menu/gp-places-menu.c
-modules/menu/gp-recent-menu.c
-modules/menu/main-menu-menu.ui
-modules/menu/menu-bar-menu.ui
-modules/menu/org.gnome.gnome-panel.applet.menu-button.gschema.xml
 modules/notification-area/na-applet.c
 modules/notification-area/na-module.c
 modules/notification-area/na-tray-manager.c
 modules/separator/separator-module.c
-modules/status-notifier/sn-module.c
 modules/wncklet/showdesktop.c
 modules/wncklet/window-list.c
 modules/wncklet/window-list-menu.ui
