diff --git a/src/net.cpp b/src/net.cpp
index 2b4b601..be721ea 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -1079,13 +1079,15 @@ void ThreadMapPort()
     struct UPNPDev * devlist = 0;
     char lanaddr[64];
 
-#ifndef UPNPDISCOVER_SUCCESS
-    /* miniupnpc 1.5 */
-    devlist = upnpDiscover(2000, multicastif, minissdpdpath, 0);
-#else
-    /* miniupnpc 1.6 */
+// see apiversions.txt in miniupnpc
+#if MINIUPNPC_API_VERSION >= 14
+    int error = 0;
+    devlist = upnpDiscover(2000, multicastif, minissdpdpath, 0, 0, 2, &error);
+#elif defined UPNPDISCOVER_SUCCESS
     int error = 0;
     devlist = upnpDiscover(2000, multicastif, minissdpdpath, 0, 0, &error);
+#elif
+    devlist = upnpDiscover(2000, multicastif, minissdpdpath, 0);
 #endif
 
     struct UPNPUrls urls;
