diff --git a/base/containers/id_map.h b/base/containers/id_map.h
index 042a21881c054..32e12baf944c8 100644
--- a/base/containers/id_map.h
+++ b/base/containers/id_map.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <limits>
 #include <iterator>
 #include <memory>
 #include <ostream>
diff --git a/base/memory/ref_counted.h b/base/memory/ref_counted.h
index 9ef94d84769df..f70e30748a389 100644
--- a/base/memory/ref_counted.h
+++ b/base/memory/ref_counted.h
@@ -7,6 +7,7 @@
 
 #include <stddef.h>
 
+#include <limits>
 #include <utility>
 
 #include "base/atomic_ref_count.h"
