--- python/cuopt/CMakeLists.txt	2026-06-10 06:05:39.000000000 +0800
+++ python/cuopt/CMakeLists.txt.new	2026-06-10 15:53:51.660905298 +0800
@@ -23,6 +23,10 @@
 set(CMAKE_CUDA_STANDARD 20)
 set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
+find_package(nvtx3 REQUIRED)
+find_package(rapids_logger REQUIRED)
+find_package(rmm REQUIRED)
+find_package(raft REQUIRED)
 find_package(cuopt "${RAPIDS_VERSION}")
 
 include(rapids-cython-core)
