diff --git a/include/hpmvs/Patch3d.h b/include/hpmvs/Patch3d.h
index e64a89d..68c9579 100644
--- a/include/hpmvs/Patch3d.h
+++ b/include/hpmvs/Patch3d.h
@@ -24,6 +24,7 @@
 #include <Eigen/Dense>
 #include <atomic>
 #include <memory>
+#include <vector>
 
 namespace mo3d {
 
diff --git a/src/hpmvs/PatchOptimizer.cpp b/src/hpmvs/PatchOptimizer.cpp
index b1ba7d0..4939d05 100644
--- a/src/hpmvs/PatchOptimizer.cpp
+++ b/src/hpmvs/PatchOptimizer.cpp
@@ -20,6 +20,7 @@
 
 #include <set>
 #include <iomanip>
+#include <numeric>
 
 // 3rd party
 #include <nlopt.hpp>
