--- python/cuvs/CMakeLists.txt	2026-02-05 05:46:31.000000000 +0800
+++ python/cuvs/CMakeLists.txt.new	2026-03-11 16:22:30.546984430 +0800
@@ -36,10 +36,16 @@
 rapids_cpm_init()
 
 # --- cuVS ---#
+find_package(nvtx3 REQUIRED)
+find_package(rapids_logger REQUIRED)
+find_package(cuco REQUIRED)
+find_package(rmm REQUIRED)
+find_package(cuco REQUIRED)
+find_package(raft REQUIRED)
 find_package(cuvs "${RAPIDS_VERSION}" REQUIRED COMPONENTS cuvs_cpp_headers cuvs cuvs_shared c_api)
 
 # --- dlpack ---#
-include(../../cpp/cmake/thirdparty/get_dlpack.cmake)
+find_package(dlpack REQUIRED)
 
 # ensure Cython targets can find dlpack headers (these do not come installed with with cuVS)
 target_include_directories(cuvs::cuvs INTERFACE "$<BUILD_INTERFACE:${DLPACK_INCLUDE_DIR}>")
