diff -aur otrtool.orig/Makefile otrtool/Makefile
--- otrtool.orig/Makefile	2015-10-28 16:57:39.761881708 +0100
+++ otrtool/Makefile	2015-10-28 16:58:09.018201634 +0100
@@ -52,6 +52,8 @@
 	$(RM) $(OBJS) $(MAIN) $(MAIN).1.gz
 
 install: $(MAIN) $(MAIN).1.gz
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
 	install -m 0755 $(MAIN) $(DESTDIR)$(PREFIX)/bin
 	install -m 0644 $(MAIN).1.gz $(DESTDIR)$(PREFIX)/share/man/man1
 
