--- mlmmj-1.3.0/include/mlmmj.h	2013-01-11 09:40:19.000000000 +0100
+++ mlmmj-1.3.0/include/mlmmj.h	2021-06-24 15:42:53.799363500 +0200
@@ -81,7 +81,7 @@
 	SUB_NONE /* For when an address is not subscribed at all */
 };
 
-char *subtype_strs[7]; /* count matches enum above; defined in subscriberfuncs.c */
+extern char *subtype_strs[7]; /* count matches enum above; defined in subscriberfuncs.c */
 
 enum subreason {
 	SUB_REQUEST,
@@ -92,7 +92,7 @@
 	SUB_SWITCH
 };
 
-char * subreason_strs[6]; /* count matches enum above; defined in subscriberfuncs.c */
+extern char * subreason_strs[6]; /* count matches enum above; defined in subscriberfuncs.c */
 
 void print_version(const char *prg);
 
