diff --git a/lib/Slic3r.pm b/lib/Slic3r.pm
index 5f4357a9..e37a5e17 100644
--- a/lib/Slic3r.pm
+++ b/lib/Slic3r.pm
@@ -49,7 +49,7 @@ warn "Running Slic3r under Perl 5.16 is neither supported nor recommended\n"
 
 use FindBin;
 # Path to the images.
-our $var = sub { decode_path($FindBin::Bin) . "/var/" . $_[0] };
+our $var = sub { '/usr/share/slic3r-prusa3d/' . $_[0] };
 
 use Moo 1.003001;
 
diff --git a/slic3r.pl b/slic3r.pl
index c1371472..6cd50804 100755
--- a/slic3r.pl
+++ b/slic3r.pl
@@ -5,7 +5,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/lib";
+    use lib "$FindBin::Bin/lib", "/usr/lib/slic3r-prusa3d/";
 }
 
 use File::Basename qw(basename);
