diff -ru msc-generator-4.6/src/csh.cpp msc-generator-4.6.patched/src/csh.cpp
--- msc-generator-4.6/src/csh.cpp	2015-07-19 12:49:59.390042073 +0200
+++ msc-generator-4.6.patched/src/csh.cpp	2015-07-19 12:49:10.900427416 +0200
@@ -440,7 +440,7 @@
             byte_offset += UTF8TrailingBytes(input[byte_offset]) + 1;
             char_offset++;
         }
-        ret.append(input+byte_from, input+std::min(byte_offset, len));
+        ret.append(input+byte_from, input+std::min<size_t>(byte_offset, len));
         if (byte_offset >= len) break; // color codes past the end of the string
         ret.append(i->second.Print());
         lastState = i->second;
