--- xyplayer_orig/xyplayer/util.py	2015-08-04 10:43:03.551022240 -0400
+++ xyplayer/xyplayer/util.py	2015-08-04 10:43:27.004469117 -0400
@@ -1,8 +1,8 @@
 # -*- coding = utf-8 -*-
 import os, re, time
 from urllib import parse, request
-from mutagenx.mp3 import MP3
-from mutagenx.easyid3 import EasyID3
+from mutagen.mp3 import MP3
+from mutagen.easyid3 import EasyID3
 from PyQt4.QtCore import QTime
 
 def read_music_info(path):
