diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc	1970-01-01 01:00:00.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc"	2023-05-18 10:19:04.384265100 +0200
@@ -0,0 +1,49 @@
+#! /bin/make
+##############################################
+# common definitions for wx and fltk conf tool
+##############################################
+#-----------------------------------------
+TARGET:=$(shell basename `pwd`)
+
+#-----------------------------------------
+# You may need to adapt this path for your system.
+# If $WX_CONF_TOOL is set by the environment then it is just used.
+# Otherwise, by default we use "which" except if a special purpose-built wxWidgets exists.
+#-----------------------------------------
+ifeq ($(strip $(WX_CONF_TOOL)),)
+	# check for specially-built static wxWidgets
+	WX_SPECIAL_CONF_TOOL:=$(shell ls /opt/wx/2.6/$(TARGET)/bin/wx-config 2>/dev/null)
+	ifneq ($(WX_SPECIAL_CONF_TOOL),)
+		WX_CONF_TOOL:=$(WX_SPECIAL_CONF_TOOL)
+	else
+		# use the system's wxWidgets
+		WX_CONF_TOOL:=$(shell which wx-config-3.0 2>/dev/null)
+		ifeq ($(WX_CONF_TOOL),)
+			WX_CONF_TOOL:=$(shell which wx-config-2.8 2>/dev/null)
+			ifeq ($(WX_CONF_TOOL),)
+				WX_CONF_TOOL:=$(shell which wx-config-2.6 2>/dev/null)
+				ifeq ($(WX_CONF_TOOL),)
+					WX_CONF_TOOL:=$(shell which wx-config 2>/dev/null)
+				endif
+			endif
+		endif
+	endif
+endif
+
+#-----------------------------------------
+# You may need to adapt this path for your system.
+# By default uses "which" except when crosscompiling for 32-bit on 64-bit systems
+#-----------------------------------------
+ifeq ($(strip $(FLTK_CONF_TOOL)),)
+	ifeq ($(filter $(TARGET),i686 x86),$(TARGET))
+		FLTK_CONF_TOOL=$(shell which fltk-config 2>/dev/null)
+	else
+		ifeq ($(HOSTTYPE),x86_64)
+			ifeq ($(TARGET),x86)
+				FLTK_CONF_TOOL=$(WORK)/tools/fltk32/bin/fltk-config
+			else
+				FLTK_CONF_TOOL=$(shell which fltk-config 2>/dev/null)
+			endif
+		endif
+	endif
+endif
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.samp.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.samp.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.samp.inc	2023-03-01 19:17:22.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.samp.inc"	2023-05-18 10:24:49.619089801 +0200
@@ -94,49 +94,6 @@
 	-lm \
 	-lpthread
 
-#-----------------------------------------
-# You may need to adapt this path for your system.
-# If $WX_CONF_TOOL is set by the environment then it is just used.
-# Otherwise, by default we use "which" except if a special purpose-built wxWidgets exists.
-#-----------------------------------------
-ifeq ($(strip $(WX_CONF_TOOL)),)
-	# check for specially-built static wxWidgets
-	WX_SPECIAL_CONF_TOOL:=$(shell ls /opt/wx/2.6/$(TARGET)/bin/wx-config 2>/dev/null)
-	ifneq ($(WX_SPECIAL_CONF_TOOL),)
-		WX_CONF_TOOL:=$(WX_SPECIAL_CONF_TOOL)
-	else
-		# use the system's wxWidgets
-		WX_CONF_TOOL:=$(shell which wx-config-3.0 2>/dev/null)
-		ifeq ($(WX_CONF_TOOL),)
-			WX_CONF_TOOL:=$(shell which wx-config-2.8 2>/dev/null)
-			ifeq ($(WX_CONF_TOOL),)
-				WX_CONF_TOOL:=$(shell which wx-config-2.6 2>/dev/null)
-				ifeq ($(WX_CONF_TOOL),)
-					WX_CONF_TOOL:=$(shell which wx-config 2>/dev/null)
-				endif
-			endif
-		endif
-	endif
-	ADD_LIBS=
-endif
-
-#-----------------------------------------
-# You may need to adapt this path for your system.
-# By default uses "which" except when crosscompiling for 32-bit on 64-bit systems
-#-----------------------------------------
-ifeq ($(strip $(FLTK_CONF_TOOL)),)
-	ifeq ($(filter $(TARGET),i686 x86),$(TARGET))
-		FLTK_CONF_TOOL=$(shell which fltk-config 2>/dev/null)
-	else
-		ifeq ($(HOSTTYPE),x86_64)
-			ifeq ($(TARGET),x86)
-				FLTK_CONF_TOOL=$(WORK)/tools/fltk32/bin/fltk-config
-			else
-				FLTK_CONF_TOOL=$(shell which fltk-config 2>/dev/null)
-			endif
-		endif
-	endif
-endif
 
 ifeq ($(strip $(WX_CONF_TOOL)),)
 	ifeq ($(MODULE_NAME),$(filter $(MODULE_NAME),PropViewFrame DeviceConfigureFrame IPConfigureFrame MainFrame))
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvDeviceConfigure/Makefile.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvDeviceConfigure/Makefile.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvDeviceConfigure/Makefile.inc	2023-03-01 19:17:22.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvDeviceConfigure/Makefile.inc"	2023-05-18 10:07:22.292821873 +0200
@@ -45,10 +45,10 @@
 LOC_LIB_PATH=$(WORK)/lib/$(TARGET)
 
 #-------------------------------------------
-# 'WX_CONF_TOOL' will be set in Makefile.samp.inc
-#-------------------------------------------
+include $(ROOT_PATH)/Makefile.conf-tools.inc
 WXRELEASE=$(shell $(WX_CONF_TOOL) --release)
-WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs)
+WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs std, webview)
+# WXOPTIONALLIBFLAGS=$(shell $(WX_CONF_TOOL) --optional-libs webview)
 WXCXXFLAGS=$(shell $(WX_CONF_TOOL) --cxxflags)
 
 #-------------------------------------------
@@ -62,6 +62,7 @@
 #-------------------------------------------
 LDOPT_1 =				\
 	$(WXLIBFLAGS)		\
+	# $(WXOPTIONALLIBFLAGS)
 
 #-------------------------------------------
 # Local typedefs warning since gcc version 4.8 (wxWidgets code)
@@ -85,5 +86,3 @@
 
 #-------------------------------------------
 include $(ROOT_PATH)/Makefile.samp.inc
-
-#-------------------------------------------
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvIPConfigure/Makefile.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvIPConfigure/Makefile.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvIPConfigure/Makefile.inc	2023-03-01 19:17:22.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvIPConfigure/Makefile.inc"	2023-05-18 10:07:27.806166644 +0200
@@ -35,10 +35,10 @@
 LOC_LIB_PATH=$(WORK)/lib/$(TARGET)
 
 #-------------------------------------------
-# 'WX_CONF_TOOL' will be set in Makefile.samp.inc
-#-------------------------------------------
+include $(ROOT_PATH)/Makefile.conf-tools.inc
 WXRELEASE=$(shell $(WX_CONF_TOOL) --release)
-WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs)
+WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs std, webview) 
+# WXOPTIONALLIBFLAGS=$(shell $(WX_CONF_TOOL) --optional-libs webview)
 WXCXXFLAGS=$(shell $(WX_CONF_TOOL) --cxxflags)
 
 #-------------------------------------------
@@ -51,6 +51,7 @@
 #-------------------------------------------
 LDOPT_1 =				\
 	$(WXLIBFLAGS)			\
+	# $(WXOPTIONALLIBFLAGS)
 
 #-------------------------------------------
 # Local typedefs warning since gcc version 4.8 (wxWidgets code)
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvPropView/Makefile.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvPropView/Makefile.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvPropView/Makefile.inc	2023-03-01 19:17:22.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/mvPropView/Makefile.inc"	2023-05-18 10:07:33.799512410 +0200
@@ -66,11 +66,10 @@
 LOC_LIB_PATH=$(WORK)/lib/$(TARGET)
 
 #-------------------------------------------
-# 'WX_CONF_TOOL' will be set in Makefile.samp.inc
-#-------------------------------------------
+include $(ROOT_PATH)/Makefile.conf-tools.inc
 WXRELEASE=$(shell $(WX_CONF_TOOL) --release)
-WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs)
-WXOPTIONALLIBFLAGS=$(shell $(WX_CONF_TOOL) --optional-libs propgrid)
+WXLIBFLAGS=$(shell $(WX_CONF_TOOL) --libs std, propgrid)
+# WXOPTIONALLIBFLAGS=$(shell $(WX_CONF_TOOL) --optional-libs propgrid)
 WXCXXFLAGS=$(shell $(WX_CONF_TOOL) --cxxflags)
 
 #-------------------------------------------
@@ -84,7 +83,7 @@
 LDOPT_1 = \
 	-fopenmp \
 	$(WXLIBFLAGS) \
-	$(WXOPTIONALLIBFLAGS)
+	# $(WXOPTIONALLIBFLAGS)
 
 #-------------------------------------------
 # Local typedefs warning since gcc version 4.8 (wxWidgets code)
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/ContinuousCaptureFLTK/Makefile.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/ContinuousCaptureFLTK/Makefile.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/ContinuousCaptureFLTK/Makefile.inc	2023-03-01 19:17:21.000000000 +0100
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/ContinuousCaptureFLTK/Makefile.inc"	2023-05-18 13:10:39.759365445 +0200
@@ -37,4 +37,5 @@
 #		-DMALLOC_TRACE
 
 #-------------------------------------------
+include $(ROOT_PATH)/Makefile.conf-tools.inc
 include $(ROOT_PATH)/Makefile.samp.inc
diff -ruN src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc"
--- src/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc	2023-05-21 14:35:32.782053830 +0200
+++ "src copy/mvIMPACT_Acquire-x86_64-2.49.0/apps/Makefile.conf-tools.inc"	2023-05-21 14:45:57.452846132 +0200
@@ -4,6 +4,8 @@
 ##############################################
 #-----------------------------------------
 TARGET:=$(shell basename `pwd`)
+OSTYPE := $(shell uname | cut -d _ -f 1 | tr [:upper:] [:lower:])
+HOSTTYPE := $(shell uname -m)
 
 #-----------------------------------------
 # You may need to adapt this path for your system.
 