--- python/wrapper.cpp.old	2024-08-08 14:56:23.226590879 +0100
+++ python/wrapper.cpp	2024-08-08 14:57:14.154373405 +0100
@@ -96,7 +96,7 @@
              mdspan_t<const T, 2> _x(x.data(), x.shape(0), x.shape(1));
              return as_nbarrayp(self.tabulate(n, _x));
            })
-      .def("__eq__", &FiniteElement<T>::operator==, nb::sig("def __eq__(self, arg: object, /) -> bool"))
+      .def("__eq__", &FiniteElement<T>::operator==)
       .def("hash", &FiniteElement<T>::hash)
       .def("permute_subentity_closure",
            [](const FiniteElement<T>& self,
