--- ./build-root/build-vpp-native/external/src-nasm/include/nasmlib.h.orig	2018-11-01 18:00:15.456221458 -0400
+++ ./build-root/build-vpp-native/external/src-nasm/include/nasmlib.h	2018-11-01 18:02:15.617516579 -0400
@@ -188,10 +188,8 @@
 int64_t readstrnum(char *str, int length, bool *warn);
 
 /*
- * seg_init: Initialise the segment-number allocator.
  * seg_alloc: allocate a hitherto unused segment number.
  */
-int32_t pure_func seg_init(void);
 int32_t pure_func seg_alloc(void);
 
 /*
