diff -aur ncbi-vdb-2.5.7.pristine/build/Makefile.install ncbi-vdb-2.5.7.new/build/Makefile.install
--- ncbi-vdb-2.5.7.pristine/build/Makefile.install	2016-01-26 21:10:12.713909719 +0000
+++ ncbi-vdb-2.5.7.new/build/Makefile.install	2016-01-26 21:18:28.630558667 +0000
@@ -66,11 +66,7 @@
 PROFILE_FILE = $(ROOT)/etc/profile.d/ncbi-vdb
 KONFIG_DIR = $(ROOT)/etc/ncbi/
 
-ifeq (linux, $(OS))
-    ifeq (0, $(shell id -u))
-        LINUX_ROOT = true
-    endif
-endif
+LINUX_ROOT = false
 
 install: copylibs copyincludes 
 ifeq (true, $(LINUX_ROOT))
diff -aur ncbi-vdb-2.5.7.pristine/setup/install.perl ncbi-vdb-2.5.7.new/setup/install.perl
--- ncbi-vdb-2.5.7.pristine/setup/install.perl	2016-01-26 21:10:12.810576379 +0000
+++ ncbi-vdb-2.5.7.new/setup/install.perl	2016-01-26 21:11:20.967240611 +0000
@@ -81,11 +81,9 @@
 prepare();
 
 my $LINUX_ROOT;
-++$LINUX_ROOT if (linux_root());
 my $ROOT = '';
 if ($OPT{root}) {
     $ROOT = "$OPT{root}/root";
-    ++$LINUX_ROOT;
     foreach ("$ROOT/usr/include", "$ROOT/etc/profile.d") {
         unless (-e $_) {
             print "mkdir -p $_... ";
