--- src/debug/debug_gui.cpp
+++ src/debug/debug_gui.cpp
@@ -69,7 +69,7 @@
 	if (logBuffPos!=logBuff.end()) {
 		logBuffPos=logBuff.end();
 		DEBUG_RefreshPage(0);
-//		mvwprintw(dbg.win_out,dbg.win_out->_maxy-1, 0, "");
+//		mvwprintw(dbg.win_out,getmaxy(dbg.win_out)-1, 0, "");
 	}
 	logBuff.push_back(buf);
 	if (logBuff.size() > MAX_LOG_BUFFER)
@@ -146,15 +146,15 @@
 		attrset(COLOR_PAIR(PAIR_BLACK_BLUE));
 	}
 	/* Show the Register bar */
-	mvaddstr(dbg.win_reg->_begy-1,0, "---(Register Overview                   )---");
+	mvaddstr(getbegy(dbg.win_reg)-1,0, "---(Register Overview                   )---");
 	/* Show the Data Overview bar perhaps with more special stuff in the end */
-	mvaddstr(dbg.win_data->_begy-1,0,"---(Data Overview   Scroll: page up/down)---");
+	mvaddstr(getbegy(dbg.win_data)-1,0,"---(Data Overview   Scroll: page up/down)---");
 	/* Show the Code Overview perhaps with special stuff in bar too */
-	mvaddstr(dbg.win_code->_begy-1,0,"---(Code Overview   Scroll: up/down     )---");
+	mvaddstr(getbegy(dbg.win_code)-1,0,"---(Code Overview   Scroll: up/down     )---");
 	/* Show the Variable Overview bar */
-	mvaddstr(dbg.win_var->_begy-1,0, "---(Variable Overview                   )---");
+	mvaddstr(getbegy(dbg.win_var)-1,0, "---(Variable Overview                   )---");
 	/* Show the Output OverView */
-	mvaddstr(dbg.win_out->_begy-1,0, "---(OutPut/Input    Scroll: home/end    )---");
+	mvaddstr(getbegy(dbg.win_out)-1,0, "---(OutPut/Input    Scroll: home/end    )---");
 	attrset(0);
 }
