diff --git a/ndless-sdk/lib/zehn_loader.tns b/ndless-sdk/lib/zehn_loader.tns
new file mode 120000
index 0000000..64c36ae
--- /dev/null
+++ b/ndless-sdk/lib/zehn_loader.tns
@@ -0,0 +1 @@
+../tools/zehn_loader/zehn_loader.tns
\ No newline at end of file
diff --git a/ndless-sdk/lib/zehn_loader_compressed.tns b/ndless-sdk/lib/zehn_loader_compressed.tns
new file mode 120000
index 0000000..55c4a8c
--- /dev/null
+++ b/ndless-sdk/lib/zehn_loader_compressed.tns
@@ -0,0 +1 @@
+../tools/zehn_loader/zehn_loader_compressed.tns
\ No newline at end of file
