--- netextender.old/NEService.service	2026-04-08 22:55:10.000000000 -0400
+++ netextender.new/NEService.service	2026-06-18 16:36:11.700128650 -0400
@@ -2,9 +2,9 @@
 Description=SonicWall NetExtender Service
 
 [Service]
-PIDFile=/var/run/NEService.pid
-ExecStartPre=/bin/rm -f /var/run/NEService.pid
-ExecStart=/usr/local/netextender/NEService
+PIDFile=/run/NEService.pid
+ExecStartPre=/bin/rm -f /run/NEService.pid
+ExecStart=/opt/netextender/NEService
 Restart=on-failure
 KillMode=process
 
