Only in netkit-telnet-0.17/telnetd: telnetd
diff -ru netkit-telnet-0.17.orig/telnetd/utility.c netkit-telnet-0.17/telnetd/utility.c
--- netkit-telnet-0.17.orig/telnetd/utility.c	2017-09-05 08:20:17.150075017 +0300
+++ netkit-telnet-0.17/telnetd/utility.c	2017-09-05 08:35:37.659899201 +0300
@@ -39,6 +39,7 @@
 
 #define PRINTOPTIONS
 
+#define _GNU_SOURCE
 #include <stdarg.h>
 #include <sys/utsname.h>
 #include <sys/time.h>
@@ -55,7 +56,7 @@
 	size_t len;
 };
 
-static struct buflist head = { next: &head, buf: 0, len: 0 };
+static struct buflist head = { .next = &head, .buf = 0, .len = 0 };
 static struct buflist *tail = &head;
 static size_t skip;
 static int trailing;
@@ -1347,7 +1348,7 @@
 void
 netopen() {
 	static const cookie_io_functions_t funcs = {
-		read: 0, write: netwrite, seek: 0, close: 0
+		.read = 0, .write = netwrite, .seek = 0, .close = 0
 	};
 
 	netfile = fopencookie(0, "w", funcs);
Only in netkit-telnet-0.17/telnetd: utility.o
diff -ru netkit-telnet-0.17.orig/telnetlogin/telnetlogin.c netkit-telnet-0.17/telnetlogin/telnetlogin.c
--- netkit-telnet-0.17.orig/telnetlogin/telnetlogin.c	2017-09-05 09:13:42.675718525 +0300
+++ netkit-telnet-0.17/telnetlogin/telnetlogin.c	2017-09-05 09:13:47.395788056 +0300
@@ -217,7 +217,7 @@
     * but, should we insist that ruid==nobody?
     */
 
-#ifdef debian
+#if 1
    /*
     * Debian's /bin/login doesn't work properly unless we're really root.
     */
