diff -ruN pkg/cms-germany-git/usr/lib/python3.10/site-packages/cms/grading/ParameterTypes.py /usr/lib/python3.10/site-packages/cms/grading/ParameterTypes.py
--- pkg/cms-germany-git/usr/lib/python3.10/site-packages/cms/grading/ParameterTypes.py	2023-02-03 16:35:06.000000000 +0100
+++ /usr/lib/python3.10/site-packages/cms/grading/ParameterTypes.py	2023-02-03 17:05:45.781269085 +0100
@@ -29,7 +29,7 @@
 
 from abc import ABCMeta, abstractmethod
 
-from jinja2 import Markup
+from markupsafe import Markup
 
 from cms.server.jinja2_toolbox import GLOBAL_ENVIRONMENT
