--- freenginx-1.30.0/src/core/nginx.c	2026-04-14 05:23:27.000000000 -0300
+++ freenginx-1.30.0_patched/src/core/nginx.c	2026-04-14 20:29:08.838726028 -0300
@@ -399,7 +399,7 @@
 static void
 ngx_show_version_info(void)
 {
-    ngx_write_stderr("nginx version: " NGINX_VER_BUILD NGX_LINEFEED);
+    ngx_write_stderr(NGINX_NAME " version: " NGINX_VER_BUILD NGX_LINEFEED);
 
     if (ngx_show_help) {
         ngx_write_stderr(
