diff --git a/configure b/configure
index 1582758..ee9af70 100755
--- a/configure
+++ b/configure
@@ -16720,7 +16720,7 @@ _ACEOF
 	*) test -f "$ac_im_incroot/X11/Xos.h" && ac_x_includes=$ac_im_incroot;;
     esac
     case $ac_im_usrlibdir in
-	/usr/lib | /usr/lib64 | /lib | /lib64) ;;
+	/usr/lib | /usr/lib | /lib | /lib) ;;
 	*) test -d "$ac_im_usrlibdir" && ac_x_libraries=$ac_im_usrlibdir ;;
     esac
   fi
@@ -21493,7 +21493,7 @@ fi
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/lib/vice"
        if false; then
   ALTERNATE_DOCDIR_TRUE=
   ALTERNATE_DOCDIR_FALSE='#'
diff --git a/configure.ac b/configure.ac
index 59f03b6..f0b043d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2257,12 +2257,7 @@ if test x"$enable_lame" != "xno" ; then
 
 fi
 
-AC_PATH_PROGS(FCCACHE, fc-cache, false)
-if test x"$FCCACHE" = "xfalse"; then
-  AM_CONDITIONAL(HAVE_FC_CACHE, false)
-else
-  AM_CONDITIONAL(HAVE_FC_CACHE, true)
-fi
+AM_CONDITIONAL(HAVE_FC_CACHE, false)
 
 if test x"$is_amigaos4" = "xyes"; then
   if test x"$enable_sdlui" != "xyes"; then
@@ -3729,7 +3724,7 @@ case "$host_os" in
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/lib/vice"
       AM_CONDITIONAL(ALTERNATE_DOCDIR, false)
     else
       VICEDIR="$PREFIX/lib/vice"
diff --git a/configure.proto b/configure.proto
index 59f03b6..f0b043d 100644
--- a/configure.proto
+++ b/configure.proto
@@ -2257,12 +2257,7 @@ if test x"$enable_lame" != "xno" ; then
 
 fi
 
-AC_PATH_PROGS(FCCACHE, fc-cache, false)
-if test x"$FCCACHE" = "xfalse"; then
-  AM_CONDITIONAL(HAVE_FC_CACHE, false)
-else
-  AM_CONDITIONAL(HAVE_FC_CACHE, true)
-fi
+AM_CONDITIONAL(HAVE_FC_CACHE, false)
 
 if test x"$is_amigaos4" = "xyes"; then
   if test x"$enable_sdlui" != "xyes"; then
@@ -3729,7 +3724,7 @@ case "$host_os" in
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/lib/vice"
       AM_CONDITIONAL(ALTERNATE_DOCDIR, false)
     else
       VICEDIR="$PREFIX/lib/vice"
