diff -U3 -r a/g3/g3cat.c b/g3/g3cat.c
--- a/g3/g3cat.c	2005-02-27 14:03:37.000000000 -0500
+++ b/g3/g3cat.c	2018-01-13 22:38:01.598767655 -0500
@@ -43,9 +43,6 @@
 static int b_written = 0;		/* bytes of a line already */
 					/* written */
 
-#ifdef __GNUC__
-inline
-#endif
 void putcode _P2( (code, len), int code, int len )
 {
 #ifdef DEBUG
@@ -66,9 +63,6 @@
     }
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void putwhitespan _P1( (l), int l )
 {
     if ( l >= 64 )
@@ -97,9 +91,6 @@
     putcode( t_white[l].bit_code, t_white[l].bit_length );
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void putblackspan _P1( (l), int l )
 {
     if ( l >= 64 )
@@ -128,9 +119,6 @@
     putcode( t_black[l].bit_code, t_black[l].bit_length );
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void puteol _P0( void )			/* write byte-aligned EOL */
 {
     static int last_buflen = 0;
diff -U3 -r a/g3/pbm2g3.c b/g3/pbm2g3.c
--- a/g3/pbm2g3.c	1998-05-07 07:32:04.000000000 -0400
+++ b/g3/pbm2g3.c	2018-01-13 22:38:11.742884611 -0500
@@ -39,9 +39,6 @@
 
 static int out_byte_tab[ 256 ];			/* for g3 byte reversal */
 
-#ifdef __GNUC__
-inline
-#endif
 void putcode _P2( (code, len), int code, int len )
 {
     out_data |= ( code << out_hibit );
@@ -59,18 +56,12 @@
     }
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void puteol _P0( void )			/* write byte-aligned EOL */
 {
     if ( byte_align ) while( out_hibit != 4 ) putcode( 0, 1 );
     putcode( 0x800, 12 );
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void putwhitespan _P1( (l), int l )
 {
     if ( l >= 64 )
@@ -99,9 +90,6 @@
     putcode( t_white[l].bit_code, t_white[l].bit_length );
 }
 
-#ifdef __GNUC__
-inline
-#endif
 void putblackspan _P1( (l), int l )
 {
     if ( l >= 64 )
