diff --git a/configure.ac b/configure.ac
index 103c3fa2..12a90e97 100644
--- a/configure.ac
+++ b/configure.ac
@@ -763,7 +763,7 @@ AC_SUBST(JPEGXL_LIBS)
 #  Markdown support
 # ----------------------------------------------------------------------
 
-AM_CONDITIONAL(HAVE_MARKDOWN, [ "$(command -v markdown)" ])
+AM_CONDITIONAL(HAVE_MARKDOWN, [ FALSE ])
 
 # _NL_TIME_FIRST_WEEKDAY support
 # note that it is an enum and not a define
