diff --git setup_support.py setup_support.py
index 102f8191..9ae009b6 100644
--- setup_support.py
+++ setup_support.py
@@ -234,6 +234,8 @@ class SetupApp(object):
             config.save_data()
 
         for gpr_args, gpr_vars in self.variants(config, 'build'):
+            if args.no_rpath:
+                gpr_args.append('-R')
             config.gprbuild(
                 self.project,
                 *gpr_args,
@@ -286,6 +288,8 @@ class SetupApp(object):
         self.build_cmd = self.parser.add_parser('build',
                                                 help='build %s' % self.name)
         Config.add_arguments(self.build_cmd)
+        self.build_cmd.add_argument('--no-rpath', '-R', action='store_true',
+                                    help='do not use rpath')
         self.build_cmd.set_defaults(command=self.build)
 
         # Clean command
