diff --git i/lib/scm_breeze.sh w/lib/scm_breeze.sh
index 89eeded..05969b0 100644
--- i/lib/scm_breeze.sh
+++ w/lib/scm_breeze.sh
@@ -106,19 +106,6 @@ find_binary() {
 export GIT_BINARY=$(find_binary git)

 # Updates SCM Breeze from GitHub.
-update_scm_breeze() {
-  currDir=$PWD
-  cd "$scmbDir"
-  oldHEAD=$(git rev-parse HEAD 2> /dev/null)
-  git pull origin main
-  # Reload latest version of '_create_or_patch_scmbrc' function
-  source "$scmbDir/lib/scm_breeze.sh"
-  _create_or_patch_scmbrc $oldHEAD
-  # Reload SCM Breeze
-  source "$scmbDir/scm_breeze.sh"
-  cd "$currDir"
-}
-
 # Create '~/.*.scmbrc' files, or attempt to patch them if passed a previous revision
 _create_or_patch_scmbrc() {
   patchfile=$(mktemp -t tmp.XXXXXXXXXX)
@@ -145,6 +132,3 @@ _create_or_patch_scmbrc() {
     fi
   done
 }
-
-# Update ~/.scmbrc, ~/.git.scmbrc, etc. from latest commit
-alias update_scmbrc_from_latest_commit="_create_or_patch_scmbrc HEAD~"
