diff -aur package.pristine/TWAIN_DSM/src/CMakeLists.txt package.new/TWAIN_DSM/src/CMakeLists.txt
--- package.pristine/TWAIN_DSM/src/CMakeLists.txt	2016-05-28 17:58:21.374985878 +0200
+++ package.new/TWAIN_DSM/src/CMakeLists.txt	2016-05-28 17:59:16.161254474 +0200
@@ -14,9 +14,9 @@
 #add definitions, compiler switches, etc.
 IF(APPLE)
 	SET(CMAKE_OSX_ARCHITECTURES "i386;x86_64")
-	ADD_DEFINITIONS(-Wall -Wextra -Werror -isysroot /Developer/SDKs/MacOSX10.6.sdk -mmacosx-version-min=10.6 -fexceptions -fPIC)
+	ADD_DEFINITIONS(-Wall -Wextra -isysroot /Developer/SDKs/MacOSX10.6.sdk -mmacosx-version-min=10.6 -fexceptions -fPIC)
 ELSE()
-	ADD_DEFINITIONS(-Wall -Wextra -Werror)
+	ADD_DEFINITIONS(-Wall -Wextra)
 ENDIF()
 
 #let's not be sharing our symbols...
