--- tpm-tools-1.3.7/configure	2011-09-07 16:48:29.000000000 +0200
+++ tpm-tools-1.3.7_NEW/configure	2012-01-06 14:59:52.970136356 +0100
@@ -13037,7 +13037,7 @@
 fi
 
 
-CFLAGS="$CFLAGS -Wall -Werror -Wreturn-type -Wsign-compare"
+CFLAGS="$CFLAGS -Wall -Wreturn-type -Wsign-compare"
 
 ac_config_files="$ac_config_files ./Makefile po/Makefile.in m4/Makefile include/Makefile lib/Makefile src/Makefile src/tpm_mgmt/Makefile src/cmds/Makefile src/data_mgmt/Makefile dist/tpm-tools.spec man/Makefile man/man1/Makefile man/man3/Makefile man/man8/Makefile"
 
