diff --git a/gtk/gtkimcontextsimple.c b/gtk/gtkimcontextsimple.c
index a7ac95e47c..237ddc0e72 100644
--- a/gtk/gtkimcontextsimple.c
+++ b/gtk/gtkimcontextsimple.c
@@ -1115,7 +1115,7 @@ gtk_im_context_simple_get_preedit_string (GtkIMContext   *context,
                   gunichar ch;
                   gboolean need_space;

-                  if (is_dead_key (priv->compose_buffer[i]))
+                  if (FALSE && is_dead_key (priv->compose_buffer[i]))
                     {
                       ch = dead_key_to_unicode (priv->compose_buffer[i], &need_space);
                       if (ch)
