diff -uNrp nouveau-orig/dispnv50/disp.c nouveau-mod/dispnv50/disp.c
--- a/dispnv50/disp.c	2026-06-12 01:03:26.000000000 +0300
+++ b/dispnv50/disp.c	2026-08-29 12:20:10.200475751 +0300
@@ -228,6 +228,10 @@ MODULE_PARM_DESC(kms_vram_pushbuf, "Plac
 static int nv50_dmac_vram_pushbuf = -1;
 module_param_named(kms_vram_pushbuf, nv50_dmac_vram_pushbuf, int, 0400);
 
+MODULE_PARM_DESC(vblank_continuous, "Keep vblank IRQs continuously enabled for high-refresh displays (default: 0)");
+static int nouveau_vblank_continuous = 0;
+module_param_named(vblank_continuous, nouveau_vblank_continuous, int, 0400);
+
 int
 nv50_dmac_create(struct nouveau_drm *drm,
 		 const s32 *oclass, u8 head, void *data, u32 size, s64 syncbuf,
@@ -353,8 +357,7 @@ nv50_outp_atomic_check_view(struct drm_e
 			 * refresh rate)
 			 */
 			if (mode->hdisplay == native_mode->hdisplay &&
-			    mode->vdisplay == native_mode->vdisplay &&
-			    mode->type & DRM_MODE_TYPE_DRIVER)
+			    mode->vdisplay == native_mode->vdisplay)
 				break;
 			mode = native_mode;
 			asyc->scaler.full = true;
@@ -3012,7 +3015,7 @@ nv50_display_create(struct drm_device *d
 	}
 
 	/* Disable vblank irqs aggressively for power-saving, safe on nv50+ */
-	dev->vblank_disable_immediate = true;
+	dev->vblank_disable_immediate = !nouveau_vblank_continuous;
 
 	nv50_audio_component_init(drm);
 
diff -uNrp nouveau-orig/nouveau_backlight.c nouveau-mod/nouveau_backlight.c
--- a/nouveau_backlight.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nouveau_backlight.c	2026-08-29 12:20:10.188748787 +0300
@@ -344,10 +344,7 @@ nouveau_backlight_init(struct drm_connec
 		goto fail_alloc;
 	}
 
-	if (!nouveau_acpi_video_backlight_use_native()) {
-		NV_INFO(drm, "Skipping nv_backlight registration\n");
-		goto fail_alloc;
-	}
+	/* Always register nv_backlight to allow daemon-based sync workarounds */
 
 	if (!nouveau_get_backlight_name(backlight_name, bl)) {
 		NV_ERROR(drm, "Failed to retrieve a unique name for the backlight interface\n");
diff -uNrp nouveau-orig/nvkm/subdev/clk/base.c nouveau-mod/nvkm/subdev/clk/base.c
--- a/nvkm/subdev/clk/base.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nvkm/subdev/clk/base.c	2026-08-29 12:20:10.188885186 +0300
@@ -282,7 +282,8 @@ nvkm_pstate_prog(struct nvkm_clk *clk, i
 
 	nvkm_pcie_set_link(pci, pstate->pcie_speed, pstate->pcie_width);
 
-	if (fb && fb->ram && fb->ram->func->calc) {
+	bool fermi_mem_reclock = nvkm_boolopt(subdev->device->cfgopt, "NvFermiMemReclock", false);
+	if (fb && fb->ram && fb->ram->func->calc && (subdev->device->card_type != NV_C0 || fermi_mem_reclock)) {
 		struct nvkm_ram *ram = fb->ram;
 		int khz = pstate->base.domain[nv_clk_src_mem];
 		do {
@@ -291,6 +292,8 @@ nvkm_pstate_prog(struct nvkm_clk *clk, i
 				ret = ram->func->prog(ram);
 		} while (ret > 0);
 		ram->func->tidy(ram);
+		if (ret < 0)
+			nvkm_error(subdev, "failed to reclock memory to %d kHz: %d\n", khz, ret);
 	}
 
 	return nvkm_cstate_prog(clk, pstate, NVKM_CLK_CSTATE_HIGHEST);
diff -uNrp nouveau-orig/nvkm/subdev/clk/gf100.c nouveau-mod/nvkm/subdev/clk/gf100.c
--- a/nvkm/subdev/clk/gf100.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nvkm/subdev/clk/gf100.c	2026-08-29 12:20:10.189273994 +0300
@@ -329,9 +329,6 @@ gf100_clk_calc(struct nvkm_clk *base, st
 
 	if ((ret = calc_clk(clk, cstate, 0x00, nv_clk_src_gpc)) ||
 	    (ret = calc_clk(clk, cstate, 0x01, nv_clk_src_rop)) ||
-	    (ret = calc_clk(clk, cstate, 0x02, nv_clk_src_hubk07)) ||
-	    (ret = calc_clk(clk, cstate, 0x07, nv_clk_src_hubk06)) ||
-	    (ret = calc_clk(clk, cstate, 0x08, nv_clk_src_hubk01)) ||
 	    (ret = calc_clk(clk, cstate, 0x09, nv_clk_src_copy)) ||
 	    (ret = calc_clk(clk, cstate, 0x0c, nv_clk_src_pmu)) ||
 	    (ret = calc_clk(clk, cstate, 0x0e, nv_clk_src_vdec)))
@@ -457,9 +454,10 @@ gf100_clk = {
 		{ nv_clk_src_hubk06 , 0x00 },
 		{ nv_clk_src_hubk01 , 0x01 },
 		{ nv_clk_src_copy   , 0x02 },
-		{ nv_clk_src_gpc    , 0x03, NVKM_CLK_DOM_FLAG_VPSTATE, "core", 2000 },
+		{ nv_clk_src_gpc    , 0x03, NVKM_CLK_DOM_FLAG_CORE |
+					    NVKM_CLK_DOM_FLAG_VPSTATE, "core", 2000 },
 		{ nv_clk_src_rop    , 0x04 },
-		{ nv_clk_src_mem    , 0x05, 0, "memory", 1000 },
+		{ nv_clk_src_mem    , 0x05, NVKM_CLK_DOM_FLAG_CORE, "memory", 1000 },
 		{ nv_clk_src_vdec   , 0x06 },
 		{ nv_clk_src_pmu    , 0x0a },
 		{ nv_clk_src_hubk07 , 0x0b },
@@ -477,5 +475,5 @@ gf100_clk_new(struct nvkm_device *device
 		return -ENOMEM;
 	*pclk = &clk->base;
 
-	return nvkm_clk_ctor(&gf100_clk, device, type, inst, false, &clk->base);
+	return nvkm_clk_ctor(&gf100_clk, device, type, inst, true, &clk->base);
 }
diff -uNrp nouveau-orig/nvkm/subdev/fb/ramgf100.c nouveau-mod/nvkm/subdev/fb/ramgf100.c
--- a/nvkm/subdev/fb/ramgf100.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nvkm/subdev/fb/ramgf100.c	2026-08-29 12:20:10.197482437 +0300
@@ -58,6 +58,9 @@ struct gf100_ramfuc {
 	struct ramfuc_reg r_0x10f340;
 	struct ramfuc_reg r_0x10f344;
 	struct ramfuc_reg r_0x10f348;
+	struct ramfuc_reg r_0x10f440;
+	struct ramfuc_reg r_0x10f444;
+	struct ramfuc_reg r_0x10f468;
 
 	struct ramfuc_reg r_0x10f910;
 	struct ramfuc_reg r_0x10f914;
@@ -275,8 +278,10 @@ gf100_ram_calc(struct nvkm_ram *base, u3
 	ram_mask(fuc, 0x10f200, 0x00000800, 0x00000000);
 	ram_wr32(fuc, 0x10f210, 0x00000000);
 	ram_nsec(fuc, 1000);
-	if (mode == 0)
-		gf100_ram_train(fuc, 0x000c1001);
+	if (mode == 0) {
+		if (ram->base.type == NVKM_RAM_TYPE_GDDR5)
+			gf100_ram_train(fuc, 0x000c1001);
+	}
 	ram_wr32(fuc, 0x10f310, 0x00000001);
 	ram_nsec(fuc, 1000);
 	ram_wr32(fuc, 0x10f090, 0x00000061);
@@ -284,7 +289,10 @@ gf100_ram_calc(struct nvkm_ram *base, u3
 	ram_nsec(fuc, 1000);
 
 	if (from == 0) {
-		ram_wr32(fuc, 0x10f824, 0x00007fd4);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3)
+			ram_wr32(fuc, 0x10f824, 0x00021e67);
+		else
+			ram_wr32(fuc, 0x10f824, 0x00007fd4);
 	} else {
 		ram_wr32(fuc, 0x1373ec, 0x00020404);
 	}
@@ -295,16 +303,27 @@ gf100_ram_calc(struct nvkm_ram *base, u3
 		ram_wr32(fuc, 0x10f830, 0x41500010);
 		ram_mask(fuc, 0x10f830, 0x01000000, 0x00000000);
 		ram_mask(fuc, 0x132100, 0x00000100, 0x00000100);
-		ram_wr32(fuc, 0x10f050, 0xff000090);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3)
+			ram_wr32(fuc, 0x10f050, 0xff001050);
+		else
+			ram_wr32(fuc, 0x10f050, 0xff000090);
 		ram_wr32(fuc, 0x1373ec, 0x00020f0f);
 		ram_wr32(fuc, 0x1373f0, 0x00000003);
 		ram_wr32(fuc, 0x137310, 0x81201616);
 		ram_wr32(fuc, 0x132100, 0x00000001);
-// 0x00020039 // 0x000000ba
 		ram_wr32(fuc, 0x10f830, 0x00300017);
 		ram_wr32(fuc, 0x1373f0, 0x00000001);
-		ram_wr32(fuc, 0x10f824, 0x00007e77);
-		ram_wr32(fuc, 0x132000, 0x18030001);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3) {
+			ram_wr32(fuc, 0x10f824, 0x00021e67);
+			ram_wr32(fuc, 0x132000, 0x18000000);
+			ram_wr32(fuc, 0x10f440, 0x22f84f10);
+			ram_wr32(fuc, 0x10f444, 0x04cc883f);
+			ram_wr32(fuc, 0x10f468, 0x00020020);
+			ram_wr32(fuc, 0x10f808, 0x08020004);
+		} else {
+			ram_wr32(fuc, 0x10f824, 0x00007e77);
+			ram_wr32(fuc, 0x132000, 0x18030001);
+		}
 		ram_wr32(fuc, 0x10f090, 0x4000007e);
 		ram_nsec(fuc, 2000);
 		ram_wr32(fuc, 0x10f314, 0x00000001);
@@ -312,26 +331,36 @@ gf100_ram_calc(struct nvkm_ram *base, u3
 		ram_wr32(fuc, 0x10f338, 0x00300220);
 		ram_wr32(fuc, 0x10f300, 0x0000011d);
 		ram_nsec(fuc, 1000);
-		ram_wr32(fuc, 0x10f290, 0x02060505);
-		ram_wr32(fuc, 0x10f294, 0x34208288);
-		ram_wr32(fuc, 0x10f298, 0x44050411);
-		ram_wr32(fuc, 0x10f29c, 0x0000114c);
-		ram_wr32(fuc, 0x10f2a0, 0x42e10069);
-		ram_wr32(fuc, 0x10f614, 0x40044f77);
-		ram_wr32(fuc, 0x10f610, 0x40044f77);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3) {
+			ram_wr32(fuc, 0x10f290, 0x02060505);
+			ram_wr32(fuc, 0x10f294, 0x34208288);
+			ram_wr32(fuc, 0x10f298, 0x44050411);
+			ram_wr32(fuc, 0x10f29c, 0x0000114c);
+			ram_wr32(fuc, 0x10f2a0, 0x42e10069);
+			ram_wr32(fuc, 0x10f614, 0x40044f77);
+			ram_wr32(fuc, 0x10f610, 0x40044f77);
+		} else {
+			ram_wr32(fuc, 0x10f290, 0x02060505);
+			ram_wr32(fuc, 0x10f294, 0x34208288);
+			ram_wr32(fuc, 0x10f298, 0x44050411);
+			ram_wr32(fuc, 0x10f29c, 0x0000114c);
+			ram_wr32(fuc, 0x10f2a0, 0x42e10069);
+			ram_wr32(fuc, 0x10f614, 0x40044f77);
+			ram_wr32(fuc, 0x10f610, 0x40044f77);
+		}
 		ram_wr32(fuc, 0x10f344, 0x00600009);
 		ram_nsec(fuc, 1000);
 		ram_wr32(fuc, 0x10f348, 0x00700008);
 		ram_wr32(fuc, 0x61c140, 0x19240000);
 		ram_wr32(fuc, 0x10f830, 0x00300017);
-		gf100_ram_train(fuc, 0x80021001);
-		gf100_ram_train(fuc, 0x80081001);
+		if (ram->base.type == NVKM_RAM_TYPE_GDDR5) {
+			gf100_ram_train(fuc, 0x80021001);
+			gf100_ram_train(fuc, 0x80081001);
+		}
 		ram_wr32(fuc, 0x10f340, 0x00500004);
 		ram_nsec(fuc, 1000);
 		ram_wr32(fuc, 0x10f830, 0x01300017);
 		ram_wr32(fuc, 0x10f830, 0x00300017);
-// 0x00030020 // 0x00000000 // 0x00000000
-// 0x00020034 // 0x0000000b
 		ram_wr32(fuc, 0x100b0c, 0x00080028);
 		ram_wr32(fuc, 0x611200, 0x00003330);
 	} else {
@@ -345,53 +374,69 @@ gf100_ram_calc(struct nvkm_ram *base, u3
 		ram_wr32(fuc, 0x1373f8, 0x00000000);
 		ram_wr32(fuc, 0x132100, 0x00000101);
 		ram_wr32(fuc, 0x137310, 0x89201616);
-		ram_wr32(fuc, 0x10f050, 0xff000090);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3)
+			ram_wr32(fuc, 0x10f050, 0xff000450);
+		else
+			ram_wr32(fuc, 0x10f050, 0xff000090);
 		ram_wr32(fuc, 0x1373ec, 0x00030404);
 		ram_wr32(fuc, 0x1373f0, 0x00000002);
-	// 0x00020039 // 0x00000011
 		ram_wr32(fuc, 0x132100, 0x00000001);
 		ram_wr32(fuc, 0x1373f8, 0x00002000);
 		ram_nsec(fuc, 2000);
-		ram_wr32(fuc, 0x10f808, 0x7aaa0050);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3) {
+			ram_wr32(fuc, 0x10f808, 0x08020004);
+			ram_wr32(fuc, 0x10f824, 0x000279e7);
+		} else {
+			ram_wr32(fuc, 0x10f808, 0x7aaa0050);
+		}
 		ram_wr32(fuc, 0x10f830, 0x00500010);
 		ram_wr32(fuc, 0x10f200, 0x00ce1000);
 		ram_wr32(fuc, 0x10f090, 0x4000007e);
 		ram_nsec(fuc, 2000);
 		ram_wr32(fuc, 0x10f314, 0x00000001);
 		ram_wr32(fuc, 0x10f210, 0x80000000);
-		ram_wr32(fuc, 0x10f338, 0x00300200);
+		ram_wr32(fuc, 0x10f338, 0x00300220);
 		ram_wr32(fuc, 0x10f300, 0x0000084d);
 		ram_nsec(fuc, 1000);
-		ram_wr32(fuc, 0x10f290, 0x0b343825);
-		ram_wr32(fuc, 0x10f294, 0x3483028e);
-		ram_wr32(fuc, 0x10f298, 0x440c0600);
-		ram_wr32(fuc, 0x10f29c, 0x0000214c);
-		ram_wr32(fuc, 0x10f2a0, 0x42e20069);
+		if (ram->base.type == NVKM_RAM_TYPE_DDR3) {
+			ram_wr32(fuc, 0x10f290, 0x02060505);
+			ram_wr32(fuc, 0x10f294, 0x34208288);
+			ram_wr32(fuc, 0x10f298, 0x44050411);
+			ram_wr32(fuc, 0x10f29c, 0x0000114c);
+			ram_wr32(fuc, 0x10f2a0, 0x42e10069);
+			ram_wr32(fuc, 0x10f614, 0x40044f77);
+			ram_wr32(fuc, 0x10f610, 0x40044f77);
+		} else {
+			ram_wr32(fuc, 0x10f290, 0x0b343825);
+			ram_wr32(fuc, 0x10f294, 0x3483028e);
+			ram_wr32(fuc, 0x10f298, 0x440c0600);
+			ram_wr32(fuc, 0x10f29c, 0x0000214c);
+			ram_wr32(fuc, 0x10f2a0, 0x42e20069);
+			ram_wr32(fuc, 0x10f614, 0x60044e77);
+			ram_wr32(fuc, 0x10f610, 0x60044e77);
+		}
 		ram_wr32(fuc, 0x10f200, 0x00ce0000);
-		ram_wr32(fuc, 0x10f614, 0x60044e77);
-		ram_wr32(fuc, 0x10f610, 0x60044e77);
 		ram_wr32(fuc, 0x10f340, 0x00500000);
 		ram_nsec(fuc, 1000);
 		ram_wr32(fuc, 0x10f344, 0x00600228);
 		ram_nsec(fuc, 1000);
 		ram_wr32(fuc, 0x10f348, 0x00700000);
 		ram_wr32(fuc, 0x13d8f4, 0x00000000);
-		ram_wr32(fuc, 0x61c140, 0x09a40000);
-
-		gf100_ram_train(fuc, 0x800e1008);
-
+		if (ram->base.type == NVKM_RAM_TYPE_GDDR5) {
+			ram_wr32(fuc, 0x61c140, 0x09a40000);
+			gf100_ram_train(fuc, 0x800e1008);
+		}
 		ram_nsec(fuc, 1000);
 		ram_wr32(fuc, 0x10f800, 0x00001804);
-	// 0x00030020 // 0x00000000 // 0x00000000
-	// 0x00020034 // 0x0000000b
 		ram_wr32(fuc, 0x13d8f4, 0x00000000);
 		ram_wr32(fuc, 0x100b0c, 0x00080028);
 		ram_wr32(fuc, 0x611200, 0x00003330);
 		ram_nsec(fuc, 100000);
-		ram_wr32(fuc, 0x10f9b0, 0x05313f41);
-		ram_wr32(fuc, 0x10f9b4, 0x00002f50);
-
-		gf100_ram_train(fuc, 0x010c1001);
+		if (ram->base.type == NVKM_RAM_TYPE_GDDR5) {
+			ram_wr32(fuc, 0x10f9b0, 0x05313f41);
+			ram_wr32(fuc, 0x10f9b4, 0x00002f50);
+			gf100_ram_train(fuc, 0x010c1001);
+		}
 	}
 
 	ram_mask(fuc, 0x10f200, 0x00000800, 0x00000800);
diff -uNrp nouveau-orig/nvkm/subdev/pmu/gt215.c nouveau-mod/nvkm/subdev/pmu/gt215.c
--- a/nvkm/subdev/pmu/gt215.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nvkm/subdev/pmu/gt215.c	2026-08-29 12:20:10.198915419 +0300
@@ -56,9 +56,11 @@ gt215_pmu_send(struct nvkm_pmu *pmu, u32
 	}
 
 	/* acquire data segment access */
-	do {
-		nvkm_wr32(device, 0x10a580, 0x00000001);
-	} while (nvkm_rd32(device, 0x10a580) != 0x00000001);
+	if (device->card_type < NV_C0) {
+		do {
+			nvkm_wr32(device, 0x10a580, 0x00000001);
+		} while (nvkm_rd32(device, 0x10a580) != 0x00000001);
+	}
 
 	/* write the packet */
 	nvkm_wr32(device, 0x10a1c0, 0x01000000 | (((addr & 0x07) << 4) +
@@ -70,11 +72,24 @@ gt215_pmu_send(struct nvkm_pmu *pmu, u32
 	nvkm_wr32(device, 0x10a4a0, (addr + 1) & 0x0f);
 
 	/* release data segment access */
-	nvkm_wr32(device, 0x10a580, 0x00000000);
+	if (device->card_type < NV_C0)
+		nvkm_wr32(device, 0x10a580, 0x00000000);
 
 	/* wait for reply, if requested */
 	if (reply) {
-		wait_event(pmu->recv.wait, (pmu->recv.process == 0));
+		unsigned long deadline = jiffies + msecs_to_jiffies(100);
+		while (pmu->recv.process != 0 && time_before(jiffies, deadline)) {
+			gt215_pmu_recv(pmu);
+			if (pmu->recv.process == 0)
+				break;
+			usleep_range(50, 100);
+		}
+		if (pmu->recv.process != 0) {
+			nvkm_warn(subdev, "pmu reply timeout\n");
+			pmu->recv.process = 0;
+			mutex_unlock(&pmu->send.mutex);
+			return -ETIMEDOUT;
+		}
 		reply[0] = pmu->recv.data[0];
 		reply[1] = pmu->recv.data[1];
 	}
@@ -96,12 +111,14 @@ gt215_pmu_recv(struct nvkm_pmu *pmu)
 		return;
 
 	/* acquire data segment access */
-	do {
-		nvkm_wr32(device, 0x10a580, 0x00000002);
-	} while (nvkm_rd32(device, 0x10a580) != 0x00000002);
+	if (device->card_type < NV_C0) {
+		do {
+			nvkm_wr32(device, 0x10a580, 0x00000002);
+		} while (nvkm_rd32(device, 0x10a580) != 0x00000002);
+	}
 
 	/* read the packet */
-	nvkm_wr32(device, 0x10a1c0, 0x02000000 | (((addr & 0x07) << 4) +
+	nvkm_wr32(device, 0x10a1c0, 0x01000000 | (((addr & 0x07) << 4) +
 				pmu->recv.base));
 	process = nvkm_rd32(device, 0x10a1c4);
 	message = nvkm_rd32(device, 0x10a1c4);
@@ -110,7 +127,8 @@ gt215_pmu_recv(struct nvkm_pmu *pmu)
 	nvkm_wr32(device, 0x10a4cc, (addr + 1) & 0x0f);
 
 	/* release data segment access */
-	nvkm_wr32(device, 0x10a580, 0x00000000);
+	if (device->card_type < NV_C0)
+		nvkm_wr32(device, 0x10a580, 0x00000000);
 
 	/* wake process if it's waiting on a synchronous reply */
 	if (pmu->recv.process) {
diff -uNrp nouveau-orig/nvkm/subdev/pmu/memx.c nouveau-mod/nvkm/subdev/pmu/memx.c
--- a/nvkm/subdev/pmu/memx.c	2026-06-12 01:03:26.000000000 +0300
+++ b/nvkm/subdev/pmu/memx.c	2026-08-29 12:20:10.199477853 +0300
@@ -61,9 +61,11 @@ nvkm_memx_init(struct nvkm_pmu *pmu, str
 	memx->size = reply[1];
 
 	/* acquire data segment access */
-	do {
-		nvkm_wr32(device, 0x10a580, 0x00000003);
-	} while (nvkm_rd32(device, 0x10a580) != 0x00000003);
+	if (device->card_type < NV_C0) {
+		do {
+			nvkm_wr32(device, 0x10a580, 0x00000003);
+		} while (nvkm_rd32(device, 0x10a580) != 0x00000003);
+	}
 	nvkm_wr32(device, 0x10a1c0, 0x01000000 | memx->base);
 	return 0;
 }
@@ -82,7 +84,8 @@ nvkm_memx_fini(struct nvkm_memx **pmemx,
 
 	/* release data segment access */
 	finish = nvkm_rd32(device, 0x10a1c0) & 0x00ffffff;
-	nvkm_wr32(device, 0x10a580, 0x00000000);
+	if (device->card_type < NV_C0)
+		nvkm_wr32(device, 0x10a580, 0x00000000);
 
 	/* call MEMX process to execute the script, and wait for reply */
 	if (exec) {
