--- ../apertium-oc-ca-1.0.5/apertium-oc-ca.oc-ca.t1x	2008-07-21 13:02:56.000000000 +0200
+++ apertium-oc-ca-1.0.5/apertium-oc-ca.oc-ca.t1x	2010-09-15 16:49:14.000000000 +0200
@@ -1282,7 +1282,7 @@
 			</choose>
 			<choose><!-- aquests no són ref-->
 				<when>
-					<test>
+					<test><and>
 						<equal caseless="yes">
 							<clip pos="2" side="sl" part="pers"/>
 							<lit-tag v="p3"/>
@@ -1291,7 +1291,7 @@
 							<clip pos="2" side="sl" part="nbr"/>
 							<lit-tag v="sg"/>
 						</equal>
-					</test>
+					</and></test>
 					<choose>
 						<when>
 							<test>
