diff -aur OCRmyPDF-2.1-stable.pristine/OCRmyPDF.sh OCRmyPDF-2.1-stable.new/OCRmyPDF.sh
--- OCRmyPDF-2.1-stable.pristine/OCRmyPDF.sh	2014-01-25 22:14:05.000000000 +0100
+++ OCRmyPDF-2.1-stable.new/OCRmyPDF.sh	2014-01-27 11:54:14.523542415 +0100
@@ -4,8 +4,7 @@
 ##############################################################################
 
 # Import required scripts
-BASEPATH="$(dirname $(readlink -f $0))"
-. "$BASEPATH/src/config.sh"
+. "/etc/ocrmypdf.sh"
 
 # Set variables corresponding to the input parameters
 ARGUMENTS="$@"
diff -aur OCRmyPDF-2.1-stable.pristine/src/config.sh OCRmyPDF-2.1-stable.new/src/config.sh
--- OCRmyPDF-2.1-stable.pristine/src/config.sh	2014-01-25 22:14:05.000000000 +0100
+++ OCRmyPDF-2.1-stable.new/src/config.sh	2014-01-27 11:55:30.300212360 +0100
@@ -26,7 +26,7 @@
 LOG_DEBUG="3"				# debug level logging
 
 # various paths
-SRC="./src"				# location of the source folder (except source of external tools like jhove)
+SRC="/usr/lib/ocrmypdf/src"		# location of the source folder (except source of external tools like jhove)
 OCR_PAGE="$SRC/ocrPage.sh"		# path to the script aimed at OCRing one page
-JHOVE="./jhove/bin/JhoveApp.jar"	# java SW for validating the final PDF/A
-JHOVE_CFG="./jhove/conf/jhove.conf"	# location of the jhove config file
+JHOVE="/usr/lib/jhove/bin/JhoveApp.jar"	# java SW for validating the final PDF/A
+JHOVE_CFG="/usr/lib/jhove/conf/jhove.conf"	# location of the jhove config file
diff -aur OCRmyPDF-2.1-stable.pristine/src/ocrPage.sh OCRmyPDF-2.1-stable.new/src/ocrPage.sh
--- OCRmyPDF-2.1-stable.pristine/src/ocrPage.sh	2014-01-25 22:14:05.000000000 +0100
+++ OCRmyPDF-2.1-stable.new/src/ocrPage.sh	2014-01-27 12:22:59.096950348 +0100
@@ -5,7 +5,7 @@
 # Copyright (c) 2013-14: fritz-hh from Github (https://github.com/fritz-hh)
 ##############################################################################
 
-. "./src/config.sh"
+. "/etc/ocrmypdf.sh"
 
 
 # Initialization of variables passed by arguments
