From 2d5f33dd83534bbfadbe59d8ce19b6bb983ec097 Mon Sep 17 00:00:00 2001
From: qumaciel <maciel@tutanota.com>
Date: Fri, 19 Aug 2022 17:50:37 -0300
Subject: gcc12

---
 build/linux.gcc.inc | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/build/linux.gcc.inc b/build/linux.gcc.inc
index aef54f5..0c9e738 100644
--- a/build/linux.gcc.inc
+++ b/build/linux.gcc.inc
@@ -36,33 +36,33 @@ LINK_FLAGS = -Wl,-rpath-link=. -rdynamic
 C_FLAGS = $(CPLUS_FLAGS)
 
 # gcc 4.2 and higher support OpenMP
-ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^(4\.[2-9]|[5-9])"))
-    OPENMP_FLAG = -fopenmp
-endif
+# ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^(4\.[2-9]|[5-9])"))
+OPENMP_FLAG = -fopenmp
+# endif
 
 # gcc 4.8 and later support RTM intrinsics, but require command line switch to enable them
-ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^(4\.[8-9]|[5-9])"))
-    RTM_KEY = -mrtm
-endif
+# ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^(4\.[8-9]|[5-9])"))
+RTM_KEY = -mrtm
+# endif
 
 # gcc 4.0 and later have -Wextra that is used by some our customers.
-ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([4-9])"))
-    WARNING_KEY += -Wextra
-endif
+# ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([4-9])"))
+WARNING_KEY += -Wextra
+# endif
 
 # gcc 5.0 and later have -Wsuggest-override and -Wno-sized-deallocation options
-ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([5-9])"))
-    # enable -Wsuggest-override via a pre-included header in order to limit to C++11 and above
-    INCLUDE_TEST_HEADERS = -include $(tbb_root)/src/test/harness_preload.h
-    WARNING_SUPPRESS += -Wno-sized-deallocation
-endif
+# ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([5-9])"))
+# enable -Wsuggest-override via a pre-included header in order to limit to C++11 and above
+INCLUDE_TEST_HEADERS = -include $(tbb_root)/src/test/harness_preload.h
+WARNING_SUPPRESS += -Wno-sized-deallocation
+# endif
 
 # gcc 6.0 and later have -flifetime-dse option that controls
 # elimination of stores done outside the object lifetime
-ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([6-9])"))
-    # keep pre-contruction stores for zero initialization
-    DSE_KEY = -flifetime-dse=1
-endif
+# ifneq (,$(shell $(CONLY) -dumpversion | egrep  "^([6-9])"))
+# keep pre-contruction stores for zero initialization
+DSE_KEY = -flifetime-dse=1
+# endif
 
 ifeq ($(cfg), release)
         CPLUS_FLAGS = $(ITT_NOTIFY) -g -O2 -DUSE_PTHREAD
-- 
2.37.2

