--- a/onvif-gui/onvif_gui/glwidget.py
+++ b/onvif-gui/onvif_gui/glwidget.py
@@ -188,14 +188,13 @@ class GLWidget(QOpenGLWidget):
             if interval.total_seconds() > 10:
                 self.mw.alarm_states = []
 
-        if self.mw.split.sizes()[0]:
-            if len(self.mw.pm.players) or len(self.mw.timers):
-                self.buildImage()
-            else:
-                if self.buffer:
-                    self.buffer.fill(QColorConstants.Black)
+        if len(self.mw.pm.players) or len(self.mw.timers):
+            self.buildImage()
+        else:
+            if self.buffer:
+                self.buffer.fill(QColorConstants.Black)
 
-            self.update()
+        self.update()
 
     def alarmBroadcast(self):
         proxyPanel = self.mw.settingsPanel.proxy
