diff --git a/oclint-driver/CMakeLists.txt b/oclint-driver/CMakeLists.txt
index d1826f1..39b697f 100644
--- a/oclint-driver/CMakeLists.txt
+++ b/oclint-driver/CMakeLists.txt
@@ -67,12 +67,12 @@ IF((NOT NO_ANALYTICS) AND (NOT MINGW) AND (NOT DOC_GEN_BUILD) AND (NOT TEST_BUIL
             /usr/local/opt/openssl/lib/libcrypto.a
             )
     ELSE()
-        FIND_LIBRARY(SSL_STATIC_LIB NAMES libssl.a)
-        FIND_LIBRARY(CRYPTO_STATIC_LIB NAMES libcrypto.a)
+        FIND_LIBRARY(SSL_SHARED_LIB NAMES libssl.so)
+        FIND_LIBRARY(CRYPTO_SHARED_LIB NAMES libcrypto.so)
         TARGET_LINK_LIBRARIES(oclint-${OCLINT_VERSION_RELEASE}
             Countly
-            ${SSL_STATIC_LIB}
-            ${CRYPTO_STATIC_LIB}
+            ${SSL_SHARED_LIB}
+            ${CRYPTO_SHARED_LIB}
             )
     ENDIF()
 ENDIF()
