--- a/Makefile	2021-12-23 10:15:03.000000000 +0200
+++ b/Makefile	2022-03-06 16:36:03.777862228 +0200
@@ -13,7 +13,7 @@
 
 JDBC_CONFIG = jdbc_config
 
-LIBDIR=/usr/lib64/
+LIBDIR=/usr/lib/jvm/java-8-openjdk/jre/lib/amd64/server
 
 SHLIB_LINK += -L$(LIBDIR) -ljvm
 
@@ -27,12 +27,12 @@
  
 PG_CPPFLAGS=-D'PKG_LIB_DIR=$(pkglibdir)' -I$(libpq_srcdir)
 
-JFLAGS = -d $(pkglibdir)
+JFLAGS = -d .
 
 all:$(TRGTS)
 
 JAVAFILES:
-	javac $(JFLAGS) $(JAVA_SOURCES)
+	/usr/lib/jvm/java-8-openjdk/bin/javac $(JFLAGS) $(JAVA_SOURCES)
 
 # the db name is hard-coded in the tests
 override USE_MODULE_DB =
