diff -Naurp openchange/mapiproxy/dcesrv_mapiproxy.c openchange-new/mapiproxy/dcesrv_mapiproxy.c
--- openchange/mapiproxy/dcesrv_mapiproxy.c	2015-11-27 00:00:08.903834201 -0600
+++ openchange-new/mapiproxy/dcesrv_mapiproxy.c	2015-11-27 02:34:59.798268143 -0600
@@ -222,16 +222,15 @@ static NTSTATUS mapiproxy_op_bind_proxy(
  */
 static NTSTATUS mapiproxy_op_bind(struct dcesrv_call_state *dce_call, const struct dcesrv_interface *iface, uint32_t if_version)
 {
+	struct server_id_buf			idbuf;
 	struct dcesrv_mapiproxy_private		*private;
 	bool					server_mode;
 	bool					ndrdump;
-	char					*server_id_printable = NULL;
 	
-	server_id_printable = server_id_str(NULL, &(dce_call->conn->server_id));
 	OC_DEBUG(5, "[session = 0x%x] [session server id = %s]\n",
 		  dce_call->context->context_id,
-		  server_id_printable);
-	talloc_free(server_id_printable);
+	          server_id_str_buf((dce_call->conn->server_id), &idbuf));
+	talloc_free(server_id_str_buf((dce_call->conn->server_id), &idbuf));
 
 	OC_DEBUG(5, "mapiproxy::mapiproxy_op_bind: [session = 0x%x] [session server id = 0x%"PRIx64" 0x%x 0x%x]", dce_call->context->context_id,
 		  dce_call->conn->server_id.pid, dce_call->conn->server_id.task_id, dce_call->conn->server_id.vnn);
diff -Naurp openchange/mapiproxy/modules/mpm_cache.c openchange-new/mapiproxy/modules/mpm_cache.c
--- openchange/mapiproxy/modules/mpm_cache.c	2015-11-27 00:00:08.820500133 -0600
+++ openchange-new/mapiproxy/modules/mpm_cache.c	2015-11-27 02:32:06.854675099 -0600
@@ -216,40 +216,37 @@ static NTSTATUS cache_pull_Release(struc
 				   struct EcDoRpc *EcDoRpc, 
 				   uint32_t handle_idx)
 {
+	struct server_id_buf		idbuf;
 	struct mpm_message		*message;
 	struct mpm_attachment		*attach;
 	struct mpm_stream		*stream;
-	char				*server_id_printable = NULL;
 
 	/* Look over messages */
 	for (message = mpm->messages; message; message = message->next) {
 		if ((mpm_session_cmp(message->session, dce_call) == true) &&
 		    (EcDoRpc->in.mapi_request->handles[handle_idx] == message->handle)) {
-		        server_id_printable = server_id_str(NULL, &(message->session->server_id));
 			OC_DEBUG(2, "* [s(%s),c(0x%x)] Del: Message 0x%"PRIx64" 0x%"PRIx64": 0x%x",
-				  server_id_printable, message->session->context_id,
+				  server_id_str_buf((message->session->server_id), &idbuf), message->session->context_id,
 				  message->FolderId, message->MessageId, message->handle);
-			talloc_free(server_id_printable);
+			talloc_free(server_id_str_buf((message->session->server_id), &idbuf));
 
 			/* Loop over children attachments */
 			attach = mpm->attachments;
 			while (attach) {
 				if ((mpm_session_cmp(attach->session, dce_call) == true) &&
 				    (message->handle == attach->parent_handle)) {
-				        server_id_printable = server_id_str(NULL, &(attach->session->server_id));
 					OC_DEBUG(2, "* [s(%s),c(0x%x)] Del recursive 1: Attachment %d: 0x%x",
-						  server_id_printable, attach->session->context_id, attach->AttachmentID, attach->handle);
-					talloc_free(server_id_printable);
+						  server_id_str_buf((attach->session->server_id), &idbuf), attach->session->context_id, attach->AttachmentID, attach->handle);
+					talloc_free(server_id_str_buf((attach->session->server_id), &idbuf));
 
 					/* Loop over children streams */
 					stream = mpm->streams;
 					while (stream) {
 						if ((mpm_session_cmp(stream->session, dce_call) == true) &&
 						    (attach->handle == stream->parent_handle)) {
-						        server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 							OC_DEBUG(2, "* [s(%s),c(0x%x)] Del recursive 1-2: Stream 0x%x",
-								  server_id_printable, stream->session->context_id, stream->handle);
-							talloc_free(server_id_printable);
+								  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->handle);
+							talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 							mpm_session_release(stream->session);
 							mpm_cache_stream_close(stream);
 							talloc_free(stream->filename);
@@ -275,10 +272,9 @@ static NTSTATUS cache_pull_Release(struc
 			while (stream) {
 				if ((mpm_session_cmp(stream->session, dce_call) == true) &&
 				    (message->handle == stream->parent_handle)) {
-					server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 					OC_DEBUG(2, "* [s(%s),c(0x%x)] Del recursive 1: Stream 0x%x",
-						  server_id_printable, stream->session->context_id, stream->handle);
-					talloc_free(server_id_printable);
+						  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->handle);
+					talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 					mpm_session_release(stream->session);
 					mpm_cache_stream_close(stream);
 					DLIST_REMOVE(mpm->streams, stream);
@@ -301,20 +297,18 @@ static NTSTATUS cache_pull_Release(struc
 	for (attach = mpm->attachments; attach; attach = attach->next) {
 		if ((mpm_session_cmp(attach->session, dce_call) == true) &&
 		    (EcDoRpc->in.mapi_request->handles[handle_idx] == attach->handle)) {
-			server_id_printable = server_id_str(NULL, &(attach->session->server_id));
 			OC_DEBUG(2, "* [s(%s),c(0x%x)] Del: Attachment %d: 0x%x",
-				  server_id_printable, attach->session->context_id, attach->AttachmentID, attach->handle);
-			talloc_free(server_id_printable);
+				  server_id_str_buf((attach->session->server_id), &idbuf), attach->session->context_id, attach->AttachmentID, attach->handle);
+			talloc_free(server_id_str_buf((attach->session->server_id), &idbuf));
 
 			/* Loop over children streams */
 			stream = mpm->streams;
 			while (stream) {
 				if ((mpm_session_cmp(stream->session, dce_call) == true) &&
 				    (attach->handle == stream->parent_handle)) {
-					server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 					OC_DEBUG(2, "* [s(%s),c(0x%x)] Del recursive 2: Stream 0x%x",
-						  server_id_printable, stream->session->context_id, stream->handle);
-					talloc_free(server_id_printable);
+						  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->handle);
+					talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 					mpm_session_release(stream->session);
 					mpm_cache_stream_close(stream);
 					DLIST_REMOVE(mpm->streams, stream);
@@ -337,10 +331,9 @@ static NTSTATUS cache_pull_Release(struc
 	for (stream = mpm->streams; stream; stream = stream->next) {
 		if ((mpm_session_cmp(stream->session, dce_call) == true) &&
 		    (EcDoRpc->in.mapi_request->handles[handle_idx] == stream->handle)) {
-			server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 			OC_DEBUG(2, "* [s(%s),c(0x%x)] Del: Stream 0x%x\n",
-				  server_id_printable, stream->session->context_id, stream->handle);
-			talloc_free(server_id_printable);
+				  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->handle);
+			talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 			mpm_session_release(stream->session);
 			mpm_cache_stream_close(stream);
 			DLIST_REMOVE(mpm->streams, stream);
@@ -428,10 +421,10 @@ static NTSTATUS cache_push_OpenMessage(s
 				       struct EcDoRpc_MAPI_REPL mapi_repl, 
 				       struct EcDoRpc *EcDoRpc)
 {
-	struct mpm_message	*el;
-	struct mapi_response	*mapi_response;
-	struct OpenMessage_req	request;
-	char *server_id_printable = NULL;
+	struct server_id_buf		idbuf;
+	struct mpm_message		*el;
+	struct mapi_response		*mapi_response;
+	struct OpenMessage_req		request;
 
 	request = mapi_req.u.mapi_OpenMessage;
 
@@ -443,17 +436,15 @@ static NTSTATUS cache_push_OpenMessage(s
 			if (mapi_repl.error_code == MAPI_E_SUCCESS) {
 				mpm_cache_ldb_add_message((TALLOC_CTX *)mpm, mpm->ldb_ctx, el);
 				el->handle = mapi_response->handles[request.handle_idx];
-				server_id_printable = server_id_str(NULL, &(el->session->server_id));
 				OC_DEBUG(2, "* [s(%s),c(0x%x)] Add: Message 0x%"PRIx64" 0x%"PRIx64" 0x%x",
-					  server_id_printable, el->session->context_id, el->FolderId,
+					  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id, el->FolderId,
 					  el->MessageId, el->handle);
-				talloc_free(server_id_printable);
+				talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 			} else {
-				server_id_printable = server_id_str(NULL, &(el->session->server_id));
 				OC_DEBUG(0, "* [s(%s),c(0x%x)] Del: Message OpenMessage returned %s",
-					  server_id_printable, el->session->context_id,
+					  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id,
 				          mapi_get_errstr(mapi_repl.error_code));
-				talloc_free(server_id_printable);
+				talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 				DLIST_REMOVE(mpm->messages, el);
 			}
 			return NT_STATUS_OK;
@@ -484,11 +475,11 @@ static NTSTATUS cache_pull_OpenAttach(st
 				      struct EcDoRpc_MAPI_REQ mapi_req, 
 				      struct EcDoRpc *EcDoRpc)
 {
-	struct mpm_message	*el;
-	struct mpm_attachment	*attach;
-	struct mapi_request	*mapi_request;
-	struct OpenAttach_req	request;
-	char 			*server_id_printable = NULL;
+	struct server_id_buf		idbuf;
+	struct mpm_message		*el;
+	struct mpm_attachment		*attach;
+	struct mapi_request		*mapi_request;
+	struct OpenAttach_req		request;
 
 	mapi_request = EcDoRpc->in.mapi_request;
 	request = mapi_req.u.mapi_OpenAttach;
@@ -518,11 +509,10 @@ static NTSTATUS cache_pull_OpenAttach(st
 		}
 	}
 
-	server_id_printable = server_id_str(NULL, &(attach->session->server_id));
 	OC_DEBUG(2, "* [s(%s),c(0x%x)] Add [1]: Attachment %d  parent handle (0x%x) 0x%"PRIx64", 0x%"PRIx64" added to the list",
-		  server_id_printable, attach->session->context_id, request.AttachmentID, attach->parent_handle, 
+		  server_id_str_buf((attach->session->server_id), &idbuf), attach->session->context_id, request.AttachmentID, attach->parent_handle, 
 		  attach->message->FolderId, attach->message->MessageId);
-	talloc_free(server_id_printable);
+	talloc_free(server_id_str_buf((attach->session->server_id), &idbuf));
 	DLIST_ADD_END(mpm->attachments, attach, struct mpm_attachment *);
 	
 	return NT_STATUS_OK;
@@ -558,11 +548,11 @@ static NTSTATUS cache_push_OpenAttach(st
 				      struct EcDoRpc_MAPI_REPL mapi_repl, 
 				      struct EcDoRpc *EcDoRpc)
 {
+	struct server_id_buf		idbuf;
 	struct mpm_attachment		*el;
 	struct mapi_request		*mapi_request;
 	struct mapi_response		*mapi_response;
 	struct OpenAttach_req		request;
-	char				*server_id_printable = NULL;
 
 	mapi_request = EcDoRpc->in.mapi_request;
 	mapi_response = EcDoRpc->out.mapi_response;
@@ -574,18 +564,16 @@ static NTSTATUS cache_push_OpenAttach(st
 		    (request.AttachmentID == el->AttachmentID)) {
 			if (mapi_repl.error_code == MAPI_E_SUCCESS) {
 				el->handle = mapi_response->handles[request.handle_idx];
-				server_id_printable = server_id_str(NULL, &(el->session->server_id));
 				OC_DEBUG(2, "* [s(%s),c(0x%x)] Add [2]: Attachment %d with handle 0x%x and parent handle 0x%x",
-					  server_id_printable, el->session->context_id, el->AttachmentID, el->handle,
+					  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id, el->AttachmentID, el->handle,
 					  el->parent_handle);
-				talloc_free(server_id_printable);
+				talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 				mpm_cache_ldb_add_attachment((TALLOC_CTX *)mpm, mpm->ldb_ctx, el);
 			} else {
-			        server_id_printable = server_id_str(NULL, &(el->session->server_id));
 				OC_DEBUG(0, "* [s(%s),c(0x%x)] Del: Attachment OpenAttach returned %s",
-					  server_id_printable, el->session->context_id,
+					  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id,
 					  mapi_get_errstr(mapi_repl.error_code));
-				talloc_free(server_id_printable);
+				talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 				DLIST_REMOVE(mpm->attachments, el);
 			}
 			return NT_STATUS_OK;
@@ -619,12 +607,12 @@ static NTSTATUS cache_pull_OpenStream(st
 				      struct EcDoRpc_MAPI_REQ mapi_req, 
 				      struct EcDoRpc *EcDoRpc)
 {
-	struct mpm_stream	*stream;
-	struct mpm_attachment	*attach;
-	struct mpm_message	*message;
-	struct mapi_request	*mapi_request;
-	struct OpenStream_req	request;
-	char			*server_id_printable = NULL;
+	struct server_id_buf		idbuf;
+	struct mpm_stream		*stream;
+	struct mpm_attachment		*attach;
+	struct mpm_message		*message;
+	struct mapi_request		*mapi_request;
+	struct OpenStream_req		request;
 
 	mapi_request = EcDoRpc->in.mapi_request;
 	request = mapi_req.u.mapi_OpenStream;
@@ -648,11 +636,10 @@ static NTSTATUS cache_pull_OpenStream(st
 			stream->message = NULL;
 			stream->ahead = (mpm->ahead == true) ? true : false;
 			gettimeofday(&stream->tv_start, NULL);
-			server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 			OC_DEBUG(2, "* [s(%s),c(0x%x)] Stream::attachment added 0x%x 0x%"PRIx64" 0x%"PRIx64,
-				  server_id_printable, stream->session->context_id, stream->parent_handle, 
+				  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->parent_handle, 
 				  stream->attachment->message->FolderId, stream->attachment->message->MessageId);
-			talloc_free(server_id_printable);
+			talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 			DLIST_ADD_END(mpm->streams, stream, struct mpm_stream *);
 			return NT_STATUS_OK;
 		}
@@ -676,10 +663,9 @@ static NTSTATUS cache_pull_OpenStream(st
 			stream->cached = false;
 			stream->ahead = (mpm->ahead == true) ? true : false;
 			gettimeofday(&stream->tv_start, NULL);
-			server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 			OC_DEBUG(2, "* [s(%s),c(0x%x)] Stream::message added 0x%x",
-				  server_id_printable, stream->session->context_id, stream->parent_handle);
-			talloc_free(server_id_printable);
+				  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->parent_handle);
+			talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 			stream->message = message;
 			DLIST_ADD_END(mpm->streams, stream, struct mpm_stream *);
 			return NT_STATUS_OK;
@@ -719,12 +705,12 @@ static NTSTATUS cache_push_OpenStream(st
 				      struct EcDoRpc_MAPI_REPL mapi_repl, 
 				      struct EcDoRpc *EcDoRpc)
 {
-	struct mpm_stream	*el;
-	struct mapi_request	*mapi_request;
-	struct mapi_response	*mapi_response;
-	struct OpenStream_req	request;
-	struct OpenStream_repl	response;
-	char			*server_id_printable = NULL;
+	struct server_id_buf		idbuf;
+	struct mpm_stream		*el;
+	struct mapi_request		*mapi_request;
+	struct mapi_response		*mapi_response;
+	struct OpenStream_req		request;
+	struct OpenStream_repl		response;
 
 	mapi_request = EcDoRpc->in.mapi_request;
 	mapi_response = EcDoRpc->out.mapi_response;
@@ -738,17 +724,15 @@ static NTSTATUS cache_push_OpenStream(st
 				if (mapi_repl.error_code == MAPI_E_SUCCESS) {
 					el->handle = mapi_response->handles[request.handle_idx];
 					el->StreamSize = response.StreamSize;
-					server_id_printable = server_id_str(NULL, &(el->session->server_id));
 					OC_DEBUG(2, "* [s(%s),c(0x%x)] Add [2]: Stream for Property Tag 0x%x, handle 0x%x and size = %d",
-						  server_id_printable, el->session->context_id, el->PropertyTag, el->handle,
+						  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id, el->PropertyTag, el->handle,
 						  el->StreamSize);
-					talloc_free(server_id_printable);
+					talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 					mpm_cache_ldb_add_stream(mpm, mpm->ldb_ctx, el);
 				} else {
-					server_id_printable = server_id_str(NULL, &(el->session->server_id));
 					OC_DEBUG(0, "* [s(%s),c(0x%x)] Del: Stream OpenStream returned %s",
-						  server_id_printable, el->session->context_id, mapi_get_errstr(mapi_repl.error_code));
-					talloc_free(server_id_printable);
+						  server_id_str_buf((el->session->server_id), &idbuf), el->session->context_id, mapi_get_errstr(mapi_repl.error_code));
+					talloc_free(server_id_str_buf((el->session->server_id), &idbuf));
 					DLIST_REMOVE(mpm->streams, el);
 				}
 				return NT_STATUS_OK;
@@ -782,11 +766,11 @@ static NTSTATUS cache_push_ReadStream(st
 				      struct EcDoRpc_MAPI_REPL mapi_repl, 
 				      struct EcDoRpc *EcDoRpc)
 {
-	struct mpm_stream	*stream;
-	struct mapi_response	*mapi_response;
-	struct ReadStream_repl	response;
+	struct server_id_buf		idbuf;
+	struct mpm_stream		*stream;
+	struct mapi_response		*mapi_response;
+	struct ReadStream_repl		response;
 	/* struct ReadStream_req	request; */
-	char			*server_id_printable = NULL;
 
 	mapi_response = EcDoRpc->out.mapi_response;
 	response = mapi_repl.u.mapi_ReadStream;
@@ -800,10 +784,9 @@ static NTSTATUS cache_push_ReadStream(st
 				if (mpm->sync == true && stream->StreamSize > mpm->sync_min) {
 					cache_exec_sync_cmd(stream);
 				} else {
-					server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 					OC_DEBUG(5, "* [s(%s),c(0x%x)] %zd bytes from remove server",
-						  server_id_printable, stream->session->context_id, response.data.length);
-					talloc_free(server_id_printable);
+						  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, response.data.length);
+					talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 					mpm_cache_stream_write(stream, response.data.length, response.data.data);
 					if (stream->offset == stream->StreamSize) {
 						if (response.data.length) {
@@ -1074,20 +1057,19 @@ static NTSTATUS cache_dispatch(struct dc
  */
 static NTSTATUS cache_unbind(struct server_id server_id, uint32_t context_id)
 {
-	struct mpm_message	*message;
-	struct mpm_attachment	*attach;
-	struct mpm_stream	*stream;
-	char			*server_id_printable = NULL;
+	struct server_id_buf		idbuf;
+	struct mpm_message		*message;
+	struct mpm_attachment		*attach;
+	struct mpm_stream		*stream;
 
 	/* Look over messages still attached to the session */
 	message = mpm->messages;
 	while (message) {
 		if ((mpm_session_cmp_sub(message->session, server_id, context_id) == true)) {
-			server_id_printable = server_id_str(NULL, &(message->session->server_id));
 			OC_DEBUG(2, "[s(%s),c(0x%x)] Message - 0x%"PRIx64"/0x%"PRIx64" handle(0x%x)",
-				  server_id_printable, message->session->context_id,
+				  server_id_str_buf((message->session->server_id), &idbuf), message->session->context_id,
 				  message->FolderId, message->MessageId, message->handle);
-			talloc_free(server_id_printable);
+			talloc_free(server_id_str_buf((message->session->server_id), &idbuf));
 			mpm_session_release(message->session);
 			DLIST_REMOVE(mpm->messages, message);
 			talloc_free(message);
@@ -1101,11 +1083,10 @@ static NTSTATUS cache_unbind(struct serv
 	attach = mpm->attachments;
 	while (attach) {
 		if ((mpm_session_cmp_sub(attach->session, server_id, context_id) == true)) {
-			server_id_printable = server_id_str(NULL, &(attach->session->server_id));
 			OC_DEBUG(2, "[s(%s),c(0x%x)] Attachment - AttachmentID(0x%x) handle(0x%x)",
-				  server_id_printable, attach->session->context_id,
+				  server_id_str_buf((attach->session->server_id), &idbuf), attach->session->context_id,
 				  attach->AttachmentID, attach->handle);
-			talloc_free(server_id_printable);
+			talloc_free(server_id_str_buf((attach->session->server_id), &idbuf));
 			mpm_session_release(attach->session);
 			DLIST_REMOVE(mpm->attachments, attach);
 			talloc_free(attach);
@@ -1118,10 +1099,9 @@ static NTSTATUS cache_unbind(struct serv
 	stream = mpm->streams;
 	while (stream) {
 		if ((mpm_session_cmp_sub(stream->session, server_id, context_id) == true)) {
-			server_id_printable = server_id_str(NULL, &(stream->session->server_id));
 			OC_DEBUG(2, "[s(%s),c(0x%x)] Stream - handle(0x%x)",
-				  server_id_printable, stream->session->context_id, stream->handle);
-			talloc_free(server_id_printable);
+				  server_id_str_buf((stream->session->server_id), &idbuf), stream->session->context_id, stream->handle);
+			talloc_free(server_id_str_buf((stream->session->server_id), &idbuf));
 			mpm_session_release(stream->session);
 			mpm_cache_stream_close(stream);
 			talloc_free(stream->filename);
