diff --git a/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/traceSubscription/ts001.c b/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/traceSubscription/ts001.c
index 97843fc7c..a2cc3ffe6 100644
--- a/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/traceSubscription/ts001.c
+++ b/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/traceSubscription/ts001.c
@@ -35,7 +35,7 @@ static jvmtiExtensionFunction subscribe  = NULL;
 static jvmtiExtensionFunction unsubscribe  = NULL;
 static jvmtiExtensionFunction flush  = NULL;
 static jvmtiExtensionFunction metadata = NULL;
-void *subscriptionID;
+static void *subscriptionID;
 volatile static jint bufferCount = 0;
 volatile static jint bufferCountFinal = 0;
 volatile static int completed = 0;
diff --git a/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/verboseGC/vgc001.c b/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/verboseGC/vgc001.c
index 2fda9e8bd..d11047e44 100644
--- a/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/verboseGC/vgc001.c
+++ b/runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/verboseGC/vgc001.c
@@ -31,15 +31,15 @@ static agentEnv * env;
 static jvmtiExtensionFunction subscribe  = NULL;
 static jvmtiExtensionFunction unsubscribe  = NULL;
 
-void *subscriptionID;
+static void *subscriptionID;
 volatile static jint bufferCount = 0;
 volatile static int alarmed = 0;
 
-void *subscriptionID2;
+static void *subscriptionID2;
 volatile static jint bufferCount2 = 0;
 volatile static int alarmed2 = 0;
 
-void *subscriptionID3;
+static void *subscriptionID3;
 volatile static jint bufferCount3 = 0;
 volatile static int alarmed3 = 0;
 
