diff -aur openspades-0.0.12.pristine/Sources/Client/Client_Input.cpp openspades-0.0.12.new/Sources/Client/Client_Input.cpp
--- openspades-0.0.12.pristine/Sources/Client/Client_Input.cpp	2016-03-07 10:11:47.126752964 +1030
+++ openspades-0.0.12.new/Sources/Client/Client_Input.cpp	2016-03-07 10:13:31.009665382 +1030
@@ -38,6 +38,7 @@
 
 #include "NetClient.h"
 
+using namespace std;
 
 SPADES_SETTING(cg_mouseSensitivity, "1");
 SPADES_SETTING(cg_zoomedMouseSensScale, "0.6");
diff -aur openspades-0.0.12.pristine/Sources/Client/Corpse.cpp openspades-0.0.12.new/Sources/Client/Corpse.cpp
--- openspades-0.0.12.pristine/Sources/Client/Corpse.cpp	2016-03-07 10:11:47.126752964 +1030
+++ openspades-0.0.12.new/Sources/Client/Corpse.cpp	2016-03-07 10:13:31.049663957 +1030
@@ -27,6 +27,8 @@
 #include "../Core/Debug.h"
 #include "../Core/Settings.h"
 
+using namespace std;
+
 SPADES_SETTING(r_corpseLineCollision, "1");
 
 namespace spades {
diff -aur openspades-0.0.12.pristine/Sources/Draw/SWMapRenderer.cpp openspades-0.0.12.new/Sources/Draw/SWMapRenderer.cpp
--- openspades-0.0.12.pristine/Sources/Draw/SWMapRenderer.cpp	2016-03-07 10:11:47.133419386 +1030
+++ openspades-0.0.12.new/Sources/Draw/SWMapRenderer.cpp	2016-03-07 10:13:31.092995748 +1030
@@ -31,6 +31,8 @@
 #include "SWUtils.h"
 #include <cstdint>
 
+using namespace std;
+
 SPADES_SETTING(r_swUndersampling, "0");
 
 namespace spades {
