--- a/recon-ng
+++ b/recon-ng
@@ -49,7 +49,7 @@
 parser.add_argument('-w', help='load/create a workspace', metavar='workspace', dest='workspace', action='store')
 parser.add_argument('-r', help='load commands from a resource file', metavar='filename', dest='script_file', action='store')
 parser.add_argument('--no-version', help='disable version check', dest='check', default=True, action='store_false')
-parser.add_argument('--no-analytics', help='disable analytics reporting', dest='analytics', default=True, action='store_false')
+parser.add_argument('--analytics', help='enable analytics reporting', dest='analytics', default=False, action='store_true')
 parser.add_argument('--no-marketplace', help='disable remote module management', dest='marketplace', default=True, action='store_false')
 parser.add_argument('--stealth', help='disable all passive requests (--no-*)', dest='stealth', default=False, action='store_true')
 parser.add_argument('--accessible', help='Use accessible outputs when available', dest='accessible', default=False, action='store_true')

--- a/recon-cli
+++ b/recon-cli
@@ -81,7 +81,7 @@
 parser.add_argument('-o', help='set a module option (can be used more than once)', metavar='name=value', dest='options', default=[], action='append')
 parser.add_argument('-x', help='run the module', dest='run', default=False, action='store_true')
 parser.add_argument('--no-version', help='disable version check', dest='check', default=True, action='store_false')
-parser.add_argument('--no-analytics', help='disable analytics reporting', dest='analytics', default=True, action='store_false')
+parser.add_argument('--analytics', help='enable analytics reporting', dest='analytics', default=False, action='store_true')
 parser.add_argument('--no-marketplace', help='disable remote module management', dest='marketplace', default=True, action='store_false')
 parser.add_argument('--stealth', help='disable all passive requests (--no-*)', dest='stealth', default=False, action='store_true')
 parser.add_argument('--version', help='displays the current version', action='version', version=base.__version__)
