--- a/recon-ng
+++ b/recon-ng
@@ -42,6 +42,6 @@ parser = argparse.ArgumentParser(description=description, version=base.__version
 parser.add_argument('-w', help='load/create a workspace', metavar='workspace', dest='workspace', action='store')
 parser.add_argument('-r', help='load commands from a resource file', metavar='filename', dest='script_file', action='store')
 parser.add_argument('--no-check', help='disable version check', dest='check', default=True, action='store_false')
-parser.add_argument('--no-analytics', help='disable analytics reporting', dest='analytics', default=True, action='store_false')
+parser.add_argument('--analytics', help='enable analytics reporting', dest='analytics', default=False, action='store_true')
 args = parser.parse_args()
 recon_ui(args)
