--- a/system76-driver	2020-01-15 10:51:43.000000000 -0500
+++ b/system76-driver	2020-01-20 22:09:04.666956825 -0500
@@ -59,7 +59,7 @@
 
 log.info('** Process start at monotonic time %r', time.monotonic())
 log.info('system76driver.__version__: %r', system76driver.__version__)
-log.info('OS: %r', platform.dist())
+log.info('OS: Arch Linux')
 log.info('kernel: %r', platform.uname().release)
 log.info('model: %r', args.model)
 
--- a/system76driver/gtk.py	2020-01-20 21:50:46.812227180 -0500
+++ b/system76driver/gtk.py	2020-01-20 22:19:02.673782867 -0500
@@ -53,9 +53,7 @@
         self.notify_text = self.builder.get_object('notifyLabel')
         self.details = self.builder.get_object('detailsText')
         self.builder.get_object('sysModel').set_text(model)
-        self.builder.get_object('ubuntuVersion').set_text(
-            '{} {} ({})'.format(*platform.dist())
-        )
+        self.builder.get_object('ubuntuVersion').set_text('Arch Linux')
         self.builder.get_object('driverVersion').set_text(__version__)
 
         self.builder.connect_signals({
