--- osmin-1.14.3/backend/CMakeLists.txt.orig	2025-08-07 14:07:05.570116105 +0200
+++ osmin-1.14.3/backend/CMakeLists.txt	2025-08-07 14:08:38.488998271 +0200
@@ -151,26 +151,9 @@
     libosmscout_4.patch
 )
 foreach(patch IN LISTS libosmscout_patches)
-    message(STATUS "Patch ${patch} : CHECK")
-    execute_process(
-        COMMAND git apply --check ${CMAKE_CURRENT_SOURCE_DIR}/${patch}
-        WORKING_DIRECTORY ${libosmscout_SOURCE_DIR}
-        RESULT_VARIABLE presult
-        OUTPUT_QUIET ERROR_QUIET)
-    if(presult)
-        message(STATUS "Patch ${patch} : REVERT")
-        execute_process(
-            COMMAND git apply --reverse ${CMAKE_CURRENT_SOURCE_DIR}/${patch}
-            WORKING_DIRECTORY ${libosmscout_SOURCE_DIR}
-            RESULT_VARIABLE presult
-            OUTPUT_QUIET ERROR_QUIET)
-        if(presult)
-            message(FATAL_ERROR "Patch ${patch} failed")
-        endif()
-    endif()
     message(STATUS "Patch ${patch} : APPLY")
     execute_process(
-        COMMAND git apply ${CMAKE_CURRENT_SOURCE_DIR}/${patch}
+        COMMAND patch -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/${patch}
         WORKING_DIRECTORY ${libosmscout_SOURCE_DIR})
 endforeach()
 
