--- CMakeLists.txt.orig	2018-05-23 15:55:25.082206324 +0200
+++ CMakeLists.txt	2018-05-23 15:55:34.908872875 +0200
@@ -65,8 +65,8 @@
 set(CMAKE_EXE_LINKER_FLAGS_DEBUG "-pg")
 
 # release build flags (overwrite default cmake release flags)
-set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
+set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG -Doverride=")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG -Doverride= -fpermissive")
 
 # generic flags
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
