diff -aur -bZwB -x .git CoCoALib-0.99555.old/configure CoCoALib-0.99555/configure
--- CoCoALib-0.99555.old/configure	2017-09-09 09:58:50.883000041 -0400
+++ CoCoALib-0.99555/configure	2017-09-09 09:59:23.732924254 -0400
@@ -1236,8 +1236,8 @@
 then
 ####  echo "GSL_INC_DIR=\"$GSL_INC_DIR\""                     >> "$CONFIG_TMP"
 ####  echo "GSL_INCLUDE=-I \$(GSL_INC_DIR)"                   >> "$CONFIG_TMP"
-  echo "GSL_LIBS=\$(EXTLIBS)/lib/libgsl-symlink.a"            >> "$CONFIG_TMP"
-  echo "GSL_LDLIBS=-lgsl-symlink -lblas -llapack"             >> "$CONFIG_TMP"
+  echo "GSL_LIBS=\$(EXTLIBS)/lib/libgsl-symlink.so"            >> "$CONFIG_TMP"
+  echo "GSL_LDLIBS=-lgsl-symlink -lcblas -llapack"             >> "$CONFIG_TMP"
   echo "GSL_VERSION="                                         >> "$CONFIG_TMP"
 fi
 echo                                                          >> "$CONFIG_TMP"
@@ -1262,7 +1262,7 @@
 if [ $HAVE_NORMALIZ = "yes" ]
 then
   # Recall that GMPXX_LIB is the libgmpxx.a library
-  echo "NORMALIZ_LIBS=\$(EXTLIBS)/lib/libnormaliz-symlink.a \$(GMPXX_LIB)" >> "$CONFIG_TMP"
+  echo "NORMALIZ_LIBS=\$(EXTLIBS)/lib/libnormaliz-symlink.so \$(GMPXX_LIB)" >> "$CONFIG_TMP"
   echo "NORMALIZ_LDLIBS=-lnormaliz-symlink"                   >> "$CONFIG_TMP"
   echo "NORMALIZ_VERSION="                                    >> "$CONFIG_TMP"
 fi
Only in CoCoALib-0.99555: output
