diff -u a/package.py b/package.py
--- a/package.py	2018-05-15 12:06:16.517181436 -0400
+++ b/package.py	2018-05-15 12:06:40.300515181 -0400
@@ -35,7 +35,6 @@
     shutil.copy('run.py', op.join(destpath, 'run.py'))
     extra_ignores = ['*.so'] if not with_so else None
     copy_packages(packages, destpath, extra_ignores=extra_ignores)
-    os.remove(op.join(destpath, 'qt', 'run_template.py')) # It doesn't belong in the package.
     shutil.copytree(op.join('build', 'help'), op.join(destpath, 'help'))
     shutil.copytree(op.join('build', 'locale'), op.join(destpath, 'locale'))
     compileall.compile_dir(destpath)
