--- a/build.js	2020-08-08 08:32:26.749005286 +0100
+++ b/build.js	2020-08-08 08:35:51.382338803 +0100
@@ -20,7 +20,7 @@
     arch = Arch.arm64
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', 'apk', 'rpm', 'AppImage'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
@@ -55,7 +55,7 @@
   linux: {
     category: 'Network',
     icon: '_icons/icon.png',
-    target: ['deb', 'zip', 'apk', 'rpm', 'AppImage'],
+    target: ['dir'],
   },
   mac: {
     category: 'public.app-category.utilities',
