diff -r -u a/build.js b/build.js
--- a/build.js	2021-03-07 08:35:45.210329013 +0000
+++ b/build.js	2021-03-07 08:37:56.830332848 +0000
@@ -24,7 +24,7 @@
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
@@ -67,7 +67,7 @@
   linux: {
     category: 'Network',
     icon: '_icons/icon.svg',
-    target: ['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'],
+    target: ['dir'],
   },
   mac: {
     category: 'public.app-category.utilities',
