diff -r -u a/build.js b/build.js
--- a/build.js	2020-12-18 15:50:23.000000000 +0000
+++ b/build.js	2020-12-24 18:13:01.432309794 +0000
@@ -20,12 +20,13 @@
     arch = Arch.arm64
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020 freetubeapp@protonmail.com',
+  electronVersion: '11.1.0',
   // asar: false,
   // compression: 'store',
   productName,
@@ -63,7 +64,7 @@
   linux: {
     category: 'Network',
     icon: '_icons/icon.png',
-    target: ['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'],
+    target: ['dir'],
   },
   mac: {
     category: 'public.app-category.utilities',
