--- build.js	2023-08-21 18:20:50.310787096 +0100
+++ builddiff.js	2023-08-21 18:25:50.210785281 +0100
@@ -36,12 +36,13 @@
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', '7z', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020-2023 freetubeapp@protonmail.com',
+  electronDist: '/usr/lib/electron22',
   // asar: false,
   // compression: 'store',
   productName,
