diff --git a/src/alpm_config.vala b/src/alpm_config.vala
index 95ce58c..4ed8614 100644
--- a/src/alpm_config.vala
+++ b/src/alpm_config.vala
@@ -340,7 +340,7 @@ namespace Alpm {
 				if ("Never" in directive) {
 					if (affect_package) {
 						default_level &= ~Signature.Level.PACKAGE;
-						default_level |= Signature.Level.PACKAGE_SET;
+//						default_level |= Signature.Level.PACKAGE_SET;
 					}
 					if (affect_database) {
 						default_level &= ~Signature.Level.DATABASE;
@@ -349,7 +349,7 @@ namespace Alpm {
 					if (affect_package) {
 						default_level |= Signature.Level.PACKAGE;
 						default_level |= Signature.Level.PACKAGE_OPTIONAL;
-						default_level |= Signature.Level.PACKAGE_SET;
+//						default_level |= Signature.Level.PACKAGE_SET;
 					}
 					if (affect_database) {
 						default_level |= Signature.Level.DATABASE;
@@ -359,7 +359,7 @@ namespace Alpm {
 					if (affect_package) {
 						default_level |= Signature.Level.PACKAGE;
 						default_level &= ~Signature.Level.PACKAGE_OPTIONAL;
-						default_level |= Signature.Level.PACKAGE_SET;
+//						default_level |= Signature.Level.PACKAGE_SET;
 					}
 					if (affect_database) {
 						default_level |= Signature.Level.DATABASE;
@@ -369,7 +369,7 @@ namespace Alpm {
 					if (affect_package) {
 						default_level &= ~Signature.Level.PACKAGE_MARGINAL_OK;
 						default_level &= ~Signature.Level.PACKAGE_UNKNOWN_OK;
-						default_level |= Signature.Level.PACKAGE_TRUST_SET;
+//						default_level |= Signature.Level.PACKAGE_TRUST_SET;
 					}
 					if (affect_database) {
 						default_level &= ~Signature.Level.DATABASE_MARGINAL_OK;
@@ -379,7 +379,7 @@ namespace Alpm {
 					if (affect_package) {
 						default_level |= Signature.Level.PACKAGE_MARGINAL_OK;
 						default_level |= Signature.Level.PACKAGE_UNKNOWN_OK;
-						default_level |= Signature.Level.PACKAGE_TRUST_SET;
+//						default_level |= Signature.Level.PACKAGE_TRUST_SET;
 					}
 					if (affect_database) {
 						default_level |= Signature.Level.DATABASE_MARGINAL_OK;
@@ -395,7 +395,7 @@ namespace Alpm {
 
 		public Signature.Level merge_siglevel (Signature.Level base_level, Signature.Level over_level) {
 			if ((over_level & Signature.Level.USE_DEFAULT) == 0) {
-				if ((over_level & Signature.Level.PACKAGE_SET) == 0) {
+/*				if ((over_level & Signature.Level.PACKAGE_SET) == 0) {
 					over_level |= base_level & Signature.Level.PACKAGE;
 					over_level |= base_level & Signature.Level.PACKAGE_OPTIONAL;
 				}
@@ -403,7 +403,7 @@ namespace Alpm {
 					over_level |= base_level & Signature.Level.PACKAGE_MARGINAL_OK;
 					over_level |= base_level & Signature.Level.PACKAGE_UNKNOWN_OK;
 				}
-			}
+*/			}
 			return over_level;
 		}
 	}diff --git a/src/daemon.vala b/src/daemon.vala
index 4727b9c..b7b6a2e 100644
--- a/src/daemon.vala
+++ b/src/daemon.vala
@@ -1204,9 +1204,9 @@ private void cb_event (Event.Data data) {
 		case Event.Type.PACSAVE_CREATED:
 			details += data.pacsave_created_file;
 			break;
-		case Event.Type.PACORIG_CREATED:
-			details += data.pacorig_created_file;
-			break;
+//		case Event.Type.PACORIG_CREATED:
+//			details += data.pacorig_created_file;
+//			break;
 		default:
 			break;
 	}diff --git a/src/transaction.vala b/src/transaction.vala
index cae7d48..3954f55 100644
--- a/src/transaction.vala
+++ b/src/transaction.vala
@@ -980,9 +980,9 @@ namespace Pamac {
 				case Alpm.Event.Type.PACSAVE_CREATED:
 					progress_dialog.spawn_in_term ({"echo", dgettext (null, "%s installed as %s.pacsave").printf (details[0])});
 					break;
-				case Alpm.Event.Type.PACORIG_CREATED:
-					progress_dialog.spawn_in_term ({"echo", dgettext (null, "%s installed as %s.pacorig").printf (details[0])});
-					break;
+//				case Alpm.Event.Type.PACORIG_CREATED:
+//					progress_dialog.spawn_in_term ({"echo", dgettext (null, "%s installed as %s.pacorig").printf (details[0])});
+//					break;
 				default:
 					break;
 			}diff --git a/vapi/libalpm.vapi b/vapi/libalpm.vapi
index 0017ded..6f919d2 100644
--- a/vapi/libalpm.vapi
+++ b/vapi/libalpm.vapi
@@ -684,8 +684,8 @@ namespace Alpm {
 			DATABASE_MARGINAL_OK = (1 << 12),
 			DATABASE_UNKNOWN_OK = (1 << 13),
 
-			PACKAGE_SET = (1 << 27),
-			PACKAGE_TRUST_SET = (1 << 28),
+//			PACKAGE_SET = (1 << 27),
+//			PACKAGE_TRUST_SET = (1 << 28),
 
 			USE_DEFAULT = (1 << 31)
 		}
@@ -754,7 +754,11 @@ namespace Alpm {
 			INTERCONFLICTS_START,
 			/** Inter-conflicts were checked for target package. */
 			INTERCONFLICTS_DONE,
-			/** Package will be installed/upgraded/downgraded/re-installed/removed; See
+	    /** Processing the package transaction is starting. */
+	    TRANSACTION_START,
+	    /** Processing the package transaction is finished. */
+	    TRANSACTION_DONE,
+						/** Package will be installed/upgraded/downgraded/re-installed/removed; See
 			 * PackageOperation for arguments. */
 			PACKAGE_OPERATION_START,
 			/** Package was installed/upgraded/downgraded/re-installed/removed; See
@@ -824,9 +828,14 @@ namespace Alpm {
 			/** A .pacsave file was created; See PacsaveCreated for
 			 * arguments */
 			PACSAVE_CREATED,
-			/** A .pacorig file was created; See PacorigCreated for
-			 * arguments */
-			PACORIG_CREATED
+	    /** Processing hooks will be started. */
+	    EVENT_HOOK_START,
+	    /** Processing hooks is finished. */
+	    EVENT_HOOK_DONE,
+	    /** A hook is starting */
+	    EVENT_HOOK_RUN_START,
+	    /** A hook has finnished runnning */
+	    EVENT_HOOK_RUN_DONE
 		}
 
 		[CCode (cname = "alpm_event_any_t", has_type_id = false)]
@@ -1261,6 +1270,7 @@ namespace Alpm {
 		TRANS_ABORT,
 		TRANS_TYPE,
 		TRANS_NOT_LOCKED,
+		TRANS_HOOK_FAILED,
 		/* Packages */
 		PKG_NOT_FOUND,
 		PKG_IGNORED,