diff --git a/webui/index.css b/webui/index.css
index c01e9e8..bebf340 100644
--- a/webui/index.css
+++ b/webui/index.css
@@ -70,6 +70,10 @@ input[type=number] {
   -moz-appearance: textfield;
 }
 
+.modal {
+    background-color: #fff;
+}
+
 .invalid {
     border-color: red;
 }
@@ -113,4 +117,8 @@ div.description {
         color: #888;
     }
 
+    .modal {
+        background-color: #333;
+    }
+
 }
diff --git a/webui/modal.js b/webui/modal.js
index ac87b3c..796c532 100644
--- a/webui/modal.js
+++ b/webui/modal.js
@@ -33,7 +33,6 @@ class Modal {
             'outline': '1px solid black',
             'font-weight': 'bold',
             'text-align': 'center',
-            'background-color': 'white',
             'margin': 'auto',
             'position': 'relative',
             'margin-top': dialogOffset + 'px',
@@ -41,6 +40,7 @@ class Modal {
             'height': height + 'px',
             'z-index': '3'
         });
+        this.dialog.classList.add('modal');
         this.main.prepend(this.dialog);
         if (title) {
             let heading = this.create('h3', title);
