--- ./amd/amdgpu/gmc_v9_0.c.orig	2019-07-23 18:08:17.902391268 +0000
+++ ./amd/amdgpu/gmc_v9_0.c	2019-07-23 18:11:10.455718188 +0000
@@ -1046,7 +1046,11 @@
 		pci_set_consistent_dma_mask(adev->pdev, DMA_BIT_MASK(32));
 		printk(KERN_WARNING "amdgpu: No coherent DMA available.\n");
 	}
+#if DRM_VERSION_CODE >= DRM_VERSION(5, 2, 0)
+	adev->need_swiotlb = drm_need_swiotlb(dma_bits);
+#else
 	adev->need_swiotlb = drm_get_max_iomem() > ((u64)1 << dma_bits);
+#endif
 
 	if (adev->gmc.xgmi.supported) {
 		r = gfxhub_v1_1_get_xgmi_info(adev);
