diff -c -r pgmodeler-0.9.0-alpha1-orig/conf/defaults/pgmodeler.conf pgmodeler-0.9.0-alpha1/conf/defaults/pgmodeler.conf
*** pgmodeler-0.9.0-alpha1-orig/conf/defaults/pgmodeler.conf	2017-02-14 16:29:17.431695027 +0100
--- pgmodeler-0.9.0-alpha1/conf/defaults/pgmodeler.conf	2017-02-14 16:33:11.807849231 +0100
***************
*** 19,25 ****
                 code-tab-width="4"
                 canvas-corner-move="true"
                 invert-rangesel-trigger="false"
!                check-update="true"
                 save-last-position="true"
                 show-main-menu="false"
                 disable-smoothness="false"
--- 19,25 ----
                 code-tab-width="4"
                 canvas-corner-move="true"
                 invert-rangesel-trigger="false"
!                check-update="false"
                 save-last-position="true"
                 show-main-menu="false"
                 disable-smoothness="false"
diff -c -r pgmodeler-0.9.0-alpha1-orig/conf/pgmodeler.conf pgmodeler-0.9.0-alpha1/conf/pgmodeler.conf
*** pgmodeler-0.9.0-alpha1-orig/conf/pgmodeler.conf	2017-02-14 16:29:17.431695027 +0100
--- pgmodeler-0.9.0-alpha1/conf/pgmodeler.conf	2017-02-14 16:33:04.520989402 +0100
***************
*** 19,25 ****
                 code-tab-width="4"
                 canvas-corner-move="true"
                 invert-rangesel-trigger="false"
!                check-update="true"
                 save-last-position="true"
                 show-main-menu="false"
                 disable-smoothness="false"
--- 19,25 ----
                 code-tab-width="4"
                 canvas-corner-move="true"
                 invert-rangesel-trigger="false"
!                check-update="false"
                 save-last-position="true"
                 show-main-menu="false"
                 disable-smoothness="false"
