diff --git a/src/robomongo/gui/widgets/workarea/WelcomeTab.cpp b/src/robomongo/gui/widgets/workarea/WelcomeTab.cpp
index cdc1519c..3bdb6d11 100644
--- a/src/robomongo/gui/widgets/workarea/WelcomeTab.cpp
+++ b/src/robomongo/gui/widgets/workarea/WelcomeTab.cpp
@@ -73,7 +73,7 @@ namespace Robomongo
     QString const blog4 = blog.arg("http://blog.robomongo.org/robomongo-1-rc1/",
         "Robomongo RC8", "14 Apr 2016");
 
-    QUrl const pic1_URL = "http://blog.robomongo.org/content/images/2017/02/bottom.png";
+    QUrl const pic1_URL = QUrl("http://blog.robomongo.org/content/images/2017/02/bottom.png");
 
     WelcomeTab::WelcomeTab(QScrollArea *parent) :
         QWidget(parent), _parent(parent)
