--- photofilmstrip/core/ProjectFile.py	2014-10-14 22:11:01.000000000 +0000
+++ photofilmstrip/core/ProjectFilev1.py	2016-08-25 22:27:13.553841009 +0000
@@ -196,7 +196,7 @@
     def __ThumbToQuery(self, picId, pic):
         pilThumb = PILBackend.GetThumbnail(pic, height=120)
         thumbWidth, thumbHeight = pilThumb.size
-        thumbData = buffer(pilThumb.tostring())
+        thumbData = buffer(pilThumb.tobytes())
         
         query = "INSERT INTO `thumbnail` (" \
                     "picture_id, width, height, data" \
