--- simmain.cc.orig	2020-02-06 20:30:33.480323954 +0100
+++ simmain.cc	2020-02-06 20:32:05.704175036 +0100
@@ -419,6 +419,11 @@
 
 	env_t::init();
 
++    // Make simutrans re-symlinks the files when opened
+#ifdef __linux__
+    system("\cp -rs --remove-destination /usr/share/games/simutrans/ ~/.local/share/");
+#endif
+    
 	// you really want help with this?
 	if (gimme_arg(argc, argv, "-h",     0) ||
 			gimme_arg(argc, argv, "-?",     0) ||
@@ -503,8 +508,9 @@
 		strcat( env_t::program_dir, PATH_SEPARATOR );
 	}
 	else {
-		strcpy( env_t::program_dir, argv[0] );
-		*(strrchr( env_t::program_dir, PATH_SEPARATOR[0] )+1) = 0;
+        static char buffer[100];
+        sprintf(buffer, "%s/.local/share/simutrans/", getenv("HOME"));
+        strcpy(env_t::program_dir, buffer);
 
 #ifdef __APPLE__
 		// change working directory from binary dir to bundle dir
