--- simmain.cc.orig	2020-02-07 13:55:44.413381961 +0100
+++ simmain.cc	2020-02-07 14:00:06.824546151 +0100
@@ -401,6 +401,13 @@
 	std::set_new_handler(sim_new_handler);
 
 	env_t::init();
+    
+    // Make simutrans extended re-symlinks the files when opened
+#ifdef __linux__
+    system("\cp -rs --remove-destination /usr/share/games/simutrans-extended/ ~/.local/share/");
+    system("\cp -rs --remove-destination /usr/share/games/simutrans/ ~/.local/share/");
+    system("\cp -rs --remove-destination ~/.local/share/simutrans/pak* ~/.local/share/simutrans-extended/");
+#endif
 
 	// you really want help with this?
 	if (gimme_arg(argc, argv, "-h",     0) ||
@@ -495,8 +502,9 @@
 		strcat( env_t::program_dir, path_sep );
 	}
 	else {
-		strcpy( env_t::program_dir, argv[0] );
-		*(strrchr( env_t::program_dir, path_sep[0] )+1) = 0;
+        static char buffer[100];
+        sprintf(buffer, "%s/.local/share/simutrans-extended/", getenv("HOME"));
+        strcpy(env_t::program_dir, buffer);
 
 #ifdef __APPLE__
 		// change working directory from binary dir to bundle dir
