diff --git a/plugins/core/Standard/qPCL/PclUtils/filters/MLSSmoothingUpsampling.cpp b/plugins/core/Standard/qPCL/PclUtils/filters/MLSSmoothingUpsampling.cpp
index 59c803c9..04a23edf 100644
--- a/plugins/core/Standard/qPCL/PclUtils/filters/MLSSmoothingUpsampling.cpp
+++ b/plugins/core/Standard/qPCL/PclUtils/filters/MLSSmoothingUpsampling.cpp
@@ -25,6 +25,7 @@
 
 //PCL
 #include <pcl/surface/mls.h>
+#include <pcl/search/kdtree.h>
 
 //qCC_plugins
 #include <ccMainAppInterface.h>
diff --git a/plugins/core/Standard/qPCL/PclUtils/utils/my_point_types.h b/plugins/core/Standard/qPCL/PclUtils/utils/my_point_types.h
index 3128e838..7f5cd7f4 100644
--- a/plugins/core/Standard/qPCL/PclUtils/utils/my_point_types.h
+++ b/plugins/core/Standard/qPCL/PclUtils/utils/my_point_types.h
@@ -21,6 +21,7 @@
 //PCL
 #include <pcl/register_point_struct.h>
 #include <pcl/point_types.h>
+#include <boost/cstdint.hpp>
 
 //! PCL custom point type used for reading RGB data
 struct OnlyRGB
