--- src/pdf2djvu-0.9.18.2/pdf-backend.cc.orig	2022-04-29 23:29:42.368147604 +0200
+++ src/pdf2djvu-0.9.18.2/pdf-backend.cc	2022-04-29 23:29:53.624814042 +0200
@@ -130,7 +130,11 @@
  */
 
 pdf::Document::Document(const std::string &file_name)
+#if POPPLER_VERSION >= 220300
+: ::PDFDoc(std::make_unique<GooString>(file_name))
+#else
 : ::PDFDoc(new pdf::String(file_name.c_str()), nullptr, nullptr)
+#endif
 {
   if (!this->isOk())
     throw LoadError();
--- src/pdf2djvu-0.9.18.2/pdf2djvu.cc.orig	2022-04-29 23:29:28.464814542 +0200
+++ src/pdf2djvu-0.9.18.2/pdf2djvu.cc	2022-04-29 23:29:59.251480598 +0200
@@ -89,7 +89,15 @@
 #endif
   }
   else
+  {
+#if POPPLER_VERSION >= 220300
+    // Poppler removed copy() method
+    // https://cgit.freedesktop.org/poppler/poppler/commit/poppler/Link.h?id=7a429c3cf9fba67ee736a52aa4f12e0c177f1405
+    dest.reset(new LinkDest(*orig_dest));
+#else
     dest.reset(orig_dest->copy());
+#endif
+  }
   if (dest.get() != nullptr)
   {
     int page;
