--- a/Makefile	2023-05-29 16:19:43.216071155 -0400
+++ b/Makefile	2023-05-29 16:21:19.496966792 -0400
@@ -1,12 +1,12 @@
 PREFIX = .
-CC = cc
+CC ?= cc
-CFLAGS = -Wall -O2 -I$(PREFIX)/include
-LDFLAGS = -L$(PREFIX)/lib
+CFLAGS += -Wall -O2 -I$(PREFIX)/include
+LDFLAGS += -L$(PREFIX)/lib
 
 all: pdftxt
 %.o: %.c
 	$(CC) -c $(CFLAGS) $<
 pdftxt: pdftxt.o
-	$(CC) -o $@ $^ $(LDFLAGS) -lmupdf -lmupdfthird -lm
+	$(CC) -o $@ $^ $(LDFLAGS) -lfreetype -lgumbo -lharfbuzz -ljbig2dec -ljpeg -lmupdf -lopenjp2 -lz -lm
 clean:
 	rm -f *.o pdftxt
