--- pebble-tool/pebble_tool/sdk/emulator.py	2015-07-22 18:00:15.000000000 -0400
+++ pebble-tool/pebble_tool/sdk/emulator.py.new	2015-07-26 09:04:42.588688037 -0400
@@ -136,7 +136,7 @@
             "-serial", "null",
             "-serial", "tcp::{},server,nowait".format(self.qemu_port),
             "-serial", "tcp::{},server".format(self.qemu_serial_port),
-            "-pflash", qemu_micro_flash,
+            "-drive",  "if=pflash,format=raw,readonly,file=" + qemu_micro_flash,
         ]
 
         platform_args = {

--- Pebble/common/phonesim/libpebble/pebble/PebbleEmulator.py	2015-07-22 18:00:03.000000000 -0400
+++ Pebble/common/phonesim/libpebble/pebble/PebbleEmulator.py.new	2015-07-26 09:09:05.287946813 -0400
@@ -91,7 +91,7 @@
         cmdline.extend(["-serial", "tcp::{},server,nowait".format(QEMU_DEFAULT_BT_PORT)])
         cmdline.extend(["-serial", "tcp::{},server,nowait".format(QEMU_DEFAULT_CONSOLE_PORT)])
         cmdline.extend(["-machine", "pebble-bb2", "-cpu", "cortex-m3"])
-        cmdline.extend(["-pflash", qemu_micro_flash])
+        cmdline.extend(["-drive", "if=pflash,format=raw,readonly,file=" + qemu_micro_flash])
         cmdline.extend(["-mtdblock", qemu_spi_flash])
         cmdline.extend(["-pidfile", self.qemu_pid])
 
