diff -Nur beebeep-code-906.orig/src/gui/AudioManager.h beebeep-code-906/src/gui/AudioManager.h
--- beebeep-code-906.orig/src/gui/AudioManager.h	2016-10-13 20:24:35.000000000 +0200
+++ beebeep-code-906/src/gui/AudioManager.h	2017-05-31 19:49:00.079185000 +0200
@@ -26,7 +26,7 @@
 
 #include "Config.h"
 #ifdef BEEBEEP_USE_PHONON4
-  #include <phonon/MediaObject>
+  #include <KDE/Phonon/MediaObject>
 #endif
 
 
