--- photoview-2.3.6/api/example.env	2021-08-31 04:15:42.000000000 -0600
+++ photoview_new/api/example.env	2021-09-06 17:17:37.496926372 -0600
@@ -1,4 +1,4 @@
-# Copy this file to .env
+# Copy this file to /etc/photoview.env

 PHOTOVIEW_DATABASE_DRIVER=mysql
 PHOTOVIEW_MYSQL_URL=user:password@tcp(localhost)/dbname
@@ -8,7 +8,7 @@
 # PHOTOVIEW_POSTGRES_URL=postgres://user:password@host:port/dbname?sslmode=(disable|allow|...)

 # Specifies the filepath for the sqlite database, if PHOTOVIEW_DATABASE_DRIVER is set to 'sqlite'
-# PHOTOVIEW_SQLITE_PATH=photoview.db
+# PHOTOVIEW_SQLITE_PATH=/var/lib/photoview/photoview.db

 PHOTOVIEW_LISTEN_IP=localhost
 PHOTOVIEW_LISTEN_PORT=4001
@@ -18,10 +18,11 @@
 PHOTOVIEW_UI_ENDPOINT=http://localhost:1234/

 # Path where media should be cached, defaults to ./media_cache
-# PHOTOVIEW_MEDIA_CACHE=./media_cache
+PHOTOVIEW_MEDIA_CACHE=/var/cache/photoview/media_cache

 # Set to 1 for the server to also serve the built static ui files
-PHOTOVIEW_SERVE_UI=0
+PHOTOVIEW_SERVE_UI=1
+PHOTOVIEW_UI_PATH=/usr/share/webapps/photoview-ui

 # Enter a valid mapbox token, to enable maps feature
 # A token can be created for free at https://mapbox.com
@@ -29,4 +30,4 @@

 # Set to 1 to set server in development mode, this enables graphql playground
 # Remove this if running in production
-PHOTOVIEW_DEVELOPMENT_MODE=1
+PHOTOVIEW_DEVELOPMENT_MODE=0
