diff --git a/cmake/modules/Findpicojson.cmake b/cmake/modules/Findpicojson.cmake
index d81d72fb..ad193840 100644
--- a/cmake/modules/Findpicojson.cmake
+++ b/cmake/modules/Findpicojson.cmake
@@ -1,7 +1,7 @@
 # SPDX-FileCopyrightText: 2025 Contributors to the Media eXchange Layer project.
 # SPDX-License-Identifier: Apache-2.0
 
-find_path(picojson_INCLUDE_DIR NAMES picojson/picojson.h DOC "The picojson include directory")
+find_path(picojson_INCLUDE_DIR NAMES picojson.h DOC "The picojson include directory")
 mark_as_advanced(picojson_INCLUDE_DIR)
 
 include(FindPackageHandleStandardArgs)
@@ -16,7 +16,7 @@ if(picojson_FOUND)
 #pragma once
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored \"-Wmaybe-uninitialized\"
-#include <picojson/picojson.h>
+#include <picojson.h>
 #pragma GCC diagnostic pop")
 
     set(picojson_INCLUDE_DIRS ${picojson_INCLUDE_DIR})
