diff -ur Pike-v7.8.866.orig/src/configure Pike-v7.8.866/src/configure
--- Pike-v7.8.866.orig/src/configure	2015-10-16 16:48:44.065646517 +0200
+++ Pike-v7.8.866/src/configure	2015-10-16 16:49:30.248249664 +0200
@@ -77196,39 +77196,6 @@
 CONF_CPPFLAGS_EXTRA="-I$srcdir -I`pwd`"
 CPPFLAGS="${CPPFLAGS} ${CONF_CPPFLAGS_EXTRA}"
 
-# We want to know if we ought to append any abi indicator to the libdir.
-echo "$as_me:$LINENO: checking if an abi indicator would be appropriate for libdir" >&5
-echo $ECHO_N "checking if an abi indicator would be appropriate for libdir... $ECHO_C" >&6
-if test "${pike_cv_lib_abi+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-
-  pike_cv_lib_abi=""
-  for dd in "$with_root/lib" "$with_root/usr/lib" "$prefix/lib"; do
-    for d in $pike_cv_abi_suffixes; do
-      if test "x$d" = "x/."; then :;
-      elif test -d "$dd$d/."; then
-        pike_cv_lib_abi="$d"
-	break
-      fi
-    done
-  done
-
-fi
-
-echo "$as_me:$LINENO: result: $pike_cv_lib_abi" >&5
-echo "${ECHO_T}$pike_cv_lib_abi" >&6
-
-echo "$as_me:$LINENO: checking what libdir should be set to" >&5
-echo $ECHO_N "checking what libdir should be set to... $ECHO_C" >&6
-if test "x$pike_cv_lib_abi" = "x"; then :;
-elif test -d `eval echo "$libdir$pike_cv_lib_abi/."` -o \
-        "x$libdir" = 'x${exec_prefix}/lib'; then
-  libdir="$libdir$pike_cv_lib_abi"
-fi
-echo "$as_me:$LINENO: result: $libdir" >&5
-echo "${ECHO_T}$libdir" >&6
-
 #############################################################################
 
 ## At this point we have at last initialized the compiler!
diff -ur Pike-v7.8.866.orig/src/configure.in Pike-v7.8.866/src/configure.in
--- Pike-v7.8.866.orig/src/configure.in	2015-10-16 16:48:44.065646517 +0200
+++ Pike-v7.8.866/src/configure.in	2015-10-16 16:49:18.045354608 +0200
@@ -2800,30 +2800,6 @@
 CONF_CPPFLAGS_EXTRA="-I$srcdir -I`pwd`"
 CPPFLAGS="${CPPFLAGS} ${CONF_CPPFLAGS_EXTRA}"  
 
-# We want to know if we ought to append any abi indicator to the libdir.
-AC_MSG_CHECKING(if an abi indicator would be appropriate for libdir)
-AC_CACHE_VAL(pike_cv_lib_abi, [
-  pike_cv_lib_abi=""
-  for dd in "$with_root/lib" "$with_root/usr/lib" "$prefix/lib"; do
-    for d in $pike_cv_abi_suffixes; do
-      if test "x$d" = "x/."; then :;
-      elif test -d "$dd$d/."; then
-        pike_cv_lib_abi="$d"
-	break
-      fi
-    done
-  done
-])
-AC_MSG_RESULT($pike_cv_lib_abi)
-
-AC_MSG_CHECKING(what libdir should be set to)
-if test "x$pike_cv_lib_abi" = "x"; then :;
-elif test -d `eval echo "$libdir$pike_cv_lib_abi/."` -o \
-        "x$libdir" = 'x${exec_prefix}/lib'; then
-  libdir="$libdir$pike_cv_lib_abi"
-fi
-AC_MSG_RESULT($libdir)
-
 #############################################################################
 
 ## At this point we have at last initialized the compiler!
