diff -Naur ./nemo-media-columns-3.2.1+betsy.orig/nemo-media-columns.py ./nemo-media-columns-3.2.1+betsy/nemo-media-columns.py
--- ./nemo-media-columns-3.2.1+betsy.orig/nemo-media-columns.py	2017-02-02 13:16:13.056986227 -0500
+++ ./nemo-media-columns-3.2.1+betsy/nemo-media-columns.py	2017-02-02 13:16:37.710320043 -0500
@@ -33,7 +33,10 @@
 # for reading videos. for future improvement, this can also read mp3!
 import kaa.metadata
 # for reading image dimensions
-import Image
+try:
+	from PIL import Image
+except:
+	import Image
 # for reading pdf
 try:
 	from pyPdf import PdfFileReader
