diff --git a/PKGBUILD b/PKGBUILD
index bb1e5c6..8e4148d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,22 +1,36 @@
+_name=CodeIntel
+_pkg=$(tr '[:upper:]' '[:lower:]' <<< $_name)
 
-pkgname=codeintel
+pkgname=("python-${_pkg}" "python2-${_pkg}")
 pkgver=<pkgver>
 pkgrel=1
 pkgdesc="Komodo Edit CodeIntel"
 url="https://github.com/Kronuz/CodeIntel"
-depends=('python' )
-makedepends=('python3' )
+#url="https://github.com/Komodo/KomodoEdit/wiki/290-CodeIntel-OOP-Protocol#set-environment"
+makedepends=('python-setuptools' 'python2-setuptools')
 license=('MPL 1.1')
 arch=('any')
 source=('<source>')
 md5sums=('<hash>')
 
+prepare() {
+#   mv "${srcdir}/${_name}-${pkgver}/"{SilverCity,silvercity}
+    cp -a "${srcdir}/${_name}-${pkgver}"{,-python2}
+}
+
 build() {
-    cd $srcdir/CodeIntel-<pkgver>
+    cd "${srcdir}/${_name}-${pkgver}"
     python setup.py build
+    cd "${srcdir}/${_name}-${pkgver}-python2"
+    python2 setup.py build
 }
 
-package() {
-    cd $srcdir/CodeIntel-<pkgver>
-    python setup.py install --root="$pkgdir" --optimize=1 
+package_python-codeintel() {
+    cd "${srcdir}/${_name}-${pkgver}"
+    python setup.py install --root="$pkgdir" --optimize=1
+}
+
+package_python2-codeintel() {
+    cd "${srcdir}/${_name}-${pkgver}-python2"
+    python2 setup.py install --root="$pkgdir" --optimize=1
 }
