diff --unified --recursive --text imageio-ffmpeg-0.4.6.orig/setup.py imageio-ffmpeg-0.4.6.new/setup.py
--- imageio-ffmpeg-0.4.6.orig/setup.py	2022-04-13 00:00:00.000000000 +0000
+++ imageio-ffmpeg-0.4.6.new/setup.py	2022-04-14 00:00:00.000000000 +0000
@@ -57,7 +57,7 @@
     platforms="any",
     provides=["imageio_ffmpeg"],
     python_requires=">=3.5",
-    setup_requires=["pip>19"],
+    setup_requires=[],
     install_requires=[],
     packages=["imageio_ffmpeg"],
     package_dir={"imageio_ffmpeg": "imageio_ffmpeg"},
