diff --git a/main.js b/main.js
index 94d4352..a294ba4 100644
--- a/main.js
+++ b/main.js
@@ -37,9 +37,6 @@ app.commandLine.appendSwitch('disable-features', 'IOSurfaceCapturer');
 // Enable Opus RED field trial.
 app.commandLine.appendSwitch('force-fieldtrials', 'WebRTC-Audio-Red-For-Opus/Enabled/');
 
-// Enable optional PipeWire support.
-app.commandLine.appendSwitch('enable-features', 'WebRTCPipeWireCapturer');
-
 // Needed until robot.js is fixed: https://github.com/octalmage/robotjs/issues/580
 app.allowRendererProcessReuse = false;
 
