diff -Naurp a/tests/common.py b/tests/common.py
--- a/tests/common.py	2019-06-09 20:32:43.467373877 +0000
+++ b/tests/common.py	2019-06-09 20:26:01.000000000 +0000
@@ -74,7 +74,7 @@ for category in ["Gtk", "Gdk", "GLib-GOb
 
 detect_leaks = os.environ.get("PITIVI_TEST_DETECT_LEAKS", "0") not in ("0", "")
 os.environ["PITIVI_USER_CACHE_DIR"] = tempfile.mkdtemp(suffix="pitiviTestsuite")
-locale.setlocale(locale.LC_ALL, "en_US")
+locale.setlocale(locale.LC_ALL, "en_US.UTF-8")
 
 
 def __create_settings(proxyingStrategy=ProxyingStrategy.NOTHING,
