diff -aur src/python-sipsimple-release-3.5.0/deps/pjsip/pjlib/include/pj/list.h src.orig/python-sipsimple-release-3.5.0/deps/pjsip/pjlib/include/pj/list.h
--- src/python-sipsimple-release-3.5.0/deps/pjsip/pjlib/include/pj/list.h	2020-02-14 12:06:07.000000000 +0100
+++ src.orig/python-sipsimple-release-3.5.0/deps/pjsip/pjlib/include/pj/list.h	2020-08-02 10:40:15.118079182 +0200
@@ -71,7 +71,7 @@
  * must initialize the 'value' member to an appropriate value (typically the
  * owner itself).
  */
-struct pj_list
+extern struct pj_list
 {
     PJ_DECL_LIST_MEMBER(void);
 } PJ_ATTR_MAY_ALIAS; /* may_alias avoids warning with gcc-4.4 -Wall -O2 */
