diff --git a/cmake/cpp.cmake b/cmake/cpp.cmake
index 4315cd044..90c13e588 100644
--- a/cmake/cpp.cmake
+++ b/cmake/cpp.cmake
@@ -267,12 +267,12 @@ foreach(PROTO_FILE IN LISTS proto_files)
   #message(STATUS "protoc src: ${PROTO_SRC}")
   add_custom_command(
     OUTPUT ${PROTO_SRC} ${PROTO_HDR}
-    COMMAND protobuf::protoc
+    COMMAND /usr/bin/protoc
     "--proto_path=${PROJECT_SOURCE_DIR}"
     ${PROTO_DIRS}
     "--cpp_out=${PROJECT_BINARY_DIR}"
     ${PROTO_FILE}
-    DEPENDS ${PROTO_FILE} protobuf::protoc
+    DEPENDS ${PROTO_FILE} /usr/bin/protoc
     COMMENT "Generate C++ protocol buffer for ${PROTO_FILE}"
     VERBATIM)
   list(APPEND PROTO_HDRS ${PROTO_HDR})
