diff --git i/plotjuggler_plugins/DataLoadMCAP/CMakeLists.txt w/plotjuggler_plugins/DataLoadMCAP/CMakeLists.txt
index 2e8dcf00..f7005464 100644
--- i/plotjuggler_plugins/DataLoadMCAP/CMakeLists.txt
+++ w/plotjuggler_plugins/DataLoadMCAP/CMakeLists.txt
@@ -25,7 +25,7 @@ else()
     cpmaddpackage(
       NAME mcap
       GITHUB_REPOSITORY foxglove/mcap
-      GIT_TAG releases/cpp/v1.4.1
+      GIT_TAG releases/cpp/v2.0.2
       DOWNLOAD_ONLY YES)
     # mcap has no CMake support, so we create our own target
 
