diff --git a/gui/src/ocpn_platform.cpp b/gui/src/ocpn_platform.cpp
index 5d6a89cff..857f4d7cc 100644
--- a/gui/src/ocpn_platform.cpp
+++ b/gui/src/ocpn_platform.cpp
@@ -1248,6 +1248,31 @@ void OCPNPlatform::SetDefaultOptions() {
     pConfig->SetPath("/Settings/WMM");
     pConfig->Write("ShowIcon", true);
     pConfig->Write("ShowLiveIcon", true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libcelestial_navigation_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libclimatology_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libocpn_draw_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libo-charts_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libpolar_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libroute_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libweatherfax_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libweather_routing_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
   }
 #endif
 
diff --git a/model/src/config_vars.cpp b/model/src/config_vars.cpp
index c501c2962..191e0e28c 100644
--- a/model/src/config_vars.cpp
+++ b/model/src/config_vars.cpp
@@ -27,7 +27,7 @@
 #include "model/config_vars.h"
 #include "config.h"
 
-bool g_allow_arb_system_plugin = false;
+bool g_allow_arb_system_plugin = true;
 bool g_always_send_rmb_rmc = false;
 bool g_bAISRolloverShowClass = false;
 bool g_bAISRolloverShowCOG = false;
