--- a/gui/src/OCPNPlatform.cpp
+++ b/gui/src/OCPNPlatform.cpp
@@ -1321,6 +1321,30 @@ void OCPNPlatform::SetDefaultOptions(void) {
     pConfig->SetPath(_T ( "/Settings/WMM" ));
     pConfig->Write(_T ( "ShowIcon" ), true);
     pConfig->Write(_T ( "ShowLiveIcon" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libcelestial_navigation_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libclimatology_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libocpn_draw_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libo-charts_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libpolar_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libroute_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libweatherfax_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
+
+    pConfig->SetPath(_T ( "/PlugIns/libweather_routing_pi.so" ));
+    pConfig->Write(_T ( "bEnabled" ), true);
   }
 #endif
 
--- a/model/src/plugin_loader.cpp
+++ b/model/src/plugin_loader.cpp
@@ -559,17 +559,17 @@ bool PluginLoader::LoadPluginCandidate(const wxString& file_name,
   if (!base_plugin_path.EndsWith(wxFileName::GetPathSeparator()))
     base_plugin_path += wxFileName::GetPathSeparator();
 
-  if (!g_bportable) {
-    if (base_plugin_path.IsSameAs(plugin_file_path)) {
-      if (!IsSystemPluginPath(file_name.ToStdString())) {
-        DEBUG_LOG << "Skipping plugin " << file_name << " in "
-                  << g_BasePlatform->GetPluginDir();
-
-        ClearLoadStamp(plugin_loadstamp.ToStdString());  // Not a fatal error
-        return false;
-      }
-    }
-  }
+  // if (!g_bportable) {
+  //   if (base_plugin_path.IsSameAs(plugin_file_path)) {
+  //     if (!IsSystemPluginPath(file_name.ToStdString())) {
+  //       DEBUG_LOG << "Skipping plugin " << file_name << " in "
+  //                 << g_BasePlatform->GetPluginDir();
+  //
+  //       ClearLoadStamp(plugin_loadstamp.ToStdString());  // Not a fatal error
+  //       return false;
+  //     }
+  //   }
+  // }
 
   if (!IsSystemPluginPath(file_name.ToStdString()) && safe_mode::get_mode()) {
     DEBUG_LOG << "Skipping plugin " << file_name << " in safe mode";
