--- a/server/src/utils/database-backups.ts
+++ b/server/src/utils/database-backupsn.ts
@@ -149,7 +149,7 @@
   }
 
   return {
-    bin: `/usr/lib/postgresql/${databaseMajorVersion}/bin/${bin}`,
+    bin: `/usr/bin/${bin}`,
     args,
     databasePassword: isUrlConnection ? new URL(databaseConfig.url).password : databaseConfig.password,
     databaseVersion,
--- a/server/src/services/backup.service.spec.ts
+++ b/server/src/services/backup.service.specn.ts
@@ -250,7 +250,7 @@
         mocks.database.getPostgresVersion.mockResolvedValue(postgresVersion);
         await sut.handleBackupDatabase();
         expect(mocks.process.spawnDuplexStream).toHaveBeenCalledWith(
-          `/usr/lib/postgresql/${expectedVersion}/bin/pg_dump`,
+          `/usr/bin/pg_dump`,
           expect.any(Array),
           expect.any(Object),
         );
