--- postgresql-15.10/src/pl/plperl/GNUmakefile	2024-11-11 23:42:37.000000000 +0100
+++ postgresql-15.10/src/pl/plperl/GNUmakefile	2024-11-20 10:50:43.010003534 +0100
@@ -55,6 +55,9 @@

 SHLIB_LINK = $(perl_embed_ldflags)

+# Force rpath to be used even though we disable it everywhere else
+SHLIB_LINK += $(rpath)
+
 REGRESS_OPTS = --dbname=$(PL_TESTDB) --dlpath=$(top_builddir)/src/test/regress
 REGRESS = plperl_setup plperl plperl_lc plperl_trigger plperl_shared \
 	plperl_elog plperl_util plperl_init plperlu plperl_array \
