diff --git a/src/Powercord/managers/styles.js b/src/Powercord/managers/styles.js
index 65b97b7..ab4ac64 100644
--- a/src/Powercord/managers/styles.js
+++ b/src/Powercord/managers/styles.js
@@ -13,10 +13,12 @@ const ErrorTypes = Object.freeze({
   INVALID_MANIFEST: 'INVALID_MANIFEST'
 });
 
+const XDG_DATA_HOME = process.env.XDG_DATA_HOME || join(process.env.HOME, '.local', 'share');
+
 module.exports = class StyleManager {
   constructor () {
     this._coreStyles = [];
-    this.themesDir = join(__dirname, '../themes');
+    this.themesDir = join(XDG_DATA_HOME, 'powercord', 'themes');
     this.themes = new Map();
 
     if (!window.__SPLASH__) {
diff --git a/src/Powercord/plugins/pc-moduleManager/components/manage/Base.jsx b/src/Powercord/plugins/pc-moduleManager/components/manage/Base.jsx
index 8aa39d1..c126c58 100644
--- a/src/Powercord/plugins/pc-moduleManager/components/manage/Base.jsx
+++ b/src/Powercord/plugins/pc-moduleManager/components/manage/Base.jsx
@@ -5,6 +5,7 @@ const { Button, Tooltip, ContextMenu, Divider, Icons: { Overflow } } = require('
 const { TextInput } = require('powercord/components/settings');
 
 const { emptyStateImage } = getModule([ 'emptyStateImage', 'emptyStateSubtext' ], false);
+const XDG_DATA_HOME = process.env.XDG_DATA_HOME || join(process.env.HOME, '.local', 'share');
 
 class Base extends React.Component {
   constructor () {
@@ -93,7 +94,7 @@ class Base extends React.Component {
           {
             type: 'button',
             name: Messages[`POWERCORD_${this.state.key}_OPEN_FOLDER`],
-            onClick: () => shell.openPath(join(__dirname, '..', '..', '..', '..', this.constructor.name.toLowerCase()))
+            onClick: () => shell.openPath(join(XDG_DATA_HOME, 'powercord', this.constructor.name.toLowerCase()))
           },
           {
             type: 'button',
