--- "src/premake-4.4-beta5/build/gmake.unix/Premake4 copy.make"	2025-04-02 12:58:36.495930555 +0200
+++ b/build/gmake.unix/Premake4.make	2025-04-02 13:01:14.235294725 +0200
@@ -24,12 +24,12 @@
   TARGETDIR  = ../../bin/release
   TARGET     = $(TARGETDIR)/premake4
   DEFINES   += -DLUA_USE_POSIX -DLUA_USE_DLOPEN
-  INCLUDES  += -I../../src/host/lua-5.1.4/src
+  INCLUDES  += -I/usr/include/lua5.1
   ALL_CFLAGS    += $(CFLAGS)   $(DEFINES) $(INCLUDES)
   ALL_CXXFLAGS  += $(CXXFLAGS) $(DEFINES) $(INCLUDES)
   ALL_LDFLAGS   += $(LDFLAGS)
   LDDEPS    +=
-  LIBS      += $(LDDEPS) -lm -ldl
+  LIBS      += $(LDDEPS) -lm -ldl -llua5.1
   LINKCMD    = $(CC) -o $(TARGET) $(OBJECTS) $(RESOURCES) $(ALL_LDFLAGS) $(LIBS)
   define PREBUILDCMDS
   endef
@@ -79,36 +79,7 @@
 	$(OBJDIR)/premake.o \
 	$(OBJDIR)/premake_main.o \
 	$(OBJDIR)/scripts.o \
-	$(OBJDIR)/string_endswith.o \
-	$(OBJDIR)/lapi.o \
-	$(OBJDIR)/lauxlib.o \
-	$(OBJDIR)/lbaselib.o \
-	$(OBJDIR)/lcode.o \
-	$(OBJDIR)/ldblib.o \
-	$(OBJDIR)/ldebug.o \
-	$(OBJDIR)/ldo.o \
-	$(OBJDIR)/ldump.o \
-	$(OBJDIR)/lfunc.o \
-	$(OBJDIR)/lgc.o \
-	$(OBJDIR)/linit.o \
-	$(OBJDIR)/liolib.o \
-	$(OBJDIR)/llex.o \
-	$(OBJDIR)/lmathlib.o \
-	$(OBJDIR)/lmem.o \
-	$(OBJDIR)/loadlib.o \
-	$(OBJDIR)/lobject.o \
-	$(OBJDIR)/lopcodes.o \
-	$(OBJDIR)/loslib.o \
-	$(OBJDIR)/lparser.o \
-	$(OBJDIR)/lstate.o \
-	$(OBJDIR)/lstring.o \
-	$(OBJDIR)/lstrlib.o \
-	$(OBJDIR)/ltable.o \
-	$(OBJDIR)/ltablib.o \
-	$(OBJDIR)/ltm.o \
-	$(OBJDIR)/lundump.o \
-	$(OBJDIR)/lvm.o \
-	$(OBJDIR)/lzio.o \
+	$(OBJDIR)/string_endswith.o
 
 RESOURCES := \
 
@@ -240,122 +211,6 @@
 	@echo $(notdir $<)
 	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
 
-$(OBJDIR)/lapi.o: ../../src/host/lua-5.1.4/src/lapi.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lauxlib.o: ../../src/host/lua-5.1.4/src/lauxlib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lbaselib.o: ../../src/host/lua-5.1.4/src/lbaselib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lcode.o: ../../src/host/lua-5.1.4/src/lcode.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ldblib.o: ../../src/host/lua-5.1.4/src/ldblib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ldebug.o: ../../src/host/lua-5.1.4/src/ldebug.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ldo.o: ../../src/host/lua-5.1.4/src/ldo.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ldump.o: ../../src/host/lua-5.1.4/src/ldump.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lfunc.o: ../../src/host/lua-5.1.4/src/lfunc.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lgc.o: ../../src/host/lua-5.1.4/src/lgc.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/linit.o: ../../src/host/lua-5.1.4/src/linit.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/liolib.o: ../../src/host/lua-5.1.4/src/liolib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/llex.o: ../../src/host/lua-5.1.4/src/llex.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lmathlib.o: ../../src/host/lua-5.1.4/src/lmathlib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lmem.o: ../../src/host/lua-5.1.4/src/lmem.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/loadlib.o: ../../src/host/lua-5.1.4/src/loadlib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lobject.o: ../../src/host/lua-5.1.4/src/lobject.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lopcodes.o: ../../src/host/lua-5.1.4/src/lopcodes.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/loslib.o: ../../src/host/lua-5.1.4/src/loslib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lparser.o: ../../src/host/lua-5.1.4/src/lparser.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lstate.o: ../../src/host/lua-5.1.4/src/lstate.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lstring.o: ../../src/host/lua-5.1.4/src/lstring.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lstrlib.o: ../../src/host/lua-5.1.4/src/lstrlib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ltable.o: ../../src/host/lua-5.1.4/src/ltable.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ltablib.o: ../../src/host/lua-5.1.4/src/ltablib.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/ltm.o: ../../src/host/lua-5.1.4/src/ltm.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lundump.o: ../../src/host/lua-5.1.4/src/lundump.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lvm.o: ../../src/host/lua-5.1.4/src/lvm.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
-$(OBJDIR)/lzio.o: ../../src/host/lua-5.1.4/src/lzio.c
-	@echo $(notdir $<)
-	$(SILENT) $(CC) $(ALL_CFLAGS) $(FORCE_INCLUDE) -o "$@" $(@:%.o) -c "$<"
-
 -include $(OBJECTS:%.o)
 ifneq (,$(PCH))
   -include $(OBJDIR)/$(notdir $(PCH)).d
