--- CPP/7zip/7zip_gcc.mak
+++ CPP/7zip/7zip_gcc.mak
@@ -10 +10 @@
-MY_ASM = asmc
+MY_ASM = uasm
@@ -29 +29 @@
- -fPIC
+ -fPIC -Wno-maybe-uninitialized
@@ -170 +170 @@
-LFLAGS_ALL = -s $(MY_ARCH_2) $(LDFLAGS) $(LD_arch) $(OBJS) $(MY_LIBS) $(LIB2)
+LFLAGS_ALL = -s $(MY_ARCH_2) $(LDFLAGS) -Wl,-z,noexecstack $(LD_arch) $(OBJS) $(MY_LIBS) $(LIB2)

--- CPP/7zip/var_gcc_x64.mak
+++ CPP/7zip/var_gcc_x64.mak
@@ -2 +2 @@
-O=b/g_$(PLATFORM)
+O=$(BUILD_DIR)

--- C/LzmaEnc.c
+++ C/LzmaEnc.c
@@ -2993,12 +2993,13 @@
 
   nowPos64 = p->nowPos64;
   RangeEnc_Init(&p->rc);
-  p->rc.outStream = &outStream.vt;
 
   if (desiredPackSize == 0)
     return SZ_ERROR_OUTPUT_EOF;
 
+  p->rc.outStream = &outStream.vt;
   res = LzmaEnc_CodeOneBlock(p, desiredPackSize, *unpackSize);
+  p->rc.outStream = NULL;
   
   *unpackSize = (UInt32)(p->nowPos64 - nowPos64);
   *destLen -= outStream.rem;
