diff -rU1 grub4dos-efi-efi.orig/preset_menu.lst grub4dos-efi-efi/preset_menu.lst
--- grub4dos-efi-efi.orig/preset_menu.lst	2025-03-30 14:12:57.167147619 +0200
+++ grub4dos-efi-efi/preset_menu.lst	2025-03-30 14:26:41.457182204 +0200
@@ -3,6 +3,6 @@
 
-title find /efi/grub/menu.lst
+title find /EFI/GRUB4DOS/menu.lst
 #	errorcheck off
-#	find --set-root /efi/grub/menu.lst
-	configfile /efi/grub/menu.lst
+#	find --set-root /EFI/GRUB4DOS/menu.lst
+	configfile /EFI/GRUB4DOS/menu.lst
 #	errorcheck on
