diff --git a/kernel/sched/topology.c b/kernel/sched/topology.c
index 1b8791c5cc9b..007dc2c27a97 100644
--- a/kernel/sched/topology.c
+++ b/kernel/sched/topology.c
@@ -2760,5 +2760,11 @@ int sched_numa_find_nth_cpu(const struct cpumask *cpus, int cpu, int node)
 {
 	return cpumask_nth(cpu, cpus);
 }
+
+const struct cpumask *sched_numa_hop_mask(unsigned int node, unsigned int hops)
+{
+	return ERR_PTR(-EINVAL);
+}
+EXPORT_SYMBOL_GPL(sched_numa_hop_mask);
 #endif /* CONFIG_NUMA */
 #endif
