--- src/HSH-2.1.2/HSH/Command.hs	2014-10-28 13:00:59.000000000 +0000
+++ Command.hs	2016-08-20 14:48:31.149988715 +0100
@@ -301,6 +301,15 @@
 #if MIN_VERSION_process(1,2,0)
 			    , delegate_ctlc = False
 #endif
+#if MIN_VERSION_process(1,3,0)
+                            , detach_console = False
+			    , create_new_console = False
+			    , new_session = False
+#endif
+#if MIN_VERSION_process(1,4,0)
+			    , child_group = Nothing
+                            , child_user = Nothing
+#endif
 			   }
     in do (_, oh', _, ph) <- createProcess cp
           let oh = fromJust oh'
@@ -320,6 +329,15 @@
 #if MIN_VERSION_process(1,2,0)
 			    , delegate_ctlc = False
 #endif
+#if MIN_VERSION_process(1,3,0)
+                            , detach_console = False
+			    , create_new_console = False
+			    , new_session = False
+#endif
+#if MIN_VERSION_process(1,4,0)
+			    , child_group = Nothing
+                            , child_user = Nothing
+#endif
 			   }
     in do (ih', oh', _, ph) <- createProcess cp
           let ih = fromJust ih'
