--- projectlibre.sh.orig	2018-04-12 10:44:10.003981216 +0800
+++ projectlibre.sh	2018-04-12 10:55:09.516978700 +0800
@@ -52,9 +52,9 @@
 CONF_DIR="$HOME/.projectlibre"
 RUN_CONF="$CONF_DIR/run.conf"
 VERSION_FILE="$CONF_DIR/java-version"
-JAVA_INSTALL_DIR="/usr/java"
+JAVA_INSTALL_DIR="/usr/lib/jvm/default-runtime"
 JAVA_INSTALL_DIR_EXE="bin/java"
-JAVA_INSTALL_DIR2="/usr/lib/jvm"
+JAVA_INSTALL_DIR2="/usr/lib/jvm/default"
 JAVA_INSTALL_DIR_EXE2="bin/java"
 JAVA_INSTALL_DIR3="/opt/ibm"
 JAVA_INSTALL_DIR_EXE3="jre/bin/java"
@@ -63,8 +63,8 @@
 #Default conf values
 JAVA_EXE="java"
 JAVA_OPTS="-Xms128m -Xmx768m"
-#OPENPROJ_HOME0="/usr/share/projectlibre"
-OPENPROJ_HOME0=`(dirname $0)`
+OPENPROJ_HOME0="/usr/share/projectlibre"
+#OPENPROJ_HOME0=`(dirname $0)`
 OPENPROJ_HOME="$OPENPROJ_HOME0"
 LOG_LEVEL="DEBUG"
 LOG_FILE="/dev/null"
@@ -124,7 +124,8 @@
 		#Comparable format
 		MIN_JAVA_VERSION=`echo "$MIN_JAVA_VERSION" | sed -e 's;\.;0;g'`
 		JAVA_VERSION=`echo "$JAVA_VERSION" | awk '{ print substr($1, 1, 3); }' | sed -e 's;\.;0;g'`
+		JAVA_VERSION=`printf "%3d" $JAVA_VERSION | tr ' ' 1`

 		if [ "$JAVA_VERSION" ]; then
 			if [ "$JAVA_VERSION" -ge "$MIN_JAVA_VERSION" ];	then
 				echo "OK"
