diff --git a/proton b/proton
index d4a49ab..9928e21 100755
--- a/proton
+++ b/proton
@@ -656,6 +656,15 @@ if __name__ == "__main__":
         log("No compat data path?")
         sys.exit(1)
 
+    if "PROTON_USER_COMPAT_DATA" in os.environ:
+        if nonzero(os.environ["PROTON_USER_COMPAT_DATA"]):
+            compatpath = os.environ["STEAM_COMPAT_DATA_PATH"].split('/')
+            os.environ["STEAM_COMPAT_DATA_PATH"] = "{}/{}/{}/{}".format(
+                os.environ["STEAM_COMPAT_CLIENT_INSTALL_PATH"],
+                compatpath[-3], compatpath[-2], compatpath[-1])
+            if not os.path.exists(os.environ["STEAM_COMPAT_DATA_PATH"]):
+                os.makedirs(os.environ["STEAM_COMPAT_DATA_PATH"])
+
     g_proton = Proton(os.path.dirname(sys.argv[0]))
 
     g_proton.extract_tarball()
