--- src/psensor-1.1.3/src/server/server.c.old
+++ src/psensor-1.1.3/src/server/server.c
@@ -192,8 +192,8 @@
 	if (page) {
 		*rp_code = MHD_HTTP_OK;
 
-		resp = MHD_create_response_from_data(strlen(page), page,
-						     MHD_YES, MHD_NO);
+		resp = MHD_create_response_from_buffer(strlen(page), page,
+						     MHD_RESPMEM_MUST_FREE);
 
 		MHD_add_response_header(resp, MHD_HTTP_HEADER_CONTENT_TYPE,
 					"application/json");
@@ -223,8 +223,8 @@
 
 			if (!st.st_size) {
 				fclose(file);
-				return MHD_create_response_from_data
-					(0, NULL, MHD_NO, MHD_NO);
+				return MHD_create_response_from_buffer
+					(0, NULL, MHD_RESPMEM_PERSISTENT);
 			}
 
 			return MHD_create_response_from_callback
@@ -263,10 +263,9 @@
 	char *page = strdup(PAGE_NOT_FOUND);
 	*rp_code = MHD_HTTP_NOT_FOUND;
 
-	return MHD_create_response_from_data(strlen(page),
+	return MHD_create_response_from_buffer(strlen(page),
 					     page,
-					     MHD_YES,
-					     MHD_NO);
+					     MHD_RESPMEM_MUST_FREE);
 }
 
 static int cbk_http_request(void *cls,
