diff -Naur GDriveFS-0.14.3/gdrivefs/gdfs/gdfuse.py GDriveFS-0.14.3-new/gdrivefs/gdfs/gdfuse.py
--- GDriveFS-0.14.3/gdrivefs/gdfs/gdfuse.py	2015-10-18 16:09:54.708226894 -0600
+++ GDriveFS-0.14.3-new/gdrivefs/gdfs/gdfuse.py	2015-10-18 16:15:17.431235540 -0600
@@ -7,6 +7,7 @@
 import atexit
 import resource
 import pprint
+import math
 
 from errno import ENOENT, EIO, ENOTDIR, ENOTEMPTY, EPERM, EEXIST
 from fuse import FUSE, Operations, FuseOSError, c_statvfs, fuse_get_context, \
@@ -161,6 +162,8 @@
             stat_result["st_mode"] = (stat.S_IFREG | effective_permission)
             stat_result["st_nlink"] = 1
 
+        stat_result["st_blocks"]   = int(math.ceil(float(stat_result["st_size"]) / 512.0))
+
         return stat_result
 
     @dec_hint(['raw_path', 'fh'])
