diff --git a/src/plugins/python/pythonlanguageclient.cpp b/src/plugins/python/pythonlanguageclient.cpp
index da3210d5..1f06b129 100644
--- a/src/plugins/python/pythonlanguageclient.cpp
+++ b/src/plugins/python/pythonlanguageclient.cpp
@@ -91,7 +91,7 @@ static PythonLanguageServerState checkPythonLanguageServer(const FilePath &pytho
     const FilePath lspPath = pyLspPath(python);
     if (!lspPath.isEmpty()) {
         const FilePath pylsp = (lspPath / "bin" / "pylsp").withExecutableSuffix();
-        if (pylsp.exists()) {
+        if (FilePath("/usr/bin/pylsp").withExecutableSuffix().exists() || pylsp.exists()) {
             if (pipUsable) {
                 Process pythonProcess;
                 Environment env = pylsp.deviceEnvironment();
