diff --color --unified --recursive --text '--exclude=.git' '--exclude=generated' '--exclude=build' '--exclude=_build' OpenRV.clean/cmake/dependencies/python3.cmake OpenRV/cmake/dependencies/python3.cmake
--- OpenRV.clean/cmake/dependencies/python3.cmake	2023-08-31 22:15:18.786591021 -0700
+++ OpenRV/cmake/dependencies/python3.cmake	2023-09-02 18:10:15.350701802 -0700
@@ -48,6 +48,9 @@
 SET(_pyside2_download_hash
     "87841aaced763b6b52e9b549e31a493f"
 )
+SET(_pyside2_patch_command
+    cd ${PROJECT_SOURCE_DIR} && patch --forward --strip=1 --input=../pyside2-tools-root.patch
+)
 
 SET(_install_dir
     ${RV_DEPS_BASE_DIR}/${_python3_target}/install
@@ -81,6 +84,7 @@
   URL ${_pyside2_archive_url}
   URL_HASH MD5=${_pyside2_download_hash}
   SOURCE_SUBDIR "sources" # Avoids the top level CMakeLists.txt
+  PATCH_COMMAND ${_pyside2_patch_command}
 )
 
 FETCHCONTENT_MAKEAVAILABLE(${_pyside2_target})
