diff -Naur orig/LuxCore-luxcorerender_v2.0/cmake/Dependencies.cmake LuxCore-luxcorerender_v2.0/cmake/Dependencies.cmake
--- orig/LuxCore-luxcorerender_v2.0/cmake/Dependencies.cmake	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/cmake/Dependencies.cmake	2018-05-11 16:22:01.774281916 +0200
@@ -43,7 +43,7 @@
     find_package(PNG REQUIRED)
     include_directories(BEFORE SYSTEM ${PNG_PNG_INCLUDE_DIR})
 	# Find Python Libraries
-	find_package(PythonLibs 3.4)
+	find_package(PythonLibs)
 endif()
 
 find_program(PYSIDE_UIC NAME pyside-uic
diff -Naur orig/LuxCore-luxcorerender_v2.0/CMakeLists.txt LuxCore-luxcorerender_v2.0/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/CMakeLists.txt	2018-05-11 16:30:33.742293724 +0200
@@ -28,6 +28,10 @@
 cmake_minimum_required(VERSION 3.5.1)
 cmake_policy(VERSION 3.0)
 
+if(POLICY CMP0072)
+	cmake_policy(SET CMP0072 OLD)
+endif()
+
 # Remove the following when the version check is at least 2.8.4
 SET(CMAKE_LEGACY_CYGWIN_WIN32 0)
 
diff -Naur orig/LuxCore-luxcorerender_v2.0/samples/luxcoreconsole/CMakeLists.txt LuxCore-luxcorerender_v2.0/samples/luxcoreconsole/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/samples/luxcoreconsole/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/samples/luxcoreconsole/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -28,4 +28,4 @@
 
 add_executable(luxcoreconsole ${LUXCORECONSOLE_SRCS})
 
-TARGET_LINK_LIBRARIES(luxcoreconsole ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${OPENCL_LIBRARIES})
\ Brak znaku nowej linii na końcu pliku
+TARGET_LINK_LIBRARIES(luxcoreconsole ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} ${OPENCL_LIBRARIES})
diff -Naur orig/LuxCore-luxcorerender_v2.0/samples/luxcoredemo/CMakeLists.txt LuxCore-luxcorerender_v2.0/samples/luxcoredemo/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/samples/luxcoredemo/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/samples/luxcoredemo/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcoredemo ${LUXCORELIBDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcoredemo ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${OPENCL_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoredemo ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} ${OPENCL_LIBRARIES})
diff -Naur orig/LuxCore-luxcorerender_v2.0/samples/luxcorescenedemo/CMakeLists.txt LuxCore-luxcorerender_v2.0/samples/luxcorescenedemo/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/samples/luxcorescenedemo/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/samples/luxcorescenedemo/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcorescenedemo ${LUXCORESCENEDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcorescenedemo ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${OPENCL_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcorescenedemo ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} ${OPENCL_LIBRARIES})
diff -Naur orig/LuxCore-luxcorerender_v2.0/samples/luxcoreui/CMakeLists.txt LuxCore-luxcorerender_v2.0/samples/luxcoreui/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/samples/luxcoreui/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/samples/luxcoreui/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -88,4 +88,5 @@
 	${OPENGL_LIBRARIES}
 	${GTK3_LIBRARIES}
 	${Boost_LIBRARIES}
+	${PYTHON_LIBRARIES}
 	${OPENCL_LIBRARIES})
diff -Naur orig/LuxCore-luxcorerender_v2.0/tests/benchsimple/CMakeLists.txt LuxCore-luxcorerender_v2.0/tests/benchsimple/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/tests/benchsimple/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/tests/benchsimple/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -22,4 +22,4 @@
 add_executable(benchsimple benchsimple.cpp)
 add_definitions(${VISIBILITY_FLAGS})
 remove_definitions("-DLUXCORE_DLL")
-target_link_libraries(benchsimple luxrays ${EMBREE_LIBRARY} ${TBB_LIBRARY} )
+target_link_libraries(benchsimple luxrays ${EMBREE_LIBRARY} ${PYTHON_LIBRARIES} ${TBB_LIBRARY} )
diff -Naur orig/LuxCore-luxcorerender_v2.0/tests/luxcoreimplserializationdemo/CMakeLists.txt LuxCore-luxcorerender_v2.0/tests/luxcoreimplserializationdemo/CMakeLists.txt
--- orig/LuxCore-luxcorerender_v2.0/tests/luxcoreimplserializationdemo/CMakeLists.txt	2018-05-06 11:04:25.000000000 +0200
+++ LuxCore-luxcorerender_v2.0/tests/luxcoreimplserializationdemo/CMakeLists.txt	2018-05-11 16:22:01.774281916 +0200
@@ -30,4 +30,4 @@

 add_executable(luxcoreimplserializationdemo ${LUXCOREIMPL_SERIALIZATIONDEMO_SRCS})

-TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${PYTHON_LIBRARIES} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
