--- a/build/build.py	2023-07-28 00:11:05.000000000 +0300
+++ b/build/build.py	2023-07-28 03:46:42.158795877 +0300
@@ -50,11 +50,12 @@
 
 def shell(cmd):
   try:
-    output = subprocess.check_output(cmd)
+    res= subprocess.run(cmd, check=True, capture_output=True)
   except subprocess.CalledProcessError as e:
-    if e.output: print(e.output)
+    if e.stdout: print(e.stdout)
+    if e.stderr: print(e.stderr)
     raise
-  return output.decode("UTF-8").strip()
+  return res.stdout.decode("UTF-8").strip()
 
 
 # Python
