Index: Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Creates 3 column layout on A4 paper and save it under 3columnA4.sla filename. This is a simple way to demonstrate creating a doc on the fly. """
Index: Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Creates 3 column layout on US Letter  paper and save it under 3columnUSLTR.sla filename. This is a simple way to demonstrate creating a doc on the fly. """
Index: Scribus/scribus/plugins/scriptplugin/samples/Calender.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/Calender.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/Calender.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ This Script creates a Calendar Sheet for the Current Month """
Index: Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """
Index: Scribus/scribus/plugins/scriptplugin/samples/Sample1.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/Sample1.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/Sample1.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ A sample script """
Index: Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 import sys
Index: Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """Golden Mean for Scribus.
Index: Scribus/scribus/plugins/scriptplugin/samples/legende.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/legende.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/legende.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ When you have an image selected this script creates small text legende
Index: Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Make selected group smaller by 10% """
Index: Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Make selected group larger by 10% """
Index: Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ This script creates a CD Pochette - a paper pocket for CD/DVD disc """
Index: Scribus/scribus/plugins/scriptplugin/samples/quote.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/quote.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/quote.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: iso-8859-1 -*-
 
 """ This script changes quotation marks from " " to french style """
Index: Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """OK. This is a sample Scribus database connection with Python HOWTO.
Index: Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Draws a "crop marks" around selected object """
Index: Scribus/scribus/plugins/scriptplugin/samples/wordcount.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/samples/wordcount.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/samples/wordcount.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Counts the words in the whole document or in a textframe """
Index: Scribus/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Align_image_in_frame.py
 # This version 2014.04.19
 """
@@ -140,4 +140,4 @@
             scribus.redrawAll()
  
 if __name__ == '__main__':
-    main()
\ No newline at end of file
+    main()
Index: Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # File: quotes.py - changes typewriter quotes to typographic quotes
 # © 2010.08.28 Gregory Pittman
Index: Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # File: spacenquotes.py - changes typewriter quotes to typographic quotes with option to add spaces
 # © 2010.08.28 Gregory Pittman
Index: Scribus/scribus/plugins/scriptplugin/scripts/Caption.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/Caption.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/Caption.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
Index: Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 """
Index: Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ 
Index: Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 # ****************************************************************************
Index: Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
Index: Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf8 -*-
 
 """
Index: Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
Index: Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
Index: Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py
===================================================================
--- Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py	(Revision 22789)
+++ Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py	(Arbeitskopie)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
