diff --git a/src/sage/doctest/forker.py b/src/sage/doctest/forker.py
index bf6d49906de..a52485d3d91 100644
--- a/src/sage/doctest/forker.py
+++ b/src/sage/doctest/forker.py
@@ -553,6 +553,8 @@ class SageDocTestRunner(doctest.DocTestRunner):
         self.total_walltime_skips = 0
         self.total_performed_tests = 0
         self.total_walltime = 0
+        if sys.version_info.minor < 13:
+            self._stats = self._name2ft
 
     def _run(self, test, compileflags, out):
         """
@@ -830,7 +832,10 @@ class SageDocTestRunner(doctest.DocTestRunner):
         self.optionflags = original_optionflags
 
         # Record and return the number of failures and tries.
-        self._DocTestRunner__record_outcome(test, failures, tries)
+        if sys.version_info.minor < 13:
+            self._DocTestRunner__record_outcome(test, failures, tries)
+        else:
+            self._DocTestRunner__record_outcome(test, failures, tries, walltime_skips)
         self.total_walltime_skips += walltime_skips
         self.total_performed_tests += tries
         return TestResults(failures, tries)
@@ -931,7 +936,7 @@ class SageDocTestRunner(doctest.DocTestRunner):
             sage: from sage.doctest.control import DocTestDefaults; DD = DocTestDefaults()
             sage: import doctest, sys, os
             sage: DTR = SageDocTestRunner(SageOutputChecker(), verbose=False, sage_options=DD, optionflags=doctest.NORMALIZE_WHITESPACE|doctest.ELLIPSIS)
-            sage: DTR._name2ft['sage.doctest.forker'] = (1,120)
+            sage: DTR._stats['sage.doctest.forker'] = (1,120)
             sage: results = DTR.summarize()
             **********************************************************************
             1 item had failures:
@@ -946,8 +951,8 @@ class SageDocTestRunner(doctest.DocTestRunner):
         passed = []
         failed = []
         totalt = totalf = 0
-        for x in self._name2ft.items():
-            name, (f, t) = x
+        for x in self._stats.items():
+            name, (f, t, *_) = x
             assert f <= t
             totalt += t
             totalf += f
@@ -972,10 +977,10 @@ class SageDocTestRunner(doctest.DocTestRunner):
             print(self.DIVIDER, file=m)
             print(count_noun(len(failed), "item"), "had failures:", file=m)
             failed.sort()
-            for thing, (f, t) in failed:
+            for thing, (f, t, *_) in failed:
                 print(" %3d of %3d in %s" % (f, t, thing), file=m)
         if verbose:
-            print(count_noun(totalt, "test") + " in " + count_noun(len(self._name2ft), "item") + ".", file=m)
+            print(count_noun(totalt, "test") + " in " + count_noun(len(self._stats), "item") + ".", file=m)
             print("%s passed and %s failed." % (totalt - totalf, totalf), file=m)
             if totalf:
                 print("***Test Failed***", file=m)
diff --git a/src/sage/misc/replace_dot_all.py b/src/sage/misc/replace_dot_all.py
index ea51a9b3159..a196810bd44 100644
--- a/src/sage/misc/replace_dot_all.py
+++ b/src/sage/misc/replace_dot_all.py
@@ -194,14 +194,15 @@ def find_replacements(location, package_regex=None, verbose=False):
                     to_exec = to_exec.replace("'", '').replace('"', '')
                     if (to_exec[-1] == ','):
                         to_exec = to_exec[:-1]
-                    exec(to_exec)
+                    glob = dict()
+                    exec(to_exec, glob)
                 except ModuleNotFoundError as err:
                     print(f'ModuleNotFoundError: {err} found when trying to execute {to_exec}')
                 except ImportError as err:
                     print(f'ImportError: {err} found when trying to execute {to_exec}')
 
                 try:  # try to evaluate the list of module names to get a list of the modules themselves which we can call import_statements on
-                    modules = eval(to_eval)
+                    modules = eval(to_eval, glob)
                 except NameError as err:
                     print(f'NameError: {err} found when trying to evaluate {to_eval} at {location}:{row_index + 1}')
                 except SyntaxError as err:
diff --git a/src/sage/matroids/extension.pyx b/src/sage/matroids/extension.pyx
index 778e6d6ef70..83b792e8d2b 100644
--- a/src/sage/matroids/extension.pyx
+++ b/src/sage/matroids/extension.pyx
@@ -211,6 +211,9 @@ cdef class LinearSubclassesIter:
 
         self._nodes = [first_cut]
 
+    def __iter__(self):
+        return self
+
     def __next__(self):
         """
         Return the next linear subclass.
diff --git a/src/sage/matroids/set_system.pyx b/src/sage/matroids/set_system.pyx
index d9a2f631fbf..0bf9cb2b7fa 100644
--- a/src/sage/matroids/set_system.pyx
+++ b/src/sage/matroids/set_system.pyx
@@ -772,6 +772,9 @@ cdef class SetSystemIterator:
         self._pointer = -1
         self._len = len(H)
 
+    def __iter__(self):
+        return self
+
     def __next__(self):
         """
         Return the next subset of a SetSystem.
diff --git a/src/sage/misc/latex_macros.py b/src/sage/misc/latex_macros.py
index fc389b32a72..1cce2fa6f14 100644
--- a/src/sage/misc/latex_macros.py
+++ b/src/sage/misc/latex_macros.py
@@ -43,6 +43,8 @@
 contain '\newcommand' lines for each of the entries in ``macros``.
 """
 
+import importlib
+
 
 def produce_latex_macro(name, *sample_args):
     r"""
@@ -69,7 +71,7 @@ def produce_latex_macro(name, *sample_args):
          sage: produce_latex_macro('GF', 37)
          '\\newcommand{\\GF}[1]{\\Bold{F}_{#1}}'
 
-    If the Sage object is not in the global name space, describe it
+    If the Sage object is not in the global namespace, describe it
     like so::
 
          sage: produce_latex_macro('sage.rings.finite_rings.finite_field_constructor.FiniteField', 3)
@@ -84,22 +86,16 @@ def produce_latex_macro(name, *sample_args):
     else:
         module, real_name = names_split
     newcommand = '\\newcommand{\\' + real_name + '}'
-    count = 0
-    args = "("
-    for x in sample_args:
-        count += 1
-        args += str(x) + ','
-    args += ')'
-    exec('from ' + module + ' import ' + real_name)
-    if count:
-        defn = '[' + str(count) + ']{'
-        defn += eval('str(LatexCall()(' + real_name + args + '))') + '}'
+    sage_object = getattr(importlib.import_module(module), real_name)
+    if sample_args:
+        defn = '[' + str(len(sample_args)) + ']{'
+        defn += str(LatexCall()(sage_object(*sample_args))) + '}'
     else:
-        defn = '{' + eval('str(LatexCall()(' + real_name + '))') + '}'
-    count = 0
-    for x in sample_args:
-        count += 1
-        defn = defn.replace(str(x), "#" + str(count))
+        defn = '{' + str(LatexCall()(sage_object)) + '}'
+    for i, x in enumerate(sample_args):
+        s = str(x)
+        assert s in defn
+        defn = defn.replace(s, "#" + str(i+1))
     return newcommand + defn
 
 
diff --git a/pyproject.toml b/pyproject.toml
index da06db03649..ca2d483197a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -74,11 +74,12 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "Programming Language :: Python :: Implementation :: CPython",
     "Topic :: Scientific/Engineering :: Mathematics",
 ]
 urls = {Homepage = "https://www.sagemath.org"}
-requires-python = ">=3.9, <3.13"
+requires-python = ">=3.9, <3.14"
 
 [project.optional-dependencies]
 R = [
diff --git a/src/setup.cfg.m4 b/src/setup.cfg.m4
index 969793209c8..3a4ec930b8f 100644
--- a/src/setup.cfg.m4
+++ b/src/setup.cfg.m4
@@ -9,7 +9,7 @@ license_files = LICENSE.txt
 include(`setup_cfg_metadata.m4')dnl'
 
 [options]
-python_requires = >=3.9, <3.13
+python_requires = >=3.9, <3.14
 install_requires =
     SPKG_INSTALL_REQUIRES_six
 dnl From build/pkgs/sagelib/dependencies
diff --git a/src/sage/misc/temporary_file.py b/src/sage/misc/temporary_file.py
index 998260be8eb..de872802805 100644
--- a/src/sage/misc/temporary_file.py
+++ b/src/sage/misc/temporary_file.py
@@ -32,7 +32,9 @@
 # as the parent for all temporary files & directories created by them.
 # This lets us clean up after those two functions when sage exits normally
 # using an atexit hook
-TMP_DIR_FILENAME_BASE = tempfile.TemporaryDirectory()
+# Note that `TemporaryDirectory()` will cleanup on program exit;
+# we keep the atexit hook to be redundant, in case that fails.
+TMP_DIR_FILENAME_BASE = tempfile.TemporaryDirectory(prefix='sage_')
 atexit.register(lambda: TMP_DIR_FILENAME_BASE.cleanup())
 
 
@@ -533,14 +533,15 @@ def spyx_tmp() -> str:
     We cache the result of this function "by hand" so that the same
     temporary directory will always be returned. A function is used to
     delay creating a directory until (if) it is needed. The temporary
-    directory is removed when sage terminates by way of an atexit
-    hook.
+    directory is automatically removed when sage terminates.
     """
     global _spyx_tmp
     if _spyx_tmp:
         return _spyx_tmp
 
-    d = tempfile.TemporaryDirectory()
-    _spyx_tmp = os.path.join(d.name, 'spyx')
-    atexit.register(lambda: d.cleanup())
+    # We don't use `tempfile.TemporaryDirectory()` here because it
+    # will be cleaned up on child exit (e.g. for parallel testing)
+    # For some reason this doesn't affect the `TemporaryDirectory`
+    # stored in the global `TMP_DIR_FILENAME_BASE`.
+    _spyx_tmp = tmp_dir(name='spyx_')
     return _spyx_tmp
