--- abPOA-1.4.2/setup.py	2023-12-13 09:36:16.000000000 +0530
+++ abPOA-1.4.2.new/setup.py	2023-12-17 18:19:19.645303305 +0530
@@ -63,5 +63,5 @@
                     include_dirs=[inc_dir],
                     depends=[src_dir+'abpoa.h', src_dir+'abpoa_align.h', src_dir+'abpoa_graph.h', src_dir+'abpoa_output.h', src_dir+'abpoa_seed.h', src_dir+'abpoa_seq.h', src_dir+'kalloc.h', src_dir+'khash.h', src_dir+'kdq.h', src_dir+'kseq.h', src_dir+'ksort.h', src_dir+'kstring.h', src_dir+'kvec.h', src_dir+'simd_abpoa_align.h', src_dir+'simd_instruction.h', src_dir+'utils.h', 'python/cabpoa.pxd'],
                     libraries = ['z', 'm', 'pthread'],
-                    extra_compile_args=['-O3', '-Wno-error=declaration-after-statement', simde, simd_flag])]
+                    extra_compile_args=['-O3', '-Wno-error=declaration-after-statement', simde])],
 )
