--- MMCorePy_wrap/Makefile.am	2014-01-21 05:25:37.000000000 +0300
+++ MMCorePy_wrap/Makefile_new.am	2014-01-24 17:14:15.904157691 +0300
@@ -34,18 +34,14 @@
 	$(top_srcdir)/MMCore/LoadableModules/LoadedDeviceAdapter.cpp
 
 _MMCorePy.so: MMCorePy_wrap.cxx $(MMCOREPYSOURCES)
-	python setup.py build_ext --inplace
+	python2 setup.py build
 
 all:
 	make MMCorePy_wrap.cxx
 	make MMCorePy.py
 	make _MMCorePy.so
-	cp MMCorePy.py ../bin
-	cp _MMCorePy.so ../bin
 
 install:
-	cp MMCorePy.py $(DEVICEADAPTERPATH)
-	cp _MMCorePy.so $(DEVICEADAPTERPATH)
 
 clean:
 	rm MMCorePy_wrap.cxx || echo "Not found, no problem"
