diff -ur aloha/bin/aloha aloha/bin/aloha
--- a/aloha/bin/aloha
+++ b/aloha/bin/aloha
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 
 import sys
diff -ur bin/.compile.py bin/.compile.py
--- a/bin/.compile.py
+++ b/bin/.compile.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2009 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur bin/mg5 bin/mg5
--- a/bin/mg5
+++ b/bin/mg5
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur bin/mg5_aMC bin/mg5_aMC
--- a/bin/mg5_aMC
+++ b/bin/mg5_aMC
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 import time
 start = time.time()
 
diff -ur madgraph/interface/common_run_interface.py madgraph/interface/common_run_interface.py
--- a/madgraph/interface/common_run_interface.py
+++ b/madgraph/interface/common_run_interface.py
@@ -907,7 +907,7 @@
                 fsock.close()
                 return
             else:
-                subprocess.call(['python', 'write_param_card.py'],
+                subprocess.call(['python2', 'write_param_card.py'],
                              cwd=pjoin(self.me_dir,'bin','internal','ufomodel'))
                 default = pjoin(self.me_dir,'bin','internal','ufomodel','param_card.dat')
 
diff -ur madgraph/iolibs/gen_infohtml.py madgraph/iolibs/gen_infohtml.py
--- a/madgraph/iolibs/gen_infohtml.py
+++ b/madgraph/iolibs/gen_infohtml.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2009 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur madgraph/iolibs/template_files/loop_optimized/check_sa_all.py.inc madgraph/iolibs/template_files/loop_optimized/check_sa_all.py.inc
--- a/madgraph/iolibs/template_files/loop_optimized/check_sa_all.py.inc
+++ b/madgraph/iolibs/template_files/loop_optimized/check_sa_all.py.inc
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 # This is an example of how to run MadLoop from Python using the f2py compilation of the wrapper file 'f2py_wrapper.f'.
 
diff -ur madgraph/iolibs/template_files/loop_optimized/check_sa.py.inc madgraph/iolibs/template_files/loop_optimized/check_sa.py.inc
--- a/madgraph/iolibs/template_files/loop_optimized/check_sa.py.inc
+++ b/madgraph/iolibs/template_files/loop_optimized/check_sa.py.inc
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 # This is an example of how to run MadLoop from Python using the f2py compilation of the wrapper file 'f2py_wrapper.f'.
 
diff -ur madgraph/madweight/blob_solution.py madgraph/madweight/blob_solution.py
--- a/madgraph/madweight/blob_solution.py
+++ b/madgraph/madweight/blob_solution.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 try:
     import madgraph.madweight.particle_class as particle_class
diff -ur madgraph/madweight/Cards.py madgraph/madweight/Cards.py
--- a/madgraph/madweight/Cards.py
+++ b/madgraph/madweight/Cards.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur madgraph/madweight/change_tf.py madgraph/madweight/change_tf.py
--- a/madgraph/madweight/change_tf.py
+++ b/madgraph/madweight/change_tf.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #Extension
 
diff -ur madgraph/madweight/create_param.py madgraph/madweight/create_param.py
--- a/madgraph/madweight/create_param.py
+++ b/madgraph/madweight/create_param.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur madgraph/madweight/create_run.py madgraph/madweight/create_run.py
--- a/madgraph/madweight/create_run.py
+++ b/madgraph/madweight/create_run.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #Extension
 import string,os,sys,re,popen2,time,stat,filecmp
diff -ur madgraph/madweight/diagram_class.py madgraph/madweight/diagram_class.py
--- a/madgraph/madweight/diagram_class.py
+++ b/madgraph/madweight/diagram_class.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys
 import logging
diff -ur madgraph/madweight/mod_file.py madgraph/madweight/mod_file.py
--- a/madgraph/madweight/mod_file.py
+++ b/madgraph/madweight/mod_file.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python2
 ####################################################################################################
 ####################################################################################################
 ##                                                                                                ##
diff -ur madgraph/madweight/MW_driver.py madgraph/madweight/MW_driver.py
--- a/madgraph/madweight/MW_driver.py
+++ b/madgraph/madweight/MW_driver.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 # Copyright (c) 2012 The MadGraph Development team and Contributors             
 #
diff -ur madgraph/madweight/MW_info.py madgraph/madweight/MW_info.py
--- a/madgraph/madweight/MW_info.py
+++ b/madgraph/madweight/MW_info.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur madgraph/madweight/substructure_class.py madgraph/madweight/substructure_class.py
--- a/madgraph/madweight/substructure_class.py
+++ b/madgraph/madweight/substructure_class.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 
 try:
diff -ur madgraph/madweight/verif_event.py madgraph/madweight/verif_event.py
--- a/madgraph/madweight/verif_event.py
+++ b/madgraph/madweight/verif_event.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur madgraph/madweight/write_MadWeight.py madgraph/madweight/write_MadWeight.py
--- a/madgraph/madweight/write_MadWeight.py
+++ b/madgraph/madweight/write_MadWeight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##### -*- coding: cp1252 -*-
 
 #Extension
diff -ur madgraph/various/combine_plots.py madgraph/various/combine_plots.py
--- a/madgraph/various/combine_plots.py
+++ b/madgraph/various/combine_plots.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2013 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur madgraph/various/histograms.py madgraph/various/histograms.py
--- a/madgraph/various/histograms.py
+++ b/madgraph/various/histograms.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2010 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur madgraph/various/plot_djrs.py madgraph/various/plot_djrs.py
--- a/madgraph/various/plot_djrs.py
+++ b/madgraph/various/plot_djrs.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2010 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur madgraph/various/progressbar.py madgraph/various/progressbar.py
--- a/madgraph/various/progressbar.py
+++ b/madgraph/various/progressbar.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 # -*- coding: iso-8859-1 -*-
 #
 # progressbar  - Text progressbar library for python.
diff -ur MadSpin/decay.py MadSpin/decay.py
--- a/MadSpin/decay.py
+++ b/MadSpin/decay.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import division
 from madgraph.interface import reweight_interface
diff -ur MadSpin/madspin MadSpin/madspin
--- a/MadSpin/madspin
+++ b/MadSpin/madspin
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/LO/bin/generate_events Template/LO/bin/generate_events
--- a/Template/LO/bin/generate_events
+++ b/Template/LO/bin/generate_events
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2011 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur Template/LO/bin/internal/addmasses_optional.py Template/LO/bin/internal/addmasses_optional.py
--- a/Template/LO/bin/internal/addmasses_optional.py
+++ b/Template/LO/bin/internal/addmasses_optional.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 #
 # doubleswitch.py
 # Author: Stephen Mrenna
diff -ur Template/LO/bin/internal/Gridpack/gridrun Template/LO/bin/internal/Gridpack/gridrun
--- a/Template/LO/bin/internal/Gridpack/gridrun
+++ b/Template/LO/bin/internal/Gridpack/gridrun
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/LO/bin/madevent Template/LO/bin/madevent
--- a/Template/LO/bin/madevent
+++ b/Template/LO/bin/madevent
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/MadWeight/bin/madweight.py Template/MadWeight/bin/madweight.py
--- a/Template/MadWeight/bin/madweight.py
+++ b/Template/MadWeight/bin/madweight.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2011 The MadGraph Development team and Contributors
diff -ur Template/MadWeight/bin/mw_options Template/MadWeight/bin/mw_options
--- a/Template/MadWeight/bin/mw_options
+++ b/Template/MadWeight/bin/mw_options
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/MadWeight/mod_file/check_model.py Template/MadWeight/mod_file/check_model.py
--- a/Template/MadWeight/mod_file/check_model.py
+++ b/Template/MadWeight/mod_file/check_model.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #test
 
diff -ur Template/MadWeight/mod_file/mod_file.py Template/MadWeight/mod_file/mod_file.py
--- a/Template/MadWeight/mod_file/mod_file.py
+++ b/Template/MadWeight/mod_file/mod_file.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #Extension
 import string
diff -ur Template/MadWeight/mod_file/mod_setup_model Template/MadWeight/mod_file/mod_setup_model
--- a/Template/MadWeight/mod_file/mod_setup_model
+++ b/Template/MadWeight/mod_file/mod_setup_model
@@ -45,7 +45,7 @@
 
         if (-e $tfdir."/TF_".$transfer_fct.".dat") {
       print "TF $transfer_fct found  in $tfdir \n";
-      system("python ./bin/change_tf.py $transfer_fct 0");
+++ b/     system("python2 ./bin/change_tf.py $transfer_fct 0");
       }
         else{print "Error Transfer function $transfer_fct not found \n";
           exit;
diff -ur Template/MadWeight/Python/clean.py Template/MadWeight/Python/clean.py
--- a/Template/MadWeight/Python/clean.py
+++ b/Template/MadWeight/Python/clean.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur Template/MadWeight/Python/expand_MadWeight.py Template/MadWeight/Python/expand_MadWeight.py
--- a/Template/MadWeight/Python/expand_MadWeight.py
+++ b/Template/MadWeight/Python/expand_MadWeight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Module
 import string
diff -ur Template/MadWeight/Python/Info.py Template/MadWeight/Python/Info.py
--- a/Template/MadWeight/Python/Info.py
+++ b/Template/MadWeight/Python/Info.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 def giveInfo(class_):
         if type(class_)!=str:
diff -ur Template/MadWeight/Python/madweight.py Template/MadWeight/Python/madweight.py
--- a/Template/MadWeight/Python/madweight.py
+++ b/Template/MadWeight/Python/madweight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # idea have 2 script
 # madweight.py -> direct submission 
diff -ur Template/MadWeight/Python/put_banner.py Template/MadWeight/Python/put_banner.py
--- a/Template/MadWeight/Python/put_banner.py
+++ b/Template/MadWeight/Python/put_banner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MG/ME/MW                               ##
diff -ur Template/MadWeight/Python/splitbanner.py Template/MadWeight/Python/splitbanner.py
--- a/Template/MadWeight/Python/splitbanner.py
+++ b/Template/MadWeight/Python/splitbanner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ##########################################################################
 ##                                                                      ##
 ##                               MadWeight                              ##
diff -ur Template/MadWeight/Python/tests.py Template/MadWeight/Python/tests.py
--- a/Template/MadWeight/Python/tests.py
+++ b/Template/MadWeight/Python/tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python2
 ####################################################################################################
 ####################################################################################################
 ##                                                                                                ##
diff -ur Template/MadWeight/src/makefile Template/MadWeight/src/makefile
--- a/Template/MadWeight/src/makefile
+++ b/Template/MadWeight/src/makefile
@@ -24,9 +24,9 @@
 	$(FC) $(FFLAGS) -o $(PROG) $(PROCESS) $(MW_PS) $(LINKLIBS)
 
 transfer_card.inc:../../Cards/transfer_card.dat ../../Cards/MadWeight_card.dat
-	python ../../bin/internal/madweight/Cards.py
+
 madweight_card.inc:../../Cards/transfer_card.dat ../../Cards/MadWeight_card.dat
-	python ../../bin/internal/madweight/Cards.py
+
 
 driver.o: phasespace.inc nexternal.inc data.inc coupl.inc madweight_card.inc nb_tf.inc
 initialization.o: madweight_card.inc TF_param.inc coupl.inc run.inc madweight_param.inc pmass.inc nexternal.inc maxamps.inc leshouche.inc nb_tf.inc
diff -ur Template/NLO/bin/aMCatNLO Template/NLO/bin/aMCatNLO
--- a/Template/NLO/bin/aMCatNLO
+++ b/Template/NLO/bin/aMCatNLO
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/NLO/bin/calculate_xsect Template/NLO/bin/calculate_xsect
--- a/Template/NLO/bin/calculate_xsect
+++ b/Template/NLO/bin/calculate_xsect
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2011 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur Template/NLO/bin/generate_events Template/NLO/bin/generate_events
--- a/Template/NLO/bin/generate_events
+++ b/Template/NLO/bin/generate_events
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2011 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur Template/NLO/bin/internal/split_jobs.py Template/NLO/bin/internal/split_jobs.py
--- a/Template/NLO/bin/internal/split_jobs.py
+++ b/Template/NLO/bin/internal/split_jobs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #  MZ, 2012-06-14
 import os
 import sys
diff -ur Template/NLO/bin/shower Template/NLO/bin/shower
--- a/Template/NLO/bin/shower
+++ b/Template/NLO/bin/shower
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2011 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur Template/NLO/Utilities/NLO_Born3.py Template/NLO/Utilities/NLO_Born3.py
--- a/Template/NLO/Utilities/NLO_Born3.py
+++ b/Template/NLO/Utilities/NLO_Born3.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 inp = open('./MADatNLO.top', 'r')
 out = open('./MADatNLO_combined.top', 'w')
diff -ur Template/NLO/Utilities/VetoPrefactors/resum_reweighter.py Template/NLO/Utilities/VetoPrefactors/resum_reweighter.py
--- a/Template/NLO/Utilities/VetoPrefactors/resum_reweighter.py
+++ b/Template/NLO/Utilities/VetoPrefactors/resum_reweighter.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur Template/NLO/Utilities/VetoPrefactors/virt_reweighter.py Template/NLO/Utilities/VetoPrefactors/virt_reweighter.py
--- a/Template/NLO/Utilities/VetoPrefactors/virt_reweighter.py
+++ b/Template/NLO/Utilities/VetoPrefactors/virt_reweighter.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 
 ################################################################################
 #
diff -ur tests/acceptance_tests/test_histograms.py tests/acceptance_tests/test_histograms.py
--- a/tests/acceptance_tests/test_histograms.py
+++ b/tests/acceptance_tests/test_histograms.py
@@ -47,5 +47,4 @@
           histo_list.output(pjoin('%s','HistoOut'), format = 'gnuplot');
         ''' % (sys.path, _file_path, self.IOpath)
 
-        os.system('echo "%s" | python' % line) 
-
+++ b/       os.system('echo "%s" | python2' % line)
diff -ur tests/acceptance_tests/test_model_equivalence.py tests/acceptance_tests/test_model_equivalence.py
--- a/tests/acceptance_tests/test_model_equivalence.py
+++ b/tests/acceptance_tests/test_model_equivalence.py
@@ -322,7 +322,7 @@
             pass
         # prepare for a local compilation
         
-        subprocess.call(['python','write_param_card.py'], cwd=os.path.join(MG5DIR,'models','sm'),
+++ b/       subprocess.call(['python2','write_param_card.py'], cwd=os.path.join(MG5DIR,'models','sm'),
                         stdout=subprocess.PIPE)
         files.cp(os.path.join(MG5DIR,'models','sm','param_card.dat'),
                  join('param_card.dat'))
diff -ur tests/parallel_tests/loop_sample_script.py tests/parallel_tests/loop_sample_script.py
--- a/tests/parallel_tests/loop_sample_script.py
+++ b/tests/parallel_tests/loop_sample_script.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2009 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur tests/parallel_tests/sample_script.py tests/parallel_tests/sample_script.py
--- a/tests/parallel_tests/sample_script.py
+++ b/tests/parallel_tests/sample_script.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2009 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur tests/test_manager.py tests/test_manager.py
--- a/tests/test_manager.py
+++ b/tests/test_manager.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python2
 ################################################################################
 #
 # Copyright (c) 2009 The MadGraph5_aMC@NLO Development team and Contributors
diff -ur vendor/IREGI/src/oneloop/create.py vendor/IREGI/src/oneloop/create.py
--- a/vendor/IREGI/src/oneloop/create.py
+++ b/vendor/IREGI/src/oneloop/create.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import re,string,os,sys,subprocess
 locdir,thisFile = os.path.split(os.path.abspath(sys.argv[0]))
 srcdir = os.path.join(locdir,'src')
diff -ur vendor/IREGI/src/oneloop/example_cpp/create.py vendor/IREGI/src/oneloop/example_cpp/create.py
--- a/vendor/IREGI/src/oneloop/example_cpp/create.py
+++ b/vendor/IREGI/src/oneloop/example_cpp/create.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import re,string,os,sys,subprocess
 locdir,thisFile = os.path.split(os.path.abspath(sys.argv[0]))
 srcdir = os.path.join(locdir,'src')
diff -ur vendor/IREGI/src/oneloop/src/avh_olo.py vendor/IREGI/src/oneloop/src/avh_olo.py
--- a/vendor/IREGI/src/oneloop/src/avh_olo.py
+++ b/vendor/IREGI/src/oneloop/src/avh_olo.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import re,sys
 import avh_pc_olo
 
diff -ur vendor/IREGI/src/oneloop/src/avh_pc_exe.py vendor/IREGI/src/oneloop/src/avh_pc_exe.py
--- a/vendor/IREGI/src/oneloop/src/avh_pc_exe.py
+++ b/vendor/IREGI/src/oneloop/src/avh_pc_exe.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys
 import avh_pc
