diff --git i/spsdk/debuggers/__init__.py w/spsdk/debuggers/__init__.py
index c2d266b..8d28c77 100644
--- i/spsdk/debuggers/__init__.py
+++ w/spsdk/debuggers/__init__.py
@@ -8,7 +8,6 @@
 """Wrappers for various types of debuggers."""
 
 from .debug_probe_jlink import DebugProbePyLink
-from .debug_probe_pemicro import DebugProbePemicro
 from .debug_probe_pyocd import DebugProbePyOCD
 
-__all__ = ["DebugProbePyLink", "DebugProbePemicro", "DebugProbePyOCD"]
+__all__ = ["DebugProbePyLink", "DebugProbePyOCD"]
