diff -rupN oci-python-sdk-2.23.5/setup.py oci-python-sdk-2.23.5b/setup.py
--- oci-python-sdk-2.23.5/setup.py	2020-11-10 12:52:50.000000000 -0500
+++ oci-python-sdk-2.23.5b/setup.py	2020-11-11 20:00:40.763925659 -0500
@@ -31,11 +31,10 @@ with open_relative("README.rst") as f:
 
 requires = [
     "certifi",
-    "configparser==4.0.2",
-    "cryptography==2.8",
-    "pyOpenSSL>=17.5.0,<=19.1.0",
-    "python-dateutil>=2.5.3,<3.0.0",
-    "pytz>=2016.10",
+    "cryptography",
+    "pyOpenSSL",
+    "python-dateutil",
+    "pytz",
 ]
 
 setup(
