diff -rupN oci-python-sdk-2.8.0/setup.py oci-python-sdk-2.8.0b/setup.py
--- oci-python-sdk-2.8.0/setup.py	2019-12-18 15:08:27.000000000 -0500
+++ oci-python-sdk-2.8.0b/setup.py	2019-12-22 23:02:52.019437302 -0500
@@ -30,11 +30,10 @@ with open_relative("README.rst") as f:
 
 requires = [
     "certifi",
-    "configparser>=3.5.0b1",
-    "cryptography>=2.1.4",
-    "pyOpenSSL>=17.5.0",
-    "python-dateutil>=2.5.3,<3.0.0",
-    "pytz>=2016.10",
+    "cryptography",
+    "pyOpenSSL",
+    "python-dateutil",
+    "pytz",
 ]
 
 setup(
