--- timezonefinder-4.0.2/timezonefinder/timezonefinder.py	2019-05-19 13:03:23.032195742 +0200
+++ timezonefinder-4.0.2/timezonefinder/timezonefinder.py	2019-05-19 13:03:35.232238756 +0200
@@ -6,7 +6,7 @@
 from struct import unpack
 
 import numpy as np
-from importlib_resources import open_binary
+from importlib.resources import open_binary
 from numpy import array, dtype, empty, fromfile
 
 from .global_settings import (
