diff --color -ur --text scooby-0.7.1.orig/setup.py scooby-0.7.1.new/setup.py
--- scooby-0.7.1.orig/setup.py	2023-02-16 12:14:14.886929678 +0100
+++ scooby-0.7.1.new/setup.py	2023-02-16 12:14:31.566938366 +0100
@@ -28,7 +28,7 @@
         'Intended Audience :: Science/Research',
         'Natural Language :: English',
     ),
-    python_requires='>=3.7.*',
+    python_requires='>=3.7',
     extras_require={
         'cpu': ['psutil', 'mkl'],
         # 'gpu': [], # TODO: what's needed?
