diff -Naur a/daemon/Makefile.am b/daemon/Makefile.am
--- a/daemon/Makefile.am	2019-10-12 17:26:38.000000000 -0400
+++ b/daemon/Makefile.am	2019-11-13 14:56:10.154495566 -0500
@@ -14,7 +14,7 @@
 DOCS = doc
 endif
 
-PYTHONLIBDIR=$(subst site-packages,dist-packages,$(pythondir))
+PYTHONLIBDIR=$(pythondir)
 
 SUBDIRS = proto $(DOCS)
 
diff -Naur a/ns3/Makefile.am b/ns3/Makefile.am
--- a/ns3/Makefile.am	2019-10-12 17:26:38.000000000 -0400
+++ b/ns3/Makefile.am	2019-11-13 14:56:18.600662969 -0500
@@ -9,7 +9,7 @@
 
 if WANT_PYTHON
 
-PYTHONLIBDIR=$(subst site-packages,dist-packages,$(pythondir))
+PYTHONLIBDIR=$(pythondir)
 
 SETUPPY	= setup.py
 SETUPPYFLAGS = -v
