diff --git a/python-package/setup.py b/python-package/setup.py
index e1433af05..3fa20467d 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -339,7 +339,6 @@
           cmdclass={
               'build_ext': BuildExt,     # type: ignore
               'sdist': Sdist,            # type: ignore
-              'install_lib': InstallLib,  # type: ignore
               'install': Install          # type: ignore
           },
           extras_require={
diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
index 65f7403e6..67390a6bd 100644
--- a/python-package/xgboost/libpath.py
+++ b/python-package/xgboost/libpath.py
@@ -20,6 +20,7 @@ def find_lib_path():
     """
     curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
     dll_path = [
+        '/usr/lib',
         # normal, after installation `lib` is copied into Python package tree.
         os.path.join(curr_path, 'lib'),
         # editable installation, no copying is performed.
