diff -Naur ./python-system_hotkey.orig/system_hotkey/system_hotkey.py ./python-system_hotkey/system_hotkey/system_hotkey.py
--- ./python-system_hotkey.orig/system_hotkey/system_hotkey.py	2021-12-15 12:22:51.996228526 -0500
+++ ./python-system_hotkey/system_hotkey/system_hotkey.py	2021-12-15 12:23:31.192184801 -0500
@@ -285,7 +285,7 @@
 
         thread safe
         '''
-        assert isinstance(hotkey, collections.Iterable) and type(hotkey) not in (str, bytes)
+        assert isinstance(hotkey, collections.abc.Iterable) and type(hotkey) not in (str, bytes)
         if self.consumer == 'callback' and not callback:
             raise TypeError('Function register requires callback argument in non sonsumer mode')
 
