--- pytorch-rocm/torch/csrc/jit/ir/ir.cpp.bak	2023-07-08 14:23:12.514918311 +0200
+++ pytorch-rocm/torch/csrc/jit/ir/ir.cpp	2023-07-08 14:23:57.239106350 +0200
@@ -1186,10 +1186,10 @@
     case prim::rpc_remote: // It represents RPC message sent.
     case aten::wait: // It can represent RPC message received.
 #if !defined(USE_ROCM)
-    case cuda::set_stream:
-    case cuda::_set_device:
-    case cuda::_current_device:
-    case cuda::synchronize:
+    case c10::cuda::set_stream:
+    case c10::cuda::_set_device:
+    case c10::cuda::_current_device:
+    case c10::cuda::synchronize:
 #endif
     case prim::Enter:
     case prim::Exit:
