diff -aur qflow-1.0.99.pristine/configure.in qflow-1.0.99.new/configure.in
--- qflow-1.0.99.pristine/configure.in	2015-05-18 15:41:59.000000000 +0200
+++ qflow-1.0.99.new/configure.in	2015-11-20 21:43:06.667490218 +0100
@@ -178,14 +178,14 @@
 [  --with-libdir=DIR			path to qflow runtime files], [
    QFLOW_LIB_DIR=$withval
 ], [
-   QFLOW_LIB_DIR=${ac_default_prefix}/share/qflow
+   QFLOW_LIB_DIR=${prefix}/share/qflow
 ])
 
 AC_ARG_WITH(bindir,
 [  --with-bindir=DIR			path to qflow launch script], [
    QFLOW_BIN_DIR=$withval
 ], [
-   QFLOW_BIN_DIR=${ac_default_prefix}/bin
+   QFLOW_BIN_DIR=${prefix}/bin
 ])
 
 dnl AC_SUBST(QFLOW_QROUTER_PATH)
diff -aur qflow-1.0.99.pristine/scripts/makesim.sh qflow-1.0.99.new/scripts/makesim.sh
--- qflow-1.0.99.pristine/scripts/makesim.sh	2015-05-18 15:41:59.000000000 +0200
+++ qflow-1.0.99.new/scripts/makesim.sh	2015-11-20 21:44:54.860238260 +0100
@@ -22,7 +22,7 @@
 rm -f ${cellname}.ext
 rm -f ${cellname}.sim
 
-/usr/local/bin/magic -dnull -noconsole <<EOF
+/usr/bin/magic -dnull -noconsole <<EOF
 drc off
 box 0 0 0 0
 load ${cellname}
diff -aur qflow-1.0.99.pristine/src/Makefile.in qflow-1.0.99.new/src/Makefile.in
--- qflow-1.0.99.pristine/src/Makefile.in	2015-05-18 15:41:59.000000000 +0200
+++ qflow-1.0.99.new/src/Makefile.in	2015-11-20 22:11:28.854948550 +0100
@@ -9,7 +9,7 @@
 LDFLAGS = @LDFLAGS@
 INSTALL = @INSTALL@
 
-QFLOW_LIB_DIR = @QFLOW_LIB_DIR@
+QFLOW_BIN_DIR = @QFLOW_BIN_DIR@
 QFLOW_GRAYWOLF_PATH = @QFLOW_GRAYWOLF_PATH@
 QFLOW_QROUTER_PATH = @QFLOW_QROUTER_PATH@
 QFLOW_ABC_PATH = @QFLOW_ABC_PATH@
@@ -26,7 +26,7 @@
 SOURCES := $(patsubst %.o,%.c,$(OBJECTS))
 TARGETS := $(patsubst %.o,%$(EXEEXT),$(OBJECTS))
 
-BININSTALL = ${QFLOW_LIB_DIR}/bin
+BININSTALL = ${QFLOW_BIN_DIR}
 
 all: $(TARGETS)
 
@@ -57,19 +57,6 @@
 	@for target in $(TARGETS); do \
 	   $(INSTALL) $$target $(DESTDIR)${BININSTALL} ;\
 	done
-	@echo "Installing links to graywolf, qrouter, odin_ii, and abc"
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f graywolf;  ln -s $(QFLOW_GRAYWOLF_PATH) graywolf)
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f qrouter;  ln -s $(QFLOW_QROUTER_PATH) qrouter)
-	@if test "${HAVE_ABC}" = "1"; then \
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f abc;  ln -s $(QFLOW_ABC_PATH) abc); \
-	fi
-	@if test "${HAVE_ODIN_II}" = "1"; then \
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f odin_ii;  ln -s $(QFLOW_ODIN_PATH) odin_ii); \
-	fi
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f magic;  ln -s $(QFLOW_MAGIC_PATH) magic)
-	@if test "${HAVE_YOSYS}" = "1"; then \
-	(cd $(DESTDIR)${BININSTALL};  $(RM) -f yosys;  ln -s $(QFLOW_YOSYS_PATH) yosys); \
-	fi
 
 uninstall:
 	$(RM) -rf ${BININSTALL}
