From 4402207b1d498e3069d7fb0e369a20cfab157caa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Benjamin=20Chr=C3=A9tien?= <chretien@lirmm.fr>
Date: Mon, 30 Nov 2015 15:02:56 +0100
Subject: Fix for qglviewer


diff --git a/CMakeModules/FindQGLViewer.cmake b/CMakeModules/FindQGLViewer.cmake
index 2e9b64c..a3f5f1c 100644
--- a/CMakeModules/FindQGLViewer.cmake
+++ b/CMakeModules/FindQGLViewer.cmake
@@ -11,14 +11,15 @@
 
 SET( QGLViewer_FOUND 0 CACHE BOOL "Do we have QGLViewer?" )
 
-FIND_PATH( QGLVIEWER_BASE_DIR qglviewer.h
+FIND_PATH( QGLVIEWER_BASE_DIR libQGLViewer.so
+  /usr/lib
   ${CMAKE_SOURCE_DIR}/src/extern/QGLViewer
   ${CMAKE_SOURCE_DIR}/octovis/src/extern/QGLViewer
 )
 
 FIND_PATH( QGLViewer_INCLUDE_DIR qglviewer.h
-    /usr/include/qglviewer-qt4
     /usr/include/QGLViewer
+    /usr/include/qglviewer-qt4
     /opt/local/include/QGLViewer
     ${QGLVIEWER_BASE_DIR}
 )
