--- mkspecs/features/spec_pre.prf.orig	2016-06-10 08:48:56.000000000 +0200
+++ mkspecs/features/spec_pre.prf	2016-08-09 16:42:35.000000000 +0200
@@ -14,9 +14,9 @@
 QMAKE_EXT_H_MOC         = .cpp
 QMAKE_EXT_JS            = .js
 QMAKE_EXT_LEX           = .l
-QMAKE_EXT_LIBTOOL       = .la
-QMAKE_EXT_PKGCONFIG     = .pc
-QMAKE_EXT_PRL           = .prl
+QMAKE_EXT_LIBTOOL       = -static.la
+QMAKE_EXT_PKGCONFIG     = -static.pc
+QMAKE_EXT_PRL           = -static.prl
 QMAKE_EXT_UI            = .ui
 QMAKE_EXT_YACC          = .y
 
