--- Tools/qmake/mkspecs/features/features.prf.orig	2013-01-04 22:56:13.465133221 +0100
+++ Tools/qmake/mkspecs/features/features.prf	2013-01-04 22:56:39.458319734 +0100
@@ -81,7 +81,7 @@
     # Disable on Windows CE for now, as ANGLE won't compile.
     !wince*:contains(QT_CONFIG, opengl) {
         # On Windows we require GLES2/EGL through ANGLE.
-        !win32|contains(QT_CONFIG, opengles2): WEBKIT_CONFIG += use_3d_graphics
+        #!win32|contains(QT_CONFIG, opengles2): WEBKIT_CONFIG += use_3d_graphics
     }
 
     # Temporarily disable FTPDIR on Windows CE (missing functions from time.h)
--- Tools/Scripts/webkitperl/FeatureList.pm.orig	2013-01-04 23:07:11.449912367 +0100
+++ Tools/Scripts/webkitperl/FeatureList.pm	2013-01-04 23:07:21.360989539 +0100
@@ -149,7 +149,7 @@
 
 my @features = (
     { option => "3d-rendering", desc => "Toggle 3D Rendering support",
-      define => "ENABLE_3D_RENDERING", default => (isAppleMacWebKit() || isQt()), value => \$threeDRenderingSupport },
+      define => "ENABLE_3D_RENDERING", default => (isAppleMacWebKit()), value => \$threeDRenderingSupport },
 
     { option => "accelerated-2d-canvas", desc => "Toggle Accelerated 2D Canvas support",
       define => "ENABLE_ACCELERATED_2D_CANVAS", default => 0, value => \$accelerated2DCanvasSupport },
