diff --git a/src/meshlabplugins/edit_paint/paintbox.cpp b/src/meshlabplugins/edit_paint/paintbox.cpp
index 2097a5ba..09cce489 100644
--- a/src/meshlabplugins/edit_paint/paintbox.cpp
+++ b/src/meshlabplugins/edit_paint/paintbox.cpp
@@ -32,11 +32,11 @@ Paintbox::Paintbox(QWidget * parent, Qt::WindowFlags flags) : QWidget(parent, fl
 	
 	QIcon undo = undo_button->icon();
 	undo_button->setDefaultAction(stacks->createUndoAction(undo_button));
-	undo_button->defaultAction()->setIcon(undo);
+	undo_button->setIcon(undo);
 
 	QIcon redo = redo_button->icon();
 	redo_button->setDefaultAction(stacks->createRedoAction(redo_button));
-	redo_button->defaultAction()->setIcon(redo);
+	redo_button->setIcon(redo);
 
 	setUndoStack(parent);	
 	
diff --git a/src/meshlabplugins/render_gdp/shaderDialog.h b/src/meshlabplugins/render_gdp/shaderDialog.h
index a62d3b51..9ff9e5a5 100644
--- a/src/meshlabplugins/render_gdp/shaderDialog.h
+++ b/src/meshlabplugins/render_gdp/shaderDialog.h
@@ -29,6 +29,8 @@
 #include <common/meshmodel.h>
 #include <QLabel>
 #include <QDockWidget>
+#include <QSlider>
+#include <QLineEdit>
 #include "shaderStructs.h"
 #include "ui_shaderDialog.h"
 #include <QMap>
