diff --git a/src/meshlabserver/mainserver.cpp b/src/meshlabserver/mainserver.cpp
index ab9a7c08..5bbd859a 100644
--- a/src/meshlabserver/mainserver.cpp
+++ b/src/meshlabserver/mainserver.cpp
@@ -897,7 +897,7 @@ int main(int argc, char *argv[])
     }
 
 	if (meshDocument.size() < outmeshlist.size())
-		fprintf(logfp, "Error: trying to save %i meshes, but only %i available in the project\n", qPrintable(outmeshlist.size()), qPrintable(meshDocument.size()));
+		fprintf(logfp, "Error: trying to save %i meshes, but only %i available in the project\n", outmeshlist.size(), meshDocument.size());
 	else
 	{
 		for (int ii = 0; ii < outmeshlist.size(); ++ii)
@@ -911,7 +911,7 @@ int main(int argc, char *argv[])
 					fprintf(logfp, "Output mesh %s has NOT been saved\n", qPrintable(outmeshlist[ii].filename));
 			}
 			else
-				fprintf(logfp, "Invalid layer %i. Output mesh %s will not be saved\n", qPrintable(ii), qPrintable(outmeshlist[ii].filename));
+				fprintf(logfp, "Invalid layer %i. Output mesh %s will not be saved\n", ii, qPrintable(outmeshlist[ii].filename));
 		}
 
 	}
diff --git a/src/meshlabplugins/edit_paint/paintbox.cpp b/src/meshlabplugins/edit_paint/paintbox.cpp
index 2097a5ba..09cce489 100644
--- a/src/meshlabplugins/edit_paint/paintbox.cpp
+++ b/src/meshlabplugins/edit_paint/paintbox.cpp
@@ -32,11 +32,11 @@ Paintbox::Paintbox(QWidget * parent, Qt::WindowFlags flags) : QWidget(parent, fl
 	
 	QIcon undo = undo_button->icon();
 	undo_button->setDefaultAction(stacks->createUndoAction(undo_button));
-	undo_button->defaultAction()->setIcon(undo);
+	undo_button->setIcon(undo);
 
 	QIcon redo = redo_button->icon();
 	redo_button->setDefaultAction(stacks->createRedoAction(redo_button));
-	redo_button->defaultAction()->setIcon(redo);
+	redo_button->setIcon(redo);
 
 	setUndoStack(parent);	
 	
diff --git a/src/meshlabplugins/render_gdp/shaderDialog.h b/src/meshlabplugins/render_gdp/shaderDialog.h
index a62d3b51..9ff9e5a5 100644
--- a/src/meshlabplugins/render_gdp/shaderDialog.h
+++ b/src/meshlabplugins/render_gdp/shaderDialog.h
@@ -29,6 +29,8 @@
 #include <common/meshmodel.h>
 #include <QLabel>
 #include <QDockWidget>
+#include <QSlider>
+#include <QLineEdit>
 #include "shaderStructs.h"
 #include "ui_shaderDialog.h"
 #include <QMap>
