From 99295fc0b9858b08d379a86625902d92f5129542 Mon Sep 17 00:00:00 2001
From: bartus <szczepaniak.bartek+github@gmail.com>
Date: Sun, 7 Jun 2020 23:22:49 +0200
Subject: [PATCH] Fix build against qt>=5.15

---
 src/MEGASync/gui/MegaSpeedGraph.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/MEGASync/gui/MegaSpeedGraph.h b/src/MEGASync/gui/MegaSpeedGraph.h
index d98f582a..b7ffbcbb 100644
--- a/src/MEGASync/gui/MegaSpeedGraph.h
+++ b/src/MEGASync/gui/MegaSpeedGraph.h
@@ -2,6 +2,7 @@
 #define MEGASPEEDGRAPH_H
 
 #include <QWidget>
+#include <QPainterPath>
 #include "megaapi.h"
 
 namespace Ui {
-- 
2.27.0

