--- a/buildsystem/check_qml_module.py
+++ b/buildsystem/check_qml_module.py
@@ -82,11 +82,18 @@
             print("qmake not found")
             return False
 
-        ret = subprocess.run(
-            [ qmake, "-qtconf", qtconf, "-query"],
-            capture_output=True,
-            encoding="utf8"
-        )
+        if os.path.isfile(qtconf):
+            ret = subprocess.run(
+                [ qmake, "-qtconf", qtconf, "-query"],
+                capture_output=True,
+                encoding="utf8"
+            )
+        else:
+            ret = subprocess.run(
+                [ qmake, "-query"],
+                capture_output=True,
+                encoding="utf8"
+            )
 
         if ret.returncode != 0:
             return None
--- a/configure
+++ b/configure
@@ -61836,7 +61836,7 @@
 
             { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if contribs provide qmake" >&5
 printf %s "checking if contribs provide qmake... " >&6; }
-      if test -x "${CONTRIB_DIR}/../bin/qmake${BUILDEXEEXT}"
+      if [ ! -z ${CONTRIB_DIR} ] && [ test -x "${CONTRIB_DIR}/../bin/qmake${BUILDEXEEXT}" ]
 then :
 
         QMAKE="${CONTRIB_DIR}/../bin/qmake${BUILDEXEEXT}"
