--- a/src/3rdparty/chromium/third_party/WebKit/Source/build/scripts/rule_bison.py
+++ b/src/3rdparty/chromium/third_party/WebKit/Source/build/scripts/rule_bison.py
@@ -100,7 +100,7 @@
 outputHFile = open(outputHTmp)
 outputHContents = outputHFile.read()
 outputHFile.close()
-os.unlink(outputHTmp)
+# not working for bison-3.7: os.unlink(outputHTmp)
 
 # Rewrite the generated header with #include guards.
 outputH = os.path.join(outputDir, inputRoot + '.h')
