diff --git a/share/qtcreator/themes/dark.creatortheme b/share/qtcreator/themes/dark.creatortheme
index 5da67fa236..6b1f2fba96 100644
--- a/share/qtcreator/themes/dark.creatortheme
+++ b/share/qtcreator/themes/dark.creatortheme
@@ -9,7 +9,7 @@ text=ffe7e7e7
 textDisabled=7fffffff
 textHighlighted=ffe7e7e7
 hoverBackground=18ffffff
-selectedBackground=66000000
+selectedBackground=aa1f75cc
 selectedBackgroundText=aa1f75cc
 normalBackground=ff333333
 alternateBackground=ff515151
@@ -205,7 +205,7 @@ PaletteBrightText=ffff0000
 PaletteText=text
 PaletteButtonText=text
 PaletteButtonTextDisabled=textDisabled
-PaletteToolTipBase=selectedBackground
+PaletteToolTipBase=66000000
 PaletteHighlight=selectedBackgroundText
 PaletteDark=shadowBackground
 PaletteHighlightedText=textHighlighted
diff --git a/share/qtcreator/themes/flat-dark.creatortheme b/share/qtcreator/themes/flat-dark.creatortheme
index e73d10b0ff..a4f263c1f7 100644
--- a/share/qtcreator/themes/flat-dark.creatortheme
+++ b/share/qtcreator/themes/flat-dark.creatortheme
@@ -13,7 +13,7 @@ toolBarItemDisabled=60a4a6a8
 fancyBarsNormalTextColor=ffd0d0d0
 fancyBarsBoldTextColor=b6fbfdff
 hoverBackground=22ffffff
-selectedBackground=66000000
+selectedBackground=aa1d545c
 selectedBackgroundText=ff1d545c
 normalBackground=ff2E2F30
 alternateBackground=ff353637
@@ -223,7 +223,7 @@ PaletteBrightText=ffff3333
 PaletteText=text
 PaletteButtonText=text
 PaletteButtonTextDisabled=textDisabled
-PaletteToolTipBase=selectedBackground
+PaletteToolTipBase=66000000
 PaletteHighlight=selectedBackgroundText
 PaletteDark=shadowBackground
 PaletteHighlightedText=textHighlighted
diff --git a/share/qtcreator/themes/flat-light.creatortheme b/share/qtcreator/themes/flat-light.creatortheme
index 29edca63fd..826c798faa 100644
--- a/share/qtcreator/themes/flat-light.creatortheme
+++ b/share/qtcreator/themes/flat-light.creatortheme
@@ -11,7 +11,7 @@ toolBarItemDisabled=38000000
 fancyBarsNormalTextColor=ff000000
 fancyBarsBoldTextColor=a0010508
 hoverBackground=1a000000
-selectedBackground=a8ffffff
+selectedBackground=553daee9
 normalBackground=ffffffff
 alternateBackground=ff515151
 stop_error=ffec7373
diff --git a/share/qtcreator/themes/flat.creatortheme b/share/qtcreator/themes/flat.creatortheme
index ab0189528f..668d4bf612 100644
--- a/share/qtcreator/themes/flat.creatortheme
+++ b/share/qtcreator/themes/flat.creatortheme
@@ -11,7 +11,7 @@ toolBarItemDisabled=60a4a6a8
 fancyBarsNormalTextColor=ffffffff
 fancyBarsBoldTextColor=b6fbfdff
 hoverBackground=22ffffff
-selectedBackground=66000000
+selectedBackground=aa3daee9
 normalBackground=ffffffff
 alternateBackground=ff515151
 error=ffdf4f4f
