diff --git a/include/ClientGUICanvas.py b/include/ClientGUICanvas.py
index 7b176bb..15894d7 100755
--- a/include/ClientGUICanvas.py
+++ b/include/ClientGUICanvas.py
@@ -5300,7 +5300,7 @@ class MediaContainer( QW.QWidget ):
         self._static_image_window = StaticImage( self )
         
         self._volume_control.adjustSize()
-        self._volume_control.setCursor( QG.Qt.ArrowCursor )
+        self._volume_control.setCursor( QC.Qt.ArrowCursor )
         
         self._animation_window.hide()
         self._animation_bar.hide()
