diff --git a/configure.rb b/configure.rb
index 23f3dcc9..951db1fa 100755
--- a/configure.rb
+++ b/configure.rb
@@ -108,7 +108,7 @@ _EOH_
        quazipDir = conf.argumentValue("with-quazip")
        if File.directory? quazipDir
           quazipLib = quazipDir + "/lib"
-          quazipInclude = quazipDir + "/include"
+          quazipInclude = quazipDir + "/include/quazip5"
           config.addLib("-L" + quazipLib)
           config.addIncludePath(quazipInclude)
        else
diff --git a/configure.tests/quazip/main.cpp b/configure.tests/quazip/main.cpp
index f9ec3753..cf1820da 100644
--- a/configure.tests/quazip/main.cpp
+++ b/configure.tests/quazip/main.cpp
@@ -33,8 +33,8 @@
  *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  ***************************************************************************/
 
-#include "quazip/quazip.h"
-#include "quazip/quazipfile.h"
+#include "quazip5/quazip.h"
+#include "quazip5/quazipfile.h"
 #include <QFile>
 
 int main()
diff --git a/configure.tests/quazip/quazip.pro b/configure.tests/quazip/quazip.pro
index 4db2eeb8..b9e21faa 100755
--- a/configure.tests/quazip/quazip.pro
+++ b/configure.tests/quazip/quazip.pro
@@ -10,7 +10,7 @@ macx {
 }
 
 unix:!macx {
-  INCLUDEPATH += /usr/include/quazip
+  INCLUDEPATH += /usr/include/quazip5
 }
 
 LIBS += -lz
diff --git a/configure.tests/quazip/quazip.qonf b/configure.tests/quazip/quazip.qonf
index 7d259a82..da8aa2ca 100755
--- a/configure.tests/quazip/quazip.qonf
+++ b/configure.tests/quazip/quazip.qonf
@@ -8,7 +8,7 @@
 			<define value="HAVE_QUAZIP" />
 		</defines>
                 <includes>
-                        <include path="/usr/include/quazip/" />
+                        <include path="/usr/include/quazip5/" />
                 </includes>
 	</provide>
 	<solution>
diff --git a/qonf/test.rb b/qonf/test.rb
index ba63f319..d746377c 100755
--- a/qonf/test.rb
+++ b/qonf/test.rb
@@ -98,12 +98,12 @@ class Test
                       if conf.hasArgument?("with-quazip")
                          quazipDir = conf.argumentValue("with-quazip")
                          quazipLib = quazipDir + "/lib"
-                         extraLib += "-L#{quazipLib} -lquazip"
-                         extraInclude = quazipDir + "/include"
+                         extraLib += "-L#{quazipLib} -lquazip5"
+                         extraInclude = quazipDir + "/include/quazip5"
                          qmakeLine = "'LIBS += #{extraLib}'"
                          qmakeLine += " 'INCLUDEPATH += #{extraInclude}'"
                       else
-                         extraLib = "-lquazip-qt5"
+                         extraLib = "-lquazip5"
                          qmakeLine = "'LIBS += #{extraLib}'"
                       end
                    else
@@ -174,9 +174,9 @@ class Test
         }
 
         if conf.hasArgument?("with-quazip")
-           config.addLib("-lquazip")
+           config.addLib("-lquazip5")
         else
-           config.addLib("-lquazip-qt5")
+           config.addLib("-lquazip5")
         end
 
         # config.addLib("-lquazip-qt5")
diff --git a/src/libtupi/tuppackagehandler.cpp b/src/libtupi/tuppackagehandler.cpp
37,39c37,39
< #include "quazip/quazip.h"
< #include "quazip/quazipfile.h"
< #include "quazip/JlCompress.h"
---
> #include "quazip5/quazip.h"
> #include "quazip5/quazipfile.h"
> #include "quazip5/JlCompress.h"
