--- mesa-20.3.0-rc2/src/amd/vulkan/radv_device.c	2020-11-19 13:19:06.000000000 -0500
+++ radv_device.c	2020-11-20 21:56:24.981971238 -0500
@@ -571,6 +571,7 @@
 	{"pswave32", RADV_PERFTEST_PS_WAVE_32},
 	{"gewave32", RADV_PERFTEST_GE_WAVE_32},
 	{"dfsm", RADV_PERFTEST_DFSM},
+	{"aco", RADV_PERFTEST_ACO},
 	{NULL, 0}
 };
 
@@ -701,6 +702,8 @@
 	instance->debug_flags = parse_debug_string(getenv("RADV_DEBUG"),
 						   radv_debug_options);
 
+  instance->debug_flags |= RADV_DEBUG_LLVM;
+
 	const char *radv_perftest_str = getenv("RADV_PERFTEST");
 	instance->perftest_flags = parse_debug_string(radv_perftest_str,
 						      radv_perftest_options);
