diff --git a/Common/Core/CMakeLists.txt b/Common/Core/CMakeLists.txt
index a4ff12aa54b..105812d3e62 100644
--- a/Common/Core/CMakeLists.txt
+++ b/Common/Core/CMakeLists.txt
@@ -493,6 +493,10 @@ if(MSVC)
       COMPILE_FLAGS /bigobj)
 endif()
 
+if (MINGW)
+  set_source_files_properties(vtkDataArray.cxx PROPERTIES COMPILE_FLAGS "-O1")
+endif ()
+
 vtk_module_add_module(VTK::CommonCore
   HEADER_DIRECTORIES
   CLASSES           ${classes}
