diff -ura --color orig/src/Client/GUI3/Interface.cpp new/src/Client/GUI3/Interface.cpp
--- orig/src/Client/GUI3/Interface.cpp	2018-02-12 13:51:37.779567572 -0500
+++ new/src/Client/GUI3/Interface.cpp	2018-02-12 13:52:51.364407378 -0500
@@ -316,7 +316,8 @@
 
 	for (unsigned int i = 0; i < modes.size(); ++i)
 	{
-		const int curModeQuality = std::abs(modes[i].width - compare.x) + std::abs(modes[i].height - compare.y)
+		const int curModeQuality = std::abs((int) modes[i].width - (int) compare.x)
+			+ std::abs((int) modes[i].height - (int) compare.y)
 			- modes[i].bitsPerPixel;
 
 		if (curModeQuality < bestModeQuality)
diff -ura --color orig/src/Client/RankCheck/PersistentLogReader.cpp new/src/Client/RankCheck/PersistentLogReader.cpp
--- orig/src/Client/RankCheck/PersistentLogReader.cpp	2018-02-12 13:51:37.782900974 -0500
+++ new/src/Client/RankCheck/PersistentLogReader.cpp	2018-02-12 13:53:33.005256833 -0500
@@ -3,6 +3,7 @@
 #include <Shared/Config/Config.hpp>
 #include <Shared/Utils/DebugLog.hpp>
 #include <Shared/Utils/StrNumCon.hpp>
+#include <cmath>
 #include <utility>
 
 static cfg::Int cfgMatchStartPos("rankcheck.persistentLogReader.currentOffset");
