--- alttoolbar_rb3compat.py.orig	2016-08-22 09:16:26.609766000 +0200
+++ alttoolbar_rb3compat.py	2016-08-22 09:18:17.856062014 +0200
@@ -540,9 +540,9 @@
                 self.shell.props.window.add_action(action)
                 self.actiongroup.add_action(action)
 
-            if accel:
+            '''if accel:
                 app.add_accelerator(accel, action_type + "." + action_name,
-                                    None)
+                                    None)'''
         else:
             if 'stock_id' in args:
                 stock_id = args['stock_id']
