diff --git a/src/RadioTray.py b/src/RadioTray.py
index cc6eec2..77f738a 100644
--- a/src/RadioTray.py
+++ b/src/RadioTray.py
@@ -167,7 +167,7 @@ class RadioTray(object):
     def configLogging(self):
         # config logging
         self.logger = logging.getLogger('radiotray')
-        self.logger.setLevel(logging.DEBUG)
+        self.logger.setLevel(logging.WARN)
         handler = logging.handlers.RotatingFileHandler(LOGFILE, maxBytes=2000000, backupCount=1)
         formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
         handler.setFormatter(formatter)
