From f542ad5eb07d9f129d7160d9e46ab07bda2afa7a Mon Sep 17 00:00:00 2001
From: Jose <jose@zeroc.com>
Date: Sat, 4 Dec 2021 09:54:36 +0100
Subject: [PATCH] Ignore redundant declarations with PHP > 8

---
 php/src/php/Config.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/php/src/php/Config.h b/php/src/php/Config.h
index c5d9db61d57..3f308d45ce7 100644
--- a/php/src/php/Config.h
+++ b/php/src/php/Config.h
@@ -50,6 +50,7 @@ extern "C"
 
 #if defined(__GNUC__) && ((__GNUC__ >= 8))
 #   pragma GCC diagnostic ignored "-Wignored-qualifiers"
+#   pragma GCC diagnostic ignored "-Wredundant-decls"
 #endif
 
 //
