diff -aur package.pristine/crashreporter/crashreporter_posix.c package.new/crashreporter/crashreporter_posix.c
--- package.pristine/crashreporter/crashreporter_posix.c	2016-06-05 23:57:15.720344670 +0200
+++ package.new/crashreporter/crashreporter_posix.c	2016-06-05 23:57:33.723548061 +0200
@@ -28,8 +28,8 @@
 #include <stdio.h>
 
 
-//#define PACKAGE 1 // workaround for bug in libbfd
-//#define PACKAGE_VERSION 1 // workaround for bug in libbfd
+#define PACKAGE 1 // workaround for bug in libbfd
+#define PACKAGE_VERSION 1 // workaround for bug in libbfd
 #include "backtrace-symbols.c"
 
 #elif defined(FOR_MACOSX)
