diff -Naur rem-0.6.0.orig/Makefile rem-0.6.0/Makefile
--- rem-0.6.0.orig/Makefile	2019-05-06 19:12:54.745092607 -0400
+++ rem-0.6.0/Makefile	2019-05-06 19:13:30.841902449 -0400
@@ -83,7 +83,7 @@
 
 $(SHARED): $(OBJS)
 	@echo "  LD      $@"
-	@$(LD) $(LFLAGS) $(SH_LFLAGS) $^ -L$(LIBRE_SO) -lre $(LIBS) -o $@
+	@$(LD) $(LDFLAGS) $(LFLAGS) $(SH_LFLAGS) $^ -L$(LIBRE_SO) -lre $(LIBS) -o $@
 
 
 $(STATIC): $(OBJS)
