--- old/usr/share/remarkable/ui/RemarkableWindow.ui	2015-06-21 17:04:59.000000000 +0200
+++ new/usr/share/remarkable/ui/RemarkableWindow.ui	2015-10-18 13:18:06.956900467 +0200
@@ -832,7 +832,6 @@
                     </child>
                     <child>
                       <object class="GtkMenuItem" id="mnu_about">
-                        <property name="related_action"/>
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
                         <property name="label" translatable="yes">About</property>
