--- a/src/egl/drivers/dri2/platform_drm.c	2017-05-10 16:13:57.000000000 +0200
+++ b/src/egl/drivers/dri2/platform_drm.c	2017-05-28 20:52:57.507271447 +0200
@@ -230,21 +230,10 @@
 
    if (dri2_surf->back == NULL)
       return -1;
-   if (dri2_surf->back->bo == NULL) {
-      if (surf->base.modifiers)
-         dri2_surf->back->bo = gbm_bo_create_with_modifiers(&dri2_dpy->gbm_dri->base.base,
-                                                            surf->base.width, surf->base.height,
-                                                            surf->base.format,
-                                                            surf->base.modifiers,
-                                                            surf->base.count);
-      else
-         dri2_surf->back->bo = gbm_bo_create(&dri2_dpy->gbm_dri->base.base,
-                                             surf->base.width,
-                                             surf->base.height,
-                                             surf->base.format,
-                                             surf->base.flags);
-
-   }
+   if (dri2_surf->back->bo == NULL)
+      dri2_surf->back->bo = gbm_bo_create(&dri2_dpy->gbm_dri->base.base,
+					  surf->base.width, surf->base.height,
+					  surf->base.format, surf->base.flags);
    if (dri2_surf->back->bo == NULL)
       return -1;
 
