--- a/services/menuManager.js
+++ b/services/menuManager.js
@@ -13,11 +13,6 @@
         label: appName,
         submenu: [
           { role: 'about' },
-          {
-            label: 'Check for Updates...',
-            click: function (menuItem, browserWindow, options) { menuManager.handleMenuAction('checkElectronUpdates', null, options); }
-          },
-          { type: 'separator' },
           {
             label: 'Preferences',
             accelerator: 'CmdOrCtrl+,',
@@ -364,11 +359,6 @@
         label: 'Help',
         role: 'help',
         submenu: [
-          {
-            label: 'Check for Updates',
-            click: function (menuItem, browserWindow, options) { menuManager.handleMenuAction('checkElectronUpdates', null, options); }
-          },
-          { type: 'separator' },
           {
             label: 'Documentation',
             click: function (menuItem, browserWindow, options) { menuManager.handleMenuAction('openCustomUrl', 'https://www.getpostman.com/docs/', options); }
--- a/js/requester.js
+++ b/js/requester.js
@@ -424105,11 +424105,6 @@
                 { key: _SettingsTypeConstants.SETTINGS_PROXY },
                 _react2.default.createElement(_SettingsProxyContainer2.default, null)
               ),
-              _react2.default.createElement(
-                _SettingsTabContent2.default,
-                { key: _SettingsTypeConstants.SETTINGS_UPDATE },
-                _react2.default.createElement(_SettingsUpdateContainer2.default, null)
-              ),
               _react2.default.createElement(
                 _SettingsTabContent2.default,
                 { key: _SettingsTypeConstants.SETTINGS_ABOUT },
@@ -424231,11 +424226,6 @@
             { refKey: _SettingsTypeConstants.SETTINGS_PROXY },
             'Proxy'
           ),
-          _react2.default.createElement(
-            _Tabs.Tab,
-            { refKey: _SettingsTypeConstants.SETTINGS_UPDATE },
-            'Update'
-          ),
           _react2.default.createElement(
             _Tabs.Tab,
             { refKey: _SettingsTypeConstants.SETTINGS_ABOUT },
