diff --git a/dune-project b/dune-project
index 4805fbb..f5be52a 100644
--- a/dune-project
+++ b/dune-project
@@ -11,7 +11,6 @@
  (synopsis "Collection of parallelism-safe data structures for Multicore OCaml")
  (depends
   (ocaml (>= 4.12))
-  (domain_shims (>= 0.1.0))
   (saturn_lockfree (= :version))
   (qcheck (and (>= 0.18.1) :with-test))
   (qcheck-stm (and (>= 0.2) :with-test))
@@ -23,7 +22,6 @@
  (synopsis "Collection of lock-free data structures for Multicore OCaml")
  (depends
   (ocaml (>= 4.12))
-  (domain_shims (>= 0.1.0))
   (qcheck (and (>= 0.18.1) :with-test))
   (qcheck-stm (and (>= 0.2) :with-test))
   (qcheck-alcotest (and (>= 0.18.1) :with-test))
diff --git a/saturn.opam b/saturn.opam
index 20dbb25..66aae1a 100644
--- a/saturn.opam
+++ b/saturn.opam
@@ -11,7 +11,6 @@ bug-reports: "https://github.com/ocaml-multicore/saturn/issues"
 depends: [
   "dune" {>= "3.2"}
   "ocaml" {>= "4.12"}
-  "domain_shims" {>= "0.1.0"}
   "saturn_lockfree" {= version}
   "qcheck" {>= "0.18.1" & with-test}
   "qcheck-stm" {>= "0.2" & with-test}
diff --git a/saturn_lockfree.opam b/saturn_lockfree.opam
index b5c2d30..44009c4 100644
--- a/saturn_lockfree.opam
+++ b/saturn_lockfree.opam
@@ -10,7 +10,6 @@ bug-reports: "https://github.com/ocaml-multicore/saturn/issues"
 depends: [
   "dune" {>= "3.2"}
   "ocaml" {>= "4.12"}
-  "domain_shims" {>= "0.1.0"}
   "qcheck" {>= "0.18.1" & with-test}
   "qcheck-stm" {>= "0.2" & with-test}
   "qcheck-alcotest" {>= "0.18.1" & with-test}
diff --git a/src_lockfree/dune b/src_lockfree/dune
index 7a76425..2a9802f 100644
--- a/src_lockfree/dune
+++ b/src_lockfree/dune
@@ -1,4 +1,3 @@
 (library
  (name saturn_lockfree)
- (public_name saturn_lockfree)
- (libraries domain_shims))
+ (public_name saturn_lockfree))
