diff --git configure.ac configure.ac
index 9f20a8bc..dfe989da 100644
--- configure.ac
+++ configure.ac
@@ -53,8 +53,8 @@ PKG_CHECK_MODULES(AUDACIOUS, [audacious >= 3.5.0],  have_audacious=yes,
 )
 AM_CONDITIONAL(HAVE_AUDACIOUS, test "$have_audacious" = yes)

-PKG_CHECK_MODULES(GTK, [glib-2.0 >= 2.6.0 gtk+-2.0 >= 2.6.0 gthread-2.0 pango],
-    , [AC_MSG_ERROR([Cannot find glib2/gtk2/pango])]
+PKG_CHECK_MODULES(GTK, [glib-2.0 >= 2.6.0 gthread-2.0 pango],
+    , [AC_MSG_ERROR([Cannot find glib2/pango])]
 )

 have_libao=no
