diff --git a/src/common/errors.ml b/src/common/errors.ml
index 6d42476..4f5ed6c 100644
--- a/src/common/errors.ml
+++ b/src/common/errors.ml
@@ -618,7 +618,6 @@ module ErrorSet = Set.Make(Error)
  *)
 module ErrorSuppressions = struct
   open Span
-  module Ast = Spider_monkey_ast
 
   type error_suppressions = Loc.t SpanMap.t
   type t = {
diff --git a/src/common/reason.ml b/src/common/reason.ml
index f8d1835..da6dd77 100644
--- a/src/common/reason.ml
+++ b/src/common/reason.ml
@@ -21,7 +21,6 @@
 
 open Utils_js
 open String_utils
-module Ast = Spider_monkey_ast
 
 let mk_id () = Ident.make ""
 
diff --git a/src/parser/parser_env.ml b/src/parser/parser_env.ml
index d3b2350..182ea31 100644
--- a/src/parser/parser_env.ml
+++ b/src/parser/parser_env.ml
@@ -14,7 +14,6 @@ module Lex_result = Lexer_flow.Lex_result
 open Ast
 module Error = Parse_error
 module SSet = Set.Make(String)
-module SMap = Map.Make(String)
 
 module Lex_mode = struct
   type t =
diff --git a/src/services/inference/module_js.ml b/src/services/inference/module_js.ml
index 112ff75..d84d95f 100644
--- a/src/services/inference/module_js.ml
+++ b/src/services/inference/module_js.ml
@@ -19,7 +19,6 @@
 open Utils_js
 
 module Ast = Spider_monkey_ast
-module Flow = Flow_js
 module ErrorSet = Errors.ErrorSet
 module FlowError = Flow_error
 
diff --git a/src/typing/func_params.ml b/src/typing/func_params.ml
index c67add7..bfb7142 100644
--- a/src/typing/func_params.ml
+++ b/src/typing/func_params.ml
@@ -1,7 +1,6 @@
 module Ast = Spider_monkey_ast
 module Anno = Type_annotation
 module Flow = Flow_js
-module Utils = Utils_js
 
 open Reason
 open Type
