diff --unified --recursive --text a/data/meson.build b/data/meson.build
--- a/data/meson.build	2022-04-07 14:22:31.000000000 -0600
+++ b/data/meson.build	2022-04-07 16:25:46.259954821 -0600
@@ -114,16 +114,3 @@
 else
   resources_path = resources.full_path()
 endif
-
-# Post install
-
-meson.add_install_script(
-    find_program('update-desktop-database').path(),
-    usrdir / 'applications'
-)
-
-meson.add_install_script(
-    find_program('gtk-update-icon-cache').path(),
-    '-qtf',
-    usrdir / 'icons' / 'hicolor'
-)
