diff -u --recursive a/QuickCheck-safe.cabal b/QuickCheck-safe.cabal
--- a/QuickCheck-safe.cabal	2019-04-29 14:48:08.000000000 +0000
+++ b/QuickCheck-safe.cabal	2022-03-18 17:22:51.265312272 +0000
@@ -33,7 +33,7 @@
 library
   hs-source-dirs:    src
   build-depends:
-    QuickCheck >= 2.7.3 && < 2.14,
+    QuickCheck >= 2.7.3,
     containers >= 0.5 && < 0.7,
     base >= 4.6 && < 5
   exposed-modules:
