diff -rupN a/src/supermodel/supermodel.amk b/src/supermodel/supermodel.amk
--- a/src/supermodel/supermodel.amk	2019-04-02 03:31:36.000000000 +0900
+++ b/src/supermodel/supermodel.amk	2021-10-07 13:52:02.243669379 +0900
@@ -36,5 +36,4 @@ src/supermodel/supermodel/sumo_params.h
 src/supermodel/supermodel/sumo_smdata.h \
 src/supermodel/supermodel/sumo_squarks.h \
 src/supermodel/supermodel/supermodel_defs.h \
-src/supermodel/supermodel/supermodel.h \
-src/supermodel/tsil/tsil.h 
+src/supermodel/supermodel/supermodel.h
