diff --color --unified --recursive --text app/index.js app2/index.js
--- app/index.js	2022-10-10 00:16:46.949457385 +0200
+++ app2/index.js	2022-10-10 00:22:26.917985986 +0200
@@ -9,7 +9,6 @@
 
 // expose globals to electron
 global.env = require('./src/utils/globals');
-const autoUpdate = require('./src/auto-update');
 const windowConfig = require('./src/window-config');
 const ipcEventListeners = require('./src/ipc-event-listeners');
 const keyboardShortcuts = require('./src/keyboard-shortcuts');
@@ -21,7 +20,6 @@
 
 // references
 let window = null;
-let autoUpdateCronJob = null;
 // expose primary window instance id globally
 global.env.windowInstanceID = null;
 global.env.windowStateEventDebounce = null;
@@ -190,14 +188,6 @@
 
         // ----- PRODUCTION ----- //
         }
-
-        // check for updates after launch
-        autoUpdate.init();
-
-        // hourly update check
-        autoUpdateCronJob = new CronJob('0 15 * * *', () => autoUpdate.checkForUpdate());
-        autoUpdateCronJob.start();
-
     });
 
     // dereference window object upon close