diff --unified --recursive --text environ-config-23.1.0/tests/test_packaging.py environ-config-23.1.0-patched/tests/test_packaging.py
--- environ-config-23.1.0/tests/test_packaging.py	2023-01-27 15:59:10.000000000 +0100
+++ environ-config-23.1.0-patched/tests/test_packaging.py	2023-01-31 16:31:04.317079774 +0100
@@ -44,20 +44,6 @@
                 == environ.__description__
             )
 
-    def test_uri(self):
-        """
-        environ.__uri__ returns the correct project URL.
-        """
-        with pytest.deprecated_call():
-            assert "https://environ-config.readthedocs.io/" == environ.__uri__
-
-    def test_email(self):
-        """
-        environ.__email__ returns Hynek's email address.
-        """
-        with pytest.deprecated_call():
-            assert "hs@ox.cx" == environ.__email__
-
     def test_does_not_exist(self):
         """
         Asking for unsupported dunders raises an AttributeError.
