Submodule Telegram/lib_ui contains modified content
diff --git a/Telegram/lib_ui/qt_conf/linux.qrc b/Telegram/lib_ui/qt_conf/linux.qrc
index b9f5692..8a9bbb3 100644
--- a/Telegram/lib_ui/qt_conf/linux.qrc
+++ b/Telegram/lib_ui/qt_conf/linux.qrc
@@ -1,7 +1,4 @@
 <RCC>
-  <qresource prefix="/qt/etc">
-    <file alias="qt.conf">qt_linux.conf</file>
-  </qresource>
   <qresource prefix="/fc">
     <file>fc-custom.conf</file>
   </qresource>
