Subject: Remove modeline for XF86Config-4 from README

--- a/README
+++ b/README
@@ -24,14 +24,6 @@ Thanks to ben for making patches to run
 Thanks to ivo for doing debian packages.
 
 f10 change between windowed/fullscreen mode
-(see the XF86Config-4, copy n paste the modeline 400x256
-then add the mode "400x256" to whatever bpp you have by default, restart x)
-example: add following line in Section "Monitor" of your 
-/etc/X11/XF86Config (or XF86Config-4)
-Modeline "400x256"     22.00  400 416 480 504  256 259 262 269 doublescan
-then in Section "Screen" if your DefaultDepth is 24 do be sure you have
-something like:
-Modes           "1024x768" "640x480" "512x384" "400x256" "320x240"
 esc/f12 exit
 
 For compilation you will need:
