diff -Naur cuttlefish.orig/data/ui/CuttlefishWindow.ui cuttlefish/data/ui/CuttlefishWindow.ui
--- cuttlefish.orig/data/ui/CuttlefishWindow.ui	2012-07-30 01:59:11.000000000 +0200
+++ cuttlefish/data/ui/CuttlefishWindow.ui	2015-06-13 20:54:31.750518028 +0200
@@ -606,7 +606,6 @@
                             <property name="can_focus">True</property>
                             <property name="headers_clickable">False</property>
                             <property name="search_column">0</property>
-                            <property name="ubuntu_almost_fixed_height_mode">True</property>
                             <signal name="cursor-changed" handler="on_tvActions_cursor_changed" swapped="no"/>
                             <child internal-child="selection">
                               <object class="GtkTreeSelection" id="treeview-selection1"/>
