diff -ura package.orig/src/html/scripts/ui.pug package.new/src/html/scripts/ui.pug
--- package.orig/src/html/scripts/ui.pug	2019-03-23 21:06:10.110844584 +0100
+++ package.new/src/html/scripts/ui.pug	2019-03-23 21:17:27.400580327 +0100
@@ -224,18 +224,3 @@
   if (process.platform === "win32") {
     modals.show('windows-drivers');
   }
-
-  utils.getUpdateAvailable().then((updateAvailable) => {
-    if(updateAvailable) {
-      utils.log.warn("This is not the latest stable release!")
-      if (utils.isSnap()) {
-        $("#snap-update-instructions").show();
-        $("#btn-update-installer").hide();
-      } else {
-        $("#generic-update-instructions").show();
-      }
-      modals.show('new-update');
-    } else {
-      utils.log.debug("Running latest stable version.");
-    }
-  });
