From 51c182c0de4ee3a8c7056975189d0ee4afbe487a Mon Sep 17 00:00:00 2001
From: egnappahz <egnappah@gmail.com>
Date: Fri, 9 Sep 2022 09:25:02 +0200
Subject: [PATCH] No.

Signed-off-by: egnappahz <egnappah@gmail.com>
---
 cmake/scripts/common/Platform.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/scripts/common/Platform.cmake b/cmake/scripts/common/Platform.cmake
index 397d8d46e8..87852ef9fc 100644
--- a/cmake/scripts/common/Platform.cmake
+++ b/cmake/scripts/common/Platform.cmake
@@ -8,7 +8,8 @@ if(CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd)
   set(_DEFAULT_PLATFORM X11 WAYLAND GBM)
 
   if(NOT APP_RENDER_SYSTEM)
-    message(SEND_ERROR "You need to decide whether you want to use GL- or GLES-based rendering. Please set APP_RENDER_SYSTEM to either \"gl\" or \"gles\". For normal desktop systems, you will usually want to use \"gl\".")
+    # message(SEND_ERROR "You need to decide whether you want to use GL- or GLES-based rendering. Please set APP_RENDER_SYSTEM to either \"gl\" or \"gles\". For normal desktop systems, you will usually want to use \"gl\".")
+    set(APP_RENDER_SYSTEM "gl")
   endif()
 else()
   string(TOLOWER ${CORE_SYSTEM_NAME} _DEFAULT_PLATFORM)
-- 
2.37.3

