diff -rupN a/tools/repl/swift/CMakeLists.txt b/tools/repl/swift/CMakeLists.txt
--- a/tools/repl/swift/CMakeLists.txt	2019-09-22 09:58:11.416432071 -0700
+++ b/tools/repl/swift/CMakeLists.txt	2019-09-22 09:58:36.799766163 -0700
@@ -19,6 +19,8 @@ add_lldb_tool(repl_swift
   main.c
   )
 
+target_link_libraries(repl_swift PRIVATE dl)
+
 # The dummy repl executable is a C program, but we always look for a mangled
 # swift symbol (corresponding to main). If we build the repl with debug info,
 # the debugger looks at the frame language (looking up the compile unit) and gets
