diff --git a/Makefile b/Makefile
index 268c112..87873dc 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ build:
 	$(GODOT) --path . --export "Linux/X11" target/debug/luxtorpeda.x86_64 --no-window
 
 release:
-	cargo build --release
+	cargo build --release --locked
 	mkdir -p target/debug
 	cp -r target/release/* target/debug
 	$(GODOT) --path . --export "Linux/X11" target/release/luxtorpeda.x86_64 --no-window
@@ -52,7 +52,7 @@ lint:
 	cargo clippy -- -D warnings
 
 test:
-	cargo test
+	cargo test --release --locked
 
 clean:
 	cargo clean
