diff --unified --recursive --text a/setup.py b/setup.py
--- a/setup.py	2020-03-12 11:51:30.759503274 +0200
+++ b/setup.py	2020-03-12 11:51:46.215411606 +0200
@@ -12,8 +12,8 @@
     url = 'https://github.com/blakev/python-syncthing',
     license = 'The MIT License',
     install_requires = [
-        'python-dateutil==2.6.1',
-        'requests==2.18.4'
+        'python-dateutil',
+        'requests'
     ],
     extras_require = {
         'dev': [
@@ -43,4 +43,4 @@
         'Programming Language :: Python :: 3.7',
         'Topic :: System :: Archiving :: Mirroring'
     ],
-)
\ No newline at end of file
+)
