diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index fb06b9556..0180e201e 100644
--- a/v8/build/config/compiler/BUILD.gn
+++ b/v8/build/config/compiler/BUILD.gn
@@ -499,11 +499,11 @@ config("compiler") {
       "-instcombine-lower-dbg-declare=0",
     ]
 
-    if (!is_chromeos && default_toolchain != "//build/toolchain/cros:target") {
-      # TODO(https://crbug.com/1049161): Remove '-DCLANG_SPAWN_CC1=ON' from build.py instead
-      # once this change has marinated a bit.
-      cflags += [ "-fintegrated-cc1" ]
-    }
+    # if (!is_chromeos && default_toolchain != "//build/toolchain/cros:target") {
+    #   # TODO(https://crbug.com/1049161): Remove '-DCLANG_SPAWN_CC1=ON' from build.py instead
+    #   # once this change has marinated a bit.
+    #   cflags += [ "-fintegrated-cc1" ]
+    # }
   }
 
   # C11/C++11 compiler flags setup.
