--- a/FreeFileSync/Source/ffs_paths.cpp    2019-11-19 23:52:20.000000000 +0100
+++ b/FreeFileSync/Source/ffs_paths.cpp    2019-11-24 18:14:17.627328553 +0100
@@ -80,6 +80,4 @@
 
-        //OS standard path (XDG layout): ~/.config/FreeFileSync
-        //wxBug: wxStandardPaths::GetUserDataDir() does not honor FileLayout_XDG flag
-        wxStandardPaths::Get().SetFileLayout(wxStandardPaths::FileLayout_XDG);
-        const Zstring cfgFolderPath = appendSeparator(utfTo<Zstring>(wxStandardPaths::Get().GetUserConfigDir())) + "FreeFileSync";
+        //OS standard path (classic layout): ~/.FreeFileSync
+        Zstring cfgFolderPath = utfTo<Zstring>(wxStandardPaths::Get().GetUserDataDir());
