diff -ur a/FreeFileSync/Source/base/ffs_paths.cpp b/FreeFileSync/Source/base/ffs_paths.cpp
--- a/FreeFileSync/Source/base/ffs_paths.cpp	2019-01-02 23:57:37.768501228 +0100
+++ b/FreeFileSync/Source/base/ffs_paths.cpp	2019-01-03 00:02:28.632322663 +0100
@@ -65,10 +65,10 @@
     wxTheApp->SetAppName(L"FreeFileSync");
     ZEN_ON_SCOPE_EXIT(wxTheApp->SetAppName(appName));
 
-    //if (isPortableVersion())
-    return appendSeparator(getExeFolderParentPath());
-    //else //use OS' standard paths
-    //    return appendSeparator(utfTo<Zstring>(wxStandardPathsBase::Get().GetResourcesDir()));
+    if (isPortableVersion())
+        return appendSeparator(getExeFolderParentPath());
+    else  // use OS' standard paths
+       return appendSeparator(utfTo<Zstring>(wxStandardPathsBase::Get().GetResourcesDir()));
 }
