--- langkit/libmanage.py	2024-04-04 19:26:44.432829600 +1100
+++ langkit/libmanage.py-new	2025-05-19 21:46:50.334525472 +1000
@@ -8,7 +8,7 @@
 import os
 from os import path
 import pdb
-import pipes
+import shlex
 import shutil
 import subprocess
 import sys
@@ -1580,7 +1580,7 @@
         """
         if self.verbosity.debug:
             printcol('Executing: {}'.format(
-                ' '.join(pipes.quote(arg) for arg in argv)
+                ' '.join(shlex.quote(arg) for arg in argv)
             ), Colors.CYAN)
 
     def log_info(self, msg: str, color: str) -> None:
