--- Rivet-2.7.2/bin/rivet	2019-05-11 16:41:04.000000000 +0200
+++ /usr/bin/rivet	2019-05-23 11:27:24.561229428 +0200
@@ -99,7 +99,7 @@
 extragroup.add_argument("-d", "--dump", "--histo-interval", dest="DUMP_PERIOD", type=int,
                         default=1000, metavar="NUM",
                         help="specify the number of events between histogram file updates, "
-                        "default = %default. Set to 0 to only write out at the end of the run. "
+                        "default = %(default)s. Set to 0 to only write out at the end of the run. "
                         "Note that intermediate histograms will be those from the analyze step "
                         "only, except for analyses explicitly declared Reentrant for which the "
                         "finalize function is executed first.")
@@ -367,7 +367,7 @@
             import tempfile, subprocess
             tffd, tfpath = tempfile.mkstemp(prefix="rivet-show.")
             msgsum = u"\n\n".join(msgs)
-            if type(u'') is not str:
+            if type(u'') is str:
                 msgsum = msgsum.encode('utf-8')
             os.write(tffd, msgsum)
             if sys.stdout.isatty():
