diff --git a/sound/pci/hda/cs35l41_hda_property.c b/sound/pci/hda/cs35l41_hda_property.c
index b39f944..feb4852 100644
--- a/sound/pci/hda/cs35l41_hda_property.c
+++ b/sound/pci/hda/cs35l41_hda_property.c
@@ -72,6 +72,13 @@ static int asus_rog_2023_spkr_id2(struct cs35l41_hda *cs35l41, struct device *ph
 	} else {
 		cs35l41->reset_gpio = gpiod_get_index(physdev, NULL, 0, GPIOD_OUT_HIGH);
 	}
+	
+	if (strcmp(cs35l41->acpi_subsystem_id, "104317F3") == 0){
+		hw_cfg->bst_type = CS35L41_INT_BOOST; /* ROG ALLY specific config */
+		hw_cfg->bst_ind = 1000; /* 1,000nH Inductance value */
+		hw_cfg->bst_ipk = 4500; /* 4,500mA peak current */
+		hw_cfg->bst_cap = 24; /* 24 microFarad cap value */
+	}
 
 	hw_cfg->valid = true;
 
