--- Gate-7.2/source/digits_hits/src/GateSecondaryProductionActor.cc	2016-02-15 16:37:41.000000000 +0100
+++ Gate-7.2/source/digits_hits/src/GateSecondaryProductionActor2.cc	2016-05-14 14:27:30.829183952 +0200
@@ -58,7 +58,6 @@ void GateSecondaryProductionActor::Const
 
    pFrag = new TH1F("fragments","Fragment production",3,0,1);
    pFrag->SetXTitle("Fragment name");
-   pFrag->SetBit(TH1::kCanRebin);
 
 
    pFrag->Fill("e- Ioni",0.);
