diff --git a/build-tools/proprietary/proprietary.projitems b/build-tools/proprietary/proprietary.projitems
index 9789a987..684212ea 100644
--- a/build-tools/proprietary/proprietary.projitems
+++ b/build-tools/proprietary/proprietary.projitems
@@ -18,6 +18,26 @@
       <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
+    <Content Include="$(_SourceDir)Mono.Android.DebugRuntime-*.xml">
+      <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(_SourceDir)ICSharpCode.SharpZipLib.dll">
+      <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(_SourceDir)INIFileParser.dll">
+      <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(_SourceDir)Xamarin.Android.Bindings.*.targets">
+      <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(_SourceDir)Xamarin.Installer.*">
+      <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
     <Content Include="$(_SourceDir)Mono.AndroidTools.*">
       <Link>$(_DestinationDir)%(Filename)%(Extension)</Link>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
