diff -ruN scidavis.1.D9.orig/config.pri scidavis.1.D9/config.pri
--- scidavis.1.D9.orig/config.pri	2015-11-24 08:07:15.000000000 +0300
+++ scidavis.1.D9/config.pri	2016-01-11 21:44:21.933009085 +0300
@@ -254,7 +254,7 @@
 	INCLUDEPATH  += /usr/include/qwtplot3d
 	LIBS         += -lqwtplot3d$${qwtsuff}
 
-	LIBS         += -lqwt$${qwtsuff} -lz -lGLU 
+	LIBS         += -lqwt5 -lz -lGLU
 
         !noassistant {LIBS+=-lQtAssistantClient}
 
