diff -ruN scidavis.1.D8.orig/config.pri scidavis.1.D8/config.pri
--- scidavis.1.D8.orig/config.pri	2014-07-22 12:01:44.000000000 +0400
+++ scidavis.1.D8/config.pri	2014-07-23 10:29:36.111933787 +0400
@@ -261,7 +261,7 @@
 	} else {
 		INCLUDEPATH  += /usr/include/qwt$${qwtsuff}
 	}
-	LIBS         += -lqwt$${qwtsuff} -lz -lGLU -lQtAssistantClient
+	LIBS         += -lqwt5 -lz -lGLU -lQtAssistantClient
 
 	INCLUDEPATH  += /usr/include/qwtplot3d$${qwtsuff}
 	LIBS         += -lqwtplot3d$${qwtsuff}
