diff --color --color -uraBN scilab-2025.1.0/scilab/bin/scilab scilab-2025.1.0-patched/scilab/bin/scilab
--- scilab-2025.1.0/scilab/bin/scilab	2025-05-22 05:05:24.000000000 -0500
+++ scilab-2025.1.0-patched/scilab/bin/scilab	2025-09-03 15:31:45.074186806 -0500
@@ -691,6 +691,7 @@
         export TCL_LIBRARY
         export TK_LIBRARY
     fi
+    LD_LIBRARY_PATH=/usr/lib/scilab${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
     PATH="$SCI:$SCI/bin:$PATH"
 fi
 
diff --color --color -uraBN scilab-2025.1.0/scilab/bin/scilab-adv-cli scilab-2025.1.0-patched/scilab/bin/scilab-adv-cli
--- scilab-2025.1.0/scilab/bin/scilab-adv-cli	2025-05-22 05:05:24.000000000 -0500
+++ scilab-2025.1.0-patched/scilab/bin/scilab-adv-cli	2025-09-03 15:31:45.074186806 -0500
@@ -691,6 +691,7 @@
         export TCL_LIBRARY
         export TK_LIBRARY
     fi
+    LD_LIBRARY_PATH=/usr/lib/scilab${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
     PATH="$SCI:$SCI/bin:$PATH"
 fi
 
diff --color --color -uraBN scilab-2025.1.0/scilab/bin/scilab-cli scilab-2025.1.0-patched/scilab/bin/scilab-cli
--- scilab-2025.1.0/scilab/bin/scilab-cli	2025-05-22 05:05:24.000000000 -0500
+++ scilab-2025.1.0-patched/scilab/bin/scilab-cli	2025-09-03 15:31:45.074186806 -0500
@@ -691,6 +691,7 @@
         export TCL_LIBRARY
         export TK_LIBRARY
     fi
+    LD_LIBRARY_PATH=/usr/lib/scilab${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
     PATH="$SCI:$SCI/bin:$PATH"
 fi
 
diff --color --color -uraBN scilab-2025.1.0/scilab/bin/scinotes scilab-2025.1.0-patched/scilab/bin/scinotes
--- scilab-2025.1.0/scilab/bin/scinotes	2025-05-22 05:05:24.000000000 -0500
+++ scilab-2025.1.0-patched/scilab/bin/scinotes	2025-09-03 15:31:45.074186806 -0500
@@ -691,6 +691,7 @@
         export TCL_LIBRARY
         export TK_LIBRARY
     fi
+    LD_LIBRARY_PATH=/usr/lib/scilab${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
     PATH="$SCI:$SCI/bin:$PATH"
 fi
 
diff --color --color -uraBN scilab-2025.1.0/scilab/bin/xcos scilab-2025.1.0-patched/scilab/bin/xcos
--- scilab-2025.1.0/scilab/bin/xcos	2025-05-22 05:05:24.000000000 -0500
+++ scilab-2025.1.0-patched/scilab/bin/xcos	2025-09-03 15:31:45.074186806 -0500
@@ -691,6 +691,7 @@
         export TCL_LIBRARY
         export TK_LIBRARY
     fi
+    LD_LIBRARY_PATH=/usr/lib/scilab${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
     PATH="$SCI:$SCI/bin:$PATH"
 fi
 
