diff --git a/src/meshlabplugins/filter_screened_poisson/filter_screened_poisson.pro b/src/meshlabplugins/filter_screened_poisson/filter_screened_poisson.pro
index 814bd31f..e04ad802 100644
--- a/src/meshlabplugins/filter_screened_poisson/filter_screened_poisson.pro
+++ meshlab/src/meshlabplugins/filter_screened_poisson/filter_screened_poisson.pro
@@ -2,6 +2,7 @@ include (../../shared.pri)
 
 macx:QMAKE_CXX = clang++-mp-3.9
 macx:QMAKE_LFLAGS += -L/opt/local/lib/libomp -lomp
+linux:QMAKE_LFLAGS += -lgomp
 macx:QMAKE_CXXFLAGS_RELEASE+= -O3 -DRELEASE -funroll-loops -ffast-math  -Wno-sign-compare -Wno-unused-parameter
 
 QMAKE_CXXFLAGS+=-fopenmp
@@ -22,4 +23,5 @@ DEFINES += FOR_RELEASE
 
 #PRE_TARGETDEPS += ./filter_screened_poisson.xml
 macx:QMAKE_POST_LINK = "cp "$$_PRO_FILE_PWD_/$$TARGET".xml ../../distrib/plugins/"$$TARGET".xml"
+linux:QMAKE_POST_LINK = "cp "$$_PRO_FILE_PWD_/$$TARGET".xml ../../distrib/plugins/"$$TARGET".xml; cd ../../distrib/plugins/ ; ln -s "$$TARGET".xml lib"$$TARGET".xml"
 
