--- dosbox/src/gui/sdlmain.cpp  2016-10-08 00:26:46.634598790 -0400
+++ dosbox/src/gui/sdlmain.cpp  2016-10-08 00:34:18.071285934 -0400
@@ -1042,10 +1042,6 @@
    //if (!(flags&GFX_CAN_32) || (flags & GFX_RGBONLY)) goto dosurface;
    // SDLScreen_Reset();
    int texsize=2 << int_log2(width > height ? width : height);
-   if (texsize>sdl.opengl.max_texsize) {
-       LOG_MSG("SDL:OPENGL:No support for texturesize of %d, falling back to surface",texsize);
-       goto dosurface;
-   }
    SDL_GL_SetAttribute( SDL_GL_DOUBLEBUFFER, 1 );
 #if SDL_VERSION_ATLEAST(1, 2, 11)
    Section_prop * sec=static_cast<Section_prop *>(control->GetSection("vsync"));


