diff -ru a/QTfrontend/CMakeLists.txt b/QTfrontend/CMakeLists.txt
--- a/QTfrontend/CMakeLists.txt	2024-01-15 23:35:38.544868616 +0100
+++ b/QTfrontend/CMakeLists.txt	2024-01-16 11:46:47.795047328 +0100
@@ -14,7 +14,11 @@
 include(CheckLibraryExists)
 
 find_package(SDL2 REQUIRED CONFIG)
-find_package(SDL2_mixer REQUIRED CONFIG) #audio in SDLInteraction
+if(WIN32 AND VCPKG_TOOLCHAIN)
+	find_package(SDL2_mixer REQUIRED CONFIG) #audio in SDLInteraction
+else()
+	find_package(SDL2_mixer 2 REQUIRED) #audio in SDLInteraction
+endif()
 include_directories(${SDL2_INCLUDE_DIRS})
 include_directories(${SDL2_MIXER_INCLUDE_DIRS})
 
