diff --git a/JASP-Desktop/mainwindow.cpp b/JASP-Desktop/mainwindow.cpp
index 14e56a9..5aed19c 100644
--- a/JASP-Desktop/mainwindow.cpp
+++ b/JASP-Desktop/mainwindow.cpp
@@ -120,9 +120,7 @@ MainWindow::MainWindow(QWidget *parent) :
 	ui->tabBar->addTab("Variables"); // variables view
 #endif
 	ui->tabBar->addTab("Common");
-#ifndef __linux__
 	ui->tabBar->addOptionsTab(); // no SEM under linux for now
-#endif
 	ui->tabBar->addHelpTab();
 
 	connect(ui->tabBar, SIGNAL(currentChanged(int)), this, SLOT(tabChanged(int)));
@@ -670,7 +668,6 @@ void MainWindow::tabChanged(int index)
 		ui->topLevelWidgets->setCurrentIndex(1);
 	}
 #endif
-#ifndef __linux__
 	else if (index == ui->tabBar->count() - 1)
 	{
 		if (_optionsForm == NULL)
@@ -682,7 +679,6 @@ void MainWindow::tabChanged(int index)
 
 		ui->topLevelWidgets->setCurrentWidget(_optionsForm);
 	}
-#endif
 	else
 	{
 #ifdef QT_DEBUG  // if variables tab enabled
@@ -988,15 +984,11 @@ void MainWindow::updateMenuEnabledDisabledStatus()
 
 void MainWindow::updateUIFromOptions()
 {
-#ifdef __linux__
-	ui->tabBar->removeTab("SEM");
-#else
 	QVariant sem = _settings.value("plugins/sem", false);
 	if (sem.canConvert(QVariant::Bool) && sem.toBool())
 		ui->tabBar->addTab("SEM");
 	else
 		ui->tabBar->removeTab("SEM");
-#endif
 
 	QVariant rl = _settings.value("toolboxes/r11tLearn", false);
 	if (rl.canConvert(QVariant::Bool) && rl.toBool())
