diff --git a/mec/protoname.c b/mec/protoname.c
index 4fee90e..f4183b4 100644
--- a/mec/protoname.c
+++ b/mec/protoname.c
@@ -14,6 +14,7 @@
 #include <string.h>
 #include <netdb.h>
 #include <sys/types.h>
+#include <ctype.h>

 #include "mec/ipv6ext.h"

diff --git a/rip.c b/rip.c
index bda372d..a13a9e2 100644
--- a/rip.c
+++ b/rip.c
@@ -63,7 +63,8 @@ bool do_opt(char *opt, char *arg, sendip_data *pack) {
 		ripopt = RIP_OPTION(pack);
 		p=q=arg;
 		/* TODO: if arg is malformed, this could segfault */
-		while(*(q++)!=':') /* do nothing */; *(--q)='\0';
+		while(*(q++)!=':') /* do nothing */;
+		*(--q)='\0';
 		rippack->addressFamily= htons((p==q)?2:(u_int16_t)strtoul(p, (char **)0, 0));
 		pack->modified |= RIP_MOD_ADDRFAM;
 		p=++q; while(*(q++)!=':') /* do nothing */; *(--q)='\0';
diff --git a/ripng.c b/ripng.c
index 1dab2a1..fec2de6 100644
--- a/ripng.c
+++ b/ripng.c
@@ -77,7 +77,8 @@ bool do_opt(char *opt, char *arg, sendip_data *pack) {
 		ripopt = RIPNG_ENTRY(pack);
 		p=q=arg;
 		/* TODO: if arg is malformed, this could segfault */
-		while(*(q++)!='/') /* do nothing */; *(--q)='\0';
+		while(*(q++)!='/') /* do nothing */;
+		*(--q)='\0';
 		ripopt->prefix = (p==q)?in6addr_any:inet6_addr(p);
 
 		p=++q; while(*(q++)!='/') /* do nothing */; *(--q)='\0';
