--- server/Makefile.orig	2019-05-30 17:01:26.463718143 +0000
+++ server/Makefile	2019-05-30 17:01:13.127062905 +0000
@@ -173,36 +173,36 @@
 	rm -rf $(CORE_FONTS) $(OUTPUT) $(GRUNT_FILES)
 
 install:
-	mkdir -pv ${DESTDIR}/var/www/onlyoffice
-	if ! id -u onlyoffice > /dev/null 2>&1; then useradd -m -d /var/www/onlyoffice -r -U onlyoffice; fi
+	#mkdir -pv ${DESTDIR}/var/www/onlyoffice
+	#if ! id -u onlyoffice > /dev/null 2>&1; then useradd -m -d /var/www/onlyoffice -r -U onlyoffice; fi
 
 	mkdir -p ${DESTDIR}${DOCUMENT_ROOT}/fonts
 	mkdir -p ${DESTDIR}/var/log/onlyoffice/documentserver
 	mkdir -p ${DESTDIR}/var/lib/onlyoffice/documentserver/App_Data
 
 	cp -fr -t ${DESTDIR}${DOCUMENT_ROOT} build/* ../web-apps-pro/deploy/*
-	mkdir -p ${DESTDIR}/etc/onlyoffice/documentserver
-	mv ${DESTDIR}${DOCUMENT_ROOT}/server/Common/config/* ${DESTDIR}/etc/onlyoffice/documentserver
+	mkdir -p ${DESTDIR}/etc/webapps/onlyoffice/documentserver
+	mv ${DESTDIR}${DOCUMENT_ROOT}/server/Common/config/* ${DESTDIR}/etc/webapps/onlyoffice/documentserver
 
-	chown onlyoffice:onlyoffice -R ${DESTDIR}/var/www/onlyoffice
-	chown onlyoffice:onlyoffice -R ${DESTDIR}/var/log/onlyoffice
-	chown onlyoffice:onlyoffice -R ${DESTDIR}/var/lib/onlyoffice
+	#chown onlyoffice:onlyoffice -R ${DESTDIR}/var/www/onlyoffice
+	#chown onlyoffice:onlyoffice -R ${DESTDIR}/var/log/onlyoffice
+	#chown onlyoffice:onlyoffice -R ${DESTDIR}/var/lib/onlyoffice
 
 	# Make symlinks for shared libs
 	find \
 		${DESTDIR}${DOCUMENT_ROOT}/server/FileConverter/bin \
 		-maxdepth 1 \
 		-name *$(SHARED_EXT) \
-		-exec sh -c 'ln -sf {} ${DESTDIR}/lib/$$(basename {})' \;
+		-exec sh -c 'ln -sf ${DOCUMENT_ROOT}/server/FileConverter/bin/$$(basename {}) ${DESTDIR}/usr/lib/$$(basename {})' \;
 
-	sudo -u onlyoffice "${DESTDIR}${DOCUMENT_ROOT}/server/tools/AllFontsGen"\
-		--input="${DESTDIR}${DOCUMENT_ROOT}/core-fonts"\
-		--allfonts-web="${DESTDIR}${DOCUMENT_ROOT}/sdkjs/common/AllFonts.js"\
-		--allfonts="${DESTDIR}${DOCUMENT_ROOT}/server/FileConverter/bin/AllFonts.js"\
-		--images="${DESTDIR}${DOCUMENT_ROOT}/sdkjs/common/Images"\
-		--selection="${DESTDIR}${DOCUMENT_ROOT}/server/FileConverter/bin/font_selection.bin"\
-		--output-web="${DESTDIR}${DOCUMENT_ROOT}/fonts"\
-		--use-system="true"
+	#sudo -u onlyoffice "${DESTDIR}${DOCUMENT_ROOT}/server/tools/AllFontsGen"\
+	#	--input="${DESTDIR}${DOCUMENT_ROOT}/core-fonts"\
+	#	--allfonts-web="${DESTDIR}${DOCUMENT_ROOT}/sdkjs/common/AllFonts.js"\
+	#	--allfonts="${DESTDIR}${DOCUMENT_ROOT}/server/FileConverter/bin/AllFonts.js"\
+	#	--images="${DESTDIR}${DOCUMENT_ROOT}/sdkjs/common/Images"\
+	#	--selection="${DESTDIR}${DOCUMENT_ROOT}/server/FileConverter/bin/font_selection.bin"\
+	#	--output-web="${DESTDIR}${DOCUMENT_ROOT}/fonts"\
+	#	--use-system="true"
 
 uninstall:
 	userdel onlyoffice
