diff -Naur original/data/gnome-flashback-compiz.session.in modified/data/gnome-flashback-compiz.session.in
--- original/data/gnome-flashback-compiz.session.in	2014-10-18 10:34:42.789009081 +0100
+++ modified/data/gnome-flashback-compiz.session.in	2014-10-18 10:37:31.023080204 +0100
@@ -1,3 +1,3 @@
 [GNOME Session]
 _Name=GNOME Flashback (Compiz)
-RequiredComponents=gnome-flashback-init;gnome-flashback;gnome-panel;compiz;gnome-settings-daemon;
+RequiredComponents=gnome-flashback-init;gnome-flashback;gnome-panel;compiz;gnome-settings-daemon;polkit-gnome-authentication-agent-1;nautilus-classic;
diff -Naur original/data/gnome-flashback-metacity.session.in modified/data/gnome-flashback-metacity.session.in
--- original/data/gnome-flashback-metacity.session.in	2014-10-18 10:34:42.789009081 +0100
+++ modified/data/gnome-flashback-metacity.session.in	2014-10-18 10:38:05.066399469 +0100
@@ -1,3 +1,3 @@
 [GNOME Session]
 _Name=GNOME Flashback (Metacity)
-RequiredComponents=gnome-flashback-init;gnome-flashback;gnome-panel;metacity;gnome-settings-daemon;
+RequiredComponents=gnome-flashback-init;gnome-flashback;gnome-panel;metacity;gnome-settings-daemon;polkit-gnome-authentication-agent-1;nautilus-classic;
