diff --git a/look-and-feel/Moe/contents/defaults b/look-and-feel/Moe/contents/defaults
index dc59ce5..8dd13bd 100644
--- a/look-and-feel/Moe/contents/defaults
+++ b/look-and-feel/Moe/contents/defaults
@@ -17,7 +17,7 @@ LayoutName=org.kde.breeze.desktop
 LayoutName=preview_stack
 
 [kwinrc][org.kde.kdecoration2]
-library=org.kde.sierrabreeze
+library=org.kde.sierrabreezejomada
 theme=Brisa
 
 [plasmarc][Theme]
