--- CMakeLists.txt
+++ CMakeLists.txt

@@ -20,10 +20,8 @@ Change the runtime path assigned during packaging and enable building tools and shared library at the same time
 option(BUILD_SHARED_LIBS "Build LASlib as DLL" OFF)
 
 if (BUILD_SHARED_LIBS AND UNIX AND NOT APPLE)
-	set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib/LASlib")
+	set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
 endif()
 
 add_subdirectory(LASlib/src)
-if (NOT BUILD_SHARED_LIBS)
 add_subdirectory(src)
-endif()

--- LASlib/src/CMakeLists.txt
+++ LASlib/src/CMakeLists.txt

@@ -114,8 +114,8 @@ Change the installation directory
 else()
 	install(TARGETS LASlib COMPONENT laslib EXPORT laslib-targets
-		ARCHIVE DESTINATION lib/LASlib
-		LIBRARY DESTINATION lib/LASlib
-		RUNTIME DESTINATION lib/LASlib)
+		ARCHIVE DESTINATION lib
+		LIBRARY DESTINATION lib
+		RUNTIME DESTINATION lib)
 	install(EXPORT laslib-targets COMPONENT laslib DESTINATION lib/cmake/LASlib)
 	install(FILES ${CMAKE_SOURCE_DIR}/LASlib/src/laslib-config.cmake COMPONENT laslib DESTINATION lib/cmake/LASlib)
 endif(MSVC)
