diff -ru --color old/setup.py new/setup.py
--- old/setup.py	2020-05-30 20:52:59.944499004 +0200
+++ new/setup.py	2020-05-30 20:54:13.063630153 +0200
@@ -22,7 +22,7 @@
 install_requires = [
     'pycurl',
     'pyparsing<3;python_version<="3.4"',
-    'pyparsing>3*;python_version>="3.5"',
+    'pyparsing;python_version>="3.5"',
     'future',
     'six',
     'configparser;python_version<"3.5"',
@@ -36,7 +36,7 @@
 setup(
     name="wfuzz",
     include_package_data=True,
-    data_files=[('docs/user', ['docs/user/advanced.rst'])],
+   #data_files=[('docs/user', ['docs/user/advanced.rst'])],
     packages=find_packages(where='src'),
     package_dir={'wfuzz': 'src/wfuzz'},
     entry_points={
