--- a/setup.py
+++ b/setup.py
@@ -12,14 +12,9 @@
 
 def get_version():
-    with open(version_file, 'r') as f:
-        exec(compile(f.read(), version_file, 'exec'))
-    import sys
-
-    # return short version for sdist
-    if 'sdist' in sys.argv or 'bdist_wheel' in sys.argv:
-        return locals()['short_version']
-    else:
-        return locals()['__version__']
+    version_ns = {}
+    with open(version_file, 'r') as f:
+        exec(f.read(), version_ns)
+    return version_ns['__version__']
 
 
 def parse_requirements(fname='requirements.txt', with_version=True):
