--- leo-editor-6.1/setup.py
+++ leo-editor-6.1/setup.py
@@ -113,7 +113,6 @@
     'meta',  # for livecode.py plugin, which is enabled by default
     'nbformat',  # for Jupyter notebook integration
     'pylint', 'pyflakes', 'black',  # coding syntax standards
-    'setupext-janitor >= 1.1',  # extend `setup.py clean` #1055,#1255
     'pyshortcuts >= 1.7',  # desktop integration (#1243)
     'sphinx',  # rST plugin
     'windows-curses; platform_system=="Windows"',  # for console mode on Windows
@@ -165,7 +164,7 @@
     setup_requires=setup_requires,
     install_requires=user_requires,
     entry_points=define_entry_points(),
-    cmdclass={'clean': janitor.CleanCommand},  # clean more than setuptools, #1055
+    cmdclass=cmd_classes,  # clean more than setuptools, #1055
     python_requires='>=3.6',
 )
 
