--- /home/JohannaNIX/Downloads/test/setupPrintcapij	2020-01-21 03:17:26.000000000 +0100
+++ /home/JohannaNIX/Downloads/test/setupPrintcapij_mod	2025-01-24 13:32:31.703664176 +0100
@@ -26,8 +26,4 @@
 DEVICE_IF=/dev/usb/lp0
 
-if [ ! -f $PRINTCAP_NAME ]; then
-	PRINTCAP_NAME=/etc/printcap
-fi
-
 case "$2" in
 	-i)
@@ -35,5 +31,5 @@
 	        mkdir -p $SPOOLER_NAME
 	fi
-	chown lp $SPOOLER_NAME
+	chown daemon $SPOOLER_NAME
 	chgrp lp $SPOOLER_NAME
 	chmod 700 $SPOOLER_NAME
