diff -urN abPOA-1.5.7.orig/CMakeLists.txt abPOA-1.5.7/CMakeLists.txt
--- abPOA-1.5.7.orig/CMakeLists.txt	2026-06-26 12:49:17.000000000 +0530
+++ abPOA-1.5.7/CMakeLists.txt	2026-09-10 09:03:42.602865404 +0530
@@ -183,7 +183,7 @@
 else()
     # Fallback: single compile with -march=native
     target_sources(abpoa PRIVATE src/abpoa_align_simd.c)
-    target_compile_options(abpoa PRIVATE -march=native)
+    # target_compile_options(abpoa PRIVATE -march=native)
 endif()
 
 if(ABPOA_BUILD_EXE)
diff -urN abPOA-1.5.7.orig/include/simd_instruction.h abPOA-1.5.7/include/simd_instruction.h
--- abPOA-1.5.7.orig/include/simd_instruction.h	2026-06-26 12:49:17.000000000 +0530
+++ abPOA-1.5.7/include/simd_instruction.h	2026-09-10 09:06:09.464942647 +0530
@@ -19,15 +19,15 @@
 #include <immintrin.h>
 #else // use SIMDE
 #ifdef __AVX512BW__
-#include "simde/simde/x86/avx512.h"
+#include <simde/x86/avx512.h>
 #else
 #ifdef __AVX2__
-#include "simde/simde/x86/avx2.h"
+#include <simde/x86/avx2.h>
 #else
 #ifdef __SSE4_1__
-#include "simde/simde/x86/sse4.1.h"
+#include <simde/x86/sse4.1.h>
 #else
-#include "simde/simde/x86/sse2.h"
+#include <simde/x86/sse2.h>
 #endif // end of sse41
 #endif // end of AVX2
 #endif // end of 512F
diff -urN abPOA-1.5.7.orig/Makefile abPOA-1.5.7/Makefile
--- abPOA-1.5.7.orig/Makefile	2026-06-26 12:49:17.000000000 +0530
+++ abPOA-1.5.7/Makefile	2026-09-10 09:04:31.808003043 +0530
@@ -26,7 +26,7 @@
 	OPT_FLAGS = -O3
 endif
 
-CFLAGS        = $(OPT_FLAGS) $(EXTRA_FLAGS)
+CFLAGS        = $(CPPFLAGS) $(OPT_FLAGS) $(EXTRA_FLAGS) $(LDFLAGS)
 
 # for gprof
 ifneq ($(pg),)
diff -urN abPOA-1.5.7.orig/src/simd_instruction.h abPOA-1.5.7/src/simd_instruction.h
--- abPOA-1.5.7.orig/src/simd_instruction.h	2026-06-26 12:49:17.000000000 +0530
+++ abPOA-1.5.7/src/simd_instruction.h	2026-09-10 09:06:17.217239789 +0530
@@ -19,15 +19,15 @@
 #include <immintrin.h>
 #else // use SIMDE
 #ifdef __AVX512BW__
-#include "simde/simde/x86/avx512.h"
+#include <simde/x86/avx512.h>
 #else
 #ifdef __AVX2__
-#include "simde/simde/x86/avx2.h"
+#include <simde/x86/avx2.h>
 #else
 #ifdef __SSE4_1__
-#include "simde/simde/x86/sse4.1.h"
+#include <simde/x86/sse4.1.h>
 #else
-#include "simde/simde/x86/sse2.h"
+#include <simde/x86/sse2.h>
 #endif // end of sse41
 #endif // end of AVX2
 #endif // end of 512F
