diff -aur SimplicityStudio_v3/setup.sh SimplicityStudio_v3_patched/setup.sh
--- SimplicityStudio_v3/setup.sh	2015-10-20 22:24:35.000000000 +0200
+++ SimplicityStudio_v3_patched/setup.sh	2015-11-18 16:09:49.410240104 +0100
@@ -48,8 +48,8 @@
 	chmod $mode "$folder/studio.desktop"
 }
 
-if [ -d "/usr/share/applications" ]; then
-	create_desktop_file "/usr/share/applications" root 644
+if [ -d "$DESTDIR/usr/share/applications" ]; then
+	create_desktop_file "$DESTDIR/usr/share/applications" root 644
 elif [ -d "$SUDO_HOME/.local/share/applications" ]; then
 	create_desktop_file "$SUDO_HOME/.local/share/applications" "$SUDO_USER" 755
 elif [ -d "$SUDO_HOME/Desktop" ]; then
@@ -61,8 +61,8 @@
 # Exit if /etc/udev/rules.d does not exist
 #
 
-if [ ! -d "/etc/udev/rules.d" ]; then
-	echo ERROR /etc/udev/rules.d does not exist
+if [ ! -d "$DESTDIR/etc/udev/rules.d" ]; then
+	echo ERROR $DESTDIR/etc/udev/rules.d does not exist
 	exit 1
 fi
 
@@ -77,7 +77,7 @@
 do
 	rules_file=`basename "$rules_path"`
 	echo "Installing $rules_file...";
-	cp "$rules_path" /etc/udev/rules.d
+	cp "$rules_path" "$DESTDIR/etc/udev/rules.d"
 done
 
 # ------------------------------------------------------------------
