diff -aur SimplicityStudio_v4/setup.sh SimplicityStudio_v4_patched/setup.sh
--- SimplicityStudio_v4/setup.sh	2016-10-21 13:55:59.000000000 -0700
+++ SimplicityStudio_v4_patched/setup.sh	2016-11-01 10:35:31.898186828 -0700
@@ -49,8 +49,8 @@
 	chmod $mode "$folder/studio.desktop"
 }
 
-if [ -d "/usr/share/applications" ]; then
-	create_desktop_file "/usr/share/applications" root 644
+if [ -d "$DESTDIR/usr/share/applications" ]; then
+	create_desktop_file "$DESTDIR/usr/share/applications" root 644
 elif [ -d "$SUDO_HOME/.local/share/applications" ]; then
 	create_desktop_file "$SUDO_HOME/.local/share/applications" "$SUDO_USER" 755
 elif [ -d "$SUDO_HOME/Desktop" ]; then
@@ -62,8 +62,8 @@
 # Exit if /etc/udev/rules.d does not exist
 #
 
-if [ ! -d "/etc/udev/rules.d" ]; then
-	echo ERROR /etc/udev/rules.d does not exist
+if [ ! -d "$DESTDIR/etc/udev/rules.d" ]; then
+	echo ERROR $DESTDIR/etc/udev/rules.d does not exist
 	exit 1
 fi
 
@@ -78,7 +78,7 @@
 do
 	rules_file=`basename "$rules_path"`
 	echo "Installing udev $rules_file...";
-	cp "$rules_path" /etc/udev/rules.d
+	cp "$rules_path" "$DESTDIR/etc/udev/rules.d"
 done
 
 # ------------------------------------------------------------------
