diff -uNr skafos-1.2/CMakeLists.txt skafos-1.2-1/CMakeLists.txt
--- skafos-1.2/CMakeLists.txt	2018-02-09 12:37:44.000000000 -0600
+++ skafos-1.2-1/CMakeLists.txt	2018-02-10 10:29:03.772920865 -0600
@@ -107,14 +107,4 @@
 install(TARGETS ${EXE_NAME} RUNTIME DESTINATION bin)
 
 include(InstallRequiredSystemLibraries)
-set(CPACK_PACKAGE_VERSION ${RELEASE_VERSION})
-set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-set(CPACK_GENERATOR "DEB")
-set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
-set(CPACK_PACKAGE_DESCRIPTION "Commandline tool for the Metis Machine (www.metismachine.com) Skafos platform.")
-set(CPACK_PACKAGE_CONTACT "support@metismachine.com")
-endif()
-
-include(CPack)
diff -uNr skafos-1.2/Makefile skafos-1.2-1/Makefile
--- skafos-1.2/Makefile	2018-02-09 12:37:44.000000000 -0600
+++ skafos-1.2-1/Makefile	2018-02-10 10:29:03.772920865 -0600
@@ -43,8 +43,7 @@
 	@cd $(BUILD_DIR) \
 	&& echo "Building skafos..." \
 	&& cmake .. -Wno-dev \
-	&& make	\
-	&& cpack --config CPackConfig.cmake
+	&& make
 
 clean:
 	@rm -rf _build/* 
