diff --git a/KikitanTranslator.Photino/KikitanTranslator.Photino.csproj b/KikitanTranslator.Photino/KikitanTranslator.Photino.csproj
index 94be314..e833eda 100644
--- a/KikitanTranslator.Photino/KikitanTranslator.Photino.csproj
+++ b/KikitanTranslator.Photino/KikitanTranslator.Photino.csproj
@@ -87,7 +87,7 @@
 
     <Target Name="FlatpakBundleLinux"
             AfterTargets="VelopackPackLinux"
-            Condition="'$(Configuration)' == 'Release' AND '$(RuntimeIdentifier)' == 'linux-x64'">
+            Condition="'$(Configuration)' == 'Release' AND '$(RuntimeIdentifier)' == 'linux-x64' AND '$(SkipFlatpak)' != 'true'">
         <Message Importance="High" Text="Building and bundling Flatpak package..." />
         
         <Exec Command="flatpak-builder &quot;$(OutDir)flatpak-build&quot; &quot;$(MSBuildProjectDirectory)/flatpak/$(PackageId).json&quot; --repo=&quot;$(OutDir)flatpak-repo&quot;" />
