From d49a12caed22fc684de822942fcf29d85d19ade1 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Thu, 19 Aug 2021 13:52:36 +0200
Subject: [PATCH] Skip test requiring RPM database if not present

---
 test/test_lint.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/test_lint.py b/test/test_lint.py
index fe4ebfb..5e8c719 100644
--- a/test/test_lint.py
+++ b/test/test_lint.py
@@ -423,6 +423,7 @@ def test_run_rpmlintrc_single_file(capsys, packages):
     assert 'no-%build-section' not in out
 
 
+@pytest.mark.skipif(not HAS_RPMDB, reason='No RPM database present')
 def test_installed_package(capsys):
     additional_options = {
         'installed': ['bzip2'],
-- 
2.33.0

