diff -ur Slingshot-0.8.1p-linux.orig/slingshot/planet.py Slingshot-0.8.1p-linux/slingshot/planet.py
--- Slingshot-0.8.1p-linux.orig/slingshot/planet.py	2007-06-22 12:52:33.000000000 +0100
+++ Slingshot-0.8.1p-linux/slingshot/planet.py	2012-08-23 22:39:42.366476664 +0100
@@ -56,7 +56,7 @@
 				if d < (self.r + p.get_radius()) * 1.5 + 0.1 * (self.mass + p.get_mass()):
 					positioned = False
 		
-		s = round(2 * self.r / 0.96)
+		s = int(round(2 * self.r / 0.96))
 		self.orig = pygame.transform.scale(self.image, (s, s))
 		
 		self.image = self.orig
diff -ur Slingshot-0.8.1p-linux.orig/slingshot/player.py Slingshot-0.8.1p-linux/slingshot/player.py
--- Slingshot-0.8.1p-linux.orig/slingshot/player.py	2007-07-13 19:27:18.000000000 +0100
+++ Slingshot-0.8.1p-linux/slingshot/player.py	2012-08-23 22:39:48.449904152 +0100
@@ -237,8 +237,8 @@
 		y = round(pos[1] - self.rect.topleft[1])
 		if x <= 1 or y <= 1:
 			return False
-		x = x - 1
-		y = y - 1
+		x = int(x - 1)
+		y = int(y - 1)
 		if not self.image.get_at((x,y)) == (0,0,0,0):
 			self.shot = True
 			return True
diff -ur Slingshot-0.8.1p-linux.orig/slingshot/slingshot.py Slingshot-0.8.1p-linux/slingshot/slingshot.py
--- Slingshot-0.8.1p-linux.orig/slingshot/slingshot.py	2007-07-13 20:57:05.000000000 +0100
+++ Slingshot-0.8.1p-linux/slingshot/slingshot.py	2012-08-23 22:40:05.066827632 +0100
@@ -442,7 +442,7 @@
 				tmp = tmp.convert_alpha()
 				rect = tmp.get_rect()
 				s = (100 - self.show_round) * rect.h / 25
-				tmp = pygame.transform.scale(tmp, (rect.w / rect.h * s, s ))
+				tmp = pygame.transform.scale(tmp, (int(rect.w / rect.h * s), int(s) ))
 				rect = tmp.get_rect()
 				rect.center = (399,299)
 				self.screen.blit(tmp, rect.topleft)
diff -ur Slingshot-0.8.1p-linux.orig/slingshot.desktop Slingshot-0.8.1p-linux/slingshot.desktop
--- Slingshot-0.8.1p-linux.orig/slingshot.desktop	2007-06-23 21:45:09.000000000 +0100
+++ Slingshot-0.8.1p-linux/slingshot.desktop	2012-08-23 22:46:50.806435681 +0100
@@ -4,6 +4,6 @@
 Comment=A Newtonian Strategy Game
 Exec=slingshot
 Categories=Application;Game;
-Icon=/opt/slingshot/data/icon64x64.png
+Icon=/usr/share/slingshot/data/icon64x64.png
 Terminal=false
 Type=Application
