--- a/smc/smc/src/overworld/world_manager.cpp	2014-07-13 17:11:53.337301000 +0200
+++ b/smc/smc/src/overworld/world_manager.cpp	2014-07-13 17:32:06.634548009 +0200
@@ -204,6 +204,16 @@
 		{
 			return obj;
 		}
+		else if(obj->m_description->m_path.at(0) == '/')
+		{
+			std::string newPath ( obj->m_description->m_path );
+			newPath.erase(0,1);
+			if(newPath.compare( path ) == 0)
+			{
+				return obj;
+			}
+		}
+
 	}
 
 	return NULL;
