diff --git a/code/renderergl2/tr_extensions.c b/code/renderergl2/tr_extensions.c
index 10a1cb3..36f6a57 100644
--- a/code/renderergl2/tr_extensions.c
+++ b/code/renderergl2/tr_extensions.c
@@ -127,31 +127,34 @@ void (APIENTRY * qglGetCompressedTexImageARB)(GLenum target, GLint lod,

 // GL_EXT_framebuffer_object
 GLboolean (APIENTRY * qglIsRenderbufferEXT)(GLuint renderbuffer);
-void (APIENTRY * qglBindRenderbufferEXT)(GLenum target, GLuint renderbuffer);
-void (APIENTRY * qglDeleteRenderbuffersEXT)(GLsizei n, const GLuint *renderbuffers);
-void (APIENTRY * qglGenRenderbuffersEXT)(GLsizei n, GLuint *renderbuffers);
+//void (APIENTRY * qglBindRenderbufferEXT)(GLenum target, GLuint renderbuffer);
+//void (APIENTRY * qglDeleteRenderbuffersEXT)(GLsizei n, const GLuint *renderbuffers);
+//void (APIENTRY * qglGenRenderbuffersEXT)(GLsizei n, GLuint *renderbuffers);

-void (APIENTRY * qglRenderbufferStorageEXT)(GLenum target, GLenum internalformat, GLsizei width, GLsizei height);
+//void (APIENTRY * qglRenderbufferStorageEXT)(GLenum target, GLenum internalformat, GLsizei width, GLsizei height);

 void (APIENTRY * qglGetRenderbufferParameterivEXT)(GLenum target, GLenum pname, GLint *params);

 GLboolean (APIENTRY * qglIsFramebufferEXT)(GLuint framebuffer);
-void (APIENTRY * qglBindFramebufferEXT)(GLenum target, GLuint framebuffer);
-void (APIENTRY * qglDeleteFramebuffersEXT)(GLsizei n, const GLuint *framebuffers);
-void (APIENTRY * qglGenFramebuffersEXT)(GLsizei n, GLuint *framebuffers);
+//void (APIENTRY * qglBindFramebufferEXT)(GLenum target, GLuint framebuffer);
+//void (APIENTRY * qglDeleteFramebuffersEXT)(GLsizei n, const GLuint *framebuffers);
+//void (APIENTRY * qglGenFramebuffersEXT)(GLsizei n, GLuint *framebuffers);

-GLenum (APIENTRY * qglCheckFramebufferStatusEXT)(GLenum target);
+//GLenum (APIENTRY * qglCheckFramebufferStatusEXT)(GLenum target);

 void (APIENTRY * qglFramebufferTexture1DEXT)(GLenum target, GLenum attachment, GLenum textarget, GLuint texture,
 	GLint level);
+/*
 void (APIENTRY * qglFramebufferTexture2DEXT)(GLenum target, GLenum attachment, GLenum textarget, GLuint texture,
 	GLint level);
+*/
 void (APIENTRY * qglFramebufferTexture3DEXT)(GLenum target, GLenum attachment, GLenum textarget, GLuint texture,
 	GLint level, GLint zoffset);

+/*
 void (APIENTRY * qglFramebufferRenderbufferEXT)(GLenum target, GLenum attachment, GLenum renderbuffertarget,
 	GLuint renderbuffer);
-
+*/
 void (APIENTRY * qglGetFramebufferAttachmentParameterivEXT)(GLenum target, GLenum attachment, GLenum pname, GLint *params);

 void (APIENTRY * qglGenerateMipmapEXT)(GLenum target);
@@ -167,14 +170,15 @@ void (APIENTRY * qglGetQueryObjectivARB)(GLuint id, GLenum pname, GLint *params)
 void (APIENTRY * qglGetQueryObjectuivARB)(GLuint id, GLenum pname, GLuint *params);

 // GL_EXT_framebuffer_blit
-void (APIENTRY * qglBlitFramebufferEXT)(GLint srcX0, GLint srcY0, GLint srcX1, GLint srcY1,
+/*void (APIENTRY * qglBlitFramebufferEXT)(GLint srcX0, GLint srcY0, GLint srcX1, GLint srcY1,
                             GLint dstX0, GLint dstY0, GLint dstX1, GLint dstY1,
                             GLbitfield mask, GLenum filter);
-
+*/
 // GL_EXT_framebuffer_multisample
+/*
 void (APIENTRY * qglRenderbufferStorageMultisampleEXT)(GLenum target, GLsizei samples,
 	GLenum internalformat, GLsizei width, GLsizei height);
-
+*/
 // GL_ARB_draw_buffers
 void (APIENTRY * qglDrawBuffersARB)(GLsizei n, const GLenum *bufs);
