--- src/qt4/main.cpp	2011-05-11 12:40:55.000000000 +0200
+++ src/qt4/main.cpp	2014-03-14 16:26:05.068095390 +0100
@@ -38,6 +38,8 @@
 #include "main_window.h"
 #include "options.h"
 
+#include <locale>
+
 int
 main( int argc, char ** argv )
 {
@@ -51,6 +53,8 @@ main( int argc, char ** argv )
     QApplication app( argc, argv );
     //QApplication::setStyle( "plastique" );
 
+    std::locale::global( std::locale::classic() );
+
     if ( ! Options::instance().parseCmdLine( argc, argv ) )
     {
         return 1;
