diff --git a/CMakeLists.txt b/CMakeLists.txt
index f98a4e2..b2a0ea8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,9 @@ set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/minizip CACHE PATH "Installa
 
 set(VERSION "2.3.3")
 
+# Minizip library SOVERSION
+set(SOVERSION "2")
+
 # Set cmake debug postfix to d
 set(CMAKE_DEBUG_POSTFIX "d")
 
@@ -328,6 +331,8 @@ if (MINGW AND BUILD_SHARED_LIBS)
   set_target_properties(${PROJECT_NAME} PROPERTIES ARCHIVE_OUTPUT_NAME "minizip")
 endif ()
 
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${VERSION} SOVERSION ${SOVERSION})
+
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE C PREFIX ""
                       POSITION_INDEPENDENT_CODE 1)
 if(USE_ZLIB)
