diff --git a/sphinxcontrib/tikz.py b/sphinxcontrib/tikz.py
index 4d4f8ce..bf21992 100644
--- a/sphinxcontrib/tikz.py
+++ b/sphinxcontrib/tikz.py
@@ -58,6 +58,7 @@ except ImportError:
 
 from docutils import nodes, utils
 from docutils.parsers.rst import directives
+from docutils.parsers.rst import Directive
 
 from sphinx.errors import SphinxError
 try:
@@ -65,8 +66,6 @@ try:
 except:
     from sphinx.util import ensuredir, ENOENT
 
-from sphinx.util.compat import Directive
-
 from glob import glob
 
 _Win_ = sys.platform[0:3] == 'win'
