--- CMakeLists.txt	2025-04-15 09:20:58.288947488 -0400
+++ "CMakeLists (copy).txt"	2025-04-15 09:25:04.697680848 -0400
@@ -15,12 +15,8 @@
 set_directory_properties(PROPERTIES EXCLUDE_FROM_ALL ON)
 
 # SPIRV Headers
-if (NOT TARGET SPIRV-Headers::SPIRV-Headers)
-    find_package(SPIRV-Headers)
-    if (NOT TARGET SPIRV-Headers::SPIRV-Headers)
-        add_subdirectory(SPIRV-Headers)
-    endif()
-endif()
+find_package(SPIRV-Headers)
+add_subdirectory(SPIRV-Headers)
 
 # fmt (also used by Dynarmic, so needs to be added first)
 if (NOT TARGET fmt::fmt)
