--- a/46sshd/sshd.service
+++ b/46sshd/sshd.service
@@ -11,7 +11,7 @@ Before=cryptsetup.target
 #
 # On distributions such as Gentoo, sshd doesn't come with
 # systemd support, thus, one has to set `Type=simple` there.
-Type=notify
+Type=simple
 
 EnvironmentFile=-/etc/crypto-policies/back-ends/opensshserver.config
 EnvironmentFile=-/etc/crypto-policies/back-ends/openssh-server.config
@@ -21,7 +21,7 @@ EnvironmentFile=-/etc/sysconfig/dracut-sshd
 
 # Start command requires the `-e` option if and only if `Type=simple`
 # is configured, see above.
-ExecStart=/usr/sbin/sshd -D $SSHD_OPTS $OPTIONS $CRYPTO_POLICY
+ExecStart=/usr/sbin/sshd -e -D $SSHD_OPTS $OPTIONS $CRYPTO_POLICY
 
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process
