diff -ur zdoom-2.6.1.orig/src/asm_ia32/a.asm zdoom-2.6.1/src/asm_ia32/a.asm
--- zdoom-2.6.1.orig/src/asm_ia32/a.asm	2012-07-02 00:49:34.000000000 +0200
+++ zdoom-2.6.1/src/asm_ia32/a.asm	2012-11-06 14:43:00.501239342 +0100
@@ -810,3 +810,13 @@
 GLOBAL _rtext_a_end
 _rtext_a_end:
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64  
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -ur zdoom-2.6.1.orig/src/asm_ia32/misc.asm zdoom-2.6.1/src/asm_ia32/misc.asm
--- zdoom-2.6.1.orig/src/asm_ia32/misc.asm	2012-07-02 00:49:34.000000000 +0200
+++ zdoom-2.6.1/src/asm_ia32/misc.asm	2012-11-06 14:43:10.124572789 +0100
@@ -198,3 +198,13 @@
 		pop		ebx
 		emms
 		ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64  
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -ur zdoom-2.6.1.orig/src/asm_ia32/tmap2.asm zdoom-2.6.1/src/asm_ia32/tmap2.asm
--- zdoom-2.6.1.orig/src/asm_ia32/tmap2.asm	2012-07-02 00:49:34.000000000 +0200
+++ zdoom-2.6.1/src/asm_ia32/tmap2.asm	2012-11-06 14:43:39.091239794 +0100
@@ -638,3 +638,13 @@
 GLOBAL _rtext_tmap2_end
 _rtext_tmap2_end:
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64  
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -ur zdoom-2.6.1.orig/src/asm_ia32/tmap3.asm zdoom-2.6.1/src/asm_ia32/tmap3.asm
--- zdoom-2.6.1.orig/src/asm_ia32/tmap3.asm	2012-07-02 00:49:34.000000000 +0200
+++ zdoom-2.6.1/src/asm_ia32/tmap3.asm	2012-11-06 14:35:26.421234034 +0100
@@ -342,3 +342,13 @@
 GLOBAL _rtext_tmap3_end
 _rtext_tmap3_end:
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -ur zdoom-2.6.1.orig/src/asm_ia32/tmap.asm zdoom-2.6.1/src/asm_ia32/tmap.asm
--- zdoom-2.6.1.orig/src/asm_ia32/tmap.asm	2012-07-02 00:49:34.000000000 +0200
+++ zdoom-2.6.1/src/asm_ia32/tmap.asm	2012-11-06 14:43:28.244573001 +0100
@@ -1891,3 +1891,13 @@
 		selfmod rtext_start, rtext_end
 		call	setpitch3
 		jmp		setvlinebpl_
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64  
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -ur zdoom-2.6.1.orig/src/asm_x86_64/tmap3.s zdoom-2.6.1/src/asm_x86_64/tmap3.s
--- zdoom-2.6.1.orig/src/asm_x86_64/tmap3.s	2012-07-02 00:49:36.000000000 +0200
+++ zdoom-2.6.1/src/asm_x86_64/tmap3.s	2012-11-06 14:34:28.807900032 +0100
@@ -139,3 +139,4 @@
 		.align 16
 
 
+		.section	.note.GNU-stack,"",@progbits
