--- a/Stackless/core/slp_transfer.c
+++ b/Stackless/core/slp_transfer.c
@@ -81,7 +81,7 @@
 #endif
 
 /* a write only variable used to prevent overly optimisation */
-intptr_t *global_goobledigoobs;
+static volatile intptr_t *global_goobledigoobs;
 static int
 climb_stack_and_transfer(PyCStackObject **cstprev, PyCStackObject *cst,
                          PyTaskletObject *prev)
--- a/Stackless/core/stacklesseval.c
+++ b/Stackless/core/stacklesseval.c
@@ -268,7 +268,7 @@
 }
 
 /* a write only variable used to prevent overly optimisation */
-intptr_t *global_goobledigoobs;
+static volatile intptr_t *global_goobledigoobs;
 static PyObject *
 climb_stack_and_eval_frame(PyFrameObject *f)
 {
