--- a/start-flashpoint.sh.old
+++ b/start-flashpoint.sh
@@ -2,18 +2,29 @@
 
 [ `id -u` -ne 0 ] && cd -P -- "`dirname "$0"`" || exit 1
 
-if [ "`file -b --mime-type Libraries/libgtk-3.so.0`" = 'application/x-sharedlib' ]; then
-  export GALLIUM_DRIVER="i915,i965,iris,nouveau,nouveau_vieux,r200,r300,r600,radeon,radeonsi,zink,swrast" GDK_PIXBUF_MODULE_FILE="$PWD/Libraries/loader.cache" GSETTINGS_SCHEMA_DIR="$PWD/Libraries" LD_LIBRARY_PATH="$PWD/Libraries" LIBGL_DRIVERS_PATH="$PWD/Libraries"
-  if [ ! -d FPSoftware/Wine/bin ]; then
+# Set configurable Flashpoint root directory.
+# Default: absolute path to script's directory
+[ "$FP_DIR" ] || FP_DIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
+
+
+if [ "`file -b --mime-type $FP_DIR/Libraries/libgtk-3.so.0`" = 'application/x-sharedlib' ]; then
+  export GALLIUM_DRIVER="i915,i965,iris,nouveau,nouveau_vieux,r200,r300,r600,radeon,radeonsi,zink,swrast" GDK_PIXBUF_MODULE_FILE="$FP_DIR/Libraries/loader.cache" GSETTINGS_SCHEMA_DIR="$FP_DIR/Libraries" LD_LIBRARY_PATH="$FP_DIR/Libraries" LIBGL_DRIVERS_PATH="$FP_DIR/Libraries"
+  if [ ! -d $FP_DIR/FPSoftware/Wine/bin ]; then
     CMD="`command -v wine`"
-    [ "$CMD" ] && export PATH="$PWD/Libraries:${CMD%/*}"
+    [ "$CMD" ] && export PATH="$FP_DIR/Libraries:${CMD%/*}"
   else
-    export PATH="$PWD/Libraries"
+    export PATH="$FP_DIR/Libraries"
   fi
 fi
 
-export WINEPREFIX="$PWD/FPSoftware/Wine"
+export WINEPREFIX="$FP_DIR/FPSoftware/Wine"
+
+# Check for correct ownership
+if [[ "$(stat -c %U "$FP_DIR/Data/flashpoint.sqlite")" != "$USER" ]]; then
+    echo -e "WARNING: Flashpoint directory is not owned by current user! Expect issues.
+    Run \"sudo USER=\$USER chown -R \$USER $FP_DIR\" to correct this."
+fi
 
 [ $# -eq 0 ] && FP_FLAGS="--js-flags=--lite_mode --ozone-platform-hint=auto" || FP_FLAGS="$@"
 cd Launcher
-./flashpoint-launcher $FP_FLAGS &
\ No newline at end of file
+./flashpoint-launcher $FP_FLAGS &
