diff -ruN AviSynthPlus-3.7.4/avs_core/CMakeLists.txt patched/avs_core/CMakeLists.txt
--- AviSynthPlus-3.7.4/avs_core/CMakeLists.txt	2025-03-26 16:56:09.563990577 +0100
+++ patched/avs_core/CMakeLists.txt	2025-03-26 16:56:22.999721313 +0100
@@ -236,3 +236,22 @@
 
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/avisynth.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+
+if (BUILD_STATIC_LIB)
+  add_library("AvsCoreStatic" STATIC ${AvsCore_Sources})
+  set_target_properties("AvsCoreStatic" PROPERTIES "OUTPUT_NAME" "${CoreName}")
+
+  # Specify include directories
+  target_include_directories("AvsCoreStatic" PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+  # Specify preprocessor definitions
+  target_compile_definitions("AvsCoreStatic" PRIVATE BUILDING_AVSCORE)
+  # Windows DLL dependencies
+  if (MSVC OR MINGW)
+    target_link_libraries("AvsCoreStatic" "uuid" "winmm" "vfw32" "msacm32" "gdi32" "user32" "advapi32" "ole32" "imagehlp" "ssp")
+  else()
+    target_link_libraries("AvsCoreStatic" "pthread" "dl" "${FSLIB}")
+  endif()
+
+  INSTALL(TARGETS AvsCoreStatic
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+endif()
diff -ruN AviSynthPlus-3.7.4/CMakeLists.txt patched/CMakeLists.txt
--- AviSynthPlus-3.7.4/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/CMakeLists.txt	2025-03-26 16:56:23.000418553 +0100
@@ -80,6 +80,11 @@
 
   # message("Compiler ID: ${CMAKE_CXX_COMPILER_ID} ") 
 
+  option(WITH_STATIC_LIB "Build with a static library" OFF)
+  if (WITH_STATIC_LIB)
+    set(BUILD_STATIC_LIB ON)
+  endif (WITH_STATIC_LIB)
+
   include(GNUInstallDirs)
 
   # Avoid uselessly linking to unused libraries
diff -ruN AviSynthPlus-3.7.4/plugins/ConvertStacked/CMakeLists.txt patched/plugins/ConvertStacked/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/ConvertStacked/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/plugins/ConvertStacked/CMakeLists.txt	2025-03-26 17:00:50.983234725 +0100
@@ -36,3 +36,17 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static)
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR})
+
+  INSTALL(TARGETS "${ProjectName}Static"
+          DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/DirectShowSource/CMakeLists.txt patched/plugins/DirectShowSource/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/DirectShowSource/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/plugins/DirectShowSource/CMakeLists.txt	2025-03-26 17:06:51.678600140 +0100
@@ -50,3 +50,17 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static "Winmm.lib" "Quartz.lib" "Ole32.lib" "User32.lib" "Oleaut32.lib" "Advapi32.lib" ${DSHOWSRC_BASECLASSES_LIB})
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR} ${DSHOWSRC_BASECLASSES_PATH})
+
+  INSTALL(TARGETS "${ProjectName}Static"
+        DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/ImageSeq/CMakeLists.txt patched/plugins/ImageSeq/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/ImageSeq/CMakeLists.txt	2025-03-26 16:56:09.568634051 +0100
+++ patched/plugins/ImageSeq/CMakeLists.txt	2025-03-26 17:05:50.028152549 +0100
@@ -70,3 +70,24 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static)
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR})
+
+  if(DevIL_FOUND)
+      target_link_libraries(${ProjectName}Static "${IL_LIBRARIES}" "${FSLIB}" "ssp")
+      target_include_directories(${ProjectName}Static PRIVATE "${IL_INCLUDE_DIR}" ${AvsCore_SOURCE_DIR})
+  else()
+      target_link_libraries(${ProjectName}Static "${FSLIB}" "ssp")
+  endif()
+
+  INSTALL(TARGETS "${ProjectName}Static"
+        DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/Shibatch/CMakeLists.txt patched/plugins/Shibatch/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/Shibatch/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/plugins/Shibatch/CMakeLists.txt	2025-03-26 17:03:48.377135543 +0100
@@ -46,3 +46,18 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName} STATIC ${SourceFiles})
+  set_target_properties(${ProjectName} PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static "PFC")
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR})
+
+
+  INSTALL(TARGETS "${ProjectName}Static"
+          DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/TimeStretch/CMakeLists.txt patched/plugins/TimeStretch/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/TimeStretch/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/plugins/TimeStretch/CMakeLists.txt	2025-03-26 17:11:56.233111126 +0100
@@ -36,3 +36,17 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static  ${SOUNDTOUCH_LINK_LIBRARIES})
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${CMAKE_CURRENT_SOURCE_DIR} ${AvsCore_SOURCE_DIR} ${SOUNDTOUCH_INCLUDE_DIRS})
+
+  INSTALL(TARGETS "${ProjectName}Static"
+          DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/VDubFilter/CMakeLists.txt patched/plugins/VDubFilter/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/VDubFilter/CMakeLists.txt	2025-03-26 16:56:09.569015009 +0100
+++ patched/plugins/VDubFilter/CMakeLists.txt	2025-03-26 17:02:35.015552734 +0100
@@ -38,3 +38,17 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static "user32" "ssp")
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR})
+
+  INSTALL(TARGETS "${ProjectName}Static"
+          DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
diff -ruN AviSynthPlus-3.7.4/plugins/VFAPIFilter/CMakeLists.txt patched/plugins/VFAPIFilter/CMakeLists.txt
--- AviSynthPlus-3.7.4/plugins/VFAPIFilter/CMakeLists.txt	2025-03-24 02:26:29.000000000 +0100
+++ patched/plugins/VFAPIFilter/CMakeLists.txt	2025-03-26 17:01:56.127424325 +0100
@@ -37,3 +37,17 @@
 
 INSTALL(TARGETS "${ProjectName}"
         DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+
+if (BUILD_STATIC_LIB)
+  add_library(${ProjectName}Static STATIC ${SourceFiles})
+  set_target_properties(${ProjectName}Static PROPERTIES "OUTPUT_NAME" ${PluginName})
+
+  # Library dependencies
+  target_link_libraries(${ProjectName}Static "User32.lib")
+  
+  # Include directories
+  target_include_directories(${ProjectName}Static PRIVATE ${AvsCore_SOURCE_DIR} ${DSHOWSRC_BASECLASSES_PATH} ${DSHOWSRC_DX_INCLUDE_PATH})
+
+  INSTALL(TARGETS "${ProjectName}Static"
+          DESTINATION "${CORE_PLUGIN_INSTALL_PATH}/avisynth")
+endif()
