diff -aur a/plugins/dfxvideo/soft.c b/plugins/dfxvideo/soft.c
--- a/plugins/dfxvideo/soft.c	2020-02-28 18:21:22.796527201 +0000
+++ b/plugins/dfxvideo/soft.c	2020-02-28 18:49:28.832669798 +0000
@@ -235,7 +235,7 @@
 /////////////////////////////////////////////////////////////////
 /////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol_Dither(unsigned short * pdest, int32_t m1, int32_t m2, int32_t m3)
+static __inline void GetShadeTransCol_Dither(unsigned short * pdest, int32_t m1, int32_t m2, int32_t m3)
 {
  int32_t r,g,b;
 
@@ -299,7 +299,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol(unsigned short * pdest,unsigned short color)
+static __inline void GetShadeTransCol(unsigned short * pdest,unsigned short color)
 {
  if(bCheckMask && (*pdest & HOST2LE16(0x8000))) return;
 
@@ -358,7 +358,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol32(uint32_t * pdest,uint32_t color)
+static __inline void GetShadeTransCol32(uint32_t * pdest,uint32_t color)
 {
  if(DrawSemiTrans)
   {
@@ -1200,11 +1200,11 @@
 // NASM version (external):
 #define shl10idiv i386_shl10idiv
 
-__inline int shl10idiv(int x, int y);
+static __inline int shl10idiv(int x, int y);
 
 #else
 
-__inline int shl10idiv(int x, int y)
+static __inline int shl10idiv(int x, int y)
 {
  __int64 bi=x;
  bi<<=10;
@@ -1217,7 +1217,7 @@
 
 // GNUC long long int version:
 
-__inline int shl10idiv(int x, int y) 
+static __inline int shl10idiv(int x, int y) 
 { 
  long long int bi=x; 
  bi<<=10; 
@@ -1230,7 +1230,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
                         
-__inline int RightSection_F(void)
+static __inline int RightSection_F(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1246,7 +1246,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_F(void)
+static __inline int LeftSection_F(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1262,7 +1262,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_F(void)
+static __inline BOOL NextRow_F(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1288,7 +1288,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_F(short x1, short y1, short x2, short y2, short x3, short y3)
+static __inline BOOL SetupSections_F(short x1, short y1, short x2, short y2, short x3, short y3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest;
@@ -1350,7 +1350,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_G(void)
+static __inline int RightSection_G(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1366,7 +1366,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_G(void)
+static __inline int LeftSection_G(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1389,7 +1389,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_G(void)
+static __inline BOOL NextRow_G(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1418,7 +1418,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_G(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb1, int32_t rgb2, int32_t rgb3)
+static __inline BOOL SetupSections_G(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb1, int32_t rgb2, int32_t rgb3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest,temp;
@@ -1496,7 +1496,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_FT(void)
+static __inline int RightSection_FT(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1512,7 +1512,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_FT(void)
+static __inline int LeftSection_FT(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1533,7 +1533,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_FT(void)
+static __inline BOOL NextRow_FT(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1561,7 +1561,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_FT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3)
+static __inline BOOL SetupSections_FT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest,temp;
@@ -1649,7 +1649,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_GT(void)
+static __inline int RightSection_GT(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1665,7 +1665,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_GT(void)
+static __inline int LeftSection_GT(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1693,7 +1693,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_GT(void)
+static __inline BOOL NextRow_GT(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1724,7 +1724,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_GT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, int32_t rgb1, int32_t rgb2, int32_t rgb3)
+static __inline BOOL SetupSections_GT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, int32_t rgb1, int32_t rgb2, int32_t rgb3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest,temp;
@@ -1829,7 +1829,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_F4(void)
+static __inline int RightSection_F4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1848,7 +1848,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_F4(void)
+static __inline int LeftSection_F4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1867,7 +1867,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_F4(void)
+static __inline BOOL NextRow_F4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1899,7 +1899,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_F4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4)
+static __inline BOOL SetupSections_F4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -2041,7 +2041,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_FT4(void)
+static __inline int RightSection_FT4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -2064,7 +2064,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_FT4(void)
+static __inline int LeftSection_FT4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -2087,7 +2087,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_FT4(void)
+static __inline BOOL NextRow_FT4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -2123,7 +2123,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_FT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4)
+static __inline BOOL SetupSections_FT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -2272,7 +2272,7 @@
 ////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////
 
-__inline int RightSection_GT4(void)
+static __inline int RightSection_GT4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -2302,7 +2302,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline int LeftSection_GT4(void)
+static __inline int LeftSection_GT4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -2332,7 +2332,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL NextRow_GT4(void)
+static __inline BOOL NextRow_GT4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -2374,7 +2374,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline BOOL SetupSections_GT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4,int32_t rgb1,int32_t rgb2,int32_t rgb3,int32_t rgb4)
+static __inline BOOL SetupSections_GT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4,int32_t rgb1,int32_t rgb2,int32_t rgb3,int32_t rgb4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -2544,7 +2544,7 @@
 // POLY 3/4 FLAT SHADED
 ////////////////////////////////////////////////////////////////////////
 
-__inline void drawPoly3Fi(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb)
+static __inline void drawPoly3Fi(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb)
 {
  int i,j,xmin,xmax,ymin,ymax;
  unsigned short color;uint32_t lcolor;
@@ -5182,7 +5182,7 @@
 // POLY 3/4 G-SHADED
 ////////////////////////////////////////////////////////////////////////
  
-__inline void drawPoly3Gi(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb1, int32_t rgb2, int32_t rgb3)
+static __inline void drawPoly3Gi(short x1,short y1,short x2,short y2,short x3,short y3,int32_t rgb1, int32_t rgb2, int32_t rgb3)
 {
  int i,j,xmin,xmax,ymin,ymax;
  int32_t cR1,cG1,cB1;
diff -aur a/plugins/peopsxgl/soft.c b/plugins/peopsxgl/soft.c
--- a/plugins/peopsxgl/soft.c	2020-02-28 18:21:22.796527201 +0000
+++ b/plugins/peopsxgl/soft.c	2020-02-28 18:49:04.812975453 +0000
@@ -232,7 +232,7 @@
 
 /////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol_Dither(unsigned short *pdest, int m1, int m2, int m3)
+static __inline void GetShadeTransCol_Dither(unsigned short *pdest, int m1, int m2, int m3)
 {
  int r,g,b;
 
@@ -296,7 +296,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol(unsigned short * pdest,unsigned short color)
+static __inline void GetShadeTransCol(unsigned short * pdest,unsigned short color)
 {
  if(bCheckMask && *pdest&0x8000) return;
 
@@ -355,7 +355,7 @@
 
 ////////////////////////////////////////////////////////////////////////
 
-__inline void GetShadeTransCol32(uint32_t *pdest, uint32_t color)
+static __inline void GetShadeTransCol32(uint32_t *pdest, uint32_t color)
 {
  if (DrawSemiTrans)
   {
@@ -1188,14 +1188,14 @@
 static int left_G, delta_left_G, right_G, delta_right_G;
 static int left_B, delta_left_B, right_B, delta_right_B;
 
-__inline int shl10idiv(int x, int y)
+static __inline int shl10idiv(int x, int y)
 {
  long long int bi=x;
  bi<<=10;
  return bi/y;
 }
 
-__inline int RightSection_F(void)
+static __inline int RightSection_F(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1209,7 +1209,7 @@
  return height;
 }
 
-__inline int LeftSection_F(void)
+static __inline int LeftSection_F(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1223,7 +1223,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_F(void)
+static __inline BOOL NextRow_F(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1247,7 +1247,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_F(short x1, short y1, short x2, short y2, short x3, short y3)
+static __inline BOOL SetupSections_F(short x1, short y1, short x2, short y2, short x3, short y3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest;
@@ -1306,7 +1306,7 @@
  return TRUE;
 }
 
-__inline int RightSection_G(void)
+static __inline int RightSection_G(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1320,7 +1320,7 @@
  return height;
 }
 
-__inline int LeftSection_G(void)
+static __inline int LeftSection_G(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1341,7 +1341,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_G(void)
+static __inline BOOL NextRow_G(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1368,7 +1368,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_G(short x1,short y1,short x2,short y2,short x3,short y3, int rgb1, int rgb2, int rgb3){
+static __inline BOOL SetupSections_G(short x1,short y1,short x2,short y2,short x3,short y3, int rgb1, int rgb2, int rgb3){
  soft_vertex *v1, *v2, *v3;
  int height, longest, temp;
 
@@ -1442,7 +1442,7 @@
  return TRUE;
 }
 
-__inline int RightSection_FT(void)
+static __inline int RightSection_FT(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1456,7 +1456,7 @@
  return height;
 }
 
-__inline int LeftSection_FT(void)
+static __inline int LeftSection_FT(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1475,7 +1475,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_FT(void)
+static __inline BOOL NextRow_FT(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1501,7 +1501,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_FT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3)
+static __inline BOOL SetupSections_FT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest,temp;
@@ -1586,7 +1586,7 @@
  return TRUE;
 }
 
-__inline int RightSection_GT(void)
+static __inline int RightSection_GT(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1600,7 +1600,7 @@
  return height;
 }
 
-__inline int LeftSection_GT(void)
+static __inline int LeftSection_GT(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1626,7 +1626,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_GT(void)
+static __inline BOOL NextRow_GT(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1655,7 +1655,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_GT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, int rgb1, int rgb2, int rgb3)
+static __inline BOOL SetupSections_GT(short x1, short y1, short x2, short y2, short x3, short y3, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, int rgb1, int rgb2, int rgb3)
 {
  soft_vertex * v1, * v2, * v3;
  int height,longest,temp;
@@ -1757,7 +1757,7 @@
  return TRUE;
 }
 
-__inline int RightSection_F4(void)
+static __inline int RightSection_F4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1774,7 +1774,7 @@
  return height;
 }
 
-__inline int LeftSection_F4(void)
+static __inline int LeftSection_F4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -1791,7 +1791,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_F4(void)
+static __inline BOOL NextRow_F4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -1821,7 +1821,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_F4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4)
+static __inline BOOL SetupSections_F4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -1960,7 +1960,7 @@
  return TRUE;
 }
 
-__inline int RightSection_FT4(void)
+static __inline int RightSection_FT4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -1981,7 +1981,7 @@
  return height;
 }
 
-__inline int LeftSection_FT4(void)
+static __inline int LeftSection_FT4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -2002,7 +2002,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_FT4(void)
+static __inline BOOL NextRow_FT4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -2036,7 +2036,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_FT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4)
+static __inline BOOL SetupSections_FT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -2182,7 +2182,7 @@
  return TRUE;
 }
 
-__inline int RightSection_GT4(void)
+static __inline int RightSection_GT4(void)
 {
  soft_vertex * v1 = right_array[ right_section ];
  soft_vertex * v2 = right_array[ right_section-1 ];
@@ -2210,7 +2210,7 @@
  return height;
 }
 
-__inline int LeftSection_GT4(void)
+static __inline int LeftSection_GT4(void)
 {
  soft_vertex * v1 = left_array[ left_section ];
  soft_vertex * v2 = left_array[ left_section-1 ];
@@ -2238,7 +2238,7 @@
  return height;  
 }
 
-__inline BOOL NextRow_GT4(void)
+static __inline BOOL NextRow_GT4(void)
 {
  if(--left_section_height<=0) 
   {
@@ -2278,7 +2278,7 @@
  return FALSE;
 }
 
-__inline BOOL SetupSections_GT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4, int rgb1, int rgb2, int rgb3, int rgb4)
+static __inline BOOL SetupSections_GT4(short x1, short y1, short x2, short y2, short x3, short y3, short x4, short y4, short tx1, short ty1, short tx2, short ty2, short tx3, short ty3, short tx4, short ty4, int rgb1, int rgb2, int rgb3, int rgb4)
 {
  soft_vertex * v1, * v2, * v3, * v4;
  int height,width,longest1,longest2;
@@ -2444,7 +2444,7 @@
 // POLY 3/4 FLAT SHADED
 ////////////////////////////////////////////////////////////////////////
 
-__inline void drawPoly3Fi(short x1, short y1, short x2, short y2, short x3, short y3, int rgb)
+static __inline void drawPoly3Fi(short x1, short y1, short x2, short y2, short x3, short y3, int rgb)
 {
  int i,j,xmin,xmax,ymin,ymax;
  unsigned short color;
@@ -5083,7 +5083,7 @@
 // POLY 3/4 G-SHADED
 ////////////////////////////////////////////////////////////////////////
  
-__inline void drawPoly3Gi(short x1,short y1,short x2,short y2,short x3,short y3, int rgb1, int rgb2, int rgb3)
+static __inline void drawPoly3Gi(short x1,short y1,short x2,short y2,short x3,short y3, int rgb1, int rgb2, int rgb3)
 {
  int i,j,xmin,xmax,ymin,ymax;
  int cR1,cG1,cB1;
