diff --unified --recursive --text zenoh-c-1.7.2.orig/build-resources/opaque-types/Cargo.toml zenoh-c-1.7.2.new/build-resources/opaque-types/Cargo.toml
--- zenoh-c-1.7.2.orig/build-resources/opaque-types/Cargo.toml	2026-01-08 19:24:13.000000000 +0100
+++ zenoh-c-1.7.2.new/build-resources/opaque-types/Cargo.toml	2026-02-15 20:30:47.295744602 +0100
@@ -38,3 +38,6 @@
 const_format = "0.2.32"
 flume = "*"
 tokio = "*"
+
+[patch.crates-io]
+static_init = { git = "https://gitlab.com/okannen/static_init.git", rev = "a15b9fc6eeed143f9bf91cb26a8c7bd5f6f82914" }
diff --unified --recursive --text zenoh-c-1.7.2.orig/Cargo.toml zenoh-c-1.7.2.new/Cargo.toml
--- zenoh-c-1.7.2.orig/Cargo.toml	2026-01-08 19:24:13.000000000 +0100
+++ zenoh-c-1.7.2.new/Cargo.toml	2026-02-15 20:31:12.582467048 +0100
@@ -125,3 +125,6 @@
 codegen-units = 1
 opt-level = 3
 panic = "abort"
+
+[patch.crates-io]
+static_init = { git = "https://gitlab.com/okannen/static_init.git", rev = "a15b9fc6eeed143f9bf91cb26a8c7bd5f6f82914" }
diff --unified --recursive --text zenoh-c-1.7.2.orig/Cargo.toml.in zenoh-c-1.7.2.new/Cargo.toml.in
--- zenoh-c-1.7.2.orig/Cargo.toml.in	2026-01-08 19:24:13.000000000 +0100
+++ zenoh-c-1.7.2.new/Cargo.toml.in	2026-02-15 20:30:58.702436437 +0100
@@ -125,3 +125,6 @@
 codegen-units = 1
 opt-level = 3
 panic = "abort"
+
+[patch.crates-io]
+static_init = { git = "https://gitlab.com/okannen/static_init.git", rev = "a15b9fc6eeed143f9bf91cb26a8c7bd5f6f82914" }
