--- Makefile.orig	2014-04-02 20:13:16.000000000 +0900
+++ Makefile	2014-04-22 18:46:03.625971898 +0900
@@ -4,7 +4,7 @@
 include $(OSL_HOME)/makefile.conf
 
 OSL_HOME_FLAGS = -DOSL_HOME=\"$(shell dirname $(shell dirname `pwd`))\"
-CXXFLAGS += $(OSL_HOME_FLAGS)
+CXXFLAGS += $(OSL_HOME_FLAGS) -fPIC -pipe
 INCLUDES += -I.. -I../../core
 
 HASH_SRC = \
@@ -46,7 +46,7 @@
 OBJS = $(patsubst %.cc,%.o,$(SRCS))
 DEPS = $(patsubst %.cc,.deps/%.cc.d,$(SRCS))
 
-all: libosl_std.a
+all: libosl_std.so.1.0
 
 light-clean:
 	-rm -rf .deps
@@ -56,3 +56,6 @@
 
 -include $(DEPS)
 libosl_std.a : $(OBJS) ../../core/osl/libosl_core.a
+
+libosl_std.so.1.0: $(OBJS)
+	$(CXX) -shared -Wl,-soname=libosl_std.so.1 $(OBJS) -o $@
