Index: SRC/element/shell/ShellMITC4.cpp
===================================================================
--- SRC/element/shell/ShellMITC4.cpp	(revision 6236)
+++ SRC/element/shell/ShellMITC4.cpp	(working copy)
@@ -1126,7 +1126,7 @@
   resid.Zero( ) ;
   
 //start Yuli Huang (yulihuang@gmail.com) & Xinzheng Lu (luxz@tsinghua.edu.cn)
-  updateBasis( );
+//  updateBasis( );
 //end Yuli Huang (yulihuang@gmail.com) & Xinzheng Lu (luxz@tsinghua.edu.cn)
 
   double dx34 = xl[0][2]-xl[0][3];
Index: SRC/material/uniaxial/STEELDR.f
===================================================================
--- SRC/material/uniaxial/STEELDR.f	(revision 6236)
+++ SRC/material/uniaxial/STEELDR.f	(working copy)
@@ -288,6 +288,8 @@
 					if(ep_s.le.ep_sushift(k)) then
 					fp_s=s1*fp_su
 					fps_s=fps_su
+					ep_sushift(k)=ep_s
+					ep_o(k)=ep_sushift(k)+ep_su
 					else
 	    call Bauschinger(1,icheat,P_major,P_minor,region,
      *  ep_sushift,s1,fp_su,fps_su,Ep_u,f_y,ep_a,fp_a,
