--- src/OpenFOAM-3.0.1/src/surfMesh/surfaceFormats/stl/STLsurfaceFormatASCII.L
+++ src/OpenFOAM-3.0.1/src/surfMesh/surfaceFormats/stl/STLsurfaceFormatASCII.L
@@ -50,7 +50,7 @@
 // It is called by yylex but is not used as the mechanism to change file.
 // See <<EOF>>
 //! \cond dummy
-#if YY_FLEX_SUBMINOR_VERSION < 34
+#if YY_FLEX_SUBMINOR_VERSION < 34 && YY_FLEX_MINOR_VERSION < 6
 extern "C" int yywrap()
 #else
 int yyFlexLexer::yywrap()
