diff -Naur gmsh-2.12.0-source.orig/Geo/GModelIO_MED.cpp gmsh-2.12.0-source/Geo/GModelIO_MED.cpp
--- gmsh-2.12.0-source.orig/Geo/GModelIO_MED.cpp	2016-01-06 13:59:42.000000000 +0100
+++ gmsh-2.12.0-source/Geo/GModelIO_MED.cpp	2016-04-30 20:50:20.965186247 +0200
@@ -522,7 +522,8 @@
     return 0;
   }
 
-  char *meshName = (char*)getName().c_str();
+  std::string strMeshName = getName();
+  char * meshName = (char*)strMeshName.c_str();
 
   // Gmsh always writes 3D unstructured meshes
 #if (MED_MAJOR_NUM == 3)
