diff --git a/include/install-vitasdk.sh b/include/install-vitasdk.sh
index d0f4a6f..55fdc01 100755
--- a/include/install-vitasdk.sh
+++ b/include/install-vitasdk.sh
@@ -19,8 +19,7 @@ install_vitasdk () {
       fi
       command -v curl || { echo "curl missing (install using: apt install curl)" ; exit 1; }
       if [ ! -d "$INSTALLDIR" ]; then
-        sudo mkdir -p $INSTALLDIR
-        sudo chown $USER:$(id -gn $USER) $INSTALLDIR
+        mkdir -p $INSTALLDIR
       fi
       wget -O- "$(get_download_link master linux)" | tar xj -C $INSTALLDIR --strip-components=1
      ;;
