Index: setup.py
===================================================================
--- setup.py	(revision 875)
+++ setup.py	(working copy)
@@ -426,7 +426,7 @@
             ext_list[-1].library_dirs = [self.libdirs]
             
             if self.SUNDIALS_version >= (3,0,0):
-                ext_list[-1].libraries = ["sundials_cvodes", "sundials_nvecserial", "sundials_idas", "sundials_sunlinsoldense", "sundials_sunlinsolspgmr", "sundials_sunmatrixdense", "sundials_sunmatrixsparse"]
+                ext_list[-1].libraries = ["sundials_cvodes", "sundials_nvecserial", "sundials_idas", "sundials_sunlinsoldense", "sundials_sunlinsolspgmr", "sundials_sunmatrixdense", "sundials_sunmatrixsparse", "sundials_kinsol"]
             else:
                 ext_list[-1].libraries = ["sundials_cvodes", "sundials_nvecserial", "sundials_idas"]
             if self.sundials_with_superlu and self.with_SLU: #If SUNDIALS is compiled with support for SuperLU
@@ -444,7 +444,7 @@
                         compile_time_env=compile_time_env, force=True)
             ext_list[-1].include_dirs = [np.get_include(), "assimulo","assimulo"+os.sep+"lib", self.incdirs]
             ext_list[-1].library_dirs = [self.libdirs]
-            ext_list[-1].libraries = ["sundials_kinsol", "sundials_nvecserial"]
+            ext_list[-1].libraries = ["sundials_kinsol", "sundials_nvecserial", "sundials_cvodes", "sundials_idas"]
             
             if self.sundials_with_superlu and self.with_SLU: #If SUNDIALS is compiled with support for SuperLU
                 ext_list[-1].include_dirs.append(self.SLUincdir)
Index: src/lib/sundials_includes.pxd
===================================================================
--- src/lib/sundials_includes.pxd	(revision 875)
+++ src/lib/sundials_includes.pxd	(working copy)
@@ -237,7 +237,7 @@
 
 
 cdef extern from "cvodes/cvodes.h":
-    void* CVodeCreate(int lmm, int iter)
+    void* CVodeCreate(int lmm)
     ctypedef int (*CVRhsFn)(realtype t, N_Vector y, N_Vector ydot, void *f_data)
     int CVodeInit(void *cvode_mem, CVRhsFn f, realtype t0, N_Vector y0)
     int CVodeReInit(void *cvode_mem, realtype t0, N_Vector y0)
Index: src/solvers/sundials.pyx
===================================================================
--- src/solvers/sundials.pyx	(revision 875)
+++ src/solvers/sundials.pyx	(working copy)
@@ -1699,7 +1699,7 @@
         if self.cvode_mem == NULL: #The solver is not initialized
             
             #Create the solver
-            self.cvode_mem= SUNDIALS.CVodeCreate(CV_BDF if self.options["discr"] == "BDF" else CV_ADAMS, CV_NEWTON if self.options["iter"] == "Newton" else CV_FUNCTIONAL)
+            self.cvode_mem= SUNDIALS.CVodeCreate(CV_BDF if self.options["discr"] == "BDF" else CV_ADAMS)
             if self.cvode_mem == NULL:
                 raise CVodeError(CV_MEM_FAIL)
             
