diff -ur a/include/linux/swap.h b/include/linux/swap.h
--- a/include/linux/swap.h	2016-07-12 20:41:53.000000000 +0300
+++ b/include/linux/swap.h	2016-07-12 20:44:44.801050945 +0300
@@ -416,14 +416,18 @@
 extern int free_swap_and_cache(swp_entry_t);
 extern int swap_type_of(dev_t, sector_t, struct block_device **);
 extern unsigned int count_swap_pages(int, int);
+extern sector_t map_swap_entry(swp_entry_t entry, struct block_device **);
 extern sector_t map_swap_page(struct page *, struct block_device **);
 extern sector_t swapdev_block(int, pgoff_t);
+extern struct swap_info_struct *get_swap_info_struct(unsigned);
 extern int page_swapcount(struct page *);
 extern int swp_swapcount(swp_entry_t entry);
 extern struct swap_info_struct *page_swap_info(struct page *);
 extern bool reuse_swap_page(struct page *, int *);
 extern int try_to_free_swap(struct page *);
 struct backing_dev_info;
+extern void get_swap_range_of_type(int type, swp_entry_t *start,
+               swp_entry_t *end, unsigned int limit);
 
 #else /* CONFIG_SWAP */
 
