diff -rupN a/docs/CMakeLists.txt b/docs/CMakeLists.txt
--- a/docs/CMakeLists.txt	2016-02-11 13:07:07.451027973 -0800
+++ b/docs/CMakeLists.txt	2016-02-11 13:06:30.881026867 -0800
@@ -14,7 +14,7 @@ SET(SPHINX_ARGS
   -D latex_paper_size=${SWIFT_SPHINX_PAPER_SIZE}
   -d ${CMAKE_BINARY_DIR}/doctrees)
 
-if(SPHINX_EXECUTABLE)
+if(FALSE)
   add_custom_target(docs_html ALL
     ${SPHINX_EXECUTABLE} ${SPHINX_ARGS} -b html
       . ${CMAKE_BINARY_DIR}/docs/html
