diff -ru syasokoban-2.0.1.orig/src/Main.cpp syasokoban-2.0.1/src/Main.cpp
--- syasokoban-2.0.1.orig/src/Main.cpp	2007-07-21 22:22:10.000000000 +0200
+++ syasokoban-2.0.1/src/Main.cpp	2016-03-16 09:40:33.094285986 +0100
@@ -10,6 +10,7 @@
 #include "GameRegistry.h"
 #include "Games.h"
 #include <stdlib.h>
+#include <unistd.h>
 
 bool g_bMustRedraw = true;
 
diff -ru syasokoban-2.0.1.orig/src/djSettings.cpp syasokoban-2.0.1/src/djSettings.cpp
--- syasokoban-2.0.1.orig/src/djSettings.cpp	2007-07-21 22:22:10.000000000 +0200
+++ syasokoban-2.0.1/src/djSettings.cpp	2016-03-16 09:49:44.686833463 +0100
@@ -6,7 +6,9 @@
 //---------------------------------------------------------------------------
 #include "djSettings.h"
 #include "djUtils.h"
+#include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 char g_szConfigFile[2048] = { 0 };
 
diff -ru syasokoban-2.0.1.orig/src/djDesktop.cpp syasokoban-2.0.1/src/djDesktop.cpp
--- syasokoban-2.0.1.orig/src/djDesktop.cpp	2007-07-21 22:22:10.000000000 +0200
+++ syasokoban-2.0.1/src/djDesktop.cpp	2016-03-16 10:17:44.332718779 +0100
@@ -15,7 +15,7 @@
 
 std::string GetDataPath()
 {
-	return std::string("data/");
+	return std::string("/usr/share/syasokoban/");
 }
 
 CdjDesktop::CdjDesktop()
