diff --git a/PKGBUILD b/PKGBUILD
index 4baa079..f767e54 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ pkgdesc="Console-based editor designed for simplified use like gedit"
 arch=("x86_64")
 url="https://github.com/istoph/editor"
 license=('Boost')
-depends=('tuiwidgets-git' 'qt5-base' 'icu' 'gcc-libs')
+depends=('tuiwidgets-git' 'qt5-base' 'icu' 'gcc-libs' 'syntax-highlighting')
 makedepends=(meson git)
 provides=(chr-editor)
 conflicts=(chr-editor)
@@ -27,7 +27,7 @@ prepare() {
 }
 
 build() {
-    arch-meson editor _build
+    arch-meson -D syntax_highlighting=true editor _build
     meson compile -C _build
 }
 
