From 0c3cae194b43c4f9278e2c20b36cc59c5b4a36ba Mon Sep 17 00:00:00 2001
From: Alexander Epaneshnikov <aarnaarn2@gmail.com>
Date: Fri, 26 Feb 2021 22:08:15 +0300
Subject: [PATCH] Use systemd to pass a credential

---
 main.py | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/main.py b/main.py
index 5d3cd26..b91eb3c 100644
--- a/main.py
+++ b/main.py
@@ -6,17 +6,22 @@
 import sys
 import time
 import urllib
-from os import getenv
+from os import path, getenv
 import logging
 import requests
 from telebot import TeleBot, types, apihelper
 
+if getenv("CREDENTIALS_DIRECTORY"):
+    print("using bot token provided by init")
+    with open(path.join(getenv("CREDENTIALS_DIRECTORY"), "bot-token"), "r") as token:
+        Bot_TOKEN = token.readline()
+else:
+    Bot_TOKEN = getenv("arch_man_bot_token")
 
-Bot_TOKEN = getenv("arch_man_bot_token")
 if Bot_TOKEN is None:
     sys.exit("You must set <arch_man_bot_token> environment variable!")
 
-bot = TeleBot(Bot_TOKEN)
+bot = TeleBot(Bot_TOKEN.rstrip("\n"))
 shouldBotLog = True  # if False bot will not create and keep log.log file
 
 
--
2.30.1
