--- cpp/CMakeLists.txt	2026-02-05 05:49:40.000000000 +0800
+++ cpp/CMakeLists.txt.new	2026-03-11 22:22:53.815616471 +0800
@@ -72,7 +72,7 @@
 set(CUGRAPH_CUDA_FLAGS "")
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-    list(APPEND CUGRAPH_CXX_FLAGS -Werror -Wno-error=deprecated-declarations -Wno-deprecated-declarations -DRAFT_HIDE_DEPRECATION_WARNINGS)
+    list(APPEND CUGRAPH_CXX_FLAGS -Wno-error=deprecated-declarations -Wno-deprecated-declarations -DRAFT_HIDE_DEPRECATION_WARNINGS)
 endif(CMAKE_COMPILER_IS_GNUCXX)
 
 
@@ -117,12 +117,11 @@
 
 # Need CCCL, then rmm, then cuCollections, then RAFT.
 # This ensures that libraries can be overridden for testing.
-include(cmake/thirdparty/get_cccl.cmake)
-include(${rapids-cmake-dir}/cpm/rmm.cmake)
-rapids_cpm_rmm(BUILD_EXPORT_SET cugraph-exports)
-include(${rapids-cmake-dir}/cpm/cuco.cmake)
-rapids_cpm_cuco(BUILD_EXPORT_SET cugraph-exports)
-include(cmake/thirdparty/get_raft.cmake)
+find_package(rapids_logger REQUIRED)
+find_package(nvtx3 REQUIRED)
+find_package(rmm REQUIRED)
+find_package(raft REQUIRED)
+find_package(cuco REQUIRED)
 
 # Always include cuVS for spectral clustering enhancements
 include(cmake/thirdparty/get_cuvs.cmake)
