--- cpp/CMakeLists.txt	2026-06-04 03:08:02.000000000 +0800
+++ cpp/CMakeLists.txt.new	2026-06-05 11:01:38.669979593 +0800
@@ -102,11 +102,13 @@
   message(DEPRECATION "UCXX_ENABLE_RMM is deprecated and will be removed in a future release. "
                       "Use UCXX_ENABLE_CCCL instead."
   )
-  include(cmake/thirdparty/get_rmm.cmake)
+  find_package(rapids_logger REQUIRED)
+  find_package(nvtx3 REQUIRED)
+  find_package(rmm REQUIRED)
 endif()
 # find cccl
 if(UCXX_ENABLE_CCCL)
-  include(cmake/thirdparty/get_cccl.cmake)
+  find_package(CCCL REQUIRED)
 endif()
 # find or install GoogleTest
 if(BUILD_TESTS)
