--- cpp/CMakeLists.txt.orig	2026-08-06 12:27:31.731962762 +0800
+++ cpp/CMakeLists.txt	2026-08-06 12:29:51.506953141 +0800
@@ -188,8 +188,7 @@
 rapids_cpm_init()
 rapids_cmake_install_lib_dir(lib_dir)
 
-include(${rapids-cmake-dir}/cpm/rapids_logger.cmake)
-rapids_cpm_rapids_logger(BUILD_EXPORT_SET nvforest-exports INSTALL_EXPORT_SET nvforest-exports)
+find_package(rapids_logger REQUIRED)
 create_logger_macros(NVFOREST "nvforest::default_logger()" include/nvforest/common)
 
 if(BUILD_NVFOREST_TESTS)
@@ -197,12 +196,13 @@
 endif()
 
 # CCCL before RMM, and RMM before RAFT
-include(cmake/thirdparty/get_cccl.cmake)
+find_package(CCCL REQUIRED)
 if(NVFOREST_ENABLE_GPU)
-  include(cmake/thirdparty/get_rmm.cmake)
-  include(cmake/thirdparty/get_raft.cmake)
+  find_package(nvtx3 REQUIRED)
+  find_package(rmm REQUIRED)
+  find_package(raft REQUIRED)
 endif()
-include(cmake/thirdparty/get_treelite.cmake)
+find_package(Treelite REQUIRED)
 
 if(BUILD_NVFOREST_TESTS)
   include(${rapids-cmake-dir}/cpm/gtest.cmake)
