diff --git a/setup.py b/setup.py
index c17d552..0d20456 100644
--- a/setup.py
+++ b/setup.py
@@ -35,14 +35,7 @@ class libuv_build_ext(build_ext):
         subprocess.run(['make', j_flag], cwd=LIBUV_DIR, env=env, check=True)
 
     def build_extensions(self):
-        libuv_lib = os.path.join(LIBUV_DIR, '.libs', 'libuv.a')
-        if not os.path.exists(libuv_lib):
-            self.build_libuv()
-        if not os.path.exists(libuv_lib):
-            raise RuntimeError('failed to build libuv')
-
-        self.extensions[-1].extra_objects.extend([libuv_lib])
-        self.compiler.add_include_dir(os.path.join(LIBUV_DIR, 'include'))
+        self.compiler.add_library('uv')
 
         if sys.platform.startswith('linux'):
             self.compiler.add_library('rt')
