--- cpp/CMakeLists.txt.orig	2026-08-06 12:02:56.547489249 +0800
+++ cpp/CMakeLists.txt	2026-08-06 12:04:37.883340326 +0800
@@ -102,11 +102,13 @@
        AND UCXX_BENCHMARKS_ENABLE_RMM
       )
 )
-  include(cmake/thirdparty/get_rmm.cmake)
+  find_package(rapids_logger REQUIRED)
+  find_package(nvtx3 REQUIRED)
+  find_package(rmm REQUIRED)
 endif()
 # find cccl
 if(UCXX_ENABLE_CCCL)
-  include(cmake/thirdparty/get_cccl.cmake)
+  find_package(CCCL REQUIRED)
 endif()
 # find or install GoogleTest
 if(BUILD_TESTS)
