diff --git a/radicle-httpd/systemd/radicle-httpd.service b/radicle-httpd/systemd/radicle-httpd.service
deleted file mode 100644
index d6266207..00000000
--- a/radicle-httpd/systemd/radicle-httpd.service
+++ /dev/null
@@ -1,23 +0,0 @@
-# Example systemd unit file for `radicle-httpd`.
-#
-# When running radicle-httpd on a server, it should be run as a separate user.
-#
-# Copy this file into /etc/systemd/system and set the User/Group parameters
-# under [Service] appropriately, as well as the `RAD_HOME` environment variable.
-#
-[Unit]
-Description=Radicle HTTP Daemon
-After=network.target network-online.target
-Requires=network-online.target
-
-[Service]
-User=seed
-Group=seed
-ExecStart=/usr/local/bin/radicle-httpd --listen 127.0.0.1:8080
-Environment=RAD_HOME=/home/seed/.radicle RUST_BACKTRACE=1 RUST_LOG=info
-KillMode=process
-Restart=always
-RestartSec=1
-
-[Install]
-WantedBy=multi-user.target
diff --git a/radicle-httpd/systemd/system/radicle-httpd.service b/radicle-httpd/systemd/system/radicle-httpd.service
new file mode 100644
index 00000000..e073db01
--- /dev/null
+++ b/radicle-httpd/systemd/system/radicle-httpd.service
@@ -0,0 +1,35 @@
+# Example systemd system service unit file for `radicle-httpd`.
+#
+# When running radicle-httpd for just one user, e.g. for a private
+# radicle-explorer deployment, it may be run as a user service, see
+# `../user/*.service`.
+#
+[Unit]
+Description=Radicle HTTP Daemon
+After=network.target network-online.target
+Wants=network-online.target
+
+[Service]
+ExecStart=/usr/bin/radicle-httpd $RADICLE_HTTPD_ARGS
+KillMode=mixed
+Restart=always
+RestartSec=3
+
+EnvironmentFile=-%E/conf.d/radicle-httpd
+Environment=RADICLE_HTTPD_ARGS=--listen 127.0.0.1:7235
+Environment=RAD_HOME=%S/radicle
+Environment=RUST_BACKTRACE=1
+Environment=RUST_LOG=info
+
+# Basic hardening options. For more, please refer to `systemd-analyze security`.
+PrivateTmp=true
+ProtectSystem=strict
+ProtectHome=true
+NoNewPrivileges=true
+MemoryDenyWriteExecute=true
+
+User=radicle
+Group=radicle
+
+[Install]
+WantedBy=multi-user.target
diff --git a/radicle-httpd/systemd/user/radicle-httpd.service b/radicle-httpd/systemd/user/radicle-httpd.service
new file mode 100644
index 00000000..358e7566
--- /dev/null
+++ b/radicle-httpd/systemd/user/radicle-httpd.service
@@ -0,0 +1,29 @@
+# Example systemd user service unit file for `radicle-httpd`.
+#
+# When running radicle-httpd on a server, it should be run as a system service
+# as a separate user, see `../system/*.service`.
+#
+[Unit]
+Description=Radicle HTTP Daemon
+
+[Service]
+ExecStart=/usr/bin/radicle-httpd $RADICLE_HTTPD_ARGS
+KillMode=mixed
+Restart=always
+RestartSec=3
+
+#ConfigurationDirectory=radicle
+#StateDirectory=radicle
+
+#EnvironmentFile=-%E/radicle/radicle-httpd.env
+Environment=RADICLE_HTTPD_ARGS=--listen 127.0.0.1:7235
+#Environment=RAD_HOME=%S/radicle
+Environment=RUST_BACKTRACE=1
+Environment=RUST_LOG=info
+
+# Basic hardening options. For more, please refer to `systemd-analyze security`.
+NoNewPrivileges=true
+MemoryDenyWriteExecute=true
+
+[Install]
+WantedBy=default.target
