diff --git a/linux/playit.service b/linux/playit.service
index b69376a..774a2d3 100644
--- a/linux/playit.service
+++ b/linux/playit.service
@@ -5,7 +5,7 @@ Wants=network-pre.target
 After=network-pre.target NetworkManager.service systemd-resolved.service
 
 [Service]
-ExecStart=/opt/playit/playitd --secret-path /etc/playit/playit.toml -l /var/log/playit/playit.log
+ExecStart=/usr/bin/playitd --secret-path /etc/playit/playit.toml -l /var/log/playit/playit.log
 Restart=on-failure
 
 [Install]
