diff --git a/src/histedit.c b/src/histedit.c
index 5ab718b..2e5c7a4 100644
--- a/src/histedit.c
+++ b/src/histedit.c
@@ -132,6 +132,8 @@ bad:
 				el_set(el, EL_EDITOR, "vi");
 			else if (Eflag)
 				el_set(el, EL_EDITOR, "emacs");
+			el_set(el, EL_ADDFN, "ed-complete", "Complete argument", _el_fn_complete);
+			el_set(el, EL_BIND, "^I", "ed-complete", NULL);
 			el_source(el, NULL);
 		}
 	} else {
