--- a/ext/taglib/gstid3v2mux.cc	2024-02-27 05:56:34.025810071 +0300
+++ b/ext/taglib/gstid3v2mux.cc	2024-02-27 05:51:40.297482458 +0300
@@ -381,7 +381,13 @@
         GST_DEBUG ("Injecting ID3v2.%u frame %u/%u of length %u and type %"
             GST_PTR_FORMAT, version, i, num_tags, GST_BUFFER_SIZE (buf), s);
 
+#if TAGLIB_MAJOR_VERSION > 1 || (TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION >= 5)
+        ID3v2::Header header;
+        header.setMajorVersion (version);
+        frame = factory->createFrame (bytes, &header);
+#else
         frame = factory->createFrame (bytes, (TagLib::uint) version);
+#endif
         if (frame)
           id3v2tag->addFrame (frame);
       }
@@ -546,7 +552,14 @@
 
       g_free (data);
 
+#if TAGLIB_MAJOR_VERSION > 1 || (TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION >= 5)
+      ID3v2::Header header;
+      header.setMajorVersion (4);
+      frame = factory->createFrame (bytes, &header);
+#else
       frame = factory->createFrame (bytes, (TagLib::uint) 4);
+#endif
+
       if (frame) {
         id3v2tag->addFrame (frame);
 
