diff --unified --recursive --text mesa.orig/src/gallium/targets/opencl/meson.build mesa.new/src/gallium/targets/opencl/meson.build
--- mesa.orig/src/gallium/targets/opencl/meson.build	2019-06-29 20:56:04.000000000 +0200
+++ mesa.new/src/gallium/targets/opencl/meson.build	2019-06-29 21:10:45.934703892 +0200
@@ -51,6 +51,7 @@
     cpp.find_library('clangSema', dirs : llvm_libdir),
     cpp.find_library('clangAnalysis', dirs : llvm_libdir),
     cpp.find_library('clangAST', dirs : llvm_libdir),
+    cpp.find_library('clangASTMatchers', dirs : llvm_libdir),
     cpp.find_library('clangEdit', dirs : llvm_libdir),
     cpp.find_library('clangLex', dirs : llvm_libdir),
     cpp.find_library('clangBasic', dirs : llvm_libdir),
