diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4fe0ea2..0ddb3a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,8 +68,6 @@ else()
                 ${CMAKE_SOURCE_DIR}/vendor/libserialport/linux_termios.c)
         target_compile_definitions(libblisp_obj PRIVATE
                 LIBSERIALPORT_ATBUILD
-                HAVE_TERMIOS2_SPEED
-                HAVE_STRUCT_TERMIOS2
                 HAVE_DECL_BOTHER
                 "SP_API=__attribute__((visibility(\"default\")))"
                 "SP_PRIV=__attribute__((visibility(\"hidden\")))")
