diff --git a/tests/test_contrib.py b/tests/test_contrib.py
index 87155e9..ad9f64a 100644
--- a/tests/test_contrib.py
+++ b/tests/test_contrib.py
@@ -32,7 +32,7 @@ class TestComputeGT(unittest.TestCase):
         Dnew, Inew = knn_ground_truth(xq, matrix_iterator(xb, 1000), 10)
 
         np.testing.assert_array_equal(Iref, Inew)
-        np.testing.assert_almost_equal(Dref, Dnew, decimal=5)
+        np.testing.assert_almost_equal(Dref, Dnew, decimal=4)
 
 
 class TestDatasets(unittest.TestCase):
