diff -aur DOT_0.3.original/dark/cursor.theme DOT_0.3.modified/dark/cursor.theme
--- DOT_0.3.original/dark/cursor.theme	2015-10-16 05:37:05.000000000 +0300
+++ DOT_0.3.modified/dark/cursor.theme	2016-01-15 12:39:20.884053000 +0200
@@ -1,3 +1,3 @@
 [Icon Theme]
-Inherits=TheDOT 0.3 dark
+Inherits=TheDOT-0.3-light
 
diff -aur DOT_0.3.original/dark/index.theme DOT_0.3.modified/dark/index.theme
--- DOT_0.3.original/dark/index.theme	2015-10-16 05:37:22.000000000 +0300
+++ DOT_0.3.modified/dark/index.theme	2016-01-15 12:38:56.004052000 +0200
@@ -1,5 +1,5 @@
 [Icon Theme]
-Name=TheDOT 0.3 dark
+Name=TheDOT-0.3-dark
 Example=arrow
-Inherits=TheDOT 0.3 dark
+Inherits=TheDOT-0.3-light
 
diff -aur DOT_0.3.original/light/cursor.theme DOT_0.3.modified/light/cursor.theme
--- DOT_0.3.original/light/cursor.theme	2015-10-16 05:36:44.000000000 +0300
+++ DOT_0.3.modified/light/cursor.theme	2016-01-15 12:38:36.730718000 +0200
@@ -1,3 +1,3 @@
 [Icon Theme]
-Inherits=TheDOT 0.3 light
+Inherits=TheDOT-0.3-dark
 
diff -aur DOT_0.3.original/light/index.theme DOT_0.3.modified/light/index.theme
--- DOT_0.3.original/light/index.theme	2015-10-16 05:36:35.000000000 +0300
+++ DOT_0.3.modified/light/index.theme	2016-01-15 12:38:26.980718000 +0200
@@ -1,5 +1,5 @@
 [Icon Theme]
-Name=TheDOT 0.3 light
+Name=TheDOT-0.3-light
 Example=arrow
-Inherits=TheDOT 0.3 light
+Inherits=TheDOT-0.3-dark
 
