--- therion.ini_raw	2017-04-03 22:32:08.000000000 +0200
+++ therion.ini	2017-04-11 09:46:37.830780039 +0200
@@ -1,5 +1,6 @@
 ### Output character encodings ###
 # encoding-default  ASCII
+encoding-default  UTF-8
 # encoding-sql  ASCII
 
 ### Default regional settings ###
@@ -23,7 +24,7 @@
 ### Tex initialization ###
 # tex-env off
 # tex-fonts <encoding> <roman> <italic> <bold> <sansserif> <sansserifoblique>
-# tex-fonts raw cmr10 cmti10 cmbx10 cmss10 cmssi10
+tex-fonts raw cmr10 cmti10 cmbx10 cmss10 cmssi10
 # tex-fonts xl2 csr10 csti10 csbx10 csss10 csssi10
 
 ### PDF fonts initialization
