--- OrcaSlicer/deps/TIFF/TIFF.cmake.orig	2023-03-30 02:45:39.957686074 +0800
+++ OrcaSlicer/deps/TIFF/TIFF.cmake	2023-03-17 22:30:57.000000000 +0800
@@ -16,7 +16,7 @@
 else()
     bambustudio_add_cmake_project(TIFF
         URL https://gitlab.com/libtiff/libtiff/-/archive/v4.1.0/libtiff-v4.1.0.zip
-        URL_HASH SHA256=1c100e5a6a25f43961321adb4a1891fc33758556654a7f2f5901d8b587770357
+        URL_HASH SHA256=c56edfacef0a60c0de3e6489194fcb2f24c03dbb550a8a7de5938642d045bd32
         DEPENDS ${ZLIB_PKG} ${PNG_PKG} dep_JPEG
         CMAKE_ARGS
             -Dlzma:BOOL=OFF
