--- a/tllocalmgr	2020-12-02 16:46:49.214712042 +0100
+++ b/tllocalmgr	2020-12-02 17:05:02.476194535 +0100
@@ -171,7 +171,10 @@
 		$installedcol_versions{$col} = $ver;
 		my @list = ();
 		unless ( -f "$INSTALLEDPKGS/texlive-$col\_$ver.pkgs" ) {
-			die "Cannot open $INSTALLEDPKGS/texlive-$col\_$ver.pkgs\n"
+			print BOLD RED "Cannot open $INSTALLEDPKGS/texlive-$col\_$ver.pkgs!\n";
+			print RED "-- Likely the package texlive-$col does not create the 'pkgs' file.\n";
+			print LOG "Cannot open $INSTALLEDPKGS/texlive-$col\_$ver.pkgs!\n";
+			next;
 		};
 		open INST, "<$INSTALLEDPKGS/texlive-$col\_$ver.pkgs" ;
 		while (<INST>) {
