--- rc/trafficserver.service.in.orig	2016-08-17 07:55:35.673822545 -1000
+++ rc/trafficserver.service.in	2016-08-17 07:56:10.830103516 -1000
@@ -21,7 +21,7 @@
 
 [Service]
 Type=simple
-EnvironmentFile=-/etc/sysconfig/trafficserver
+EnvironmentFile=-/etc/conf.d/trafficserver
 PIDFile=@exp_runtimedir@/cop.pid
 ExecStart=@exp_bindir@/traffic_cop $TC_DAEMON_ARGS
 ExecReload=@exp_bindir@/traffic_ctl config reload
