diff -Nur original/plugins/trailfocus/src/trailfocus.cpp modified/plugins/trailfocus/src/trailfocus.cpp
--- original/plugins/trailfocus/src/trailfocus.cpp	2015-08-20 14:01:09.000000000 +0100
+++ modified/plugins/trailfocus/src/trailfocus.cpp	2016-05-08 13:57:35.867366143 +0100
@@ -195,7 +195,7 @@
 	}
 	else
 	{
-	    distance = abs (cur->activeNum () - best->activeNum ());
+	    distance = abs ((int)(cur->activeNum () - best->activeNum ()));
 	    if (distance < bestDist)
 	    {
 		best     = cur;
