diff --git a/python/pyproject.toml b/python/pyproject.toml
--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -10,7 +10,8 @@ dependencies = [
     "pyarrow>=16",
     "pydantic>=2.7.4,<3",
     "tqdm>=4.27.0",
+    "typing-extensions>=4.0.0",
-    "lance-namespace>=0.3.2"
+    "lance-namespace>=0.12,<0.13"
 ]
 description = "lancedb"
 authors = [{ name = "LanceDB Devs", email = "dev@lancedb.com" }]
