diff --git a/Rendering/OpenGL2/vtkTestOpenGLVersion.cxx b/Rendering/OpenGL2/vtkTestOpenGLVersion.cxx
index 2d753bd51df..6ee4a3195ed 100644
--- a/Rendering/OpenGL2/vtkTestOpenGLVersion.cxx
+++ b/Rendering/OpenGL2/vtkTestOpenGLVersion.cxx
@@ -25,7 +25,7 @@ with additional system information.
 #include <vector>
 
 #include <windows.h>
-#include <gl/GL.h>
+#include <GL/gl.h>
 
 class CheckOpenGLVersion {
 public:
