diff -ura uberwriter.orig/de.wolfvollprecht.UberWriter.desktop uberwriter.new/de.wolfvollprecht.UberWriter.desktop
--- uberwriter.orig/de.wolfvollprecht.UberWriter.desktop	2018-08-09 21:34:15.253054010 +0200
+++ uberwriter.new/de.wolfvollprecht.UberWriter.desktop	2018-08-09 21:35:30.953109625 +0200
@@ -3,7 +3,7 @@
 Comment=UberWriter, a simple and distraction free Markdown Editor
 Categories=GNOME;Office;
 Exec=uberwriter %U
-Icon=/usr/share/uberwriter/media/uberwriter.svg
+Icon=uberwriter
 Terminal=false
 Type=Application
 MimeType=text/x-markdown;text/plain;
diff -ura uberwriter.orig/setup.py uberwriter.new/setup.py
--- uberwriter.orig/setup.py	2018-08-09 21:35:46.873123653 +0200
+++ uberwriter.new/setup.py	2018-08-09 21:34:15.283054000 +0200
@@ -76,10 +76,10 @@
         'uberwriter_lib.pylocales' : ['locales.db'],
     },
     data_files=[
-        (app_prefix + '/bin/', ['bin/uberwriter']),
-        (app_prefix + '/share/glib-2.0/schemas', ['data/de.wolfvollprecht.UberWriter.gschema.xml']),
-        (app_prefix + '/share/icons/hicolor/scalable/apps', ['data/media/uberwriter.svg']),
-        (app_prefix + '/share/applications', ['de.wolfvollprecht.UberWriter.desktop']),
+        (app_prefix + '/usr/bin/', ['bin/uberwriter']),
+        (app_prefix + '/usr/share/glib-2.0/schemas', ['data/de.wolfvollprecht.UberWriter.gschema.xml']),
+        (app_prefix + '/usr/share/icons/hicolor/scalable/apps', ['data/media/uberwriter.svg']),
+        (app_prefix + '/usr/share/applications', ['de.wolfvollprecht.UberWriter.desktop']),
         (app_prefix + '/opt/uberwriter/data/ui', extra_files_ui),
         (app_prefix + '/opt/uberwriter/data/media', extra_files_media),
         (app_prefix + '/opt/uberwriter/data/lua', extra_files_scripts)
