--- src/qdmr/dist/99-qdmr.rules.orig	2025-11-30 19:39:26.302016776 +0100
+++ src/qdmr/dist/99-qdmr.rules	2025-11-30 19:41:14.401533766 +0100
@@ -1,12 +1,12 @@
 # Radioddity GD-73
-SUBSYSTEM=="usb", ATTRS{idVendor}=="1206", ATTRS{idProduct}=="0227", MODE="660", GROUP="dialout"
+SUBSYSTEM=="usb", ATTRS{idVendor}=="1206", ATTRS{idProduct}=="0227", MODE="660", GROUP="uucp"
 
 # TYT MD-UV380
-SUBSYSTEM=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="df11", MODE="660", GROUP="dialout"
+SUBSYSTEM=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="df11", MODE="660", GROUP="uucp"
 
 # Baofeng RD-5R, TD-5R
 # Radioddity GD-77
-SUBSYSTEM=="usb", ATTRS{idVendor}=="15a2", ATTRS{idProduct}=="0073", MODE="660", GROUP="dialout"
+SUBSYSTEM=="usb", ATTRS{idVendor}=="15a2", ATTRS{idProduct}=="0073", MODE="660", GROUP="uucp"
 
 # Ignore this device in Modem Manager
 # Anytone AT-D868UV, AT-D878UV, AT-D578UV (II) (GD32)
