--- a/uefitool/UEFIPatch/uefipatch.cpp	2016-10-07 23:39:16.579188249 +0200
+++ b/uefitool/UEFIPatch/uefipatch.cpp	2016-10-07 23:40:00.707034736 +0200
@@ -27,13 +27,13 @@
 
 UINT8 UEFIPatch::patchFromFile(QString path)
 {
-    QFileInfo patchInfo = QFileInfo("patches.txt");
+    QFileInfo patchInfo = QFileInfo("/usr/share/uefitool/patches.txt");
 
     if (!patchInfo.exists())
         return ERR_INVALID_FILE;
 
     QFile file;
-    file.setFileName("patches.txt");
+    file.setFileName("/usr/share/uefitool/patches.txt");
 
     if (!file.open(QFile::ReadOnly | QFile::Text))
         return ERR_INVALID_FILE;
