diff --git a/ibdump.c b/ibdump.c
index a9405b7..500579d 100644
--- a/ibdump.c
+++ b/ibdump.c
@@ -998,9 +998,9 @@ int set_sw_sniffer(struct resources *res, int mode, int is_tx, int is_rx)
     if ((res->dev_rev_id == DI_CIB) || (res->dev_rev_id == DI_CX4) ||
         (res->dev_rev_id == DI_CX4LX) || (res->dev_rev_id == DI_CX5) ||
         (res->dev_rev_id == DI_CX6) || (res->dev_rev_id == DI_CX6DX) ||
-        (res->dev_rev_id == DI_CX7) || (res->dev_rev_id != DI_CX8) ||
-        (res->dev_rev_id != DI_BF2) || (res->dev_rev_id != DI_BF2A1) ||
-        (res->dev_rev_id != DI_BF3) || (res->dev_rev_id != DI_BF4))
+        (res->dev_rev_id == DI_CX7) || (res->dev_rev_id == DI_CX8) ||
+        (res->dev_rev_id == DI_BF2) || (res->dev_rev_id == DI_BF2A1) ||
+        (res->dev_rev_id == DI_BF3) || (res->dev_rev_id == DI_BF4))
     {
         return fifth_gen_set_sw_sniffer(res, mode);
     }
