--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -786,6 +786,11 @@ add_definitions(-DBOOST_ERROR_CODE_HEADER_ONLY
     -DBOOST_REGEX_NO_LIB
 )
 
+if (ARCHITECTURE_x86_64)
+    find_package(dynarmic REQUIRED)
+    add_library(dynarmic ALIAS dynarmic::dynarmic)
+endif()
+
 enable_testing()
 add_subdirectory(externals)
 add_subdirectory(src)
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -18,6 +18,6 @@ add_library(catch-single-include INTERFACE)
 target_include_directories(catch-single-include INTERFACE catch/single_include)
 
 # Dynarmic
-if (ARCHITECTURE_x86_64)
+if (ARCHITECTURE_x86_64 AND NOT TARGET dynarmic)
     set(DYNARMIC_NO_BUNDLED_FMT ON)
     set(DYNARMIC_IGNORE_ASSERTS ON CACHE BOOL "" FORCE)
