--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,6 +136,10 @@ if (DYNARMIC_TESTS_USE_UNICORN)
     find_package(Unicorn REQUIRED)
 endif()
 
+find_library(ZYDIS_LIBRARY Zydis REQUIRED)
+add_library(Zydis UNKNOWN IMPORTED)
+set_property(TARGET Zydis PROPERTY IMPORTED_LOCATION "${ZYDIS_LIBRARY}")
+
 # Pull in externals CMakeLists for libs where available
 add_subdirectory(externals)
 
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -42,10 +42,3 @@ if (NOT TARGET xbyak)
         target_compile_definitions(xbyak INTERFACE XBYAK_NO_OP_NAMES)
     endif()
 endif()
-
-# zydis
-
-option(ZYDIS_BUILD_TOOLS "" OFF)
-option(ZYDIS_BUILD_EXAMPLES "" OFF)
-set(ZYDIS_ZYCORE_PATH "${CMAKE_CURRENT_LIST_DIR}/zycore" CACHE PATH "")
-add_subdirectory(zydis EXCLUDE_FROM_ALL)
