diff --git a/Cargo.toml b/Cargo.toml
index ad7e70eb38..ebffe0716f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -195,7 +195,7 @@ ref-cast = "1.0.0"
 regex = "1.5.4"
 relative-path = { version = "1.7.0", features = ["serde"] }
 reqwest = { version = "0.11.4", features = ["stream", "rustls-tls", "rustls-tls-webpki-roots"], default-features = false }
-rusqlite = { version = "0.28.0", features = ["bundled"] }
+rusqlite = { version = "0.28.0" }
 rustyline = "11.0"
 sequence_trie = "0.3.6"
 serde = { version = "1.0", features = ["derive"] }
