--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -841,6 +841,9 @@ add_definitions(-DBOOST_ERROR_CODE_HEADER_ONLY
     -DBOOST_REGEX_NO_LIB
 )
 
+find_package(sirit REQUIRED)
+add_library(sirit ALIAS sirit::sirit)
+
 enable_testing()
 add_subdirectory(externals)
 add_subdirectory(src)
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -93,3 +92,0 @@ endif()
-# Sirit
-add_subdirectory(sirit)
-
