diff --unified --recursive --text orig/serialplot/src/updatechecker.cpp new/serialplot/src/updatechecker.cpp
--- orig/serialplot/src/updatechecker.cpp	2019-03-19 10:39:03.674912791 +0100
+++ new/serialplot/src/updatechecker.cpp	2019-03-19 10:39:22.542052005 +0100
@@ -47,7 +47,7 @@
 
 void UpdateChecker::checkUpdate()
 {
-    if (isChecking()) return;
+    return;
 
     auto req = QNetworkRequest(QUrl(BB_DOWNLOADS_URL));
     activeReply = nam.get(req);
