Index: tunslip6-3.1/tunslip6.c
===================================================================
--- tunslip6-3.1.orig/tunslip6.c
+++ tunslip6-3.1/tunslip6.c
@@ -849,6 +851,7 @@ fprintf(stderr," -H             Hardware
 fprintf(stderr," -I             Inquire IP address\n");
 fprintf(stderr," -X             Software XON/XOFF flow control (default disabled)\n");
 fprintf(stderr," -L             Log output format (adds time stamps)\n");
+fprintf(stderr," -P             Log progress debug information\n");
 fprintf(stderr," -s siodev      Serial device (default /dev/ttyUSB0)\n");
 fprintf(stderr," -M             Interface MTU (default and min: 1280)\n");
 fprintf(stderr," -T             Make tap interface (default is tun interface)\n");
@@ -860,7 +863,7 @@ fprintf(stderr,"    -v2         Printabl
 fprintf(stderr,"    -v3         Printable strings and SLIP packet notifications\n");
 fprintf(stderr,"    -v4         All printable characters as they are received\n");
 fprintf(stderr,"    -v5         All SLIP packets in hex\n");
-fprintf(stderr,"    -v          Equivalent to -v3\n");
+fprintf(stderr,"    -v          Equivalent to -v2\n");
 fprintf(stderr," -d[basedelay]  Minimum delay between outgoing SLIP packets.\n");
 fprintf(stderr,"                Actual delay is basedelay*(#6LowPAN fragments) milliseconds.\n");
 fprintf(stderr,"                -d is equivalent to -d10.\n");
@@ -874,7 +877,7 @@ exit(1);
   argv += (optind - 1);
 
   if(argc != 2 && argc != 3) {
-    err(1, "usage: %s [-B baudrate] [-H] [-L] [-s siodev] [-t tundev] [-T] [-v verbosity] [-d delay] [-a serveraddress] [-p serverport] ipaddress", prog);
+    err(1, "usage: %s [-B baudrate] [-H] [-L] [-P] [-VERBOSITY [-t tundev] [-T] [-vVERBOSITY] [-d delay] [-a serveraddress] [-p serverport] ipaddress", prog);
   }
   ipaddr = argv[1];
 
