diff --git a/pyproject.toml b/pyproject.toml
index 3d8540f7..8bac4513 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "ninja", "wheel", "pybind11>=2.10.1"]
+requires = ["setuptools>=40.8.0", "wheel", "pybind11>=2.10.1"]
 build-backend = "setuptools.build_meta"
 
 [tool.pytype]
diff --git a/pytype/tools/analyze_project/pytype_runner.py b/pytype/tools/analyze_project/pytype_runner.py
index a2065c09..3330d990 100644
--- a/pytype/tools/analyze_project/pytype_runner.py
+++ b/pytype/tools/analyze_project/pytype_runner.py
@@ -405,8 +405,7 @@ class PytypeRunner:
     # relpath() prevents possibly sensitive directory info from appearing in
     # ninja's "Entering directory" message.
     c = path_utils.relpath(path_utils.dirname(self.ninja_file))
-    command = _get_executable('ninja') + [
-        '-k', k, '-C', c, '-j', str(self.jobs)]
+    command = ['ninja', '-k', k, '-C', c, '-j', str(self.jobs)]
     if logging.getLogger().isEnabledFor(logging.INFO):
       command.append('-v')
     ret = subprocess.call(command)
diff --git a/setup.cfg b/setup.cfg
index 9857d67c..5e2c42d0 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -37,7 +37,6 @@ install_requires =
     libcst>=1.0.1
     msgspec>=0.18.6
     networkx>=2.8
-    ninja>=1.10.0.post2
     pycnite>=2024.07.31
     pydot>=1.4.2
     tabulate>=0.8.10
