diff --git a/src/MadnessInteractiveReloaded/MIR.csproj b/src/MadnessInteractiveReloaded/MIR.csproj
index c69724a..1fb2424 100644
--- a/src/MadnessInteractiveReloaded/MIR.csproj
+++ b/src/MadnessInteractiveReloaded/MIR.csproj
@@ -79,6 +79,7 @@
 	
 	<ItemGroup>
 		<ProjectReference Include="..\BigGustave\src\BigGustave\BigGustave.csproj" />
+		<ProjectReference Include="..\Walgelijk\Walgelijk.csproj" />
 	</ItemGroup>
 	
 	<ItemGroup>
