diff --git a/tools/mxl-info/main.cpp b/tools/mxl-info/main.cpp
index 866236e..c7818f4 100644
--- a/tools/mxl-info/main.cpp
+++ b/tools/mxl-info/main.cpp
@@ -5,6 +5,7 @@
 #include <cstdlib>
 #include <filesystem>
 #include <iostream>
+#include <span>
 #include <sstream>
 #include <string>
 #include <unistd.h>
@@ -13,7 +14,6 @@
 #include <CLI/CLI.hpp>
 #include <fmt/color.h>
 #include <fmt/format.h>
-#include <gsl/span>
 #include <picojson/picojson.h>
 #include <mxl/flow.h>
 #include <mxl/flowinfo.h>
@@ -98,7 +98,7 @@ namespace
 
         std::ostream& operator<<(std::ostream& os, mxlFlowInfo const& info)
         {
-            auto const span = uuids::span<std::uint8_t, sizeof info.config.common.id>{
+            auto const span = std::span<std::uint8_t, sizeof info.config.common.id>{
                 const_cast<std::uint8_t*>(info.config.common.id), sizeof info.config.common.id};
             auto const id = uuids::uuid(span);
             os << "- Flow [" << uuids::to_string(id) << ']' << '\n'
