diff -aur --no-dereference radium-5.9.60.pristine/bin/packages/build.sh radium-5.9.60.new/bin/packages/build.sh
--- radium-5.9.60.pristine/bin/packages/build.sh	2019-05-11 22:54:53.259293886 +0100
+++ radium-5.9.60.new/bin/packages/build.sh	2019-05-11 22:55:12.335959902 +0100
@@ -110,6 +110,7 @@
 rm -fr libpd-master
 tar xvzf libpd-master.tar.gz
 cd libpd-master/
+sed -i '/define CFLAGS/ s|")| -I/usr/include/tirpc ")|' make.scm
 make clean
 make -j`nproc`
 cd ..
