diff --git a/CMakeLists.txt b/CMakeLists.txt
index df7f51d38..50e8108a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -392,11 +392,13 @@ endif()
 
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
 set(XXHASH_BUILD_XXHSUM OFF CACHE BOOL "Build the xxhsum binary")
-add_subdirectory(core/deps/xxHash/cmake_unofficial)
-target_link_libraries(${PROJECT_NAME} PRIVATE xxHash::xxhash)
+find_package(PkgConfig)
+pkg_search_module(xxhash REQUIRED IMPORTED_TARGET libxxhash)
+target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::xxhash)
 
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
-add_subdirectory(core/deps/glm)
+find_package(glm REQUIRED)
+target_compile_definitions(glm::glm INTERFACE GLM_ENABLE_EXPERIMENTAL)
 target_link_libraries(${PROJECT_NAME} PRIVATE glm::glm)
 
 if(USE_VULKAN)
@@ -413,7 +415,7 @@ if(USE_VULKAN)
 		add_library(glslang::glslang-default-resource-limits ALIAS glslang-default-resource-limits)
 		add_library(glslang::SPIRV ALIAS SPIRV)
 	endif()
-	target_link_libraries(${PROJECT_NAME} PRIVATE glslang::glslang-default-resource-limits glslang::SPIRV)
+	target_link_libraries(${PROJECT_NAME} PRIVATE glslang::glslang-default-resource-limits glslang::glslang)
 endif()
 
 if(NOT LIBRETRO)
@@ -467,13 +469,13 @@ if(NOT LIBRETRO)
 			target_link_libraries(${PROJECT_NAME} PRIVATE CURL::libcurl)
 		endif()
 	endif()
-
+endif()
 	find_package(ZLIB)
 	if(TARGET ZLIB::ZLIB AND NOT ANDROID AND (NOT WIN32 OR WINDOWS_STORE))
 		set(WITH_SYSTEM_ZLIB ON CACHE BOOL "Use system provided zlib library")
 		target_link_libraries(${PROJECT_NAME} PRIVATE ZLIB::ZLIB)
 	endif()
-
+if(FALSE)
 	if(USE_LUA)
 		find_package(Lua 5.2)
 		if(NOT APPLE AND LUA_FOUND)
@@ -537,25 +539,24 @@ if(PKG_CONFIG_FOUND AND NOT ANDROID AND NOT APPLE AND NOT LIBRETRO)
 			target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBPULSE)
 		endif()
 	endif()
-
+endif()
 	if(USE_HOST_LIBZIP)
 		pkg_check_modules(LIBZIP IMPORTED_TARGET libzip)
 		if(LIBZIP_FOUND)
 			target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBZIP)
 		endif()
 	endif()
-
+if(FALSE)
 	if(ENABLE_OPROFILE)
 		target_compile_definitions(${PROJECT_NAME} PRIVATE DYNA_OPROF)
 		target_link_libraries(${PROJECT_NAME} PRIVATE opagent)
 	endif()
-
+endif()
 	find_package(MiniUPnPc)
 	if(MINIUPNP_FOUND)
 		target_include_directories(${PROJECT_NAME} PRIVATE ${MINIUPNP_INCLUDE_DIRS})
 		target_link_libraries(${PROJECT_NAME} PRIVATE ${MINIUPNP_LIBRARIES})
 	endif()
-endif()
 
 if(UNIX AND NOT APPLE AND NOT ANDROID)
 	add_definitions(
@@ -1364,8 +1365,8 @@ if(USE_VULKAN)
 	add_subdirectory(core/deps/Vulkan-Headers)
 	target_link_libraries(${PROJECT_NAME} PRIVATE Vulkan::Headers)
 
-	add_subdirectory(core/deps/VulkanMemoryAllocator)
-	target_compile_options(VulkanMemoryAllocator INTERFACE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:AppleClang,Clang>>:-Wno-nullability-completeness>)
+	find_package(VulkanMemoryAllocator REQUIRED)
+	target_compile_options(GPUOpen::VulkanMemoryAllocator INTERFACE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:AppleClang,Clang>>:-Wno-nullability-completeness>)
 	target_link_libraries(${PROJECT_NAME} PRIVATE GPUOpen::VulkanMemoryAllocator)
 
 	if(ANDROID AND NOT LIBRETRO AND "arm64" IN_LIST ARCHITECTURE)
