diff -aur --no-dereference package.pristine/bin/packages/build.sh package.new/bin/packages/build.sh
--- package.pristine/bin/packages/build.sh	2017-03-26 19:27:43.000000000 +0000
+++ package.new/bin/packages/build.sh	2017-03-30 05:14:35.867793829 +0000
@@ -189,27 +189,27 @@
     cd ../
 fi
 
-if [[ $RADIUM_QT_VERSION == 5 ]]
-then
-
-    rm -fr xcb-proto-1.12/
-    tar xvjf xcb-proto-1.12.tar.bz2
-    cd xcb-proto-1.12/
-    mkdir install
-    ./configure --prefix=`pwd`/install
-    make
-    make install
-    cd ..
-    
-    rm -fr libxcb-1.12
-    tar xvjf libxcb-1.12.tar.bz2 
-    cd libxcb-1.12
-    export PKG_CONFIG_PATH=`pwd`/../xcb-proto-1.12/install/lib/pkgconfig:$PKG_CONFIG_PATH
-    ./configure
-    make
-    cd ..
-    
-fi
+#if [[ $RADIUM_QT_VERSION == 5 ]]
+#then
+#
+#    rm -fr xcb-proto-1.12/
+#    tar xvjf xcb-proto-1.12.tar.bz2
+#    cd xcb-proto-1.12/
+#    mkdir install
+#    ./configure --prefix=`pwd`/install
+#    make
+#    make install
+#    cd ..
+#    
+#    rm -fr libxcb-1.12
+#    tar xvjf libxcb-1.12.tar.bz2 
+#    cd libxcb-1.12
+#    export PKG_CONFIG_PATH=`pwd`/../xcb-proto-1.12/install/lib/pkgconfig:$PKG_CONFIG_PATH
+#    ./configure
+#    make
+#    cd ..
+#    
+#fi
 
 
 touch deletemetorebuild
diff -aur --no-dereference package.pristine/Qt/Qt_Main.cpp package.new/Qt/Qt_Main.cpp
--- package.pristine/Qt/Qt_Main.cpp	2017-03-26 19:27:43.000000000 +0000
+++ package.new/Qt/Qt_Main.cpp	2017-03-30 05:12:36.587795134 +0000
@@ -2523,6 +2523,7 @@
 
 #endif // !defined(RELEASE)
 
+/*
 #if defined(FOR_LINUX)
   {
     QString name(dlpi_name);
@@ -2534,7 +2535,7 @@
     }
   }
 #endif
- 
+*/
  
 
   if (is_main_root)
