diff -aur --no-dereference radium-5.9.67.pristine/bin/packages/build.sh radium-5.9.67.new/bin/packages/build.sh
--- radium-5.9.67.pristine/bin/packages/build.sh	2019-06-19 13:56:47.000000000 +0100
+++ radium-5.9.67.new/bin/packages/build.sh	2019-06-19 17:55:42.121298953 +0100
@@ -13,12 +13,12 @@
 export CPPFLAGS="-mtune=generic -msse2 -mfpmath=sse -fPIC -fno-strict-aliasing"
 export CXXFLAGS="-mtune=generic -msse2 -mfpmath=sse -fPIC -fno-strict-aliasing"
 
-DASCC=gcc
-DASCXX=g++
+#DASCC=gcc
+#DASCXX=g++
 
 # These two lines should be uncommented if setting CLANG=1 in build_linux_common.sh
-#DASCC=clang
-#DASCXX=clang++
+DASCC=clang
+DASCXX=clang++
 
 if ! env |grep RADIUM_QT_VERSION ; then
     echo "Must define RADIUM_QT_VERSION to either 4 or 5. For instance: \"RADIUM_QT_VERSION=5 make packages\""
diff -aur --no-dereference radium-5.9.67.pristine/build_linux_common.sh radium-5.9.67.new/build_linux_common.sh
--- radium-5.9.67.pristine/build_linux_common.sh	2019-06-19 13:56:47.000000000 +0100
+++ radium-5.9.67.new/build_linux_common.sh	2019-06-19 17:55:11.894631896 +0100
@@ -57,7 +57,7 @@
 # REQUIRES_RTTI=1 cmake -DLLVM_ENABLE_PROJECTS="clang;compiler-rt" -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=$GCC_PREFIX/bin/gcc -DCMAKE_CXX_COMPILER=$GCC_PREFIX/bin/g++ -DGCC_INSTALL_PREFIX=$GCC_PREFIX -DCMAKE_INSTALL_PREFIX=/home/kjetil/site -DLLVM_ENABLE_RTTI=ON ../llvm 
 # REQUIRES_RTTI=1 make REQUIRES_RTTI=1 -j2
 
-USE_CLANG=0
+USE_CLANG=1
 
 if [[ $USE_CLANG == 1 ]] ; then
     export CLANG_PREFIX=$(dirname `which clang`)/../
