--- imgui-sfml-3.0/CMakeLists.txt	2025-01-03 19:53:12.000000000 +0100
+++ imgui-sfml-3.0.patched/CMakeLists.txt	2025-02-12 16:35:53.397212934 +0100
@@ -20,7 +20,7 @@
 set(IMGUI_SFML_CONFIG_INSTALL_DIR "" CACHE PATH "Path where user's config header will be installed")
 
 # For FindImGui.cmake
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+# list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
 if(IMGUI_SFML_FIND_SFML)
   if(NOT BUILD_SHARED_LIBS)
@@ -32,24 +32,25 @@
 # ImGui does not provide native support for CMakeLists, workaround for now to have
 # users specify IMGUI_DIR. Waiting for this PR to get merged...
 #    https://github.com/ocornut/imgui/pull/1713
-if(NOT IMGUI_DIR)
-  set(IMGUI_DIR "" CACHE PATH "imgui top-level directory")
-  message(FATAL_ERROR "ImGui directory not found. Set IMGUI_DIR to imgui's top-level path (containing 'imgui.h' and other files).\n")
-endif()
+# if(NOT IMGUI_DIR)
+#   set(IMGUI_DIR "" CACHE PATH "imgui top-level directory")
+#   message(FATAL_ERROR "ImGui directory not found. Set IMGUI_DIR to imgui's top-level path (containing 'imgui.h' and other files).\n")
+# endif()
 
 # This uses FindImGui.cmake provided in ImGui-SFML repo for now
-find_package(ImGui 1.91.1 REQUIRED)
+find_package(imgui REQUIRED)
 
 # These headers will be installed alongside ImGui-SFML
-set(IMGUI_PUBLIC_HEADERS
-  ${IMGUI_INCLUDE_DIR}/imconfig.h
-  ${IMGUI_INCLUDE_DIR}/imgui.h
-  ${IMGUI_INCLUDE_DIR}/imgui_internal.h # not actually public, but users might need it
-  ${IMGUI_INCLUDE_DIR}/imstb_rectpack.h
-  ${IMGUI_INCLUDE_DIR}/imstb_textedit.h
-  ${IMGUI_INCLUDE_DIR}/imstb_truetype.h
-  ${IMGUI_INCLUDE_DIR}/misc/cpp/imgui_stdlib.h
-)
+
+# set(IMGUI_PUBLIC_HEADERS
+#   ${IMGUI_INCLUDE_DIR}/imconfig.h
+#   ${IMGUI_INCLUDE_DIR}/imgui.h
+#   ${IMGUI_INCLUDE_DIR}/imgui_internal.h # not actually public, but users might need it
+#   ${IMGUI_INCLUDE_DIR}/imstb_rectpack.h
+#   ${IMGUI_INCLUDE_DIR}/imstb_textedit.h
+#   ${IMGUI_INCLUDE_DIR}/imstb_truetype.h
+#   ${IMGUI_INCLUDE_DIR}/misc/cpp/imgui_stdlib.h
+# )
 
 # CMake 3.11 and later prefer to choose GLVND, but we choose legacy OpenGL just because it's safer
 # (unless the OpenGL_GL_PREFERENCE was explicitly set)
@@ -63,7 +64,7 @@
 include(GNUInstallDirs)
 
 # Define ImGui-SFML
-add_library(ImGui-SFML imgui-SFML.cpp ${IMGUI_SOURCES})
+add_library(ImGui-SFML imgui-SFML.cpp)
 add_library(ImGui-SFML::ImGui-SFML ALIAS ImGui-SFML)
 target_include_directories(ImGui-SFML PUBLIC
   $<BUILD_INTERFACE:${IMGUI_INCLUDE_DIR}>
