diff --git a/cpp/src/command_classes/DoorLockLogging.cpp b/cpp/src/command_classes/DoorLockLogging.cpp
index c235a849..be0ea5ac 100644
--- a/cpp/src/command_classes/DoorLockLogging.cpp
+++ b/cpp/src/command_classes/DoorLockLogging.cpp
@@ -304,16 +304,15 @@ bool DoorLockLogging::HandleMsg
 			}
 			uint8 userid = (_data[10]);
 			uint8 usercodelength = (_data[11]);
-			char usercode[254];
-			snprintf(usercode, sizeof(usercode), "UserCode:");
+			char usercode[254] = { 0 };
 			if (usercodelength > 0)
-				for (int i = 0; i < usercodelength; i++ )
+				for (int i = 0, j = 0; i < usercodelength && j < (int)sizeof(usercode); i++ )
 				{
-					snprintf(usercode, sizeof(usercode), "%s %d", usercode, (int)_data[12+i]);
+					j += snprintf(usercode + j, sizeof(usercode) - j, " %d", (int)_data[12+i]);
 				}
 
 			if (valid) {
-				snprintf(msg, sizeof(msg), "%02d/%02d/%02d %02d:%02d:%02d \tMessage: %s \tUserID: %d \t%s", (int)day, (int)month, (int)year, (int)hour, (int)minute, (int)second, c_DoorLockEventType[EventType], (int)userid, usercode);
+				snprintf(msg, sizeof(msg), "%02d/%02d/%02d %02d:%02d:%02d \tMessage: %s \tUserID: %d \tUserCode:%s", (int)day, (int)month, (int)year, (int)hour, (int)minute, (int)second, c_DoorLockEventType[EventType], (int)userid, usercode);
 			} else
 				snprintf(msg, sizeof(msg), "Invalid Record");
 			value->OnValueRefreshed(msg);
