diff --unified --recursive --text asus-nb-ctrl/Makefile new/Makefile
--- asus-nb-ctrl/Makefile	2020-09-27 12:34:08.648921861 -0400
+++ new/Makefile	2020-09-27 12:49:03.576026582 -0400
@@ -42,13 +42,13 @@
 	install -D -m 0755 "target/release/$(BIN_C)" "$(DESTDIR)$(bindir)/$(BIN_C)"
 	install -D -m 0755 "target/release/$(BIN_D)" "$(DESTDIR)$(bindir)/$(BIN_D)"
 	install -D -m 0755 "target/release/$(BIN_N)" "$(DESTDIR)$(bindir)/$(BIN_N)"
-	install -D -m 0644 "data/$(PMRULES)" "$(DESTDIR)/lib/udev/rules.d/$(PMRULES)"
-	install -D -m 0644 "data/$(BIN_D).rules" "$(DESTDIR)/lib/udev/rules.d/99-$(BIN_D).rules"
+	install -D -m 0644 "data/$(PMRULES)" "$(DESTDIR)/usr/lib/udev/rules.d/$(PMRULES)"
+	install -D -m 0644 "data/$(BIN_D).rules" "$(DESTDIR)/usr/lib/udev/rules.d/99-$(BIN_D).rules"
 	install -D -m 0644 "data/$(LEDCFG)" "$(DESTDIR)$(sysconfdir)/asusd/$(LEDCFG)"
 	install -D -m 0644 "data/$(BIN_D).conf" "$(DESTDIR)$(sysconfdir)/dbus-1/system.d/$(BIN_D).conf"
 	install -D -m 0644 "data/$(X11CFG)" "$(DESTDIR)$(sysconfdir)/X11/xorg.conf.d/$(X11CFG)"
-	install -D -m 0644 "data/$(BIN_D).service" "$(DESTDIR)/lib/systemd/system/$(BIN_D).service"
-	install -D -m 0644 "data/$(BIN_N).service" "$(DESTDIR)/lib/systemd/user/$(BIN_N).service"
+	install -D -m 0644 "data/$(BIN_D).service" "$(DESTDIR)/usr/lib/systemd/system/$(BIN_D).service"
+	install -D -m 0644 "data/$(BIN_N).service" "$(DESTDIR)/usr/lib/systemd/user/$(BIN_N).service"
 	install -D -m 0644 "data/icons/asus_notif_yellow.png" "$(DESTDIR)/usr/share/icons/hicolor/512x512/apps/asus_notif_yellow.png"
 	install -D -m 0644 "data/icons/asus_notif_green.png" "$(DESTDIR)/usr/share/icons/hicolor/512x512/apps/asus_notif_green.png"
 	install -D -m 0644 "data/icons/asus_notif_red.png" "$(DESTDIR)/usr/share/icons/hicolor/512x512/apps/asus_notif_red.png"
@@ -57,12 +57,12 @@
 	rm -f "$(DESTDIR)$(bindir)/$(BIN_C)"
 	rm -f "$(DESTDIR)$(bindir)/$(BIN_D)"
 	rm -f "$(DESTDIR)$(bindir)/$(BIN_N)"
-	rm -f "$(DESTDIR)/lib/udev/rules.d/$(PMRULES)"
-	rm -f "$(DESTDIR)/lib/udev/rules.d/99-$(BIN_D).rules"
+	rm -f "$(DESTDIR)/usr/lib/udev/rules.d/$(PMRULES)"
+	rm -f "$(DESTDIR)/usr/lib/udev/rules.d/99-$(BIN_D).rules"
 	rm -f "$(DESTDIR)$(sysconfdir)/dbus-1/system.d/$(BIN_D).conf"
 	rm -f "$(DESTDIR)$(sysconfdir)/X11/xorg.conf.d/$(X11CFG)"
-	rm -f "$(DESTDIR)/lib/systemd/system/$(BIN_D).service"
-	rm -r "$(DESTDIR)/lib/systemd/user/$(BIN_N).service"
+	rm -f "$(DESTDIR)/usr/lib/systemd/system/$(BIN_D).service"
+	rm -r "$(DESTDIR)/usr/lib/systemd/user/$(BIN_N).service"
 	rm -r "$(DESTDIR)/usr/share/icons/hicolor/512x512/apps/asus_notif_*"
 
 update:
