--- a/src/Makefile	2013-06-03 03:03:05.321690000 +0200
+++ b/src/Makefile	2013-06-04 03:45:13.514436898 +0200
@@ -96,8 +96,8 @@
 	${CC} $(CFLAGS) ${CPPFLAGS} -c $< -o $@
 
 install : all
-	$(INSTALL) -d ${DESTDIR}$(prefix)/sbin
-	$(INSTALL) -s mosquitto ${DESTDIR}${prefix}/sbin/mosquitto
+	$(INSTALL) -d ${DESTDIR}$(prefix)/bin
+	$(INSTALL) -s mosquitto ${DESTDIR}${prefix}/bin/mosquitto
 	$(INSTALL) mosquitto_plugin.h ${DESTDIR}${prefix}/include/mosquitto_plugin.h
 ifeq ($(WITH_TLS),yes)
 	$(INSTALL) -s mosquitto_passwd ${DESTDIR}${prefix}/bin/mosquitto_passwd
