diff -rupN mjpg-streamer-r63/plugins/input_uvc/input_uvc.c mjpg-streamer-r63-patch/plugins/input_uvc/input_uvc.c
--- mjpg-streamer-r63/plugins/input_uvc/input_uvc.c	2008-06-16 18:23:51.000000000 +0700
+++ mjpg-streamer-r63-patch/plugins/input_uvc/input_uvc.c	2015-02-24 14:46:17.544148293 +0700
@@ -706,7 +706,7 @@ void *cam_thread( void *arg ) {
     }
     else {
       DBG("copying frame\n");
-      pglobal->size = memcpy_picture(pglobal->buf, videoIn->tmpbuffer, videoIn->buf.bytesused);
+      pglobal->size = memcpy_picture(pglobal->buf, videoIn->tmpbuffer, videoIn->currPictSize);
     }
 
 #if 0
diff -rupN mjpg-streamer-r63/plugins/input_uvc/v4l2uvc.c mjpg-streamer-r63-patch/plugins/input_uvc/v4l2uvc.c
--- mjpg-streamer-r63/plugins/input_uvc/v4l2uvc.c	2008-06-16 18:23:51.000000000 +0700
+++ mjpg-streamer-r63-patch/plugins/input_uvc/v4l2uvc.c	2015-02-24 14:48:42.381979926 +0700
@@ -333,6 +333,7 @@ int uvcGrab(struct vdIn *vd)
       */
 
       memcpy(vd->tmpbuffer, vd->mem[vd->buf.index], vd->buf.bytesused);
+      vd->currPictSize = vd->buf.bytesused;
 
       if (debug)
         fprintf(stderr, "bytes in used %d \n", vd->buf.bytesused);
diff -rupN mjpg-streamer-r63/plugins/input_uvc/v4l2uvc.h mjpg-streamer-r63-patch/plugins/input_uvc/v4l2uvc.h
--- mjpg-streamer-r63/plugins/input_uvc/v4l2uvc.h	2008-06-16 18:23:51.000000000 +0700
+++ mjpg-streamer-r63-patch/plugins/input_uvc/v4l2uvc.h	2015-02-24 14:49:26.164355180 +0700
@@ -73,6 +73,7 @@ struct vdIn {
     int framecount;
     int recordstart;
     int recordtime;
+    int currPictSize
 };
 
 int init_videoIn(struct vdIn *vd, char *device, int width, int height, int fps, int format, int grabmethod);
