--- ../build/vboxnetadp/linux/VBoxNetAdp-linux.c	2015-09-26 14:28:05.092622068 +0200
+++ ../build/vboxnetadp/linux/VBoxNetAdp-linux.new.c	2015-09-26 14:33:16.320075314 +0200
@@ -436,11 +436,7 @@
 
     vboxNetAdpShutdown();
     /* Remove control device */
-    rc = misc_deregister(&g_CtlDev);
-    if (rc < 0)
-    {
-        printk(KERN_ERR "misc_deregister failed with rc=%x\n", rc);
-    }
+    misc_deregister(&g_CtlDev);
 
     RTR0Term();
 
