diff --git a/source/blender/blenkernel/BKE_blender_version.h b/source/blender/blenkernel/BKE_blender_version.h
index 7139c00edf6..5ac0c705c52 100644
--- a/source/blender/blenkernel/BKE_blender_version.h
+++ b/source/blender/blenkernel/BKE_blender_version.h
@@ -27,10 +27,10 @@
 /* these lines are grep'd, watch out for our not-so-awesome regex
  * and keep comment above the defines.
  * Use STRINGIFY() rather than defining with quotes */
-#define BLENDER_VERSION         279
+#define BLENDER_VERSION         281
 #define BLENDER_SUBVERSION      0
 /* Several breakages with 270, e.g. constraint deg vs rad */
-#define BLENDER_MINVERSION      270
+#define BLENDER_MINVERSION      281
 #define BLENDER_MINSUBVERSION   6
 
 /* used by packaging tools */
