Index: src/bareos/core/src/lib/CMakeLists.txt
===================================================================
--- src.orig/bareos/core/src/lib/CMakeLists.txt
+++ src/bareos/core/src/lib/CMakeLists.txt
@@ -140,6 +140,9 @@ target_compile_definitions(
           -DBAREOS_SHORT_DATE="${BAREOS_SHORT_DATE}"
           -DBAREOS_PROG_DATE_TIME="${BAREOS_PROG_DATE_TIME}"
           -DBAREOS_YEAR="${BAREOS_YEAR}"
+          -DBAREOS_BINARY_INFO="${BAREOS_BINARY_INFO}"
+          -DBAREOS_SERVICES_MESSAGE="${BAREOS_SERVICES_MESSAGE}"
+          -DBAREOS_JOBLOG_MESSAGE="${BAREOS_JOBLOG_MESSAGE}"
 )
 
 target_link_libraries(
