diff --unified --recursive --text virtualbricks.orig/virtualbricks/gui/dialogs.py virtualbricks.new/virtualbricks/gui/dialogs.py
--- virtualbricks.orig/virtualbricks/gui/dialogs.py	2024-03-14 05:18:43.483389522 +0100
+++ virtualbricks.new/virtualbricks/gui/dialogs.py	2024-03-14 05:20:20.271336337 +0100
@@ -2224,12 +2224,12 @@
         vbox.foreach(vbox.remove)
         for i, (name, dest) in enumerate(iter_model(store)):
             nlabel = Gtk.Label(name + ":")
-            nlabel.props.halign = 0.0
-            nlabel.props.valign = 0.5
+            nlabel.props.halign = Gtk.Align.START
+            nlabel.props.valign = Gtk.Align.CENTER
             dlabel = Gtk.Label(dest.path)
             dlabel.set_tooltip_text(dest.path)
-            dlabel.props.halign = 0.0
-            dlabel.props.valign = 0.5
+            dlabel.props.halign = Gtk.Align.START
+            dlabel.props.valign = Gtk.Align.CENTER
             dlabel.set_ellipsize(Pango.EllipsizeMode.MIDDLE)
             box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=5)
             box.pack_start(nlabel, False, True, 0)
diff --unified --recursive --text virtualbricks.orig/virtualbricks/_settings.py virtualbricks.new/virtualbricks/_settings.py
--- virtualbricks.orig/virtualbricks/_settings.py	2024-03-14 05:18:43.483389522 +0100
+++ virtualbricks.new/virtualbricks/_settings.py	2024-03-14 05:20:40.831607944 +0100
@@ -40,7 +40,7 @@
 DEFAULT_CONF = {
     "term": "/usr/bin/xterm",
     "alt-term": "/usr/bin/gnome-terminal",
-    "sudo": "/usr/bin/gksu",
+    "sudo": "/usr/bin/pkexec",
     "kvm": False,
     "ksm": False,
     "cdroms": "",
