--- src/visit2.9.0/src/CMake/FindIceT.cmake	2015-01-20 18:51:20.000000000 +0100
+++ src/visit2.9.0/src/CMake/FindIceT.cmake.new	2015-03-21 16:36:45.938774650 +0100
@@ -43,14 +43,14 @@
 # Ice-T provides an ICETConfig.cmake file that we could include but it appears 
 # to have errors in the computed library and include paths. If those problems
 # get resolved, we can uncomment these lines and remove our own detection code.
-#IF(ICET_DIR)
-#INCLUDE(${ICET_DIR}/lib/ICETConfig.cmake)
-#ENDIF(ICET_DIR)
+IF(ICET_DIR)
+  INCLUDE(${ICET_DIR}/lib/IceTConfig.cmake)
+ENDIF(ICET_DIR)
 
 # Use the ICET_DIR hint from the config-site .cmake file 
 # Except on windows, where it is part of the repo.
 
-INCLUDE(${VISIT_SOURCE_DIR}/CMake/SetUpThirdParty.cmake)
+#INCLUDE(${VISIT_SOURCE_DIR}/CMake/SetUpThirdParty.cmake)
 
-SET_UP_THIRD_PARTY(ICET lib include icet icet_strategies icet_mpi)
+#SET_UP_THIRD_PARTY(ICET lib include icet icet_strategies icet_mpi)
 
