diff --git a/ElmerGUI/CMakeLists.txt b/ElmerGUI/CMakeLists.txt
index 8b520162..68b0a6c3 100644
--- a/ElmerGUI/CMakeLists.txt
+++ b/ElmerGUI/CMakeLists.txt
@@ -90,7 +90,7 @@ ENDIF()
 
 IF(WITH_VTK)
   # ADD_DEFINITIONS(-DEG_VTK)
-  FIND_PACKAGE(VTK REQUIRED)
+  FIND_PACKAGE(VTK REQUIRED ViewsQt FiltersParallel RenderingAnnotation RenderingFreeType IOXML IOImage FiltersParallelFlowPaths)
   INCLUDE(${VTK_USE_FILE})
   MESSAGE(STATUS "  [ElmerGUI] VTK version:       " ${VTK_VERSION})
   MESSAGE(STATUS "  [ElmerGUI] VTK_LIBS:          " "${VTK_LIBRARIES}")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b8d2abc..e552ece4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -364,6 +364,7 @@ ENDIF(WITH_Zoltan)
 MESSAGE(STATUS "------------------------------------------------")
 MESSAGE(STATUS "Mesh adaptation 2D/3D looking for [Mmg] tools ")
 
+FIND_PACKAGE(VTK REQUIRED ViewsQt)
 FIND_PACKAGE(MMG)
 # INCLUDE("cmake/Modules/FindMMG.cmake")
 # INCLUDE_DIRECTORIES(${MMG_INCLUDE_DIR})
