diff --git a/src/main.rs b/src/main.rs
index 774fb5d..0e23f42 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -300,7 +300,7 @@ fn parse_w_time(time_str: &str) -> Result<u32, CircadianError> {
 fn idle_w() -> IdleResult {
     let w_stdout = Stdio::piped();
     let mut w_output = Command::new("w")
-        .arg("-hus")
+        .arg("-husf")
         .stdout(w_stdout).spawn()?;
     let _ = w_output.wait()?;
     let w_stdout = w_output.stdout
@@ -334,7 +334,7 @@ fn idle_fn(cmd: &str, args: Vec<&str>) -> IdleResult {
         };
         let display = format!(":{}", device.chars().rev().next().unwrap_or('0'));
         let mut output = Command::new("w")
-            .arg("-hus")
+            .arg("-husf")
             .stdout(Stdio::piped()).spawn()?;
         let _ = output.wait()?;
         let w_stdout = output.stdout
