diff --git i/Release/src/http/client/http_client_asio.cpp w/Release/src/http/client/http_client_asio.cpp
index 07bb4885..f3153b6d 100644
--- i/Release/src/http/client/http_client_asio.cpp
+++ w/Release/src/http/client/http_client_asio.cpp
@@ -30,7 +30,7 @@
 #include <boost/asio/ssl.hpp>
 #include <boost/asio/ssl/error.hpp>
 #include <boost/asio/steady_timer.hpp>
-#include <boost/bind.hpp>
+#include <boost/bind/bind.hpp>
 #if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
diff --git i/Release/src/http/common/http_helpers.cpp w/Release/src/http/common/http_helpers.cpp
index 9ffbd20d..2faceb94 100644
--- i/Release/src/http/common/http_helpers.cpp
+++ w/Release/src/http/common/http_helpers.cpp
@@ -84,7 +84,7 @@ size_t chunked_encoding::add_chunked_delimiters(_Out_writes_(buffer_size) uint8_
     }
     else
     {
-        char buffer[9];
+        char buffer[17];
 #ifdef _WIN32
         sprintf_s(buffer, sizeof(buffer), "%8IX", bytes_read);
 #else
