diff -crB azure-cli-azure-cli-2.0.52/src/azure-cli-core/azure/cli/core/extension/operations.py azure-cli-azure-cli-2.0.52_fixed/src/azure-cli-core/azure/cli/core/extension/operations.py
*** azure-cli-azure-cli-2.0.52/src/azure-cli-core/azure/cli/core/extension/operations.py	2018-11-29 13:32:10.000000000 -0800
--- azure-cli-azure-cli-2.0.52_fixed/src/azure-cli-core/azure/cli/core/extension/operations.py	2019-01-01 10:40:31.594122272 -0800
***************
*** 14,20 ****
  from collections import OrderedDict
  
  import requests
! from wheel.install import WHEEL_INFO_RE
  from pkg_resources import parse_version
  
  from knack.log import get_logger
--- 14,20 ----
  from collections import OrderedDict
  
  import requests
! from wheel.wheelfile import WHEEL_INFO_RE
  from pkg_resources import parse_version
  
  from knack.log import get_logger
diff -crB azure-cli-azure-cli-2.0.52/src/azure-cli-core/azure/cli/core/extension/_resolve.py azure-cli-azure-cli-2.0.52_fixed/src/azure-cli-core/azure/cli/core/extension/_resolve.py
*** azure-cli-azure-cli-2.0.52/src/azure-cli-core/azure/cli/core/extension/_resolve.py	2018-11-29 13:32:10.000000000 -0800
--- azure-cli-azure-cli-2.0.52_fixed/src/azure-cli-core/azure/cli/core/extension/_resolve.py	2019-01-01 10:40:46.814055369 -0800
***************
*** 2,8 ****
  # Copyright (c) Microsoft Corporation. All rights reserved.
  # Licensed under the MIT License. See License.txt in the project root for license information.
  # --------------------------------------------------------------------------------------------
! from wheel.install import WHEEL_INFO_RE
  from pkg_resources import parse_version
  
  from knack.log import get_logger
--- 2,8 ----
  # Copyright (c) Microsoft Corporation. All rights reserved.
  # Licensed under the MIT License. See License.txt in the project root for license information.
  # --------------------------------------------------------------------------------------------
! from wheel.wheelfile import WHEEL_INFO_RE
  from pkg_resources import parse_version
  
  from knack.log import get_logger
