diff -ur golded3-old/gcarea.cpp golded3/gcarea.cpp
--- golded3-old/gcarea.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gcarea.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -184,7 +184,7 @@
   }
 
   Desc desc;
-  char buf[256];
+  char buf[2560];
   bool newarea = true;
   static int serial = 0;
   static int net1st = false;
@@ -870,7 +870,7 @@
 {
   char* key;
   Path file;
-  char buf[256], options[80];
+  char buf[2560], options[80];
   bool is_sqafix = false;
   bool is_dz = false;
 
diff -ur golded3-old/gccfgg2.cpp golded3/gccfgg2.cpp
--- golded3-old/gccfgg2.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gccfgg2.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -210,7 +210,7 @@
 
 void CfgAttributes() {
 
-  char buf[256];
+  char buf[2560];
   strxcpy(buf, val, 256);
   if(cfgingroup)
     CFG->grp.AddItm(GRP_ATTRIBUTES, buf, strlen(buf)+1);
diff -ur golded3-old/gccfgg6.cpp golded3/gccfgg6.cpp
--- golded3-old/gccfgg6.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gccfgg6.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -202,7 +202,7 @@
 
 //  ------------------------------------------------------------------
 
-void CfgOrigin(const char* v)  { char buf[256]; val = strxcpy(buf, v, sizeof(buf)); CfgOrigin(); }
+void CfgOrigin(const char* v)  { char buf[2560]; val = strxcpy(buf, v, sizeof(buf)); CfgOrigin(); }
 void CfgOrigin() {
 
   if(not strblank(val)) {
diff -ur golded3-old/gccfgg8.cpp golded3/gccfgg8.cpp
--- golded3-old/gccfgg8.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gccfgg8.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -433,7 +433,7 @@
   StripQuotes(val);
   if(not strblank(val)) {
     if(cfgingroup) {
-      char buf[256];
+      char buf[2560];
       strxcpy(buf, val, sizeof(buf));
       CFG->grp.AddItm(GRP_USERNAME, buf, strlen(buf)+1);
     }
diff -ur golded3-old/gclang.cpp golded3/gclang.cpp
--- golded3-old/gclang.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gclang.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -669,7 +669,7 @@
   char* ptr;
   char* str;
   int line = 0;
-  char buf[256];
+  char buf[2560];
 
   if (file==NULL || *file==0)
   {
diff -ur golded3-old/gcmisc.cpp golded3/gcmisc.cpp
--- golded3-old/gcmisc.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gcmisc.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -618,7 +618,7 @@
   {
     Esc EscTable;
     Chs ChsTable;
-    char buf[256];
+    char buf[2560];
     char* ptr;
     char* ptr2;
     int line, n, x, y, ch=0;
@@ -864,7 +864,7 @@
     if (ofp.isopen())
     {
       ofp.SetvBuf(NULL, _IOFBF, 16000);
-      char buf[256];
+      char buf[2560];
       long fpos = 0;
       long tpos = 0;
       bool was_blank = false;
diff -ur golded3-old/gearea.cpp golded3/gearea.cpp
--- golded3-old/gearea.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gearea.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -439,7 +439,7 @@
 void GPickArealist::print_line(uint idx, uint pos, bool isbar) {
 
   vchar vbuf[256];
-  char buf[256];
+  char buf[2560];
 
   if(AL[idx]->isseparator()) {
     Area* area = AL.AreaNoToPtr(idx);
@@ -527,7 +527,7 @@
   for(AL.item = AL.idx.begin(); AL.item != AL.idx.end(); AL.item++)
     nummarks += (*AL.item)->Mark.Count();
 
-  char buf[256];
+  char buf[2560];
   gsprintf(PRINTF_DECLARE_BUFFER(buf), LNG->DropMarksInfo, longdotstr(nummarks));
 
   w_info(buf);
@@ -555,7 +555,7 @@
   uint n;
   uint x;
   const char* adesc;
-  char buf[256], tmp[256];
+  char buf[2560], tmp[256];
 
   int mode, changed, currno;
 
diff -ur golded3-old/gecarb.cpp golded3/gecarb.cpp
--- golded3-old/gecarb.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gecarb.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -95,7 +95,7 @@
   Attr attr;
   bool cchide;
   bool ignorecc = false;
-  char buf[256], buf2[256];
+  char buf[2560], buf2[2560];
   Line* line;
   Line* newline;
   Line* ccline = NULL;
@@ -334,7 +334,7 @@
 
   if(CFG->crosspost == NO)
     return;
-  char buf[256];
+  char buf[2560];
   const char *ptr;
   bool ignorexc = false;
   Line* newline;
diff -ur golded3-old/gecmfd.cpp golded3/gecmfd.cpp
--- golded3-old/gecmfd.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gecmfd.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -617,7 +617,7 @@
       }
 
       if(need_netmail_kludges) {
-        char buf[256] = "";
+        char buf[2560] = "";
 
         if (not strstr(msg->txt, "\001INTL"))
           // The INTL kludge for zone crossing
diff -ur golded3-old/gectrl.cpp golded3/gectrl.cpp
--- golded3-old/gectrl.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gectrl.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -60,7 +60,7 @@
 
 char* MakeOrigin(GMsg* msg, const char* orig) {
 
-  char buf[256];
+  char buf[2560];
   char origin[100];
 
   strxcpy(origin, orig, sizeof(origin));
diff -ur golded3-old/gedoit.cpp golded3/gedoit.cpp
--- golded3-old/gedoit.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gedoit.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -142,7 +142,7 @@
   }
   else
   {
-    char buf[256];
+    char buf[2560];
     gsprintf(PRINTF_DECLARE_BUFFER(buf), LNG->CouldNotOpen, fnam);
     w_info(buf);
     waitkeyt(10000);
@@ -162,7 +162,7 @@
 
   GFTRK("WriteMsgs");
 
-  char buf[256];
+  char buf[2560];
   char fname[GMAXPATH], ofname[GMAXPATH];
 
   int overwrite = NO;
diff -ur golded3-old/gefile.cpp golded3/gefile.cpp
--- golded3-old/gefile.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/gefile.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -377,7 +377,7 @@
   gfileselect* p = new gfileselect;
   throw_new(p);
 
-  char buf[256];
+  char buf[2560];
   Path fbuf;
   uint MIN_POS=0, MAX_POS=MIN_POS+MAXROW-10;
   bool done, winop = false;
@@ -732,7 +732,7 @@
   if (AA->Msgn.Count() and msg->line and msg->lines)
   {
     gstrarray freqfile;
-    char buf[256];
+    char buf[2560];
     const char* ptr;
     const char* ptr1 = NULL;
     const char* ptr2;
diff -ur golded3-old/geinit.cpp golded3/geinit.cpp
--- golded3-old/geinit.cpp	2019-09-23 14:45:14.980000000 +0300
+++ golded3/geinit.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -175,7 +175,7 @@
   gfile fp(AddPath(CFG->areapath, CFG->semaphore.importlist), "rt", CFG->sharemode);
   if (fp.isopen())
   {
-    char buf[256];
+    char buf[2560];
     int echonums = 0;
     update_statusline(LNG->ReadingEcholist);
     while (fp.Fgets(buf, sizeof(buf)))
@@ -193,7 +193,7 @@
   // Mark the areas from the import taglist
   for (uint n = 0; n < AL.size(); n++)
   {
-    char buf[256];
+    char buf[2560];
     strcpy(buf, AL[n]->echoid());
     int x = SearchTaglist(echoin, buf);
     if(*echoin[x])
@@ -244,7 +244,7 @@
 static void ReadAddrMacros()
 {
   char* ptr;
-  char buf[256], path[GMAXPATH];
+  char buf[2560], path[GMAXPATH];
 
   ptr = getenv("FD");
   if(ptr)
@@ -806,10 +806,7 @@
   oldscreen = vsave();
   #endif
 
-  if(CFG->screensize > 0xFF) {
-    gvid->setmode(CFG->screensize >> 8);    // Set video mode
-  }
-  else if(CFG->screensize) {
+  if(CFG->screensize) {
     gvid->setrows(CFG->screensize);         // Just set rows
   }
 
diff -ur golded3-old/geline.cpp golded3/geline.cpp
--- golded3-old/geline.cpp	2019-09-23 14:45:15.060000000 +0300
+++ golded3/geline.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -3186,7 +3186,7 @@
 
 Line* AddLineF(Line*& line, const char* format, ...) {
 
-  char buf[256];
+  char buf[2560];
   va_list argptr;
   va_start(argptr, format);
   vsprintf(buf, format, argptr);
@@ -3200,7 +3200,7 @@
 
 Line* AddHexdump(Line*& line, void* data, size_t datalen) {
 
-  char buf[256];
+  char buf[2560];
   uint pos = 0;
   char* ptr = (char*)data;
 
@@ -3306,7 +3306,7 @@
 void InvalidateControlInfo(GMsg* msg) {
 
   Line* line = msg->lin;
-  char buf[256];
+  char buf[2560];
 
   while(line) {
 
diff -ur golded3-old/gemenu.cpp golded3/gemenu.cpp
--- golded3-old/gemenu.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gemenu.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -605,7 +605,7 @@
 
 int SelectFromFile(const char* file, char* selection, const char* title, const char* nolines)
 {
-  char buf[256];
+  char buf[2560];
   bool retval=false;
   int lines = 0;
 
@@ -664,7 +664,7 @@
 
   if (not CFG->tagline.empty())
   {
-    char buf[256];
+    char buf[2560];
     gstrarray Listi;
 
     gstrarray::iterator it = CFG->tagline.begin();
@@ -724,7 +724,7 @@
 
   if (not CFG->origin.empty())
   {
-    char buf[256];
+    char buf[2560];
     gstrarray Listi;
 
     gstrarray::iterator it = CFG->origin.begin();
@@ -782,7 +782,7 @@
 {
   if(not CFG->username.empty())
   {
-    char buf[256];
+    char buf[2560];
     char adrs[40];
     gstrarray Listi;
 
@@ -833,7 +833,7 @@
 {
   if (not CFG->tpl.empty())
   {
-    char buf[256];
+    char buf[2560];
     char adrs[40];
     gstrarray Listi;
 
diff -ur golded3-old/gemlst.cpp golded3/gemlst.cpp
--- golded3-old/gemlst.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gemlst.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -333,7 +333,7 @@
     mattr_ = hattr;
   }
 
-  char buf[256];
+  char buf[2560];
 
   if(AA->Msglistwidesubj()) {
     resiz += tosiz + 1;
@@ -865,7 +865,7 @@
 
 void GThreadlist::print_line(uint idx, uint pos, bool isbar)
 {
-  char buf[256];
+  char buf[2560];
   ThreadEntry &t = treeEntryList[idx];
   size_t tdlen = xlen - ((AA->Msglistdate() == MSGLISTDATE_NONE) ? 8 : 18);
 
diff -ur golded3-old/genode.cpp golded3/genode.cpp
--- golded3-old/genode.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/genode.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -1133,7 +1133,7 @@
     location = it->loc;
   else
   {
-    char buf[256];
+    char buf[2560];
     location_item item(addr);
 
     addr.make_string(buf);
@@ -1191,7 +1191,7 @@
   w_info(LNG->Wait);
 
   Addr addr;
-  char buf[256];
+  char buf[2560];
   char* ptr = buf;
 
   switch(what) {
@@ -1257,7 +1257,7 @@
     if (fp.isopen())
     {
       // Read the list index
-      char buf[256];
+      char buf[2560];
       NODE->nodelists = 0;
       while (fp.Fgets(buf, sizeof(buf)))
       {
diff -ur golded3-old/gepost.cpp golded3/gepost.cpp
--- golded3-old/gepost.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gepost.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -482,7 +482,7 @@
 static void MakeMsg2(int& mode, int& status, int& forwstat, int& topline, GMsg* msg, GMsg* oldmsg, GMsg* cmpmsg) {
 
   uint n;
-  char buf[256], buf2[256];
+  char buf[2560], buf2[2560];
   Line* line;
   Line* newline;
 
diff -ur golded3-old/geqwks.cpp golded3/geqwks.cpp
--- golded3-old/geqwks.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/geqwks.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -44,7 +44,7 @@
     {
       char* key;
       char* val;
-      char buf[256];
+      char buf[2560];
       while (fp.Fgets(buf, sizeof(buf)))
       {
         val = strtrim(buf);
@@ -75,7 +75,7 @@
   gfile fp(file, "rt");
   if (fp.isopen())
   {
-    char buf[256];
+    char buf[2560];
     int line = 0;
     int confno = 0;
     int confcnt = 0;
@@ -342,7 +342,7 @@
 
     if (imported and *QWK->ReplyLinker())
     {
-      char buf[256];
+      char buf[2560];
       gsprintf(PRINTF_DECLARE_BUFFER(buf), LNG->Replylinker, QWK->ReplyLinker());
       ShellToDos(QWK->ReplyLinker(), buf, LGREY_|_BLACK, YES);
     }
@@ -542,7 +542,7 @@
   fp.Fopen(scanfile, "rt");
   if (fp.isopen())
   {
-    char buf[256];
+    char buf[2560];
     while (fp.Fgets(buf, sizeof(buf)))
     {
       char* ptr = strchr(buf, ' ');
diff -ur golded3-old/gerand.cpp golded3/gerand.cpp
--- golded3-old/gerand.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gerand.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -277,7 +277,7 @@
   }
 
   if(found) {
-    char buf[256];
+    char buf[2560];
 
     if(CFG->grp.GetItm(GRP_AKA, &adat->aka, sizeof(gaka)))
       SetAka(adat->aka.addr);
diff -ur golded3-old/geread.cpp golded3/geread.cpp
--- golded3-old/geread.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/geread.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -141,7 +141,7 @@
 
   int istwit;
   bool istwitto, istwitsubj;
-  char buf[256], buf2[200];
+  char buf[2560], buf2[2560];
 
   GMsg* msg = (GMsg*)throw_calloc(1, sizeof(GMsg));
 
diff -ur golded3-old/gesoup.cpp golded3/gesoup.cpp
--- golded3-old/gesoup.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gesoup.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -762,7 +762,7 @@
     fp.Fopen(scanfile, "rt");
     if (fp.isopen())
     {
-      char buf[256];
+      char buf[2560];
       while (fp.Fgets(buf, sizeof(buf)))
       {
         char* ptr = strchr(buf, ' ');
diff -ur golded3-old/gesrch.cpp golded3/gesrch.cpp
--- golded3-old/gesrch.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/gesrch.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -166,7 +166,7 @@
   default_item.where.kludges   = false;
   default_item.where.signature = false;
 
-  char buf[256];
+  char buf[2560];
   char* b = buf;
   bool item_complete = false;
 
diff -ur golded3-old/geutil2.cpp golded3/geutil2.cpp
--- golded3-old/geutil2.cpp	2019-09-23 14:45:14.990000000 +0300
+++ golded3/geutil2.cpp	2019-09-23 14:45:42.450000000 +0300
@@ -159,7 +159,7 @@
 
   int n = strlen(find);
   if(strnieql(text, find, n) and (not is_tearline or (text[n] == NUL) or isspace(text[n]))) {
-    char buf[256];
+    char buf[2560];
 
     strcpy(buf, text);
     strcpy(stpcpy(text, replace), &buf[n]);
