diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 814d49f5a94..648956efffb 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -3461,19 +3461,6 @@ BOOL CDECL __wine_needs_override_large_address_aware(void)
     return needs_override;
 }

-BOOL CDECL __wine_needs_override_large_address_aware(void)
-{
-    static int needs_override = -1;
-
-    if (needs_override == -1)
-    {
-        const char *str = getenv( "WINE_LARGE_ADDRESS_AWARE" );
-
-        needs_override = !str || atoi(str) == 1;
-    }
-    return needs_override;
-}
-

 /***********************************************************************
  *           virtual_setup_exception
diff --git a/dlls/ntdll/heap.c b/dlls/ntdll/heap.c
index b1006e5f99d..4f95dd39e11 100644
--- a/dlls/ntdll/heap.c
+++ b/dlls/ntdll/heap.c
@@ -2349,7 +2349,3 @@ void HEAP_notify_thread_destroy( BOOLEAN last )
 {
     HEAP_lfh_notify_thread_destroy( last );
 }
-
-void HEAP_notify_thread_destroy( BOOLEAN last )
-{
-}
diff --git a/dlls/ntdll/string.c b/dlls/ntdll/string.c
index 156f61d3e81..ca3125da3e9 100644
--- a/dlls/ntdll/string.c
+++ b/dlls/ntdll/string.c
@@ -218,130 +218,6 @@ static FORCEINLINE void *memmove_c( char *d, const char *s, size_t n )
 }


-static FORCEINLINE void memmove_c_unaligned_32( char *d, const char *s, size_t n )
-{
-    uint64_t tmp0, tmp1, tmp2, tmpn;
-
-    if (n >= 24)
-    {
-        tmp0 = *(uint64_t *)s;
-        tmp1 = *(uint64_t *)(s + 8);
-        tmp2 = *(uint64_t *)(s + 16);
-        tmpn = *(uint64_t *)(s + n - 8);
-        *(uint64_t *)d = tmp0;
-        *(uint64_t *)(d + 8) = tmp1;
-        *(uint64_t *)(d + 16) = tmp2;
-        *(uint64_t *)(d + n - 8) = tmpn;
-    }
-    else if (n >= 16)
-    {
-        tmp0 = *(uint64_t *)s;
-        tmp1 = *(uint64_t *)(s + 8);
-        tmpn = *(uint64_t *)(s + n - 8);
-        *(uint64_t *)d = tmp0;
-        *(uint64_t *)(d + 8) = tmp1;
-        *(uint64_t *)(d + n - 8) = tmpn;
-    }
-    else if (n >= 8)
-    {
-        tmp0 = *(uint64_t *)s;
-        tmpn = *(uint64_t *)(s + n - 8);
-        *(uint64_t *)d = tmp0;
-        *(uint64_t *)(d + n - 8) = tmpn;
-    }
-    else if (n >= 4)
-    {
-        tmp0 = *(uint32_t *)s;
-        tmpn = *(uint32_t *)(s + n - 4);
-        *(uint32_t *)d = tmp0;
-        *(uint32_t *)(d + n - 4) = tmpn;
-    }
-    else if (n >= 2)
-    {
-        tmp0 = *(uint16_t *)s;
-        tmpn = *(uint16_t *)(s + n - 2);
-        *(uint16_t *)d = tmp0;
-        *(uint16_t *)(d + n - 2) = tmpn;
-    }
-    else if (n >= 1)
-    {
-        *(uint8_t *)d = *(uint8_t *)s;
-    }
-}
-
-
-static FORCEINLINE void *memmove_c( char *d, const char *s, size_t n )
-{
-    if (n <= 32) memmove_c_unaligned_32( d, s, n );
-    else if (d <= s)
-    {
-        uint64_t tmp0, tmp1, tmp2;
-        size_t k = 0;
-        while (n >= 48)
-        {
-            tmp0 = *(uint64_t *)(s +  0);
-            tmp1 = *(uint64_t *)(s +  8);
-            tmp2 = *(uint64_t *)(s + 16);
-            *(uint64_t*)(d +  0) = tmp0;
-            *(uint64_t*)(d +  8) = tmp1;
-            *(uint64_t*)(d + 16) = tmp2;
-            tmp0 = *(uint64_t *)(s + 24);
-            tmp1 = *(uint64_t *)(s + 32);
-            tmp2 = *(uint64_t *)(s + 40);
-            *(uint64_t*)(d + 24) = tmp0;
-            *(uint64_t*)(d + 32) = tmp1;
-            *(uint64_t*)(d + 40) = tmp2;
-            d += 48; s += 48; n -= 48; k += 48;
-        }
-        while (n >= 24)
-        {
-            tmp0 = *(uint64_t *)(s +  0);
-            tmp1 = *(uint64_t *)(s +  8);
-            tmp2 = *(uint64_t *)(s + 16);
-            *(uint64_t*)(d +  0) = tmp0;
-            *(uint64_t*)(d +  8) = tmp1;
-            *(uint64_t*)(d + 16) = tmp2;
-            d += 24; s += 24; n -= 24; k += 24;
-        }
-        memmove_c_unaligned_32( d, s, n );
-        return d - k;
-    }
-    else
-    {
-        uint64_t tmp0, tmp1, tmp2;
-        size_t k = n;
-        while (k >= 48)
-        {
-            tmp0 = *(uint64_t *)(s + k -  8);
-            tmp1 = *(uint64_t *)(s + k - 16);
-            tmp2 = *(uint64_t *)(s + k - 24);
-            *(uint64_t*)(d + k -  8) = tmp0;
-            *(uint64_t*)(d + k - 16) = tmp1;
-            *(uint64_t*)(d + k - 24) = tmp2;
-            tmp0 = *(uint64_t *)(s + k - 32);
-            tmp1 = *(uint64_t *)(s + k - 40);
-            tmp2 = *(uint64_t *)(s + k - 48);
-            *(uint64_t*)(d + k - 32) = tmp0;
-            *(uint64_t*)(d + k - 40) = tmp1;
-            *(uint64_t*)(d + k - 48) = tmp2;
-            k -= 48;
-        }
-        while (k >= 24)
-        {
-            tmp0 = *(uint64_t *)(s + k -  8);
-            tmp1 = *(uint64_t *)(s + k - 16);
-            tmp2 = *(uint64_t *)(s + k - 24);
-            *(uint64_t*)(d + k -  8) = tmp0;
-            *(uint64_t*)(d + k - 16) = tmp1;
-            *(uint64_t*)(d + k - 24) = tmp2;
-            k -= 24;
-        }
-        memmove_c_unaligned_32( d, s, k );
-    }
-    return d;
-}
-
-
 /*********************************************************************
  *                  memcpy   (NTDLL.@)
  *
diff --git a/dlls/winex11.drv/vulkan.c b/dlls/winex11.drv/vulkan.c
index ef7717d725b..4bf31909992 100644
--- a/dlls/winex11.drv/vulkan.c
+++ b/dlls/winex11.drv/vulkan.c
@@ -360,24 +360,6 @@ Window wine_vk_active_surface(HWND hwnd)
     return window;
 }

-Window wine_vk_active_surface( HWND hwnd )
-{
-    struct wine_vk_surface *surface, *active = NULL;
-    Window window;
-
-    EnterCriticalSection(&context_section);
-    LIST_FOR_EACH_ENTRY(surface, &surface_list, struct wine_vk_surface, entry)
-    {
-        if (surface->hwnd != hwnd) continue;
-        active = surface;
-    }
-    if (!active) window = None;
-    else window = active->window;
-    LeaveCriticalSection(&context_section);
-
-    return window;
-}
-
 void vulkan_thread_detach(void)
 {
     struct wine_vk_surface *surface, *next;
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index dfe521d62c0..f5e1e0aa2c3 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1509,22 +1509,6 @@ void update_client_window( HWND hwnd )
 }


-/**********************************************************************
- *		update_client_window
- */
-void update_client_window( HWND hwnd )
-{
-    struct x11drv_win_data *data;
-    if ((data = get_win_data( hwnd )))
-    {
-        data->client_window = wine_vk_active_surface( hwnd );
-        /* make sure any request that could use old client window has been flushed */
-        XFlush( data->display );
-        release_win_data( data );
-    }
-}
-
-
 /**********************************************************************
  *		create_dummy_client_window
  */
diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 2323bd4e61d..2891f997a51 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -626,63 +626,6 @@ static void x11drv_load_xpresent(void)
 }
 #endif /* SONAME_LIBXPRESENT */

-#ifdef SONAME_LIBXFIXES
-
-#define MAKE_FUNCPTR(f) typeof(f) * p##f;
-MAKE_FUNCPTR(XFixesQueryExtension)
-MAKE_FUNCPTR(XFixesQueryVersion)
-MAKE_FUNCPTR(XFixesCreateRegion)
-MAKE_FUNCPTR(XFixesCreateRegionFromGC)
-MAKE_FUNCPTR(XFixesSelectSelectionInput)
-#undef MAKE_FUNCPTR
-
-static void x11drv_load_xfixes(void)
-{
-    int event, error, major = 3, minor = 0;
-    void *xfixes;
-
-    if (!(xfixes = dlopen(SONAME_LIBXFIXES, RTLD_NOW)))
-    {
-        WARN("Xfixes library %s not found, disabled.\n", SONAME_LIBXFIXES);
-        return;
-    }
-
-#define LOAD_FUNCPTR(f) \
-    if (!(p##f = dlsym(xfixes, #f)))                          \
-    {                                                         \
-        WARN("Xfixes function %s not found, disabled\n", #f); \
-        dlclose(xfixes);                                      \
-        return;                                               \
-    }
-    LOAD_FUNCPTR(XFixesQueryExtension)
-    LOAD_FUNCPTR(XFixesQueryVersion)
-    LOAD_FUNCPTR(XFixesCreateRegion)
-    LOAD_FUNCPTR(XFixesCreateRegionFromGC)
-    LOAD_FUNCPTR(XFixesSelectSelectionInput)
-#undef LOAD_FUNCPTR
-
-    if (!pXFixesQueryExtension(gdi_display, &event, &error))
-    {
-        WARN("Xfixes extension not found, disabled.\n");
-        dlclose(xfixes);
-        return;
-    }
-
-    if (!pXFixesQueryVersion(gdi_display, &major, &minor) ||
-        major < 2)
-    {
-        WARN("Xfixes version 2.0 not found, disabled.\n");
-        dlclose(xfixes);
-        return;
-    }
-
-    TRACE("Xfixes, error %d, event %d, version %d.%d found\n",
-          error, event, major, minor);
-    use_xfixes = TRUE;
-    xfixes_event_base = event;
-}
-#endif /* SONAME_LIBXFIXES */
-
 static void init_visuals( Display *display, int screen )
 {
     int count;
