From 87399f9e40c8475004924732432e677356edfa36 Mon Sep 17 00:00:00 2001
From: Sebastian Krzyszkowiak <sebastian.krzyszkowiak@puri.sm>
Date: Wed, 11 Nov 2020 05:35:28 +0100
Subject: [PATCH] home test

---
 src/home.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/home.c b/src/home.c
index 417be36..44de64d 100644
--- a/src/home.c
+++ b/src/home.c
@@ -387,6 +387,8 @@ GtkWidget *
 phosh_home_new (struct zwlr_layer_shell_v1 *layer_shell,
                 struct wl_output *wl_output)
 {
+  int height;
+  phosh_shell_get_usable_area (phosh_shell_get_default (), NULL, NULL, NULL, &height);
   return g_object_new (PHOSH_TYPE_HOME,
                        "layer-shell", layer_shell,
                        "wl-output", wl_output,
@@ -396,6 +398,7 @@ phosh_home_new (struct zwlr_layer_shell_v1 *layer_shell,
                        "layer", ZWLR_LAYER_SHELL_V1_LAYER_TOP,
                        "kbd-interactivity", FALSE,
                        "exclusive-zone", PHOSH_HOME_BUTTON_HEIGHT,
+                       "height", height + PHOSH_HOME_BUTTON_HEIGHT,
                        "namespace", "phosh home",
                        NULL);
 }
-- 
2.26.2

