diff --git a/include/i3-atoms_NET_SUPPORTED.xmacro.h b/include/i3-atoms_NET_SUPPORTED.xmacro.h
index b491da98..620f9b8d 100644
--- a/include/i3-atoms_NET_SUPPORTED.xmacro.h
+++ b/include/i3-atoms_NET_SUPPORTED.xmacro.h
@@ -24,6 +24,7 @@ xmacro(_NET_WM_WINDOW_TYPE_DROPDOWN_MENU) \
 xmacro(_NET_WM_WINDOW_TYPE_POPUP_MENU) \
 xmacro(_NET_WM_WINDOW_TYPE_TOOLTIP) \
 xmacro(_NET_WM_WINDOW_TYPE_NOTIFICATION) \
+xmacro(_NET_WM_FULLSCREEN_MONITORS) \
 xmacro(_NET_WM_DESKTOP) \
 xmacro(_NET_WM_STRUT_PARTIAL) \
 xmacro(_NET_CLIENT_LIST) \
