diff -up src/wmsmixer-0.5.1/Imakefile.orig src/wmsmixer-0.5.1/Imakefile
--- src/wmsmixer-0.5.1/Imakefile.orig	2003-04-12 16:08:16.000000000 -0600
+++ src/wmsmixer-0.5.1/Imakefile	2013-09-02 18:40:52.000000000 -0600
@@ -6,8 +6,8 @@ OSSAUDIO =
 
 DEPLIBS = $(DEPXLIB) 
 LOCAL_LIBRARIES = $(XPMLIB) $(XLIB) $(OSSAUDIO)
-CC = c++ -m32
-LD = c++ -m32 -nostdlib
+CC = c++
+LD = c++ -nostdlib
 
 SRCS = wmsmixer.cc
 OBJS = wmsmixer.o
