--- wordcloud.original	2015-01-08 08:04:20.760754809 -0800
+++ wordcloud.modified	2015-01-08 08:14:58.750537478 -0800
@@ -18,7 +18,9 @@
 
 item1 = itemgetter(1)
 
-FONT_PATH = os.environ.get("FONT_PATH", "/usr/share/fonts/truetype/droid/DroidSansMono.ttf")
+# Patch to work on Arch Linux
+FONT_PATH = "/usr/share/fonts/TTF/DroidSansMono.ttf"
+# FONT_PATH = os.environ.get("FONT_PATH", "/usr/share/fonts/truetype/droid/DroidSansMono.ttf")
 STOPWORDS = set([x.strip() for x in open(os.path.join(os.path.dirname(__file__),
                                                       'stopwords')).read().split('\n')])
 
