diff --git a/hid-playstation.c b/hid-playstation.c
index 2f989da..0b7075d 100644
--- a/hid-playstation.c
+++ b/hid-playstation.c
@@ -1401,8 +1401,11 @@ static const struct hid_device_id ps_devices[] = {
 		.driver_data = 0 },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_SONY, USB_DEVICE_ID_SONY_PS5_CONTROLLER),
 		.driver_data = 0 },
+	{ 0, 0 }
 };
 
+MODULE_DEVICE_TABLE(hid, ps_devices);
+
 static struct hid_driver ps_driver = {
 	.name             = "playstation",
 	.id_table         = ps_devices,
