diff -Naur mate-menu-5.6.6.orig/lib/mate-menu-config.py mate-menu-5.6.6/lib/mate-menu-config.py
--- mate-menu-5.6.6.orig/lib/mate-menu-config.py	2015-12-17 02:33:58.891764000 +0100
+++ mate-menu-5.6.6/lib/mate-menu-config.py	2015-12-17 02:35:01.542250580 +0100
@@ -297,7 +297,6 @@
 
         self.toggleUseCustomColors(self.useCustomColors)
         self.mainWindow.present()
-        self.getBackgroundColor()
 
     def set_theme(self, widget):
         model = widget.get_model()
