diff --git src/pp-formatting.adb src/pp-formatting.adb
index c4539a77..5da22b54 100644
--- src/pp-formatting.adb
+++ src/pp-formatting.adb
@@ -60,7 +60,8 @@ package body Pp.Formatting is
 
    subtype Ada_Node is Libadalang.Analysis.Ada_Node;
    use type Ada_Node;
-   function Is_Null (Tree : Ada_Node) return Boolean is (Tree.Is_Null);
+   function Is_Null (Tree : Ada_Node) return Boolean is
+     (Libadalang.Analysis.Is_Null (Tree));
    function T_Img (Tree : Ada_Node) return String is
      (Libadalang.Analysis.Short_Image (Tree));
