diff --git a/other/wrapper/nzbhydra2wrapperPy3.py b/other/wrapper/nzbhydra2wrapperPy3.py
index e64c80d4..50678569 100644
--- a/other/wrapper/nzbhydra2wrapperPy3.py
+++ b/other/wrapper/nzbhydra2wrapperPy3.py
@@ -640,8 +640,7 @@ def main(arguments):
     global internalApiKey
     internalApiKey = args.internalApiKey
     if args.update:
-        logger.info("Executing update")
-        update()
+        logger.error("Updating is not supported.")
         sys.exit(0)
     if "--version" in unknownArgs or "--help" in unknownArgs:
         # no fancy shit, just start the file
@@ -693,7 +692,7 @@ def main(arguments):
 
             if controlCode == 11:
                 logger.info("NZBHydra main process has terminated for updating")
-                update()
+                logger.warning("Updating is not supported. Restarting...")
                 doStart = True
             elif controlCode == 22:
                 logger.info("NZBHydra main process has terminated for restart")
