diff --git a/src/TenacityApp.cpp b/src/TenacityApp.cpp
index 8136aa48b..cb0bfb629 100644
--- a/src/TenacityApp.cpp
+++ b/src/TenacityApp.cpp
@@ -862,7 +862,12 @@ void TenacityApp::OnTimer(wxTimerEvent &WXUNUSED(event)) {
 #define WL(lang, sublang)
 #endif

-#if wxCHECK_VERSION(3, 0, 1)
+#if wxCHECK_VERSION(3, 1, 6)
+wxLanguageInfo userLangs[] =
+    {
+        {wxLANGUAGE_USER_DEFINED, wxT("eu"), wxT("eu"), wxT(""), WL(0, SUBLANG_DEFAULT) wxT("Basque"), wxT("Euskara"), wxLayout_LeftToRight},
+};
+#elif wxCHECK_VERSION(3, 0, 1)
 wxLanguageInfo userLangs[] =
     {
         // Bosnian is defined in wxWidgets already