diff --git a/src/aria2p/utils.py b/src/aria2p/utils.py
index 20834d5..855ae72 100644
--- a/src/aria2p/utils.py
+++ b/src/aria2p/utils.py
@@ -13,7 +13,7 @@ from typing import Any, Dict, List
 import pkg_resources
 import toml
 from loguru import logger
-from xdg import XDG_CONFIG_HOME
+from xdg import BaseDirectory

 from aria2p.types import PathOrStr

@@ -240,7 +240,7 @@ def load_configuration() -> Dict[str, Any]:
     config_dict["DEFAULT"] = toml.loads(default_config)

     # Check for configuration file
-    config_file_path = Path(XDG_CONFIG_HOME) / "aria2p" / "config.toml"
+    config_file_path = Path(BaseDirectory.save_data_path("aria2p")) / "config.toml"

     if config_file_path.exists():
         try:
