--- log.old.cpp	2016-03-13 00:32:24.210852190 +0300
+++ log.cpp	2016-03-13 00:33:04.384183674 +0300
@@ -68,6 +68,7 @@
 	g_log_set_default_handler(log, this);
 }
 
+#ifndef g_info
 void g_info(const gchar *fmt, ...)
 {
 	va_list va;
@@ -76,3 +77,4 @@
 	g_logv(G_LOG_DOMAIN, G_LOG_LEVEL_INFO, fmt, va);
 	va_end(va);
 }
+#endif
--- log.old.hpp	2016-03-13 00:35:42.544176395 +0300
+++ log.hpp	2016-03-13 00:35:47.154176180 +0300
@@ -19,6 +19,8 @@
 			gpointer user_data);
 };
 
+#ifndef g_info
 extern void g_info(const gchar *fmt, ...);
+#endif
 
 #endif//!_LOG_HPP_
