--- src/wizardpen-orig.c        2011-05-03 03:28:23.000000000 +0200
+++ src/wizardpen.c     2018-04-27 16:07:52.257449224 +0200
@@ -573,7 +573,7 @@
 
     if (local->fd != -1)
     {
-        RemoveEnabledDevice (local->fd);
+        xf86RemoveEnabledDevice (local);
         if (priv->buffer)
         {
             XisbFree(priv->buffer);
@@ -581,7 +581,7 @@
         }
         xf86CloseSerial(local->fd);
     }
-    RemoveEnabledDevice (local->fd);
+    xf86RemoveEnabledDevice (local);
     local->fd = -1;
     return rc;
 
@@ -679,7 +679,7 @@
 
     if (local->fd != -1)
     {
-        RemoveEnabledDevice (local->fd);
+        xf86RemoveEnabledDevice (local);
         if (priv->buffer)
         {
             XisbFree(priv->buffer);
