--- xfce4-mixer-4.11.0/panel-plugin/Makefile.am~	2014-04-09 19:07:46.000000000 +0000
+++ xfce4-mixer-4.11.0/panel-plugin/Makefile.am	2018-06-10 20:54:10.332489213 +0000
@@ -22,6 +22,7 @@
 libmixer_la_CFLAGS =							\
 	$(PLATFORM_CFLAGS)						\
 	$(GLIB_CFLAGS)							\
+	$(DBUS_GLIB_CFLAGS)							\
 	$(GTK_CFLAGS)							\
 	$(LIBXFCE4UTIL_CFLAGS)						\
 	$(LIBXFCE4UI_CFLAGS)						\
@@ -42,6 +43,7 @@
 	$(PLATFORM_LDFLAGS)
 
 libmixer_la_LIBADD =							\
+	$(DBUS_GLIB_LIBS)							\
 	$(GLIB_LIBS)							\
 	$(GTK_LIBS)							\
 	$(LIBXFCE4UTIL_LIBS)						\
--- xfce4-mixer-4.11.0/xfce4-mixer/Makefile.am~	2014-04-09 19:07:46.000000000 +0000
+++ xfce4-mixer-4.11.0/xfce4-mixer/Makefile.am	2018-06-10 20:56:41.439687277 +0000
@@ -27,6 +27,7 @@
 	-DG_LOG_DOMAIN=\"xfce4-mixer\"
 
 xfce4_mixer_CFLAGS = 							\
+	$(DBUS_GLIB_CFLAGS)							\
 	$(GLIB_CFLAGS)							\
 	$(GTK_CFLAGS)							\
 	$(UNIQUE_CFLAGS)						\
@@ -41,6 +42,7 @@
 xfce4_mixer_LDFLAGS = 							\
 	$(top_builddir)/libxfce4mixer/libxfce4mixer.la			\
 	$(GLIB_LIBS)							\
+	$(DBUS_GLIB_LIBS)							\
 	$(GTHREAD_LIBS)							\
 	$(GTK_LIBS)							\
 	$(UNIQUE_LIBS)							\
