diff -Nur original/plugins/animation/animation.xml.in modified/plugins/animation/animation.xml.in
--- original/plugins/animation/animation.xml.in	2015-08-20 14:01:09.000000000 +0100
+++ modified/plugins/animation/animation.xml.in	2015-09-14 14:12:42.950728084 +0100
@@ -126,7 +126,7 @@
 	    <default>
 	      <value>(type=Normal | Dialog | ModalDialog | Unknown) &amp; !(name=gnome-screensaver)</value>
 	      <value>(type=Menu | PopupMenu | DropdownMenu | Combo)</value>
-	      <value>(type=Tooltip | Notification | Utility) &amp; !(name=compiz) &amp; !(title=notify-osd)</value>
+	      <value>(type=Tooltip | Notification | Utility) &amp; !(name=compiz) &amp; !(title=notify-osd) &amp; !(name=xfce4-notifyd)</value>
 	    </default>
 	  </option>
 	  <option name="close_options" type="list">
