diff --git a/xine_frontend_cec.c b/xine_frontend_cec.c
index 02e44f0..461bc4f 100644
--- a/xine_frontend_cec.c
+++ b/xine_frontend_cec.c
@@ -272,7 +272,7 @@ static void libcec_config_clear(libcec_configuration *p)
   p->bMonitorOnly = 0;
   p->cecVersion = CEC_DEFAULT_SETTING_CEC_VERSION;
   p->adapterType = ADAPTERTYPE_UNKNOWN;
-  p->iDoubleTapTimeoutMs = CEC_DOUBLE_TAP_TIMEOUT_MS;
+  p->iDoubleTapTimeout50Ms = CEC_DOUBLE_TAP_TIMEOUT_50_MS;
   p->comboKey = CEC_USER_CONTROL_CODE_STOP;
   p->iComboKeyTimeoutMs = CEC_DEFAULT_COMBO_TIMEOUT_MS;
 
