diff -ura package.old/svglib/svglib.py package.new/svglib/svglib.py
--- package.old/svglib/svglib.py	2018-08-05 13:06:04.914558175 +0300
+++ package.new/svglib/svglib.py	2018-08-05 13:06:38.733596856 +0300
@@ -744,7 +744,7 @@
         subpath_start = []
         lastop = ''
 
-        for i in xrange(0, len(normPath), 2):
+        for i in range(0, len(normPath), 2):
             op, nums = normPath[i:i+2]
 
             if op in ('m', 'M') and i > 0 and path.operators[-1] != _CLOSEPATH:
Только в package.new/svglib: svglib.py~
diff -ura package.old/svglib/utils.py package.new/svglib/utils.py
--- package.old/svglib/utils.py	2018-08-05 13:06:04.914558175 +0300
+++ package.new/svglib/utils.py	2018-08-05 13:06:49.629953774 +0300
@@ -196,7 +196,7 @@
 
     c1 = cos(theta1)
     s1 = sin(theta1)
-    for i in xrange(nfrag):
+    for i in range(nfrag):
         c0 = c1
         s0 = s1
         theta1 = start_rad + i * frag_rad
Только в package.new/svglib: utils.py~
