--- a/src/conky.cc
+++ b/src/conky.cc
@@ -60,6 +60,9 @@
 #ifdef BUILD_IMLIB2
 #include "imlib2.h"
 #endif /* BUILD_IMLIB2 */
+#ifdef BUILD_XSHAPE
+#include <X11/extensions/shape.h>
+#endif /* BUILD_XSHAPE */
 #endif /* BUILD_X11 */
 #ifdef BUILD_NCURSES
 #include <ncurses.h>
@@ -2046,6 +2049,25 @@
 int inotify_fd = -1;
 #endif
 
+#ifdef BUILD_XSHAPE
+static bool need_null_shape = true;
+	/* allow only decorated windows to be given mouse input */
+void set_null_shape(void){
+	int major_version, minor_version;
+	if (!XShapeQueryVersion(display, &major_version, &minor_version)) {
+		NORM_ERR("Input shapes are not supported");
+	} else {
+		if (own_window.get(*state) &&
+		    (own_window_type.get(*state) != TYPE_NORMAL ||
+		     (TEST_HINT(own_window_hints.get(*state), HINT_UNDECORATED)))) {
+			XShapeCombineRectangles(display, window.window, ShapeInput, 0, 0,
+			   NULL, 0, ShapeSet, Unsorted);
+		}
+	}
+}
+#endif /* BUILD_XSHAPE */
+
+
 static void main_loop(void)
 {
 	int terminate = 0;
@@ -2067,6 +2089,10 @@
 	sigaddset(&newmask, SIGTERM);
 	sigaddset(&newmask, SIGUSR1);
 #endif
+
+#ifdef BUILD_XSHAPE
+	set_null_shape();
+#endif /* BUILD_XSHAPE */
 
 	last_update_time = 0.0;
 	next_update_time = get_time() - fmod(get_time(), active_update_interval());
@@ -2335,7 +2361,12 @@
 									own_window_type.get(*state) == TYPE_DESKTOP) {
 								/* allow conky to hold input focus. */
 								break;
-							} else {
+							}
+#ifdef BUILD_XSHAPE
+						else if (need_null_shape)
+							set_null_shape();
+#endif /* BUILD_XSHAPE */
+						else {
 								/* forward the click to the desktop window */
 								XUngrabPointer(display, ev.xbutton.time);
 								ev.xbutton.window = window.desktop;
@@ -2356,7 +2387,12 @@
 									not TEST_HINT(own_window_hints.get(*state), HINT_UNDECORATED)) {
 								/* allow conky to hold input focus. */
 								break;
-							} else {
+							}
+#ifdef BUILD_XSHAPE
+						else if (need_null_shape)
+							set_null_shape();
+#endif /* BUILD_XSHAPE */
+						else {
 								/* forward the release to the desktop window */
 								ev.xbutton.window = window.desktop;
 								ev.xbutton.x = ev.xbutton.x_root;

