diff -urN fastparquet-0.1.5-orig/fastparquet/compression.py fastparquet-0.1.5/fastparquet/compression.py
--- fastparquet-0.1.5-orig/fastparquet/compression.py	2018-08-06 12:54:34.784680188 -0400
+++ fastparquet-0.1.5/fastparquet/compression.py	2018-08-06 13:00:39.692135158 -0400
@@ -78,7 +78,7 @@
     def zstd_compress(data, **kwargs):
         kwargs['write_content_size'] = False
         cctx = zstd.ZstdCompressor(**kwargs)
-        return cctx.compress(data, allow_empty=True)
+        return cctx.compress(data)
     def zstd_decompress(data, uncompressed_size):
         dctx = zstd.ZstdDecompressor()
         return dctx.decompress(data, max_output_size=uncompressed_size)
